* http://bugs.gentoo.org/show_bug.cgi?id=234549

--- modules/inotify-path.c
+++ modules/inotify-path.c
@@ -37,6 +37,14 @@
 #include "inotify-path.h"
 #include "inotify-missing.h"
 
+/* missing on glib-2.4 */
+#ifndef  IN_ONLYDIR
+#define  IN_ONLYDIR 0x01000000 
+#endif
+#ifndef IN_MOVE_SELF
+#define IN_MOVE_SELF 0x00000800
+#endif
+
 #define IP_INOTIFY_MASK (IN_MODIFY|IN_ATTRIB|IN_MOVED_FROM|IN_MOVED_TO|IN_DELETE|IN_CREATE|IN_DELETE_SELF|IN_UNMOUNT|IN_MOVE_SELF)
 
 typedef struct ip_watched_dir_s {
