diff -ru -x '*.Po' -x '*.Plo' gnome-vfs-2.20.1.orig/libgnomevfs/gnome-vfs-resolve.c gnome-vfs-2.20.1/libgnomevfs/gnome-vfs-resolve.c
--- gnome-vfs-2.20.1.orig/libgnomevfs/gnome-vfs-resolve.c	2008-03-28 12:55:15 +0100
+++ gnome-vfs-2.20.1/libgnomevfs/gnome-vfs-resolve.c	2008-03-28 12:59:09 +0100
@@ -174,7 +174,9 @@
 #ifdef EAI_ADDRFAMILY
 	   case EAI_ADDRFAMILY: return GNOME_VFS_ERROR_HOST_HAS_NO_ADDRESS;
 #endif
-#ifdef EAI_NODATA
+	   /* on interix, EAI_NODATA is defined to EAI_NONAME and makes
+		* the compiler bail out with "duplicate case label" */
+#if defined(EAI_NODATA) && (EAI_NODATA != EAI_NONAME)
 	   case EAI_NODATA: return GNOME_VFS_ERROR_HOST_HAS_NO_ADDRESS;
 #endif
 #ifdef EAI_SYSTEM
