diff -ru -x '*.Po' -x '*.Plo' gnome-vfs-2.20.1.orig/modules/ftp-method.c gnome-vfs-2.20.1/modules/ftp-method.c
--- gnome-vfs-2.20.1.orig/modules/ftp-method.c	2008-04-11 09:31:11 +0200
+++ gnome-vfs-2.20.1/modules/ftp-method.c	2008-04-11 09:33:26 +0200
@@ -663,6 +663,10 @@
 	return result;
 }
 
+#ifdef __INTERIX
+	typedef int socklen_t;
+#endif
+
 static GnomeVFSResult
 do_transfer_command (FtpConnection *conn, gchar *command, GnomeVFSContext *context) 
 {
diff -ru -x '*.Po' -x '*.Plo' gnome-vfs-2.20.1.orig/test/test-long-cancel.c gnome-vfs-2.20.1/test/test-long-cancel.c
--- gnome-vfs-2.20.1.orig/test/test-long-cancel.c	2008-04-11 09:31:16 +0200
+++ gnome-vfs-2.20.1/test/test-long-cancel.c	2008-04-11 09:36:11 +0200
@@ -46,6 +46,10 @@
 #include <unistd.h>
 #include <stdlib.h>
 
+#ifdef __INTERIX
+typedef int socklen_t;
+#endif
+
 #include <glib/gmain.h>
 #include <libgnomevfs/gnome-vfs-async-ops.h>
 #include <libgnomevfs/gnome-vfs-init.h>
