diff --git a/src/linux26.gmk b/src/linux26.gmk
index edceb3a..4ca8a53 100644
--- a/src/linux26.gmk
+++ b/src/linux26.gmk
@@ -59,12 +59,12 @@ # figure out if termcap is avaiable or i
 # accomplish this try to compile test.c and see if it can link termcap. For
 # Linux 2.6 then preference would appear to be "ncurses" rather than "termcap". 
 #
-test = $(shell echo "main() { printf(\"HW\n\"); }" > _t.c ; $(LD) $(LDFLAGS) -o /dev/null -lncurses _t.c 2>&1 ; rm _t.c)
-ifneq "$(strip $(test))" ""
-CONSOLE_LIBS  = -ltermcap
-else
+#test = $(shell echo "main() { printf(\"HW\n\"); }" > _t.c ; $(LD) $(LDFLAGS) -o /dev/null -lncurses _t.c 2>&1 ; rm _t.c)
+#ifneq "$(strip $(test))" ""
+#CONSOLE_LIBS  = -ltermcap
+#else
 CONSOLE_LIBS  = -lncurses
-endif
+#endif
 #
 # Rules
 #
