Index: xmlto-0.0.22/Makefile.am
===================================================================
--- xmlto-0.0.22.orig/Makefile.am
+++ xmlto-0.0.22/Makefile.am
@@ -67,11 +67,9 @@ EXTRA_DIST = xmlto.spec \
 	doc/xmlif.xml \
 	xmlto.mak
 
-$(man1_MANS): $(top_srcdir)/doc/xmlto.xml $(top_srcdir)/doc/xmlif.xml
-	for xml in xmlif.xml xmlto.xml; do \
-	  FORMAT_DIR=$(top_srcdir)/format \
-	   $(BASH) ./xmlto -o man/man1 man $(top_srcdir)/doc/$$xml ; \
-	done || ( RC=$$?; exit $$RC )
+man/man1/%.1: doc/%.xml
+	FORMAT_DIR=$(top_srcdir)/format \
+	$(BASH) ./xmlto -o $(dir $@) man $<
 
 TESTS_ENVIRONMENT = top_srcdir=$(top_srcdir) top_builddir=$(top_builddir)
 TESTS = xmlif/test/run-test
