diff -up gnupg-1.4.9/keyserver/gpgkeys_curl.c.BAD gnupg-1.4.9/keyserver/gpgkeys_curl.c
--- gnupg-1.4.9/keyserver/gpgkeys_curl.c.BAD	2008-05-24 12:41:24.000000000 -0500
+++ gnupg-1.4.9/keyserver/gpgkeys_curl.c	2008-05-24 12:41:41.000000000 -0500
@@ -286,7 +286,7 @@ main(int argc,char *argv[])
 
   if(follow_redirects)
     {
-      curl_easy_setopt(curl,CURLOPT_FOLLOWLOCATION,1);
+      curl_easy_setopt(curl,CURLOPT_FOLLOWLOCATION,1L);
       if(follow_redirects>0)
 	curl_easy_setopt(curl,CURLOPT_MAXREDIRS,follow_redirects);
     }
@@ -298,10 +298,10 @@ main(int argc,char *argv[])
     {
       fprintf(console,"gpgkeys: curl version = %s\n",curl_version());
       curl_easy_setopt(curl,CURLOPT_STDERR,console);
-      curl_easy_setopt(curl,CURLOPT_VERBOSE,1);
+      curl_easy_setopt(curl,CURLOPT_VERBOSE,1L);
     }
 
-  curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER,opt->flags.check_cert);
+  curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER,(long)opt->flags.check_cert);
   curl_easy_setopt(curl,CURLOPT_CAINFO,opt->ca_cert_file);
 
   if(proxy)
diff -up gnupg-1.4.9/keyserver/gpgkeys_hkp.c.BAD gnupg-1.4.9/keyserver/gpgkeys_hkp.c
--- gnupg-1.4.9/keyserver/gpgkeys_hkp.c.BAD	2008-05-24 12:41:47.000000000 -0500
+++ gnupg-1.4.9/keyserver/gpgkeys_hkp.c	2008-05-24 12:42:10.000000000 -0500
@@ -202,9 +202,9 @@ send_key(int *eof)
     fprintf(console,"gpgkeys: HTTP URL is `%s'\n",request);
 
   curl_easy_setopt(curl,CURLOPT_URL,request);
-  curl_easy_setopt(curl,CURLOPT_POST,1);
+  curl_easy_setopt(curl,CURLOPT_POST,1L);
   curl_easy_setopt(curl,CURLOPT_POSTFIELDS,key);
-  curl_easy_setopt(curl,CURLOPT_FAILONERROR,1);
+  curl_easy_setopt(curl,CURLOPT_FAILONERROR,1L);
 
   res=curl_easy_perform(curl);
   if(res!=0)
@@ -661,7 +661,7 @@ main(int argc,char *argv[])
     {
       fprintf(console,"gpgkeys: curl version = %s\n",curl_version());
       curl_easy_setopt(curl,CURLOPT_STDERR,console);
-      curl_easy_setopt(curl,CURLOPT_VERBOSE,1);
+      curl_easy_setopt(curl,CURLOPT_VERBOSE,1L);
     }
 
   if(proxy)
