diff -ruN dbus-1.1.4.orig/tools/dbus-launch.c dbus-1.1.4/tools/dbus-launch.c
--- dbus-1.1.4.orig/tools/dbus-launch.c	2008-02-05 08:53:15.000000000 -0600
+++ dbus-1.1.4/tools/dbus-launch.c	2008-02-05 08:54:04.000000000 -0600
@@ -966,6 +966,8 @@
       char write_pid_fd_as_string[MAX_FD_LEN];
       char write_address_fd_as_string[MAX_FD_LEN];
 
+      xdisplay = NULL;
+
       if (close_stderr)
 	do_close_stderr ();
 
