diff -ru -x '*.Po' -x '*.Plo' dbus-1.1.20.orig/dbus/dbus-sysdeps-util-unix.c dbus-1.1.20/dbus/dbus-sysdeps-util-unix.c
--- dbus-1.1.20.orig/dbus/dbus-sysdeps-util-unix.c	2008-03-19 12:08:32 +0100
+++ dbus-1.1.20/dbus/dbus-sysdeps-util-unix.c	2008-03-19 11:49:29 +0100
@@ -383,6 +383,7 @@
     }
 #endif /* HAVE_LIBAUDIT */
   
+#ifndef __INTERIX
   /* setgroups() only works if we are a privileged process,
    * so we don't return error on failure; the only possible
    * failure is that we don't have perms to do it.
@@ -393,6 +394,7 @@
   if (setgroups (0, NULL) < 0)
     _dbus_warn ("Failed to drop supplementary groups: %s\n",
                 _dbus_strerror (errno));
+#endif
   
   /* Set GID first, or the setuid may remove our permission
    * to change the GID
diff -ru -x '*.Po' -x '*.Plo' dbus-1.1.20.orig/bus/activation-helper.c dbus-1.1.20/bus/activation-helper.c
--- dbus-1.1.20.orig/bus/activation-helper.c	2008-03-28 07:39:27 +0100
+++ dbus-1.1.20/bus/activation-helper.c	2008-03-28 07:40:47 +0100
@@ -321,6 +321,7 @@
       return FALSE;
     }
 
+  #ifndef __INTERIX
   /* initialize the group access list */
   if (initgroups (user, pw->pw_gid))
     {
@@ -328,6 +329,7 @@
                       "could not initialize groups");
       return FALSE;
     }
+  #endif
 
   /* change to the primary group for the user */
   if (setgid (pw->pw_gid))
diff -ru -x '*.Po' -x '*.Plo' dbus-1.1.20.orig/tools/dbus-send.c dbus-1.1.20/tools/dbus-send.c
--- dbus-1.1.20.orig/tools/dbus-send.c	2008-03-28 07:39:32 +0100
+++ dbus-1.1.20/tools/dbus-send.c	2008-03-28 07:49:38 +0100
@@ -28,6 +28,11 @@
 
 #include "dbus-print-message.h"
 
+#if defined(__INTERIX)
+# define strtoll strtoq
+# define strtoull strtouq
+#endif
+
 static const char *appname;
 
 static void
