From c1a31b6b30dc683f219ffb8c655138b07a12a187 Mon Sep 17 00:00:00 2001
From: Benjamin Reed <rangerrick@befunk.com>
Date: Wed, 22 Jul 2009 12:47:03 +0000
Subject: missing HAVE_UNIX_FD_PASSING check

Signed-off-by: Thiago Macieira <thiago@kde.org>
---
diff --git a/dbus/dbus-message.c b/dbus/dbus-message.c
index 3373229..72d3be8 100644
--- a/dbus/dbus-message.c
+++ b/dbus/dbus-message.c
@@ -3628,9 +3628,13 @@ dbus_set_error_from_message (DBusError   *error,
 dbus_bool_t
 dbus_message_contains_unix_fds(DBusMessage *message)
 {
+#ifdef HAVE_UNIX_FD_PASSING
   _dbus_assert(message);
 
   return message->n_unix_fds > 0;
+#else
+  return FALSE;
+#endif
 }
 
 /** @} */
--
cgit v0.8.2
