--- dbus-1.4.0.orig/dbus/dbus-sysdeps-unix.c
+++ dbus-1.4.0/dbus/dbus-sysdeps-unix.c
@@ -1700,8 +1700,10 @@ _dbus_read_credentials_socket  (int
                                 DBusCredentials *credentials,
                                 DBusError       *error)
 {
+#if defined(HAVE_CMSGCRED) || defined(LOCAL_CREDS)
   struct msghdr msg;
   struct iovec iov;
+#endif
   char buf;
   dbus_uid_t uid_read;
   dbus_pid_t pid_read;
@@ -1735,6 +1737,7 @@ _dbus_read_credentials_socket  (int
 
   _dbus_credentials_clear (credentials);
 
+#if defined(HAVE_CMSGCRED) || defined(LOCAL_CREDS)
   /* Systems supporting LOCAL_CREDS are configured to have this feature
    * enabled (if it does not conflict with HAVE_CMSGCRED) prior accepting
    * the connection.  Therefore, the received message must carry the
@@ -1748,14 +1751,17 @@ _dbus_read_credentials_socket  (int
   msg.msg_iov = &iov;
   msg.msg_iovlen = 1;
 
-#if defined(HAVE_CMSGCRED) || defined(LOCAL_CREDS)
   _DBUS_ZERO(cmsg);
   msg.msg_control = (caddr_t) &cmsg;
   msg.msg_controllen = CMSG_SPACE (sizeof (struct cmsgcred));
 #endif
 
  again:
+#if defined(HAVE_CMSGCRED) || defined(LOCAL_CREDS)
   bytes_read = recvmsg (client_fd, &msg, 0);
+#else
+  bytes_read = read(client_fd, &buf, 0);
+#endif
 
   if (bytes_read < 0)
     {
