--- Makefile.in.orig	2010-03-19 13:00:24 +0100
+++ Makefile.in	2010-03-19 13:11:06 +0100
@@ -108,6 +108,9 @@
 libsrtp.so: $(srtpobj) $(cryptobj) $(gdoi)
 	$(CC) $(LDFLAGS) -shared -Wl,-soname,libsrtp.so.1 -o libsrtp.so.1.0  $^
 
+libsrtp.dylib: $(srtpobj) $(cryptobj) $(gdoi)
+	$(CC) $(LDFLAGS) -dynamiclib -install_name $(libdir)/libsrtp.1.dylib -o libsrtp.1.0.dylib  $^
+
 # libcryptomath.a contains general-purpose routines that are used to
 # generate tables and verify cryptoalgorithm implementations - this
 # library is not meant to be included in production code
@@ -195,6 +198,11 @@
 		ln -s libsrtp.so.1.0 $(DESTDIR)$(libdir)/libsrtp.so.1; \
 		ln -s libsrtp.so.1.0 $(DESTDIR)$(libdir)/libsrtp.so; \
 	fi
+	if [ -f libsrtp.1.0.dylib ]; then \
+		cp libsrtp.1.0.dylib $(DESTDIR)$(libdir)/; \
+		ln -s libsrtp.1.0.dylib $(DESTDIR)$(libdir)/libsrtp.1.dylib; \
+		ln -s libsrtp.1.0.dylib $(DESTDIR)$(libdir)/libsrtp.dylib; \
+	fi
 
 uninstall:
 	rm -rf $(DESTDIR)$(includedir)/srtp
