--- db-6.0.35/dist/aclocal/cxx.m4.orig	2017-09-21 14:38:27.091198600 +0200
+++ db-6.0.35/dist/aclocal/cxx.m4	2017-09-21 14:38:38.481785600 +0200
@@ -171,6 +171,7 @@
 
 	t44 = t4;
 	t55 = t5;
+	return 0;
 }
 
 template <typename T4>
@@ -178,6 +179,7 @@
 {
 	T4 t44 = t4;
 	std::vector<T4> abc = t55;
+	return 0;
 }
 
 // maximum of two int values 
--- db-6.0.35/dist/aclocal/tls.m4.orig	2017-09-21 15:41:28.325565600 +0200
+++ db-6.0.35/dist/aclocal/tls.m4	2017-09-21 15:41:56.388089900 +0200
@@ -21,10 +21,10 @@
               class TLSClass2 {
               public: static ] $ax_tls_decl_keyword [int tlsvar;
               };
-              template<typename T> ] $ax_tls_defn_keyword [ T* TLSClass<T>::tlsvar = NULL;]
+              template<typename T> ] $ax_tls_defn_keyword [ T* TLSClass<T>::tlsvar = 0;]
               $ax_tls_defn_keyword [int TLSClass2::tlsvar = 1; 
               static $ax_tls_decl_keyword int x = 0;],
-              [TLSClass<int>::tlsvar = NULL; TLSClass2::tlsvar = 1;],
+              [TLSClass<int>::tlsvar = 0; TLSClass2::tlsvar = 1;],
               [ac_cv_tls=modifier ; break])
       done
       test "$ac_cv_tls" = none || break
@@ -39,7 +39,7 @@
          static pthread_key_t key;
 
          static void init_once(void) {
-             pthread_key_create(&key, NULL);
+             pthread_key_create(&key, 0);
          }
          static void *get_tls() {
              return (void *)pthread_getspecific(&key);
--- db-6.0.35/dist/configure.ac.orig	2017-09-21 16:12:45.638028200 +0200
+++ db-6.0.35/dist/configure.ac	2017-09-21 16:13:11.731822700 +0200
@@ -371,11 +371,11 @@
 INSTALLER="\$(LIBTOOL) --mode=install cp -p"
 
 MAKEFILE_CC="\$(LIBTOOL) --mode=compile ${MAKEFILE_CC}"
-MAKEFILE_SOLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CCLINK} -avoid-version"
-MAKEFILE_CCLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CCLINK}"
+MAKEFILE_SOLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CCLINK} -no-undefined -avoid-version"
+MAKEFILE_CCLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CCLINK} -no-undefined"
 MAKEFILE_CXX="\$(LIBTOOL) --mode=compile ${MAKEFILE_CXX}"
-MAKEFILE_XSOLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CXXLINK} -avoid-version"
-MAKEFILE_CXXLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CXXLINK}"
+MAKEFILE_XSOLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CXXLINK} -no-undefined -avoid-version"
+MAKEFILE_CXXLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CXXLINK} -no-undefined"
 
 
 case "$host_os" in
@@ -411,9 +411,6 @@
 	o=".lo"
 fi
 INSTALL_LIBS="$DEFAULT_LIB"
-if test "$enable_static" = "yes"; then
-	INSTALL_LIBS="$INSTALL_LIBS \$(libdb)"
-fi
 
 # Optional C++ API.
 if test "$db_cv_cxx" = "yes"; then
@@ -424,9 +421,6 @@
 		DEFAULT_LIB_CXX="\$(libxso_target)"
 	fi
 	INSTALL_LIBS="$INSTALL_LIBS $DEFAULT_LIB_CXX"
-	if test "$enable_static" = "yes"; then
-		INSTALL_LIBS="$INSTALL_LIBS \$(libcxx)"
-	fi
 fi
 
 # Optional Java API / JDBC.
@@ -553,9 +547,6 @@
 		DEFAULT_LIB_SQL="\$(libsqlso_target)"
 	fi
 	INSTALL_LIBS="$INSTALL_LIBS $DEFAULT_LIB_SQL"
-	if test "$enable_static" = "yes"; then
-		INSTALL_LIBS="$INSTALL_LIBS \$(libsql)"
-	fi
 
 	if test "$db_cv_test" = "yes"; then
 		subdirs="$subdirs sql"
@@ -593,9 +584,6 @@
 	if test "$enable_shared" = "yes"; then
 		INSTALL_LIBS_EXTRA="$INSTALL_LIBS_EXTRA \$(libsqliteso)"
 	fi
-	if test "$enable_static" = "yes"; then
-		INSTALL_LIBS="$INSTALL_LIBS \$(libsqlite)"
-	fi
 fi
 
 # Optional SQL code generation tool.
@@ -619,9 +607,6 @@
 		ADDITIONAL_INCS="$ADDITIONAL_INCS \$(topdir)/lang/cxx/stl/$f"
 	done
 	INSTALL_LIBS="$INSTALL_LIBS $DEFAULT_LIB_STL"
-	if test "$enable_static" = "yes"; then
-		INSTALL_LIBS="$INSTALL_LIBS \$(libstl)"
-	fi
 fi
 
 # Checks for include files, structures, C types.
