--- setup.py
+++ setup.py
@@ -3,22 +3,24 @@
 from distutils.command.install import install
 from distutils.core import setup
 
-headers = (glob( os.path.join( "CXX","*.hxx" ) )
-          +glob( os.path.join( "CXX","*.h" ) ))
-sources = (glob( os.path.join( "Src", "*.cxx" ) )
-          +glob( os.path.join( "Src", "*.c" ) ))
+headers = (glob( os.path.join( "CXX", "Python%s" % sys.version_info[0], "*.hxx" ) )
+          +glob( os.path.join( "CXX", "Python%s" % sys.version_info[0], "*.h" ) )
+          +[os.path.join( "CXX", "Version.hxx" )]
+          +[os.path.join( "CXX", "WrapPython.h" )])
+sources = (glob( os.path.join( "Src", "Python%s" % sys.version_info[0], "*.cxx" ) )
+          +glob( os.path.join( "Src", "Python%s" % sys.version_info[0], "*.c" ) ))
 
 
 class my_install (install):
 
     def finalize_options (self):
-	if not self.install_data or (len(self.install_data) < 8) :
-	    self.install_data = "$base/share/python$py_version_short"
+        if not self.install_data or (len(self.install_data) < 8):
+            self.install_data = "$base/share/python$py_version_short"
         install.finalize_options (self)
 
     def run (self):
-	self.distribution.data_files = [("CXX", sources)]
-	self.distribution.headers = headers
+        self.distribution.data_files = [("CXX", sources)]
+        self.distribution.headers = headers
         install.run (self)
 
 
