--- groff-1.18/doc/Makefile.in.orig	Mon Jul 29 12:50:04 2002
+++ groff-1.18/doc/Makefile.in	Mon Jul 29 12:54:49 2002
@@ -104,7 +104,7 @@
 prepare_examples: grnexmpl.g
 	test -f grnexmpl.g || cp $(srcdir)/grnexmpl.g .
 
-groff: groff.texinfo
+groff.info: groff.texinfo
 	$(MAKEINFO) -I$(srcdir) $(srcdir)/groff.texinfo
 
 gnu.eps: gnu.xpm
--- groff-1.18/doc/groff.texinfo.orig	Mon Jul 29 12:48:56 2002
+++ groff-1.18/doc/groff.texinfo	Mon Jul 29 12:49:07 2002
@@ -8,7 +8,7 @@
 @c
 
 @c %**start of header (This is for running Texinfo on a region.)
-@setfilename groff
+@setfilename groff.info
 @settitle The GNU Troff Manual
 @setchapternewpage odd
 @footnotestyle separate
