--- src/utils/pfbtops/pfbtops.c.orig	2009-07-17 13:42:50.000000000 +0200
+++ src/utils/pfbtops/pfbtops.c	2009-07-17 13:44:55.000000000 +0200
@@ -19,6 +19,10 @@
 
 /* This translates ps fonts in .pfb format to ASCII ps files. */
 
+/* define early in case getopt.h gets included
+ * by system headers already (seen on IRIX). */
+#define __GETOPT_PREFIX groff_
+
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
@@ -27,7 +31,6 @@
 #include <stdlib.h>
 #include <limits.h>
 
-#define __GETOPT_PREFIX groff_
 #include <getopt.h>
 
 #include "nonposix.h"
--- src/utils/xtotroff/xtotroff.c.orig	2009-07-17 16:27:17.000000000 +0200
+++ src/utils/xtotroff/xtotroff.c	2009-07-17 16:27:39.000000000 +0200
@@ -8,6 +8,10 @@
 #include <config.h>
 #endif
 
+/* define early in case getopt.h gets included
+ * by system headers already (seen on IRIX). */
+#define __GETOPT_PREFIX groff_
+
 #include <X11/Xlib.h>
 #include <stdio.h>
 #include <ctype.h>
@@ -17,7 +21,6 @@
 #include <fcntl.h>
 #include <limits.h>
 
-#define __GETOPT_PREFIX groff_
 #include <getopt.h>
 
 #include "XFontName.h"
