--- groff-1.19.1/tmac/s.tmac	2004-04-16 06:34:21.000000000 +0200
+++ groff/tmac/s.tmac	2004-08-05 18:38:23.000000000 +0200
@@ -188,7 +188,10 @@
 .di cov*au-div!\\n[cov*n-au]
 .nf
 .ft I
-.ps \\n[PS]
+.ie (\\n[PS] >= 1000) \
+.	ps (\\n[PS]z / 1000u)
+.el \
+.	ps \\n[PS]
 ..
 .de @AI
 .if !'\*(.T'html' .par@reset
@@ -201,7 +204,10 @@
 .	di cov*ai-div!\\n[cov*n-au]
 .	nf
 .	ft R
-.	ps \\n[PS]
+.	ie (\\n[PS] >= 1000) \
+.		ps (\\n[PS]z / 1000u)
+.	el \
+.		ps \\n[PS]
 .\}
 ..
 .de LP
@@ -499,7 +505,6 @@
 .nr PN \\n%
 .nr pg*col-num 0
 .nr pg@fn-bottom-margin 0
-.nr pg*saved-po \\n[PO]
 .po \\n[PO]u
 .ev h
 .par@reset
@@ -536,12 +541,12 @@
 ..
 .de pg*end-col
 'sp |\\n[pg*col-top]u
-.po (u;\\n[pg*saved-po]+(\\n[pg@colw]+\\n[pg*gutw]*\\n[pg*col-num]))
+.po (u;\\n[PO]+(\\n[pg@colw]+\\n[pg*gutw]*\\n[pg*col-num]))
 .\"po +(u;\\n[pg@colw]+\\n[pg*gutw])
 .pg*start-col
 ..
 .de pg*end-page
-.po \\n[pg*saved-po]u
+.po \\n[PO]u
 .\" Make sure we don't exit if there are still floats or footnotes left-over.
 .ie \\n[kp@head]<\\n[kp@tail]:\\n[fn@have-overflow] \{\
 .	\" Switching environments ensures that we don't get an unnecessary
@@ -873,7 +878,6 @@
 .ds*end!\\n[\\n[.ev]:ds-type]
 .nr \\n[.ev]:ds-type 0
 ..
-.als De DE
 .de ds@auto-end
 .if \\n[\\n[.ev]:ds-type] \{\
 .	@error automatically terminating display
@@ -951,7 +955,6 @@
 .	nr \\n[.ev]:ds-type 3
 .\}
 ..
-.als Ds DS
 .de ds@need
 .if '\\n(.z'' \{\
 .	while \\n[.t]<=(\\$1)&(\\n[nl]>\\n[pg@header-bottom]) \{\
@@ -999,10 +1002,23 @@
 .\" don't set LT so that it can be defaulted from LL
 .ie rLT .lt \\n[LT]u
 .el .lt \\n[LL]u
-.ps \\n[PS]
+.ie (\\n[PS] >= 1000) \
+.	ps (\\n[PS]z / 1000u)
+.el \
+.	ps \\n[PS]
 .\" don't set VS so that it can be defaulted from PS
-.ie rVS .par*vs \\n[VS]
-.el .par*vs \\n[PS]+2
+.ie rVS \{\
+.	ie (\\n[VS] >= 1000) \
+.		par*vs "(\\n[VS]p / 1000u)"
+.	el \
+.		par*vs \\n[VS]
+.\}
+.el \{\
+.	ie (\\n[PS] >= 1000) \
+.		par*vs "((\\n[PS]p / 1000u) + 2p)"
+.	el \
+.		par*vs "(\\n[PS] + 2)"
+.\}
 .if dFAM .fam \\*[FAM]
 .if !rHY .nr HY 14
 .hy \\n[HY]
@@ -1034,16 +1050,37 @@
 .de par@init
 .if !rLT .nr LT \\n[LL]
 .if !rFL .nr FL \\n[LL]*5/6
-.if !rVS .nr VS \\n[PS]+2
+.if !rVS \{\
+.	ie (\\n[PS] >= 1000) \
+.		nr VS (\\n[PS] + 2000)
+.	el \
+.		nr VS (\\n[PS] + 2)
+.\}
 .if !rDI .nr DI .5i
-.if !rFPS .nr FPS \\n[PS]-2
-.if !rFVS .nr FVS \\n[FPS]+2
+.if !rFPS \{\
+.	ie (\\n[PS] >= 1000) \
+.		nr FPS (\\n[PS] - 2000)
+.	el \
+.		nr FPS (\\n[PS] - 2)
+.\}
+.if !rFVS \{\
+.	ie (\\n[FPS] >= 1000) \
+.		nr FVS (\\n[FPS] + 2000)
+.	el \
+.		nr FVS (\\n[FPS] + 2)
+.\}
 .\" don't change environment 0
 .ev h
-.ps \\n[PS]
+.ie (\\n[PS] >= 1000) \
+.	ps (\\n[PS]z / 1000u)
+.el \
+.	ps \\n[PS]
 .if !rQI .nr QI 5n
 .if !rPI .nr PI 5n
-.par*vs \\n[VS]
+.ie (\\n[VS] >= 1000) \
+.	par*vs "(\\n[VS]p / 1000u)"
+.el \
+.	par*vs \\n[VS]
 .if !rPD .nr PD .3v>?\n(.V
 .if !rDD .nr DD .5v>?\n(.V
 .if !rFI .nr FI 2n
@@ -1101,8 +1138,14 @@
 .in (u;\\n[\\n[.ev]:li]+\\n[\\n[.ev]:pli])
 .ft 1
 .fam \\*[FAM]
-.ps \\n[\\n[.ev]:PS]
-.par*vs \\n[\\n[.ev]:VS]
+.ie (\\n[\\n[.ev]:PS] >= 1000) \
+.	ps (\\n[\\n[.ev]:PS]z / 1000u)
+.el \
+.	ps \\n[\\n[.ev]:PS]
+.ie (\\n[\\n[.ev]:VS] >= 1000) \
+.	par*vs "(\\n[\\n[.ev]:VS]p / 1000u)"
+.el \
+.	par*vs \\n[\\n[.ev]:VS]
 .ls 1
 .TA
 .hy \\n[HY]
@@ -1357,7 +1400,10 @@
 ..
 .\" In paragraph macros.
 .de NL
-.ps \\n[\\n[.ev]:PS]
+.ie (\\n[\\n[.ev]:PS] >= 1000) \
+.	ps (\\n[\\n[.ev]:PS]z / 1000u)
+.el \
+.	ps \\n[\\n[.ev]:PS]
 ..
 .de SM
 .ps -2
@@ -1545,7 +1591,10 @@
 .1C
 .if !'\\$1'no' \{\
 .	ce 1
-.	ps \\n[PS]+2
+.	ie (\\n[PS] >= 1000) \
+.		ps ((\\n[PS]z / 1000u) + 2z)
+.	el \
+.		ps \\n[PS]+2
 .	ft B
 \\*[TOC]
 .	ft
