diff -urN diffutils-2.8.4.orig/man/Makefile.am diffutils-2.8.4/man/Makefile.am
--- diffutils-2.8.4.orig/man/Makefile.am	2002-03-10 20:53:34.000000000 -0800
+++ diffutils-2.8.4/man/Makefile.am	2003-12-09 19:54:47.000000000 -0800
@@ -17,12 +17,11 @@
 ## Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
 ## 02111-1307, USA.
 
-dist_man_MANS =	cmp.1 diff.1 diff3.1 sdiff.1
+dist_man_MANS =	cmp.1 diff3.1 sdiff.1
 MAINTAINERCLEANFILES = $(dist_man_MANS)
 
 S = $(top_srcdir)/src
 cmp.1: $S/cmp.c
-diff.1: $S/diff.c
 diff3.1: $S/diff3.c
 sdiff.1: $S/sdiff.c
 
diff -urN diffutils-2.8.4.orig/man/Makefile.in diffutils-2.8.4/man/Makefile.in
--- diffutils-2.8.4.orig/man/Makefile.in	2002-06-28 00:04:58.000000000 -0700
+++ diffutils-2.8.4/man/Makefile.in	2003-12-09 19:54:58.000000000 -0800
@@ -108,7 +108,7 @@
 am__quote = @am__quote@
 install_sh = @install_sh@
 
-dist_man_MANS = cmp.1 diff.1 diff3.1 sdiff.1
+dist_man_MANS = cmp.1 diff3.1 sdiff.1
 MAINTAINERCLEANFILES = $(dist_man_MANS)
 
 S = $(top_srcdir)/src
@@ -282,7 +282,6 @@
 	uninstall-info-am uninstall-man uninstall-man1
 
 cmp.1: $S/cmp.c
-diff.1: $S/diff.c
 diff3.1: $S/diff3.c
 sdiff.1: $S/sdiff.c
 
