diff -urN diffutils-2.8.4/src/Makefile.am diffutils-2.8.4.azarah/src/Makefile.am
--- diffutils-2.8.4/src/Makefile.am	2002-06-17 07:55:42.000000000 +0200
+++ diffutils-2.8.4.azarah/src/Makefile.am	2002-10-19 08:53:41.000000000 +0200
@@ -37,7 +37,7 @@
 
 MOSTLYCLEANFILES = paths.h
 
-cmp.$(OBJEXT) diff3.$(OBJECT) diff.$(OBJECT) sdiff.$(OBJECT): paths.h
+cmp.$(OBJEXT) diff3.$(OBJEXT) diff.$(OBJEXT) sdiff.$(OBJEXT): paths.h
 paths.h:
 	(echo '#define DEFAULT_DIFF_PROGRAM "$(bindir)/'`echo diff|sed '$(transform)'`'"' && \
 	 echo '#define LOCALEDIR "$(localedir)"') >$@
diff -urN diffutils-2.8.4/src/Makefile.in diffutils-2.8.4.azarah/src/Makefile.in
--- diffutils-2.8.4/src/Makefile.in	2002-06-28 09:04:20.000000000 +0200
+++ diffutils-2.8.4.azarah/src/Makefile.in	2002-10-19 08:54:09.000000000 +0200
@@ -410,7 +410,7 @@
 	uninstall-binPROGRAMS uninstall-info-am
 
 
-cmp.$(OBJEXT) diff3.$(OBJECT) diff.$(OBJECT) sdiff.$(OBJECT): paths.h
+cmp.$(OBJEXT) diff3.$(OBJEXT) diff.$(OBJEXT) sdiff.$(OBJEXT): paths.h
 paths.h:
 	(echo '#define DEFAULT_DIFF_PROGRAM "$(bindir)/'`echo diff|sed '$(transform)'`'"' && \
 	 echo '#define LOCALEDIR "$(localedir)"') >$@
