diff -ru -x config* -x *.Po diffutils-2.8.7.orig/lib/inttostr.h diffutils-2.8.7/lib/inttostr.h
--- diffutils-2.8.7.orig/lib/inttostr.h	Fri Jan 18 14:33:37 2008
+++ diffutils-2.8.7/lib/inttostr.h	Fri Jan 18 14:50:09 2008
@@ -22,7 +22,7 @@
 # include <config.h>
 #endif
 
-#if HAVE_INTTYPES_H
+#if HAVE_INTTYPES_H_WITH_UINTMAX
 # include <inttypes.h>
 #endif
 #if HAVE_STDINT_H
diff -ru -x config* -x *.Po diffutils-2.8.7.orig/lib/strtoimax.c diffutils-2.8.7/lib/strtoimax.c
--- diffutils-2.8.7.orig/lib/strtoimax.c	Fri Jan 18 14:33:37 2008
+++ diffutils-2.8.7/lib/strtoimax.c	Fri Jan 18 14:44:42 2008
@@ -21,9 +21,9 @@
 # include <config.h>
 #endif
 
-#if HAVE_INTTYPES_H
+#if HAVE_INTTYPES_H_WITH_UINTMAX
 # include <inttypes.h>
-#elif HAVE_STDINT_H
+#elif HAVE_STDINT_H_WITH_UINTMAX
 # include <stdint.h>
 #endif
 
diff -ru -x config* -x *.Po diffutils-2.8.7.orig/lib/xstrtol.h diffutils-2.8.7/lib/xstrtol.h
--- diffutils-2.8.7.orig/lib/xstrtol.h	Fri Jan 18 14:33:37 2008
+++ diffutils-2.8.7/lib/xstrtol.h	Fri Jan 18 14:41:29 2008
@@ -23,10 +23,10 @@
 # include "exitfail.h"
 
 /* Get uintmax_t.  */
-# if HAVE_INTTYPES_H
+# if HAVE_INTTYPES_H_WITH_UINTMAX
 #  include <inttypes.h>
 # else
-#  if HAVE_STDINT_H
+#  if HAVE_STDINT_H_WITH_UINTMAX
 #   include <stdint.h>
 #  endif
 # endif
diff -ru -x config* -x *.Po diffutils-2.8.7.orig/lib/xstrtoumax.c diffutils-2.8.7/lib/xstrtoumax.c
--- diffutils-2.8.7.orig/lib/xstrtoumax.c	Fri Jan 18 14:33:37 2008
+++ diffutils-2.8.7/lib/xstrtoumax.c	Fri Jan 18 14:42:19 2008
@@ -21,9 +21,9 @@
 # include <config.h>
 #endif
 
-#if HAVE_INTTYPES_H
+#if HAVE_INTTYPES_H_WITH_UINTMAX
 # include <inttypes.h>
-#elif HAVE_STDINT_H
+#elif HAVE_STDINT_H_WITH_UINTMAX
 # include <stdint.h>
 #endif
 
diff -ru -x config* -x *.Po diffutils-2.8.7.orig/src/system.h diffutils-2.8.7/src/system.h
--- diffutils-2.8.7.orig/src/system.h	Fri Jan 18 14:33:37 2008
+++ diffutils-2.8.7/src/system.h	Fri Jan 18 14:55:06 2008
@@ -22,6 +22,10 @@
 
 #include <config.h>
 
+#if HAVE_STDINT_H_WITH_UINTMAX
+#  include <stdint.h>
+#endif
+
 /* Don't bother to support K&R C compilers any more; it's not worth
    the trouble.  These macros prevent some library modules from being
    compiled in K&R C mode.  */
@@ -160,8 +164,9 @@
 
 #include <limits.h>
 
-#if HAVE_INTTYPES_H
+#if HAVE_INTTYPES_H_WITH_UINTMAX
 # include <inttypes.h>
+#elif HAVE_INTTYPES_H_WITH_UINTMAX
 #endif
 #ifndef PTRDIFF_MAX
 # define PTRDIFF_MAX TYPE_MAXIMUM (ptrdiff_t)
