diff -ur xscreensaver-5.05.orig/driver/XScreenSaver.ad.in xscreensaver-5.05/driver/XScreenSaver.ad.in
--- xscreensaver-5.05.orig/driver/XScreenSaver.ad.in	2008-03-01 23:33:47.000000000 +0200
+++ xscreensaver-5.05/driver/XScreenSaver.ad.in	2008-05-07 20:47:16.000000000 +0300
@@ -31,36 +31,36 @@
 *mode:			random
 *timeout:		0:10:00
 *cycle:			0:10:00
-*lockTimeout:		0:00:00
+*lockTimeout:		0:20:00
 *passwdTimeout:		0:00:30
-*dpmsEnabled:		False
-*dpmsStandby:		2:00:00
-*dpmsSuspend:		2:00:00
-*dpmsOff:		4:00:00
-*grabDesktopImages:	True
+*dpmsEnabled:		True
+*dpmsStandby:		1:00:00
+*dpmsSuspend:		1:00:00
+*dpmsOff:		2:00:00
+*grabDesktopImages:	False
 *grabVideoFrames:	False
 *chooseRandomImages:	@DEFAULT_IMAGES_P@
 *imageDirectory:	@DEFAULT_IMAGE_DIRECTORY@
 *nice:			10
 *memoryLimit:		0
-*lock:			False
+*lock:			True
 *verbose:		False
 *timestamp:		True
 *fade:			True
 *unfade:		False
 *fadeSeconds:		0:00:03
 *fadeTicks:		20
-*splash:		True
+*splash:		False
 *splashDuration:	0:00:05
 *visualID:		default
 *captureStderr: 	True
 *ignoreUninstalledPrograms: False
 
-*textMode:		file
-*textLiteral:		XScreenSaver
-*textFile:		@DEFAULT_TEXT_FILE@
-*textProgram:		fortune
-*textURL:		http://www.livejournal.com/stats/latest-rss.bml
+*textMode:		literal
+*textLiteral:		Gentoo Linux
+*textFile:		/etc/gentoo-release
+*textProgram:		/usr/bin/fortune
+*textURL:		http://planet.gentoo.org/rss20.xml
 
 *overlayTextForeground:	#FFFF00
 *overlayTextBackground:	#000000
@@ -117,7 +117,10 @@
 
 ! The format used for printing the date and time in the password dialog box
 ! (see the strftime(3) manual page for details.)
-*dateFormat:		%d-%b-%y (%a); %I:%M %p
+! Preferred time representation for the current locale
+*dateFormat:    %x %X
+! Xscreensaver's default
+! *dateFormat:		%d-%b-%y (%a); %I:%M %p
 ! To show the time only:
 ! *dateFormat:		%I:%M %p
 ! For 24 hour time:
