From cc21d43a5991764a2dc2bfc5b22a19cec2f62180 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Sun, 13 Apr 2008 23:57:55 -0400
Subject: [PATCH] git ignores

---
 .gitignore |   22 ++++++++++++++++++++++
 1 files changed, 22 insertions(+), 0 deletions(-)
 create mode 100644 .gitignore

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..f011661
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,22 @@
+*.o
+
+/config.h
+/config.make
+/config.status
+
+/lib/libnet-tools.a
+
+/version.h
+
+/arp
+/hostname
+/ifconfig
+/ipmaddr
+/iptunnel
+/mii-tool
+/nameif
+/netstat
+/plipconfig
+/rarp
+/route
+/slattach
-- 
1.5.5

