From 396d2d99db901296449663782a7ea13b1d0d6b51 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 14 Apr 2008 00:17:39 -0400
Subject: [PATCH] libdir needs version.h

---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index a2f0c77..382cbf2 100644
--- a/Makefile
+++ b/Makefile
@@ -172,7 +172,7 @@ $(NET_LIB):	config.h version.h intl.h libdir
 
 i18n.h:		i18ndir
 
-libdir:
+libdir:		version.h
 		@$(MAKE) -C $(NET_LIB_PATH)
 
 i18ndir:
-- 
1.5.5

