From ea596e184e45d3cb08fbb0f4269f9bcbf3256ed3 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 14 Apr 2008 00:32:40 -0400
Subject: [PATCH] document the -e option

---
 arp.c           |    1 +
 man/en_US/arp.8 |    7 +++++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arp.c b/arp.c
index 4ee5c58..26ab326 100644
--- a/arp.c
+++ b/arp.c
@@ -630,6 +630,7 @@ static void usage(void)
     fprintf(stderr, _("  arp [-v]   [<HW>] [-i <if>] -Ds <host> <if> [netmask <nm>] pub          <-''-\n\n"));
     
     fprintf(stderr, _("        -a                       display (all) hosts in alternative (BSD) style\n"));
+    fprintf(stderr, _("        -e                       display (all) hosts in default (Linux) style\n"));
     fprintf(stderr, _("        -s, --set                set a new ARP entry\n"));
     fprintf(stderr, _("        -d, --delete             delete a specified entry\n"));
     fprintf(stderr, _("        -v, --verbose            be verbose\n"));
diff --git a/man/en_US/arp.8 b/man/en_US/arp.8
index 4d20ed3..281047d 100644
--- a/man/en_US/arp.8
+++ b/man/en_US/arp.8
@@ -8,7 +8,7 @@ arp \- manipulate the system ARP cache
 .IR type ] 
 .RB [ \-i
 .IR if ] 
-.RB [ \-a ] 
+.RB [ \-ae ]
 .RI [ hostname ]
 .PP
 .B arp 
@@ -115,7 +115,7 @@ Tell the user what is going on by being verbose.
 shows numerical addresses instead of trying to determine symbolic host, port
 or user names.
 .TP
-.B "\-H type, \-\-hw-type type"
+.B "\-H type, \-\-hw-type type, \-t type"
 When setting or reading the ARP cache, this optional parameter tells
 .B arp
 which class of entries it should check for.  The default value of
@@ -134,6 +134,9 @@ and
 .B \-a
 Use alternate BSD style output format (with no fixed columns).
 .TP
+.B \-e
+Use default Linux style output format (with fixed columns).
+.TP
 .B "\-D, \-\-use-device"
 Instead of a hw_addr, the given argument is the name of an interface. 
 .B arp
-- 
1.5.5

