From bef310c8274c7df6e004ae0e927e0d757a85c586 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 14 Apr 2008 00:39:09 -0400
Subject: [PATCH] prevent overflows in interface buffers

---
 netstat.c |   18 ++++++++++++------
 1 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/netstat.c b/netstat.c
index dcd1f8a..8057d78 100644
--- a/netstat.c
+++ b/netstat.c
@@ -784,7 +784,8 @@ static void tcp_do_one(int lnr, const char *line)
 	    local_addr[22 - strlen(buffer)] = '\0';
 
 	strcat(local_addr, ":");
-	strcat(local_addr, buffer);
+	strncat(local_addr, buffer, sizeof(local_addr));
+	local_addr[sizeof(local_addr)-1] = 0;
 	snprintf(buffer, sizeof(buffer), "%s",
 		 get_sname(htons(rem_port), "tcp", flag_not & FLAG_NUM_PORT));
 
@@ -792,7 +793,8 @@ static void tcp_do_one(int lnr, const char *line)
 	    rem_addr[22 - strlen(buffer)] = '\0';
 
 	strcat(rem_addr, ":");
-	strcat(rem_addr, buffer);
+	strncat(rem_addr, buffer, sizeof(rem_addr));
+	rem_addr[sizeof(rem_addr)-1] = 0;
 	timers[0] = '\0';
 
 	if (flag_opt)
@@ -936,7 +938,8 @@ static void udp_do_one(int lnr, const char *line)
 	if ((strlen(local_addr) + strlen(buffer)) > 22)
 	    local_addr[22 - strlen(buffer)] = '\0';
 	strcat(local_addr, ":");
-	strcat(local_addr, buffer);
+	strncat(local_addr, buffer, sizeof(local_addr));
+	local_addr[sizeof(local_addr)-1] = 0;
 
 	snprintf(buffer, sizeof(buffer), "%s",
 		 get_sname(htons(rem_port), "udp", flag_not & FLAG_NUM_PORT));
@@ -945,7 +948,8 @@ static void udp_do_one(int lnr, const char *line)
 	if ((strlen(rem_addr) + strlen(buffer)) > 22)
 	    rem_addr[22 - strlen(buffer)] = '\0';
 	strcat(rem_addr, ":");
-	strcat(rem_addr, buffer);
+	strncat(rem_addr, buffer, sizeof(rem_addr));
+	rem_addr[sizeof(rem_addr)-1] = 0;
 
 	timers[0] = '\0';
 	if (flag_opt)
@@ -1058,7 +1062,8 @@ static void raw_do_one(int lnr, const char *line)
 	if ((strlen(local_addr) + strlen(buffer)) > 22)
 	    local_addr[22 - strlen(buffer)] = '\0';
 	strcat(local_addr, ":");
-	strcat(local_addr, buffer);
+	strncat(local_addr, buffer, sizeof(local_addr));
+	local_addr[sizeof(local_addr)-1] = 0;
 
 	snprintf(buffer, sizeof(buffer), "%s",
 		 get_sname(htons(rem_port), "raw", flag_not & FLAG_NUM_PORT));
@@ -1067,7 +1072,8 @@ static void raw_do_one(int lnr, const char *line)
 	if ((strlen(rem_addr) + strlen(buffer)) > 22)
 	    rem_addr[22 - strlen(buffer)] = '\0';
 	strcat(rem_addr, ":");
-	strcat(rem_addr, buffer);
+	strncat(rem_addr, buffer, sizeof(rem_addr));
+	rem_addr[sizeof(rem_addr)-1] = 0;
 
 	timers[0] = '\0';
 	if (flag_opt)
-- 
1.5.5

