Index: raptor-1.4.16/src/Makefile.in
===================================================================
--- raptor-1.4.16.orig/src/Makefile.in
+++ raptor-1.4.16/src/Makefile.in
@@ -233,7 +233,7 @@ LEX = @LEX@
 LEXLIB = @LEXLIB@
 LEX_OUTPUT_ROOT = @LEX_OUTPUT_ROOT@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@ $(MEM_LIBS)
+LIBS = @LIBS@ $(MEM_LIBS) -lm
 LIBTOOL = @LIBTOOL@
 LN_S = @LN_S@
 LTLIBOBJS = @LTLIBOBJS@
