Make it look into the Prefix.

--- esearch-0.7.1/esearch.py
+++ esearch-0.7.1/esearch.py
@@ -9,9 +9,11 @@
 
 from getopt import *
 import sys
+import os
+from portage.const import EPREFIX
 
-sys.path.insert(0, "/usr/lib/portage/pym")
-sys.path.insert(0, "/usr/lib/esearch")
+sys.path.insert(0, EPREFIX + "/usr/lib/portage/pym")
+sys.path.insert(0, EPREFIX + "/usr/lib/esearch")
 
 try:
     from portage.output import bold, red, green, darkgreen, turquoise, nocolor
@@ -22,7 +24,7 @@
 
 from common import needdbversion
 
-esearchdbdir =  "/var/cache/edb/"
+esearchdbdir =  EPREFIX + "/var/cache/edb/"
 
 def usage():
     print "esearch (0.7.1) - Replacement for 'emerge search' with search-index"
--- esearch-0.7.1/eupdatedb.py
+++ esearch-0.7.1/eupdatedb.py
@@ -16,9 +16,10 @@
 from os.path import exists
 from shutil import copyfile
 from getopt import *
+from portage.const import EPREFIX
 
-sys.path.insert(0, "/usr/lib/portage/pym")
-sys.path.insert(0, "/usr/lib/esearch")
+sys.path.insert(0, EPREFIX + "/usr/lib/portage/pym")
+sys.path.insert(0, EPREFIX + "/usr/lib/esearch")
 
 import portage
 try:
@@ -32,8 +33,8 @@
 
 from common import needdbversion, version
 
-esearchdbdir =  "/var/cache/edb/"
-tmpfile =       "/tmp/esearchdb.py.tmp"
+esearchdbdir =  EPREFIX + "/var/cache/edb/"
+tmpfile =       EPREFIX + "/tmp/esearchdb.py.tmp"
 
 vartree = portage.vartree()
 
