Index: subversion-1.5.0-rc9/configure.ac
===================================================================
--- subversion-1.5.0-rc9.orig/configure.ac
+++ subversion-1.5.0-rc9/configure.ac
@@ -84,9 +84,19 @@ AC_ARG_ENABLE(neon-version-check,
   [])
 SVN_LIB_NEON($NEON_ALLOWED_LIST, $NEON_RECOMMENDED_VER, $NEON_URL)
 
+if test "$svn_lib_neon" = "yes"; then
+  AC_DEFINE([SVN_HAVE_NEON], 1,
+            [Defined if support for Neon is enabled])
+fi
+
 dnl Search for serf as an alternative to neon
 SVN_LIB_SERF
 
+if test "$svn_lib_serf" = "yes"; then
+  AC_DEFINE([SVN_HAVE_SERF], 1,
+            [Defined if support for Serf is enabled])
+fi
+
 dnl Set up a number of directories ---------------------
 
 dnl Create SVN_BINDIR for proper substitution
Index: subversion-1.5.0-rc9/subversion/libsvn_ra/ra_loader.c
===================================================================
--- subversion-1.5.0-rc9.orig/subversion/libsvn_ra/ra_loader.c
+++ subversion-1.5.0-rc9/subversion/libsvn_ra/ra_loader.c
@@ -59,7 +59,7 @@
    ### any code which uses the pre-1.2 API svn_ra_get_ra_library
    ### instead of svn_ra_open. */
 
-#if defined(SVN_LIBSVN_CLIENT_LINKS_RA_NEON) && defined (SVN_LIBSVN_CLIENT_LINKS_RA_SERF)
+#if defined(SVN_HAVE_NEON) && defined(SVN_HAVE_SERF)
 #define MUST_CHOOSE_DAV
 #endif
 
