--- daemon-1.0.1/src/native/unix/support/apsupport.m4.orig	2006-05-07 15:28:22.000000000 +0200
+++ daemon-1.0.1/src/native/unix/support/apsupport.m4	2006-05-07 15:34:42.000000000 +0200
@@ -35,7 +35,7 @@
   linux*)
     CFLAGS="$CFLAGS -DOS_LINUX -DDSO_DLFCN"
     supported_os="linux"
-    LDFLAGS="$LDFLAGS -ldl -lpthread"
+    LIBS="$LIBS -ldl -lpthread"
     ;;
   cygwin)
     CFLAGS="$CFLAGS -DOS_CYGWIN -DDSO_DLFCN -DNO_SETSID"
@@ -94,4 +94,5 @@
   AC_MSG_RESULT([ok])
   AC_SUBST(CFLAGS)
   AC_SUBST(LDFLAGS)
+  AC_SUBST(LIBS)
 ])
--- daemon-1.0.1/src/native/unix/Makedefs.in.orig	2006-05-07 15:50:07.000000000 +0200
+++ daemon-1.0.1/src/native/unix/Makedefs.in	2004-02-27 09:40:46.000000000 +0100
@@ -22,6 +22,7 @@
 JAR = @JAR@
 CFLAGS = @CFLAGS@
 LDFLAGS = @LDFLAGS@
+LIBS = @LIBS@
 JAVACFLAGS = @JAVACFLAGS@
 RANLIB = @RANLIB@
 LDCMD = @LDCMD@
--- daemon-1.0.1/src/native/unix/native/Makefile.in.orig	2006-05-07 15:36:54.000000000 +0200
+++ daemon-1.0.1/src/native/unix/native/Makefile.in	2006-05-07 15:37:44.000000000 +0200
@@ -36,7 +36,7 @@
 	$(RANLIB) libservice.a
 
 jsvc: jsvc-unix.o libservice.a
-	$(LDCMD) $(LDFLAGS) jsvc-unix.o libservice.a -o ../jsvc
+	$(LDCMD) $(LDFLAGS) jsvc-unix.o libservice.a $(LIBS) -o ../jsvc
 
 clean:
 	rm -f $(OBJS) ../jsvc jsvc-unix.o libservice.a
