--- postgresql-8.2.3.orig/src/bin/Makefile	2006-03-05 16:58:50.000000000 +0100
+++ postgresql-8.2.3/src/bin/Makefile	2007-03-29 12:12:26.000000000 +0200
@@ -14,7 +14,7 @@
 include $(top_builddir)/src/Makefile.global
 
 DIRS := initdb ipcclean pg_ctl pg_dump \
-	psql scripts pg_config pg_controldata pg_resetxlog
+	psql scripts pg_controldata pg_resetxlog
 ifeq ($(PORTNAME), win32)
 DIRS+=pgevent
 endif
--- postgresql-8.2.3.orig/src/interfaces/Makefile	2004-04-20 02:33:51.000000000 +0200
+++ postgresql-8.2.3/src/interfaces/Makefile	2007-03-29 12:14:01.000000000 +0200
@@ -12,7 +12,7 @@
 top_builddir = ../..
 include $(top_builddir)/src/Makefile.global
 
-DIRS := libpq ecpg 
+DIRS := ecpg 
 
 ALLDIRS := $(DIRS) 

--- postgresql-8.2.3.orig/src/Makefile	2006-06-23 01:50:35.000000000 +0200
+++ postgresql-8.2.3/src/Makefile	2007-03-29 12:54:59.000000000 +0200
@@ -18,12 +18,10 @@
 	$(MAKE) -C timezone $@
 	$(MAKE) -C backend $@
 	$(MAKE) -C backend/utils/mb/conversion_procs $@
-	$(MAKE) -C include $@
 	$(MAKE) -C interfaces $@
 	$(MAKE) -C bin $@
 	$(MAKE) -C pl $@
 	$(MAKE) -C makefiles $@
-	$(MAKE) -C test/regress $@
 
 install: install-local
 
@@ -47,7 +45,6 @@
 	$(MAKE) -C port $@
 	$(MAKE) -C timezone $@
 	$(MAKE) -C backend $@
-	$(MAKE) -C include $@
 	$(MAKE) -C interfaces $@
 	$(MAKE) -C bin $@
 	$(MAKE) -C pl $@
@@ -60,12 +57,10 @@
 	-$(MAKE) -C port $@
 	-$(MAKE) -C timezone $@
 	-$(MAKE) -C backend $@
-	-$(MAKE) -C include $@
 	-$(MAKE) -C interfaces $@
 	-$(MAKE) -C bin $@
 	-$(MAKE) -C pl $@
 	-$(MAKE) -C makefiles $@
-	-$(MAKE) -C test $@
 	-$(MAKE) -C tutorial NO_PGXS=1 $@
 	-$(MAKE) -C test/thread $@
 	rm -f Makefile.port Makefile.global
