diff -ur a/config/python.m4 b/config/python.m4
--- a/config/python.m4	2006-10-16 18:24:54 +0100
+++ b/config/python.m4	2007-08-17 15:37:00 +0100
@@ -78,18 +78,4 @@
 AC_SUBST(python_libspec)[]dnl
 AC_SUBST(python_additional_libs)[]dnl
 
-# threaded python is not supported on bsd's
-AC_MSG_CHECKING(whether Python is compiled with thread support)
-pythreads=`${PYTHON} -c "import sys; print int('thread' in sys.builtin_module_names)"`
-if test "$pythreads" = "1"; then
-  AC_MSG_RESULT(yes)
-  case $host_os in
-  openbsd*|freebsd*)
-    AC_MSG_ERROR([threaded Python not supported on this platform])
-    ;;
-  esac
-else
-  AC_MSG_RESULT(no)
-fi
-
 ])# PGAC_CHECK_PYTHON_EMBED_SETUP
diff -ur a/configure b/configure
--- a/configure	2007-04-20 04:20:41 +0100
+++ b/configure	2007-08-17 15:37:20 +0100
@@ -5002,24 +5002,6 @@
 echo "${ECHO_T}${python_libspec} ${python_additional_libs}" >&6
 
 
-# threaded python is not supported on bsd's
-echo "$as_me:$LINENO: checking whether Python is compiled with thread support" >&5
-echo $ECHO_N "checking whether Python is compiled with thread support... $ECHO_C" >&6
-pythreads=`${PYTHON} -c "import sys; print int('thread' in sys.builtin_module_names)"`
-if test "$pythreads" = "1"; then
-  echo "$as_me:$LINENO: result: yes" >&5
-echo "${ECHO_T}yes" >&6
-  case $host_os in
-  openbsd*|freebsd*)
-    { { echo "$as_me:$LINENO: error: threaded Python not supported on this platform" >&5
-echo "$as_me: error: threaded Python not supported on this platform" >&2;}
-   { (exit 1); exit 1; }; }
-    ;;
-  esac
-else
-  echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6
-fi
 
 
 fi
