--- plugin/auth-pam/auth-pam.c
+++ plugin/auth-pam/auth-pam.c
@@ -31,7 +31,11 @@
 #include <dlfcn.h>
 #include "pamdl.h"
 #else
+#ifndef __APPLE__
 #include <security/pam_appl.h>
+#else
+#include <pam/pam_appl.h>
+#endif
 #endif
 
 #include <stdio.h>
--- plugin/auth-pam/pamdl.c
+++ plugin/auth-pam/pamdl.c
@@ -9,7 +9,11 @@
 
 #include <stdio.h>
 #include <dlfcn.h>
+#ifndef __APPLE__
 #include <security/pam_appl.h>
+#else
+#include <pam/pam_appl.h>
+#endif
 
 #include "pamdl.h"
 
--- plugin/auth-pam/pamdl.h
+++ plugin/auth-pam/pamdl.h
@@ -1,5 +1,10 @@
 #if DLOPEN_PAM
+#ifndef __APPLE__
 #include <security/pam_appl.h>
+#else
+#include <pam/pam_appl.h>
+#endif
+
 
 /* Dynamically load and unload the PAM library */
 int dlopen_pam (const char *so);
--- plugin/down-root/Makefile
+++ plugin/down-root/Makefile
@@ -7,8 +7,14 @@
 
 CC_FLAGS=-O2 -Wall
 
+TARGET_DIR=
+
 down-root.so : down-root.o
+ifeq ($(shell uname -s), Darwin)
+	gcc ${CC_FLAGS} -fPIC -dynamiclib -install_name $(TARGET_DIR)/openvpn-down-root.so -o openvpn-down-root.so down-root.o -lc
+else
 	gcc ${CC_FLAGS} -fPIC -shared -Wl,-soname,openvpn-down-root.so -o openvpn-down-root.so down-root.o -lc
+endif
 
 down-root.o : down-root.c
 	gcc ${CC_FLAGS} -fPIC -c ${INCLUDE} down-root.c
--- plugin/auth-pam/Makefile
+++ plugin/auth-pam/Makefile
@@ -17,8 +17,14 @@
 
 CC_FLAGS=-O2 -Wall -DDLOPEN_PAM=$(DLOPEN_PAM)
 
+TARGET_DIR=
+
 openvpn-auth-pam.so : auth-pam.o pamdl.o
+ifeq ($(shell uname -s), Darwin)
+	gcc ${CC_FLAGS} -fPIC -dynamiclib -install_name $(TARGET_DIR)/openvpn-auth-pam.so -o openvpn-auth-pam.so auth-pam.o pamdl.o -lc $(LIBPAM)
+else
 	gcc ${CC_FLAGS} -fPIC -shared -Wl,-soname,openvpn-auth-pam.so -o openvpn-auth-pam.so auth-pam.o pamdl.o -lc $(LIBPAM)
+endif
 
 auth-pam.o : auth-pam.c pamdl.h
 	gcc ${CC_FLAGS} -fPIC -c ${INCLUDE} auth-pam.c
