diff -ru openvpn-2.1.0~/basic.h openvpn-2.1.0/basic.h
--- openvpn-2.1.0~/basic.h	2009-12-19 20:31:43.000000000 +0100
+++ openvpn-2.1.0/basic.h	2009-12-19 20:31:56.000000000 +0100
@@ -22,22 +22,11 @@
  *  59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <stdbool.h>
+
 #ifndef BASIC_H
 #define BASIC_H
 
-/* bool definitions */
-#ifndef bool
-#define bool int
-#endif
-
-#ifndef true
-#define true 1
-#endif
-
-#ifndef false
-#define false 0
-#endif
-
 #define BOOL_CAST(x) ((x) ? (true) : (false))
 
 /* size of an array */
