--- openssl-1.0.2l/Configure	2017-10-02 13:14:38.701296300 +0200
+++ openssl-1.0.2l-.default/Configure	2017-10-02 13:19:34.325506800 +0200
@@ -602,6 +602,7 @@
 
 # MinGW
 "mingw", "gcc:-mno-cygwin -DL_ENDIAN -DWIN32_LEAN_AND_MEAN -fomit-frame-pointer -O3 -march=i486 -Wall::-D_MT:MINGW32:-lws2_32 -lgdi32 -lcrypt32:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts} EXPORT_VAR_AS_FN:${x86_asm}:coff:win32:cygwin-shared:-D_WINDLL -DOPENSSL_USE_APPLINK:-mno-cygwin:.dll.a",
+"mingwvc-parity","parity.gnu.gcc:-DL_ENDIAN -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_DEPRECATE:::MINGW32:-lws2_32 -lgdi32 -lcrypt32:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts} EXPORT_VAR_AS_FN:${no_asm}:win32:cygwin-shared:-D_WINDLL -fPIC -DPIC::.dll.a",
 # As for OPENSSL_USE_APPLINK. Applink makes it possible to use .dll
 # compiled with one compiler with application compiled with another
 # compiler. It's possible to engage Applink support in mingw64 build,
--- openssl-1.0.2l/e_os.h	2017-05-25 14:54:34.000000000 +0200
+++ openssl-1.0.2l-.default/e_os.h	2017-10-02 15:13:18.575589700 +0200
@@ -652,7 +652,7 @@
  * Some IPv6 implementations are broken, disable them in known bad versions.
  */
 #  if !defined(OPENSSL_USE_IPV6)
-#   if defined(AF_INET6) && !defined(OPENSSL_SYS_BEOS_BONE) && !defined(NETWARE_CLIB)
+#   if defined(AF_INET6) && !defined(OPENSSL_SYS_BEOS_BONE) && !defined(NETWARE_CLIB) && !defined(_WIN32)
 #    define OPENSSL_USE_IPV6 1
 #   else
 #    define OPENSSL_USE_IPV6 0
--- openssl-1.0.2l/util/mklink.pl	2017-05-25 14:54:34.000000000 +0200
+++ openssl-1.0.2l-.default/util/mklink.pl	2017-09-27 17:22:29.528648000 +0200
@@ -52,6 +52,7 @@
 my $file;
 $symlink_exists=eval {symlink("",""); 1};
 if ($^O eq "msys") { $symlink_exists=0 };
+if ($^O eq "cygwin") { $symlink_exists=0 }; # may use cl.exe on cygwin
 foreach $file (@files) {
     my $err = "";
     if ($symlink_exists) {
