diff -Naur openssl-0.9.7e.orig/Configure openssl-0.9.7e/Configure
--- openssl-0.9.7e.orig/Configure	2005-09-01 21:48:06.000000000 +0900
+++ openssl-0.9.7e/Configure	2005-09-01 22:05:34.000000000 +0900
@@ -424,6 +424,7 @@
 # Linux on ARM
 "linux-elf-arm","gcc:-DL_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -Wall::-D_REENTRANT::-ldl:BN_LLONG::::::::::dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 "linux-elf-armb","gcc:-DB_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -Wall::-D_REENTRANT::-ldl:BN_LLONG::::::::::dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
+"linux-elf-sh","gcc:-DTERMIO -O3 -fomit-frame-pointer -Wall::-D_REENTRANT::-ldl:BN_LLONG::::::::::dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 
 # SCO/Caldera targets.
 #
diff -Naur openssl-0.9.7e.orig/config openssl-0.9.7e/config
--- openssl-0.9.7e.orig/config	2005-09-01 21:48:04.000000000 +0900
+++ openssl-0.9.7e/config	2005-09-01 22:07:26.000000000 +0900
@@ -595,6 +595,8 @@
 	OUT="linux-parisc" ;;
   arm*b*-*-linux2) OUT="linux-elf-armb" ;;
   arm*-*-linux2) OUT="linux-elf-arm" ;;
+  sh*b*-*-linux2) OUT="linux-elf-sh"; options="$options -DB_ENDIAN" ;;
+  sh*-*-linux2) OUT="linux-elf-sh"; options="$options -DL_ENDIAN" ;;
   s390-*-linux2) OUT="linux-s390" ;;
   s390x-*-linux?) OUT="linux-s390x" ;;
   x86_64-*-linux?) OUT="linux-x86_64" ;;
