diff -ur openssl-0.9.7e.orig/Configure openssl-0.9.7e/Configure
--- openssl-0.9.7e.orig/Configure	2004-12-12 17:44:17.030893680 -0500
+++ openssl-0.9.7e/Configure	2004-12-12 17:44:39.143532048 -0500
@@ -423,6 +423,7 @@
 
 # Linux on ARM
 "linux-elf-arm","gcc:-DL_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -Wall::-D_REENTRANT::-ldl:BN_LLONG::::::::::dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
+"linux-elf-armb","gcc:-DB_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -Wall::-D_REENTRANT::-ldl:BN_LLONG::::::::::dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 
 # SCO/Caldera targets.
 #
diff -ur openssl-0.9.7e.orig/config openssl-0.9.7e/config
--- openssl-0.9.7e.orig/config	2004-12-12 17:44:17.029893832 -0500
+++ openssl-0.9.7e/config	2004-12-12 17:44:30.935779816 -0500
@@ -593,6 +593,7 @@
 
 	options="$options -mschedule=$CPUSCHEDULE -march=$CPUARCH"
 	OUT="linux-parisc" ;;
+  arm*b*-*-linux2) OUT="linux-elf-armb" ;;
   arm*-*-linux2) OUT="linux-elf-arm" ;;
   s390-*-linux2) OUT="linux-s390" ;;
   s390x-*-linux?) OUT="linux-s390x" ;;
