diff -r -c -C 2 openssl-0.9.6g-orig/crypto/des/fcrypt.c openssl-0.9.6g/crypto/des/fcrypt.c
*** openssl-0.9.6g-orig/crypto/des/fcrypt.c	Thu Sep 26 15:20:47 2002
--- openssl-0.9.6g/crypto/des/fcrypt.c	Thu Sep 26 15:21:24 2002
***************
*** 62,66 ****
  	DES_LONG Eswap0, DES_LONG Eswap1);
  
! #if !defined(PERL5) && !defined(__FreeBSD__) && !defined(NeXT) && !defined(_DARWIN)
  char *crypt(const char *buf, const char *salt)
  	{
--- 62,66 ----
  	DES_LONG Eswap0, DES_LONG Eswap1);
  
! #if !defined(PERL5) && !defined(__FreeBSD__) && !defined(NeXT) && !defined(_DARWIN) && !defined(__GLIBC__)
  char *crypt(const char *buf, const char *salt)
  	{
diff -r -c -C 2 openssl-0.9.6g-orig/test/Makefile openssl-0.9.6g/test/Makefile
*** openssl-0.9.6g-orig/test/Makefile	Thu Sep 26 15:20:47 2002
--- openssl-0.9.6g/test/Makefile	Thu Sep 26 15:23:26 2002
***************
*** 28,32 ****
  DLIBCRYPTO= ../libcrypto.a
  DLIBSSL= ../libssl.a
! LIBCRYPTO= -L.. -lcrypto
  LIBSSL= -L.. -lssl
  
--- 28,32 ----
  DLIBCRYPTO= ../libcrypto.a
  DLIBSSL= ../libssl.a
! LIBCRYPTO= -L.. -lcrypto -lcrypt
  LIBSSL= -L.. -lssl
  
diff -r -c -C 2 openssl-0.9.6g-orig/test/Makefile.ssl openssl-0.9.6g/test/Makefile.ssl
*** openssl-0.9.6g-orig/test/Makefile.ssl	Thu Sep 26 15:20:47 2002
--- openssl-0.9.6g/test/Makefile.ssl	Thu Sep 26 15:23:26 2002
***************
*** 28,32 ****
  DLIBCRYPTO= ../libcrypto.a
  DLIBSSL= ../libssl.a
! LIBCRYPTO= -L.. -lcrypto
  LIBSSL= -L.. -lssl
  
--- 28,32 ----
  DLIBCRYPTO= ../libcrypto.a
  DLIBSSL= ../libssl.a
! LIBCRYPTO= -L.. -lcrypto -lcrypt
  LIBSSL= -L.. -lssl
  
