diff -ru openssl-1.1.0f/Configurations/10-main.conf openssl-1.1.0f-.default/Configurations/10-main.conf
--- openssl-1.1.0f/Configurations/10-main.conf	2017-05-25 14:46:17.000000000 +0200
+++ openssl-1.1.0f-.default/Configurations/10-main.conf	2017-09-28 12:56:22.544307300 +0200
@@ -1463,6 +1463,28 @@
         multilib         => "",
         apps_aux_src     => add("win32_init.c"),
     },
+    "mingwvc-parity" => {
+        inherit_from     => [ "BASE_unix", asm("x86_asm"),
+                              sub { $disabled{shared} ? () : "uplink_common" } ],
+        cc               => "gcc",
+        cflags           => combine(picker(default => "-DL_ENDIAN -DWIN32_LEAN_AND_MEAN -DUNICODE -D_UNICODE -m32 -Wall",
+                                           debug   => "-g -O0",
+                                           release => "-O3 -fomit-frame-pointer"),
+                                    threads("-D_MT")),
+        sys_id           => "WIN32",
+        ex_libs          => add("-lws2_32 -lgdi32 -lcrypt32"),
+        bn_ops           => "BN_LLONG EXPORT_VAR_AS_FN",
+        thread_scheme    => "winthreads",
+        perlasm_scheme   => "win32",
+        dso_scheme       => "win32",
+        shared_target    => "mingw-shared",
+        shared_cflag     => add("-D_WINDLL"),
+        shared_ldflag    => "-static-libgcc",
+        shared_rcflag    => "--target=pe-i386",
+        shared_extension => ".dll",
+        multilib         => "",
+        apps_aux_src     => add("win32_init.c"),
+    },
     "mingw64" => {
         # As for OPENSSL_USE_APPLINK. Applink makes it possible to use
         # .dll compiled with one compiler with application compiled with
