--- Configure
+++ Configure
@@ -967,6 +967,10 @@
 				{
 				$cross_compile_prefix=$1;
 				}
+			elsif (/^--with-aix-soname=(aix|both|svr4)$/)
+				{
+				$withargs{"aix-soname"}=$1;
+				}
 			elsif (/^-[lL](.*)$/ or /^-Wl,/)
 				{
 				$libs.=$_." ";
@@ -1431,6 +1435,10 @@
 	$no_shared_warn = 1 if !$no_shared;
 	$no_shared = 1;
 	}
+elsif ($shared_target eq "aix-shared" && defined($withargs{"aix-soname"}))
+	{
+	$shared_target .= "-" . $withargs{"aix-soname"};
+	}
 if (!$no_shared)
 	{
 	if ($shared_cflag ne "")
--- Makefile.shared
+++ Makefile.shared
@@ -545,6 +545,38 @@
 	NOALLSYMSFLAGS=''; \
 	SHAREDFLAGS='$(CFLAGS) $(SHARED_LDFLAGS) -Wl,-bexpall,-bnolibpath,-bM:SRE'; \
 	$(LINK_SO_O);
+link_a.aix.svr4:
+	@ $(CALC_VERSIONS); \
+	OBJECT_MODE=`expr "x$(SHARED_LDFLAGS)" : 'x\-[a-z]*\(64\)'` || : ; \
+	OBJECT_MODE=$${OBJECT_MODE:-32}; export OBJECT_MODE; \
+	SHLIB=lib$(LIBNAME).so; \
+	SHLIB_SUFFIX=; \
+	test "$$OBJECT_MODE" = 64 && shr='shr_64' || shr='shr'; \
+	shrexp=$$SHLIB$$SHLIB_SOVER.d/$$shr.exp; \
+	shrimp=$$SHLIB$$SHLIB_SOVER.d/$$shr.imp; \
+	ALLSYMSFLAGS="-Wl,-bexport:$$shrexp"; \
+	NOALLSYMSFLAGS=''; \
+	SHAREDFLAGS='$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-G,-bernotok,-bsymbolic,-bnolibpath'; \
+	NM='$(NM)'; : $${NM:=nm}; \
+	AR='$(AR)'; : $${AR:=ar r}; \
+	STRIP='$(STRIP)'; : $${STRIP:=strip}; \
+	SHOBJECTS="lib$(LIBNAME).a $(LIBEXTRAS)"; \
+	( rm -rf $$SHLIB$$SHLIB_SOVER.d && \
+	  mkdir $$SHLIB$$SHLIB_SOVER.d && \
+	  $${NM} -PCpgl $$SHOBJECTS \
+	    | awk '{ \
+		if ((($$ 2 == "T") || ($$ 2 == "D") || ($$ 2 == "B") || ($$ 2 == "W") || ($$ 2 == "V") || ($$ 2 == "Z")) && (substr($$ 1, 1, 1) != ".")) { \
+		  if (($$ 2 == "W") || ($$ 2 == "V") || ($$ 2 == "Z")) { print $$ 1 " weak" } else { print $$ 1 } \
+	      } }' \
+	    | sort -u > $$shrexp && \
+	  { echo "#! $$SHLIB$$SHLIB_SOVER($$shr.o)"; echo "# $$OBJECT_MODE"; cat $$shrexp; } > $$shrimp && \
+	  INHIBIT_SYMLINKS=1; \
+	  SHLIB_SUFFIX=".d/$$shr.o"; \
+	  $(LINK_SO) && \
+	  $${STRIP} -e $$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX && \
+	  $${AR} $$SHLIB$$SHLIB_SOVER $$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX $$shrimp && \
+	  rm -rf $$SHLIB$$SHLIB_SOVER.d \
+	) && $(SYMLINK_SO)
 link_a.aix:
 	@ $(CALC_VERSIONS); \
 	OBJECT_MODE=`expr "x$(SHARED_LDFLAGS)" : 'x\-[a-z]*\(64\)'` || : ; \
@@ -649,6 +681,10 @@
 link_a.aix-shared: link_a.aix
 link_app.aix-shared: link_app.aix
 symlink.aix-shared: symlink.aix
+link_o.aix-shared-svr4: link_o.aix
+link_a.aix-shared-svr4: link_a.aix.svr4
+link_app.aix-shared-svr4: link_app.aix
+symlink.aix-shared-svr4: symlink.aix
 link_o.reliantunix-shared: link_o.reliantunix
 link_a.reliantunix-shared: link_a.reliantunix
 link_app.reliantunix-shared: link_app.reliantunix
