diff -ur openssl-0.9.7c.orig/apps/der_chop openssl-0.9.7c/apps/der_chop
--- openssl-0.9.7c.orig/apps/der_chop	2003-09-30 14:38:15.000000000 +0200
+++ openssl-0.9.7c/apps/der_chop	2004-09-13 13:46:06.000000000 +0200
@@ -65,7 +65,7 @@
 	}
 else
 	{
-	$file="/tmp/a$$.DER";
+	$file=`mktemp -t a$$.DER.XXXXXX` || die $!;
 	open(OUT,">$file") || die "unable to open $file:$!\n";
 	for (;;)
 		{
@@ -139,7 +139,7 @@
 	{
 	local($str)=@_;
 	local(*OUT,*IN,@a,$t,$d,$ret);
-	local($file)="/tmp/b$$.DER";
+	local($file)=`mktemp -t b$$.DER.XXXXXX` || die $!;
 	local(@ret);
 
 	open(OUT,">$file");
@@ -225,7 +225,7 @@
 sub do_private_key
 	{
 	local($data,@struct)=@_;
-	local($file)="/tmp/b$$.DER";
+	local($file)=`mktemp -t b$$.DER.XXXXXX` || die $!;
 	local($off,$d,$hl,$len,$_,$b,@p,$s);
 
 	($type)=($struct[4] =~ /OBJECT_IDENTIFIER :(.*)\s*$/);
@@ -260,7 +260,7 @@
 sub do_certificate
 	{
 	local($data,@struct)=@_;
-	local($file)="/tmp/b$$.DER";
+	local($file)=`mktemp -t b$$.DER.XXXXXX` || die $!;
 	local($off,$d,$hl,$len,$_,$b,@p,$s);
 
 	($off,$d,$hl,$len)=&parse_line($struct[2]);
diff -ur openssl-0.9.7c.orig/apps/der_chop.in openssl-0.9.7c/apps/der_chop.in
--- openssl-0.9.7c.orig/apps/der_chop.in	1999-05-26 00:31:49.000000000 +0200
+++ openssl-0.9.7c/apps/der_chop.in	2004-09-13 13:43:46.000000000 +0200
@@ -65,7 +65,7 @@
 	}
 else
 	{
-	$file="/tmp/a$$.DER";
+	$file=`mktemp -t a$$.DER.XXXXXX` || die $!;
 	open(OUT,">$file") || die "unable to open $file:$!\n";
 	for (;;)
 		{
@@ -139,7 +139,7 @@
 	{
 	local($str)=@_;
 	local(*OUT,*IN,@a,$t,$d,$ret);
-	local($file)="/tmp/b$$.DER";
+	local($file)=`mktemp -t b$$.DER.XXXXXX` || die $!;
 	local(@ret);
 
 	open(OUT,">$file");
@@ -225,7 +225,7 @@
 sub do_private_key
 	{
 	local($data,@struct)=@_;
-	local($file)="/tmp/b$$.DER";
+	local($file)=`mktemp -t b$$.DER.XXXXXX` || die $!;
 	local($off,$d,$hl,$len,$_,$b,@p,$s);
 
 	($type)=($struct[4] =~ /OBJECT_IDENTIFIER :(.*)\s*$/);
@@ -260,7 +260,7 @@
 sub do_certificate
 	{
 	local($data,@struct)=@_;
-	local($file)="/tmp/b$$.DER";
+	local($file)=`mktemp -t b$$.DER.XXXXXX` || die $!;
 	local($off,$d,$hl,$len,$_,$b,@p,$s);
 
 	($off,$d,$hl,$len)=&parse_line($struct[2]);
