--- apps/s_socket.c
+++ apps/s_socket.c
@@ -718,9 +718,14 @@
 		if (domain == AF_INET)
 			ret=gethostbyname(name);
 #if OPENSSL_USE_IPV6
+#if defined (__SVR4) && defined (__sun)
+		else
+			ret=getipnodebyname(name, AF_INET6, AI_DEFAULT, NULL);
+#else
 		else
 			ret=gethostbyname2(name, AF_INET6);
 #endif
+#endif
 		if (ret == NULL) return(NULL);
 		/* else add to cache */
 		if(strlen(name) < sizeof ghbn_cache[0].name)
