diff -ur openssl-0.9.7d.orig/Configure openssl-0.9.7d/Configure
--- openssl-0.9.7d.orig/Configure	2004-12-11 19:01:11.078248352 -0500
+++ openssl-0.9.7d/Configure	2004-12-11 19:06:46.998180760 -0500
@@ -315,7 +315,7 @@
 "MPE/iX-gcc", "gcc:-D_ENDIAN -DBN_DIV2W -O3 -D_POSIX_SOURCE -D_SOCKET_SOURCE -I/SYSLOG/PUB::(unknown):MPE:-L/SYSLOG/PUB -lsyslog -lsocket -lcurses:BN_LLONG DES_PTR DES_UNROLL DES_RISC1:::",
 
 #### PARISC Linux setups
-"linux-parisc","gcc:-DB_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -Wall -DBN_DIV2W::-D_REENTRANT:::BN_LLONG RC4_CHAR RC4_CHUNK DES_UNROLL BF_PTR::",
+"linux-parisc","gcc:-DB_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -Wall -DBN_DIV2W::-D_REENTRANT::-ldl:BN_LLONG RC4_CHAR RC4_CHUNK DES_UNROLL BF_PTR::::::::::dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 
 # Dec Alpha, OSF/1 - the alpha164-cc is historical, for the conversion
 # from the older DEC C Compiler to the newer compiler.  It's now the
diff -ur openssl-0.9.7d.orig/config openssl-0.9.7d/config
--- openssl-0.9.7d.orig/config	2004-12-11 19:01:11.077248504 -0500
+++ openssl-0.9.7d/config	2004-12-11 19:06:55.621869760 -0500
@@ -564,9 +564,9 @@
 	sun4d)	OUT="linux-sparcv8" ;;
 	*)	OUT="linux-sparcv7" ;;
 	esac ;;
-  parisc-*-linux2)
+  parisc*-*-linux2)
-        CPUARCH=`awk '/cpu family/{print substr($5,1,3)}' /proc/cpuinfo`
-	CPUSCHEDULE=`awk '/^cpu.[ 	]: PA/{print substr($3,3)}' /proc/cpuinfo`
+        CPUARCH=`awk '/cpu family/{print substr($5,1,3)}' /proc/cpuinfo | head -n 1`
+	CPUSCHEDULE=`awk '/^cpu[[:space:]]*: PA/{print substr($3,3)}' /proc/cpuinfo | head -n 1`
 
 	# ??TODO ??  Model transformations
 	# 0. CPU Architecture for the 1.1 processor has letter suffixes. We strip that off
