--- openssl-0.9.8a/engines/Makefile	2005-06-28 04:10:18.000000000 -0500
+++ openssl-0.9.8a-new/engines/Makefile	2005-10-29 16:16:23.000000000 -0500
@@ -88,15 +88,24 @@
 	@[ -n "$(INSTALLTOP)" ] # should be set by top Makefile...
 	@if [ -n "$(SHARED_LIBS)" ]; then \
 		set -e; \
+		$(PERL) $(TOP)/util/mkdir-p.pl $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines; \
+		if [ "$(PLATFORM)" == "Cygwin" ]; then \
+			SHLIB="dll"; \
+		elif [ "$(PLATFORM)" == "Darwin" ]; then \
+			SHLIB="dylib"; \
+		else \
+			SHLIB="so"; \
+		fi; \
 		for l in $(LIBNAMES); do \
 			( echo installing $$l; \
-			  if [ "$(PLATFORM)" != "Cygwin" ]; then \
-				  cp lib$$l.so $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines/lib$$l.so.new; \
-			  else \
-				  cp cyg$$l.dll $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines/lib$$l.so.new; \
-			  fi; \
-			  chmod 555 $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines/lib$$l.so.new; \
-			  mv -f $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines/lib$$l.so.new $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines/lib$$l.so ); \
+			if [ "$(PLATFORM)" != "Cygwin" ]; then \
+			  cp lib$$l.$(SHLIB) $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines/lib$$l.$(SHLIB).new; \
+			else \
+			  cp cyg$$l.dll $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines/lib$$l.so.new; \
+			fi; \
+			  cp lib$$l.$(SHLIB) $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines/lib$$l.$(SHLIB).new; \
+			  chmod 555 $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines/lib$$l.$(SHLIB).new; \
+			  mv -f $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines/lib$$l.$(SHLIB).new $(INSTALL_PREFIX)$(INSTALLTOP)/lib/engines/lib$$l.$(SHLIB) ); \
 		done; \
 	fi
 
