--- icecream-create-env.orig	2008-09-15 19:43:50 +0200
+++ icecream-create-env	2008-09-15 19:52:09 +0200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 #
 # icecream-create-env - helper script to create icecc environments(mostly for cross-compiling)
 #
@@ -49,9 +49,9 @@
 
 if [ "`gcc -dumpmachine`" = "${prefix}" ]
 then
-	/usr/lib/icecc/icecc-create-env /usr/${prefix}/gcc-bin/${version}/gcc /usr/${prefix}/gcc-bin/${version}/g++ | tee ${tmpfile}
+	"@GENTOO_PORTAGE_EPREFIX@"/usr/lib/icecc/icecc-create-env "@GENTOO_PORTAGE_EPREFIX@"/usr/${prefix}/gcc-bin/${version}/gcc "@GENTOO_PORTAGE_EPREFIX@"/usr/${prefix}/gcc-bin/${version}/g++ | tee ${tmpfile}
 else
-	/usr/lib/icecc/icecc-create-env /usr/${CHOST}/${prefix}/gcc-bin/${version}/${prefix}-gcc /usr/${CHOST}/${prefix}/gcc-bin/${version}/${prefix}-g++ | tee ${tmpfile}
+	"@GENTOO_PORTAGE_EPREFIX@"/usr/lib/icecc/icecc-create-env "@GENTOO_PORTAGE_EPREFIX@"/usr/${CHOST}/${prefix}/gcc-bin/${version}/${prefix}-gcc "@GENTOO_PORTAGE_EPREFIX@"/usr/${CHOST}/${prefix}/gcc-bin/${version}/${prefix}-g++ | tee ${tmpfile}
 fi
 
 # figure out the name of the archive
@@ -60,7 +60,7 @@
 echo "Testing icecc environment..."
 tar -x -z -f ${icecc_envname} -C ${tmpdir}
 touch ${tmpdir}/empty.c
-chroot ${tmpdir}/ /usr/bin/gcc -c /empty.c
+chroot ${tmpdir}/ "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/gcc -c /empty.c
 tested=${?}
 rm ${tmpdir}/empty.c
 
