--- icecream-config
+++ icecream-config
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 #
 # icecream-config - helper script for icecream and its ebuild
 #
@@ -24,7 +24,7 @@
 #  automatically update icecream's links when upgrading toolchains
 #
 icecc_remove_link() {
-	local t="/usr/lib/icecc/bin/${1}"
+	local t="@GENTOO_PORTAGE_EPREFIX@/usr/lib/icecc/bin/${1}"
 	if [ -L ${t} ]; then
 		icecc_echo "Removing ${t}..."
 		rm -f "${t}"
@@ -39,9 +39,9 @@
 		ICECC_QUIET=1 icecc_remove_link "${1}"
 
 		# then create the new link
-		local t="/usr/lib/icecc/bin/${1}"
+		local t="@GENTOO_PORTAGE_EPREFIX@/usr/lib/icecc/bin/${1}"
 		icecc_echo "Creating icecream shadow link: ${t}..."
-		ln -s /usr/bin/icecc "${t}"
+		ln -s "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/icecc "${t}"
 	fi
 }
 icecc_links() {
