--- icecream/client/icecc-create-env
+++ icecream/client/icecc-create-env
@@ -4,6 +4,8 @@
 # Copyright (C) 2004 by the Icecream Authors
 # GPL
 
+prefix="@GENTOO_PORTAGE_EPREFIX@"
+
 target_files=
 
 is_darwin=0
@@ -95,24 +97,24 @@
   exit 1
 fi
 
-add_file $added_gcc /usr/bin/gcc
-add_file $added_gxx /usr/bin/g++
-add_file /usr/bin/as
+add_file $added_gcc "${prefix}"/usr/bin/gcc
+add_file $added_gxx "${prefix}"/usr/bin/g++
+add_file "${eprefix}"/usr/bin/as
 
 if test "$is_darwin" = 1; then
     # add dynamic linker
     add_file /usr/lib/dyld
-    real_file=`/usr/bin/gcc --version | head -n 1 2>&1 | cut -d" " -f1`
-    add_file /usr/bin/$real_file
-    real_file=`/usr/bin/g++ --version | head -n 1 2>&1 | cut -d" " -f1`
-    add_file /usr/bin/$real_file
-    real_file=`/usr/bin/as -micha -- < /dev/null 2>&1 | cut -d: -f1`
+    real_file=`"${prefix}"/usr/bin/gcc --version | head -n 1 2>&1 | cut -d" " -f1`
+    add_file "${prefix}"/usr/bin/$real_file
+    real_file=`"${prefix}"/usr/bin/g++ --version | head -n 1 2>&1 | cut -d" " -f1`
+    add_file "${prefix}"/usr/bin/$real_file
+    real_file=`"${prefix}"/usr/bin/as -micha -- < /dev/null 2>&1 | cut -d: -f1`
     add_file $real_file
 fi
 
 
-add_file `$added_gcc -print-prog-name=cc1` /usr/bin/cc1
-add_file `$added_gxx -print-prog-name=cc1plus` /usr/bin/cc1plus
+add_file `$added_gcc -print-prog-name=cc1` "${eprefix}"/usr/bin/cc1
+add_file `$added_gxx -print-prog-name=cc1plus` "${eprefix}"/usr/bin/cc1plus
 specfile=`$added_gcc -print-file-name=specs`
 if test -n "$specfile" && test -e "$specfile"; then
   add_file "$specfile"
@@ -142,8 +144,8 @@
 
 md5sum=NONE
 for file in /usr/bin/md5sum /bin/md5 /usr/bin/md5 /sbin/md5; do
-   if test -x $file; then
-	md5sum=$file
+   if test -x "${prefix}"$file; then
+	md5sum="${prefix}"$file
         break
    fi
 done
--- icecream/client/main.cpp
+++ icecream/client/main.cpp
@@ -141,8 +141,8 @@
     string gcc, gpp;
 
     // perhaps we're on gentoo
-    if ( !lstat("/usr/bin/gcc-config", &st) ) {
-        string gccpath=read_output("/usr/bin/gcc-config -B") + "/";
+    if ( !lstat("@GENTOO_PORTAGE_EPREFIX@/usr/bin/gcc-config", &st) ) {
+        string gccpath=read_output("@GENTOO_PORTAGE_EPREFIX@/usr/bin/gcc-config -B") + "/";
         gcc=gccpath + "gcc";
         gpp=gccpath + "g++";
     } else {
--- icecream/daemon/environment.cpp
+++ icecream/daemon/environment.cpp
@@ -205,7 +205,7 @@
     native_environment = "";
     string nativedir = basedir + "/native/";
 
-    if ( ::access( "/usr/bin/gcc", X_OK ) || ::access( "/usr/bin/g++", X_OK ) ) 
+    if ( ::access( "@GENTOO_PORTAGE_EPREFIX@/usr/bin/gcc", X_OK ) || ::access( "@GENTOO_PORTAGE_EPREFIX@/usr/bin/g++", X_OK ) ) 
 	return 0;
 
     if ( mkdir( nativedir.c_str(), 0755 ) )
--- icecream/daemon/workit.cpp
+++ icecream/daemon/workit.cpp
@@ -237,9 +237,9 @@
         char **argv = new char*[argc + 1];
 	int i = 0;
         if (j.language() == CompileJob::Lang_C)
-            argv[i++] = strdup( "usr/bin/gcc" );
+            argv[i++] = strdup( "@GENTOO_PORTAGE_EPREFIX@/usr/bin/gcc" );
         else if (j.language() == CompileJob::Lang_CXX)
-            argv[i++] = strdup( "usr/bin/g++" );
+            argv[i++] = strdup( "@GENTOO_PORTAGE_EPREFIX@/usr/bin/g++" );
         else
             assert(0);
 
