diff -Naur poppler-0.5.9.orig/goo/FixedPoint.h poppler-0.5.9/goo/FixedPoint.h
--- poppler-0.5.9.orig/goo/FixedPoint.h	2007-03-30 20:57:17.000000000 +0200
+++ poppler-0.5.9/goo/FixedPoint.h	2007-06-26 00:50:03.000000000 +0200
@@ -11,7 +11,7 @@
 #ifndef FIXEDPOINT_H
 #define FIXEDPOINT_H
 
-#include <config.h>
+#include <poppler/poppler-config.h>
 
 #if USE_FIXEDPOINT
 
diff -Naur poppler-0.5.9.orig/goo/gmem.h poppler-0.5.9/goo/gmem.h
--- poppler-0.5.9.orig/goo/gmem.h	2007-04-01 01:17:26.000000000 +0200
+++ poppler-0.5.9/goo/gmem.h	2007-06-26 00:49:35.000000000 +0200
@@ -10,7 +10,7 @@
 #define GMEM_H
 
 #include <stdio.h>
-#include <config.h>
+#include <poppler/poppler-config.h>
 
 #if USE_EXCEPTIONS
 
diff -Naur poppler-0.5.9.orig/poppler/poppler-config.h.in poppler-0.5.9/poppler/poppler-config.h.in
--- poppler-0.5.9.orig/poppler/poppler-config.h.in	2007-05-30 20:33:56.000000000 +0200
+++ poppler-0.5.9/poppler/poppler-config.h.in	2007-06-26 00:48:56.000000000 +0200
@@ -15,10 +15,20 @@
 // redefining those symbols.
 
 /* Enable multithreading support. */
-#ifndef MULTITHREADING
+#ifndef MULTITHREADED
 #undef MULTITHREADED
 #endif
 
+/* Enable exceptions. */
+#ifndef USE_EXCEPTIONS
+#undef USE_EXCEPTIONS
+#endif
+
+/* Use fixedpoint. */
+#ifndef USE_FIXEDPOINT
+#undef USE_FIXEDPOINT
+#endif
+
 /* Full path for the system-wide xpdfrc file. */
 #ifndef SYSTEM_XPDFRC
 #undef SYSTEM_XPDFRC
diff -Naur poppler-0.5.9.orig/poppler/SecurityHandler.h poppler-0.5.9/poppler/SecurityHandler.h
--- poppler-0.5.9.orig/poppler/SecurityHandler.h	2007-04-05 01:11:52.000000000 +0200
+++ poppler-0.5.9/poppler/SecurityHandler.h	2007-06-26 00:47:40.000000000 +0200
@@ -9,7 +9,7 @@
 #ifndef SECURITYHANDLER_H
 #define SECURITYHANDLER_H
 
-#include <config.h>
+#include "poppler-config.h"
 
 #ifdef USE_GCC_PRAGMAS
 #pragma interface
diff -Naur poppler-0.5.9.orig/utils/ImageOutputDev.h poppler-0.5.9/utils/ImageOutputDev.h
--- poppler-0.5.9.orig/utils/ImageOutputDev.h	2007-02-01 21:01:00.000000000 +0100
+++ poppler-0.5.9/utils/ImageOutputDev.h	2007-06-26 00:50:38.000000000 +0200
@@ -9,7 +9,7 @@
 #ifndef IMAGEOUTPUTDEV_H
 #define IMAGEOUTPUTDEV_H
 
-#include <poppler-config.h>
+#include <poppler/poppler-config.h>
 
 #ifdef USE_GCC_PRAGMAS
 #pragma interface
