--- ghc-updater
+++ ghc-updater
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 #
 # This script has been modified by kosmikus and is based on 
 # python-updater by liquidx.
@@ -24,17 +24,17 @@
 shopt -s nullglob
 
 # fix the PATH to include the dirs where portage installs ghc
-PATH="/usr/bin:/opt/ghc/bin:${PATH}"
+PATH="@GENTOO_PORTAGE_EPREFIX@/usr/bin:@GENTOO_PORTAGE_EPREFIX@/opt/ghc/bin:${PATH}"
 
 NEW_GHC_VER=$(ghc --version | sed 's:^.*version ::')
 NEW_GHC_LIBDIR=$(ghc --print-libdir)
 
 PKGS_EXCEPTIONS="dev-lang/ghc dev-lang/ghc-bin"
 PKGS_MANUAL=""
-LOGFILE="/var/log/ghc-updater.log"
+LOGFILE="@GENTOO_PORTAGE_EPREFIX@/var/log/ghc-updater.log"
 
 # portage variables
-PKG_DBDIR=/var/db/pkg
+PKG_DBDIR=@GENTOO_PORTAGE_EPREFIX@/var/db/pkg
 
 # moved the portageq checks into a function to make command
 # line parsing immediate
@@ -47,7 +47,7 @@
 PRETEND=0
 PKGS_TO_REMERGE=""
 PKGS_COUNT_REMERGE=0
-PORTAGE_PYTHON="/usr/bin/python"
+PORTAGE_PYTHON="@GENTOO_PORTAGE_EPREFIX@/usr/bin/python"
 
 usage() {
 	echo "usage: ghc-updater [options]"
@@ -83,7 +83,7 @@
 
 # /etc/init.d/functions.sh always points to the correct functions.sh no
 # matter which version of baselayout
-source /etc/init.d/functions.sh
+source @GENTOO_PORTAGE_EPREFIX@/etc/init.d/functions.sh
 
 # misc helper functions
 eloginfo() {
@@ -152,7 +152,7 @@
 #
 OLD_PACKAGES_DIR=""
 # Exclude new library dir and lib symlinks:
-for d in /{usr,opt/ghc}/lib{,64}/ghc{,-bin}-*; do
+for d in @GENTOO_PORTAGE_EPREFIX@/{usr,opt/ghc}/lib{,64}/ghc{,-bin}-*; do
     [[ "${d}" == ${NEW_GHC_LIBDIR} ]] || [[ -L ${d%/*} ]] || OLD_PACKAGES_DIR="${OLD_PACKAGES_DIR}${d} "
 done
 
