diff -ru libX11-1.2.2.orig/configure.ac libX11-1.2.2/configure.ac
--- libX11-1.2.2.orig/configure.ac	2009-09-25 08:28:41 +0200
+++ libX11-1.2.2/configure.ac	2009-09-25 08:35:57 +0200
@@ -93,6 +93,11 @@
 		# Unix domain sockets are faster now.
 		DEFAULT_LOCAL_TRANS="UNIX_TRANS,LOCAL_TRANS,TCP_TRANS"
 		;;
+	winnt*)
+		# On Windows, there is no local transport, since we have no
+		# local transports other than tcp.
+		DEFAULT_LOCAL_TRANS="TCP_TRANS"
+		;;
 	*)
 		if test "$LOCALCONN" = "yes"; then
 			DEFAULT_LOCAL_TRANS="LOCAL_TRANS,UNIX_TRANS,TCP_TRANS"
