diff -ru -x '*.Po' -x '*.Plo' libX11-1.2.2.orig/src/xcb_io.c libX11-1.2.2/src/xcb_io.c
--- libX11-1.2.2.orig/src/xcb_io.c	2009-08-21 08:11:22 +0200
+++ libX11-1.2.2/src/xcb_io.c	2009-08-21 08:22:53 +0200
@@ -13,6 +13,11 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifdef __INTERIX
+/* required for fd_set */
+# include <sys/time.h>
+#endif
+
 static void return_socket(void *closure)
 {
 	Display *dpy = closure;
