diff -ru -x '*.Po' -x '*.Plo' libX11-1.2.2.orig/include/X11/Xlib.h libX11-1.2.2/include/X11/Xlib.h
--- libX11-1.2.2.orig/include/X11/Xlib.h	2009-09-24 11:30:37 +0200
+++ libX11-1.2.2/include/X11/Xlib.h	2009-09-24 17:07:22 +0200
@@ -1387,7 +1387,13 @@
 #define _Xdebug (*_Xdebug_p)
 #endif
 
-extern int _Xdebug;
+#ifdef __PARITY__
+# define X11_IMPORT __declspec(dllimport)
+#else
+# define X11_IMPORT
+#endif
+
+extern X11_IMPORT int _Xdebug;
 
 extern XFontStruct *XLoadQueryFont(
     Display*		/* display */,
diff -ru -x '*.Po' -x '*.Plo' libX11-1.2.2.orig/include/X11/Xlibint.h libX11-1.2.2/include/X11/Xlibint.h
--- libX11-1.2.2.orig/include/X11/Xlibint.h	2009-09-24 11:30:37 +0200
+++ libX11-1.2.2/include/X11/Xlibint.h	2009-09-24 17:07:22 +0200
@@ -274,21 +274,27 @@
 #define _Xglobal_lock (*_Xglobal_lock_p)
 #endif
 
+#ifdef __PARITY__
+# define X11_IMPORT __declspec(dllimport)
+#else
+# define X11_IMPORT
+#endif
+
 /* in XlibInt.c */
-extern void (*_XCreateMutex_fn)(
+extern X11_IMPORT void (*_XCreateMutex_fn)(
     LockInfoPtr /* lock */
 );
-extern void (*_XFreeMutex_fn)(
+extern X11_IMPORT void (*_XFreeMutex_fn)(
     LockInfoPtr /* lock */
 );
-extern void (*_XLockMutex_fn)(
+extern X11_IMPORT void (*_XLockMutex_fn)(
     LockInfoPtr	/* lock */
 #if defined(XTHREADS_WARN) || defined(XTHREADS_FILE_LINE)
     , char * /* file */
     , int /* line */
 #endif
 );
-extern void (*_XUnlockMutex_fn)(
+extern X11_IMPORT void (*_XUnlockMutex_fn)(
     LockInfoPtr	/* lock */
 #if defined(XTHREADS_WARN) || defined(XTHREADS_FILE_LINE)
     , char * /* file */
@@ -296,7 +302,7 @@
 #endif
 );
 
-extern LockInfoPtr _Xglobal_lock;
+extern X11_IMPORT LockInfoPtr _Xglobal_lock;
 
 #if defined(XTHREADS_WARN) || defined(XTHREADS_FILE_LINE)
 #define LockDisplay(d)	     if ((d)->lock_fns) (*(d)->lock_fns->lock_display)((d),__FILE__,__LINE__)
diff -ru -x '*.Po' -x '*.Plo' libX11-1.2.2.orig/modules/im/ximcp/imLcLkup.c libX11-1.2.2/modules/im/ximcp/imLcLkup.c
--- libX11-1.2.2.orig/modules/im/ximcp/imLcLkup.c	2009-09-24 11:30:38 +0200
+++ libX11-1.2.2/modules/im/ximcp/imLcLkup.c	2009-09-24 17:07:22 +0200
@@ -45,6 +45,10 @@
 #include "XlcPubI.h"
 #include "Ximint.h"
 
+#ifdef _WIN32
+# include <alloca.h>
+#endif
+
 Public int
 _XimLocalMbLookupString(XIC xic, XKeyEvent *ev, char *buffer, int bytes,
 			KeySym *keysym, Status *status)
@@ -63,7 +67,11 @@
 	    ||(ic->private.local.brl_committed != 0))) {
 	if (ic->private.local.brl_committed != 0) { /* Braille Event */
 	    unsigned char pattern = ic->private.local.brl_committed;
+		#ifndef _WIN32
 	    char mb[XLC_PUBLIC(ic->core.im->core.lcd, mb_cur_max)];
+		#else
+		char* mb = (char*)alloca(XLC_PUBLIC(ic->core.im->core.lcd, mb_cur_max));
+		#endif
 	    ret = _Xlcwctomb(ic->core.im->core.lcd, mb, BRL_UC_ROW | pattern);
 	    if(ret > bytes) {
 		if(status) *status = XBufferOverflow;
diff -ru -x '*.Po' -x '*.Plo' libX11-1.2.2.orig/modules/im/ximcp/imTrans.c libX11-1.2.2/modules/im/ximcp/imTrans.c
--- libX11-1.2.2.orig/modules/im/ximcp/imTrans.c	2009-09-24 11:30:38 +0200
+++ libX11-1.2.2/modules/im/ximcp/imTrans.c	2009-09-24 17:07:22 +0200
@@ -33,6 +33,11 @@
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
+
+#ifdef WIN32
+#include <X11/Xwindows.h>
+#endif
+
 #include <stdio.h>
 #include <X11/Xatom.h>
 #include <X11/Xmd.h>
@@ -44,10 +49,6 @@
 #include "XimTrans.h"
 #include "XimTrInt.h"
 
-#ifdef WIN32
-#include <X11/Xwindows.h>
-#endif
-
 
 #ifndef XIM_CONNECTION_RETRIES
 #define XIM_CONNECTION_RETRIES 5
diff -ru -x '*.Po' -x '*.Plo' libX11-1.2.2.orig/src/Makefile.am libX11-1.2.2/src/Makefile.am
--- libX11-1.2.2.orig/src/Makefile.am	2009-09-24 11:30:39 +0200
+++ libX11-1.2.2/src/Makefile.am	2009-09-24 17:07:22 +0200
@@ -422,7 +422,7 @@
 KEYSYMDEF=@KEYSYMDEF@
 
 ks_tables.h: $(KEYSYMDEF) $(top_builddir)/src/util/makekeys$(EXEEXT)
-	$(top_builddir)/src/util/makekeys < $(KEYSYMDEF) > ks_tables_h
+	$(top_builddir)/src/util/makekeys$(EXEEXT) < $(KEYSYMDEF) > ks_tables_h
 	mv ks_tables_h $@
 
 $(top_builddir)/src/util/makekeys$(EXEEXT): force
diff -ru -x '*.Po' -x '*.Plo' libX11-1.2.2.orig/src/RdBitF.c libX11-1.2.2/src/RdBitF.c
--- libX11-1.2.2.orig/src/RdBitF.c	2009-09-24 11:30:39 +0200
+++ libX11-1.2.2/src/RdBitF.c	2009-09-24 17:07:22 +0200
@@ -56,6 +56,7 @@
 #define MAX_SIZE 255
 
 /* shared data for the image read/parse logic */
+#ifdef __GNUC__
 static const short hexTable[256] = {
     ['0'] = 0,  ['1'] = 1,
     ['2'] = 2,  ['3'] = 3,
@@ -73,6 +74,9 @@
     ['}'] = -1, ['\n'] = -1,
     ['\t'] = -1
 };
+#else
+static short hexTable[256] = { 0 };
+#endif
 
 /*
  *	read next hex value in the input stream, return -1 if EOF
@@ -88,6 +92,26 @@
 
     /* loop, accumulate hex value until find delimiter  */
     /* skip any initial delimiters found in read stream */
+	#ifndef __GNUC__
+	if(hexTable['1'] == 0) {
+		/* init for MS */
+		hexTable['0'] = 0;  hexTable['1'] = 1;
+		hexTable['2'] = 2;  hexTable['3'] = 3;
+		hexTable['4'] = 4;  hexTable['5'] = 5;
+		hexTable['6'] = 6;  hexTable['7'] = 7;
+		hexTable['8'] = 8;  hexTable['9'] = 9;
+		hexTable['A'] = 10; hexTable['B'] = 11;
+		hexTable['C'] = 12; hexTable['D'] = 13;
+		hexTable['E'] = 14; hexTable['F'] = 15;
+		hexTable['a'] = 10; hexTable['b'] = 11;
+		hexTable['c'] = 12; hexTable['d'] = 13;
+		hexTable['e'] = 14; hexTable['f'] = 15;
+
+		hexTable[' '] = -1; hexTable[','] = -1;
+		hexTable['}'] = -1; hexTable['\n'] = -1;
+		hexTable['\t'] = -1;
+	}
+	#endif
 
     while (!done) {
 	ch = getc(fstream);
diff -ru -x '*.Po' -x '*.Plo' libX11-1.2.2.orig/src/UIThrStubs.c libX11-1.2.2/src/UIThrStubs.c
--- libX11-1.2.2.orig/src/UIThrStubs.c	2009-09-24 11:30:39 +0200
+++ libX11-1.2.2/src/UIThrStubs.c	2009-09-24 17:07:22 +0200
@@ -85,6 +85,9 @@
     /*
      * Don't know what to do here. Is there something do be done at all?
      */
+#ifdef __PARITY__
+typedef long xthread_t;
+#endif
 #else /* !WIN32 */
 #ifdef USE_TIS_SUPPORT
 #include <tis.h>
diff -ru -x '*.Po' -x '*.Plo' libX11-1.2.2.orig/src/XlibInt.c libX11-1.2.2/src/XlibInt.c
--- libX11-1.2.2.orig/src/XlibInt.c	2009-09-24 11:30:39 +0200
+++ libX11-1.2.2/src/XlibInt.c	2009-09-24 17:08:00 +0200
@@ -42,6 +42,11 @@
 #include <config.h>
 #endif
 #include "Xlibint.h"
+
+#ifdef __PARITY__
+#pragma push_macro("min")
+#endif
+
 #include "Xprivate.h"
 #include <X11/Xpoll.h>
 #if !USE_XCB
@@ -52,11 +57,19 @@
 #include <stdio.h>
 #ifdef WIN32
 #include <direct.h>
+#ifdef __PARITY__
+# include <unistd.h>
+# include <stdlib.h>
+#endif
 #endif
 
 #ifdef XTHREADS
 #include "locking.h"
 
+#ifdef __PARITY__
+#pragma pop_macro("min")
+#endif
+
 /* these pointers get initialized by XInitThreads */
 LockInfoPtr _Xglobal_lock = NULL;
 void (*_XCreateMutex_fn)(LockInfoPtr) = NULL;
diff -ru -x '*.Po' -x '*.Plo' libX11-1.2.2.orig/src/util/Makefile.am libX11-1.2.2/src/util/Makefile.am
--- libX11-1.2.2.orig/src/util/Makefile.am	2009-09-24 11:30:39 +0200
+++ libX11-1.2.2/src/util/Makefile.am	2009-09-24 17:07:22 +0200
@@ -1,6 +1,6 @@
 # $XdotOrg: lib/X11/src/util/Makefile.am,v 1.4 2006-02-19 02:14:12 jamey Exp $
 
-noinst_PROGRAMS=makekeys
+noinst_PROGRAMS=makekeys$(EXEEXT)
 
 makekeys_CFLAGS=$(X11_CFLAGS)
 
