diff -ur libxspf-1.2.0.orig/test/OfficialSuite.cpp libxspf-1.2.0/test/OfficialSuite.cpp
--- libxspf-1.2.0.orig/test/OfficialSuite.cpp	2009-02-21 22:13:25.000000000 +0200
+++ libxspf-1.2.0/test/OfficialSuite.cpp	2009-05-10 22:39:16.000000000 +0300
@@ -21,6 +21,7 @@
  * Sebastian Pipping, sping@xiph.org
  */
 
+#include <cstdio>
 #include "OfficialSuite.h"
 #include <xspf/XspfReaderCallback.h>
 using namespace Xspf;
diff -ur libxspf-1.2.0.orig/test/ProjectOpus/ProjectOpusSuite.cpp libxspf-1.2.0/test/ProjectOpus/ProjectOpusSuite.cpp
--- libxspf-1.2.0.orig/test/ProjectOpus/ProjectOpusSuite.cpp	2009-02-21 22:13:25.000000000 +0200
+++ libxspf-1.2.0/test/ProjectOpus/ProjectOpusSuite.cpp	2009-05-10 22:37:16.000000000 +0300
@@ -21,6 +21,7 @@
  * Sebastian Pipping, sping@xiph.org
  */
 
+#include <cstdio>
 #include "ProjectOpusSuite.h"
 #include <xspf/XspfReader.h>
 #include <xspf/XspfTrack.h>
