diff -Naur soundtouch-1.3.1-orig/source/example/bpm/Makefile.am soundtouch-1.3.1/source/example/bpm/Makefile.am
--- soundtouch-1.3.1-orig/source/example/bpm/Makefile.am	2006-03-19 06:08:36.000000000 -0600
+++ soundtouch-1.3.1/source/example/bpm/Makefile.am	2007-02-24 11:52:13.000000000 -0600
@@ -33,7 +33,7 @@
 
 
 # other compiler flags to add
-AM_CXXFLAGS=-O3 -I../../../include
+AM_CXXFLAGS=-I../../../include
 
 # other linking flags to add
 #libBPM_la_LIBADD=
diff -Naur soundtouch-1.3.1-orig/source/example/SoundStretch/Makefile.am soundtouch-1.3.1/source/example/SoundStretch/Makefile.am
--- soundtouch-1.3.1-orig/source/example/SoundStretch/Makefile.am	2006-03-19 06:08:36.000000000 -0600
+++ soundtouch-1.3.1/source/example/SoundStretch/Makefile.am	2007-02-24 11:51:52.000000000 -0600
@@ -44,10 +44,10 @@
 soundstretch_LDADD=../../SoundTouch/libSoundTouch.la ../bpm/libBPM.la -lm
 
 ## linker flags
-soundstretch_LDFLAGS=-s
+soundstretch_LDFLAGS=
 
 ## additional compiler flags
-soundstretch_CXXFLAGS=-O3 -I../bpm
+soundstretch_CXXFLAGS=-I../bpm
 
 #clean-local: 
 #	-rm -f additional-files-to-remove-on-make-clean
diff -Naur soundtouch-1.3.1-orig/source/SoundTouch/Makefile.am soundtouch-1.3.1/source/SoundTouch/Makefile.am
--- soundtouch-1.3.1-orig/source/SoundTouch/Makefile.am	2006-03-19 06:08:36.000000000 -0600
+++ soundtouch-1.3.1/source/SoundTouch/Makefile.am	2007-02-24 11:50:21.000000000 -0600
@@ -35,7 +35,7 @@
 
 # ??? test for -fcheck-new in configure.ac
 # other compiler flags to add
-AM_CXXFLAGS=-O3 -msse -fcheck-new -I../../include
+AM_CXXFLAGS=-fcheck-new -I../../include
 
 # other linking flags to add
 #libSoundTouch_la_LIBADD=
