diff -Naur otp_src_R13A~orig/erts/emulator/Makefile.in otp_src_R13A/erts/emulator/Makefile.in
--- otp_src_R13A~orig/erts/emulator/Makefile.in	2009-04-04 14:25:34.000000000 +0200
+++ otp_src_R13A/erts/emulator/Makefile.in	2009-04-04 16:53:23.000000000 +0200
@@ -256,6 +256,7 @@
 else
 CS_CFLAGS = $(CS_CFLAGS_)
 endif
+CS_LDFLAGS = $(LDFLAGS)
 CS_LIBS = -L../lib/internal/$(TARGET) -lerts_internal @ERTS_INTERNAL_X_LIBS@
 
 LIBS	+= @TERMCAP_LIB@ -L../lib/internal/$(TARGET) @ERTS_INTERNAL_X_LIBS@
@@ -654,8 +655,8 @@
 CS_SRC = sys/$(ERLANG_OSTYPE)/erl_child_setup.c
 
 $(BINDIR)/$(CS_EXECUTABLE): $(CS_SRC)
-	$(CS_PURIFY) $(CC) -o $(BINDIR)/$(CS_EXECUTABLE) \
-	$(CS_CFLAGS) $(COMMON_INCLUDES) $(CS_SRC) $(CS_LIBS)
+	$(CS_PURIFY) $(CC) $(CS_LDFLAGS) -o $(BINDIR)/$(CS_EXECUTABLE) \
+		$(CS_CFLAGS) $(COMMON_INCLUDES) $(CS_SRC) $(CS_LIBS)
 
 $(OBJDIR)/%.elib.o: beam/%.c
 	$(CC) $(ELIB_FLAGS) $(subst -O2, $(GEN_OPT_FLGS), $(CFLAGS)) $(INCLUDES) -c $< -o $@
diff -Naur otp_src_R13A~orig/lib/asn1/c_src/Makefile.in otp_src_R13A/lib/asn1/c_src/Makefile.in
--- otp_src_R13A~orig/lib/asn1/c_src/Makefile.in	2009-04-04 14:25:15.000000000 +0200
+++ otp_src_R13A/lib/asn1/c_src/Makefile.in	2009-04-04 17:48:14.000000000 +0200
@@ -60,7 +60,7 @@
 DRIVER_INCLUDES = -I$(ERL_TOP)/erts/emulator/beam \
 	-I$(ERL_TOP)/erts/emulator/sys/$(ERLANG_OSTYPE)
 CFLAGS = $(DRIVER_INCLUDES) $(EI_INCLUDES) @DED_CFLAGS@
-LDFLAGS = @DED_LDFLAGS@
+LDFLAGS += @DED_LDFLAGS@
 
 LD_INCL_EI = -L$(EI_LIBDIR)
 
diff -Naur otp_src_R13A~orig/lib/crypto/c_src/Makefile.in otp_src_R13A/lib/crypto/c_src/Makefile.in
--- otp_src_R13A~orig/lib/crypto/c_src/Makefile.in	2009-04-04 14:25:16.000000000 +0200
+++ otp_src_R13A/lib/crypto/c_src/Makefile.in	2009-04-04 17:48:32.000000000 +0200
@@ -33,7 +33,7 @@
 LD = @DED_LD@
 SHELL = /bin/sh
 LIBS = @LIBS@
-LDFLAGS = @DED_LDFLAGS@
+LDFLAGS += @DED_LDFLAGS@
 CFLAGS = @DED_CFLAGS@
 ERLANG_OSTYPE = @ERLANG_OSTYPE@
 TARGET = @host@
diff -Naur otp_src_R13A~orig/lib/erl_interface/src/Makefile.in otp_src_R13A/lib/erl_interface/src/Makefile.in
--- otp_src_R13A~orig/lib/erl_interface/src/Makefile.in	2009-04-04 14:25:16.000000000 +0200
+++ otp_src_R13A/lib/erl_interface/src/Makefile.in	2009-04-04 16:41:46.000000000 +0200
@@ -715,11 +715,11 @@
 else
 ifdef THR_DEFS
 $(ERL_CALL): $(ERLCALL) ../include/ei.h $(MT_EILIB)
-	$(PURIFY) $(CC) $(PROG_CFLAGS) $(THR_DEFS) -o $@ $(ERLCALL) \
+	$(PURIFY) $(CC) $(PROG_CFLAGS) $(THR_DEFS) $(LDFLAGS) -o $@ $(ERLCALL) \
 		-L$(OBJDIR) -lei $(THR_LIBS) $(LIBS)
 else
 $(ERL_CALL): $(ERLCALL) ../include/ei.h $(ST_EILIB)
-	$(PURIFY) $(CC) $(PROG_CFLAGS) -o $@ $(ERLCALL) \
+	$(PURIFY) $(CC) $(PROG_CFLAGS) $(LDFLAGS) -o $@ $(ERLCALL) \
 		-L$(OBJDIR) -lei $(LIBS)
 endif
 endif
diff -Naur otp_src_R13A~orig/lib/megaco/src/flex/Makefile.in otp_src_R13A/lib/megaco/src/flex/Makefile.in
--- otp_src_R13A~orig/lib/megaco/src/flex/Makefile.in	2009-04-04 14:25:16.000000000 +0200
+++ otp_src_R13A/lib/megaco/src/flex/Makefile.in	2009-04-04 17:57:24.000000000 +0200
@@ -48,7 +48,7 @@
 CC      = @CC@
 CFLAGS_MT = $(CFLAGS) -D_THREAD_SAFE -D_REENTRANT
 LD      = @DED_LD@
-LDFLAGS = @DED_LDFLAGS@
+LDFLAGS += @DED_LDFLAGS@
 LEX     = @LEX@
 LEXLIB  = @LEXLIB@
 PERL    = @PERL@
diff -Naur otp_src_R13A~orig/lib/runtime_tools/c_src/Makefile.in otp_src_R13A/lib/runtime_tools/c_src/Makefile.in
--- otp_src_R13A~orig/lib/runtime_tools/c_src/Makefile.in	2009-04-04 14:25:34.000000000 +0200
+++ otp_src_R13A/lib/runtime_tools/c_src/Makefile.in	2009-04-04 16:23:59.000000000 +0200
@@ -35,7 +35,7 @@
 LD = @DED_LD@
 SHELL = /bin/sh
 LIBS = @LIBS@
-LDFLAGS = @DED_LDFLAGS@
+LDFLAGS += @DED_LDFLAGS@
 ERLANG_OSTYPE = @ERLANG_OSTYPE@
 
 SYSINCLUDE = 	-I$(ERL_TOP)/erts/emulator/beam \
diff -Naur otp_src_R13A~orig/lib/ssl/c_src/Makefile.in otp_src_R13A/lib/ssl/c_src/Makefile.in
--- otp_src_R13A~orig/lib/ssl/c_src/Makefile.in	2009-04-04 14:25:15.000000000 +0200
+++ otp_src_R13A/lib/ssl/c_src/Makefile.in	2009-04-04 16:33:37.000000000 +0200
@@ -163,7 +163,7 @@
 
 # Unix
 $(BINDIR)/ssl_esock: $(OBJS)
-	$(CC) $(CC_R_OPT) $(PLAIN_CFLAGS) -o $@ $^ $(LIBS) $(SSL_LINK_LIB)
+	$(CC) $(CC_R_OPT) $(PLAIN_CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS) $(SSL_LINK_LIB)
 
 # Win32/Cygwin
 $(BINDIR)/ssl_esock.exe: $(OBJS)
