diff -urN otp_src_R11B-5/lib/odbc/c_src/Makefile.in otp_src_R11B-5.odbc/lib/odbc/c_src/Makefile.in
--- otp_src_R11B-5/lib/odbc/c_src/Makefile.in	2006-05-03 11:18:45.000000000 +0300
+++ otp_src_R11B-5.odbc/lib/odbc/c_src/Makefile.in	2007-11-07 00:24:36.000000000 +0200
@@ -57,11 +57,11 @@
 WIN32_TARGET = $(WIN_BIN_DIR)/odbcserver.exe
 EXE_TARGET = $(WIN32_TARGET)
 else
-ifneq ($(BITS64),yes)
+#ifneq ($(BITS64),yes)
 EI_LIB = -lerl_interface -lei
 UNIX_TARGET = $(BIN_DIR)/odbcserver
 EXE_TARGET = $(UNIX_TARGET)
-endif
+#endif
 endif
 
 C_FILES = odbcserver.c 
