diff -Naur otp_src_R11B-4.orig/lib/erl_interface/src/connect/ei_resolve.c otp_src_R11B-4/lib/erl_interface/src/connect/ei_resolve.c
--- otp_src_R11B-4.orig/lib/erl_interface/src/connect/ei_resolve.c	2006-11-06 14:53:29 +0100
+++ otp_src_R11B-4/lib/erl_interface/src/connect/ei_resolve.c	2007-06-13 23:14:06 +0200
@@ -44,6 +44,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
+#include <sys/param.h>
 #endif 
 
 /* common to all platforms */
@@ -614,7 +615,7 @@
 #ifndef HAVE_GETHOSTBYNAME_R
   return my_gethostbyname_r(name,hostp,buffer,buflen,h_errnop);
 #else
-#ifdef __GLIBC__
+#if (defined(__GLIBC__) || (__FreeBSD_version >= 602000))
   struct hostent *result;
 
   gethostbyname_r(name, hostp, buffer, buflen, &result, h_errnop);
