respect the configure nls flag

http://bugs.gentoo.org/192819


--- dpkg-1.15.1/man/Makefile.in.orig	2009-05-21 06:49:55.000000000 +0200
+++ dpkg-1.15.1/man/Makefile.in	2009-05-21 22:42:28.000000000 +0200
@@ -197,7 +197,11 @@
 top_srcdir = @top_srcdir@
 
 # Extract the list of languages from the po4a config file.
+ifeq ($(USE_NLS),yes)
 LINGUAS = `sed -ne 's/^.*\[po4a_langs\] \(.*\)$$/\1/p' $(srcdir)/po/po4a.cfg`
+else
+LINGUAS =
+endif
 dist_man_MANS = deb-control.5 deb-version.5 deb-shlibs.5 \
 	deb-substvars.5 deb-symbols.5 deb-triggers.5 deb-old.5 \
 	deb-override.5 deb.5 dpkg-architecture.1 dpkg-buildpackage.1 \
