diff -Naurp dpkg-1.15.5.2.orig/m4/dpkg-libs.m4 dpkg-1.15.5.2/m4/dpkg-libs.m4
--- dpkg-1.15.5.2.orig/m4/dpkg-libs.m4	2009-11-21 09:40:57.000000000 +0000
+++ dpkg-1.15.5.2/m4/dpkg-libs.m4	2009-11-21 09:37:55.000000000 +0000
@@ -100,14 +100,23 @@ fi
 # DPKG_LIB_CURSES
 # ---------------
 # Check for curses library.
-AC_DEFUN([DPKG_LIB_CURSES],
-[AC_ARG_VAR([CURSES_LIBS], [linker flags for curses library])dnl
-AC_CHECK_HEADERS([ncursesw/ncurses.h ncurses/ncurses.h ncurses.h curses.h \
-                  ncursesw/term.h ncurses/term.h term.h])
-AC_CHECK_LIB([ncursesw], [initscr], [CURSES_LIBS="${CURSES_LIBS:+$CURSES_LIBS }-lncursesw"],
-	[AC_CHECK_LIB([ncurses], [initscr], [CURSES_LIBS="${CURSES_LIBS:+$CURSES_LIBS }-lncurses"],
+AC_DEFUN([DPKG_LIB_CURSES], [
+AC_REQUIRE([AM_UNICODE])
+AC_ARG_VAR([CURSES_LIBS], [linker flags for curses library])dnl
+if test "x$USE_UNICODE" = "xyes"; then
+	AC_CHECK_HEADERS([ncursesw/ncurses.h ncurses/ncurses.h ncurses.h curses.h \
+	                  ncursesw/term.h ncurses/term.h term.h])
+	AC_CHECK_LIB([ncursesw], [initscr], [CURSES_LIBS="${CURSES_LIBS:+$CURSES_LIBS }-lncursesw"],
+		[AC_CHECK_LIB([ncurses], [initscr], [CURSES_LIBS="${CURSES_LIBS:+$CURSES_LIBS }-lncurses"],
+			[AC_CHECK_LIB([curses], [initscr], [CURSES_LIBS="${CURSES_LIBS:+$CURSES_LIBS }-lcurses"],
+				[AC_MSG_WARN([no curses library found])])])])
+else
+	AC_CHECK_HEADERS([ncurses/ncurses.h ncurses.h curses.h \
+	                  ncurses/term.h term.h])
+	AC_CHECK_LIB([ncurses], [initscr], [CURSES_LIBS="${CURSES_LIBS:+$CURSES_LIBS }-lncurses"],
 		[AC_CHECK_LIB([curses], [initscr], [CURSES_LIBS="${CURSES_LIBS:+$CURSES_LIBS }-lcurses"],
-			[AC_MSG_WARN([no curses library found])])])])
+			[AC_MSG_WARN([no curses library found])])])
+fi
 ])# DPKG_LIB_CURSES
 
 # DPKG_LIB_SSD
diff -Naurp dpkg-1.15.5.2.orig/m4/unicode.m4 dpkg-1.15.5.2/m4/unicode.m4
--- dpkg-1.15.5.2.orig/m4/unicode.m4	1970-01-01 00:00:00.000000000 +0000
+++ dpkg-1.15.5.2/m4/unicode.m4	2009-11-21 09:46:53.000000000 +0000
@@ -0,0 +1,21 @@
+# unicode.m4
+dnl Copyright (C) 2009 Free Software Foundation, Inc.
+dnl This file is free software; the Free Software Foundation
+dnl gives unlimited permission to copy and/or distribute it,
+dnl with or without modifications, as long as this notice is preserved.
+dnl
+dnl Author:
+dnl   Yuri Vasilevski <yvasilev@gentoo.org>, 2009
+
+AC_PREREQ(2.50)
+
+AC_DEFUN([AM_UNICODE],
+[
+  AC_MSG_CHECKING([whether UNICODE is requested])
+  dnl Default: UNICODE is enabled
+  AC_ARG_ENABLE(unicode,
+    [  --disable-unicode       do not use Unicode (wide characters) support],
+    USE_UNICODE=$enableval, USE_UNICODE=yes)
+  AC_MSG_RESULT($USE_UNICODE)
+  AC_SUBST(USE_UNICODE)
+])
