fixes from upstream

http://bugs.gentoo.org/150579

--- beecrypt/c++/security/Security.h
+++ beecrypt/c++/security/Security.h
@@ -59,16 +59,16 @@
 		{
 			friend class AlgorithmParameterGenerator;
 			friend class AlgorithmParameters;
-			friend class CertificateFactory;
-			friend class CertPathValidator;
-			friend class Cipher;
-			friend class KeyAgreement;
+			friend class ::CertificateFactory;
+			friend class ::CertPathValidator;
+			friend class ::Cipher;
+			friend class ::KeyAgreement;
 			friend class KeyFactory;
 			friend class KeyPairGenerator;
 			friend class KeyStore;
-			friend class Mac;
+			friend class ::Mac;
 			friend class MessageDigest;
-			friend class SecretKeyFactory;
+			friend class ::SecretKeyFactory;
 			friend class SecureRandom;
 			friend class Signature;
 
--- beecrypt/c++/beeyond/BeeCertificate.h
+++ beecrypt/c++/beeyond/BeeCertificate.h
@@ -61,7 +61,7 @@
 		 */
 		class BEECRYPTCXXAPI BeeCertificate : public beecrypt::security::cert::Certificate, public beecrypt::lang::Cloneable
 		{
-			friend class BeeCertificateFactory;
+			friend class ::BeeCertificateFactory;
 
 		public:
 			static const Date FOREVER;
