diff -Nurp quilt-0.46.orig/Makefile.in quilt-0.46/Makefile.in
--- quilt-0.46.orig/Makefile.in	2006-10-11 14:36:10.000000000 +0200
+++ quilt-0.46/Makefile.in	2006-12-27 15:26:38.000000000 +0100
@@ -81,8 +81,7 @@ QUILT :=	$(QUILT_IN)
 SRC +=		$(QUILT_SRC:%=quilt/%)
 DIRT +=		$(QUILT_IN:%=quilt/%)
 
-SCRIPTS_IN :=	patchfns parse-patch inspect dependency-graph edmail	\
-		remove-trailing-ws
+SCRIPTS_IN :=	patchfns parse-patch inspect edmail remove-trailing-ws
 
 SCRIPTS_SRC :=	$(SCRIPTS_IN:%=%.in)
 SCRIPTS :=	$(SCRIPTS_IN)
diff -Nurp quilt-0.46.orig/bash_completion quilt-0.46/bash_completion
--- quilt-0.46.orig/bash_completion	2006-02-03 08:38:15.000000000 +0100
+++ quilt-0.46/bash_completion	2006-12-27 15:26:38.000000000 +0100
@@ -95,7 +95,7 @@ _quilt_completion()
     prev=${COMP_WORDS[COMP_CWORD-1]}
 
     # quilt sub commands 
-    cmds='add annotate applied delete diff edit files fold fork graph \
+    cmds='add annotate applied delete diff edit files fold fork \
           grep header import mail new next patches pop previous push refresh \
 	  remove rename series setup snapshot top unapplied'
 
@@ -188,19 +188,6 @@ _quilt_completion()
 	        ;;
 	   esac
 	   ;;
-	graph)
-	   case $prev in
-	     -T)
-	     	COMPREPLY=( $( compgen -W "ps" -- $cur ) )
-		;;
-	     --edge-labels)
-	        COMPREPLY=( $( compgen -W "files" -- $cur ) )
-		;;
-	     *)
-		COMPREPLY=( $( compgen -W "-T -h --all --reduce --lines --edge-labels $(quilt applied)" -- $cur ) )
-		;;
-	   esac
-	   ;;
 	grep)
 	   _longopt grep
 	   COMPREPLY=( ${COMPREPLY[@]:-} $( compgen -W "-h" -- $cur ) )
