--- quilt-0.47/Makefile.in.orig	2008-11-17 14:53:21.000000000 -0500
+++ quilt-0.47/Makefile.in	2008-11-17 14:53:49.000000000 -0500
@@ -82,8 +82,7 @@ QUILT :=	$(QUILT_IN)
 SRC +=		$(QUILT_SRC:%=quilt/%)
 DIRT +=		$(QUILT_IN:%=quilt/%)
 
-SCRIPTS_IN :=	patchfns parse-patch inspect dependency-graph edmail	\
-		remove-trailing-ws
+SCRIPTS_IN :=	patchfns parse-patch inspect edmail	remove-trailing-ws
 
 SCRIPTS_SRC :=	$(SCRIPTS_IN:%=%.in)
 SCRIPTS :=	$(SCRIPTS_IN)
--- quilt-0.47/bash_completion.orig	2008-11-17 14:56:25.000000000 -0500
+++ quilt-0.47/bash_completion	2008-11-17 15:02:50.000000000 -0500
@@ -95,7 +95,7 @@ _quilt_completion()
     prev=${COMP_WORDS[COMP_CWORD-1]}
 
     # quilt sub commands 
-    cmds='add annotate applied delete diff edit files fold fork graph \
+    cmds='add annotate applied delete diff edit files fold fork \
           grep header import mail new next patches pop previous push refresh \
 	  remove rename series setup snapshot top unapplied'
 
@@ -188,19 +188,6 @@ _quilt_completion()
 	        ;;
 	   esac
 	   ;;
-	graph)
-	   case $prev in
-	     -T)
-	     	COMPREPLY=( $( compgen -W "ps" -- $cur ) )
-		;;
-	     --edge-labels)
-	        COMPREPLY=( $( compgen -W "files" -- $cur ) )
-		;;
-	     *)
-		COMPREPLY=( $( compgen -W "-T -h --all --reduce --lines --edge-labels $(quilt applied)" -- $cur ) )
-		;;
-	   esac
-	   ;;
 	grep)
 	   _longopt grep
 	   COMPREPLY=( ${COMPREPLY[@]:-} $( compgen -W "-h" -- $cur ) )
