http://bugs.gentoo.org/show_bug.cgi?id=211499

--- gnu/getopt.c.dist	2009-04-20 13:03:00.742170400 +0100
+++ gnu/getopt.c	2009-04-20 13:03:57.935443360 +0100
@@ -69,12 +69,7 @@
 #include <unistd.h>
 #endif	/* GNU C library.  */
 
-#ifdef VMS
-#include <unixlib.h>
-#if HAVE_STRING_H - 0
 #include <string.h>
-#endif
-#endif
 
 #if defined (WIN32) && !defined (__CYGWIN32__)
 /* It's not Unix, really.  See?  Capital letters.  */
