diff -ur spim-7.5-orig/spim/Makefile spim-7.5/spim/Makefile
--- spim-7.5-orig/spim/Makefile	2009-08-15 06:16:36.000000000 +0300
+++ spim-7.5/spim/Makefile	2009-12-22 21:35:38.679967575 +0200
@@ -54,13 +54,13 @@
 
 
 # Full path for the directory that will hold the executable files:
-BIN_DIR = /usr/bin
+BIN_DIR = $(DESTDIR)/usr/bin
 
 # Full path for the directory that will hold the exception handler:
-EXCEPTION_DIR = /usr/lib/spim
+EXCEPTION_DIR = $(DESTDIR)/var/lib/spim
 
 # Full path for the directory that will hold the man files:
-MAN_DIR = /usr/share/man/en
+MAN_DIR = $(DESTDIR)/usr/share/man
 
 
 # If you have flex, use it instead of lex.  If you use flex, define this
@@ -92,11 +92,9 @@
 
 DEFINES = $(ENDIAN) $(MEM_SIZES) -DDEFAULT_EXCEPTION_HANDLER="\"$(EXCEPTION_DIR)/exceptions.s\"" -DSPIM_VERSION="\"`cat ../VERSION`\""
 
-CC = gcc
-CFLAGS = -I. -I$(CPU_DIR) $(DEFINES) -g -Wall
+CFLAGS += -I. -I$(CPU_DIR) $(DEFINES) -Wall
 YFLAGS = -d --file-prefix=y
 YCFLAGS =
-LDFLAGS = -lm
 CSH = bash
 
 # lex.yy.c is usually compiled with -O to speed it up.
@@ -110,10 +108,10 @@
 
 
 spim:   force
-	make -f Makefile spim2
+	$(MAKE) -f Makefile spim2
 
 spim2:	$(OBJS) spim.o
-	$(CC) -g $(OBJS) spim.o $(LDFLAGS) -o spim -lm
+	$(CC) $(OBJS) spim.o $(LDFLAGS) -o spim -lm
 
 
 force:	configuration
@@ -131,19 +129,19 @@
 	@echo
 	@echo "Testing tt.bare.s:"
 	$(CSH) -c "./spim -delayed_branches -delayed_loads -noexception -file $(TEST_DIR)/tt.bare.s >& test.out"
-	@tail -2 test.out
+	@tail -n2 test.out
 	@echo
 
 	@echo
 	@echo "Testing tt.core.s:"
 	$(CSH) -c "./spim -ef $(CPU_DIR)/exceptions.s -file $(TEST_DIR)/tt.core.s < $(TEST_DIR)/tt.in >& test.out"
-	@tail -2 test.out
+	@tail -n2 test.out
 	@echo
 
 	@echo
 	@echo "Testing tt.endian.s:"
 	$(CSH) -c "./spim -ef $(CPU_DIR)/exceptions.s -file $(TEST_DIR)/tt.endian.s  >& test.out"
-	@tail -2 test.out
+	@tail -n2 test.out
 	@echo
 	@echo
 
@@ -154,13 +152,13 @@
 	@echo
 	@echo "Testing tt.alu.bare.s:"
 	$(CSH) -c "./spim -bare -noexception -file $(TEST_DIR)/tt.alu.bare.s >& test.out"
-	@tail -2 test.out
+	@tail -n2 test.out
 	@echo
 
 	@echo
 	@echo "Testing tt.fpt.bare.s:"
 	$(CSH) -c "./spim -bare -noexception -file $(TEST_DIR)/tt.fpu.bare.s >& test.out"
-	@tail -2 test.out
+	@tail -n2 test.out
 	@echo
 	@echo
 
@@ -178,7 +176,6 @@
 	install -D -m 0444 $(CPU_DIR)/exceptions.s $(EXCEPTION_DIR)/exceptions.s
 
 install-man:
-	install -D -m 0444 $(DOC_DIR)/spim.man $(MAN_DIR)
 
 very-clean: clean
 	rm -f configuration
diff -ur spim-7.5-orig/xspim/Makefile spim-7.5/xspim/Makefile
--- spim-7.5-orig/xspim/Makefile	2009-08-15 06:16:37.000000000 +0300
+++ spim-7.5/xspim/Makefile	2009-12-22 21:36:52.583871429 +0200
@@ -175,8 +175,7 @@
 # from  top Makefile
   BOOTSTRAPCFLAGS =
 
-               CC = gcc -m32
-               AS = gcc -m32 -c -x assembler
+               AS = $(CC) -m32 -c -x assembler
 
 .SUFFIXES: .cc
 
@@ -201,12 +200,12 @@
 
               CPP = cpp $(STD_CPP_DEFINES)
            RAWCPP = cpp -undef $(STD_CPP_OPTIONS)
-    PREPROCESSCMD = gcc -m32 -E $(STD_CPP_DEFINES)
+    PREPROCESSCMD = $(CC) -m32 -E $(STD_CPP_DEFINES)
 
           INSTALL = install
      INSTALLFLAGS = -c
 
-               LD = gcc -m32 -nostdlib
+               LD = $(CC) -m32 -nostdlib
 
               LEX = flex -l
                M4 = m4
@@ -220,7 +219,6 @@
       LINTLIBFLAG = -C
          LINTOPTS = -axz
                LN = ln -s
-             MAKE = make
                MV = mv -f
                CP = cp
 
@@ -270,14 +268,14 @@
               COL = col
          COLFLAGS = -b
 
-            MODCC = gcc -m32
+            MODCC = $(CC) -m32
 
            MODCPP = cpp
         MODCFLAGS = $(CFLAGS)
-            MODAS = gcc -m32 -c -x assembler
+            MODAS = $(CC) -m32 -c -x assembler
        MODASFLAGS =
 
-            MODLD = gcc -m32 -nostdlib
+            MODLD = $(CC) -m32 -nostdlib
 
        MODLDFLAGS =
 MODLDCOMBINEFLAGS = -r
@@ -330,7 +328,7 @@
 
       ALLINCLUDES = $(INCLUDES) $(EXTRA_INCLUDES) $(TOP_INCLUDES) $(INSTALLED_INCLUDES) $(STD_INCLUDES)
        ALLDEFINES = $(ALLINCLUDES) $(STD_DEFINES) $(PROTO_DEFINES) $(THREADS_DEFINES) $(MODULE_DEFINES) $(DEFINES) $(EXTRA_DEFINES)
-           CFLAGS = $(CDEBUGFLAGS) $(CCOPTIONS) $(THREADS_CFLAGS) $(MODULE_CFLAGS) $(ALLDEFINES)
+           CFLAGS += $(MODULE_CFLAGS) $(ALLDEFINES)
         LINTFLAGS = $(LINTOPTS) -DLINT $(ALLDEFINES) $(DEPEND_DEFINES)
          LDPRELIB =  $(INSTALLED_LIBS)
         LDPOSTLIB =
@@ -339,7 +337,7 @@
 
            LDLIBS = $(LDPOSTLIBS) $(THREADS_LIBS) $(SYS_LIBRARIES) $(EXTRA_LIBRARIES)
 
-           CCLINK = $(CC)
+           CCLINK = $(CC) $(LDFLAGS)
 
           CXXLINK = $(CXX)
 
@@ -1079,10 +1077,10 @@
 BIN_DIR = /usr/bin
 
 # Full path for the directory that will hold the exception handler:
-EXCEPTION_DIR = /usr/lib/spim
+EXCEPTION_DIR = /var/lib/spim
 
 # Full path for the directory that will hold the man files:
-MAN_DIR = /usr/share/man/en
+MAN_DIR = /usr/share/man
 
 # If you have flex, use it instead of lex.  If you use flex, define this
 # variable and set LEXFLAGS.
