diff -ru -x '*.Po' -x '*.Plo' gst-plugins-base-0.10.19.orig/gst-libs/gst/sdp/gstsdpmessage.c gst-plugins-base-0.10.19/gst-libs/gst/sdp/gstsdpmessage.c
--- gst-plugins-base-0.10.19.orig/gst-libs/gst/sdp/gstsdpmessage.c	2008-04-17 10:20:10 +0200
+++ gst-plugins-base-0.10.19/gst-libs/gst/sdp/gstsdpmessage.c	2008-04-17 10:30:45 +0200
@@ -320,6 +320,10 @@
 static gboolean
 is_multicast_address (const gchar * host_name, guint * family)
 {
+#ifdef __INTERIX
+  /* only applies to interix 3.5 and 5.2, 6.0 has addinfo stuff! */
+  return FALSE;
+#else
   struct addrinfo hints;
   struct addrinfo *ai;
   struct addrinfo *res;
@@ -350,6 +354,7 @@
   freeaddrinfo (res);
 
   return ret;
+#endif
 }
 
 /**
