diff -ru -x '*.Po' -x '*.Plo' gst-plugins-base-0.10.17.orig/gst/tcp/gstfdset.c gst-plugins-base-0.10.17/gst/tcp/gstfdset.c
--- gst-plugins-base-0.10.17.orig/gst/tcp/gstfdset.c	2008-03-17 13:46:29 +0100
+++ gst-plugins-base-0.10.17/gst/tcp/gstfdset.c	2008-03-17 14:00:49 +0100
@@ -23,7 +23,13 @@
 #define MIN_POLLFDS     32
 #define INIT_POLLFDS    MIN_POLLFDS
 
+// CAREFULL: on interix, poll() works only on /proc/*
+#if defined(__INTERIX)
+#include <poll.h>
+#else
 #include <sys/poll.h>
+#endif
+
 #include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
diff -ru -x '*.Po' -x '*.Plo' gst-plugins-base-0.10.17.orig/gst/tcp/gsttcp.c gst-plugins-base-0.10.17/gst/tcp/gsttcp.c
--- gst-plugins-base-0.10.17.orig/gst/tcp/gsttcp.c	2008-03-17 13:46:29 +0100
+++ gst-plugins-base-0.10.17/gst/tcp/gsttcp.c	2008-03-17 13:56:48 +0100
@@ -37,6 +37,10 @@
 #include <sys/filio.h>
 #endif
 
+#if defined(__INTERIX)
+# include <sys/time.h>
+#endif
+
 #include "gsttcp.h"
 #include <gst/gst-i18n-plugin.h>
 
diff -ru -x '*.Po' -x '*.Plo' gst-plugins-base-0.10.17.orig/gst/tcp/gsttcpserversrc.c gst-plugins-base-0.10.17/gst/tcp/gsttcpserversrc.c
--- gst-plugins-base-0.10.17.orig/gst/tcp/gsttcpserversrc.c	2008-03-17 13:46:29 +0100
+++ gst-plugins-base-0.10.17/gst/tcp/gsttcpserversrc.c	2008-03-17 14:01:41 +0100
@@ -31,6 +31,10 @@
 #include <sys/ioctl.h>
 #include <fcntl.h>
 
+#if defined(__INTERIX)
+# include <sys/time.h>
+#endif
+
 
 /* control stuff stolen from fdsrc */
 #define CONTROL_STOP            'S'     /* stop the select call */
diff -ru -x '*.Po' -x '*.Plo' gst-plugins-base-0.10.17.orig/gst-libs/gst/rtsp/gstrtspconnection.c gst-plugins-base-0.10.17/gst-libs/gst/rtsp/gstrtspconnection.c
--- gst-plugins-base-0.10.17.orig/gst-libs/gst/rtsp/gstrtspconnection.c	2008-03-17 13:46:30 +0100
+++ gst-plugins-base-0.10.17/gst-libs/gst/rtsp/gstrtspconnection.c	2008-03-17 13:48:26 +0100
@@ -87,6 +87,10 @@
 #include <sys/filio.h>
 #endif
 
+#if defined(__INTERIX)
+# include <sys/time.h>
+#endif
+
 #include "gstrtspconnection.h"
 #include "gstrtspbase64.h"
 
@@ -229,6 +233,11 @@
   unsigned long flags;
   struct in_addr *addrp;
 #else
+
+#if defined(__INTERIX) && !defined(INET_ADDRSTRLEN)
+# define INET_ADDRSTRLEN 1024 // sufficuent space...
+#endif
+
   char **addrs;
   gchar ipbuf[INET_ADDRSTRLEN];
 #endif /* G_OS_WIN32 */
diff -ru -x '*.Po' -x '*.Plo' gst-plugins-base-0.10.17.orig/gst-libs/gst/sdp/gstsdpmessage.c gst-plugins-base-0.10.17/gst-libs/gst/sdp/gstsdpmessage.c
--- gst-plugins-base-0.10.17.orig/gst-libs/gst/sdp/gstsdpmessage.c	2008-03-17 13:46:30 +0100
+++ gst-plugins-base-0.10.17/gst-libs/gst/sdp/gstsdpmessage.c	2008-03-17 13:50:52 +0100
@@ -319,6 +319,9 @@
 static gboolean
 is_multicast_address (const gchar * host_name, guint * family)
 {
+#if defined(__INTERIX)
+  return FALSE;
+#else
   struct addrinfo hints;
   struct addrinfo *ai;
   struct addrinfo *res;
@@ -348,6 +351,7 @@
   freeaddrinfo (res);
 
   return ret;
+#endif
 }
 
 /**
