diff -ru -x '*.Po' -x '*.Plo' gst-plugins-base-0.10.20.orig/gst/tcp/gstmultifdsink.c gst-plugins-base-0.10.20/gst/tcp/gstmultifdsink.c
--- gst-plugins-base-0.10.20.orig/gst/tcp/gstmultifdsink.c	2008-07-30 08:20:45 +0200
+++ gst-plugins-base-0.10.20/gst/tcp/gstmultifdsink.c	2008-07-30 09:16:46 +0200
@@ -712,6 +712,7 @@
 static gint
 setup_dscp_client (GstMultiFdSink * sink, GstTCPClient * client)
 {
+#ifndef __INTERIX
   gint tos;
   gint ret;
   struct sockaddr_storage ssaddr;
@@ -765,6 +766,10 @@
     GST_DEBUG_OBJECT (sink, "could not set DSCP: %s", g_strerror (errno));
 
   return ret;
+#else
+  GST_ERROR_OBJECT(sink, "not supported on interix systems!");
+  return 0;
+#endif
 }
 
 
diff -ru -x '*.Po' -x '*.Plo' gst-plugins-base-0.10.20.orig/gst/tcp/gsttcpserversrc.h gst-plugins-base-0.10.20/gst/tcp/gsttcpserversrc.h
--- gst-plugins-base-0.10.20.orig/gst/tcp/gsttcpserversrc.h	2008-07-30 08:20:45 +0200
+++ gst-plugins-base-0.10.20/gst/tcp/gsttcpserversrc.h	2008-07-30 09:00:23 +0200
@@ -69,7 +69,7 @@
 
   /* client information */
   struct sockaddr_in client_sin;
-  socklen_t client_sin_len;
+  int client_sin_len;
   GstPollFD client_sock_fd;
 
   GstPoll *fdset;
diff -ru -x '*.Po' -x '*.Plo' gst-plugins-base-0.10.20.orig/gst-libs/gst/rtsp/gstrtspconnection.c gst-plugins-base-0.10.20/gst-libs/gst/rtsp/gstrtspconnection.c
--- gst-plugins-base-0.10.20.orig/gst-libs/gst/rtsp/gstrtspconnection.c	2008-07-30 08:20:51 +0200
+++ gst-plugins-base-0.10.20/gst-libs/gst/rtsp/gstrtspconnection.c	2008-07-30 08:49:28 +0200
@@ -1558,6 +1558,7 @@
 GstRTSPResult
 gst_rtsp_connection_set_qos_dscp (GstRTSPConnection * conn, guint qos_dscp)
 {
+#ifndef __INTERIX
   struct sockaddr_storage sa_s;
   socklen_t sa_sl = sizeof (sa_s);
   gint af;
@@ -1600,6 +1601,7 @@
 
   return GST_RTSP_OK;
 
+#endif
   /* ERRORS */
 no_getsockname:
 no_setsockopt:
