diff -ru -x '*.Po' -x '*.Plo' gst-plugins-base-0.10.19.orig/gst-libs/gst/rtsp/gstrtspconnection.c gst-plugins-base-0.10.19/gst-libs/gst/rtsp/gstrtspconnection.c
--- gst-plugins-base-0.10.19.orig/gst-libs/gst/rtsp/gstrtspconnection.c	2008-04-17 10:20:10 +0200
+++ gst-plugins-base-0.10.19/gst-libs/gst/rtsp/gstrtspconnection.c	2008-04-17 10:27:57 +0200
@@ -202,6 +202,10 @@
   GstClockTime to;
   gint retval;
 
+#if defined(__INTERIX) && !defined(INET_ADDRSTRLEN)
+# define INET_ADDRSTRLEN 1024
+#endif
+
 #ifdef G_OS_WIN32
   unsigned long flags = 1;
   struct in_addr *addrp;
@@ -230,6 +234,10 @@
     addrp = (struct in_addr *) hostinfo->h_addr_list[0];
     /* this is not threadsafe */
     ip = inet_ntoa (*addrp);
+#elif defined(__INTERIX)
+	addrs = hostinfo->h_addr_list;
+    /* this is not threadsafe */
+	ip = inet_ntoa (*(struct in_addr*)addrs[0]);
 #else
     addrs = hostinfo->h_addr_list;
     ip = inet_ntop (AF_INET, (struct in_addr *) addrs[0], ipbuf,
