Index: ocaml-3.10.2/configure
===================================================================
--- ocaml-3.10.2.orig/configure
+++ ocaml-3.10.2/configure
@@ -1339,6 +1339,8 @@ if test $has_tk = true; then
   tcl_version=`sh ./runtest $tk_defs $tk_x11_include tclversion.c`
   for tk_incs in \
     "-I/usr/local/include" \
+    "-I/usr/local/include/tcl8.5 -I/usr/local/include/tk8.5" \
+    "-I/usr/include/tcl8.5 -I/usr/include/tk8.5" \
     "-I/usr/local/include/tcl8.4 -I/usr/local/include/tk8.4" \
     "-I/usr/include/tcl8.4 -I/usr/include/tk8.4" \
     "-I/usr/local/include/tcl8.3 -I/usr/local/include/tk8.3" \
@@ -1361,6 +1363,7 @@ if test $has_tk = true; then
     8.2) tclmaj=8 tclmin=2 tkmaj=8 tkmin=2 ;;
     8.3) tclmaj=8 tclmin=3 tkmaj=8 tkmin=3 ;;
     8.4) tclmaj=8 tclmin=4 tkmaj=8 tkmin=4 ;;
+    8.5) tclmaj=8 tclmin=5 tkmaj=8 tkmin=5 ;;
     *) echo "This version is not known."; has_tk=false ;;
     esac
   else
Index: ocaml-3.10.2/otherlibs/labltk/support/cltkImg.c
===================================================================
--- ocaml-3.10.2.orig/otherlibs/labltk/support/cltkImg.c
+++ ocaml-3.10.2/otherlibs/labltk/support/cltkImg.c
@@ -98,7 +98,11 @@ camltk_setimgdata_native (value imgname,
   pib.offset[0] = 0;
   pib.offset[1] = 1;
   pib.offset[2] = 2;
+#if (TK_MAJOR_VERSION == 8 && TK_MINOR_VERSION >= 5 || TK_MAJOR_VERSION > 8)
+  Tk_PhotoPutBlock(cltclinterp,ph,&pib,Int_val(x),Int_val(y),Int_val(w),Int_val(h)
+#else
   Tk_PhotoPutBlock(ph,&pib,Int_val(x),Int_val(y),Int_val(w),Int_val(h)
+#endif
 #if (TK_MAJOR_VERSION == 8 && TK_MINOR_VERSION >= 4 || TK_MAJOR_VERSION > 8)
                    , TK_PHOTO_COMPOSITE_SET
 #endif
Index: ocaml-3.10.2/otherlibs/labltk/tkanim/tkAnimGIF.c
===================================================================
--- ocaml-3.10.2.orig/otherlibs/labltk/tkanim/tkAnimGIF.c
+++ ocaml-3.10.2/otherlibs/labltk/tkanim/tkAnimGIF.c
@@ -334,7 +334,11 @@ FileReadGIF(interp, f, fileName, formatS
                 goto error;
             }
         }
+#if (TK_MAJOR_VERSION == 8 && TK_MINOR_VERSION >= 5 || TK_MAJOR_VERSION > 8)
+        Tk_PhotoPutBlock(interp, photoHandle, &block, 0, 0, imageWidth, imageHeight
+#else
         Tk_PhotoPutBlock(photoHandle, &block, 0, 0, imageWidth, imageHeight
+#endif
 #if (TK_MAJOR_VERSION == 8 && TK_MINOR_VERSION >= 4 || TK_MAJOR_VERSION > 8)
                    , TK_PHOTO_COMPOSITE_SET
 #endif
