Index: ocaml-3.11.0+beta1/configure
===================================================================
--- ocaml-3.11.0+beta1.orig/configure
+++ ocaml-3.11.0+beta1/configure
@@ -245,7 +245,7 @@ esac
 # Configure the bytecode compiler
 
 bytecc="$cc"
-mkexe="\$(BYTECC)"
+mkexe="\$(BYTECC) \$(BYTECCLINKOPTS)"
 bytecccompopts=""
 bytecclinkopts=""
 dllccompopts=""
@@ -1500,7 +1500,15 @@ fi
 
 # Final twiddling of compiler options to work around known bugs
 
+bytecccompopts="$CFLAGS $bytecccompopts"
+bytecclinkopts="$LDFLAGS $bytecclinkopts"
+natdynlinkopts="$LDFLAGS $natdynlinkopts"
+nativeccrawlinkopts="$RAW_LDFLAGS $nativecclinkopts"
+nativecclinkopts="$LDFLAGS $nativecclinkopts"
+nativecccompopts="$CFLAGS $nativecccompopts"
 nativeccprofopts="$nativecccompopts"
+mksharedlib="$mksharedlib $LDFLAGS"
+mkmaindll="$mkmaindll $LDFLAGS"
 case "$buggycc" in
   gcc.2.96)
     bytecccompopts="$bytecccompopts -fomit-frame-pointer"
@@ -1548,7 +1556,7 @@ echo "DEBUGGER=$debugger" >> Makefile
 echo "CC_PROFILE=$cc_profile" >> Makefile
 echo "SYSTHREAD_SUPPORT=$systhread_support" >> Makefile
 echo "PARTIALLD=$partialld" >> Makefile
-echo "PACKLD=\$(PARTIALLD) \$(NATIVECCLINKOPTS) -o " >> Makefile
+echo "PACKLD=\$(PARTIALLD) $nativeccrawlinkopts -o " >> Makefile
 echo "DLLCCCOMPOPTS=$dllccompopts" >> Makefile
 echo "IFLEXDIR=$iflexdir" >> Makefile
 echo "O=o" >> Makefile
