diff -urN yelp-2.20.0-orig/configure.in yelp-2.20.0/configure.in
--- yelp-2.20.0-orig/configure.in	2008-03-12 10:17:17.000000000 +0530
+++ yelp-2.20.0/configure.in	2008-03-12 10:19:22.000000000 +0530
@@ -133,7 +133,12 @@
 dnl beagle
 dnl ******
 
-BEAGLE_MODULES="libbeagle-0.0 >= 0.2.4"
+BEAGLE_MODULES=
+if $PKG_CONFIG --exists libbeagle-1.0; then
+	BEAGLE_MODULES="libbeagle-1.0 >= 0.3.0"
+else
+	BEAGLE_MODULES="libbeagle-0.0 >= 0.2.4"
+fi
 
 AC_ARG_WITH([search],
 	[AC_HELP_STRING([--with-search=basic|beagle|auto],
diff -urN yelp-2.20.0-orig/src/yelp-search-parser.c yelp-2.20.0/src/yelp-search-parser.c
--- yelp-2.20.0-orig/src/yelp-search-parser.c	2008-03-12 10:17:16.000000000 +0530
+++ yelp-2.20.0/src/yelp-search-parser.c	2008-03-12 10:19:54.000000000 +0530
@@ -304,7 +313,8 @@
     g_free (check);
     xmlNewTextChild (parser->root, NULL, BAD_CAST "online1", BAD_CAST split[1]);
 
-    parser->finished = TRUE;
+    if (parser->snippet_request_count == 0)
+      parser->finished = TRUE;
 
 }
 
@@ -357,7 +367,7 @@
 
     const char *xml = beagle_snippet_response_get_snippet (response);
 
-    if (xml == NULL) {
+    if (xml == NULL || strlen (xml) == 0) {
 	debug_print (DB_DEBUG, "snippet_response empty\n");
 	return;
     }
@@ -552,7 +562,7 @@
 
 	beagle_query_set_max_hits (query, 10000);
 	beagle_query_add_text (query, parser->search_terms);
-	beagle_query_add_source (query, "documentation");
+	beagle_query_add_text (query, "source:documentation OR source:manpages");
 
 	parser->hits = g_ptr_array_new ();
 
