diff -ur indent-2.2.9.orig/src/indent.c indent-2.2.9/src/indent.c
--- indent-2.2.9.orig/src/indent.c	2002-10-28 12:00:56.000000000 -0800
+++ indent-2.2.9/src/indent.c	2006-08-01 16:13:42.000000000 -0700
@@ -103,6 +103,7 @@
 unsigned long    in_prog_size   = 0U;
 char           * in_prog        = NULL;
 int              break_line     = 0;
+BOOLEAN          inhibited    = 0;  /* Whether or not reformatting is currently disabled */
 
 /* The position that we will line the current line up with when it comes time
  * to print it (if we are lining up to parentheses).  */
@@ -2036,12 +2038,24 @@
     BOOLEAN        * force_nl,
     BOOLEAN        * flushed_nl)
 {
+            BOOLEAN save_inhibited;
+
             if (parser_state_tos->last_saw_nl && (s_code != e_code))
             {
+                /* remember if this comment is actually *INDENT-OFF* directive */
+                save_inhibited = inhibited;
+
                 *flushed_nl = false;
                 dump_line(true, &paren_target);
                 parser_state_tos->want_blank = false;
                 *force_nl = false;
+
+                /* if the comment was *INDENT-OFF* directive, it has been used up already
+                 * by dump_line() above so return now */
+                if (save_inhibited)
+                {
+                    return;
+                }
             }
             print_comment (&paren_target);
 }
diff -ur indent-2.2.9.orig/src/indent.h indent-2.2.9/src/indent.h
--- indent-2.2.9.orig/src/indent.h	2002-11-10 13:02:48.000000000 -0800
+++ indent-2.2.9/src/indent.h	2006-08-01 16:13:37.000000000 -0700
@@ -124,6 +122,8 @@
 
 extern int break_line;          /* Whether or not we should break the line. */
 
+extern BOOLEAN inhibited;       /* Whether or not reformatting is currently disabled */
+
 /* pointer to the token that lexi() has just found */
 extern char *token;
 
diff -ur indent-2.2.9.orig/src/output.c indent-2.2.9/src/output.c
--- indent-2.2.9.orig/src/output.c	2002-12-12 09:36:49.000000000 -0800
+++ indent-2.2.9/src/output.c	2006-08-01 16:13:34.000000000 -0700
@@ -24,7 +26,6 @@
 RCSTAG_CC ("$Id: 2.2.9-indent-off-segfault.patch,v 1.1 2006/07/31 06:10:31 wormo Exp $");
 
 static FILE            * output       = NULL;
-static BOOLEAN           inhibited    = 0;
 static buf_break_st_ty * buf_break_list = NULL;
 
 /* Priority mask bits */
