--- indent-2.2.10/man/texinfo2man.c.orig	2008-03-11 19:50:42.000000000 +0100
+++ indent-2.2.10/man/texinfo2man.c	2009-11-23 06:06:43.000000000 +0100
@@ -3,6 +3,7 @@
 #include <malloc.h>
 #include <string.h>
 #include <ctype.h>
+#include <stdlib.h>
 
 /* texinfo2man.
  * Convert a texinfo document to man format.
@@ -163,7 +164,7 @@
 
 static char value_updated[64], value_edition[64], value_version[64];
 
-process_texi (FILE * in)
+void process_texi (FILE * in)
 {
   char buf[1024];
   int in_block = 0;
@@ -253,6 +254,12 @@
     char buf[1024];
     int line_no = 0;
 
+	if (argc != 3) {
+		fprintf(stderr, "Usage: %s <man template> <info file>\n", argv[0]);
+		fprintf(stderr, "  The man page will be written to stdout.\n");
+		return -1;
+	}
+
     texinfoname = argv[2];
 
     in = fopen (argv[2], "r");
