--- configure.in.orig 
+++ configure.in 
@@ -3932,23 +3932,6 @@
     if test "$DB_INCLUDES" = ""; then
        AC_MSG_ERROR(no. install the db4-dev package)
     fi
-    AC_MSG_CHECKING([whether db is at least 4.1])
-    for v in `seq 1 7`; do
-       AC_TRY_RUN([
-#include <db.h>
-
-int main(int argc, char **argv) {
-       if(DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR == $v) return 0;
-       else return 1;
-}
-       ], [DB_VERSION_MINOR=$v], [])
-    done
-    if test "$DB_VERSION_MINOR" -gt "1"; then
-       AC_MSG_RESULT([OK])
-       DB_VERSION=4.$DB_VERSION_MINOR
-    else
-       AC_MSG_ERROR([no. you need at least db 4.1])
-    fi
     save_LIBS="$LIBS"
     for dbver in -4.8 4.8 -4.7 4.7 -4.6 4.6 -4.5 4.5 -4.4 4.4 -4.3 4.3 -4.2 4.2 -4.1 4.1 -4 4 ''; do
        AC_CHECK_LIB(db$dbver, db_create, [ DB_LIB="db$dbver"; DB_CPPLIB="db_cxx$dbver"; LIBS="-ldb$dbver $LIBS"; break ])

