--- patches/dev300/apply 
+++ patches/dev300/apply 
@@ -452,10 +452,6 @@
 
 # avoid error messages about missing directories with templates
 sfx2-check-existing-template-dirs.diff, i#100518, pmladek
-
-[ FedoraLinuxOnlyFixes ]
-# fix for new SQLLEN in unixodbc. Update internal headers.
-new-SQLLEN.diff, i#104088
 
 [ LinuxOnly ]
 # fix that allows OOo to work with a standard (unpatched) libjpeg,

