--- patches/dev300/apply 
+++ patches/dev300/apply
@@ -1596,7 +1596,7 @@
 SectionOwner => aprosky
 
 # Allow build to proceed with PaX enabled
-gentoo-pax-fix.diff, aprosky
+#gentoo-pax-fix.diff, aprosky
 # support server-only versions of 64bit JDKs
 64bit-jdk-server-paths.diff, i#65974, hmth
 # use IBM j9vm over IBM classic JVM
@@ -3561,7 +3561,6 @@
 [ BuildBits ]
 # Allow oowintool to detect java sdk 1.6.x
 oowintool-java6.diff, fridrich
-connectivity-build-fix-mac.diff, thorsten
 vcl-xinerama-clone-fix.diff, n#578730, thorsten
 
 [ HyphenFix ]
--- configure.in 
+++ configure.in 
@@ -1326,12 +1326,18 @@
 	done
     fi
     SUNTEMPLATES_LANG=
-    # check whether the langs are requested at all
-    for lang in $wanted_sun_templates ; do
-	if test -n "`echo $with_lang | grep "$lang"`" ; then
-		SUNTEMPLATES_LANG="$SUNTEMPLATES_LANG $lang"
-	fi
-    done
+
+    # --with-lang="" now used for en-US
+    if test "z$with_lang" = "z" ; then
+        SUNTEMPLATES_LANG="en-US"
+    else
+        # check whether the langs are requested at all
+        for lang in $wanted_sun_templates ; do
+	    if test -n "`echo $with_lang | grep "$lang"`" ; then
+		  SUNTEMPLATES_LANG="$SUNTEMPLATES_LANG $lang"
+	    fi
+        done
+    fi
     AC_MSG_RESULT([$SUNTEMPLATES_LANG])
 fi
 AC_SUBST(SUNTEMPLATES_LANG)

