--- bin/package-ooo 
+++ bin/package-ooo 
@@ -33,6 +33,7 @@
 {
     cat <<EOT >$1
 #!/bin/sh
+unset PYTHONPATH
 $OOINSTBASE/program/soffice $2 "\$@"
 EOT
     chmod 755 $1
--- patches/dev300/apply 
+++ patches/dev300/apply 
@@ -1375,12 +1375,10 @@
 system-db-check.diff, i#65979, hmth
 # User ~/.ooo3 as user dir
 scp2-user-config-ooo3.diff, pmladek
+#fix build without java
+lucene-no-java.diff
-# unused java code
-disable-parser-jaxp.diff
 # default inc/lib
 config_office-XINC-XLIB-defaults.diff, i#65976
-# gdi/font.cxx: stl list error
-vcl_font-NO_LIST.diff
 # linux-headers-2.6.22 issue?!
 linux-headers.diff

--- distro-configs/Gentoo.conf.in 
+++ distro-configs/Gentoo.conf.in 
@@ -1,15 +1,23 @@
 --disable-fontooo
 --disable-qadevooo
 --enable-libsn
+--enable-neon
 --enable-ogltrans
 --enable-xsltproc
 --with-dynamic-xinerama
+--with-external-dict-dir=/usr/share/myspell
+--with-external-hyph-dir=/usr/share/myspell
+--with-external-thes-dir=/usr/share/myspell
 --with-system-boost
 --with-system-curl
 --with-system-db
+--with-system-dicts
 --with-system-expat
+--with-system-hunspell
 --with-system-icu
 --with-system-libxslt
+--with-system-neon
+--with-system-openssl
 --with-system-vigra
 --with-system-xmlsec
 --with-system-xrender-headers

