Index: makefile.mk
===================================================================
RCS file: /cvs/external/lucene/makefile.mk,v
retrieving revision 1.3.2.1
diff -u -r1.3.2.1 makefile.mk
--- lucene/makefile.mk	31 Jul 2008 20:07:34 -0000	1.3.2.1
+++ lucene/makefile.mk	19 Sep 2008 11:23:37 -0000
@@ -41,6 +41,8 @@
 .INCLUDE :	antsettings.mk
 # --- Files --------------------------------------------------------
 
+.IF "$(SOLAR_JAVA)" != ""
+
 LUCENE_MAJOR=2
 LUCENE_MINOR=3
 LUCENE_MICRO=2
@@ -58,9 +60,13 @@
 
 OUT2BIN=.$/build$/$(LUCENE_CORE_JAR) .$/build$/contrib$/analyzers$/$(LUCENE_ANALYZERS_JAR)
 
+.ENDIF
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE : set_ext.mk
 .INCLUDE : target.mk
-.INCLUDE : tg_ext.mk 
 
+.IF "$(SOLAR_JAVA)" != ""
+.INCLUDE : tg_ext.mk 
+.ENDIF
