--- ooo-build-2.4.1.8/bin/package-ooo.orig
+++ ooo-build-2.4.1.8/bin/package-ooo
@@ -92,12 +92,12 @@
 fi
 
 # create bash completion
-mkdir -p $OODESTDIR/etc/bash_completion.d
+mkdir -p $SYSCONFDIR/bash_completion.d
 gen_bash_comp_opts=--binsuffix="$BINSUFFIX"
 if test "z$ENABLE_OPENXML" = "zyes" ; then
     gen_bash_comp_opts="$gen_bash_comp_opts --enable-openxml"
 fi
-$TOOLSDIR/bin/generate-bash-completion $gen_bash_comp_opts $TOOLSDIR/bin/bash-completion.in $OODESTDIR/etc/bash_completion.d/ooffice${BINSUFFIX}.sh
+$TOOLSDIR/bin/generate-bash-completion $gen_bash_comp_opts $TOOLSDIR/bin/bash-completion.in $SYSCONFDIR/bash_completion.d/ooffice${BINSUFFIX}.sh
 
 if test "z$VENDORNAME" != "zRedHat"; then
 	mkdir -p $MANDIR/man1
@@ -161,11 +161,11 @@
         # Icons and menu stuff is handled in actions.py
         ;;
   *)
-	mkdir -p $OODESTDIR/usr/share/applications
+	mkdir -p $DATADIR/applications
 	cd $TOOLSDIR/desktop
 	for source in *.desktop ; do
 	    dest=`echo $source | sed "s|.desktop\$||"`
-	    dest="$OODESTDIR/usr/share/applications/$dest$BINSUFFIX.desktop"
+	    dest="$DATADIR/applications/$dest$BINSUFFIX.desktop"
 	    add_version=
 	    test "z$VENDORNAME" = "zNovell" -a "z$BINSUFFIX" != "z" && add_version=" ($VERSION)" || :
 	    sed -e "s|\(^Name.*\)\$|\1$add_version|
@@ -173,7 +173,7 @@
 	done
 
 	# icons
-	icondir=/usr/share/icons/hicolor
+	icondir=$DATADIRBASE/icons/hicolor
 	for size in 16x16 22x22 24x24 32x32 48x48 scalable ; do
 	    mkdir -p $OODESTDIR/$icondir/$size/apps
 	    cd $TOOLSDIR/desktop/$size/
@@ -187,7 +187,7 @@
 	done
 
 	# create symlinks below share/pixmaps to keep the backward compatibility
-	pixmapsdir=$OODESTDIR/usr/share/pixmaps
+	pixmapsdir=$DATADIR/pixmaps
 	mkdir -p $pixmapsdir
 	cd $OODESTDIR/$icondir/48x48/apps
 	for icon in ooo-*.png ; do
@@ -195,9 +195,9 @@
 	done
 	
 	# shared MIME info
-	mkdir -p $OODESTDIR/usr/share/mime/packages
+	mkdir -p $DATADIR/mime/packages
 	cd $TOOLSDIR/desktop
-	cp openoffice.xml $OODESTDIR/usr/share/mime/packages
+	cp openoffice.xml $DATADIR/mime/packages
 	if test "z$RUN_POST_INSTALL_SCRIPTS" = "zyes" && 
 	   which update-mime-database >/dev/null 2>&1 ; then
 	    update-mime-database /usr/share/mime || :
