--- sbcl-1.0.15-orig.ebuild	2008-04-09 17:56:30.347838000 -0500
+++ sbcl-1.0.15.ebuild	2008-04-09 17:57:00.290529000 -0500
@@ -102,6 +102,7 @@
 	cd "${S}"
 
 	epatch "${FILESDIR}"/${PN}-1.0.6-solaris.patch
+	epatch "${FILESDIR}"/${PN}-binutils.patch
 #	epatch "${FILESDIR}/disable-tests-gentoo-${PV}.patch"
 	use source && sed 's%"$(build_root)%$(module).lisp "$(build_root)%' -i contrib/vanilla-module.mk
 
