diff -ur sbcl-0.9.17.orig/contrib/vanilla-module.mk sbcl-0.9.17/contrib/vanilla-module.mk
--- sbcl-0.9.17.orig/contrib/vanilla-module.mk	2006-09-08 08:36:47.000000000 -0500
+++ sbcl-0.9.17/contrib/vanilla-module.mk	2006-09-29 21:53:50.000000000 -0500
@@ -5,4 +5,4 @@
 test:: $(MODULE).fasl
 
 install:
-	cp $(MODULE).fasl "$(BUILD_ROOT)$(INSTALL_DIR)"
+	cp $(MODULE).fasl $(MODULE).lisp "$(BUILD_ROOT)$(INSTALL_DIR)"
Only in sbcl-0.9.17/contrib: vanilla-module.mk.~1.6.~
