--- setupext.py
+++ setupext.py
@@ -46,12 +46,11 @@
 import subprocess
 
 basedir = {
-    'win32'  : ['win32_static',],
-    'linux2' : ['/usr/local', '/usr'],
-    'linux'  : ['/usr/local', '/usr',],
-    'cygwin' : ['/usr/local', '/usr',],
-    '_darwin' : ['/sw/lib/freetype2', '/sw/lib/freetype219', '/usr/local',
-                '/usr', '/sw'],
+    'win32'  : [],
+    'linux2' : [],
+    'linux'  : [],
+    'cygwin' : [],
+    '_darwin' : [],
     # it appears builds with darwin are broken because of all the
     # different flags the deps can be compile with, so I am pushing
     # people to :
@@ -59,13 +58,13 @@
 
     'darwin' : [],
 
-    'freebsd4' : ['/usr/local', '/usr'],
-    'freebsd5' : ['/usr/local', '/usr'],
-    'freebsd6' : ['/usr/local', '/usr'],
-    'sunos5' : [os.getenv('MPLIB_BASE') or '/usr/local',],
-    'gnukfreebsd5' : ['/usr/local', '/usr'],
-    'gnukfreebsd6' : ['/usr/local', '/usr'],
-    'aix5' : ['/usr/local'],
+    'freebsd4' : [],
+    'freebsd5' : [],
+    'freebsd6' : [],
+    'sunos5' : [],
+    'gnukfreebsd5' : [],
+    'gnukfreebsd6' : [],
+    'aix5' : [],
 }
 
 import sys, os, stat
@@ -328,6 +327,7 @@
 
     module.include_dirs.extend(incdirs)
     module.include_dirs.append('.')
+    module.include_dirs.append(os.path.join(sys.prefix, 'include'))
     module.library_dirs.extend(libdirs)
 
 def getoutput(s):
