diff -ru bash_completion.orig/bash_completion bash_completion/bash_completion
--- bash_completion.orig/bash_completion	2008-04-03 08:33:39 +0200
+++ bash_completion/bash_completion	2008-04-03 08:37:16 +0200
@@ -162,7 +162,7 @@
 have()
 {
 	unset -v have
-	PATH=$PATH:/sbin:/usr/sbin:/usr/local/sbin type $1 &>/dev/null &&
+	PATH=$PATH:@GENTOO_PORTAGE_EPREFIX@/sbin:@GENTOO_PORTAGE_EPREFIX@/usr/sbin type $1 &>/dev/null &&
 		have="yes"
 }
 
@@ -1319,7 +1319,7 @@
 	else
 		len=${#cur}
 		idx=0
-		for pval in $( /usr/sbin/postconf -m ); do
+		for pval in $( @GENTOO_PORTAGE_EPREFIX@/usr/sbin/postconf -m ); do
 			if [[ "$cur" == "${pval:0:$len}" ]]; then
 				COMPREPLY[$idx]="$pval:"
 				idx=$(($idx+1))
@@ -1395,7 +1395,7 @@
 	fi
 	len=${#cur}
 	idx=0
-	for pval in $( /usr/sbin/postconf | cut -d ' ' -f 1 ); do
+	for pval in $( @GENTOO_PORTAGE_EPREFIX@/usr/sbin/postconf | cut -d ' ' -f 1 ); do
 		if [[ "$cur" == "${pval:0:$len}" ]]; then
 			COMPREPLY[$idx]="$pval$eqext"
 			idx=$(($idx+1))
@@ -1825,7 +1825,7 @@
 	--eval)
 		# get a list of macros
 		COMPREPLY=( $( sed -ne 's|^\(%'${cur#\%}'[^ '$'\t'']*\).*$|\1|p' \
-			       /usr/lib/rpm/macros ) )
+			       @GENTOO_PORTAGE_EPREFIX@/usr/lib/rpm/macros ) )
 		return 0
 		;;
 	--pipe)
@@ -3176,7 +3176,7 @@
 
 _root_command()
 {
-	PATH=$PATH:/sbin:/usr/sbin:/usr/local/sbin _command $1 $2 $3
+	PATH=$PATH:@GENTOO_PORTAGE_EPREFIX@/sbin:@GENTOO_PORTAGE_EPREFIX@/usr/sbin _command $1 $2 $3
 }
 complete -F _root_command $filenames sudo fakeroot really
 
@@ -3369,7 +3369,7 @@
 	local config
 
 	config=/etc/ssl/openssl.cnf
-	[ ! -f $config ] && config=/usr/share/ssl/openssl.cnf
+	[ ! -f $config ] && config=@GENTOO_PORTAGE_EPREFIX@/usr/share/ssl/openssl.cnf
 	for (( i=2; i < COMP_CWORD; i++ )); do
 		if [[ "${COMP_WORDS[i]}" == -config ]]; then
 			config=${COMP_WORDS[i+1]}
@@ -4308,7 +4308,7 @@
 
 	case "$prev" in
 	    -@(f|-frontend))
-		opt=( $( echo /usr/share/perl5/Debconf/FrontEnd/* ) )
+		opt=( $( echo @GENTOO_PORTAGE_EPREFIX@/usr/share/perl5/Debconf/FrontEnd/* ) )
 		opt=( ${opt[@]##*/} )
 		opt=( ${opt[@]%.pm} )
 		COMPREPLY=( $( compgen -W '${opt[@]}' -- $cur ) )
@@ -4410,7 +4410,7 @@
 					;;
 				"-F")
 					# -F: force change log format
-					COMPREPLY=( $( ( cd /usr/lib/dpkg/parsechangelog; compgen -f "$cur" ) ) )
+					COMPREPLY=( $( ( cd @GENTOO_PORTAGE_EPREFIX@/usr/lib/dpkg/parsechangelog; compgen -f "$cur" ) ) )
 					return 0
 					;;
 				"-V"|"-D")
@@ -5490,7 +5490,7 @@
 	local cur portsdir prev indexfile
 	local -a COMPREPLY2
 
-	portsdir=${PORTSDIR:-/usr/ports}/
+	portsdir=${PORTSDIR:-@GENTOO_PORTAGE_EPREFIX@/usr/ports}/
 	cur=${COMP_WORDS[COMP_CWORD]}
 	prev=${COMP_WORDS[COMP_CWORD-1]}
 	# First try INDEX-5
@@ -5639,7 +5639,7 @@
 			if [ -n "$MPLAYER_SKINS_DIR" ]; then
 				skinsdir=$MPLAYER_SKINS_DIR
 			else
-				skinsdir=/usr/local/share/mplayer/Skin
+				skinsdir=@GENTOO_PORTAGE_EPREFIX@/usr/local/share/mplayer/Skin
 			fi
 
 			IFS=$'\t\n' 
@@ -6049,7 +6049,7 @@
 	COMPREPLY=()
 	cur=${COMP_WORDS[COMP_CWORD]}
 	prev=${COMP_WORDS[COMP_CWORD-1]}
-	dictfile=/usr/share/dict/words
+	dictfile=@GENTOO_PORTAGE_EPREFIX@/usr/share/dict/words
 
 	for (( i=1; i < COMP_CWORD; i++ )); do
 		case "${COMP_WORDS[i]}" in
@@ -6812,7 +6812,7 @@
 _aspell_dictionary()
 {
 	local datadir
-	datadir=/usr/lib/aspell
+	datadir=@GENTOO_PORTAGE_EPREFIX@/usr/lib/aspell
 	COMPREPLY=( $( command ls $datadir/*.@(multi|alias) ) )
 	COMPREPLY=( ${COMPREPLY[@]%.@(multi|alias)} )
 	COMPREPLY=( $( compgen -W '${COMPREPLY[@]#$datadir/}' -- $cur ) )
@@ -6947,7 +6947,7 @@
 		return 0
 	fi
 
-	infopath='/usr/share/info'
+	infopath='@GENTOO_PORTAGE_EPREFIX@/usr/share/info'
 
 	if [ "${INFOPATH: -1:1}" == ':' ]; then
 		infopath=${INFOPATH}${infopath}
