--- util/Makefile.in
+++ util/Makefile.in
@@ -99,1 +99,1 @@
-AM_CPPFLAGS = -I. -I.. -I$(top_srcdir)/lib '-DDEFAULT_CRACKLIB_DICT="$(pkgdatadir)/pw_dict"'
+AM_CPPFLAGS = -I. -I.. -I$(top_srcdir)/lib '-DDEFAULT_CRACKLIB_DICT="$(libdir)/cracklib_dict"'
--- lib/Makefile.in
+++ lib/Makefile.in
@@ -103,1 +103,1 @@
-AM_CPPFLAGS = -I. -I.. -I$(top_srcdir)/lib -DIN_CRACKLIB '-DDEFAULT_CRACKLIB_DICT="$(pkgdatadir)/pw_dict"'
+AM_CPPFLAGS = -I. -I.. -I$(top_srcdir)/lib -DIN_CRACKLIB '-DDEFAULT_CRACKLIB_DICT="$(libdir)/cracklib_dict"'
--- python/Makefile.in
+++ python/Makefile.in
@@ -41 +41 @@
-@BUILD_PYTHON_TRUE@am__append_1 = '-DDEFAULT_CRACKLIB_DICT="$(pkgdatadir)/pw_dict"' \
+@BUILD_PYTHON_TRUE@am__append_1 = '-DDEFAULT_CRACKLIB_DICT="$(libdir)/cracklib_dict"' \
