--- mailnews/extensions/enigmail/ui/content/enigmailMessengerOverlay.js.orig	2007-05-22 17:08:53.000000000 +0000
+++ mailnews/extensions/enigmail/ui/content/enigmailMessengerOverlay.js	2007-05-22 17:09:32.000000000 +0000
@@ -139,7 +139,8 @@
 // the following headers are added for compatibility with DispMUA extension:
     "x-mimeole",
     "x-bugzilla-reason",
-    "x-php-bug" ];
+    "x-php-bug",
+    "list-post" ];
 
   var addHdr = [];
 
--- mailnews/extensions/enigmail/package/prefs/enigmail.js.orig	2007-05-22 17:10:41.000000000 +0000
+++ mailnews/extensions/enigmail/package/prefs/enigmail.js 2007-05-22 17:10:53.000000000 +0000
@@ -69,7 +69,7 @@
 pref("extensions.enigmail.handleDoubleClick",false);
 
 // headers to be retrieved from the mail backend, but hidden from the users
-pref("extensions.enigmail.hideHeaders", "x-enigmail-version openpgp content-transfer-encoding x-mimeole x-bugzilla-reason x-php-bug");
+pref("extensions.enigmail.hideHeaders", "x-enigmail-version openpgp content-transfer-encoding x-mimeole x-bugzilla-reason x-php-bug list-post");
 
 // disable '<' and '>' around email addresses
 pref("extensions.enigmail.hushMailSupport",false);