--- mailnews/extensions/enigmail/package/enigmail.js	2010-06-26 19:51:30.000000000 +0200
+++ mailnews/extensions/enigmail/package/enigmail.js	2010-07-10 17:34:46.000000000 +0200
@@ -1231,7 +1231,10 @@ function () {
                .getInstallLocation(ENIGMAIL_EXTENSION_ID);
       var extensionLoc = installLoc.getItemFile(ENIGMAIL_EXTENSION_ID, "wrappers");
       extensionLoc.append("gpg-agent-wrapper.sh");
-			extensionLoc.permissions=0755;
+      try {
+        extensionLoc.permissions=0755;
+      }
+      catch(ex) {}
       agentProcess = Components.classes[NS_PROCESS_UTIL_CONTRACTID].createInstance(Components.interfaces.nsIProcess);
       agentProcess.init(extensionLoc);
       agentProcess.run(true, [ "stop", this.gpgAgentProcess ], 2);
@@ -1758,7 +1761,10 @@ function (domWindow) {
                    .getInstallLocation(ENIGMAIL_EXTENSION_ID);
           var extensionLoc = installLoc.getItemFile(ENIGMAIL_EXTENSION_ID, "wrappers");
           extensionLoc.append("gpg-agent-wrapper.sh");
-          extensionLoc.permissions=0755;
+          try {
+            extensionLoc.permissions=0755;
+          }
+          catch(ex) {}
           args.unshift(command.path);
           args.unshift("start");
           command = extensionLoc;

