Index: configure
===================================================================
--- configure	(revision 26439)
+++ configure	(working copy)
@@ -2300,6 +2300,24 @@
     _altivec=yes
   fi
 fi
+if ! x86 ; then
+  _mmx=
+  _3dnow=
+  _3dnowext=
+  _mmxext=
+  _sse=
+  _sse2=
+  _ssse3=
+  _cmov=
+fi
+if ! ppc ; then
+  _altivec=
+fi
+if ! arm ; then
+  _armv5te=
+  _armv6=
+  _iwmmxt=
+fi
 
 
 echocheck "extern symbol prefix"
