diff -ur mplayer-1.0_rc2_p27725.orig/libavcodec/libx264.c mplayer-1.0_rc2_p27725/libavcodec/libx264.c
--- mplayer-1.0_rc2_p27725.orig/libavcodec/libx264.c	2008-10-06 21:04:25.000000000 -0400
+++ mplayer-1.0_rc2_p27725/libavcodec/libx264.c	2008-10-07 01:38:28.325921893 -0400
@@ -162,7 +162,11 @@
 
     x4->params.i_bframe = avctx->max_b_frames;
     x4->params.b_cabac = avctx->coder_type == FF_CODER_TYPE_AC;
+#if X264_BUILD > 62
     x4->params.i_bframe_adaptive = avctx->b_frame_strategy;
+#else
+    x4->params.b_bframe_adaptive = avctx->b_frame_strategy;
+#endif
     x4->params.i_bframe_bias = avctx->bframebias;
     x4->params.b_bframe_pyramid = avctx->flags2 & CODEC_FLAG2_BPYRAMID;
     avctx->has_b_frames= avctx->flags2 & CODEC_FLAG2_BPYRAMID ? 2 : !!avctx->max_b_frames;
