--- configure
+++ configure
@@ -1832,7 +1832,15 @@
       cc_check -march=$proc $cpuopt=$proc || proc=pentium4
     fi
     if test "$proc" = "core2" ; then
-      cc_check -march=$proc $cpuopt=$proc || proc=pentium-m
+      if ! cc_check -march=$proc $cpuopt=$proc; then
+        proc=nocona;
+        if ! cc_check -march=$proc $cpuopt=$proc; then
+          proc=prescott;
+          if ! cc_check -march=$proc $cpuopt=$proc; then
+            proc=pentium-m;
+          fi
+        fi
+      fi
     fi
     if test "$proc" = "pentium4" || test "$proc" = "pentium-m" || test "$proc" = "pentium3" || test "$proc" = "pentium2" || test "$proc" = "athlon" || test "$proc" = "c3-2" || test "$proc" = "geode"; then
       cc_check -march=$proc $cpuopt=$proc  || proc=i686
