diff -uNr mplayer-1.0_rc2_p24929-orig/libao2/ao_pulse.c mplayer-1.0_rc2_p24929/libao2/ao_pulse.c
--- mplayer-1.0_rc2_p24929-orig/libao2/ao_pulse.c	2007-10-21 12:00:54.000000000 -0700
+++ mplayer-1.0_rc2_p24929/libao2/ao_pulse.c	2008-01-04 15:46:22.000000000 -0800
@@ -253,11 +253,16 @@
 /** Play the specified data to the pulseaudio server */
 static int play(void* data, int len, int flags) {
     if (len) {
-        if (pa_stream_write(stream, data, len, NULL, 0, PA_SEEK_RELATIVE) < 0) {
+        int rc;
+        pa_threaded_mainloop_lock(mainloop);
+        rc = pa_stream_write(stream, data, len, NULL, 0, PA_SEEK_RELATIVE);
+        pa_threaded_mainloop_unlock(mainloop);
+        if (rc < 0) {
             GENERIC_ERR_MSG(context, "pa_stream_write() failed");
             return -1;
         }
     } else {
+        pa_threaded_mainloop_lock(mainloop);
         pa_operation *o = pa_stream_trigger(stream, NULL, NULL);
         if (!o) {
             GENERIC_ERR_MSG(context, "pa_stream_trigger() failed");
@@ -265,6 +270,7 @@
         }
         /* We don't wait for this operation to complete */
         pa_operation_unref(o);
+        pa_threaded_mainloop_unlock(mainloop);
     }
     return len;
 }
