--- configure.orig	2007-09-01 20:01:18 +0200
+++ configure	2007-09-01 20:08:04 +0200
@@ -1272,16 +1272,6 @@
   _ldconfig="ldconfig -R"
 fi
 
-if freebsd ; then
-  _ld_extra="$_ld_extra -L/usr/local/lib"
-  _inc_extra="$_inc_extra -I/usr/local/include"
-fi
-
-if darwin; then
-  _ld_extra="$_ld_extra -L/usr/local/lib"
-  _inc_extra="$_inc_extra -I/usr/local/include"
-fi
-
 if aix ; then
   _ld_extra="$_ld_extra -lC"
 fi
@@ -3850,7 +3840,7 @@
 
 
 echocheck "X11 headers presence"
-  for I in `echo $_inc_extra | sed s/-I//g` /usr/X11/include /usr/X11R6/include /usr/include/X11R6 /usr/include /usr/openwin/include ; do
+  for I in `echo $_inc_extra | sed s/-I//g` ; do
     if test -f "$I/X11/Xlib.h" ; then
       _inc_x11="-I$I"
       _x11_headers="yes"
@@ -3873,8 +3863,7 @@
 #include <X11/Xutil.h>
 int main(void) { (void) XCreateWindow(0,0,0,0,0,0,0,0,0,0,0,0); return 0; }
 EOF
-  for I in "" -L/usr/X11R6/lib -L/usr/lib/X11R6 -L/usr/X11/lib \
-           -L/usr/lib32 -L/usr/openwin/lib -L/usr/X11R6/lib64 -L/usr/lib ; do
+  for I in "" ; do
     if netbsd; then
       _ld_tmp="$I -lXext -lX11 $_ld_pthread -Wl,-R`echo $I | sed s/^-L//`"
     else
