--- libdvdread/nav_read.c
+++ libdvdread/nav_read.c
@@ -20,6 +20,7 @@
 #include "config.h"
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #if defined(HAVE_INTTYPES_H)
 #include <inttypes.h>
--- tremor/misc.h
+++ tremor/misc.h
@@ -28,9 +28,12 @@
 #ifndef  _LOW_ACCURACY_
 /* 64 bit multiply */
 
+#include <stdlib.h>
 #include <sys/types.h>
 #include "config.h"
 
+extern int _ilog(unsigned int v);
+
 #ifndef WORDS_BIGENDIAN
 union magic {
   struct {
