--- configure
+++ configure
@@ -1329,16 +1329,6 @@
   _ldconfig="ldconfig -R"
 fi
 
-if freebsd ; then
-  _ld_extra="$_ld_extra -L/usr/local/lib"
-  _inc_extra="$_inc_extra -I/usr/local/include"
-fi
-
-if darwin; then
-  _ld_extra="$_ld_extra -L/usr/local/lib"
-  _inc_extra="$_inc_extra -I/usr/local/include"
-fi
-
 if aix ; then
   _ld_extra="$_ld_extra -lC"
 fi
@@ -3845,7 +3835,7 @@
 echocheck "X11 headers presence"
   _x11_headers="no"
   _res_comment="check if the dev(el) packages are installed"
-  for I in `echo $_inc_extra | sed s/-I//g` /usr/include ; do
+  for I in `echo $_inc_extra | sed s/-I//g` ; do
     if test -f "$I/X11/Xlib.h" ; then
       _x11_headers="yes"
       _res_comment=""
--- DOCS/xml/configure
+++ DOCS/xml/configure
@@ -8,17 +8,7 @@
 
 echo "Searching for XML catalogs..."
 for _try_catalog in \
-  /etc/sgml/catalog \
-  /usr/share/xml/docbook/*/catalog.xml \
-  /opt/local/share/xml/docbook-xml/*/catalog.xml \
-  /usr/share/sgml/docbook/xml-dtd-4.1.2/xmlcatalog \
-  /usr/share/apps/ksgmltools2/customization/en/catalog \
-  /usr/share/sgml/catalog \
-  /usr/local/share/sgml/catalog \
-  /usr/lib/sgml/catalog \
-  /usr/local/lib/sgml/catalog \
-  /usr/share/docbook-xml42/catalog.xml \
-  /usr/share/sgml/docbook/xmlcatalog
+  "@GENTOO_PORTAGE_EPREFIX@"/etc/sgml/catalog
 do
   if test -f "$_try_catalog"
   then
@@ -40,15 +30,7 @@
 echo "Searching for stylesheets..."
 echo "Searching for html/chunk.xsl..."
 for _try_chunk_xsl in \
-  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/html/chunk.xsl \
-  /usr/share/sgml/docbook/yelp/docbook/html/chunk.xsl \
-  /usr/local/share/sgml/docbook/stylesheet/xsl/nwalsh/html/chunk.xsl \
-  /usr/local/share/sgml/docbook/yelp/docbook/html/chunk.xsl \
-  /usr/share/docbook-xsl/html/chunk.xsl \
-  /usr/share/sgml/docbook/xsl-stylesheets*/html/chunk.xsl \
-  /usr/share/xml/docbook/stylesheet/nwalsh/current/html/chunk.xsl \
-  /opt/local/share/xsl/docbook-xsl/html/chunk.xsl \
-
+  "@GENTOO_PORTAGE_EPREFIX@"/usr/share/sgml/docbook/xsl-stylesheets*/html/chunk.xsl
 do
   if test -f "$_try_chunk_xsl"
   then
@@ -68,14 +50,7 @@
 
 echo "Searching for html/docbook.xsl..."
 for _try_docbook_xsl in \
-  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/html/docbook.xsl \
-  /usr/share/sgml/docbook/yelp/docbook/html/docbook.xsl \
-  /usr/local/share/sgml/docbook/stylesheet/xsl/nwalsh/html/docbook.xsl \
-  /usr/local/share/sgml/docbook/yelp/docbook/html/docbook.xsl \
-  /usr/share/docbook-xsl/html/docbook.xsl \
-  /usr/share/sgml/docbook/xsl-stylesheets*/html/docbook.xsl \
-  /usr/share/xml/docbook/stylesheet/nwalsh/current/html/docbook.xsl \
-  /opt/local/share/xsl/docbook-xsl/html/docbook.xsl \
+  "@GENTOO_PORTAGE_EPREFIX@"/usr/share/sgml/docbook/xsl-stylesheets*/html/docbook.xsl
 
 do
   if test -f "$_try_docbook_xsl"
@@ -126,14 +101,7 @@
 echo "Searching for DTD..."
 #FIXME: This should prefer higher version numbers, not the other way around ..
 for _try_dtd in \
-  /usr/share/xml/docbook/*/dtd/4*/docbookx.dtd \
-  /usr/share/xml/docbook/*/docbookx.dtd \
-  /usr/share/sgml/docbook/*/docbookx.dtd \
-  /usr/share/sgml/docbook/dtd/*/docbookx.dtd \
-  /usr/share/sgml/docbook/dtd/xml/*/docbookx.dtd \
-  /usr/share/docbook-xml*/docbookx.dtd \
-  /opt/local/share/xml/docbook-xml/*/docbookx.dtd \
-  /usr/share/apps/ksgmltools2/docbook/*/docbookx.dtd
+  "@GENTOO_PORTAGE_EPREFIX@"/usr/share/sgml/docbook/*/docbookx.dtd
 do
   if test -f "$_try_dtd"
   then
