--- Makefile	2008-04-16 21:06:20.000000000 +0200
+++ Makefile.new	2008-04-16 21:11:02.000000000 +0200
@@ -344,8 +344,8 @@
 		if test "$$lang" = en ; then \
 			$(INSTALL) -c -m 644 DOCS/man/en/mplayer.1 $(MANDIR)/man1/ ; \
 		else \
-			$(INSTALL) -d $(MANDIR)/$$i/man1 ; \
-			$(INSTALL) -c -m 644 DOCS/man/$$lang/mplayer.1 $(MANDIR)/$$i/man1/ ; \
+			$(INSTALL) -d $(MANDIR)/$$lang/man1 ; \
+			$(INSTALL) -c -m 644 DOCS/man/$$lang/mplayer.1 $(MANDIR)/$$lang/man1/ ; \
 		fi ; \
 	done
 
