diff -ru ico-1.0.2.orig/ico.c ico-1.0.2/ico.c
--- ico-1.0.2.orig/ico.c	2009-01-29 14:58:21 +0100
+++ ico-1.0.2/ico.c	2009-01-29 14:59:14 +0100
@@ -77,19 +77,27 @@
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>	/* for time_t */
+#include <sys/time.h>	/* for struct timeval */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #include <X11/XlibConf.h>
 #ifdef XTHREADS
 # define MULTITHREAD
 #endif
 #endif /* HAVE_CONFIG_H / autoconf */
 
-#include <math.h>
 #include <X11/Xlib.h>
 #include <X11/Xatom.h>
 #include <X11/Xutil.h>
 #include <X11/Xfuncs.h>
 #include <X11/keysym.h>
-#include <stdio.h>
 #ifdef MULTIBUFFER
 #include <X11/extensions/multibuf.h>
 #endif /* MULTIBUFFER */
@@ -113,10 +121,6 @@
 };
 #define NumberPolygons sizeof(polygons)/sizeof(polygons[0])
 
-#include <stdlib.h>
-#include <time.h>	/* for time_t */
-#include <sys/time.h>	/* for struct timeval */
-
 typedef double Transform3D[4][4];
 
 typedef struct {
@@ -1328,3 +1332,7 @@
 	XCloseDisplay(dpy);
 	return 0;
 }
+
+#ifdef __cplusplus
+} /* extern "C" */
+#endif
