--- src/module.cc.orig	2008-11-27 06:56:41 +0100
+++ src/module.cc	2009-06-13 22:50:33 +0200
@@ -117,14 +117,23 @@
 
 static ResDecl res_mod_path("module:path", PKGLIBDIR"/"VERSION":"PKGLIBDIR, 0,0);
 
+/* dlopen can take a file without extension and automatically do the
+ * right thing, however that doesn't fit with this code that tries to
+ * stat before the dlopen call, hence need some help here */
+#if defined(__MACH__) && defined(__APPLE__)
+static char *ext = ".bundle";
+#else
+static char *ext = ".so";
+#endif
+
 static int access_so(xstring &fullpath)
 {
    int res=access(fullpath,F_OK);
    if(res==-1)
    {
       int len=fullpath.length();
-      if(len>3 && strcmp(fullpath+len-3,".so"))
-	 fullpath.append(".so");
+      if(len>strlen(ext) && strcmp(fullpath+len-strlen(ext),ext))
+	 fullpath.append(ext);
       res=access(fullpath,F_OK);
    }
    return res;
