diff -ruN pngcrush-1.6.10-nolib.orig/Makefile  pngcrush-1.6.10-nolib/Makefile
--- pngcrush-1.6.10-nolib.orig/Makefile	1970-01-01 02:00:00.000000000 +0200
+++ pngcrush-1.6.10-nolib/Makefile	2008-07-30 23:01:16.000000000 +0300
@@ -0,0 +1,69 @@
+# Sample makefile for pngcrush using gcc and GNU make.
+# Glenn Randers-Pehrson
+# Last modified:  19 February 2005
+# Kapil Hari Paranjape
+# Last modified:  12 June 2008
+#
+# Invoke this makefile from a shell prompt in the usual way; for example:
+#
+#	make -f Makefile
+#
+# This makefile builds a dynamically linked executable.
+
+# macros --------------------------------------------------------------------
+
+GAS_VERSION := $(shell as --version | sed -n -e's/GNU assembler (.*) //p')
+
+PNGINC = /usr/include
+
+CC ?= gcc
+RM = rm -f
+
+CFLAGS += -I. -Wall
+
+O = .o
+E =
+
+# additional defines
+DEFINES = -DPNG_USE_PNGGCCRD -DPNG_iCCP_SUPPORTED \
+	-DPNG_iTXt_SUPPORTED -DPNG_USE_GLOBAL_ARRAYS \
+	-DGAS_VERSION="\"${GAS_VERSION}\"" \
+
+PNGCRUSH  = pngcrush
+
+LIBS = -lpng -lz
+
+OBJS  = pngcrush$(O)
+
+EXES = $(PNGCRUSH)$(E)
+
+
+# implicit make rules -------------------------------------------------------
+
+.c$(O): png.h cexcept.h $(ZHDR)
+	$(CC) -c $(CFLAGS) $(DEFINES) $<
+
+
+# dependencies --------------------------------------------------------------
+
+all:  $(EXES)
+
+png.h:
+	ln -s $(PNGINC)/png.h png.h
+
+pngcrush$(O): pngcrush.c cexcept.h
+	$(CC) -c $(CFLAGS) $(DEFINES) $<
+
+$(PNGCRUSH)$(E): $(OBJS)
+	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LIBS)
+
+# maintenance ---------------------------------------------------------------
+
+clean:
+	$(RM) $(EXES) $(OBJS) png.h
+
+install:
+	mkdir -p $(DESTDIR)/usr/bin/
+	cp $(PNGCRUSH)$(E) $(DESTDIR)/usr/bin/
+	chmod 0755 $(DESTDIR)/usr/bin/$(PNGCRUSH)$(E)
+

diff -ruN pngcrush-1.6.10-nolib.orig/pngcrush.c  pngcrush-1.6.10-nolib/pngcrush.c
--- pngcrush-1.6.10-nolib.orig/pngcrush.c	2008-10-15 01:00:58.000000000 +0400
+++ pngcrush-1.6.10-nolib/pngcrush.c	2008-10-15 01:02:30.000000000 +0400
@@ -2821,7 +2821,7 @@
 #if !defined(PNG_cHRM_SUPPORTED) || !defined(PNG_hIST_SUPPORTED) || \
     !defined(PNG_iCCP_SUPPORTED) || !defined(PNG_sCAL_SUPPORTED) || \
     !defined(PNG_pCAL_SUPPORTED) || !defined(PNG_sPLT_SUPPORTED) || \
-    !defined(PNG_tIME_SUPPORTED)
+    !defined(PNG_sTER_SUPPORTED) || !defined(PNG_tIME_SUPPORTED)
                         png_byte chunk_name[5];
                         chunk_name[4] = '\0';
 #endif

