--- test/test_ssl.py
+++ test/test_ssl.py
@@ -10,12 +10,7 @@
 from os import makedirs, symlink
 from os.path import join
 
-try:
-    # Prefer Twisted's TestCase, since it supports things like skips.
-    from twisted.trial.unittest import TestCase
-except ImportError:
-    # Fall back to the stdlib TestCase though, since it kind of works.
-    from unittest import TestCase, main
+from unittest import TestCase, main
 
 from OpenSSL.crypto import TYPE_RSA, FILETYPE_PEM, PKey, dump_privatekey, load_certificate, load_privatekey
 from OpenSSL.SSL import WantReadError, Context, Connection, Error
@@ -221,7 +221,7 @@
         self._load_verify_locations_test(None, capath)
 
 
-    def test_set_default_verify_paths(self):
+    def _test_set_default_verify_paths(self):
         """
         L{Context.set_default_verify_paths} causes the platform-specific CA
         certificate locations to be used for verification purposes.
