--- pyOpenSSL-0.7/test/test_crypto.py	2008-04-11 18:45:54.000000000 +0300
+++ pyOpenSSL-0.7/test/test_crypto.py	2008-04-16 13:14:19.088659137 +0300
@@ -534,3 +534,7 @@
         self.assertEqual(
             cert.digest("md5"),
             "A8:EB:07:F8:53:25:0A:F2:56:05:C5:A5:C4:C4:C7:15")
+
+if __name__ == '__main__':
+    import unittest
+    unittest.main()
--- pyOpenSSL-0.7/test/test_ssl.py	2008-03-22 00:34:42.000000000 +0200
+++ pyOpenSSL-0.7/test/test_ssl.py	2008-04-16 13:15:33.345659292 +0300
@@ -35,3 +35,7 @@
         ctx = Context(TLSv1_METHOD)
         ctx.use_privatekey(key)
         self.assertRaises(TypeError, ctx.use_privatekey, "")
+
+if __name__ == '__main__':
+    import unittest
+    unittest.main()
