--- mirrorselect~	2007-08-20 19:54:09 +0200
+++ mirrorselect	2007-08-20 19:54:01 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@GENTOO_PORTAGE_EPREFIX@/usr/bin/python
 
 """
 Mirrorselect 1.x written by Colin Kingsley (tercel@gentoo.org)
@@ -10,6 +10,7 @@
 import os, time, popen2, re, shutil, signal
 from HTMLParser import HTMLParser
 from optparse import IndentedHelpFormatter, OptionParser
+from portage.const import rootuid
 	
 class Output:
 	"""Handles text output. Only prints messages with level <= verbosity.
@@ -764,7 +765,7 @@
 			'You do not appear to have netselect on your system. '
 			'You must use the -D flag')
 	
-	if (os.getuid() != 0) and not options.output:
+	if (os.getuid() != rootuid) and not options.output:
 		output.print_err('Must be root to write to /etc/make.conf!\n')
 	
 	if args:
@@ -777,7 +778,7 @@
 output = Output()	#the only FUCKING global. Damnit.
 def main(argv):
 	"""Lets Rock!"""
-	config_path = '/etc/make.conf'
+	config_path = '@GENTOO_PORTAGE_EPREFIX@/etc/make.conf'
 	list_url = 'http://www.gentoo.org/main/en/mirrors.xml?passthru=1'
 
 	signal.signal(signal.SIGINT, handler)
