--- main.py.orig	2009-08-03 19:02:08 +0200
+++ main.py	2009-10-09 18:03:12 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@GENTOO_PORTAGE_EPREFIX@/usr/bin/python
 
 # Mirrorselect 1.x
 # Tool for selecting Gentoo source and rsync mirrors.
@@ -39,6 +39,7 @@
 from optparse import IndentedHelpFormatter, OptionParser
 from mirrorselect.mirrorparser3 import MirrorParser3, MIRRORS_3_XML
 import codecs
+from portage.const import rootuid
 
 class Output(object):
 	"""Handles text output. Only prints messages with level <= verbosity.
@@ -861,7 +862,7 @@
 			'You do not appear to have netselect on your system. '
 			'You must use the -D flag')
 	
-	if (os.getuid() != 0) and not options.output:
+	if (os.getuid() != rootuid) and not options.output:
 		output.print_err('Must be root to write to /etc/make.conf!\n')
 	
 	if args:
@@ -874,7 +875,7 @@
 output = Output()	#the only FUCKING global. Damnit.
 def main(argv):
 	"""Lets Rock!"""
-	config_path = '/etc/make.conf'
+	config_path = '@GENTOO_PORTAGE_EPREFIX@/etc/make.conf'
 
 	signal.signal(signal.SIGINT, handler)
 
