--- Jamroot.orig	2009-04-03 19:23:02.876426258 +0200
+++ Jamroot	2009-04-03 19:37:05.899697787 +0200
@@ -444,14 +444,10 @@
         }
         return $(result) ;
     }    
-            
-    generate stage-unversioned : stage-proper : 
-      <generating-rule>@make-unversioned-links ;
-    explicit stage-unversioned ;
-    
-    generate install-unversioned : install-proper : 
-      <generating-rule>@make-unversioned-links ;
-    explicit install-unversioned ;
+
+
+    alias stage-unversioned ;
+    alias install-unversioned ;
 }
 else
 {
