Index: boost/graph/distributed/detail/mpi_process_group.ipp
===================================================================
--- boost/graph/distributed/detail/mpi_process_group.ipp	(revision 57744)
+++ boost/graph/distributed/detail/mpi_process_group.ipp	(working copy)
@@ -28,6 +28,7 @@
 #include <queue>
 #include <stack>
 #include <boost/graph/distributed/detail/tag_allocator.hpp>
+#include <stdio.h>

 // #define PBGL_PROCESS_GROUP_DEBUG

