diff -u findutils-4.5.1/gnulib/lib/fpurge.c findutils-4.5.1/gnulib/lib/fpurge.c
--- findutils-4.5.1/gnulib/lib/fpurge.c	2007-12-02 11:57:51.000000000 +0000
+++ findutils-4.5.1/gnulib/lib/fpurge.c	2008-05-23 21:33:50.000000000 +0100
@@ -113,6 +113,12 @@
     /* fp->_Buf <= fp->_Next <= fp->_Rend */
     fp->_Rend = fp->_Next;
   return 0;
+# elif defined __MINT__
+  fp->__pushback_bufp = 0;
+  if (fp->__mode.__write)
+    fp->__put_limit = fp->__buffer;
+  fp->__bufp = fp->__get_limit;
+  return 0;
 # else
  #error "Please port gnulib fpurge.c to your platform! Look at the definitions of fflush, setvbuf and ungetc on your system, then report this to bug-gnulib."
 # endif
diff -u findutils-4.5.1/gnulib/lib/freadahead.c findutils-4.5.1/gnulib/lib/freadahead.c
--- findutils-4.5.1/gnulib/lib/freadahead.c	2007-12-02 11:57:51.000000000 +0000
+++ findutils-4.5.1/gnulib/lib/freadahead.c	2008-05-23 21:37:24.000000000 +0100
@@ -78,6 +78,12 @@
 	 + (fp->_Mode & 0x4000 /* _MBYTE */
 	    ? (fp->_Back + sizeof (fp->_Back)) - fp->_Rback
 	    : 0);
+#elif defined __MINT__
+  if (fp->__mode.__write)
+    return 0;
+  if (fp->__pushed_back)
+    return (fp->__get_limit - fp->__pushback_bufp + 1);
+  return (fp->__get_limit - fp->__bufp);
 #else
  #error "Please port gnulib freadahead.c to your platform! Look at the definition of fflush, fread, ungetc on your system, then report this to bug-gnulib."
 #endif
diff -u findutils-4.5.1/gnulib/lib/freading.c findutils-4.5.1/gnulib/lib/freading.c
--- findutils-4.5.1/gnulib/lib/freading.c	2007-12-02 11:57:51.000000000 +0000
+++ findutils-4.5.1/gnulib/lib/freading.c	2008-05-23 21:52:11.000000000 +0100
@@ -42,6 +42,13 @@
 #elif defined __QNX__               /* QNX */
   return ((fp->_Mode & 0x2 /* _MOPENW */) == 0
 	  || (fp->_Mode & 0x1000 /* _MREAD */) != 0);
+#elif defined __MINT__
+  if (!fp->__mode.__write && fp->__mode.__read)
+    return 1;
+  return (fp->__mode.__read &&
+          fp->__get_limit > fp->__bufp &&
+          fp->__get_limit > fp->__put_limit &&
+          fp->__bufp > fp->__buffer);
 #else
  #error "Please port gnulib freading.c to your platform!"
 #endif
diff -u findutils-4.5.1/gnulib/lib/fseeko.c findutils-4.5.1/gnulib/lib/fseeko.c
--- findutils-4.5.1/gnulib/lib/fseeko.c	2008-02-13 23:31:13.000000000 +0000
+++ findutils-4.5.1/gnulib/lib/fseeko.c	2008-05-23 22:37:13.000000000 +0100
@@ -94,6 +94,8 @@
   if ((fp->_Mode & _MWRITE ? fp->_Next == fp->_Buf : fp->_Next == fp->_Rend)
       && fp->_Rback == fp->_Back + sizeof (fp->_Back)
       && fp->_Rsave == NULL)
+#elif defined __MINT__
+  if (0)
 #else
   #error "Please port gnulib fseeko.c to your platform! Look at the code in fpurge.c, then report this to bug-gnulib."
 #endif
