diff -ur gnulib/lib/fpurge.c gnulib/lib/fpurge.c
--- gnulib/lib/fpurge.c	2008-11-30 22:03:02.000000000 +0000
+++ gnulib/lib/fpurge.c	2008-12-29 12:31:20.000000000 +0000
@@ -114,6 +114,12 @@
     /* fp->_Buf <= fp->_Next <= fp->_Rend */
     fp->_Rend = fp->_Next;
   return 0;
+# elif defined __MINT__
+  fp->__pushback_bufp = 0;
+  if (fp->__mode.__write)
+    fp->__put_limit = fp->__buffer;
+  fp->__bufp = fp->__get_limit;
+  return 0;
 # else
  #error "Please port gnulib fpurge.c to your platform! Look at the definitions of fflush, setvbuf and ungetc on your system, then report this to bug-gnulib."
 # endif
diff -ur gnulib/lib/freadahead.c gnulib/lib/freadahead.c
--- gnulib/lib/freadahead.c	2008-11-30 22:03:02.000000000 +0000
+++ gnulib/lib/freadahead.c	2008-12-29 12:32:22.000000000 +0000
@@ -70,6 +70,12 @@
 	 + (fp->_Mode & 0x4000 /* _MBYTE */
 	    ? (fp->_Back + sizeof (fp->_Back)) - fp->_Rback
 	    : 0);
+#elif defined __MINT__
+  if (fp->__mode.__write)
+    return 0;
+  if (fp->__pushed_back)
+    return (fp->__get_limit - fp->__pushback_bufp + 1);
+  return (fp->__get_limit - fp->__bufp);
 #elif defined SLOW_BUT_NO_HACKS     /* users can define this */
   abort ();
   return 0;
diff -ur gnulib/lib/freading.c gnulib/lib/freading.c
--- gnulib/lib/freading.c	2008-11-30 22:03:02.000000000 +0000
+++ gnulib/lib/freading.c	2008-12-29 12:31:20.000000000 +0000
@@ -46,6 +46,13 @@
 #elif defined __QNX__               /* QNX */
   return ((fp->_Mode & 0x2 /* _MOPENW */) == 0
 	  || (fp->_Mode & 0x1000 /* _MREAD */) != 0);
+#elif defined __MINT__
+  if (!fp->__mode.__write && fp->__mode.__read)
+    return 1;
+  return (fp->__mode.__read &&
+          fp->__get_limit > fp->__bufp &&
+          fp->__get_limit > fp->__put_limit &&
+          fp->__bufp > fp->__buffer);
 #else
  #error "Please port gnulib freading.c to your platform!"
 #endif
diff -ur gnulib/lib/fseeko.c gnulib/lib/fseeko.c
--- gnulib/lib/fseeko.c	2008-11-30 22:05:25.000000000 +0000
+++ gnulib/lib/fseeko.c	2008-12-29 12:31:20.000000000 +0000
@@ -82,6 +82,8 @@
   if ((fp->_Mode & _MWRITE ? fp->_Next == fp->_Buf : fp->_Next == fp->_Rend)
       && fp->_Rback == fp->_Back + sizeof (fp->_Back)
       && fp->_Rsave == NULL)
+#elif defined __MINT__
+  if (0)
 #else
   #error "Please port gnulib fseeko.c to your platform! Look at the code in fpurge.c, then report this to bug-gnulib."
 #endif
