diff -ruN findutils-4.3.11.orig/gnulib/m4/ls-mntd-fs.m4 findutils-4.3.11/gnulib/m4/ls-mntd-fs.m4
--- findutils-4.3.11.orig/gnulib/m4/ls-mntd-fs.m4	Fri Jan 18 10:31:32 2008
+++ findutils-4.3.11/gnulib/m4/ls-mntd-fs.m4	Fri Jan 18 10:29:45 2008
@@ -326,6 +326,12 @@
   fi
 fi
 
+case $host_os in
+interix*)
+  ac_list_mounted_fs="not_supported_on_windows"
+  ;;
+esac
+
 if test -z "$ac_list_mounted_fs"; then
   AC_MSG_ERROR([could not determine how to read list of mounted file systems])
   # FIXME -- no need to abort building the whole package
diff -ruN findutils-4.3.11.orig/configure findutils-4.3.11/configure
--- findutils-4.3.11.orig/configure	Fri Jan 18 10:31:24 2008
+++ findutils-4.3.11/configure	Fri Jan 18 10:32:20 2008
@@ -23007,6 +23007,12 @@
   fi
 fi
 
+case $host_os in
+interix*)
+  ac_list_mounted_fs="not_supported_on_windows"
+  ;;
+esac
+
 if test -z "$ac_list_mounted_fs"; then
   { { echo "$as_me:$LINENO: error: could not determine how to read list of mounted file systems" >&5
 echo "$as_me: error: could not determine how to read list of mounted file systems" >&2;}
diff -ruN findutils-4.3.11.orig/find/fstype.c findutils-4.3.11/find/fstype.c
--- findutils-4.3.11.orig/find/fstype.c	Fri Jan 18 10:31:32 2008
+++ findutils-4.3.11/find/fstype.c	Fri Jan 18 10:29:45 2008
@@ -193,8 +193,10 @@
 static struct mount_entry *
 must_read_fs_list(bool need_fs_type)
 {
+#ifndef __INTERIX
   struct mount_entry *entries = read_file_system_list(need_fs_type);
   if (NULL == entries)
+#endif
     {
       /* We cannot determine for sure which file we were trying to
        * use because gnulib has extracted all that stuff away. 
@@ -202,7 +204,11 @@
        */
       error(1, 0, "Cannot read mounted file system list");
     }
+#ifndef __INTERIX
   return entries;
+#else
+  return NULL;
+#endif
 }
 
 
@@ -295,6 +301,7 @@
    * result of this operation.   If we can't get the fs list 
    * but we never need the information, there is no need to fail.
    */
+#ifndef __INTERIX
   for (entry = entries = read_file_system_list(false);
        entry;
        entry = entry->me_next)
@@ -306,6 +313,7 @@
     }
   free_file_system_list(entries);
   *n = used;
+#endif
   return result;
 }
 
diff -ruN findutils-4.3.11.orig/gnulib/lib/mountlist.c findutils-4.3.11/gnulib/lib/mountlist.c
--- findutils-4.3.11.orig/gnulib/lib/mountlist.c	Fri Jan 18 10:31:32 2008
+++ findutils-4.3.11/gnulib/lib/mountlist.c	Fri Jan 18 10:29:45 2008
@@ -341,6 +341,9 @@
 struct mount_entry *
 read_file_system_list (bool need_fs_type)
 {
+#ifdef __INTERIX
+  return NULL;
+#else
   struct mount_entry *mount_list;
   struct mount_entry *me;
   struct mount_entry **mtail = &mount_list;
@@ -881,4 +884,5 @@
     errno = saved_errno;
     return NULL;
   }
+#endif /* !__INTERIX */
 }
