Ripped from Fedora.

--- gawk-3.1.4/awkgram.y.nextc	2004-11-09 15:53:03.355118792 +0100
+++ gawk-3.1.4/awkgram.y	2004-11-09 16:05:46.091165288 +0100
@@ -1433,6 +1433,11 @@
 nextc(void)
 {
 	if (gawk_mb_cur_max > 1)	{
+		if (!lexptr || lexptr >= lexend)
+			if (!get_src_buf()) {
+				return EOF;
+			}
+
 		/* Update the buffer index.  */
 		cur_ring_idx = (cur_ring_idx == RING_BUFFER_SIZE - 1)? 0 :
 			cur_ring_idx + 1;
@@ -1444,11 +1449,6 @@
 			mbstate_t tmp_state;
 			size_t mbclen;
 
-			if (!lexptr || lexptr >= lexend)
-				if (!get_src_buf()) {
-					return EOF;
-				}
-
 			for (idx = 0 ; lexptr + idx < lexend ; idx++) {
 				tmp_state = cur_mbstate;
 				mbclen = mbrlen(lexptr, idx + 1, &tmp_state);
