Patch by Alan Hourihane for FreeMiNT

--- gawk-3.1.6/array.c
+++ gawk-3.1.6/array.c
@@ -702,7 +702,7 @@
 	 * just jumping from 8K to 64K.
 	 */
 	static const long sizes[] = { 13, 127, 1021, 8191, 16381, 32749, 65497,
-#if ! defined(MSDOS) && ! defined(OS2) && ! defined(atarist)
+#if ! defined(MSDOS) && ! defined(OS2) 
 				131101, 262147, 524309, 1048583, 2097169,
 				4194319, 8388617, 16777259, 33554467, 
 				67108879, 134217757, 268435459, 536870923,
--- gawk-3.1.6/awk.h
+++ gawk-3.1.6/awk.h
@@ -195,9 +195,9 @@
 #define GFMT_WORKAROUND
 #endif	/* NeXT */
 
-#if defined(atarist) || defined(VMS)
+#if defined(VMS)
 #include <unixlib.h>
-#endif	/* atarist || VMS */
+#endif	/* VMS */
 
 #if ! defined(MSDOS) && ! defined(OS2) && ! defined(WIN32) && ! defined(__EMX__) && ! defined(__CYGWIN__) && ! defined(O_BINARY) /*duh*/
 #define O_BINARY	0
@@ -216,10 +216,6 @@
 #include "vms/redirect.h"
 #endif  /*VMS*/
 
-#ifdef atarist
-#include "unsupported/atari/redirect.h"
-#endif
-
 #ifndef HAVE_VPRINTF
 /* if you don't have vprintf, try this and cross your fingers. */
 #ifdef	HAVE_DOPRNT
@@ -1235,19 +1231,7 @@
 extern int strncasecmp P((const char *s1, const char *s2, register size_t n));
 #endif
 
-#if defined(atarist)
-#if defined(PIPES_SIMULATED)
-/* unsupported/atari/tmpnam.c */
-extern char *tmpnam P((char *buf));
-extern char *tempnam P((const char *path, const char *base));
-#else
-#include <wait.h>
-#endif
-#include <fcntl.h>
-#define INVALID_HANDLE  (__SMALLEST_VALID_HANDLE - 1)
-#else
 #define INVALID_HANDLE (-1)
-#endif /* atarist */
 
 #ifdef HAVE_SYS_WAIT_H
 #include <sys/wait.h>
--- gawk-3.1.6/custom.h
+++ gawk-3.1.6/custom.h
@@ -102,3 +102,8 @@
 #define HAVE_TZSET 1
 #define _TZSET 1
 #endif
+
+/* It's slightly busted, but we'll use it anyway */
+#ifdef __MINT__
+#define HAVE_MKTIME 1
+#endif
--- gawk-3.1.6/gawkmisc.c
+++ gawk-3.1.6/gawkmisc.c
@@ -40,15 +40,11 @@
 #if defined(VMS)
 #include "vms/gawkmisc.vms"
 #else /* not VMS */
-#if defined(atarist)
-#include "unsupported/atari/gawkmisc.atr"
-#else /* not atarist */
 #if defined(TANDEM)
 #include "tmiscc"
 #else /* not TANDEM */
 #include "posix/gawkmisc.c"
 #endif /* not TANDEM */
-#endif /* not atarist */
 #endif /* not VMS */
 #endif /* not MSDOS, not OS2, not WIN32 */
 #endif /* not __EMX__ */
--- gawk-3.1.6/io.c
+++ gawk-3.1.6/io.c
@@ -111,10 +111,6 @@
 
 #endif /* HAVE_SOCKETS */
 
-#ifdef atarist
-#include <stddef.h>
-#endif
-
 #if defined(GAWK_AIX)
 #undef TANDEM	/* AIX defines this in one of its header files */
 #undef MSDOS	/* For good measure, */
--- gawk-3.1.6/replace.c
+++ gawk-3.1.6/replace.c
@@ -6,21 +6,8 @@
 #include "awk.h"
 
 
-#ifdef atarist
-/*
- * this will work with gcc compiler - for other compilers you may
- * have to replace path separators in this file into backslashes
- */
-#include "unsupported/atari/stack.c"
-#include "unsupported/atari/tmpnam.c"
-#endif /* atarist */
-
 #ifndef HAVE_SYSTEM
-#ifdef atarist
-#include "unsupported/atari/system.c"
-#else
 #include "missing_d/system.c"
-#endif
 #endif /* HAVE_SYSTEM */
 
 #ifndef HAVE_MEMCMP
