--- gawk-3.1.5/bisonfix.awk.orig	2006-11-16 12:00:58.140294000 +0100
+++ gawk-3.1.5/bisonfix.awk	2006-11-16 12:01:20.870294000 +0100
@@ -19,16 +19,4 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
 
-/^#if \(! defined \(yyoverflow\) \\/ {
-	line = $0
-	sub(/\\$/, "", line)
-	getline line2
-	sub(/\\$/, "", line2)
-	getline line3
-
-	line = line line2 line3
-	print line
-	next
-}
-
 { print }
