diff -Nru mDNSResponder-107.6.vanilla/Clients/Java/nmakefile mDNSResponder-107.6/Clients/Java/nmakefile
--- mDNSResponder-107.6.vanilla/Clients/Java/nmakefile	2007-04-28 15:32:09.000000000 +0200
+++ mDNSResponder-107.6/Clients/Java/nmakefile	2007-04-28 15:32:50.000000000 +0200
@@ -94,7 +94,7 @@
 JAVASRC	= .
 .SUFFIXES : .java
 {$(JAVASRC)}.java{$(BAOBJ)}.class:	
-	$(JAVAC) -d $(BAOBJ) -classpath $(BAOBJ);$(DNS_SD) $<
+	$(JAVAC) $(JAVACFLAGS) -d $(BAOBJ) -classpath $(BAOBJ);$(DNS_SD) $<
 {$(JAVASRC)}.java{$(SCOBJ)}.class:	
-	$(JAVAC) -d $(SCOBJ) -classpath $(SCOBJ);$(DNS_SD) $<
+	$(JAVAC) $(JAVACFLAGS) -d $(SCOBJ) -classpath $(SCOBJ);$(DNS_SD) $<
 
diff -Nru mDNSResponder-107.6.vanilla/mDNSPosix/Makefile mDNSResponder-107.6/mDNSPosix/Makefile
--- mDNSResponder-107.6.vanilla/mDNSPosix/Makefile	2007-04-28 15:32:09.000000000 +0200
+++ mDNSResponder-107.6/mDNSPosix/Makefile	2007-04-28 15:33:34.000000000 +0200
@@ -585,7 +585,7 @@
 JAVAH = $(JDK)/bin/javah
 JAVADOC = $(JDK)/bin/javadoc
 JAR = $(JDK)/bin/jar
-JAVACFLAGS = $(CFLAGS) $(JAVACFLAGS_OS) -I$(JDK)/include
+JAVACFLAGS_C = $(CFLAGS) $(JAVACFLAGS_OS) -I$(JDK)/include
 
 Java: setup $(BUILDDIR)/dns_sd.jar $(BUILDDIR)/libjdns_sd.$(LDSUFFIX)
 	@echo "Java wrappers done"
@@ -610,10 +610,10 @@
 	$(JAR) -cf $@ -C $(OBJDIR) com
 
 $(BUILDDIR)/libjdns_sd.$(LDSUFFIX): $(JAVASRC)/JNISupport.c $(OBJDIR)/DNSSD.java.h setup libdns_sd
-	$(CC) -o $@ $< $(JAVACFLAGS) -I$(OBJDIR) -L$(BUILDDIR)
+	$(CC) -o $@ $< $(JAVACFLAGS_C) -I$(OBJDIR) -L$(BUILDDIR)
 
 $(OBJDIR)/com/apple/dnssd/%.class:	$(JAVASRC)/%.java
-	$(JAVAC) -d $(OBJDIR) -classpath $(OBJDIR) $<
+	$(JAVAC) $(JAVACFLAGS) -d $(OBJDIR) -classpath $(OBJDIR) $<
 
 $(OBJDIR)/DNSSD.java.h: $(OBJDIR)/com/apple/dnssd/DNSSD.class
 	$(JAVAH) -force -classpath $(OBJDIR) -o $@ \
