Index: sox-14.1.0/src/ffmpeg.c
===================================================================
--- sox-14.1.0.orig/src/ffmpeg.c
+++ sox-14.1.0/src/ffmpeg.c
@@ -86,7 +86,11 @@ static int stream_component_open(priv_t 
 
   codec = avcodec_find_decoder(enc->codec_id);
   enc->workaround_bugs = 1;
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
   enc->error_resilience = 1;
+#else
+  enc->error_recognition = 1;
+#endif
   if (!codec || avcodec_open(enc, codec) < 0)
     return -1;
   if (enc->codec_type != CODEC_TYPE_AUDIO) {
