Index: sox-14.1.0/src/Makefile.in
===================================================================
--- sox-14.1.0.orig/src/Makefile.in
+++ sox-14.1.0/src/Makefile.in
@@ -2899,7 +2899,6 @@ uninstall-am: uninstall-binPROGRAMS unin
 all: sox$(EXEEXT) play rec soxi sox_sample_test$(EXEEXT) example0$(EXEEXT) example1$(EXEEXT) example2$(EXEEXT)
 
 play rec soxi: sox$(EXEEXT)
-	./sox --help > /dev/null
 	test -f .libs/lt-sox && ( $(RM) $@ && $(LN_S) .libs/lt-sox $@ ) || $(LN_S) sox$(EXEEXT) $@
 
 install-exec-hook:
