--- sox-14.0.1/src/Makefile.am.orig	2008-07-06 23:59:55.000000000 -0700
+++ sox-14.0.1/src/Makefile.am	2008-07-06 23:59:38.000000000 -0700
@@ -291,7 +291,7 @@
 all: sox$(EXEEXT) play rec sox_sample_test$(EXEEXT) example1$(EXEEXT)
 
 play rec: sox$(EXEEXT)
-	./sox --help > /dev/null
+	@#./sox --help > /dev/null
 	test -f .libs/lt-sox && ( $(RM) $@ && $(LN_S) .libs/lt-sox $@ ) || $(LN_S) sox$(EXEEXT) $@
 
 install-exec-hook:
