From a61e394d59e96267c6cef12be796f6832d212c39 Mon Sep 17 00:00:00 2001
From: Paolo Bacchilega <paobac@src.gnome.org>
Date: Wed, 14 Jul 2010 11:51:57 +0000
Subject: do not link to the exiv2 extension if it's not built

[bug #624326]
---
diff --git a/extensions/exiv2_tools/Makefile.am b/extensions/exiv2_tools/Makefile.am
index 7bf5ae6..26a127f 100644
--- a/extensions/exiv2_tools/Makefile.am
+++ b/extensions/exiv2_tools/Makefile.am
@@ -5,7 +5,7 @@ SUBDIRS = data
 extensiondir = $(pkglibdir)/extensions
 extension_LTLIBRARIES = libexiv2_tools.la
 
-libexiv2_tools_la_SOURCES = 			\
+libexiv2_tools_la_SOURCES = 		\
 	exiv2-utils.h			\
 	exiv2-utils.cpp			\
 	gth-edit-exiv2-page.c		\
diff --git a/extensions/importer/Makefile.am b/extensions/importer/Makefile.am
index de5095c..87b2d06 100644
--- a/extensions/importer/Makefile.am
+++ b/extensions/importer/Makefile.am
@@ -45,7 +45,10 @@ libimporter_la_SOURCES = 			\
 
 libimporter_la_CFLAGS = $(GTHUMB_CFLAGS) -I$(top_srcdir) -I$(top_builddir)/gthumb 
 libimporter_la_LDFLAGS = $(EXTENSION_LIBTOOL_FLAGS)
-libimporter_la_LIBADD = $(GTHUMB_LIBS) ../catalogs/libcatalogs.la ../exiv2_tools/libexiv2_tools.la ../image_rotation/libimage_rotation.la
+libimporter_la_LIBADD = $(GTHUMB_LIBS) ../catalogs/libcatalogs.la ../image_rotation/libimage_rotation.la
+if ENABLE_EXIV2
+libimporter_la_LIBADD += ../exiv2_tools/libexiv2_tools.la
+endif
 libimporter_la_DEPENDENCIES = $(top_builddir)/gthumb/gthumb$(EXEEXT)
 
 extensioninidir = $(extensiondir)
--
cgit v0.8.3.1
