diff -ru gthumb-2.10.10.orig/libgthumb/file-utils.c gthumb-2.10.10/libgthumb/file-utils.c
--- gthumb-2.10.10.orig/libgthumb/file-utils.c	2008-10-22 13:18:29.771795250 +0200
+++ gthumb-2.10.10/libgthumb/file-utils.c	2008-10-22 13:25:10.977704838 +0200
@@ -2459,12 +2459,18 @@
         template = g_strconcat (best_folder, "/.gth-XXXXXX", NULL);
         g_free (best_folder);
         
-        result = mkdtemp (template);
+	/* mkdtemp not in Solaris 10 */
+	{
+	   int status;
 
-        if ((result == NULL) || (*result == '\0')) {
-                g_free (template);
-                result = NULL;
-        }
+	   result = mktemp(template);
+	   status = mkdir (result,0700);
+	    
+	   if ((result == NULL) || (*result == '\0') || (status != 0)) {
+	                   g_free (template);
+	          result = NULL;
+	   }
+	}
 
         return result;
 }
