diff -ur gcc-4.7.2-orig/gcc/configure gcc-4.7.2/gcc/configure
--- gcc-4.7.2-orig/gcc/configure	2014-04-11 21:54:51.000000000 +0200
+++ gcc-4.7.2/gcc/configure	2014-04-11 22:01:49.000000000 +0200
@@ -23323,16 +23323,12 @@
 	tls_first_major=2
 	tls_first_minor=17
 	;;
-  i[34567]86-*-* | x86_64-*-solaris2.1[0-9]*)
+  i[34567]86-*-*)
     case "$target" in
       i[34567]86-*-solaris2.*)
 	on_solaris=yes
 	tga_func=___tls_get_addr
 	;;
-      x86_64-*-solaris2.1[0-9]*)
-	on_solaris=yes
-	tga_func=__tls_get_addr
-        ;;
       *)
 	on_solaris=no
 	;;
@@ -23368,8 +23364,31 @@
 	leal	foo@ntpoff(%ecx), %eax"
     ;;
   x86_64-*-*)
-    conftest_s='
-	.section ".tdata","awT",@progbits
+    case "$target" in
+      x86_64-*-solaris2.1[0-9]*)
+	on_solaris=yes
+	tga_func=__tls_get_addr
+        ;;
+      *)
+	on_solaris=no
+	;;
+    esac
+    if test x$on_solaris = xyes && test x$gas_flag = xno; then
+      conftest_s='
+	.section .tdata,"awt",@progbits'
+      tls_first_major=0
+      tls_first_minor=0
+
+$as_echo "#define TLS_SECTION_ASM_FLAG 't'" >>confdefs.h
+
+    else
+      conftest_s='
+	.section ".tdata","awT",@progbits'
+      tls_first_major=2
+      tls_first_minor=14
+      tls_as_opt="--fatal-warnings"
+    fi
+    conftest_s="$conftest_s
 foo:	.long	25
 	.text
 	movq	%fs:0, %rax
@@ -23377,10 +23396,7 @@
 	leaq	foo@TLSLD(%rip), %rdi
 	leaq	foo@DTPOFF(%rax), %rdx
 	movq	foo@GOTTPOFF(%rip), %rax
-	movq	$foo@TPOFF, %rax'
-	tls_first_major=2
-	tls_first_minor=14
-	tls_as_opt=--fatal-warnings
+	movq	\$foo@TPOFF, %rax"
 	;;
   ia64-*-*)
     conftest_s='
diff -ur gcc-4.7.2-orig/gcc/configure.ac gcc-4.7.2/gcc/configure.ac
--- gcc-4.7.2-orig/gcc/configure.ac	2014-04-11 21:54:51.000000000 +0200
+++ gcc-4.7.2/gcc/configure.ac	2014-04-11 21:56:27.000000000 +0200
@@ -2871,16 +2871,12 @@
 	tls_first_major=2
 	tls_first_minor=17
 	;;
-  i[34567]86-*-* | x86_64-*-solaris2.1[0-9]*)
+  i[34567]86-*-*)
     case "$target" in
       i[34567]86-*-solaris2.*)
 	on_solaris=yes
 	tga_func=___tls_get_addr
 	;;
-      x86_64-*-solaris2.1[0-9]*)
-	on_solaris=yes
-	tga_func=__tls_get_addr
-        ;;
       *)
 	on_solaris=no
 	;;
@@ -2917,8 +2913,32 @@
 	leal	foo@ntpoff(%ecx), %eax"
     ;;
   x86_64-*-*)
-    conftest_s='
-	.section ".tdata","awT",@progbits
+    case "$target" in
+      x86_64-*-solaris2.1[0-9]*)
+	on_solaris=yes
+	tga_func=__tls_get_addr
+        ;;
+      *)
+	on_solaris=no
+	;;
+    esac
+    if test x$on_solaris = xyes && test x$gas_flag = xno; then
+      conftest_s='
+	.section .tdata,"awt",@progbits'
+      tls_first_major=0
+      tls_first_minor=0
+changequote([,])dnl
+      AC_DEFINE(TLS_SECTION_ASM_FLAG, 't',
+[Define to the flag used to mark TLS sections if the default (`T') doesn't work.])
+changequote(,)dnl
+    else
+      conftest_s='
+	.section ".tdata","awT",@progbits'
+      tls_first_major=2
+      tls_first_minor=14
+      tls_as_opt="--fatal-warnings"
+    fi
+    conftest_s="$conftest_s
 foo:	.long	25
 	.text
 	movq	%fs:0, %rax
@@ -2926,10 +2946,7 @@
 	leaq	foo@TLSLD(%rip), %rdi
 	leaq	foo@DTPOFF(%rax), %rdx
 	movq	foo@GOTTPOFF(%rip), %rax
-	movq	$foo@TPOFF, %rax'
-	tls_first_major=2
-	tls_first_minor=14
-	tls_as_opt=--fatal-warnings
+	movq	\$foo@TPOFF, %rax"
 	;;
   ia64-*-*)
     conftest_s='
