--- texmf-update.orig	2007-12-03 15:33:34.019436271 +0100
+++ texmf-update	2007-12-03 16:54:19.160771307 +0100
@@ -1,11 +1,11 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 #
 # Utility to update Gentoo TeXLive distribution configuration files
 #
 
 echo "Configuring TeXLive ..."
 
-PATH=/bin:/usr/bin
+PATH="@GENTOO_PORTAGE_EPREFIX@"/bin:"@GENTOO_PORTAGE_EPREFIX@"/usr/bin
 
 # Fix for all those with altered umask for root
 umask 022
@@ -29,31 +29,31 @@
 done
 
 if [ "$TEXINPUTS" ]; then
-	if $(echo ${TEXINPUTS} | grep '/usr/share/texmf' &>/dev/null) ; then
-		export TEXINPUTS=$(echo ${TEXINPUTS} | sed -e 's|/usr/share/texmf/*:\?||g')
-	elif $(echo ${TEXINPUTS} | grep '/var/lib/texmf' &>/dev/null) ; then
-		export TEXINPUTS=$(echo ${TEXINPUTS} | sed -e 's|/var/lib/texmf/*:\?||g')
+	if $(echo ${TEXINPUTS} | grep '@GENTOO_PORTAGE_EPREFIX@/usr/share/texmf' &>/dev/null) ; then
+		export TEXINPUTS=$(echo ${TEXINPUTS} | sed -e 's|@GENTOO_PORTAGE_EPREFIX@/usr/share/texmf/*:\?||g')
+	elif $(echo ${TEXINPUTS} | grep '@GENTOO_PORTAGE_EPREFIX@/var/lib/texmf' &>/dev/null) ; then
+		export TEXINPUTS=$(echo ${TEXINPUTS} | sed -e 's|@GENTOO_PORTAGE_EPREFIX@/var/lib/texmf/*:\?||g')
 	fi
 fi
 
-if [ -d /etc/texmf/texmf.d ]; then
-    echo "Generating /etc/texmf/web2c/texmf.cnf from /etc/texmf/texmf.d ..."
-    cat /etc/texmf/texmf.d/*.cnf > "/etc/texmf/web2c/texmf.cnf"
+if [ -d "@GENTOO_PORTAGE_EPREFIX@"/etc/texmf/texmf.d ]; then
+    echo "Generating @GENTOO_PORTAGE_EPREFIX@/etc/texmf/web2c/texmf.cnf from @GENTOO_PORTAGE_EPREFIX@/etc/texmf/texmf.d ..."
+    cat "@GENTOO_PORTAGE_EPREFIX@"/etc/texmf/texmf.d/*.cnf > "@GENTOO_PORTAGE_EPREFIX@/etc/texmf/web2c/texmf.cnf"
 fi
 
-if [ -d /etc/texmf/fmtutil.d ]; then
-    echo "Generating /etc/texmf/web2c/fmtutil.cnf from /etc/texmf/fmtutil.d ..."
-    cat /etc/texmf/fmtutil.d/*.cnf > "/etc/texmf/web2c/fmtutil.cnf"
+if [ -d "@GENTOO_PORTAGE_EPREFIX@"/etc/texmf/fmtutil.d ]; then
+    echo "Generating @GENTOO_PORTAGE_EPREFIX@/etc/texmf/web2c/fmtutil.cnf from @GENTOO_PORTAGE_EPREFIX@/etc/texmf/fmtutil.d ..."
+    cat "@GENTOO_PORTAGE_EPREFIX@"/etc/texmf/fmtutil.d/*.cnf > "@GENTOO_PORTAGE_EPREFIX@/etc/texmf/web2c/fmtutil.cnf"
 fi
 
-if [ -d /etc/texmf/updmap.d ]; then
-    echo "Generating /etc/texmf/web2c/updmap.cfg from /etc/texmf/updmap.d ..."
-    cat /etc/texmf/updmap.d/*.cfg > "/etc/texmf/web2c/updmap.cfg"
+if [ -d "@GENTOO_PORTAGE_EPREFIX@"/etc/texmf/updmap.d ]; then
+    echo "Generating @GENTOO_PORTAGE_EPREFIX@/etc/texmf/web2c/updmap.cfg from @GENTOO_PORTAGE_EPREFIX@/etc/texmf/updmap.d ..."
+    cat "@GENTOO_PORTAGE_EPREFIX@"/etc/texmf/updmap.d/*.cfg > "@GENTOO_PORTAGE_EPREFIX@/etc/texmf/web2c/updmap.cfg"
 fi
 
-if [ -d /etc/texmf/dvips.d ]; then
-    echo "Generating /etc/texmf/dvips/config/config.ps from /etc/texmf/dvips.d ..."
-    cat /etc/texmf/dvips.d/*.ps > "/etc/texmf/dvips/config/config.ps"
+if [ -d "@GENTOO_PORTAGE_EPREFIX@"/etc/texmf/dvips.d ]; then
+    echo "Generating @GENTOO_PORTAGE_EPREFIX@/etc/texmf/dvips/config/config.ps from @GENTOO_PORTAGE_EPREFIX@/etc/texmf/dvips.d ..."
+    cat "@GENTOO_PORTAGE_EPREFIX@"/etc/texmf/dvips.d/*.ps > "@GENTOO_PORTAGE_EPREFIX@/etc/texmf/dvips/config/config.ps"
 fi
 
 echo "Generating ls-R files"
