--- 05searchpaths.cnf
+++ 05searchpaths.cnf
@@ -9,18 +9,18 @@
 
 % The tree containing the runtime files closely related to the specific
 % program version used:
-TEXMFMAIN = /usr/share/texmf
+TEXMFMAIN = @GENTOO_PORTAGE_EPREFIX@/usr/share/texmf
 
 % The main distribution tree:
-TEXMFDIST = /usr/share/texmf-dist
+TEXMFDIST = @GENTOO_PORTAGE_EPREFIX@/usr/share/texmf-dist
 
 % The Gentoo site tree
-TEXMFSITE = /usr/share/texmf-site
+TEXMFSITE = @GENTOO_PORTAGE_EPREFIX@/usr/share/texmf-site
 
 % A place for local additions to a "standard" texmf tree.
 % This tree is not used for local configuration maintained by
 % texconfig, it uses TEXMFCONFIG below.
-TEXMFLOCAL = /usr/local/share/texmf
+TEXMFLOCAL = @GENTOO_PORTAGE_EPREFIX@/usr/local/share/texmf
 
 % TEXMFSYSVAR, where texconfig-sys stores variable runtime data.
 % With teTeX-3.0 or later, this must be set.
@@ -28,7 +28,7 @@
 %   TEXMFSYSVAR = $TEXMFMAIN
 % For using a separate tree:
 %   TEXMFSYSVAR = $SELFAUTOPARENT/texmf-var
-TEXMFSYSVAR = /var/lib/texmf
+TEXMFSYSVAR = @GENTOO_PORTAGE_EPREFIX@/var/lib/texmf
 
 % TEXMFSYSCONFIG, where texconfig-sys stores configuration data.
 % With teTeX-3.0 or later, this must be set.
@@ -36,7 +36,7 @@
 %   TEXMFSYSCONFIG = $TEXMFMAIN
 % For using a separate tree:
 %   TEXMFSYSCONFIG = $SELFAUTOPARENT/texmf-config
-TEXMFSYSCONFIG = /etc/texmf
+TEXMFSYSCONFIG = @GENTOO_PORTAGE_EPREFIX@/etc/texmf
 
 % User texmf trees are allowed as follows.
 % This used to be HOMETEXMF.
@@ -78,7 +78,7 @@
 % Where generated fonts may be written.  This tree is used when the sources
 % were found in a system tree and either that tree wasn't writable, or the
 % varfonts feature was enabled in MT_FEATURES in mktex.cnf.
-VARTEXFONTS = /var/tmp/texfonts
+VARTEXFONTS = @GENTOO_PORTAGE_EPREFIX@/var/tmp/texfonts
 
 % Where to look for ls-R files.  There need not be an ls-R in the
 % directories in this path, but if there is one, Kpathsea will use it.
@@ -104,7 +104,7 @@
 %
 % On some systems, there will be a system tree which contains all the font
 % files that may be created as well as the formats.  For example
-%   TEXMFVAR = /var/lib/texmf
+%   TEXMFVAR = @GENTOO_PORTAGE_EPREFIX@/var/lib/texmf
 % is used on many Linux systems.  In this case, set VARTEXFONTS like this
 % VARTEXFONTS = $TEXMFVAR/fonts
 % and do not mention it in TEXMFDBS (but _do_ mention TEXMFVAR).
