diff -ru -x '*.Po' -x '*.Plo' libsoup-2.4.1.orig/libsoup/soup-socket.c libsoup-2.4.1/libsoup/soup-socket.c
--- libsoup-2.4.1.orig/libsoup/soup-socket.c	2008-07-31 09:43:58 +0200
+++ libsoup-2.4.1/libsoup/soup-socket.c	2008-07-31 09:46:01 +0200
@@ -356,15 +356,19 @@
 	setsockopt (priv->sockfd, SOL_SOCKET,
 		    SO_REUSEADDR, (void *) &opt, sizeof (opt));
 
+#ifdef SO_RCVTIMEO
 	timeout.tv_sec = priv->timeout;
 	timeout.tv_usec = 0;
 	setsockopt (priv->sockfd, SOL_SOCKET,
 		    SO_RCVTIMEO, (void *) &timeout, sizeof (timeout));
+#endif
 
+#ifdef SO_SNDTIMEO
 	timeout.tv_sec = priv->timeout;
 	timeout.tv_usec = 0;
 	setsockopt (priv->sockfd, SOL_SOCKET,
 		    SO_SNDTIMEO, (void *) &timeout, sizeof (timeout));
+#endif
 
 #ifndef G_OS_WIN32
 	priv->iochannel =
