* lazy dylib building

--- gsm-1.0-pl13/Makefile
+++ gsm-1.0-pl13/Makefile
@@ -140,7 +140,7 @@
 # Targets
 
 LIBGSM	= $(LIB)/libgsm.a
-LIBGSMSO= $(LIB)/libgsm.so
+LIBGSMSO= libgsm
 
 TOAST	= $(BIN)/toast
 UNTOAST	= $(BIN)/untoast
@@ -323,9 +323,9 @@
 # The basic API: libgsm
 
 $(LIBGSMSO):	$(LIB) $(SHARED_GSM_OBJECTS)
-		$(LD) $(LFLAGS) -o $@.1.0.13 -shared -Xlinker -soname -Xlinker libgsm.so.1 $(SHARED_GSM_OBJECTS) -lc
-		ln -fs libgsm.so.1.0.13 lib/libgsm.so.1
-		ln -fs libgsm.so.1.0.13 lib/libgsm.so
+		$(LD) $(LFLAGS) -o $(LIB)/$@.1.0.13.dylib -dynamiclib -install_name $(GSM_INSTALL_LIB)/$@.1.dylib $(SHARED_GSM_OBJECTS) -lc
+		ln -fs libgsm.1.0.13.dylib lib/libgsm.1.dylib
+		ln -fs libgsm.1.0.13.dylib lib/libgsm.dylib
 
 $(LIBGSM):	$(LIB) $(STATIC_GSM_OBJECTS)
 		-rm $(RMFLAGS) $(LIBGSM)
@@ -336,7 +336,7 @@
 # Toast, Untoast and Tcat -- the compress-like frontends to gsm.
 
 $(TOAST):	$(BIN) $(TOAST_OBJECTS) $(LIBGSM)
-		$(LD) $(LFLAGS) -o $(TOAST) $(TOAST_OBJECTS) $(LIBGSMSO) $(LDLIB)
+		$(LD) $(LFLAGS) -o $(TOAST) $(TOAST_OBJECTS) $(LIB)/$(LIBGSMSO).dylib $(LDLIB)
 
 $(UNTOAST):	$(BIN) $(TOAST)
 		-rm $(RMFLAGS) $(UNTOAST)
@@ -454,8 +454,8 @@
 
 clean:	semi-clean
 		-rm $(RMFLAGS) $(LIBGSM) $(ADDTST)/add		\
-			$(LIBGSMSO) $(LIB)/libgsm.so.1.0.13	\
-			$(LIB)libgsm.so.1			\
+			$(LIB)/$(LIBGSMSO).dylib $(LIB)/libgsm.1.0.13.dylib	\
+			$(LIB)libgsm.1.dylib			\
 			$(TOAST) $(TCAT) $(UNTOAST)		\
 			$(ROOT)/gsm-1.0.tar.Z
 
