diff -aurN gcc-4.4.1/gcc/configure gcc-4.4.1.mint/gcc/configure
--- gcc-4.4.1/gcc/configure	Tue Apr  8 14:25:06 2008
+++ gcc-4.4.1.mint/gcc/configure	Tue Apr  8 14:26:46 2008
@@ -14578,6 +14578,10 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  mint*)
+    lt_cv_sys_max_cmd_len=8192;
+    ;;
+
   netbsd* | freebsd* | openbsd* | darwin* | dragonfly*)
     # This has been around since 386BSD, at least.  Likely further.
     if test -x /sbin/sysctl; then
