diff -ru ufed-0.40.orig/Portage.pm ufed-0.40/Portage.pm
--- ufed-0.40.orig/Portage.pm	2005-11-26 17:24:15 +0100
+++ ufed-0.40/Portage.pm	2007-03-07 13:15:32 +0100
@@ -127,7 +127,7 @@
 }
 
 sub read_make_conf() {
-	my %env = read_sh "/etc/make.conf";
+	my %env = read_sh "@GENTOO_PORTAGE_EPREFIX@/etc/make.conf";
 	merge %make_conf_flags, %{$env{USE}} if exists $env{USE};
 	@portagedirs = $environment{PORTDIR};
 	push @portagedirs, split ' ', $environment{PORTDIR_OVERLAY} if defined $environment{PORTDIR_OVERLAY};
@@ -141,20 +141,20 @@
 }
 
 sub read_make_globals() {
-	for my $dir(@profiles, '/etc') {
+	for my $dir(@profiles, '@GENTOO_PORTAGE_EPREFIX@/etc') {
 		read_sh "$dir/make.globals";
 	}
 }
 
 sub read_packages() {
-	die "Couldn't read /var/db/pkg\n" unless opendir my $pkgdir, '/var/db/pkg';
+	die "Couldn't read @GENTOO_PORTAGE_EPREFIX@/var/db/pkg\n" unless opendir my $pkgdir, '@GENTOO_PORTAGE_EPREFIX@/var/db/pkg';
 	while(my $cat = readdir $pkgdir) {
 		next if $cat eq '.' or $cat eq '..';
-		next unless opendir my $catdir, "/var/db/pkg/$cat";
+		next unless opendir my $catdir, "@GENTOO_PORTAGE_EPREFIX@/var/db/pkg/$cat";
 		while(my $pkg = readdir $catdir) {
 			next if $pkg eq '.' or $pkg eq '..';
-			if(open my $provide, '<', "/var/db/pkg/$cat/$pkg/PROVIDE") {
-				if(open my $use, '<', "/var/db/pkg/$cat/$pkg/USE") {
+			if(open my $provide, '<', "@GENTOO_PORTAGE_EPREFIX@/var/db/pkg/$cat/$pkg/PROVIDE") {
+				if(open my $use, '<', "@GENTOO_PORTAGE_EPREFIX@/var/db/pkg/$cat/$pkg/USE") {
 					# could be shortened, but make sure not to strip off part of the name
 					$pkg =~ s/-\d+(?:\.\d+)*\w?(?:_(?:alpha|beta|pre|rc|p)\d*)?(?:-r\d+)?$//;
 					$packages{"$cat/$pkg"} = 1;
@@ -195,9 +195,9 @@
 }
 
 sub read_profiles() {
-	$_ = readlink '/etc/make.profile';
-	die "/etc/make.profile is not a symlink\n" if not defined $_;
-	@profiles = norm_path '/etc', $_;
+	$_ = readlink '@GENTOO_PORTAGE_EPREFIX@/etc/make.profile';
+	die "@GENTOO_PORTAGE_EPREFIX@/etc/make.profile is not a symlink\n" if not defined $_;
+	@profiles = norm_path '@GENTOO_PORTAGE_EPREFIX@/etc', $_;
 	for (my $i = -1; $i >= -@profiles; $i--) {
 		for(noncomments "$profiles[$i]/parent") {
 			splice @profiles, $i, 0, norm_path $profiles[$i], $_;
diff -ru ufed-0.40.orig/ufed-curses-help.c ufed-0.40/ufed-curses-help.c
--- ufed-0.40.orig/ufed-curses-help.c	2005-11-26 17:24:15 +0100
+++ ufed-0.40/ufed-curses-help.c	2007-03-07 13:15:32 +0100
@@ -90,9 +90,9 @@
 "generally means that it is either a spelling error in one of the three "
 "configuration files or it is not an offically sanctioned USE flag. "
 "Sanctioned USE flags can be found in",
-" /usr/portage/profiles/use.desc",
+" @GENTOO_PORTAGE_EPREFIX@/usr/portage/profiles/use.desc",
 "and in",
-" /usr/portage/profiles/use.local.desc",
+" @GENTOO_PORTAGE_EPREFIX@/usr/portage/profiles/use.local.desc",
 "",
 "***",
 "",
diff -ru ufed-0.40.orig/ufed.pl ufed-0.40/ufed.pl
--- ufed-0.40.orig/ufed.pl	2005-11-26 17:24:15 +0100
+++ ufed-0.40/ufed.pl	2007-03-07 13:15:32 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -I/usr/lib/ufed
+#!@GENTOO_PORTAGE_EPREFIX@/usr/bin/perl -I@GENTOO_PORTAGE_EPREFIX@/usr/lib/ufed
 use strict;
 use warnings;
 
@@ -31,7 +31,7 @@
 	@{$use_descriptions{$_}} = "(Unknown)"
 	if not exists $use_descriptions{$_};
 }
-@{$use_descriptions{'-*'}} = 'Never enable any flags other than those specified in /etc/make.conf';
+@{$use_descriptions{'-*'}} = 'Never enable any flags other than those specified in @GENTOO_PORTAGE_EPREFIX@/etc/make.conf';
 
 for(keys %Portage::use_masked_flags)
 { delete $use_descriptions{$_} if $Portage::use_masked_flags{$_} }
@@ -72,7 +72,7 @@
 		POSIX::dup2 $owrite, 4;
 		POSIX::close $owrite;
 		my $interface = 'ufed-curses';
-		exec { "/usr/lib/ufed/$interface" } $interface or
+		exec { "@GENTOO_PORTAGE_EPREFIX@/usr/lib/ufed/$interface" } $interface or
 		do { print STDERR "Couldn't launch $interface\n"; exit 3 }
 	}
 	POSIX::close $iread;
@@ -152,8 +152,8 @@
 	my $contents;
 
 	{
-		open my $makeconf, '<', '/etc/make.conf' or die "Couldn't open /etc/make.conf\n";
-		open my $makeconfold, '>', '/etc/make.conf.old' or die "Couldn't open /etc/make.conf.old\n";
+		open my $makeconf, '<', '@GENTOO_PORTAGE_EPREFIX@/etc/make.conf' or die "Couldn't open @GENTOO_PORTAGE_EPREFIX@/etc/make.conf\n";
+		open my $makeconfold, '>', '@GENTOO_PORTAGE_EPREFIX@/etc/make.conf.old' or die "Couldn't open @GENTOO_PORTAGE_EPREFIX@/etc/make.conf.old\n";
 		local $/;
 		$_ = <$makeconf>;
 		print $makeconfold $_;
@@ -279,7 +279,7 @@
 	die "Parse error when writing make.conf - did you modify it while ufed was running?\n" if $@;
 
 	{
-		open my $makeconf, '>', '/etc/make.conf' or die "Couldn't open /etc/make.conf\n";
+		open my $makeconf, '>', '@GENTOO_PORTAGE_EPREFIX@/etc/make.conf' or die "Couldn't open @GENTOO_PORTAGE_EPREFIX@/etc/make.conf\n";
 		print $makeconf $_;
 		close $makeconf;
 	}
