From 6661dd08e3a08a4fd42a19d5c4359ae494cfffe2 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Mon, 20 Jul 2009 23:49:21 +0200
Subject: [PATCH] Add support for libxklavier-4

---
 configure.in                 |    4 +++-
 libgnomekbd/gkbd-indicator.c |    6 ++++++
 2 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/configure.in b/configure.in
index 121c976..b1ee544 100644
--- a/configure.in
+++ b/configure.in
@@ -82,7 +82,9 @@ PKG_CHECK_MODULES(GTK, gtk+-2.0 >= $GTK_REQUIRED gdk-x11-2.0 >= $GDK_REQUIRED)
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
-PKG_CHECK_MODULES(LIBXKLAVIER, libxklavier >= $LIBXKLAVIER_REQUIRED)
+PKG_CHECK_MODULES(LIBXKLAVIER, libxklavier >= 4,
+	[AC_DEFINE(HAVE_XKL_4, [1], [libxklavier 4 support])],
+	[PKG_CHECK_MODULES(LIBXKLAVIER, libxklavier >= $LIBXKLAVIER_REQUIRED)])
 
 AC_SUBST(LIBXKLAVIER_CFLAGS)
 AC_SUBST(LIBXKLAVIER_LIBS)
diff --git a/libgnomekbd/gkbd-indicator.c b/libgnomekbd/gkbd-indicator.c
index 48b59cb..608652e 100644
--- a/libgnomekbd/gkbd-indicator.c
+++ b/libgnomekbd/gkbd-indicator.c
@@ -17,6 +17,8 @@
  * Boston, MA 02111-1307, USA.
  */
 
+#include <config.h>
+
 #include <memory.h>
 
 #include <gdk/gdkkeysyms.h>
@@ -717,7 +719,11 @@ gkbd_indicator_global_init (void)
 	globals.registry =
 	    xkl_config_registry_get_instance (globals.engine);
 
+#ifdef HAVE_XKL_4
+	xkl_config_registry_load (globals.registry, FALSE);
+#else
 	xkl_config_registry_load (globals.registry);
+#endif
 
 	gconf_client = gconf_client_get_default ();
 
-- 
1.6.3.3

