--- configure.ac.orig	2008-12-22 16:16:38.000000000 +0100
+++ configure.ac	2009-03-11 13:50:03.000000000 +0100
@@ -2376,12 +2376,18 @@
 # Check for NSNetServices
 #--------------------------------------------------------------------
 HAVE_MDNS=0
-AC_CHECK_HEADERS(dns_sd.h, have_mdns=yes, have_mdns=no)
-if test "$have_mdns" = "yes"; then
-  AC_CHECK_LIB(dns_sd, DNSServiceBrowse, have_mdns=yes, have_mdns=no)
+AC_ARG_ENABLE(zeroconf,
+  [  --disable-zeroconf		Disable NSNetServices support],,
+  enable_zeroconf=yes)
+
+if test $enable_zeroconf = yes; then
+  AC_CHECK_HEADERS(dns_sd.h, have_mdns=yes, have_mdns=no)
   if test "$have_mdns" = "yes"; then
-    LIBS="-ldns_sd $LIBS"
-    HAVE_MDNS=1
+    AC_CHECK_LIB(dns_sd, DNSServiceBrowse, have_mdns=yes, have_mdns=no)
+    if test "$have_mdns" = "yes"; then
+      LIBS="-ldns_sd $LIBS"
+      HAVE_MDNS=1
+    fi
   fi
 fi
 AC_SUBST(HAVE_MDNS)
