diff -ru -x '*.Po' -x '*.Plo' libXmu-1.0.4.orig/include/X11/Xmu/Atoms.h libXmu-1.0.4/include/X11/Xmu/Atoms.h
--- libXmu-1.0.4.orig/include/X11/Xmu/Atoms.h	2009-01-20 16:29:41 +0100
+++ libXmu-1.0.4/include/X11/Xmu/Atoms.h	2009-01-20 16:30:26 +0100
@@ -40,7 +40,7 @@
 
 typedef struct _AtomRec *AtomPtr;
 
-extern AtomPtr
+externalref AtomPtr
     _XA_ATOM_PAIR,
     _XA_CHARACTER_POSITION,
     _XA_CLASS,
