diff -ru -x '*.Po' -x '*.Plo' libXmu-1.0.4.orig/src/EditresCom.c libXmu-1.0.4/src/EditresCom.c
--- libXmu-1.0.4.orig/src/EditresCom.c	2008-09-25 11:49:13 +0200
+++ libXmu-1.0.4/src/EditresCom.c	2008-09-25 16:07:48 +0200
@@ -132,6 +132,12 @@
 
 #define streq(a,b) (strcmp((a), (b)) == 0)
 
+#ifdef _WIN32
+// in Win32 there is a IsChild function already, but with a (of course)
+// completely different signature.
+# define IsChild _XIsChild
+#endif
+
 /*
  * Prototypes
  */
@@ -2051,9 +2057,14 @@
 {
     static XtResource resources[] = {
         {"editresBlock", "EditresBlock", XtREditresBlock, sizeof(EditresBlock),
-     XtOffsetOf(Globals, block), XtRImmediate, (XtPointer)BlockNone}
+     XtOffsetOf(Globals, block), NULL, (XtPointer)BlockNone}
     };
 
+	// required for windows... argl (initializer not a constant)
+	if(!resources[0].default_type)
+		resources[0].default_type = XtRImmediate;
+
+
   for (; XtParent(w) != NULL; w = XtParent(w))
     ;
 
diff -ru -x '*.Po' -x '*.Plo' libXmu-1.0.4.orig/src/GetHost.c libXmu-1.0.4/src/GetHost.c
--- libXmu-1.0.4.orig/src/GetHost.c	2008-09-25 11:49:13 +0200
+++ libXmu-1.0.4/src/GetHost.c	2008-09-25 11:51:59 +0200
@@ -60,6 +60,9 @@
     int len;
 #ifdef WIN32
     static WSADATA wsadata;
+	#ifndef BYTE
+	#define BYTE unsigned char
+	#endif
 
     if (!wsadata.wVersion && WSAStartup(MAKEWORD(1,1), &wsadata))
 	return -1;
