--- texk/xdvipdfmx/configure.ac
+++ texk/xdvipdfmx/configure.ac
@@ -67,17 +67,11 @@
   CHO_CHECK_LIBPAPER
 fi
 
-KPSE_CHECK_FRAMEWORK([ApplicationServices], [ATSFontRef fontRef])
 KPSE_FONTCONFIG_FLAGS
-if test "x$kpse_cv_have_ApplicationServices" = xyes; then
-  build=yes
-  AC_DEFINE([XETEX_MAC], 1, [Define if part of a teTeX/TeX Live build for Mac OS X.])
-else
   KPSE_ADD_FLAGS([fontconfig])
   CHO_CHECK_LIBFONTCONFIG
   KPSE_RESTORE_FLAGS
   build=$cho_cv_have_fontconfig
-fi
 
 if test "x$build" = xyes; then
   AC_DEFINE([XETEX], 1, [Define if part of a teTeX/TeX Live build.])
--- texk/xdvipdfmx/configure
+++ texk/xdvipdfmx/configure
@@ -18648,64 +18648,6 @@
 
 fi
 
-{ $as_echo "$as_me:$LINENO: checking for Mac OS X ApplicationServices framework" >&5
-$as_echo_n "checking for Mac OS X ApplicationServices framework... " >&6; }
-if test "${kpse_cv_have_ApplicationServices+set}" = set; then
-  $as_echo_n "(cached) " >&6
-else
-  cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-#include <ApplicationServices/ApplicationServices.h>
-int
-main ()
-{
-ATSFontRef fontRef
-  ;
-  return 0;
-}
-_ACEOF
-rm -f conftest.$ac_objext
-if { (ac_try="$ac_compile"
-case "(($ac_try" in
-  *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
-  *) ac_try_echo=$ac_try;;
-esac
-eval ac_try_echo="\"\$as_me:$LINENO: $ac_try_echo\""
-$as_echo "$ac_try_echo") >&5
-  (eval "$ac_compile") 2>conftest.er1
-  ac_status=$?
-  grep -v '^ *+' conftest.er1 >conftest.err
-  rm -f conftest.er1
-  cat conftest.err >&5
-  $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } && {
-	 test -z "$ac_c_werror_flag" ||
-	 test ! -s conftest.err
-       } && test -s conftest.$ac_objext; then
-  kpse_cv_have_ApplicationServices=yes
-else
-  $as_echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
-
-	kpse_cv_have_ApplicationServices=no
-fi
-
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-fi
-{ $as_echo "$as_me:$LINENO: result: $kpse_cv_have_ApplicationServices" >&5
-$as_echo "$kpse_cv_have_ApplicationServices" >&6; }
-if test "x$kpse_cv_have_ApplicationServices" = xyes; then
-
-cat >>confdefs.h <<\_ACEOF
-#define HAVE_APPLICATIONSERVICES 1
-_ACEOF
-
-fi
-
 if test -n "$ac_tool_prefix"; then
   # Extract the first word of "${ac_tool_prefix}pkg-config", so it can be a program name with args.
 set dummy ${ac_tool_prefix}pkg-config; ac_word=$2
