--- texlive-20091107-source/texk/kpathsea/progname.c
+++ texlive-20091107-source/texk/kpathsea/progname.c
@@ -403,6 +403,10 @@
     self = concat3 (".", DIR_SEP_STRING, argv0);
     
-  ret = xdirname (remove_dots (kpse, expand_symlinks (kpse, self)));
+  if (kpse_absolute_p (kpse, self, true)) {
+     ret = xdirname (remove_dots (kpse, self));
+  } else {
+     ret = xdirname (remove_dots (kpse, expand_symlinks (kpse, self)));
+  }
 
   free (self);
   
