--- opengl.eselect
+++ opengl.eselect
@@ -72,13 +72,17 @@
 		# First remove old symlinks
 		for file in libGL{,core}.{a,so,dylib,la} ; do
 			if [[ -e ${file} ]] ; then
-				rm -f "${file}" || die -q "Failed to delete ${file}"
+				# also delete specific so versions, not applicable for dylibs
+				rm -f "${file}"* || die -q "Failed to delete ${file}"
 			fi
 		done
 
 		# Note that we don't do .so*, just .so on purpose.  The
 		# loader knows to look in the profile dir, and the
 		# linked just needs the .so
+		# That is actually frikkin' expensive, and it doesn't work well
+		# for Prefix where it's even more expensive, so we copy the
+		# soname version too.
 		for file in ${profile_libdir}/libGL{,core}.{so,dylib,a,la}; do
 			if [[ ${ROOT} != / ]] ; then
 				rootfile="${file//${ROOT}}"
@@ -95,6 +99,12 @@
 				sed "s:${profile_libdir}:${libdir}:g" "${file}" > "$(basename "${file}")" || die -q "Failed to create ${libdir}/$(basename "${file}")"
 			else
 				ln -s "${rootfile}" || die -q "Failed to create ${libdir}/$(basename "${file}")"
+				if [[ ${file} == *.so ]] ; then
+					# copy soname target, needed for Prefix
+					soname=$(scanelf -qBF '%S|%p' ${rootfile})
+					soname=${soname%|*}  # stupid scanelf must print filename
+					ln -s "${rootfile%/*}/${soname}" || die -q "Failed to create ${libdir}/${soname}"
+				fi
 			fi
 		done
 	popd &> /dev/null
