--- opengl.eselect.orig	2009-11-13 20:57:39 +0100
+++ opengl.eselect	2009-11-13 21:08:59 +0100
@@ -85,8 +85,18 @@
 		# Note that we don't do .so*, just .so on purpose.  The
 		# loader knows to look in the profile dir, and the
 		# linked just needs the .so
+		# That is actually frikkin' expensive, and it doesn't work well
+		# for Prefix where it's even more expensive because there is no
+		# ld.so.conf in action, so we copy the soname version too.
+		# Note that this is not necessary for Darwin since install_names
+		# are absolute paths.
 		for file in ${profile_libdir}/libGL{,core}.{so,dylib,a}; do
 			upgrade_file ${file} ${target}
+			if [[ ${file} == *.so ]] ; then
+				# copy soname target, needed for loader in Prefix
+				soname=$(scanelf -qBF '%S#p' ${file})
+				upgrade_file ${file%/*}/${soname} ${target}
+			fi
 		done
 	popd &> /dev/null
 }
