--- opengl.eselect
+++ opengl.eselect
@@ -20,9 +20,9 @@
 EBUILD_VERSION="1.2.0"
 
 # Our data
-ENV_FILE="${ROOT}/etc/env.d/03opengl"
-PREFIX="${ROOT}/usr"
-DST_PREFIX="${ROOT}/usr"
+ENV_FILE="${EROOT}/etc/env.d/03opengl"
+PREFIX="${EROOT}/usr"
+DST_PREFIX="${EROOT}/usr"
 
 get_current_implementation() {
 	local ret
@@ -47,7 +47,7 @@
 	local dir_name
 
 	for libdir in $(list_libdirs) ; do
-		[[ ${ROOT} != / ]] && libdir=${libdir#${ROOT}}
+		[[ ${ROOT} != / ]] && libdir=${libdir#${EROOT}}
 		for dir in "${PREFIX}/${libdir}"/opengl/* ; do
 			dir_name=$(basename "${dir}")
 			[[ -d ${dir} && ${dir_name} != "global" ]] || continue
@@ -90,6 +90,12 @@
 		soname=$(scanelf -qBF '%S|%p' "${file}")
 		soname=${soname%|*}  # stupid scanelf must print filename
 		upgrade_file "${file%/*}"/${soname} "${target}"
+	elif [[ ${file} == *.dylib ]] ; then
+		# same as ELF, but then for Mach-O, only consider the first arch
+		# in case of FAT objects
+		local installname=$(scanmacho -qBF '%S|%p' "${file}" | head -n1)
+		installname=${installname%|*}  # stupid scanelf must print filename
+		upgrade_file "${installname}" "${target}"
 	fi
 }
 
@@ -176,7 +182,7 @@
 
 	for libdir in $(list_libdirs); do
 		# Set libdir correctly to EROOT
-		[[ ${ROOT} != / ]] && libdir=${libdir#${ROOT}}
+		[[ ${ROOT} != / ]] && libdir=${libdir#${EROOT}}
 
 		# First make sure we have an opengl directory and this
 		# is a real lib dir, not a symlink
