diff -ur apache-maven-3.0.3.orig/bin/mvn apache-maven-3.0.3/bin/mvn
--- apache-maven-3.0.3.orig/bin/mvn	2011-02-28 18:33:40.000000000 +0100
+++ apache-maven-3.0.3/bin/mvn	2011-05-01 12:59:46.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 # ----------------------------------------------------------------------------
 # Licensed to the Apache Software Foundation (ASF) under one
 # or more contributor license agreements.  See the NOTICE file
@@ -36,8 +36,8 @@
 
 if [ -z "$MAVEN_SKIP_RC" ] ; then
 
-  if [ -f /etc/mavenrc ] ; then
-    . /etc/mavenrc
+  if [ -f @GENTOO_PORTAGE_EPREFIX@/etc/mavenrc ] ; then
+    . @GENTOO_PORTAGE_EPREFIX@/etc/mavenrc
   fi
 
   if [ -f "$HOME/.mavenrc" ] ; then
@@ -57,14 +57,11 @@
            if [ -z "$JAVA_VERSION" ] ; then
              JAVA_VERSION="CurrentJDK"
            fi
-           if [ -z "$JAVA_HOME" ] ; then
-             JAVA_HOME=/System/Library/Frameworks/JavaVM.framework/Versions/${JAVA_VERSION}/Home
-           fi
            ;;
 esac
 
 if [ -z "$JAVA_HOME" ] ; then
-  if [ -r /etc/gentoo-release ] ; then
+  if [ -r @GENTOO_PORTAGE_EPREFIX@/etc/gentoo-release ] ; then
     JAVA_HOME=`java-config --jre-home`
   fi
 fi
Only in apache-maven-3.0.3: prefix.patch
