--- Makefile.in.orig	2008-10-26 13:15:44.000000000 +0100
+++ Makefile.in	2008-10-26 13:16:25.000000000 +0100
@@ -256,14 +256,14 @@ install-doc: doc
 	@mkdir -p $(DESTDIR)$(mandir)/man3
 	@echo "Installing documentation in $(DESTDIR)$(mandir)"
 # install Expectk man page if present
-	$(INSTALL_DATA) $(srcdir)/expectk.man $(mandir)/man1/expectk.1 ; \
+	$(INSTALL_DATA) $(srcdir)/expectk.man $(DESTDIR)$(mandir)/man1/expectk.1 ; \
 # install Expect man page
-	$(INSTALL_DATA) $(srcdir)/expect.man $(mandir)/man1/expect.1
+	$(INSTALL_DATA) $(srcdir)/expect.man $(DESTDIR)$(mandir)/man1/expect.1
 # install man page for Expect and Expectk libraries
-	$(INSTALL_DATA) $(srcdir)/libexpect.man $(mandir)/man3/libexpect.3
+	$(INSTALL_DATA) $(srcdir)/libexpect.man $(DESTDIR)$(mandir)/man3/libexpect.3
 	-for i in $(SCRIPT_MANPAGE_LIST) ; do \
 	  if [ -f $(srcdir)/example/$$i.man ] ; then \
-	    $(INSTALL_DATA) $(srcdir)/example/$$i.man $(mandir)/man1/$$i.1 ; \
+	    $(INSTALL_DATA) $(srcdir)/example/$$i.man $(DESTDIR)$(mandir)/man1/$$i.1 ; \
 	  else true; fi ; \
 	done
 
