diff -ruB epic4-2.8.orig/source/functions.c epic4-2.8/source/functions.c
--- epic4-2.8.orig/source/functions.c	2008-03-03 16:00:27 +0100
+++ epic4-2.8/source/functions.c	2008-03-03 16:04:36 +0100
@@ -6974,7 +6974,12 @@
 	 */
 	GET_STR_ARG(number, input);
 	
+	#ifdef __INTERIX
+	/* i know this is no good, but still.... */
+	retval = strtol(number, &after, base);
+	#else
 	retval = strtoimax(number, &after, base);
+	#endif
 	/* Argh -- do we want to return error if invalid char found? */
 	RETURN_INT(retval);
 }
@@ -7019,7 +7024,12 @@
 		number++;
 	}
 
+	#ifdef __INTERIX
+	/* i know this is no good, but still.... */
+	num = strtol(number, &after, 10);	/* Must not use GET_INT_ARG */
+	#else
 	num = strtoimax(number, &after, 10);	/* Must not use GET_INT_ARG */
+	#endif
 
 	while (pow(base, len) <= num)
 		len++;
