diff -ru -x '*.Po' -x '*.Plo' gvfs-0.2.3.orig/daemon/gvfsbackendburn.c gvfs-0.2.3/daemon/gvfsbackendburn.c
--- gvfs-0.2.3.orig/daemon/gvfsbackendburn.c	2008-04-15 08:44:21 +0200
+++ gvfs-0.2.3/daemon/gvfsbackendburn.c	2008-04-15 08:49:33 +0200
@@ -368,7 +368,11 @@
   tempdir = g_build_filename (g_get_tmp_dir (), filename, NULL);
   g_free (filename);
 
+  #ifndef __INTERIX
   dir = mkdtemp (tempdir);
+  #else
+  dir = mktemp (tempdir);
+  #endif
   if (dir == NULL)
     {
       g_free (tempdir);
diff -ru -x '*.Po' -x '*.Plo' gvfs-0.2.3.orig/daemon/gvfsbackendsftp.c gvfs-0.2.3/daemon/gvfsbackendsftp.c
--- gvfs-0.2.3.orig/daemon/gvfsbackendsftp.c	2008-04-15 08:53:30 +0200
+++ gvfs-0.2.3/daemon/gvfsbackendsftp.c	2008-04-15 08:39:35 +0200
@@ -32,6 +32,10 @@
 #include <fcntl.h>
 #include <string.h>
 
+#ifdef __INTERIX
+# include <sys/time.h>
+#endif
+
 #include <glib/gstdio.h>
 #include <glib/gi18n.h>
 #include <gio/gio.h>
diff -ru -x '*.Po' -x '*.Plo' gvfs-0.2.3.orig/daemon/gvfsbackendtrash.c gvfs-0.2.3/daemon/gvfsbackendtrash.c
--- gvfs-0.2.3.orig/daemon/gvfsbackendtrash.c	2008-04-15 08:44:21 +0200
+++ gvfs-0.2.3/daemon/gvfsbackendtrash.c	2008-04-15 08:48:03 +0200
@@ -32,6 +32,10 @@
 #include <fcntl.h>
 #include <string.h>
 
+#ifdef __INTERIX
+# include <sys/time.h>
+#endif
+
 #include <glib/gstdio.h>
 #include <glib/gi18n.h>
 #include <gio/gio.h>
