diff -ru recode-3.6.orig/src/Makefile.am recode-3.6/src/Makefile.am
--- recode-3.6.orig/src/Makefile.am	2008-07-14 15:53:21 +0200
+++ recode-3.6/src/Makefile.am	2008-07-14 15:55:52 +0200
@@ -43,11 +43,11 @@
 L_SURFACES =
 
 recode_SOURCES = main.c freeze.c mixed.c
-recode_LDADD = librecode.la ../lib/libreco.a
+recode_LDADD = librecode.la
 
 librecode_la_SOURCES = argmatch.c charname.c combine.c fr-charname.c \
 hash.c iconv.c libiconv.c localcharset.c merged.c names.c outer.c quotearg.c \
-recode.c request.c strip-pool.c task.c xmalloc.c $(C_STEPS)
+recode.c request.c strip-pool.c task.c xmalloc.c error.c xstrdup.c $(C_STEPS)
 librecode_la_LDFLAGS = -version-info 0:0:0
 
 INCLUDES = -I.. -I$(srcdir) -I$(top_srcdir)/lib -I$(top_srcdir)/libiconv
