diff -ru xmessage-1.0.2.orig/configure.ac xmessage-1.0.2/configure.ac
--- xmessage-1.0.2.orig/configure.ac	2009-09-22 14:18:44 +0200
+++ xmessage-1.0.2/configure.ac	2009-09-22 14:20:21 +0200
@@ -32,7 +32,11 @@
 AC_PROG_INSTALL
 
 # Checks for pkg-config packages
-XAW_CHECK_XPRINT_SUPPORT(XMESSAGE)
+PKG_CHECK_MODULES(XAW, xaw7)
+
+XMESSAGE_CFLAGS="${XMESSAGE_CFLAGS} ${XAW_CFLAGS}"
+XMESSAGE_LIBS="${XMESSAGE_LIBS} ${XAW_LIBS}"
+
 AC_SUBST(XMESSAGE_CFLAGS)
 AC_SUBST(XMESSAGE_LIBS)
 
