--- IPython/Extensions/jobctrl.py.orig	2008-07-05 11:04:11.672812749 +0100
+++ IPython/Extensions/jobctrl.py	2008-07-05 14:12:45.893249659 +0100
@@ -190,10 +190,11 @@
         
 if sys.platform == 'win32':
     shell_internal_commands = 'break chcp cls copy ctty date del erase dir md mkdir path prompt rd rmdir start time type ver vol'.split()
+    PopenExc = WindowsError
 else:
     # todo linux commands
     shell_internal_commands = [] 
-
+    PopenExc = OSError
 
 def jobctrl_shellcmd(ip,cmd):
     """ os.system replacement that stores process info to db['tasks/t1234'] """
@@ -208,7 +209,7 @@
     try:
         try:
             p = Popen(cmd,shell = use_shell)
-        except WindowsError:
+        except PopenExc:
             if use_shell:
                 # try with os.system
                 os.system(cmd)
