diff -urNp ccrypt-1.7.org/emacs/jka-compr-ccrypt.el ccrypt-1.7/emacs/jka-compr-ccrypt.el
--- ccrypt-1.7.org/emacs/jka-compr-ccrypt.el	2003-08-25 17:15:44.000000000 +0300
+++ ccrypt-1.7/emacs/jka-compr-ccrypt.el	2007-06-09 18:36:19.000000000 +0300
@@ -302,6 +302,7 @@ based on the filename itself and `jka-co
   "The encryption password. This variable is buffer-local.")
 
 (make-variable-buffer-local 'jka-compr-ccrypt-buffer-password)
+(put 'jka-compr-ccrypt-buffer-password 'permanent-local t)
 
 (defun jka-compr-ccrypt-read-passwd (&optional confirm default)
   (read-passwd (format "Password for %s: " (buffer-name)) confirm default))
