diff -ru colordiff-1.0.7.orig/colordiff.pl colordiff-1.0.7/colordiff.pl
--- colordiff-1.0.7.orig/colordiff.pl	2008-04-03 19:17:47 +0200
+++ colordiff-1.0.7/colordiff.pl	2008-04-03 19:19:36 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!@GENTOO_PORTAGE_EPREFIX@/usr/bin/perl -w
 
 ########################################################################
 #                                                                      #
--- colordiff-1.0.7/colordiff.pl	2008-11-25 14:22:27.000000000 +0100
+++ colordiff-1.0.7/colordiff.pl.new	2008-11-25 14:28:28.000000000 +0100
@@ -63,7 +63,7 @@
 
 # Locations for personal and system-wide colour configurations
 my $HOME   = $ENV{HOME};
-my $etcdir = '/etc';
+my $etcdir = '@GENTOO_PORTAGE_EPREFIX@/etc';
 my ($setting, $value);
 my @config_files = ("$etcdir/colordiffrc");
 push (@config_files, "$ENV{HOME}/.colordiffrc") if (defined $ENV{HOME});
