diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN ConsoleKit-0.3.0.orig/data/ConsoleKit.conf ConsoleKit-0.3.0/data/ConsoleKit.conf
--- ConsoleKit-0.3.0.orig/data/ConsoleKit.conf	2008-04-19 01:39:49.000000000 -0400
+++ ConsoleKit-0.3.0/data/ConsoleKit.conf	2009-04-27 15:20:57.000000000 -0400
@@ -90,6 +90,8 @@
     <allow send_interface="org.freedesktop.ConsoleKit.Session"
            send_member="GetIdleHint"/>
     <allow send_interface="org.freedesktop.ConsoleKit.Session"
+           send_member="SetIdleHint"/>
+    <allow send_interface="org.freedesktop.ConsoleKit.Session"
            send_member="GetIdleSinceHint"/>
   </policy>
 
