Clean up build system to make cross-compiling easy

--- scripts/makefile.linux
+++ scripts/makefile.linux
@@ -8,12 +8,12 @@
 PNGMIN = 1.2.8
 PNGVER = $(PNGMAJ).$(PNGMIN)
 
-CC=gcc
+CC?=gcc
 
 # where "make install" puts libpng12.a, libpng12.so*,
 # libpng12/png.h and libpng12/pngconf.h
 # Prefix must be a full pathname.
-prefix=/usr/local
+prefix=/usr
 
 # Where the zlib library and include files are located.
 #ZLIBLIB=/usr/local/lib
@@ -31,18 +31,25 @@
 
 # for pgcc version 2.95.1, -O3 is buggy; don't use it.
 
-CFLAGS=-I$(ZLIBINC) -Wall -O3 -funroll-loops \
-	$(ALIGN) # $(WARNMORE) -g -DPNG_DEBUG=5
+CFLAGS += -Wall $(ALIGN) 
+ifneq ($(ZLIBINC),)
+CFLAGS += -I$(ZLIBINC)
+endif
+
+ifneq ($(ZLIBLIB),)
+LDFLAGS += -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB)
+endif
+LDFLAGS    += -lz -lm
+LDFLAGS_A   = $(LDFLAGS) libpng.a
+LDFLAGS_SO  = $(LDFLAGS) -L. -Wl,-rpath,. -lpng12
 
-LDFLAGS=-L. -Wl,-rpath,. -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) -lpng12 -lz -lm
-LDFLAGS_A=-L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) libpng.a -lz -lm
-
-RANLIB=ranlib
+AR?=ar
+RANLIB?=ranlib
 #RANLIB=echo
 
 INCPATH=$(prefix)/include
 LIBPATH=$(prefix)/lib
-MANPATH=$(prefix)/man
+MANPATH=$(prefix)/share/man
 BINPATH=$(prefix)/bin
 
 # override DESTDIR= on the make install command line to easily support
@@ -74,7 +81,7 @@
 all: libpng.a $(LIBNAME).so pngtest pngtest-static libpng.pc libpng-config
 
 libpng.a: $(OBJS)
-	ar rc $@ $(OBJS)
+	$(AR) rc $@ $(OBJS)
 	$(RANLIB) $@
 
 libpng.pc:
@@ -104,10 +111,10 @@
 libpng.so.3.$(PNGMIN): $(OBJSDLL)
 	$(CC) -shared -Wl,-soname,libpng.so.3 \
 	-o libpng.so.3.$(PNGMIN) \
-	$(OBJSDLL)
+	$(OBJSDLL) $(LDFLAGS)
 
 pngtest: pngtest.o $(LIBNAME).so
-	$(CC) -o pngtest $(CFLAGS) pngtest.o $(LDFLAGS)
+	$(CC) -o pngtest $(CFLAGS) pngtest.o $(LDFLAGS_SO)
 
 pngtest-static: pngtest.o libpng.a
 	$(CC) -o pngtest-static $(CFLAGS) pngtest.o $(LDFLAGS_A)
--- scripts/makefile.darwin	2005-10-30 09:55:21.000000000 +0100
+++ scripts/makefile.darwin	2005-10-30 10:08:38.000000000 +0100
@@ -7,20 +7,30 @@
 # For conditions of distribution and use, see copyright notice in png.h
 
 # where "make install" puts libpng.a, libpng12.dylib, png.h and pngconf.h
-prefix=/usr/local
+prefix=/usr
 
 # Where the zlib library and include files are located
 #ZLIBLIB=/usr/local/lib
 #ZLIBINC=/usr/local/include
-ZLIBLIB=../zlib
-ZLIBINC=../zlib
+ZLIBLIB=./zlib
+ZLIBINC=./zlib
 
-CC=cc
-CFLAGS=-I$(ZLIBINC) -Wall -O3 -funroll-loops
-LDFLAGS=-L. -L$(ZLIBLIB) -lpng12 -lz
+CC?=gcc
+
+CFLAGS += -Wall $(ALIGN) 
+ifneq ($(ZLIBINC),)
+CFLAGS += -I$(ZLIBINC)
+endif
+
+ifneq ($(ZLIBLIB),)
+LDFLAGS += -L$(ZLIBLIB)
+endif
+LDFLAGS    += -lz
+LDFLAGS_DYLIB = $(LDFLAGS) -L. -lpng12
 
 #RANLIB=echo
-RANLIB=ranlib
+AR?=ar
+RANLIB?=ranlib
 
 PNGMAJ = 0
 PNGMIN = 1.2.8
@@ -29,7 +39,7 @@
 
 INCPATH=$(prefix)/include
 LIBPATH=$(prefix)/lib
-MANPATH=$(prefix)/man
+MANPATH=$(prefix)/share/man
 BINPATH=$(prefix)/bin
 
 # override DESTDIR= on the make install command line to easily support
@@ -61,7 +71,7 @@
 all: libpng.a $(LIBNAME).dylib pngtest libpng.pc libpng-config
 
 libpng.a: $(OBJS)
-	ar rc $@ $(OBJS)
+	$(AR) rc $@ $(OBJS)
 	$(RANLIB) $@
 
 libpng.pc:
@@ -88,17 +98,17 @@
 	 -install_name $(LIBPATH)/$(LIBNAME).$(PNGMAJ).dylib \
 	 -current_version $(PNGVER) -compatibility_version $(PNGVER) \
 	 -o $(LIBNAME).$(PNGVER).dylib \
-	 $(OBJSDLL) -L$(ZLIBLIB) -lz
+	 $(OBJSDLL) $(LDFLAGS)
 
 libpng.3.$(PNGMIN).dylib: $(OBJSDLL)
 	$(CC) -dynamiclib \
 	 -install_name $(LIBPATH)/libpng.3.dylib \
 	 -current_version 3 -compatibility_version 3 \
 	 -o libpng.3.$(PNGMIN).dylib \
-	 $(OBJSDLL) -L$(ZLIBLIB) -lz
+	 $(OBJSDLL) $(LDFLAGS)
 
 pngtest: pngtest.o $(LIBNAME).dylib
-	$(CC) -o pngtest $(CFLAGS) pngtest.o $(LDFLAGS)
+	$(CC) -o pngtest $(CFLAGS) pngtest.o $(LDFLAGS_DYLIB)
 
 test: pngtest
 	./pngtest
