diff -ru libpng-1.2.25.orig/configure.ac libpng-1.2.25/configure.ac
--- libpng-1.2.25.orig/configure.ac	2008-02-18 23:31:14 +0100
+++ libpng-1.2.25/configure.ac	2008-03-04 17:12:12 +0100
@@ -74,6 +74,13 @@
 
 AC_MSG_CHECKING([if libraries can be versioned])
 GLD=`$LD --help < /dev/null 2>/dev/null | grep version-script`
+
+# does not quite work on interix. it causes the linker to
+# crash when trying to link something that depends on libpng12.so
+case "$host_os" in
+interix*) GLD= ;;
+esac
+
 if test "$GLD"; then
     have_ld_version_script=yes
     AC_MSG_RESULT(yes)
