http://bugs.gentoo.org/195813

--- Makefile.in
+++ Makefile.in
@@ -1230,7 +1230,7 @@
 libpng.sym: png.h pngconf.h
 	rm -f $@ $@.new
 	$(CPP) @LIBPNG_DEFINES@ $(CPPFLAGS) -DPNG_BUILDSYMS $(srcdir)/png.h | \
-		$(SED) -n -e 's|^.*PNG_FUNCTION_EXPORT[ 	]*\([a-zA-Z0-9_]*\).*$$|\1|p' \
-			-e 's|^.*PNG_DATA_EXPORT[ 	]*\([a-zA-Z0-9_]*\).*$$|\1|p' \
+		$(SED) -n -e 's|^.*PNG_FUNCTION_EXPORT[ 	]*\([[:alnum:]_]*\).*$$|\1|p' \
+			-e 's|^.*PNG_DATA_EXPORT[ 	]*\([[:alnum:]_]*\).*$$|\1|p' \
 			>$@.new
 	mv $@.new $@
