--- genlop.orig	2007-04-09 16:55:46 +0200
+++ genlop	2007-08-02 22:39:58 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 #
 # $Id: genlop,v 1.21 2005/08/16 23:50:32 antonio Exp $
 #
@@ -26,7 +26,7 @@
 use File::Basename;
 
 my $version  = "0.30.8";
-my @logfiles = ("/var/log/emerge.log");
+my @logfiles = ("@GENTOO_PORTAGE_EPREFIX@/var/log/emerge.log");
 my %COLORS   = (
     'blue'  => 'bold blue',
     'green' => 'bold green',
@@ -109,7 +109,7 @@
 sub open_file
 {
     my ($file, $fh) = @_;
-    if ($file eq "/var/log/emerge.log" && !-r $file)
+    if ($file eq "@GENTOO_PORTAGE_EPREFIX@/var/log/emerge.log" && !-r $file)
     {
         print "$progname: cannot open " . $file . " for reading\n" . "maybe you are not a member of the portage group ?\n" . "try genlop -h for help\n";
         exit 1;
@@ -802,12 +802,12 @@
     }
     $e_count = 0;
     print "\n   Info about currently installed ebuild:\n";
-    opendir(DIR, "/var/db/pkg/") || die "can't open /var/db/pkg/ $!\n";
+    opendir(DIR, "@GENTOO_PORTAGE_EPREFIX@/var/db/pkg/") || die "can't open @GENTOO_PORTAGE_EPREFIX@/var/db/pkg/ $!\n";
     while (defined(my $categoria = readdir(DIR)))
     {
         if ($package =~ m/^$categoria.*/g)
         {
-            opendir(DIR2, "/var/db/pkg/$categoria");
+            opendir(DIR2, "@GENTOO_PORTAGE_EPREFIX@/var/db/pkg/$categoria");
             while (defined(my $package_dir = readdir(DIR2)))
             {
 
@@ -851,7 +851,7 @@
                     my (@potential_use, @pkg_use, @used_use, @unused_use);
 
                     # each installed package store its information here
-                    my $db_pkg_dir = "/var/db/pkg/$categoria/$package_dir/";
+                    my $db_pkg_dir = "@GENTOO_PORTAGE_EPREFIX@/var/db/pkg/$categoria/$package_dir/";
                     if ("$categoria/$package_dir" =~ m/.*\/(.*)/g)
                     {
 
