--- tetex-src-3.0/texk/xdvik/x_util.c	2005-12-06 16:04:45.000000000 +0100
+++ tetex-src-3.0/texk/xdvik/x_util.c.new	2006-01-06 13:13:22.000000000 +0100
@@ -205,15 +205,15 @@
     w = (*((xuint32 *) window_p));
 #else
 # if WORDS_BIGENDIAN
-    w = ((unsigned long)wp[0] << 24) |
-	((unsigned long)wp[1] << 16) |
-	((unsigned long)wp[2] << 8)  |
-	(unsigned long)wp[3];
+    w = ((unsigned long)window_p[0] << 24) |
+	((unsigned long)window_p[1] << 16) |
+	((unsigned long)window_p[2] << 8)  |
+	(unsigned long)window_p[3];
 # else
-    w = ((unsigned long)wp[3] << 24) |
-	((unsigned long)wp[2] << 16) |
-	((unsigned long)wp[1] << 8)  |
-	(unsigned long)wp[0];
+    w = ((unsigned long)window_p[3] << 24) |
+	((unsigned long)window_p[2] << 16) |
+	((unsigned long)window_p[1] << 8)  |
+	(unsigned long)window_p[0];
 # endif
 #endif
     return w;
