--- tetex-src-3.0/texk/tetex/epstopdf.orig	2005-11-28 01:34:05.000000000 +0100
+++ tetex-src-3.0/texk/tetex/epstopdf	2005-10-22 19:25:12.000000000 +0200
@@ -205,7 +205,8 @@
 ### open output file
 if ($::opt_gs) {
   my $pipe = "$GS -q -sDEVICE=pdfwrite $GSOPTS " .
-          "-sOutputFile=$OutputFilename - -c quit";
+      "-dAutoRotatePages=/None " .
+      "\"-sOutputFile=$OutputFilename\" - -c quit";
   debug "Ghostscript pipe:", $pipe;
   open(OUT,"|$pipe") or error "Cannot open Ghostscript for piped input";
 }
