diff -ru -x '*.Po' -x '*.Plo' tetex-src-2.0.2.orig/texk/dviljk/config.h tetex-src-2.0.2/texk/dviljk/config.h
--- tetex-src-2.0.2.orig/texk/dviljk/config.h	2008-04-13 13:16:37 +0200
+++ tetex-src-2.0.2/texk/dviljk/config.h	2008-04-13 13:41:33 +0200
@@ -259,7 +259,7 @@
 #endif
 #endif
 
-#if defined (unix) && !defined (__DJGPP__)
+#if (defined (unix) && !defined (__DJGPP__)) || defined(__INTERIX)
 #define OS "Unix"
 #define READ_BINARY     "r"
 #define WRITE_BINARY    "w"
diff -ru -x '*.Po' -x '*.Plo' tetex-src-2.0.2.orig/texk/dviljk/dvi2xx.h tetex-src-2.0.2/texk/dviljk/dvi2xx.h
--- tetex-src-2.0.2.orig/texk/dviljk/dvi2xx.h	2008-04-13 13:16:38 +0200
+++ tetex-src-2.0.2/texk/dviljk/dvi2xx.h	2008-04-13 13:40:34 +0200
@@ -289,7 +289,7 @@
 /**********************************************************************/
 
 #ifndef WIN32
-#ifndef _AMIGA
+#if !defined(_AMIGA) && !defined(__INTERIX)
 # ifndef unix
 #  if NeedFunctionPrototypes
 long    access(char *, int);      /* all the other ones known under RISC OS */
