--- src/term.h	2004-04-13 19:24:02.000000000 +0200
+++ src/term.h.new	2004-05-05 11:16:41.216200736 +0200
@@ -270,6 +270,20 @@
 /****************************************************************************/
 /* These terminals can be used on any system */
 
+/* NOTE THAT PNG AND JPEG SUPPORT REQUIRE SEPARATE LIBRARIES */
+/* There are 2 flavors of PNG support                                        */
+/* If you are using an old version of libgd in order to get true GIF support */
+/* then the best we can do is use a separate poor-quality PNG driver png.trm */
+/* But if you are using a newer libgd then it provides full blown support    */
+/* for both PNG and JPEG in term/gd.trm                                      */
+#ifdef HAVE_LIBPNG
+# if defined(HAVE_GD_PNG) || defined(HAVE_GD_JPEG)
+#  include "gd.trm"
+# else
+#  include "png.trm"
+# endif
+#endif
+
 #ifdef X11
 # include "x11.trm"		/* X Window System */
 # include "xlib.trm"		/* dumps x11 commands to gpoutfile */
@@ -351,20 +365,6 @@
 # include "pdf.trm"
 #endif
 
-/* NOTE THAT PNG AND JPEG SUPPORT REQUIRE SEPARATE LIBRARIES */
-/* There are 2 flavors of PNG support                                        */
-/* If you are using an old version of libgd in order to get true GIF support */
-/* then the best we can do is use a separate poor-quality PNG driver png.trm */
-/* But if you are using a newer libgd then it provides full blown support    */
-/* for both PNG and JPEG in term/gd.trm                                      */
-#ifdef HAVE_LIBPNG
-# if defined(HAVE_GD_PNG) || defined(HAVE_GD_JPEG)
-#  include "gd.trm"
-# else
-#  include "png.trm"
-# endif
-#endif
-
 /* postscript */
 #include "post.trm"
 
