diff -urN gnuplot-4.0.0.orig/term/pdf.trm gnuplot-4.0.0/term/pdf.trm
--- gnuplot-4.0.0.orig/term/pdf.trm	2004-04-10 08:55:17.000000000 -0700
+++ gnuplot-4.0.0/term/pdf.trm	2005-05-24 15:33:52.000000000 -0700
@@ -343,7 +343,9 @@
 	myPDF = PDF_new();
 
     /*open new PDF file */ 
-    if (PDF_open_fp(myPDF, gpoutfile) == -1)
+    /* HBB FIXME 20040708: PDF_open_file is deprecated, will have to
+     * be re-done again soon. */
+    if (PDF_open_file(myPDF, outstr) == -1)
 	int_error(NO_CARET, "Error:cannot open PDF file .\n");
 
 #ifdef PDF_DONT_COMPRESS
