diff -ru vim72.orig/src/if_cscope.c vim72/src/if_cscope.c
--- vim72.orig/src/if_cscope.c	2010-01-14 16:53:17 +0100
+++ vim72/src/if_cscope.c	2010-01-14 16:55:06 +0100
@@ -2278,7 +2278,11 @@
 	/* Use sigaction() to limit the waiting time to two seconds. */
 	sigemptyset(&sa.sa_mask);
 	sa.sa_handler = sig_handler;
+#ifdef SA_NODEFER
+	/* interix misses SA_NODEFER. don't know what effect this
+	 * will have, but it's the best i can do here */
 	sa.sa_flags = SA_NODEFER;
+#endif
 	sigaction(SIGALRM, &sa, &old);
 	alarm(2); /* 2 sec timeout */
 
