diff -ru portage.orig/pym/portage/const.py portage/pym/portage/const.py
--- portage.orig/pym/portage/const.py	2009-03-30 13:14:27.000000000 +0200
+++ portage/pym/portage/const.py	2009-03-30 13:16:50.000000000 +0200
@@ -57,8 +57,8 @@
 MISC_SH_BINARY          = PORTAGE_BIN_PATH+"/misc-functions.sh"
 SANDBOX_BINARY          = BPREFIX+"/usr/bin/sandbox"
 FAKEROOT_BINARY         = BPREFIX+"/usr/bin/fakeroot"
-BASH_BINARY             = BPREFIX+"/bin/bash"
-MOVE_BINARY             = BPREFIX+"/bin/mv"
+BASH_BINARY             = BPREFIX+"/usr/lib/portage/bin/ebuild-helpers/bash"
+MOVE_BINARY             = BPREFIX+"/usr/lib/portage/bin/ebuild-helpers/mv"
 PRELINK_BINARY          = EPREFIX+"/usr/sbin/prelink"
 
 WORLD_FILE              = PRIVATE_PATH + "/world"
