diff -ru prefix-portage-2.2.00.14153.orig/bin/dispatch-conf prefix-portage-2.2.00.14153/bin/dispatch-conf
--- prefix-portage-2.2.00.14153.orig/bin/dispatch-conf	2009-08-26 08:51:24 +0200
+++ prefix-portage-2.2.00.14153/bin/dispatch-conf	2009-08-26 09:15:08 +0200
@@ -19,6 +19,7 @@
 from random import *
 import atexit, commands, re, shutil, stat, sys
 
+import portage
 from portage import os
 from portage import dispatch_conf
 from portage.process import find_binary
diff -ru prefix-portage-2.2.00.14153.orig/bin/quickpkg prefix-portage-2.2.00.14153/bin/quickpkg
--- prefix-portage-2.2.00.14153.orig/bin/quickpkg	2009-08-26 08:51:24 +0200
+++ prefix-portage-2.2.00.14153/bin/quickpkg	2009-08-26 09:20:19 +0200
@@ -5,15 +5,15 @@
 
 import errno, signal, sys
 from itertools import izip
+import portage
+
+from portage import os
 
 # for an explanation on this logic, see pym/_emerge/__init__.py
 if os.environ.__contains__("PORTAGE_PYTHONPATH"):
 	sys.path.insert(0, os.environ["PORTAGE_PYTHONPATH"])
 else:
 	sys.path.insert(0, os.path.join(os.path.dirname(os.path.dirname(os.path.realpath(__file__))), "pym"))
-import portage
-
-from portage import os
 
 def quickpkg_main(options, args, eout):
 	from portage import catsplit, dep_expand, flatten, isvalidatom, xpak
