Index: prefix-portage-2.1.13.4497/bin/misc-functions.sh
===================================================================
--- prefix-portage-2.1.13.4497.orig/bin/misc-functions.sh
+++ prefix-portage-2.1.13.4497/bin/misc-functions.sh
@@ -182,6 +182,16 @@ install_qa_check() {
 		unset INSTALLTOD
 	fi
 
+	if [[ -d ${D}/${PREFIX} ]] ; then
+		declare -i INSTALLTOD=0
+		for i in $(find "${D}/${PREFIX}/"); do
+			echo "QA Notice: /${i##${D}/${PREFIX}} double prefix"
+			((INSTALLTOD++))
+		done
+		ewarn "QA concerns: ${INSTALLTOD} double prefix files installed "
+		unset INSTALLTOD
+	fi
+
 	local find_log="${T}/find-portage-log"
 	find "${D}"/ -user ${PORTAGE_USER:-portage} -print0 > "${find_log}"
 	if [[ -s ${find_log} ]] ; then
@@ -239,7 +249,13 @@ preinst_mask() {
 		 eerror "${FUNCNAME}: IMAGE is unset"
 		 return 1
 	else
-		IMAGE="${IMAGE}/${EPREFIX}"
+		plen=${#EPREFIX}
+		ilen=${#IMAGE}
+		# check if prefixed at end
+		pend=${IMAGE:$[ilen-plen]:${plen}}
+		if [ "${pend}" != "${EPREFIX}" ]; then
+			IMAGE="${IMAGE}${EPREFIX}"
+		fi
 	fi
 	# remove man pages, info pages, docs if requested
 	for f in man info doc; do
