Index: prefix-portage-2.1.13.4497/bin/misc-functions.sh
===================================================================
--- prefix-portage-2.1.13.4497.orig/bin/misc-functions.sh
+++ prefix-portage-2.1.13.4497/bin/misc-functions.sh
@@ -363,12 +363,14 @@ dyn_package() {
 	install_mask "${PORTAGE_BUILDDIR}/image" ${PKG_INSTALL_MASK}
 	local pkg_dest="${PKGDIR}/All/${PF}.tbz2"
 	local pkg_tmp="${PKGDIR}/All/${PF}.tbz2.$$"
+	addwrite "${PKGDIR}"
+
 	# Sandbox is disabled in case the user wants to use a symlink
 	# for $PKGDIR and/or $PKGDIR/All.
 	export SANDBOX_ON="0"
 	tar cpvf - ./ | bzip2 -f > "${pkg_tmp}" || die "Failed to create tarball"
 	cd ..
-	python -c "import xpak; t=xpak.tbz2('${pkg_tmp}'); t.recompose('${PORTAGE_BUILDDIR}/build-info')"
+	PYTHONPATH=${EPREFIX}/usr/lib/portage/pym python -c "import xpak; t=xpak.tbz2('${pkg_tmp}'); t.recompose('${PORTAGE_BUILDDIR}/build-info')"
 	if [ $? -ne 0 ]; then
 		rm -f "${pkg_tmp}"
 		die "Failed to append metadata to the tbz2 file"
