--- pym/portage/__init__.py.orig	2009-10-07 18:59:52 +0200
+++ pym/portage/__init__.py	2009-10-07 19:00:03 +0200
@@ -3841,7 +3841,7 @@
 
 # In some cases, openpty can be slow when it fails. Therefore,
 # stop trying to use it after the first failure.
-if platform.system() in ["SunOS", "Darwin"]:
+if platform.system() in ["SunOS", "Darwin", "AIX", "FreeBSD"]:
 	# disable the use of openpty on Solaris as it seems Python's openpty
 	# implementation doesn't play nice on Solaris with Portage's
 	# behaviour causing hangs/deadlocks.
