Some finetuning for the preserve-libs feature on AIX.

--- bin/misc-functions.sh.orig	2009-10-29 20:34:41 +0100
+++ bin/misc-functions.sh	2009-10-29 20:35:53 +0100
@@ -447,6 +447,13 @@
 		fi
 
 		rm -f "${PORTAGE_BUILDDIR}"/build-info/NEEDED.XCOFF.1
+
+		local neededfd
+		for neededfd in {3..1024} none; do ( : <&${neededfd} ) 2>/dev/null || break; done
+		[[ ${neededfd} != none ]] || die "cannot find free file descriptor handle"
+
+		eval "exec ${neededfd}>\"${PORTAGE_BUILDDIR}\"/build-info/NEEDED.XCOFF.1" || die "cannot open ${PORTAGE_BUILDDIR}/build-info/NEEDED.XCOFF.1"
+
 		find "${ED}" -not -type d -exec \
 			"${EPREFIX}/usr/bin/aixdll-query" '{}' FILE MEMBER FLAGS FORMAT RUNPATH DEPLIBS ';' \
 			> "${T}"/needed 2>/dev/null
@@ -483,7 +490,7 @@
 
 			if [[ ${prev_FILE} != ${FILE} ]]; then
 				# Save NEEDED information for the archive library stub
-				echo "${FORMAT##* }${FORMAT%%-*};${FILE#${D%/}};${FILE##*/};;" >> "${PORTAGE_BUILDDIR}"/build-info/NEEDED.XCOFF.1
+				echo "${FORMAT##* }${FORMAT%%-*};${FILE#${D%/}};${FILE##*/};;" >&${neededfd}
 			fi
 
 			# Make sure we disallow insecure RUNPATH's
@@ -491,8 +498,8 @@
 			# (older, broken libtools would do this).  Also check for null paths
 			# because the loader will search $PWD when it finds null paths.
 			# And we really want absolute paths only.
-			if [[ -n $(echo ":${RUNPATH}:" | grep -E "(${PORTAGE_BUILDDIR}|::|:[^/])") ]]; then
-				insecure_rpath_list="${insecure_rpath_list}\n${FILE}"
+			if [[ " ${FLAGS} " == *" SHROBJ "* && -n $(echo ":${RUNPATH}:" | grep -E "(${PORTAGE_BUILDDIR}|::|:[^/])") ]]; then
+				insecure_rpath_list="${insecure_rpath_list}\n${FILE}${MEMBER:+[${MEMBER}]}"
 			fi
 
 			# Although we do have runtime linking, we don't want undefined symbols.
@@ -511,9 +518,11 @@
 
 			[[ -n ${MEMBER} ]] && MEMBER="[${MEMBER}]"
 			# Save NEEDED information
-			echo "${FORMAT##* }${FORMAT%%-*};${FILE}${MEMBER};${FILE##*/}${MEMBER};${RUNPATH};${needed}" >> "${PORTAGE_BUILDDIR}"/build-info/NEEDED.XCOFF.1
+			echo "${FORMAT##* }${FORMAT%%-*};${FILE}${MEMBER};${FILE##*/}${MEMBER};${RUNPATH};${needed}" >&${neededfd}
 		done <"${T}"/needed
 
+		eval "exec ${neededfd}>&-" || die "cannot close handle to ${PORTAGE_BUILDDIR}/build-info/NEEDED.XCOFF.1"
+
 		if [[ -n ${undefined_symbols_list} ]]; then
 			vecho -ne '\a\n'
 			eqawarn "QA Notice: The following files contain undefined symbols."
@@ -539,7 +539,7 @@
 			eqawarn " with 'prefix' as the maintaining herd of the package."
 			eqawarn "${insecure_rpath_list}"
 			vecho -ne '\a\n'
-			if [[ -n ${x} ]] || has stricter ${FEATURES} ; then
+			if has stricter ${FEATURES} ; then
 				insecure_rpath=1
 			fi
 		fi
--- pym/portage/dbapi/vartree.py.orig	2009-10-30 22:39:13.957455749 +0100
+++ pym/portage/dbapi/vartree.py	2009-10-30 22:39:17.947318507 +0100
@@ -1798,7 +1798,7 @@
 			self.linkmap = LinkageMapMachO(self)
 		elif chost.find('interix') >= 0 or chost.find('winnt') >= 0:
 			self.linkmap = LinkageMapPeCoff(self)
-		elif chost.find('aix'):
+		elif chost.find('aix') >= 0:
 			self.linkmap = LinkageMapXCoff(self)
 		else:
 			self.linkmap = LinkageMap(self)
