--- prefix-portage-2.1.13.4497/cnf/make.globals
+++ prefix-portage-2.1.13.4497/cnf/make.globals
@@ -57,7 +57,7 @@                                              
 PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --compress --force --whole-file --delete --delete-after --stats --timeout=180 --exclude='/distfiles' --exclude='/local' --exclude='/packages'"
 
 # Minimal CONFIG_PROTECT
-CONFIG_PROTECT="@DOMAIN_PREFIX@/etc"
+CONFIG_PROTECT="/etc"
 
 # Disable auto-use
 USE_ORDER="env:pkg:conf:defaults"
--- prefix-portage-2.1.13.4497/bin/dispatch-conf
+++ prefix-portage-2.1.13.4497/bin/dispatch-conf
@@ -78,6 +78,9 @@
         #
 
         for path in config_paths.split ():
+            if not path.startswith(EPREFIX):
+                path=EPREFIX+path
+
             if not os.path.exists (path):
                 continue
 
--- prefix-portage-2.1.13.4497/bin/etc-update
+++ prefix-portage-2.1.13.4497/bin/etc-update
@@ -42,7 +42,11 @@ function scan() {
 		exit 1
 	fi
 
+	prefix_len=${#EPREFIX}
 	for path in ${CONFIG_PROTECT} ; do
+		if [ "${path::${prefix_len}}" != "${EPREFIX}" ]; then
+			path=${EPREFIX}${path}
+		fi
 		[ ! -d ${path} ] && continue
 
 		ofile=""
--- prefix-portage-2.1.13.4497/pym/dispatch_conf.py
+++ prefix-portage-2.1.13.4497/pym/dispatch_conf.py
@@ -12,6 +12,7 @@ import os, sys, commands, shutil
 
 sys.path = ["/usr/lib/portage/pym"]+sys.path
 import portage
+from portage_const import EPREFIX
 
 RCS_BRANCH = '1.1.1'
 RCS_LOCK = 'rcs -ko -M -l'
@@ -23,12 +24,12 @@ DIFF3_MERGE = 'diff3 -mE %s %s %s >%s'
 
 def read_config(mandatory_opts):
     try:
-        opts = portage.getconfig('/etc/dispatch-conf.conf')
+        opts = portage.getconfig(EPREFIX+'/etc/dispatch-conf.conf')
     except:
         opts = None
 
     if not opts:
-        print >> sys.stderr, 'dispatch-conf: Error reading /etc/dispatch-conf.conf; fatal'
+        print >> sys.stderr, 'dispatch-conf: Error reading '+EPREFIX+'/etc/dispatch-conf.conf; fatal'
         sys.exit(1)
 
     for key in mandatory_opts:
