diff -ru tcp_wrappers_7.6.orig/tcpd.h tcp_wrappers_7.6/tcpd.h
--- tcp_wrappers_7.6.orig/tcpd.h	2009-07-07 11:15:38 +0200
+++ tcp_wrappers_7.6/tcpd.h	2009-07-07 11:19:44 +0200
@@ -21,7 +21,9 @@
 #include <netinet/in.h>
 #include <stdio.h>
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 /* Structure to describe one communications endpoint. */
 
@@ -255,6 +257,8 @@
 extern char *my_strtok();
 #endif
 
-__END_DECLS
+#ifdef __cplusplus
+} /* extern "C" */
+#endif
 
 #endif /* tcpd.h */
