From 852568991b251e9366da167f1b746a0a1db6adf0 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 10 Mar 2008 09:31:51 -0400
Subject: [PATCH] Typo fix.

---
 XI.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/XI.h b/XI.h
index ec9bee2..3b11860 100644
--- a/XI.h
+++ b/XI.h
@@ -49,6 +49,8 @@ SOFTWARE.
 #ifndef _XI_H_
 #define _XI_H_
 
+#define sz_xGetExtensionVersionReq             8
+#define sz_xGetExtensionVersionReply           32
 #define sz_xListInputDevicesReq			4
 #define sz_xListInputDevicesReply		32
 #define sz_xOpenDeviceReq			8
-- 
1.5.4.3

