diff -ru xterm-248.orig/main.c xterm-248/main.c
--- xterm-248.orig/main.c	2009-09-30 15:11:48 +0200
+++ xterm-248/main.c	2009-09-30 15:14:08 +0200
@@ -135,6 +135,10 @@
 #include <grp.h>		/* initgroups() */
 #endif
 
+#ifdef __INTERIX
+extern char ** environ;
+#endif
+
 #ifdef USE_ISPTS_FLAG
 static Bool IsPts = False;
 #endif
@@ -2415,7 +2419,7 @@
     }
 }
 
-#if defined(__osf__) || (defined(__GLIBC__) && !defined(USE_USG_PTYS)) || defined(__DragonFly__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
+#if defined(__osf__) || (defined(__GLIBC__) && !defined(USE_USG_PTYS)) || defined(__DragonFly__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__) || defined(__INTERIX)
 #define USE_OPENPTY 1
 static int opened_tty = -1;
 #endif
diff -ru xterm-248.orig/misc.c xterm-248/misc.c
--- xterm-248.orig/misc.c	2009-09-30 15:11:48 +0200
+++ xterm-248/misc.c	2009-09-30 15:13:17 +0200
@@ -77,6 +77,10 @@
 #include <X11/Sunkeysym.h>
 #endif
 
+#ifdef __INTERIX
+extern char ** environ;
+#endif
+
 #ifdef HAVE_LANGINFO_CODESET
 #include <langinfo.h>
 #endif
