diff -ru xterm-232.orig/main.c xterm-232/main.c
--- xterm-232.orig/main.c	2008-03-07 09:25:43 +0100
+++ xterm-232/main.c	2008-03-07 10:12:55 +0100
@@ -135,6 +135,10 @@
 #include <grp.h>		/* initgroups() */
 #endif
 
+#ifdef __INTERIX
+extern char ** environ;
+#endif
+
 #ifdef USE_ISPTS_FLAG
 static Bool IsPts = False;
 #endif
@@ -2386,7 +2390,7 @@
     }
 }
 
-#if defined(__osf__) || (defined(__GLIBC__) && !defined(USE_USG_PTYS)) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
+#if defined(__osf__) || (defined(__GLIBC__) && !defined(USE_USG_PTYS)) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__) || defined(__INTERIX)
 #define USE_OPENPTY 1
 static int opened_tty = -1;
 #endif
diff -ru xterm-232.orig/misc.c xterm-232/misc.c
--- xterm-232.orig/misc.c	2008-03-07 09:25:43 +0100
+++ xterm-232/misc.c	2008-03-07 10:00:06 +0100
@@ -76,6 +76,10 @@
 #include <X11/Sunkeysym.h>
 #endif
 
+#ifdef __INTERIX
+extern char ** environ;
+#endif
+
 #ifdef HAVE_LANGINFO_CODESET
 #include <langinfo.h>
 #endif
