diff -ru libcddb-1.3.0.orig/lib/cddb_cmd.c libcddb-1.3.0/lib/cddb_cmd.c
--- libcddb-1.3.0.orig/lib/cddb_cmd.c	2008-03-17 16:31:14 +0100
+++ libcddb-1.3.0/lib/cddb_cmd.c	2008-03-17 16:31:50 +0100
@@ -974,7 +974,11 @@
     free(aux);                  /* free temporary buffer */
     /* extract disc ID */
     aux = cddb_regex_get_string(line, matches, 2);
+#ifdef __INTERIX
+    disc->discid = strtol(aux, NULL, 16);
+#else
     disc->discid = strtoll(aux, NULL, 16);
+#endif
     free(aux);                  /* free temporary buffer */
     /* extract artist and title */
     if (matches[4].rm_so != -1) {
diff -ru libcddb-1.3.0.orig/lib/cddb_regex.c libcddb-1.3.0/lib/cddb_regex.c
--- libcddb-1.3.0.orig/lib/cddb_regex.c	2008-03-17 16:31:14 +0100
+++ libcddb-1.3.0/lib/cddb_regex.c	2008-03-17 16:32:18 +0100
@@ -135,7 +135,11 @@
     long long h;
 
     buf = cddb_regex_get_string(s, matches, idx);
+#ifdef __INTERIX
+    h = strtol(buf, &endp, 16);
+#else
     h = strtoll(buf, &endp, 16);
+#endif
     free(buf);
     return (unsigned long)(h & 0xffffffff);
 }
