diff -ru espgs-8.15.4.orig/configure.ac espgs-8.15.4/configure.ac
--- espgs-8.15.4.orig/configure.ac	2008-04-11 09:47:49 +0200
+++ espgs-8.15.4/configure.ac	2008-04-11 10:16:59 +0200
@@ -829,7 +829,7 @@
 
 AC_ARG_ENABLE(threads, [  --enable-threads        enable threaded output (default=no)],
 [	case `uname` in
-		Linux* | IRIX* | SunOS*)
+		Linux* | IRIX* | SunOS* | Interix*)
 		THREADSYNC="posync"
 		THREADLIBS="-lpthread"
 		;;
@@ -838,7 +838,7 @@
 		THREADLIBS="-pthread"
 		;;
 		*)
-		AC_MSG_ERROR(Sorry, threading support not available on this platform!)
+		AC_MSG_ERROR([Sorry, threading support not available on this platform!])
 		;;
 	esac
 ])
