--- passook.orig	Fri Jan 26 15:02:26 2001
+++ passook	Fri Apr 26 02:03:22 2002
@@ -38,7 +38,7 @@
 }
 
 sub fourletterword {
-	my $grepstring = 'egrep "^....$" /usr/dict/words';
+	my $grepstring = 'egrep "^....$" /usr/share/dict/words';
 	my @fourletters = split(/\n/,`$grepstring`); 
 	my $word = $fourletters[rand(@fourletters)];
         $word = &tangle($word) if ($p<5);
@@ -46,7 +46,7 @@
 }
 
 sub threeletterword {
-	my $grepstring = 'egrep "^...$" /usr/dict/words';
+	my $grepstring = 'egrep "^...$" /usr/share/dict/words';
 	my @threeletters = split(/\n/,`$grepstring`);
 	my $word = $threeletters[rand(@threeletters)];
 	$word = &tangle($word) if ($p<5);
