diff -ur mpg123-1.0rc2.orig/configure.ac mpg123-1.0rc2/configure.ac
--- mpg123-1.0rc2.orig/configure.ac	2007-12-09 17:13:26.000000000 +0200
+++ mpg123-1.0rc2/configure.ac	2007-12-10 20:24:16.000000000 +0200
@@ -577,7 +577,7 @@
     more_sources="$s_gen"
   ;;
   altivec)
-    ADD_CFLAGS="$ADD_CFLAGS -maltivec -faltivec"
+    ADD_CFLAGS="$ADD_CFLAGS -maltivec"
     ADD_CPPFLAGS="$ADD_CPPFLAGS -DREAL_IS_FLOAT -DOPT_ALTIVEC"
     more_sources="$s_altivec"
   ;;
