diff -ur mpg123-1.0.1.orig/configure mpg123-1.0.1/configure
--- mpg123-1.0.1.orig/configure	2007-12-28 17:30:35.000000000 +0200
+++ mpg123-1.0.1/configure	2007-12-30 16:17:19.000000000 +0200
@@ -16790,7 +16790,7 @@
     more_sources="$s_gen"
   ;;
   altivec)
-    ADD_CFLAGS="$ADD_CFLAGS -maltivec -faltivec"
+    ADD_CFLAGS="$ADD_CFLAGS -maltivec"
     ADD_CPPFLAGS="$ADD_CPPFLAGS -DREAL_IS_FLOAT -DOPT_ALTIVEC"
     more_sources="$s_altivec"
   ;;
diff -ur mpg123-1.0.1.orig/configure.ac mpg123-1.0.1/configure.ac
--- mpg123-1.0.1.orig/configure.ac	2007-12-28 17:29:04.000000000 +0200
+++ mpg123-1.0.1/configure.ac	2007-12-30 16:17:30.000000000 +0200
@@ -584,7 +584,7 @@
     more_sources="$s_gen"
   ;;
   altivec)
-    ADD_CFLAGS="$ADD_CFLAGS -maltivec -faltivec"
+    ADD_CFLAGS="$ADD_CFLAGS -maltivec"
     ADD_CPPFLAGS="$ADD_CPPFLAGS -DREAL_IS_FLOAT -DOPT_ALTIVEC"
     more_sources="$s_altivec"
   ;;
