diff --git a/gcc/treelang/parse.y b/gcc/treelang/parse.y
index 6522520..17abfb1 100644
--- a/gcc/treelang/parse.y
+++ b/gcc/treelang/parse.y
@@ -685,9 +685,9 @@ NAME LEFT_PARENTHESIS expressions_with_commas_opt RIGHT_PARENTHESIS {
 
 expressions_with_commas_opt:
 /* Nil.  */ {
-$$ = 0
+$$ = 0;
 }
-|expressions_with_commas { $$ = $1 }
+|expressions_with_commas { $$ = $1; }
 ;
 
 expressions_with_commas:
