diff -ru gcc-4.2.4.orig/gcc/config/interix.h gcc-4.2.4/gcc/config/interix.h
--- gcc-4.2.4.orig/gcc/config/interix.h	2009-09-23 10:19:11 +0200
+++ gcc-4.2.4/gcc/config/interix.h	2009-09-23 10:09:59 +0200
@@ -57,6 +57,12 @@
 #undef STARTFILE_SPEC
 #define STARTFILE_SPEC  \
   "%{!shared:%{pg:gcrt0%O%s}%{!pg:%{p:mcrt0%O%s}%{!p:crt0%O%s}}} %{shared:crti%O%s}"
+ 
+/* include /usr/lib/x86 in the search by default, so that the x86 compiled GCC
+ * is able to run on a x64 SUA installation. */
+#undef STARTFILE_PREFIX_SPEC
+#define STARTFILE_PREFIX_SPEC \
+	"/usr/lib/x86/ /usr/lib/ /lib/"
 
 #undef WORD_SWITCH_TAKES_ARG
 #define WORD_SWITCH_TAKES_ARG(STR)					\
