--- a/fixincludes/inclhack.def
+++ b/fixincludes/inclhack.def
@@ -1325,6 +1325,19 @@
 };
 
 /*
+ * https://gcc.gnu.org/bugzilla/show_bug.cgi?id=93082
+ */
+fix = {
+    hackname  = darwin_authorization;
+    mach      = "*-*-darwin*";
+    files     = Frameworks/Security.framework/Headers/Authorization.h;
+    select    = "static const size_t kAuthorizationExternalFormLength = 32;\n";
+    c_fix     = format;
+    c_fix_arg = "enum { kAuthorizationExternalFormLength = 32 };\n";
+    test_text = "static const size_t kAuthorizationExternalFormLength = 32;\n";
+};
+
+/*
  *  For the AAB_darwin7_9_long_double_funcs fix (and later fixes for long long)
  *  to be useful, the main math.h must use <> and not "" includes.
  */
--- a/fixincludes/fixincl.x
+++ b/fixincludes/fixincl.x
@@ -2686,6 +2686,43 @@
 
 /* * * * * * * * * * * * * * * * * * * * * * * * * *
  *
+ *  Description of Darwin_Authorization fix
+ */
+tSCC zDarwin_AuthorizationName[] =
+     "darwin_authorization";
+
+/*
+ *  File name selection pattern
+ */
+tSCC zDarwin_AuthorizationList[] =
+  "Frameworks/Security.framework/Headers/Authorization.h\0";
+/*
+ *  Machine/OS name selection pattern
+ */
+tSCC* apzDarwin_AuthorizationMachs[] = {
+        "*-*-darwin*",
+        (const char*)NULL };
+
+/*
+ *  content selection pattern - do fix if pattern found
+ */
+tSCC zDarwin_AuthorizationSelect0[] =
+       "static const size_t kAuthorizationExternalFormLength = 32;\n";
+
+#define    DARWIN_AUTHORIZATION_TEST_CT  1
+static tTestDesc aDarwin_AuthorizationTests[] = {
+  { TT_EGREP,    zDarwin_AuthorizationSelect0, (regex_t*)NULL }, };
+
+/*
+ *  Fix Command Arguments for Darwin_Authorization
+ */
+static const char* apzDarwin_AuthorizationPatch[] = {
+    "format",
+    "enum { kAuthorizationExternalFormLength = 32 };\n",
+    (char*)NULL };
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * *
+ *
  *  Description of Darwin_9_Long_Double_Funcs_2 fix
  */
 tSCC zDarwin_9_Long_Double_Funcs_2Name[] =
@@ -10476,9 +10513,9 @@
  *
  *  List of all fixes
  */
-#define REGEX_COUNT          296
+#define REGEX_COUNT          297
 #define MACH_LIST_SIZE_LIMIT 187
-#define FIX_COUNT            258
+#define FIX_COUNT            259
 
 /*
  *  Enumerate the fixes
@@ -10547,6 +10584,7 @@
     CTRL_QUOTES_USE_FIXIDX,
     CXX_UNREADY_FIXIDX,
     DARWIN_AVAILABILITYINTERNAL_FIXIDX,
+    DARWIN_AUTHORIZATION_FIXIDX,
     DARWIN_9_LONG_DOUBLE_FUNCS_2_FIXIDX,
     DARWIN_EXTERNC_FIXIDX,
     DARWIN_GCC4_BREAKAGE_FIXIDX,
@@ -11060,6 +11098,11 @@
      DARWIN_AVAILABILITYINTERNAL_TEST_CT, FD_MACH_ONLY | FD_SUBROUTINE,
      aDarwin_AvailabilityinternalTests,   apzDarwin_AvailabilityinternalPatch, 0 },
 
+  {  zDarwin_AuthorizationName,    zDarwin_AuthorizationList,
+     apzDarwin_AuthorizationMachs,
+     DARWIN_AUTHORIZATION_TEST_CT, FD_MACH_ONLY | FD_SUBROUTINE,
+     aDarwin_AuthorizationTests,   apzDarwin_AuthorizationPatch, 0 },
+
   {  zDarwin_9_Long_Double_Funcs_2Name,    zDarwin_9_Long_Double_Funcs_2List,
      apzDarwin_9_Long_Double_Funcs_2Machs,
      DARWIN_9_LONG_DOUBLE_FUNCS_2_TEST_CT, FD_MACH_ONLY | FD_SUBROUTINE,
