http://bugs.gentoo.org/show_bug.cgi?id=300948

--- terminal.c.old	2010-01-14 00:37:40.000000000 +0000
+++ terminal.c	2010-01-14 00:38:03.000000000 +0000
@@ -97,7 +97,7 @@
 static int tcap_initialized;
 
 #if !defined (__linux__)
-#  if defined (__EMX__) || defined (NEED_EXTERN_PC)
+#  if defined (__EMX__) || defined(__MINT__) || defined (NEED_EXTERN_PC)
 extern 
 #  endif /* __EMX__ || NEED_EXTERN_PC */
 char PC, *BC, *UP;
