--- support/shlib-install
+++ support/shlib-install
@@ -130,7 +130,7 @@
 	fi
 	;;
 
-bsdi4*|*gnu*|darwin*|macosx*|k*bsd*-gnu)
+bsdi4*|*gnu*|darwin*|macosx*|k*bsd*-gnu|*netbsd*)
 	# libname.so.M -> libname.so.M.N
 	${echo} ${RM} ${INSTALLDIR}/$LINK2
 	if [ -z "$uninstall" ]; then
@@ -144,7 +144,7 @@
 	fi
 	;;
 
-solaris2*|aix4.[2-9]*|osf*|irix[56]*|sysv[45]*|dgux*)
+solaris2*|aix4.[2-9]*|aix[5-9].*|osf*|irix[56]*|sysv[45]*|dgux*|interix*)
 	# libname.so -> libname.so.M
 	${echo} ${RM} ${INSTALLDIR}/$LINK1
 	if [ -z "$uninstall" ]; then
