diff -NrU5 audacious-plugins-1.5.1.orig/configure.ac audacious-plugins-1.5.1/configure.ac
--- audacious-plugins-1.5.1.orig/configure.ac	2008-09-03 12:59:59.000000000 +0200
+++ audacious-plugins-1.5.1/configure.ac	2008-09-03 13:02:47.000000000 +0200
@@ -1363,10 +1363,15 @@
 if test "x$have_mtp_up" = "xyes"; then
     have_mtp_up=yes
     PKG_CHECK_MODULES(MTP, [libmtp >= 0.1.3], [GENERAL_PLUGINS="$GENERAL_PLUGINS mtp_up" ], [have_mtp_up="no"])
     MTP_LIBS=`pkg-config --libs libmtp`
     MTP_CFLAGS=`pkg-config --cflags libmtp`
+    dnl Check for libmtp 0.3.0+
+    PKG_CHECK_EXISTS( [libmtp >= 0.3.0], [libmtp_030_pkg=yes], [libmtp_030_pkg=no])
+    if test x"$libmtp_030_pkg" = "xyes" ; then
+       AC_DEFINE([HAVE_LIBMTP_030], 1, [indicates if at least libmtp 0.3.0 is installed])
+    fi
 else
     have_mtp_up=no
     AC_MSG_RESULT([*** mtp upload plugin disabled by request ***])
 fi
 AC_SUBST(MTP_LIBS)
diff -NrU5 audacious-plugins-1.5.1.orig/src/mtp_up/mtp.c audacious-plugins-1.5.1/src/mtp_up/mtp.c
--- audacious-plugins-1.5.1.orig/src/mtp_up/mtp.c	2008-09-03 12:59:59.000000000 +0200
+++ audacious-plugins-1.5.1/src/mtp_up/mtp.c	2008-09-03 13:05:55.000000000 +0200
@@ -190,16 +190,21 @@
     tmp = g_strescape(from_path,NULL);
     filename=g_filename_from_uri(tmp,NULL,NULL);
 
     g_free(from_path);
     g_free(tmp);
-    parent_id = mtp_device->default_music_folder;
 
 #if DEBUG
     g_print("Uploading track '%s'\n",filename);
 #endif
+#ifdef HAVE_LIBMTP_030
+    gentrack->parent_id = mtp_device->default_music_folder;
+    ret = LIBMTP_Send_Track_From_File(mtp_device, filename , gentrack, NULL , NULL);
+#else
+    parent_id = mtp_device->default_music_folder;
     ret = LIBMTP_Send_Track_From_File(mtp_device, filename , gentrack, NULL , NULL, parent_id);
+#endif
     LIBMTP_destroy_track_t(gentrack);
     if (ret == 0) 
         g_print("Track upload finished!\n");
     else
     {
