diff -ru audacious-plugins-1.5.0.orig/configure.ac audacious-plugins-1.5.0/configure.ac
--- audacious-plugins-1.5.0.orig/configure.ac	2008-03-18 10:51:45 +0100
+++ audacious-plugins-1.5.0/configure.ac	2008-03-18 09:56:27 +0100
@@ -18,9 +18,9 @@
 AC_SUBST(LIBBEEP_MICRO_VERSION)
 
 dnl XXX kludge for FreeBSD
-if test -d "/usr/X11R6/include"; then
-	CPPFLAGS="$CPPFLAGS -I/usr/X11R6/include"
-fi
+dnl if test -d "/usr/X11R6/include"; then
+dnl	CPPFLAGS="$CPPFLAGS -I/usr/X11R6/include"
+dnl fi
 
 dnl append audacious header path to CPPFLAGS when --prefix is specified.
 if test "$prefix" != "NONE" ; then
diff -ru audacious-plugins-1.5.0.orig/src/aosd/ghosd-main.c audacious-plugins-1.5.0/src/aosd/ghosd-main.c
--- audacious-plugins-1.5.0.orig/src/aosd/ghosd-main.c	2008-03-18 10:32:21 +0100
+++ audacious-plugins-1.5.0/src/aosd/ghosd-main.c	2008-03-18 10:48:55 +0100
@@ -13,7 +13,11 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/time.h>
-#include <sys/poll.h>
+#ifndef __INTERIX
+# include <sys/poll.h>
+#else
+# include <sys/time.h>
+#endif
 #include <string.h>
 #include <time.h>
 #include <unistd.h>
@@ -86,16 +90,27 @@
 
   for (;;) {
     gettimeofday(&tv_now, NULL);
+
+#ifndef __INTERIX
     int dt = (until->tv_sec  - tv_now.tv_sec )*1000 +
              (until->tv_usec - tv_now.tv_usec)/1000;
     if (dt <= 0) break;
 
     struct pollfd pollfd = { ghosd_get_socket(ghosd), POLLIN, 0 };
     int ret = poll(&pollfd, 1, dt);
+#else
+	// no poll() on interix (it's broken). so use select().
+	fd_set selset;
+
+	FD_ZERO(&selset);
+	FD_SET(ghosd_get_socket(ghosd), &selset);
+
+	int ret = select(FD_SETSIZE, &selset, NULL, NULL, until);
+#endif
     if (ret < 0) {
       if ( errno != EINTR )
       {
-        perror("poll");
+        perror("poll/select");
         exit(1);
       }
       /* else go on, ignore EINTR */
diff -ru audacious-plugins-1.5.0.orig/src/demac/ape.h audacious-plugins-1.5.0/src/demac/ape.h
--- audacious-plugins-1.5.0.orig/src/demac/ape.h	2008-03-18 10:51:45 +0100
+++ audacious-plugins-1.5.0/src/demac/ape.h	2008-03-18 09:56:28 +0100
@@ -24,6 +24,10 @@
 #ifndef APE_H
 #define APE_H
 
+#if defined(__INTERIX)
+# include <stdint.h>
+#endif
+
 #define FFMIN(a,b) ((a) < (b) ? (a) : (b))
 
 #ifdef DEBUG
diff -ru audacious-plugins-1.5.0.orig/src/sexypsf/types.h audacious-plugins-1.5.0/src/sexypsf/types.h
--- audacious-plugins-1.5.0.orig/src/sexypsf/types.h	2008-03-18 10:51:45 +0100
+++ audacious-plugins-1.5.0/src/sexypsf/types.h	2008-03-18 09:56:28 +0100
@@ -3,6 +3,10 @@
 
 #include <inttypes.h>
 
+#if defined(__INTERIX)
+# include <stdint.h>
+#endif
+
 #define INLINE inline
 
 typedef int8_t s8;
diff -ru audacious-plugins-1.5.0.orig/src/neon/neon.c audacious-plugins-1.5.0/src/neon/neon.c
--- audacious-plugins-1.5.0.orig/src/neon/neon.c	2008-03-18 14:02:43 +0100
+++ audacious-plugins-1.5.0/src/neon/neon.c	2008-03-18 14:05:36 +0100
@@ -70,10 +70,13 @@
     neon_aud_vfs_metadata_impl
 };
 
+// doesn't work on interix!
+#ifndef __INTERIX
 /* bring ne_set_connect_timeout in as a weak reference, not using it
  * unless we have it available (neon 0.27) --nenolod
  */
 extern void ne_set_connect_timeout(ne_session *sess, int timeout) __attribute__ ((weak));
+#endif
 
 /*
  * ========
@@ -672,8 +675,11 @@
         ne_set_session_flag(handle->session, NE_SESSFLAG_ICYPROTO, 1);
         ne_set_session_flag(handle->session, NE_SESSFLAG_PERSIST, 0);
 
+		// doesn't work on interix
+		#ifndef __INTERIX
         if (ne_set_connect_timeout != NULL)
             ne_set_connect_timeout(handle->session, 10);
+		#endif
 
         ne_set_read_timeout(handle->session, 10);
         ne_set_useragent(handle->session, "Audacious/1.5.0");
