diff -ru -x '*.Po' -x '*.Plo' audacious-plugins-1.5.1.orig/src/sexypsf/types.h audacious-plugins-1.5.1/src/sexypsf/types.h
--- audacious-plugins-1.5.1.orig/src/sexypsf/types.h	2008-08-20 10:39:21 +0200
+++ audacious-plugins-1.5.1/src/sexypsf/types.h	2008-08-20 10:40:06 +0200
@@ -1,8 +1,6 @@
 #ifndef _SPSF_TYPES_H__
 #define _SPSF_TYPES_H__
 
-#include <inttypes.h>
-
 #if defined(__INTERIX)
 # include <stdint.h>
 #endif
diff -ru -x '*.Po' -x '*.Plo' audacious-plugins-1.5.1.orig/src/vtx/ay8912.c audacious-plugins-1.5.1/src/vtx/ay8912.c
--- audacious-plugins-1.5.1.orig/src/vtx/ay8912.c	2008-08-20 10:39:25 +0200
+++ audacious-plugins-1.5.1/src/vtx/ay8912.c	2008-08-20 10:41:30 +0200
@@ -1,6 +1,6 @@
 /* AY/YM emulator implementation. */
 
-#include <inttypes.h>
+#include <stdint.h>
 #include "ayemu.h"
 
 #define debuglog stderr;
diff -ru -x '*.Po' -x '*.Plo' audacious-plugins-1.5.1.orig/src/vtx/ayemu_8912.h audacious-plugins-1.5.1/src/vtx/ayemu_8912.h
--- audacious-plugins-1.5.1.orig/src/vtx/ayemu_8912.h	2008-08-20 10:39:26 +0200
+++ audacious-plugins-1.5.1/src/vtx/ayemu_8912.h	2008-08-20 10:40:52 +0200
@@ -6,7 +6,7 @@
 #define _AYEMU_ay8912_h
 
 #include <stddef.h>
-#include <inttypes.h>
+#include <stdint.h>
 
 BEGIN_C_DECLS
 
diff -ru -x '*.Po' -x '*.Plo' audacious-plugins-1.5.1.orig/src/vtx/ayemu_vtxfile.h audacious-plugins-1.5.1/src/vtx/ayemu_vtxfile.h
--- audacious-plugins-1.5.1.orig/src/vtx/ayemu_vtxfile.h	2008-08-20 10:39:26 +0200
+++ audacious-plugins-1.5.1/src/vtx/ayemu_vtxfile.h	2008-08-20 10:41:07 +0200
@@ -2,7 +2,7 @@
 #define _AYEMU_vtxfile_h
 
 #include <glib.h>
-#include <inttypes.h>
+#include <stdint.h>
 #include <audacious/plugin.h>
 #include "ayemu_8912.h"
 
diff -ru -x '*.Po' -x '*.Plo' audacious-plugins-1.5.1.orig/src/vtx/vtxfile.c audacious-plugins-1.5.1/src/vtx/vtxfile.c
--- audacious-plugins-1.5.1.orig/src/vtx/vtxfile.c	2008-08-20 10:39:26 +0200
+++ audacious-plugins-1.5.1/src/vtx/vtxfile.c	2008-08-20 10:41:49 +0200
@@ -3,7 +3,7 @@
 #include <errno.h>
 #include <assert.h>
 #include <ctype.h>
-#include <inttypes.h>
+#include <stdint.h>
 
 #include <audacious/plugin.h>
 
