diff -Naru postgresql-8.1.0.org/src/Makefile.shlib postgresql-8.1.0/src/Makefile.shlib
--- postgresql-8.1.0.org/src/Makefile.shlib	2005-11-12 20:26:09.000000000 +0000
+++ postgresql-8.1.0/src/Makefile.shlib	2005-11-12 20:28:28.000000000 +0000
@@ -363,10 +363,10 @@
 install-lib: install-lib-static install-lib-shared
 
 install-lib-static: lib$(NAME).a
-	$(INSTALL_STLIB) $< $(DESTDIR)$(libdir)/lib$(NAME).a
+	$(INSTALL_STLIB) $< $(DESTDIR)$(libdir)/lib$(NAME_STATIC_LIB).a
 ifeq ($(PORTNAME), darwin)
 	cd $(DESTDIR)$(libdir) && \
-	ranlib lib$(NAME).a
+	ranlib lib$(NAME_STATIC_LIB).a
 endif
 
 ifeq ($(enable_shared), yes)
diff -Naru postgresql-8.1.0.org/src/interfaces/libpq/Makefile postgresql-8.1.0/src/interfaces/libpq/Makefile
--- postgresql-8.1.0.org/src/interfaces/libpq/Makefile	2005-11-12 20:26:09.000000000 +0000
+++ postgresql-8.1.0/src/interfaces/libpq/Makefile	2005-11-12 20:28:02.000000000 +0000
@@ -20,6 +20,8 @@
 SO_MINOR_VERSION= 1
 DLTYPE= library
 
+NAME_STATIC_LIB= pq-$(SO_MAJOR_VERSION)
+
 override CPPFLAGS :=  -DFRONTEND -I$(srcdir) $(CPPFLAGS) -I$(top_builddir)/src/port
 ifneq ($(PORTNAME), win32)
 override CFLAGS += $(PTHREAD_CFLAGS)
@@ -135,7 +137,7 @@
 	$(INSTALL_DATA) $(srcdir)/libpq-fe.h $(DESTDIR)$(includedir)
 	$(INSTALL_DATA) $(srcdir)/libpq-int.h $(DESTDIR)$(includedir_internal)
 	$(INSTALL_DATA) $(srcdir)/pqexpbuffer.h $(DESTDIR)$(includedir_internal)
-	$(INSTALL_DATA) $(srcdir)/pg_service.conf.sample $(DESTDIR)$(datadir)/pg_service.conf.sample
+	$(INSTALL_DATA) $(srcdir)/pg_service.conf.sample $(DESTDIR)$(datadir)/pg_service-${SO_MAJOR_VERSION}.conf.sample
 
 installdirs:
 	$(mkinstalldirs) $(DESTDIR)$(libdir) $(DESTDIR)$(includedir) $(DESTDIR)$(includedir_internal)
diff -Naru postgresql-8.1.0.org/src/interfaces/libpq/nls.mk postgresql-8.1.0/src/interfaces/libpq/nls.mk
--- postgresql-8.1.0.org/src/interfaces/libpq/nls.mk	2005-11-12 20:26:09.000000000 +0000
+++ postgresql-8.1.0/src/interfaces/libpq/nls.mk	2005-11-12 20:28:02.000000000 +0000
@@ -1,5 +1,5 @@
 # $PostgreSQL: pgsql/src/interfaces/libpq/nls.mk,v 1.20 2005/01/14 08:57:06 petere Exp $
-CATALOG_NAME	:= libpq
+CATALOG_NAME	:= libpq-4
 AVAIL_LANGUAGES	:= af cs de es fr hr it ko nb pl pt_BR ru sk sl sv tr zh_CN zh_TW
 GETTEXT_FILES	:= fe-auth.c fe-connect.c fe-exec.c fe-lobj.c fe-misc.c fe-protocol2.c fe-protocol3.c fe-secure.c
 GETTEXT_TRIGGERS:= libpq_gettext pqInternalNotice:2
