diff -ru -x *.Po -x config* -x Makefile* -x *.Plo pkg-config-0.22.orig/glib-1.2.8/gerror.c pkg-config-0.22/glib-1.2.8/gerror.c
--- pkg-config-0.22.orig/glib-1.2.8/gerror.c	Fri Jan 18 15:15:06 2008
+++ pkg-config-0.22/glib-1.2.8/gerror.c	Fri Jan 18 15:20:36 2008
@@ -192,7 +192,7 @@
 static void
 stack_trace (char **args)
 {
-#ifndef NATIVE_WIN32
+#if !defined(NATIVE_WIN32) && !defined(__INTERIX)
   pid_t pid;
   int in_fd[2];
   int out_fd[2];
diff -ru -x *.Po -x config* -x Makefile* -x *.Plo pkg-config-0.22.orig/glib-1.2.8/gmain.c pkg-config-0.22/glib-1.2.8/gmain.c
--- pkg-config-0.22.orig/glib-1.2.8/gmain.c	Fri Jan 18 15:15:06 2008
+++ pkg-config-0.22/glib-1.2.8/gmain.c	Fri Jan 18 15:20:36 2008
@@ -54,6 +54,10 @@
 #endif /* HAVE_UNISTD_H */
 #include <errno.h>
 
+#ifdef __INTERIX
+#include <poll.h>
+#endif
+
 #ifdef NATIVE_WIN32
 #define STRICT
 #include <windows.h>
