diff -ru pkg-config-0.25.orig/glib-1.2.10/gmain.c pkg-config-0.25/glib-1.2.10/gmain.c
--- pkg-config-0.25.orig/glib-1.2.10/gmain.c	Wed Oct 20 08:19:01 2010
+++ pkg-config-0.25/glib-1.2.10/gmain.c	Wed Oct 20 08:21:35 2010
@@ -49,6 +49,9 @@
 #  undef events	 /* AIX 4.1.5 & 4.3.2 define this for SVR3,4 compatibility */
 #  undef revents /* AIX 4.1.5 & 4.3.2 define this for SVR3,4 compatibility */
 #endif /* GLIB_HAVE_SYS_POLL_H */
+#ifdef __INTERIX
+#  include <poll.h>
+#endif
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif /* HAVE_UNISTD_H */
