--- pkg.c.orig	2008-10-10 14:33:53.000000000 +0200
+++ pkg.c	2008-10-10 14:39:17.000000000 +0200
@@ -869,6 +869,8 @@
   system_directories = g_slist_append (NULL, g_strdup ("/usr/include"));
 #endif
 
+  system_directories = g_slist_append (system_directories, g_strdup ("@GENTOO_PORTAGE_EPREFIX@/usr/include"));
+
   c_include_path = g_getenv ("C_INCLUDE_PATH");
   if (c_include_path != NULL)
     {
@@ -933,23 +935,27 @@
 
 #ifdef PREFER_LIB64
 #define SYSTEM_LIBDIR "/usr/lib64"
+#define PREFIX_LIBDIR "@GENTOO_PORTAGE_EPREFIX@/usr/lib64"
 #else
 #define SYSTEM_LIBDIR "/usr/lib"
+#define PREFIX_LIBDIR "@GENTOO_PORTAGE_EPREFIX@/usr/lib"
 #endif
   count = 0;
   iter = pkg->L_libs;
   while (iter != NULL)
     {
       if (strcmp (iter->data, "-L" SYSTEM_LIBDIR) == 0 ||
-          strcmp (iter->data, "-L " SYSTEM_LIBDIR) == 0)
+          strcmp (iter->data, "-L " SYSTEM_LIBDIR) == 0 ||
+          strcmp (iter->data, "-L" PREFIX_LIBDIR) == 0 ||
+          strcmp (iter->data, "-L " PREFIX_LIBDIR) == 0)
         {
-          debug_spew ("Package %s has -L" SYSTEM_LIBDIR " in Libs\n",
+          debug_spew ("Package %s has -L" SYSTEM_LIBDIR " or -L" PREFIX_LIBDIR " in Libs\n",
                       pkg->name);
           if (g_getenv ("PKG_CONFIG_ALLOW_SYSTEM_LIBS") == NULL)
             {              
               iter->data = NULL;
               ++count;
-              debug_spew ("Removing -L" SYSTEM_LIBDIR " from libs for %s\n", pkg->key);
+              debug_spew ("Removing -L" SYSTEM_LIBDIR " and -L" PREFIX_LIBDIR " from libs for %s\n", pkg->key);
             }
         }
 
