--- src/plugin.c.orig	2009-06-16 08:41:35 +0200
+++ src/plugin.c	2009-06-16 08:41:04 +0200
@@ -135,6 +135,8 @@ int plugins_load(server *srv) {
 		buffer_append_string(srv->tmp_buf, modules);
 #if defined(__WIN32) || defined(__CYGWIN__)
 		buffer_append_string_len(srv->tmp_buf, CONST_STR_LEN(".dll"));
+#elif defined(__APPLE__)
+		buffer_append_string_len(srv->tmp_buf, CONST_STR_LEN(".bundle"));
 #else
 		buffer_append_string_len(srv->tmp_buf, CONST_STR_LEN(".so"));
 #endif
