diff -ru -x '*.Po' -x '*.Plo' lighttpd-1.4.20.orig/src/lighttpd-angel.c lighttpd-1.4.20/src/lighttpd-angel.c
--- lighttpd-1.4.20.orig/src/lighttpd-angel.c	2008-11-17 09:28:55 +0100
+++ lighttpd-1.4.20/src/lighttpd-angel.c	2008-11-17 10:05:46 +0100
@@ -74,9 +74,13 @@
 	sigaction(SIGPIPE, &act, NULL);
 	sigaction(SIGUSR1, &act, NULL);
 
+	#ifndef __INTERIX
 	act.sa_sigaction = sigaction_handler;
+	#endif
 	sigemptyset(&act.sa_mask);
+	#ifdef SA_SIGINFO
 	act.sa_flags = SA_SIGINFO;
+	#endif
 
 	sigaction(SIGINT,  &act, NULL);
 	sigaction(SIGTERM, &act, NULL);
diff -ru -x '*.Po' -x '*.Plo' lighttpd-1.4.20.orig/src/server.c lighttpd-1.4.20/src/server.c
--- lighttpd-1.4.20.orig/src/server.c	2008-11-17 09:28:57 +0100
+++ lighttpd-1.4.20/src/server.c	2008-11-17 10:03:13 +0100
@@ -752,10 +752,12 @@
 		 * */
 		if (srv->srvconf.groupname->used) {
 			setgid(grp->gr_gid);
+			#ifndef __INTERIX
 			setgroups(0, NULL);
 			if (srv->srvconf.username->used) {
 				initgroups(srv->srvconf.username->ptr, grp->gr_gid);
 			}
+			#endif
 		}
 #endif
 #ifdef HAVE_CHROOT
@@ -1151,7 +1153,7 @@
 
 				return -1;
 			} else {
-#ifdef HAVE_SIGACTION
+#if defined(HAVE_SIGACTION) && defined(SA_SIGINFO)
 				log_error_write(srv, __FILE__, __LINE__, "sdsd", 
 					"logfiles cycled UID =",
 					last_sighup_info.si_uid,
@@ -1460,7 +1462,7 @@
 		}
 	}
 
-#ifdef HAVE_SIGACTION
+#if defined(HAVE_SIGACTION) && defined(SA_SIGINFO)
 	log_error_write(srv, __FILE__, __LINE__, "sdsd", 
 			"server stopped by UID =",
 			last_sigterm_info.si_uid,
diff -ru -x '*.Po' -x '*.Plo' lighttpd-1.4.20.orig/src/spawn-fcgi.c lighttpd-1.4.20/src/spawn-fcgi.c
--- lighttpd-1.4.20.orig/src/spawn-fcgi.c	2008-11-17 09:28:57 +0100
+++ lighttpd-1.4.20/src/spawn-fcgi.c	2008-11-17 10:00:40 +0100
@@ -442,11 +442,13 @@
 
 			/* do the change before we do the chroot() */
 			setgid(grp->gr_gid);
+			#ifndef __INTERIX
 			setgroups(0, NULL); 
 
 			if (username) {
 				initgroups(username, grp->gr_gid);
 			}
+			#endif
 
 		}
 
