--- configure
+++ configure
@@ -101,15 +101,17 @@
 # check for libs
 search_lib() {
   for dir in ${DIRS};  do
-    for dbn in $@; do
-      for suffix in so dylib obj a; do
-        for lib in lib out; do
-          if test -r "${dir}/${lib}/lib${dbn}.${suffix}"; then
-            add_lib "${dir}/${lib}" "${dbn}" "${suffix}"
-	    echo "Found: lib${dbn}.${suffix} at ${dir}/${lib}"
-            return 0
-	  fi
-	done
+    for subdir in "" ${subdirs}; do
+      for dbn in $@; do
+        for suffix in so dylib obj a; do
+          for lib in lib out; do
+            if test -r "${dir}/${lib}/${subdir}/lib${dbn}.${suffix}"; then
+              add_lib "${dir}/${lib}/${subdir}" "${dbn}" "${suffix}"
+              echo "Found: lib${dbn}.${suffix} at ${dir}/${lib}/${subdir}"
+              return 0
+            fi
+          done
+        done
       done
     done
   done
@@ -121,7 +123,7 @@
 ######################### QT
 subdirs="/qt /qt4"
 search_includes Qt/qobject.h || err "The QT Library headerfiles were not found. Set QTDIR appropriately."
-subdirs="/qt/Qt /qt4/Qt /Qt"
+subdirs="/qt/Qt /qt4/Qt /Qt /qt4"
 search_includes qobject.h || err "The QT Library headerfiles were not found. Set QTDIR appropriately."
 search_lib QtGui4 QtGui || err "The QT library was not found."
 search_lib ltdl || err "Libtool ltdl not found."
