diff -urNp xca-0.6.4.org/lib/x509rev.cpp xca-0.6.4/lib/x509rev.cpp
--- xca-0.6.4.org/lib/x509rev.cpp	2007-08-13 21:55:50.000000000 +0300
+++ xca-0.6.4/lib/x509rev.cpp	2007-10-23 17:35:42.000000000 +0200
@@ -7,7 +7,12 @@
 
 #include "x509rev.h"
 
-#if OPENSSL_VERSION_NUMBER >= 0x00908000L
+#if OPENSSL_VERSION_NUMBER >= 0x0090806fL
+#define X509_REVOKED_dup(x5r) (X509_REVOKED *)ASN1_dup( \
+	(i2d_of_void*)i2d_X509_REVOKED, \
+	(d2i_of_void*)d2i_X509_REVOKED, \
+	(char *)x5r)
+#elif OPENSSL_VERSION_NUMBER >= 0x00908000L
 #define X509_REVOKED_dup(x5r) \
 	ASN1_dup_of (X509_REVOKED, i2d_X509_REVOKED, d2i_X509_REVOKED, x5r)
 
