--- xca-0.6.4/configure	2007-08-13 20:55:50.000000000 +0200
+++ xca-0.6.4.new/configure	2007-09-18 14:23:10.000000000 +0200
@@ -103,17 +103,19 @@
 # check for libs
 search_lib() {
   for dir in ${DIRS};  do
+for subdir in "" ${subdirs}; do
     for dbn in $@; do
       for suffix in so dylib obj a; do
         for lib in lib out; do
-          if test -r "${dir}/${lib}/lib${dbn}.${suffix}"; then
-            add_lib "${dir}/${lib}" "${dbn}" "${suffix}"
-	    echo "Found: lib${dbn}.${suffix} at ${dir}/${lib}"
+          if test -r "${dir}/${lib}/${subdir}/lib${dbn}.${suffix}"; then
+            add_lib "${dir}/${lib}/${subdir}" "${dbn}" "${suffix}"
+	    echo "Found: lib${dbn}.${suffix} at ${dir}/${lib}/${subdir}"
             return 0
 	  fi
 	done
       done
     done
+done
   done
   return 1
 }
@@ -123,7 +125,7 @@
 ######################### QT
 subdirs="/qt /qt4"
 search_includes Qt/qobject.h || err "The QT Library headerfiles were not found. Set QTDIR appropriately."
-subdirs="/qt/Qt /qt4/Qt"
+subdirs="/qt/Qt /qt4/Qt /qt4"
 search_includes qobject.h || err "The QT Library headerfiles were not found. Set QTDIR appropriately."
 search_lib QtGui4 QtGui || err "The QT library was not found."
 search_lib c_r || true
