commit 6bd98aeca2c27900c4b8101d6a29db84bb9db372
Author: Lasse Collin <lasse.collin@tukaani.org>
Date:   Sat Jun 21 15:18:05 2008 +0300

    Remove invalid semicolon. Thanks to Steffi Wassermann.

diff --git a/src/sdk/7zip/Compress/LZMA/LZMAEncoder.cpp b/src/sdk/7zip/Compress/LZMA/LZMAEncoder.cpp
index a88020f..5770504 100644
--- a/src/sdk/7zip/Compress/LZMA/LZMAEncoder.cpp
+++ b/src/sdk/7zip/Compress/LZMA/LZMAEncoder.cpp
@@ -182,7 +182,7 @@ UInt32 CLiteralEncoder2::GetPrice(bool matchMode, Byte matchByte, Byte symbol) c
     context = (context << 1) | bit;
   }
   return price;
-};
+}
 
 
 namespace NLength {
