--- src/lzma/Makefile.am.old	2008-04-28 22:42:51.000000000 +0100
+++ src/lzma/Makefile.am	2008-04-28 22:43:08.000000000 +0100
@@ -25,7 +26,8 @@
 	cd $(DESTDIR)$(bindir) && \
 	rm -f unlzma lzcat && \
 	$(LN_S) lzma unlzma && \
-	$(LN_S) lzma lzcat
+	$(LN_S) lzma lzcat && \
+	stack -S 512K lzma
 
 install-data-hook:
 	cd $(DESTDIR)$(mandir)/man1 && \
--- src/lzma/Makefile.in.old	2008-12-16 16:47:31.000000000 +0000
+++ src/lzma/Makefile.in	2008-12-16 16:47:42.000000000 +0000
@@ -671,7 +671,8 @@
 	cd $(DESTDIR)$(bindir) && \
 	rm -f unlzma lzcat && \
 	$(LN_S) lzma unlzma && \
-	$(LN_S) lzma lzcat
+	$(LN_S) lzma lzcat && \
+	stack -S 512K lzma
 
 install-data-hook:
 	cd $(DESTDIR)$(mandir)/man1 && \
