diff -Naur libcdio-0.79-orig/include/cdio++/iso9660.hpp libcdio-0.79/include/cdio++/iso9660.hpp
--- libcdio-0.79-orig/include/cdio++/iso9660.hpp	2006-03-07 15:04:01.000000000 -0600
+++ libcdio-0.79/include/cdio++/iso9660.hpp	2008-01-05 09:32:02.000000000 -0600
@@ -29,6 +29,7 @@
 
 #include <cdio/iso9660.h>
 #include <cdio++/cdio.hpp>
+#include <stdlib.h>
 #include <string.h>
 #include <vector>		// vector class library
 using namespace std;
