diff -urN oniguruma/regint.h oniguruma-ppc/regint.h
--- oniguruma/regint.h	2004-09-18 14:33:09.000000000 +0900
+++ oniguruma-ppc/regint.h	2004-10-10 17:04:02.280027232 +0900
@@ -539,11 +539,11 @@
 #define ARG_MEMNUM       4
 #define ARG_OPTION       5
 
-typedef int RelAddrType;
-typedef int AbsAddrType;
-typedef int LengthType;
-typedef int RepeatNumType;
+typedef short int RelAddrType;
+typedef short int AbsAddrType;
+typedef short int LengthType;
 typedef short int MemNumType;
+typedef int RepeatNumType;
 
 #define SIZE_OPCODE        1
 #define SIZE_RELADDR       sizeof(RelAddrType)
