--- ext/curses/extconf.rb
+++ ext/curses/extconf.rb
@@ -6,7 +6,7 @@
 make=false
 have_library("mytinfo", "tgetent") if /bow/ =~ RUBY_PLATFORM
-have_library("tinfo", "tgetent") or have_library("termcap", "tgetent")
+have_library("tinfo", "tgetent")
 if have_header(*curses=%w"ncurses.h") and have_library("ncurses", "initscr")
   make=true
 elsif have_header(*curses=%w"ncurses/curses.h") and have_library("ncurses", "initscr")
