--- gocr-0.46/src/Makefile.in	2008-02-05 22:20:44.000000000 +0100
+++ gocr-0.46/src/Makefile.in.new	2008-10-22 20:55:41.000000000 +0200
@@ -48,7 +48,8 @@
 CPPFLAGS=@CPPFLAGS@
 # to see the config.h
 CFLAGS=@CFLAGS@ $(CPPFLAGS) -I../include $(DEFS)
-LDFLAGS=@LDFLAGS@ @LIBS@ -L.
+LDFLAGS=@LDFLAGS@
+LIBADD=@LIBS@
 DESTDIR=@prefix@
 
 .c.o: gocr.h pgm2asc.h ../include/config.h
@@ -67,7 +68,7 @@
 $(PROGRAM): $(LIBOBJS) gocr.o
 	# make it conform to ld --as-needed
 	#$(CC) -o $@ gocr.o ./lib$(PGMASCLIB).a $(LDFLAGS)
-	$(CC) -o $@ gocr.o $(LIBOBJS) $(LDFLAGS)
+	$(CC) $(LDFLAGS) gocr.o $(LIBOBJS) $(LIBADD) -o $@
 	if test -r $(PROGRAM); then cp $@ ../bin; fi
 #$(LIBPGMASCLIB): $(LIBOBJS)
 #	-rm -f $@
