diff -Nru gocr-0.44.orig/src/Makefile.in gocr-0.44/src/Makefile.in
--- gocr-0.44.orig/src/Makefile.in	2006-12-13 17:00:05.000000000 +0100
+++ gocr-0.44/src/Makefile.in	2007-03-29 15:55:06.000000000 +0200
@@ -45,7 +45,8 @@
 CPPFLAGS=@CPPFLAGS@
 # to see the config.h
 CFLAGS=@CFLAGS@ $(CPPFLAGS) -I../include $(DEFS)
-LDFLAGS=@LDFLAGS@ @LIBS@ -L.
+LDFLAGS=@LDFLAGS@
+LIBADD=@LIBS@
 DESTDIR=@prefix@
 
 .c.o: $(INCLUDEFILES)
@@ -57,7 +58,7 @@
 
 $(PROGRAM): $(LIBPGMASCLIB) gocr.o
 	# make it conform to ld --as-needed
-	$(CC) -o $@ gocr.o ./lib$(PGMASCLIB).a $(LDFLAGS)
+	$(CC) $(LDFLAGS) gocr.o ./lib$(PGMASCLIB).a $(LIBADD) -o $@
 	if test -r $(PROGRAM); then cp $@ ../bin; fi
 
 gocr.o: gocr.h Makefile
