diff -Nru gocr-0.43.orig/frontend/Makefile gocr-0.43/frontend/Makefile
--- gocr-0.43.orig/frontend/Makefile	2007-01-20 11:07:44.000000000 +0200
+++ gocr-0.43/frontend/Makefile	2007-01-20 11:07:20.000000000 +0200
@@ -11,5 +11,5 @@
 
 
 gnome/Makefile: gnome/configure
-	(cd gnome;configure)
+	(cd gnome;./configure)
 
diff -Nru gocr-0.43.orig/src/Makefile.in gocr-0.43/src/Makefile.in
--- gocr-0.43.orig/src/Makefile.in	2006-10-31 13:20:09.000000000 +0200
+++ gocr-0.43/src/Makefile.in	2007-01-20 11:10:46.000000000 +0200
@@ -43,7 +43,8 @@
 CPPFLAGS=@CPPFLAGS@
 # to see the config.h
 CFLAGS=@CFLAGS@ $(CPPFLAGS) -I../include $(DEFS)
-LDFLAGS=@LDFLAGS@ @LIBS@ -L.
+LDFLAGS=@LDFLAGS@
+LIBADD=@LIBS@
 DESTDIR=@prefix@
 
 .c.o: $(INCLUDEFILES)
@@ -54,7 +55,7 @@
 all: $(PROGRAM) $(LIBPGMASCLIB)
 
 $(PROGRAM): $(LIBPGMASCLIB) gocr.o
-	$(CC) -o $@ gocr.o $(LDFLAGS) ./lib$(PGMASCLIB).a
+	$(CC) $(LDFLAGS) gocr.o ./lib$(PGMASCLIB).a $(LIBADD) -o $@
 	if test -r $(PROGRAM); then cp $@ ../bin; fi
 
 gocr.o: gocr.h Makefile
