--- eselect-xvmc-0.3.eselect.orig	2012-05-17 11:02:05.074353000 +0200
+++ eselect-xvmc-0.3.eselect	2012-05-17 12:39:44.416374000 +0200
@@ -7,18 +7,27 @@
 SVN_DATE='$Date: 2011/08/29 21:02:02 $'
 VERSION=$(svn_date_to_version "${SVN_DATE}" )
 
+LIBDIR=/usr/lib
+XVMCCONFIG=/etc/X11/XvMCConfig
+
+get_libname() {
+	[[ $(uname) == "Darwin" ]] \
+		&& echo ${1:+.}${1}.dylib \
+		|| echo .so${1:+.}${1}
+}
+
 XVMCLIBS=(
-"libXvMCNVIDIA_dynamic.so.1"
-"libXvMC.so.1"
-"libviaXvMC.so.1"
-"libviaXvMCPro.so.1"
-"libchromeXvMC.so.1"
-"libchromeXvMCPro.so.1"
-"libXvMCVIA.so"
-"libXvMCVIAPro.so"
-"libI810XvMC.so.1"
-"/usr/lib/libIntelXvMC.so"
-"libAMDXvBA.so.1" )
+"libXvMCNVIDIA_dynamic$(get_libname 1)"
+"libXvMC$(get_libname 1)"
+"libviaXvMC$(get_libname 1)"
+"libviaXvMCPro$(get_libname 1)"
+"libchromeXvMC$(get_libname 1)"
+"libchromeXvMCPro$(get_libname 1)"
+"libXvMCVIA$(get_libname)"
+"libXvMCVIAPro$(get_libname)"
+"libI810XvMC$(get_libname 1)"
+"/usr/lib/libIntelXvMC$(get_libname)"
+"libAMDXvBA$(get_libname 1)" )
 XVMCPRETTY=(
 "nvidia"
 "xorg-x11"
@@ -35,7 +44,7 @@
 get_implementation_indices() {
 	local ret n
 	for (( n = 0; n < ${#XVMCLIBS[@]}; ++n )); do
-		[[ -e "${ROOT}/usr/lib/${XVMCLIBS[n]}" ]] && ret+=($n)
+		[[ -e "${ROOT}${LIBDIR}/${XVMCLIBS[n]}" ]] && ret+=($n)
 	done
 
 	echo ${ret[@]}
@@ -43,8 +52,8 @@
 
 get_current_implementation_index() {
 	local n
-	if [[ -f "${ROOT}/etc/X11/XvMCConfig" ]]; then
-		local current=$(< "${ROOT}/etc/X11/XvMCConfig")
+	if [[ -f "${ROOT}${XVMCCONFIG}" ]]; then
+		local current=$(< "${ROOT}${XVMCCONFIG}")
 		for (( n = 0; n < ${#XVMCLIBS[@]}; ++n )); do
 			if [[ "${XVMCLIBS[n]}" = "${current}" ]]; then
 				echo "${n}"
@@ -58,11 +67,11 @@
 
 set_new_implementation() {
 	echo -n "Switching to ${XVMCPRETTY[$1]} XvMC implementation..."
-	touch "${ROOT}/etc/X11/XvMCConfig" 2>&1 > /dev/null
+	touch "${ROOT}${XVMCCONFIG}" 2&>1 > /dev/null
 	if [[ $? -eq 0 ]]; then
-		echo "${XVMCLIBS[$1]}" > "${ROOT}/etc/X11/XvMCConfig"
-		chmod 644 "${ROOT}/etc/X11/XvMCConfig"
-		chown 0:0 "${ROOT}/etc/X11/XvMCConfig"
+		echo "${XVMCLIBS[$1]}" > "${ROOT}${XVMCCONFIG}"
+		chmod 644 "${ROOT}${XVMCCONFIG}"
+		chown 0:0 "${ROOT}${XVMCCONFIG}"
 		echo " done"
 	else
 		echo " failed!"
