diff -ur faac/common/Makefile.am faac-new/common/Makefile.am
--- faac/common/Makefile.am	2004-07-28 10:18:21.000000000 +0200
+++ faac-new/common/Makefile.am	2006-11-22 16:40:22.000000000 +0100
@@ -1,4 +1,4 @@
-if WITH_MP4V2
+if WITH_INTERNAL_MP4V2
 SUBDIRS = mp4v2
 else
 SUBDIRS =
diff -ur faac/configure.in faac-new/configure.in
--- faac/configure.in	2006-08-13 16:17:26.000000000 +0200
+++ faac-new/configure.in	2006-11-22 16:39:18.000000000 +0100
@@ -7,7 +7,7 @@
 
 AC_ARG_WITH( mp4v2,  [  --with-mp4v2            compile libmp4v2],WITHMP4V2=$withval, WITHMP4V2=yes)
 
-AC_DEFUN(MY_DEFINE, [ AC_DEFINE($1, 1, [define if needed]) ])
+AC_DEFUN([MY_DEFINE], [ AC_DEFINE($1, 1, [define if needed]) ])
 
 CFLAGS=${CFLAGS:-"-O2 -Wall"}
 
@@ -27,6 +27,7 @@
 AC_CHECK_LIB(gnugetopt, getopt_long)
 
 AM_CONDITIONAL(WITH_MP4V2, false)
+AM_CONDITIONAL(WITH_INTERNAL_MP4V2, false)
 
 AC_CHECK_DECLS([MP4Create, MP4MetadataDelete],
                AC_CHECK_LIB(mp4v2, MP4MetadataDelete, external_mp4v2=yes,
@@ -35,10 +36,13 @@
 
 if test x$external_mp4v2 = xyes; then
   AC_MSG_NOTICE([*** Building with external mp4v2 ***])
+  MY_DEFINE(HAVE_LIBMP4V2)
+  AM_CONDITIONAL(WITH_MP4V2, true)
 else
   if test x$WITHMP4V2 = xyes; then
      AC_MSG_NOTICE([*** Building with internal mp4v2 ***])
      AM_CONDITIONAL(WITH_MP4V2, true)
+     AM_CONDITIONAL(WITH_INTERNAL_MP4V2, true)
      AC_CONFIG_LINKS(common/mp4v2/mpeg4ip_config.h:config.h)
      MY_DEFINE(HAVE_LIBMP4V2)
   else
diff -ur faac/frontend/Makefile.am faac-new/frontend/Makefile.am
--- faac/frontend/Makefile.am	2004-07-28 10:18:21.000000000 +0200
+++ faac-new/frontend/Makefile.am	2006-11-22 16:24:16.000000000 +0100
@@ -3,9 +3,14 @@
 faac_SOURCES = main.c input.c
 
 if WITH_MP4V2
+if WITH_INTERNAL_MP4V2
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/common/mp4v2
 LDADD = $(top_builddir)/libfaac/libfaac.la $(top_srcdir)/common/mp4v2/libmp4v2.la -lm -lstdc++
 else
 INCLUDES = -I$(top_srcdir)/include 
+LDADD = $(top_builddir)/libfaac/libfaac.la -lm -lmp4v2
+endif
+else
+INCLUDES = -I$(top_srcdir)/include 
 LDADD = $(top_builddir)/libfaac/libfaac.la -lm
 endif
