http://bugs.gentoo.org/show_bug.cgi?id=269650#c7
from Kent Fredric
--- po4a-0.36.4/Build.PL
+++ po4a-0.36.4/Build.PL
@@ -127,8 +127,14 @@
 	    my %options;
 	    $options{utf8} = 1;
 	    my $parser = Pod::Man->new (%options);
-
-	    system("PERL5LIB=lib perl po4a --previous po/pod.cfg") and die;
+     { # Fix for massive slowdown/Memory consumption
+       local %ENV = %ENV;
+       $ENV{PERL5LIB}='lib';
+       open( my $fh, '-|', qw( perl po4a --previous po/pod.cfg ))  or die;
+       while( defined ( my $line = <$fh> ) ){ 
+          print $line;
+       }
+     }
 	    system("mkdir -p blib/man/man7") and die;
 	    system("mkdir -p blib/man/man1") and die;
 	    system("cp doc/po4a.7.pod blib/man/man7") and die;
