--- configure.in
+++ configure.in
@@ -1015,9 +1015,9 @@
 dnl and for readline and history libraries
 
 if test x$enable_sysreadline = xyes; then
-  AC_CHECK_HEADER(readline.h, ECA_S_READLINE_INCLUDES=/usr/include, 
+  AC_CHECK_HEADER(readline.h, ECA_S_READLINE_INCLUDES="", 
     [AC_CHECK_HEADER(readline/readline.h,
-		     ECA_S_READLINE_INCLUDES=/usr/include/readline,
+		     ECA_S_READLINE_INCLUDES="",
       [
         AC_MSG_WARN([*** readline headers not installed ***])
 	enable_sysreadline=no	 
--- ecasound/eca-curses.cpp
+++ ecasound/eca-curses.cpp
@@ -42,8 +42,8 @@
 #endif
 
 #define READLINE_LIBRARY
-#include <readline.h>
-#include <history.h>
+#include <readline/readline.h>
+#include <readline/history.h>
 
 #include <eca-iamode-parser.h>
 #include <eca-version.h>
