Common subdirectories: src.orig/auto and src/auto
diff -u src.orig/gui.h src/gui.h
--- src.orig/gui.h	Mon Jul 29 23:06:39 2002
+++ src/gui.h	Mon Jul 29 23:06:52 2002
@@ -225,6 +225,10 @@
 				   be "Pixel", but that's an unsigned and we
 				   need a signed value */
 
+#define INVALCOLOR (guicolor_T)-11111	/* number for invalid color; on 32 bit
+				   displays there is a tiny chance this is an
+				   actual color */
+
 #ifdef FEAT_GUI_GTK
   typedef GdkFont	*GuiFont;	/* handle for a GUI font */
   typedef GdkFont	*GuiFontset;	/* handle for a GUI fontset */
Common subdirectories: src.orig/proto and src/proto
Common subdirectories: src.orig/testdir and src/testdir
Common subdirectories: src.orig/xxd and src/xxd
