--- vim71/src/configure.in
+++ vim71/src/configure.in
@@ -35,16 +35,10 @@
     dnl old method; fall-back for when -dumpversion doesn't work
     gccversion=`"$CC" --version | sed -e '2,$d' -e 's/darwin.//' -e 's/^[[^0-9]]*\([[0-9]]\.[[0-9.]]*\).*$/\1/g'`
   fi
-  dnl version 4.0.1 was reported to cause trouble on Macintosh by Marcin Dalecki
-  if test "$gccversion" = "3.0.1" -o "$gccversion" = "3.0.2" -o "$gccversion" = "4.0.1"; then
-    echo 'GCC [[34]].0.[[12]] has a bug in the optimizer, disabling "-O#"'
-    CFLAGS=`echo "$CFLAGS" | sed 's/-O[[23456789]]/-O/'`
-  else
     if test "$gccversion" = "3.1" -o "$gccversion" = "3.2" -o "$gccversion" = "3.2.1" && `echo "$CFLAGS" | grep -v fno-strength-reduce >/dev/null`; then
       echo 'GCC 3.1 and 3.2 have a bug in the optimizer, adding "-fno-strength-reduce"'
       CFLAGS="$CFLAGS -fno-strength-reduce"
     fi
-  fi
 fi
 
 dnl If configure thinks we are cross compiling, there is probably something
