--- 70slime-gentoo.el
+++ 70slime-gentoo.el
@@ -12,6 +12,6 @@
 ;; this allows us not to require dev-lisp/hyperspec
 ;; (which is non-free) as a hard dependency
 (setq common-lisp-hyperspec-root
-      (if (file-exists-p "/usr/share/doc/hyperspec/HyperSpec")
-	  "file:///usr/share/doc/hyperspec/HyperSpec/"
+      (if (file-exists-p "@GENTOO_PORTAGE_EPREFIX@/usr/share/doc/hyperspec/HyperSpec")
+	  "file://@GENTOO_PORTAGE_EPREFIX@/usr/share/doc/hyperspec/HyperSpec/"
 	"http://www.lispworks.com/reference/HyperSpec/"))
