--- keychain
+++ keychain
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 # Copyright 1999-2005 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # Originally authored by Daniel Robbins <drobbins@gentoo.org>
@@ -8,7 +8,7 @@
 
 version=2.6.8
 
-PATH="/usr/bin:/bin:/sbin:/usr/sbin:/usr/ucb:${PATH}"
+PATH="@GENTOO_PORTAGE_EPREFIX@/usr/bin:@GENTOO_PORTAGE_EPREFIX@/bin:@GENTOO_PORTAGE_EPREFIX@/sbin:@GENTOO_PORTAGE_EPREFIX@/usr/sbin:/usr/ucb:${PATH}"
 
 maintainer="agriffis@gentoo.org"
 zero=`basename "$0"`
