--- ext/Time/HiRes/HiRes.xs.orig	2009-06-16 15:08:45.000000000 +0200
+++ ext/Time/HiRes/HiRes.xs	2009-06-16 15:12:21.000000000 +0200
@@ -67,9 +67,9 @@
 
 #if defined(TIME_HIRES_CLOCK_GETTIME) && defined(_STRUCT_ITIMERSPEC)
 
-/* HP-UX has CLOCK_XXX values but as enums, not as defines.
+/* HP-UX before 11.31 has CLOCK_XXX values but as enums, not as defines.
  * The only way to detect these would be to test compile for each. */
-# ifdef __hpux
+# if defined(__hpux) && !defined(CLOCK_REALTIME)
 #  define CLOCK_REALTIME CLOCK_REALTIME
 #  define CLOCK_VIRTUAL  CLOCK_VIRTUAL
 #  define CLOCK_PROFILE  CLOCK_PROFILE
