diff -ur ./Configure ./Configure
--- ./Configure	Sun Jan  8 14:51:02 2006
+++ ./Configure	Wed Feb 27 15:20:26 2008
@@ -1225,6 +1209,9 @@
 elif test -n "$DJGPP"; then
 : DOS DJGPP
     _exe=".exe"
+elif test -f "/kern/cookiejar"; then
+: MiNT
+    _exe=""
 elif test -d c:/. -o -n "$is_os2" ; then
 : OS/2 or cygwin
     _exe=".exe"
diff -ur ./MANIFEST ./MANIFEST
--- ./MANIFEST	Tue Jan 31 23:27:52 2006
+++ ./MANIFEST	Wed Feb 27 14:23:18 2008
@@ -1044,7 +1044,6 @@
 hints/lynxos.sh			Hints for named architecture
 hints/machten_2.sh		Hints for named architecture
 hints/machten.sh		Hints for named architecture
-hints/mint.sh			Hints for named architecture
 hints/mips.sh			Hints for named architecture
 hints/mpc.sh			Hints for named architecture
 hints/mpeix.sh			Hints for named architecture
@@ -2123,13 +2123,6 @@
 mg.h				Magic header
 minimod.pl			Writes lib/ExtUtils/Miniperl.pm
 miniperlmain.c			Basic perl w/o dynamic loading or extensions
-mint/errno.h			MiNT port
-mint/Makefile			MiNT port
-mint/pwd.c			MiNT port
-mint/README			MiNT port
-mint/stdio.h			MiNT port
-mint/sys/time.h			MiNT port
-mint/time.h			MiNT port
 mpeix/mpeix.c			MPE/iX port
 mpeix/mpeixish.h		MPE/iX port
 mpeix/mpeix_setjmp.c		MPE/iX port
diff -ur ./malloc.c ./malloc.c
--- ./malloc.c	Wed Oct  5 15:59:40 2005
+++ ./malloc.c	Wed Feb 27 14:24:18 2008
@@ -262,7 +262,7 @@
 #define MIN_BUC_POW2 (sizeof(void*) > 4 ? 3 : 2) /* Allow for 4-byte arena. */
 #define MIN_BUCKET (MIN_BUC_POW2 * BUCKETS_PER_POW2)
 
-#if !(defined(I286) || defined(atarist) || defined(__MINT__))
+#if !(defined(I286))
 	/* take 2k unless the block is bigger than that */
 #  define LOG_OF_MIN_ARENA 11
 #else
@@ -552,8 +552,8 @@
  */
 #define u_short unsigned short
 
-/* 286 and atarist like big chunks, which gives too much overhead. */
-#if (defined(RCHECK) || defined(I286) || defined(atarist) || defined(__MINT__)) && defined(PACK_MALLOC)
+/* 286 like big chunks, which gives too much overhead. */
+#if (defined(RCHECK) || defined(I286)) && defined(PACK_MALLOC)
 #  undef PACK_MALLOC
 #endif 
 
@@ -1720,7 +1720,6 @@
 	/* Second, check alignment. */
 	slack = 0;
 
-#if !defined(atarist) && !defined(__MINT__) /* on the atari we dont have to worry about this */
 #  ifndef I286 	/* The sbrk(0) call on the I286 always returns the next segment */
 	/* WANTED_ALIGNMENT may be more than NEEDED_ALIGNMENT, but this may
 	   improve performance of memory access. */
@@ -1729,7 +1728,6 @@
 	    add += slack;
 	}
 #  endif
-#endif /* !atarist && !MINT */
 		
 	if (add) {
 	    DEBUG_m(PerlIO_printf(Perl_debug_log, 
diff -ur ./miniperlmain.c ./miniperlmain.c
--- ./miniperlmain.c	Sun Sep 18 15:56:38 2005
+++ ./miniperlmain.c	Wed Feb 27 14:33:56 2008
@@ -38,12 +38,6 @@
 static void xs_init (pTHX);
 static PerlInterpreter *my_perl;
 
-#if defined (__MINT__) || defined (atarist)
-/* The Atari operating system doesn't have a dynamic stack.  The
-   stack size is determined from this value.  */
-long _stksize = 64 * 1024;
-#endif
-
 int
 main(int argc, char **argv, char **env)
 {
diff -ur ./os2/perlrexx.c ./os2/perlrexx.c
--- ./os2/perlrexx.c	Wed Dec 19 16:02:52 2001
+++ ./os2/perlrexx.c	Wed Feb 27 14:28:00 2008
@@ -31,12 +31,6 @@
 ULONG PERLDROPALL(PCSZ name, LONG rargc, const RXSTRING *rargv, PCSZ queuename, PRXSTRING retstr);
 ULONG PERLDROPALLEXIT(PCSZ name, LONG rargc, const RXSTRING *rargv, PCSZ queuename, PRXSTRING retstr);
 
-#if defined (__MINT__) || defined (atarist)
-/* The Atari operating system doesn't have a dynamic stack.  The
-   stack size is determined from this value.  */
-long _stksize = 64 * 1024;
-#endif
-
 /* Register any extra external extensions */
 
 /* Do not delete this line--writemain depends on it */
diff -ur ./perl.c ./perl.c
--- ./perl.c	Tue Jan 31 12:34:46 2006
+++ ./perl.c	Wed Feb 27 14:25:00 2008
@@ -3325,10 +3325,6 @@
 		      "\n\nOS/2 port Copyright (c) 1990, 1991, Raymond Chen, Kai Uwe Rommel\n"
 		      "Version 5 port Copyright (c) 1994-2002, Andreas Kaiser, Ilya Zakharevich\n");
 #endif
-#ifdef atarist
-	PerlIO_printf(PerlIO_stdout(),
-		      "atariST series port, ++jrb  bammi@cadence.com\n");
-#endif
 #ifdef __BEOS__
 	PerlIO_printf(PerlIO_stdout(),
 		      "BeOS port Copyright Tom Spindler, 1997-1999\n");
diff -ur ./perl.h ./perl.h
--- ./perl.h	Fri Jan 13 18:17:12 2006
+++ ./perl.h	Wed Feb 27 14:25:24 2008
@@ -329,7 +329,7 @@
  */
 
 /* define this once if either system, instead of cluttering up the src */
-#if defined(MSDOS) || defined(atarist) || defined(WIN32) || defined(NETWARE)
+#if defined(MSDOS) || defined(WIN32) || defined(NETWARE)
 #define DOSISH 1
 #endif
 
diff -ur ./perlio.c ./perlio.c
--- ./perlio.c	Fri Jan  6 22:42:20 2006
+++ ./perlio.c	Wed Feb 27 14:27:20 2008
@@ -120,23 +120,13 @@
      * This used to be contents of do_binmode in doio.c
      */
 #ifdef DOSISH
-#  if defined(atarist) || defined(__MINT__)
-    if (!fflush(fp)) {
-	if (mode & O_BINARY)
-	    ((FILE *) fp)->_flag |= _IOBIN;
-	else
-	    ((FILE *) fp)->_flag &= ~_IOBIN;
-	return 1;
-    }
-    return 0;
-#  else
     dTHX;
 #ifdef NETWARE
     if (PerlLIO_setmode(fp, mode) != -1) {
 #else
     if (PerlLIO_setmode(fileno(fp), mode) != -1) {
 #endif
-#    if defined(WIN32) && defined(__BORLANDC__)
+#  if defined(WIN32) && defined(__BORLANDC__)
 	/*
 	 * The translation mode of the stream is maintained independent of
 	 * the translation mode of the fd in the Borland RTL (heavy
@@ -149,12 +139,11 @@
 	    fp->flags |= _F_BIN;
 	else
 	    fp->flags &= ~_F_BIN;
-#    endif
+#  endif
 	return 1;
     }
     else
 	return 0;
-#  endif
 #else
 #  if defined(USEMYBINMODE)
     dTHX;
diff -ur ./pp_sys.c ./pp_sys.c
--- ./pp_sys.c	Fri Jan 13 18:15:14 2006
+++ ./pp_sys.c	Wed Feb 27 14:10:24 2008
@@ -3281,14 +3281,6 @@
     RETPUSHNO;
 }
 
-#if defined(atarist) /* this will work with atariST. Configure will
-			make guesses for other systems. */
-# define FILE_base(f) ((f)->_base)
-# define FILE_ptr(f) ((f)->_ptr)
-# define FILE_cnt(f) ((f)->_cnt)
-# define FILE_bufsiz(f) ((f)->_cnt + ((f)->_ptr - (f)->_base))
-#endif
-
 PP(pp_fttext)
 {
     dSP;
@@ -5685,30 +5677,6 @@
     case 8:
 	retval = syscall(a[0],a[1],a[2],a[3],a[4],a[5],a[6],a[7]);
 	break;
-#ifdef atarist
-    case 9:
-	retval = syscall(a[0],a[1],a[2],a[3],a[4],a[5],a[6],a[7],a[8]);
-	break;
-    case 10:
-	retval = syscall(a[0],a[1],a[2],a[3],a[4],a[5],a[6],a[7],a[8],a[9]);
-	break;
-    case 11:
-	retval = syscall(a[0],a[1],a[2],a[3],a[4],a[5],a[6],a[7],a[8],a[9],
-	  a[10]);
-	break;
-    case 12:
-	retval = syscall(a[0],a[1],a[2],a[3],a[4],a[5],a[6],a[7],a[8],a[9],
-	  a[10],a[11]);
-	break;
-    case 13:
-	retval = syscall(a[0],a[1],a[2],a[3],a[4],a[5],a[6],a[7],a[8],a[9],
-	  a[10],a[11],a[12]);
-	break;
-    case 14:
-	retval = syscall(a[0],a[1],a[2],a[3],a[4],a[5],a[6],a[7],a[8],a[9],
-	  a[10],a[11],a[12],a[13]);
-	break;
-#endif /* atarist */
     }
     SP = ORIGMARK;
     PUSHi(retval);
diff -ur ./util.c ./util.c
--- ./util.c	Sun Jan  8 20:59:58 2006
+++ ./util.c	Wed Feb 27 14:26:06 2008
@@ -2289,7 +2289,7 @@
     return PerlIO_fdopen(p[This], mode);
 }
 #else
-#if defined(atarist) || defined(EPOC)
+#if defined(EPOC)
 FILE *popen();
 PerlIO *
 Perl_my_popen(pTHX_ char *cmd, char *mode)
@@ -2721,7 +2721,7 @@
     return;
 }
 
-#if defined(atarist) || defined(OS2) || defined(EPOC)
+#if defined(OS2) || defined(EPOC)
 int pclose();
 #ifdef HAS_FORK
 int					/* Cannot prototype with I32
@@ -2821,7 +2821,7 @@
     register char *s;
     I32 len = 0;
     int retval;
-#if defined(DOSISH) && !defined(OS2) && !defined(atarist)
+#if defined(DOSISH) && !defined(OS2)
 #  define SEARCH_EXTS ".bat", ".cmd", NULL
 #  define MAX_EXT_LEN 4
 #endif
@@ -2953,22 +2953,19 @@
 			',',
 			&len);
 #else
-#if defined(atarist) || defined(DOSISH)
+#if defined(DOSISH)
 	    for (len = 0; *s
-#  ifdef atarist
-		    && *s != ','
-#  endif
 		    && *s != ';'; len++, s++) {
 		if (len < sizeof tmpbuf)
 		    tmpbuf[len] = *s;
 	    }
 	    if (len < sizeof tmpbuf)
 		tmpbuf[len] = '\0';
-#else  /* ! (atarist || DOSISH) */
+#else  /* ! (DOSISH) */
 	    s = delimcpy(tmpbuf, tmpbuf + sizeof tmpbuf, s, PL_bufend,
 			':',
 			&len);
-#endif /* ! (atarist || DOSISH) */
+#endif /* ! (DOSISH) */
 #endif /* MACOS_TRADITIONAL */
 	    if (s < PL_bufend)
 		s++;
@@ -2979,7 +2976,7 @@
 	    	tmpbuf[len++] = ':';
 #else
 	    if (len
-#  if defined(atarist) || defined(__MINT__) || defined(DOSISH)
+#  if defined(DOSISH)
 		&& tmpbuf[len - 1] != '/'
 		&& tmpbuf[len - 1] != '\\'
 #  endif
diff -ur /dev/null ./hints/freemint.sh
--- /dev/null	Sun Jan  8 20:59:58 2006
+++ ./hints/freemint.sh	Wed Feb 27 17:09:40 2008
@@ -0,0 +1,44 @@
+# hints/freemint.sh
+#
+# talk to guido@freemint.de if you want to change this file.
+# Please read the README.mint file.
+#
+# misc stuff
+
+archname="m68k-freemint"
+
+cc='gcc'
+ccflags="$ccflags"
+cccdlflags=' '; # avoid -fPIC
+ccdlflags="-Wl,-whole-archive -lgcc -Wl,-no-whole-archive"
+
+# libs
+libpth="$prefix/lib /usr/local/lib /usr/lib"
+glibpth="$libpth"
+xlibpth="$libpth"
+
+libswanted='m dld'
+dl_src='dl_dld.xs'
+dlext='o'
+lddlflags='-r'
+ldflags='-static'
+so='none'
+
+#
+# compiler & linker flags
+#
+optimize='-O2'
+locale_cflags='optimize=" "'
+
+#sbrk() returns -1 (failure) somewhere in lib/unicore/mktables at
+#around 14M, so we need to use system malloc() as our sbrk()
+malloc_cflags='ccflags="-DUSE_PERL_SBRK -DPERL_SBRK_VIA_MALLOC $ccflags"'
+
+# Locales aren't feeling well.
+LC_ALL=C; export LC_ALL;
+LANG=C; export LANG;
+
+# Some good answers to the questions in Configure:
+usemallocwrap='define'
+usemymalloc='y'
+usevfork='true'
diff -ur --new-file mint/Makefile mint/Makefile
--- mint/Makefile	1999-01-17 09:02:07.000000000 +0000
+++ mint/Makefile	1970-01-01 01:00:00.000000000 +0100
@@ -1,15 +0,0 @@
-# IMPORTANT:  This Makefile is not intended to build Perl itself but
-#             only to replace a broken pwd command!
-
-all:	pwd
-
-pwd:	pwd.c
-	$(CC) -O3 -o pwd pwd.c
-
-install: pwd
-	(new_pwd=`which pwd` && cp -f $$new_pwd $$new_pwd.broken \
-		&& cp -f pwd $$new_pwd) 
-
-clean:
-	rm -f pwd.o pwd
-
diff -ur --new-file mint/README mint/README
--- mint/README	1999-01-17 09:02:07.000000000 +0000
+++ mint/README	1970-01-01 01:00:00.000000000 +0100
@@ -1,14 +0,0 @@
-This subdirectory contains some additional files which are necessary
-(or at least useful) when compiling Perl on MiNT.
-
-"Makefile" and "pwd.c" will build and install a fixed version of the
-pwd command if your system pwd is broken.
-
-The header files are wrappers around broken system header files.  Make
-sure that this directory stands at first place in your include path
-when compiling Perl.
-
-The file system.c is an enhanced version of the system() function
-in the MiNTLib.  It is strongly recommended that you insert this
-version into your libc before you compile Perl (see README.MiNT
-in the toplevel directory for details).
diff -ur --new-file mint/errno.h mint/errno.h
--- mint/errno.h	1999-01-17 09:02:07.000000000 +0000
+++ mint/errno.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,32 +0,0 @@
-/* Wrapper around broken system errno.h.  */
-
-#ifndef _PERL_WRAPPER_AROUND_ERRNO_H
-# define _PERL_WRAPPER_AROUND_ERRNO_H 1
-
-/* First include the system file.  */
-#include_next <errno.h> 
-
-/* Now add the missing stuff.
-#ifndef EAGAIN
-# define EAGAIN EWOULDBLOCK
-#endif
-
-/* This one is problematic.  If you open() a directory with the
-   MiNTLib you can't detect from errno if it is really a directory
-   or if the file simply doesn't exist.  You'll get ENOENT 
-   ("file not found") in either case.
-   
-   Defining EISDIR as ENOENT is actually a bad idea but works fine
-   in general.  In praxi, if code checks for errno == EISDIR it
-   will attempt an opendir() call on the file in question and this
-   call will also file if the file really can't be found.  But
-   you may get compile-time errors if the errno checking is embedded
-   in a switch statement ("duplicate case value in switch").
-   
-   Anyway, here the define works alright.  */
-#ifndef EISDIR
-# define EISDIR ENOENT
-#endif
-
-#endif
-
diff -ur --new-file mint/pwd.c mint/pwd.c
--- mint/pwd.c	1999-01-17 09:02:07.000000000 +0000
+++ mint/pwd.c	1970-01-01 01:00:00.000000000 +0100
@@ -1,43 +0,0 @@
-/* pwd.c - replacement for broken pwd command.
- * Copyright 1997 Guido Flohr, <gufl0000@stud.uni-sb.de>.
- * Do with it as you please.
- */
-#include <stdio.h>
-#include <limits.h>
-#include <unistd.h>
-#include <string.h>
-#include <errno.h>
-
-#if defined(__STDC__) || defined(__cplusplus)
-int main (int argc, char* argv[])
-#else
-int main (argc, argv)
-	int argc;
-	char* argv[];
-#endif
-{
-	char path_buf[PATH_MAX + 1];
-	
-	if (argc > 1) {
-		int i;
-		
-		fflush (stdout);
-		fputs (argv[0], stderr);
-		fputs (": ignoring garbage arguments\n", stderr);
-	}
-	
-	if (!getcwd (path_buf, PATH_MAX + 1)) {
-		fflush (stdout);
-		/* Save space, memory and the whales, avoid fprintf.  */
-		fputs (argv[0], stderr);
-		fputs (": can\'t get current working directory: ", stderr);
-		fputs (strerror (errno), stderr);
-		fputc ('\n', stderr);
-		return 1;
-	}
-	if (puts (path_buf) < 0) {
-		return 1;
-	}
-	return 0;
-}
-/* End of pwd.c.  */
diff -ur --new-file mint/stdio.h mint/stdio.h
--- mint/stdio.h	1999-01-17 09:02:07.000000000 +0000
+++ mint/stdio.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,21 +0,0 @@
-/* Wrapper around broken system stdio.h.  */
-
-#ifndef _PERL_WRAPPER_AROUND_STDIO_H
-# define _PERL_WRAPPER_AROUND_STDIO_H 1
-
-/* The MiNTLib has a macro called EOS in stdio.h.  This conflicts
-   with regnode.h.  Who had this glorious idea.  */
-#ifdef EOS
-# define PERL_EOS EOS
-#endif
-
-/* First include the system file.  */
-#include_next <stdio.h> 
-
-#ifdef EOS
-# undef EOS
-# define EOS PERL_EOS
-#endif
-
-#endif
-
diff -ur --new-file mint/sys/time.h mint/sys/time.h
--- mint/sys/time.h	1999-01-17 09:02:07.000000000 +0000
+++ mint/sys/time.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,2 +0,0 @@
-#include <time.h>
-
diff -ur --new-file mint/time.h mint/time.h
--- mint/time.h	1999-01-17 09:02:07.000000000 +0000
+++ mint/time.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,22 +0,0 @@
-/* Wrapper around broken system time.h.  */
-
-#ifndef _PERL_WRAPPER_AROUND_TIME_H
-# define _PERL_WRAPPER_AROUND_TIME_H 1
-
-/* Recent versions of the MiNTLib have a macro HAS_TZNAME in 
-   time.h resp. sys/time.h.  Wow, I wonder why they didn't
-   define HAVE_CONFIG_H ...  */
-#ifdef HAS_TZNAME 
-# define PERL_HAS_TZNAME HAS_TZNAME
-#endif
-
-/* First include the system file.  */
-#include_next <time.h> 
-
-#ifdef HAS_TZNAME
-# undef HAS_TZNAME
-# define HAS_TZNAME PERL_HAS_TZNAME
-#endif
-
-#endif
-
diff -ur --new-file hints/mint.sh hints/mint.sh
--- hints/mint.sh	Fri Sep  8 21:18:10 2000
+++ hints/mint.sh	Fri Feb 29 00:13:26 2008
@@ -1,94 +0,0 @@
-# hints/mint.sh
-#
-# talk to gufl0000@stud.uni-sb.de if you want to change this file.
-# Please read the README.mint file.
-#
-# misc stuff
-
-case `uname -m` in
-	atarist*) archname="m68000-mint"
-		  ;;
-	*)	  archname="m68k-mint"
-		  ;;
-esac		  
-
-here=`pwd | tr -d '\015'`
-
-cc='gcc'
-
-# The weird include path is really to work around some bugs in
-# broken system header files.
-ccflags="$ccflags -D__MINT__ -Uatarist -DDEBUGGING -I$here/../mint"
-
-# libs
-
-libpth="$prefix/lib /usr/lib /usr/local/lib"
-glibpth="$libpth"
-xlibpth="$libpth"
-
-libswanted='gdbm socket port m'
-so='none'
-
-#
-# compiler & linker flags
-#
-optimize='-O2 -fomit-frame-pointer -fno-defer-pop -fstrength-reduce'
-
-# The setlocale function in the MiNTLib is actually a bad joke.  We 
-# lend a workaround from Ultrix.  If neither LC_ALL nor LANG is
-# set in the environment, perl won't complain.  If one is set to
-# anything but "C" you will see a warning.  Note that you can
-# still use the GNU extension "$LANGUAGE" if you want to use
-# the i18n features of some GNU packages.
-util_cflags='ccflags="$ccflags -DLOCALE_ENVIRON_REQUIRED"'
-
-#
-# Some good answers to the questions in Configure:
-# Does Configure really get all these wrong?
-usenm='true'
-d_suidsafe='true'
-clocktype='long'
-usevfork='true'
-d_fsetpos='fpos_t'
-gidtype='gid_t'
-groupstype='gid_t'
-lseektype='long'
-modetype='mode_t'
-sizetype='size_t'
-timetype='time_t'
-uidtype='uid_t'
-
-# Don't remove that leading tab character (Configure Black Magic (TM)).
-    broken_pwd=
-case "`/bin/pwd|tr -d xy|tr '\015\012' 'xy'`" in
-*xy) broken_pwd=yes ;;
-esac
-
-if test X"$broken_pwd" = Xyes
-then
-    echo " "
-    echo "*** Building fixed 'pwd'... (as described in README.mint) ***"
-    echo " "
-    cd mint
-    make pwd
-    cd ..
-    if test -x mint/pwd -a -w /usr/bin
-    then
-	echo " "
-	echo "*** Installing fixed 'pwd'... ***"
-	echo " "
-	cd mint
-	make install
-	cd ..
-	if cmp -s mint/pwd /usr/bin/pwd
-	then
-	    echo "*** Installed fixed 'pwd' successfully. ***"
-	else
-	    echo "*** Failed to install fixed 'pwd'.  Aborting. ***"
-	    exit 1
-	fi
-    else
-	echo "*** Cannot install fixed 'pwd'.  Aborting. ***"
-	exit 1
-    fi
-fi
