--- Configure.orig	2008-02-04 13:41:20.248931500 +0100
+++ Configure	2008-02-04 13:44:02.427067000 +0100
@@ -1233,6 +1233,13 @@
 : no include file wanted by default
 inclwanted=''
 
+: determine libdir name
+: Should be */lib64 on x86-64, sparc64, ppc64
+case `arch` in
+  x86_64) lib='lib64';;
+  *) lib='lib';;
+esac
+  
 groupstype=''
 libnames=''
 : change the next line if compiling for Xenix/286 on Xenix/386
@@ -1241,8 +1248,8 @@
 loclibpth=""
 
 : general looking path for locating libraries
-glibpth="/lib /usr/lib $xlibpth"
-glibpth="$glibpth /usr/ccs/lib /usr/ucblib"
+glibpth="/$lib /usr/$lib $xlibpth"
+glibpth="$glibpth /usr/ccs/$lib /usr/ucblib"
 test -f /usr/shlib/libc.so && glibpth="/usr/shlib $glibpth"
 test -f /shlib/libc.so     && glibpth="/shlib $glibpth"
 
@@ -5927,8 +5934,8 @@
 : Reproduce behavior of 5.005 and earlier, maybe drop that in 5.7.
 case "$installstyle" in
 '')	case "$prefix" in
-		*perl*) dflt='lib';;
-		*) dflt='lib/perl5' ;;
+		*perl*) dflt='lib64';;
+		*) dflt='lib64/perl5' ;;
 	esac
 	;;
 *)	dflt="$installstyle" ;;
@@ -5944,8 +5951,8 @@
 : /opt/perl/lib/perl5... would be redundant.
 : The default "style" setting is made in installstyle.U
 case "$installstyle" in
-*lib/perl5*) set dflt privlib lib/$package/$version ;;
-*)	 set dflt privlib lib/$version ;;
+*lib64/perl5*) set dflt privlib lib64/$package/$version ;;
+*)	 set dflt privlib lib64/$version ;;
 esac
 eval $prefixit
 $cat <<EOM
@@ -6447,8 +6454,8 @@
 prog=`echo $package | $sed 's/-*[0-9.]*$//'`
 case "$sitelib" in
 '') case "$installstyle" in
-	*lib/perl5*) dflt=$siteprefix/lib/$package/site_$prog/$version ;;
-	*)	 dflt=$siteprefix/lib/site_$prog/$version ;;
+	*lib64/perl5*) dflt=$siteprefix/lib64/$package/site_$prog/$version ;;
+	*)	 dflt=$siteprefix/lib64/site_$prog/$version ;;
 	esac
 	;;
 *)	dflt="$sitelib"
@@ -6574,8 +6581,8 @@
 	'')
 		prog=`echo $package | $sed 's/-*[0-9.]*$//'`
 		case "$installstyle" in
-		*lib/perl5*) dflt=$vendorprefix/lib/$package/vendor_$prog/$version ;;
-		*)	     dflt=$vendorprefix/lib/vendor_$prog/$version ;;
+		*lib64/perl5*) dflt=$vendorprefix/lib64/$package/vendor_$prog/$version ;;
+		*)	     dflt=$vendorprefix/lib64/vendor_$prog/$version ;;
 		esac
 		;;
 	*)	dflt="$vendorlib"
@@ -10478,9 +10485,9 @@
 		else
 			for net in net socket
 			do
-				if test -f /usr/lib/lib$net$_a; then
-					( ($nm $nm_opt /usr/lib/lib$net$_a | eval $nm_extract) ||  \
-					$ar t /usr/lib/lib$net$_a) 2>/dev/null >> libc.list
+				if test -f /usr/$lib/lib$net$_a; then
+					( ($nm $nm_opt /usr/$lib/lib$net$_a | eval $nm_extract) ||  \
+					$ar t /usr/$lib/lib$net$_a) 2>/dev/null >> libc.list
 					if $contains socket libc.list >/dev/null 2>&1; then
 						d_socket="$define"
 						socketlib="-l$net"
@@ -21599,6 +21606,7 @@
 ldflags_uselargefiles='$ldflags_uselargefiles'
 ldlibpthname='$ldlibpthname'
 less='$less'
+lib='$lib'
 lib_ext='$lib_ext'
 libc='$libc'
 libperl='$libperl'
