--- lib/File/Path.pm.orig	2009-09-28 17:22:41 +0200
+++ lib/File/Path.pm	2009-09-28 17:23:17 +0200
@@ -108,6 +108,8 @@
 use Exporter ();
 use strict;
 use warnings;
+use Cwd 'getcwd';
+use Config;
 
 our $VERSION = "1.08";
 our @ISA = qw( Exporter );
@@ -273,7 +275,8 @@
 	return;
     }
 
-    opendir my $oldpwd, '.' or do {
+    my $oldpwd;
+    $Config{d_dirfd} ? do { opendir $oldpwd, "." } : do { $oldpwd = getcwd } or do {
 	carp "Can't fetch initial working directory";
 	return;
     };
