--- ext/IO/IO.xs.orig	2004-08-02 21:26:35.486883359 -0700
+++ ext/IO/IO.xs	2004-08-02 21:26:58.285155590 -0700
@@ -75,7 +75,7 @@
 	int mode = RETVAL;
 #ifdef O_NONBLOCK
 	/* POSIX style */
-#if defined(O_NDELAY) && O_NDELAY != O_NONBLOCK
+#if defined(O_NDELAY) && O_NDELAY != O_NONBLOCK && !defined(__linux__)
 	/* Ooops has O_NDELAY too - make sure we don't
 	 * get SysV behaviour by mistake. */
 
