--- configure.ac
+++ configure.ac
@@ -7,8 +7,26 @@
 AC_PATH_PROG(INSTALL, install)
 MKDIR="${MKDIR:-${INSTALL} -d}"
 
+AC_USE_SYSTEM_EXTENSIONS
+
 # strndup() was introduced in POSIX.1-2008.
-AC_CHECK_FUNCS([strndup])
+AC_MSG_CHECKING([for strndup])
+old_CFLAGS="${CFLAGS}"
+CFLAGS="${CFLAGS} -Wall -Werror -Wextra"
+AC_LINK_IFELSE(
+[#include <string.h>
+int main()
+{
+  strndup("", 0);
+  return 0;
+}], [have_strndup="1"], [have_strndup="0"])
+if test "${have_strndup}" = "1"; then
+  AC_MSG_RESULT([yes])
+  AC_DEFINE([HAVE_STRNDUP], [1], [Define to 1 if you have the 'strndup' function.])
+else
+  AC_MSG_RESULT([no])
+fi
+CFLAGS="${old_CFLAGS}"
 
 AC_SUBST([ASFLAGS])
 AC_SUBST([MKDIR])
