--- python.eselect-20080925
+++ python.eselect-20080925
@@ -7,8 +7,10 @@
 SVN_DATE='$Date: 2008-10-25 17:40:49 +0300 (Cts, 25 Eki 2008) $'
 VERSION=$(svn_date_to_version "${SVN_DATE}" )
 
+EPREFIX=@GENTOO_PORTAGE_EPREFIX@
+EROOT=${EPREFIX}${ROOT}
-INTERPRETER_PATH="${ROOT%/}/usr/bin/"
-MAN_PATH="${ROOT%/}/usr/share/man/man1/"
+INTERPRETER_PATH="${EROOT%/}/usr/bin/"
+MAN_PATH="${EROOT%/}/usr/share/man/man1/"
 
 # find a list of python versions
 find_targets() {
