--- Makefile.in.orig	2009-10-02 16:33:57.307364000 +0200
+++ Makefile.in	2009-10-02 16:33:43.477765000 +0200
@@ -4,6 +4,7 @@
 WARNINGS_FLAGS = -Wall -Wextra
 ASFLAGS = @ASFLAGS@
 LDFLAGS = @LDFLAGS@
+LIBS = @LIBS@
 EXEEXT = @EXEEXT@
 MKDIR = @MKDIR@
 INSTALL= @INSTALL@
@@ -17,7 +18,7 @@
 all: python-wrapper$(EXEEXT)
 
 python-wrapper$(EXEEXT): config.h python-wrapper.c
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(WARNINGS_FLAGS) $(ASFLAGS) $(LDFLAGS) -o $@ python-wrapper.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(WARNINGS_FLAGS) $(ASFLAGS) $(LDFLAGS) -o $@ python-wrapper.c $(LIBS)
 
 install: all
 	$(MKDIR) $(DESTDIR)$(bindir) $(DESTDIR)$(datadir)/eselect/modules
