--- python.eselect
+++ python.eselect
@@ -7,6 +7,7 @@
 SVN_DATE='$Date: 2009-06-06 17:03:14 +0000 (Sat, 06 Jun 2009) $'
 VERSION=$(svn_date_to_version "${SVN_DATE}" )
 
+ROOT="${ROOT}@GENTOO_PORTAGE_EPREFIX@"
 ENV_D_PATH="${ROOT%/}/etc/env.d"
 INTERPRETER_PATH="${ROOT%/}/usr/bin/"
 MAN_PATH="${ROOT%/}/usr/share/man/man1/"
@@ -34,7 +34,7 @@
 set_python_config() {
 	local script="${INTERPRETER_PATH}python-config" target="${1}"
 	cat << EOF > "${script}"
-#!/bin/bash
+#!/usr/bin/env bash
 # Gentoo python-config wrapper script
 
 [[ "\${EPYTHON}" =~ (/|^python\$) ]] && EPYTHON="${target/-config-/}"
--- python-wrapper.c
+++ python-wrapper.c
@@ -9,7 +9,7 @@
 #include <unistd.h>
 #include <sys/stat.h>
 
-#define ENVD_CONFIG "/etc/env.d/python/config"
+#define ENVD_CONFIG "@GENTOO_PORTAGE_EPREFIX@/etc/env.d/python/config"
 
 /* 127 is the standard return code for "command not found" */
 #define EXIT_ERROR 127
@@ -39,7 +43,10 @@
 		/* If PATH is unset, then it defaults to ":/bin:/usr/bin", per
 		 * execvp(3).
 		 */
-		PATH = ":/bin:/usr/bin";
+		/* however, if you managed to find the prefix wrapper, you
+		 * probably aren't looking for host stuff, so add prefix paths
+		 * too */
+		PATH = ":@GENTOO_PORTAGE_EPREFIX/bin:@GENTOO_PORTAGE_EPREFIX@/usr/bin:/bin:/usr/bin";
 	}
 	char* path = strdup(PATH);
 	char* state = NULL;
