--- guile-1.4-r3/work/guile-1.4/libguile/net_db.c	2000-06-14 01:26:05.000000000 +0100
+++ guile-1.4.1/work/guile-1.4.1/libguile/net_db.c	2002-05-04 03:16:13.000000000 +0100
@@ -82,9 +80,12 @@
 int close ();
 #endif /* STDC_HEADERS */
 
+#ifndef HAVE_INET_ATON
+/* for our definition in inet_aton.c, not usually needed.  */
 extern int inet_aton ();
+#endif
 
-SCM_DEFINE (scm_inet_aton, "inet-aton", 1, 0, 0, 
+SCM_DEFINE (scm_inet_aton, "inet-aton", 1, 0, 0,
             (SCM address),
 	    "Converts a string containing an Internet host address in the traditional\n"
 	    "dotted decimal notation into an integer.\n\n"
