=== modified file 'Makefile.in'
--- Makefile.in	2008-11-02 10:49:47 +0000
+++ Makefile.in	2008-11-04 18:03:48 +0000
@@ -85,6 +85,8 @@
 GENSRC = scanner.c grammar.c version.c
 LIBOBJS = @LIBOBJS@
 
+HAVE_LINUX_TPACKET_AUXDATA = @HAVE_LINUX_TPACKET_AUXDATA@
+
 SRC =	$(PSRC) $(FSRC) $(CSRC) $(SSRC) $(GENSRC)
 
 # We would like to say "OBJ = $(SRC:.c=.o)" but Ultrix's make cannot
@@ -434,6 +436,14 @@
 	    $(DESTDIR)$(includedir)/pcap/pcap.h
 	$(INSTALL_DATA) $(srcdir)/pcap/bpf.h \
 	    $(DESTDIR)$(includedir)/pcap/bpf.h
+	if test ! -z "@HAVE_LINUX_TPACKET_AUXDATA@" ; then \
+		$(INSTALL_DATA) $(srcdir)/pcap/vlan.h \
+		    $(DESTDIR)$(includedir)/pcap/vlan.h ; \
+	fi
+	if test ! -z "@BT_SRC@" ; then \
+		$(INSTALL_DATA) $(srcdir)/pcap/bluetooth.h \
+		    $(DESTDIR)$(includedir)/pcap/bluetooth.h ; \
+	fi
 	$(INSTALL_DATA) $(srcdir)/pcap/namedb.h \
 	    $(DESTDIR)$(includedir)/pcap/namedb.h
 	$(INSTALL_DATA) $(srcdir)/pcap/sll.h \

=== modified file 'aclocal.m4'
--- aclocal.m4	2008-11-02 09:45:33 +0000
+++ aclocal.m4	2008-11-04 18:05:01 +0000
@@ -930,6 +930,8 @@
 	ac_cv_lbl_linux_tpacket_auxdata_tp_vlan_tci=no))
     AC_MSG_RESULT($ac_cv_lbl_linux_tpacket_auxdata_tp_vlan_tci)
     if test $ac_cv_lbl_linux_tpacket_auxdata_tp_vlan_tci = yes ; then
+	    HAVE_LINUX_TPACKET_AUXDATA=tp_vlan_tci
+	    AC_SUBST(HAVE_LINUX_TPACKET_AUXDATA)
 	    AC_DEFINE(HAVE_LINUX_TPACKET_AUXDATA_TP_VLAN_TCI,1,[if tp_vlan_tci exists])
     fi])
 

