diff -ru libpcap-0.9.5~/configure.in libpcap-0.9.5/configure.in
--- libpcap-0.9.5~/configure.in	2007-05-05 12:43:21.000000000 +0200
+++ libpcap-0.9.5/configure.in	2007-05-05 12:43:41.000000000 +0200
@@ -178,7 +178,10 @@
 dnl XXX This could be done for cross-compiling, but for now it's not.
 dnl
 if test -z "$with_pcap" && test "$cross_compiling" = yes; then
-	AC_MSG_ERROR(pcap type not determined when cross-compiling; use --with-pcap=...)
+	case $host in
+	*-linux*) with_pcap="linux";;
+	*) AC_MSG_ERROR(pcap type not determined when cross-compiling; use --with-pcap=...);;
+	esac
 fi
 AC_ARG_WITH(pcap, [  --with-pcap=TYPE        use packet capture TYPE])
 AC_MSG_CHECKING(packet capture type)
@@ -347,8 +350,10 @@
 linux)
 	AC_MSG_CHECKING(Linux kernel version)
  	if test "$cross_compiling" = yes; then
+		dnl we could check linux/version.h here, but who runs
+		dnl versions of linux older than 2.0.x anymore to bother ?
  		AC_CACHE_VAL(ac_cv_linux_vers,
- 		    ac_cv_linux_vers=unknown)
+ 		    ac_cv_linux_vers=2)
  	else
  		AC_CACHE_VAL(ac_cv_linux_vers,
  		    ac_cv_linux_vers=`uname -r 2>&1 | \
