diff -ru xtrans-1.2.3.orig/Xtranssock.c xtrans-1.2.3/Xtranssock.c
--- xtrans-1.2.3.orig/Xtranssock.c	2009-01-16 09:40:19 +0100
+++ xtrans-1.2.3/Xtranssock.c	2009-01-16 10:59:17 +0100
@@ -637,7 +637,7 @@
 		    (char *) &one, sizeof (int));
     }
 #endif
-#ifdef IPV6_V6ONLY
+#if defined(IPV6_V6ONLY) && defined(IPPROTO_IPV6)
     if (Sockettrans2devtab[i].family == AF_INET6)
     {
 	int one = 1;
@@ -722,7 +722,7 @@
 	return NULL;
     }
 
-#ifdef IPV6_V6ONLY
+#if defined(IPV6_V6ONLY) && defined(IPPROTO_IPV6)
     if (Sockettrans2devtab[i].family == AF_INET6)
     {
 	int one = 1;
