diff -ru xtrans-1.2.1.orig/xtrans.m4 xtrans-1.2.1/xtrans.m4
--- xtrans-1.2.1.orig/xtrans.m4	2008-09-22 11:50:30 +0200
+++ xtrans-1.2.1/xtrans.m4	2008-09-22 11:51:18 +0200
@@ -68,7 +68,7 @@
  AC_REQUIRE([AC_CANONICAL_HOST])
  AC_REQUIRE([AC_TYPE_SIGNAL])
  [case $host_os in
-	mingw*)	unixdef="no"   ;;
+	mingw*|winnt*)	unixdef="no"   ;;
 	*)	unixdef="yes"  ;;
  esac]
  AC_ARG_ENABLE(unix-transport,
