--- file-4.02/magic/Magdir/elf
+++ file-4.02/magic/Magdir/elf
@@ -91,5 +91,5 @@
 >>18	leshort		48		Hitachi H8S,
 >>18	leshort		49		Hitachi H8/500,
->>18	leshort		50		IA-64 (Intel 64 bit architecture)
+>>18	leshort		50		IA-64,
 >>18	leshort		51		Stanford MIPS-X,
 >>18	leshort		52		Motorola Coldfire,
@@ -178,5 +178,5 @@
 >>18	beshort		48		Hitachi H8S,
 >>18	beshort		49		Hitachi H8/500,
->>18	beshort		50		Intel Merced Processor,
+>>18	beshort		50		IA-64,
 >>18	beshort		51		Stanford MIPS-X,
 >>18	beshort		52		Motorola Coldfire,
