--- cdrtools-2.01.01/cdrecord/cdrecord.c	2006-10-28 11:11:33.000000000 +0200
+++ cdrtools-2.01.01/cdrecord/cdrecord.c.rezero	2006-10-28 11:10:43.000000000 +0200
@@ -4042,7 +4042,8 @@
 	scsi_start_stop_unit(scgp, 1, 0, immed);
 	wait_unit_ready(scgp, 120);
 	scgp->silent++;
-	rezero_unit(scgp);	/* Is this needed? Not supported by some drives */
+    /* kernel >= 2.6.12 does not like that as user
+	rezero_unit(scgp);	Is this needed? Not supported by some drives */
 	scgp->silent--;
 	test_unit_ready(scgp);
 	scsi_start_stop_unit(scgp, 1, 0, immed);
