diff -ur cdrtools-2.01.01.orig/libfind/libfind.mk cdrtools-2.01.01/libfind/libfind.mk
--- cdrtools-2.01.01.orig/libfind/libfind.mk	2007-02-04 01:00:05.000000000 +0200
+++ cdrtools-2.01.01/libfind/libfind.mk	2007-08-28 22:33:25.000000000 +0300
@@ -20,7 +20,7 @@
 CPPOPTS +=	-DSCHILY_PRINT
 
 include		Targets
-LIBS=		
+LIBS=		$(LIB_ACL)
 
 ###########################################################################
 include		$(SRCROOT)/$(RULESDIR)/rules.lib
diff -ur cdrtools-2.01.01.orig/libfind/libfind_p.mk cdrtools-2.01.01/libfind/libfind_p.mk
--- cdrtools-2.01.01.orig/libfind/libfind_p.mk	2007-06-30 14:13:34.000000000 +0300
+++ cdrtools-2.01.01/libfind/libfind_p.mk	2007-08-28 22:34:04.000000000 +0300
@@ -22,7 +22,7 @@
 CPPOPTS +=	-DSCHILY_PRINT
 
 include		Targets
-LIBS=		
+LIBS=		$(LIB_ACL)
 
 ###########################################################################
 include		$(SRCROOT)/$(RULESDIR)/rules.lib
diff -ur cdrtools-2.01.01.orig/libfind/shlfind.mk cdrtools-2.01.01/libfind/shlfind.mk
--- cdrtools-2.01.01.orig/libfind/shlfind.mk	2007-02-04 01:00:05.000000000 +0200
+++ cdrtools-2.01.01/libfind/shlfind.mk	2007-08-28 22:45:14.000000000 +0300
@@ -21,7 +21,7 @@
 CPPOPTS +=	-DSCHILY_PRINT
 
 include		Targets
-LIBS=		-lschily -lc
+LIBS=		-lschily -lc $(LIB_ACL)
 
 ###########################################################################
 include		$(SRCROOT)/$(RULESDIR)/rules.shl
