--- vfe/vfe.cpp	2008-09-06 00:14:57.000000000 +0200
+++ vfe/vfe.cpp	2009-01-16 12:48:13.646808534 +0100
@@ -747,9 +747,9 @@
 
   // this code may not be removed or modified unless the compilation is for personal
   // use and the binary is not distributed to anyone else.
-  xtime_get(&t, TIME_UTC);
-  if (t.sec > 1230760800)
-    throw POV_EXCEPTION(kCannotHandleRequestErr, "Please obtain an updated version of this application.");
+  // xtime_get(&t, TIME_UTC);
+  // if (t.sec > 1230760800)
+  //   throw POV_EXCEPTION(kCannotHandleRequestErr, "Please obtain an updated version of this application.");
 
   state = kStarting;
 
