--- vfe/unix/unixconsole.cpp~	2008-10-07 17:06:27.000000000 -0600
+++ vfe/unix/unixconsole.cpp	2008-11-08 14:11:58.000000000 -0700
@@ -281,10 +281,21 @@
 
 		if (user_code != current_code)
 		{
-			fprintf(stderr, "%s: this pre-release version of POV-Ray for Unix %s\n",
-				PACKAGE,
-				current_week < 52 ? "has expired" : "is over twelve months old and is no longer usable"
-			);
+                        if (current_week < 52)
+                        {
+				fprintf(stderr, "%s: this pre-release version of POV-Ray for Unix %s\n",
+					PACKAGE,
+					"has expired"
+				);
+                                fprintf(stderr, "\nTo extend the license period, you can do something like the following\n(adjust syntax for your shell):\n\n    export POVRAY_BETA=`povray --betacode 2>&1`\n\n");
+                        }
+                        else
+                        {
+				fprintf(stderr, "%s: this pre-release version of POV-Ray for Unix %s\n",
+					PACKAGE,
+					"is over twelve months old and is no longer usable"
+				);
+                        }
 			exit(RETURN_ERROR);
 		}
 		current_time = EXPIRE_AT + (current_week+1)*spw - current_time;
