--- povray-3.7.0.beta.35a/configure.ac.old	2010-05-24 14:57:58.000000000 -0600
+++ povray-3.7.0.beta.35a/configure.ac	2010-05-24 15:12:20.000000000 -0600
@@ -361,11 +361,6 @@ else
   fi
   AC_MSG_RESULT([yes])
   AX_CHECK_LIB([png], [$required_libpng_version], [png12 png], [png_get_libpng_ver], [png.h], [png_get_libpng_ver(NULL)], [$with_libpng])
-  ### FIXME: do not allow for 1.4.x
-  AC_MSG_CHECKING([for libpng version < 1.4 (not supported at the moment!)])
-  AX_COMPARE_VERSION([$ax_check_lib_version], [ge], [1.4], [ax_check_lib="bad"], [ax_check_lib="ok"])
-  AC_MSG_RESULT([$ax_check_lib])
-  ###
   if test x"$ax_check_lib" != x"ok"; then
     AC_MSG_ERROR([cannot find a suitable PNG library])
   else
