diff -ru -x '*.Po' -x '*.Plo' pixman-0.15.20.orig/test/blitters-test.c pixman-0.15.20/test/blitters-test.c
--- pixman-0.15.20.orig/test/blitters-test.c	2009-08-19 17:06:40 +0200
+++ pixman-0.15.20/test/blitters-test.c	2009-08-19 17:08:15 +0200
@@ -259,7 +259,12 @@
     stride = (stride + 3) & ~3;
 
     /* do the allocation */
+	#ifndef __INTERIX
     buf = (uint32_t *)memalign (64, stride * height);
+	#else
+	/* no memalign() on interix */
+	buf = (uint32_t *)malloc(stride * height);
+	#endif
 
     /* initialize image with random data */
     for (i = 0; i < stride * height; i++)
