diff -ru -x '*.Po' -x '*.Plo' audacious-1.5.1.orig/src/audacious/af_equalizer.c audacious-1.5.1/src/audacious/af_equalizer.c
--- audacious-1.5.1.orig/src/audacious/af_equalizer.c	2008-08-20 09:19:59 +0200
+++ audacious-1.5.1/src/audacious/af_equalizer.c	2008-08-20 09:22:33 +0200
@@ -18,7 +18,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <inttypes.h>
+#include <stdint.h>
 #include <math.h>
 
 #include "af_compat.h"
diff -ru -x '*.Po' -x '*.Plo' audacious-1.5.1.orig/src/audacious/main.c audacious-1.5.1/src/audacious/main.c
--- audacious-1.5.1.orig/src/audacious/main.c	2008-08-20 09:19:59 +0200
+++ audacious-1.5.1/src/audacious/main.c	2008-08-20 09:23:10 +0200
@@ -35,7 +35,6 @@
 #include <gdk/gdk.h>
 #include <stdlib.h>
 #include <string.h>
-#include <getopt.h>
 #include <ctype.h>
 #include <time.h>
 
diff -ru -x '*.Po' -x '*.Plo' audacious-1.5.1.orig/src/libSAD/SFMT.h audacious-1.5.1/src/libSAD/SFMT.h
--- audacious-1.5.1.orig/src/libSAD/SFMT.h	2008-08-20 09:19:59 +0200
+++ audacious-1.5.1/src/libSAD/SFMT.h	2008-08-20 09:21:56 +0200
@@ -34,13 +34,13 @@
 #include <stdio.h>
 
 #if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)
-  #include <inttypes.h>
+  #include <stdint.h>
 #elif defined(_MSC_VER) || defined(__BORLANDC__)
   typedef unsigned int uint32_t;
   typedef unsigned __int64 uint64_t;
   #define inline __inline
 #else
-  #include <inttypes.h>
+  #include <stdint.h>
   #if defined(__GNUC__)
     #define inline __inline__
   #endif
diff -ru -x '*.Po' -x '*.Plo' audacious-1.5.1.orig/src/libSAD/common.h audacious-1.5.1/src/libSAD/common.h
--- audacious-1.5.1.orig/src/libSAD/common.h	2008-08-20 09:19:59 +0200
+++ audacious-1.5.1/src/libSAD/common.h	2008-08-20 09:20:57 +0200
@@ -24,7 +24,7 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <math.h>
-#include <inttypes.h>
+#include <stdint.h>
 
 #ifndef TRUE
 #define TRUE 1
diff -ru -x '*.Po' -x '*.Plo' audacious-1.5.1.orig/src/libSAD/noicegen.h audacious-1.5.1/src/libSAD/noicegen.h
--- audacious-1.5.1.orig/src/libSAD/noicegen.h	2008-08-20 09:19:59 +0200
+++ audacious-1.5.1/src/libSAD/noicegen.h	2008-08-20 09:21:23 +0200
@@ -1,7 +1,7 @@
 #ifndef NOICEGEN_H
 #define NOICEGEN_H
 
-#include <inttypes.h>
+#include <stdint.h>
 
 int triangular_dither_noise(int nbits);
 double triangular_dither_noise_f(void);
