diff -ru audacious-1.5.0.orig/src/audacious/main.c audacious-1.5.0/src/audacious/main.c
--- audacious-1.5.0.orig/src/audacious/main.c	2008-03-17 16:51:18 +0100
+++ audacious-1.5.0/src/audacious/main.c	2008-03-17 16:54:09 +0100
@@ -35,7 +35,9 @@
 #include <gdk/gdk.h>
 #include <stdlib.h>
 #include <string.h>
+#ifndef __INTERIX
 #include <getopt.h>
+#endif
 #include <ctype.h>
 #include <time.h>
 
diff -ru audacious-1.5.0.orig/src/libSAD/SFMT.h audacious-1.5.0/src/libSAD/SFMT.h
--- audacious-1.5.0.orig/src/libSAD/SFMT.h	2008-03-17 16:51:19 +0100
+++ audacious-1.5.0/src/libSAD/SFMT.h	2008-03-17 16:52:40 +0100
@@ -56,6 +56,10 @@
   #endif
 #endif
 
+#if defined(__INTERIX)
+# include <stdint.h>
+#endif
+
 #if defined(__GNUC__)
 #define ALWAYSINLINE __attribute__((always_inline))
 #endif
diff -ru audacious-1.5.0.orig/src/audacious/Makefile audacious-1.5.0/src/audacious/Makefile
--- audacious-1.5.0.orig/src/audacious/Makefile	2008-03-17 17:04:09 +0100
+++ audacious-1.5.0/src/audacious/Makefile	2008-03-17 17:05:39 +0100
@@ -180,7 +180,8 @@
 	${DBUS_LIBS}		\
 	${MOWGLI_LIBS}		\
 	${LIBMCS_LIBS}		\
-	${REGEX_LIBS}
+	${REGEX_LIBS}		\
+	${GLIB_LIBS}
 
 LDFLAGS += ${PROG_IMPLIB_LDFLAGS}
 
