diff -ru -x '*.Po' -x '*.Plo' audacious-1.5.1.orig/src/libSAD/SFMT.h audacious-1.5.1/src/libSAD/SFMT.h
--- audacious-1.5.1.orig/src/libSAD/SFMT.h	2008-07-17 16:05:52 +0200
+++ audacious-1.5.1/src/libSAD/SFMT.h	2008-07-17 16:08:47 +0200
@@ -44,6 +44,10 @@
   #if defined(__GNUC__)
     #define inline __inline__
   #endif
+
+  #if HAVE_STDINT_H
+    #include <stdint.h>
+  #endif
 #endif
 
 #ifndef PRIu64
