diff -ru -x '*.Po' -x '*.Plo' wavpack-4.41.0.orig/cli/utils.c wavpack-4.41.0/cli/utils.c
--- wavpack-4.41.0.orig/cli/utils.c	2008-03-17 10:40:33 +0100
+++ wavpack-4.41.0/cli/utils.c	2008-03-17 10:45:52 +0100
@@ -27,6 +27,10 @@
 #include <string.h>
 #include <stdio.h>
 
+#ifdef __INTERIX
+# include <utime.h>
+#endif
+
 #include "wavpack.h"
 #include "utils.h"
 
@@ -75,7 +79,11 @@
 int copy_timestamp(const char *src_filename, const char *dst_filename)
 {
     struct stat fileinfo;
+#ifndef __INTERIX
     struct timeval times[2];
+#else
+	struct utimbuf utbuf;
+#endif
 
     if (strcmp(src_filename, "-") == 0 || strcmp(dst_filename, "-") == 0)
         return TRUE;
@@ -83,6 +91,7 @@
     if (stat(src_filename, &fileinfo))
         return FALSE; /* stat failed */
 
+#ifndef __INTERIX
     times[0].tv_sec  = fileinfo.st_atime;
     times[0].tv_usec = 0;
 
@@ -91,6 +100,17 @@
 
     if (utimes(dst_filename, times))
         return FALSE; /* utimes failed */
+#else /* __INTERIX */
+
+	// have no utimes, but only utime...
+	
+	utbuf.actime = fileinfo.st_atime;
+	utbuf.modtime = fileinfo.st_mtime;
+
+	if(utime(dst_filename, &utbuf) != 0)
+		return FALSE;
+
+#endif /* __INTERIX */
 
     return TRUE;
 }
