diff -ru -x '*.Po' -x '*.Plo' wavpack-4.50.0.orig/src/wputils.c wavpack-4.50.0/src/wputils.c
--- wavpack-4.50.0.orig/src/wputils.c	2008-07-17 08:20:30 +0200
+++ wavpack-4.50.0/src/wputils.c	2008-07-17 08:21:54 +0200
@@ -41,6 +41,10 @@
 #define stricmp(x,y) strcasecmp(x,y)
 #endif
 
+#ifdef __INTERIX
+#define strnlen(a, b) strlen(a)
+#endif
+
 #ifdef DEBUG_ALLOC
 #define malloc malloc_db
 #define realloc realloc_db
