diff -ur a/src/plugins/mozjs.c b/src/plugins/mozjs.c
--- a/src/plugins/mozjs.c	Thu Jun  5 22:56:14 2008
+++ b/src/plugins/mozjs.c	Sun May 24 22:47:01 2009
@@ -32,6 +32,14 @@
 #include <jsapi.h>
 #include "pacutils.h"
 
+#ifndef INET_ADDRSTRLEN
+#define INET_ADDRSTRLEN 16
+#endif
+
+#ifndef INET6_ADDRSTRLEN
+#define INET6_ADDRSTRLEN 46
+#endif
+
 static JSBool dnsResolve(JSContext *cx, JSObject *obj, uintN argc, jsval *argv, jsval *rval) {
 	// Get hostname argument
 	char *tmp = px_strdup(JS_GetStringBytes(JS_ValueToString(cx, argv[0])));
diff -ur a/src/plugins/webkit.c b/src/plugins/webkit.c
--- a/src/plugins/webkit.c	Thu Jun  5 22:56:14 2008
+++ b/src/plugins/webkit.c	Sun May 24 22:49:06 2009
@@ -32,6 +32,14 @@
 #include <JavaScriptCore/JavaScript.h>
 #include "pacutils.h"
 
+#ifndef INET_ADDRSTRLEN
+#define INET_ADDRSTRLEN 16
+#endif
+
+#ifndef INET6_ADDRSTRLEN
+#define INET6_ADDRSTRLEN 46
+#endif
+
 static char *jstr2str(JSStringRef str, bool release)
 {
 	char *tmp = px_malloc0(JSStringGetMaximumUTF8CStringSize(str)+1);
