 configure.ac |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index bfea453..16b114a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -17,9 +17,10 @@ PKG_CHECK_MODULES(x11, x11, have_x11=yes, have_x11=no)
 PKG_CHECK_MODULES(xmu, xmu, have_xmu=yes, have_xmu=no)
 PKG_CHECK_MODULES(gconf, gconf-2.0, have_gconf=yes, have_gconf=no)
 PKG_CHECK_MODULES(webkit, webkit-1.0, have_webkit=yes, have_webkit=no)
-PKG_CHECK_MODULES(mozjs, xulrunner-js, have_mozjs=yes, 
-		[PKG_CHECK_MODULES(mozjs, firefox-js, have_mozjs=yes,
-		[PKG_CHECK_MODULES(mozjs, mozilla-js, have_mozjs=yes, have_mozjs=no)])])
+PKG_CHECK_MODULES(mozjs, xulrunner-js, have_mozjs=yes,
+	 	[PKG_CHECK_MODULES(mozjs, libxul-unstable mozilla-js, have_mozjs=yes,
+		[PKG_CHECK_MODULES(mozjs, seamonkey-js, have_mozjs=yes,
+		[PKG_CHECK_MODULES(mozjs, firefox-js, have_mozjs=yes, have_mozjs=no)])])])
 PKG_CHECK_MODULES(NetworkManager, NetworkManager, 
 					have_networkmanager=yes, have_networkmanager=no)
 PKG_CHECK_MODULES(dbus, dbus-1, have_dbus=yes, have_dbus=no)
