--- Makefile.in.orig	2008-03-27 14:19:45 +0100
+++ Makefile.in	2008-03-27 14:23:22 +0100
@@ -51,6 +51,8 @@
 
 PIC_OBJS = $(OBJS:%.o=%.lo)
 
+LIB_OBJS = $(OBJS)
+
 OBJA =
 # to use the asm code: make OBJA=match.o
 
@@ -69,8 +71,8 @@
 	  echo '		*** zlib test FAILED ***'; \
 	fi
 
-libz.a: $(OBJS) $(OBJA)
-	$(AR) rc $@ $(OBJS) $(OBJA)
+libz.a: $(LIB_OBJS) $(OBJA)
+	$(AR) rc $@ $(LIB_OBJS) $(OBJA)
 	-@ ($(RANLIB) $@ || true) >/dev/null 2>&1
 
 match.o: match.S
@@ -82,6 +84,7 @@
 %.lo: %.c
 	$(CC) $(CFLAGS) -DPIC -fPIC -c $< -o $@
 
+$(SHAREDLIBM): $(SHAREDLIBV)
 $(SHAREDLIBV): $(PIC_OBJS)
 	$(LDSHARED) -o $@ $(PIC_OBJS) -lc $(LDFLAGS)
 	rm -f $(SHAREDLIB) $(SHAREDLIBM)
--- configure.orig	2008-03-27 14:19:49 +0100
+++ configure	2008-03-27 14:25:29 +0100
@@ -97,6 +97,11 @@
                  shared_ext='.sl'
                  SHAREDLIB='libz.sl';;
          esac;;
+  AIX)
+	LDSHARED=${LDSHARED-"$cc -shared -Wl,-G"}
+	LIB_OBJS='$(SHAREDLIBM)'
+	buildSHAREDLIB=
+  	;;
   Darwin*)   shared_ext='.dylib'
              SHAREDLIB=libz$shared_ext
              SHAREDLIBV=libz.$VER$shared_ext
@@ -172,6 +177,7 @@
 SHAREDLIB=${SHAREDLIB-"libz$shared_ext"}
 SHAREDLIBV=${SHAREDLIBV-"libz$shared_ext.$VER"}
 SHAREDLIBM=${SHAREDLIBM-"libz$shared_ext.$VER1"}
+LIB_OBJS=${LIB_OBJS-'$(OBJS)'}
 
 if test $shared -eq 1; then
   echo Checking for shared library support...
@@ -179,7 +185,7 @@
   if ($CC -c $SFLAGS $test.c) 2>/dev/null &&
      ($LDSHARED -o $test$shared_ext $test.o) 2>/dev/null; then
     CFLAGS="$SFLAGS"
-    LIBS="$LIBS $SHAREDLIBV"
+    LIBS="$LIBS ${buildSHAREDLIB-${SHAREDLIBV}}"
     echo Building shared library $SHAREDLIBV with $CC.
   elif test -z "$old_cc" -a -z "$old_cflags"; then
     echo No shared library support.
@@ -463,6 +469,7 @@
 /^CPP *=/s#=.*#=$CPP#
 /^LDSHARED *=/s#=.*#=$LDSHARED#
 /^LIBS *=/s#=.*#=$LIBS#
+/^LIB_OBJS *=/s#=.*#=$LIB_OBJS#
 /^SHAREDLIB *=/s#=.*#=$SHAREDLIB#
 /^SHAREDLIBV *=/s#=.*#=$SHAREDLIBV#
 /^SHAREDLIBM *=/s#=.*#=$SHAREDLIBM#
