diff -ur layman-1.3.2.orig//etc/layman.cfg layman-1.3.2/etc/layman.cfg
--- layman-1.3.2.orig//etc/layman.cfg	2010-03-30 13:25:00.000000000 -0500
+++ layman-1.3.2/etc/layman.cfg	2010-03-30 13:25:50.000000000 -0500
@@ -3,7 +3,7 @@
 #-----------------------------------------------------------
 # Defines the directory where overlays should be installed
 
-storage   : /var/lib/layman
+storage   : @GENTOO_PORTAGE_EPREFIX@/var/lib/layman
 
 #-----------------------------------------------------------
 # Remote overlay lists will be stored here
@@ -31,7 +31,7 @@
 #            http://mydomain.org/my-layman-list.xml
 #            file:///var/lib/layman/my-list.xml
 
-overlays  : http://www.gentoo.org/proj/en/overlays/repositories.xml
+overlays  : http://www.gentoo.org/proj/en/gentoo-alt/prefix/layman-prefix.txt
 
 #-----------------------------------------------------------
 # Proxy support
diff -ur layman-1.3.2.orig//layman/config.py layman-1.3.2/layman/config.py
--- layman-1.3.2.orig//layman/config.py	2010-03-30 13:25:00.000000000 -0500
+++ layman-1.3.2/layman/config.py	2010-03-30 13:28:53.000000000 -0500
@@ -65,8 +65,8 @@
         if args == None:
             args = sys.argv
 
-        self.defaults = {'config'    : '/etc/layman/layman.cfg',
-                         'storage'   : '/var/lib/layman',
+        self.defaults = {'config'    : '@GENTOO_PORTAGE_EPREFIX@/etc/layman/layman.cfg',
+                         'storage'   : '@GENTOO_PORTAGE_EPREFIX@/var/lib/layman',
                          'cache'     : '%(storage)s/cache',
                          'local_list': '%(storage)s/overlays.xml',
                          'make_conf' : '%(storage)s/make.conf',
@@ -74,15 +74,15 @@
                          'proxy'     : '',
                          'umask'     : '0022',
                          'overlays'  :
-                         'http://www.gentoo.org/proj/en/overlays/repositories.xml',
-                         'bzr_command': '/usr/bin/bzr',
-                         'cvs_command': '/usr/bin/cvs',
-                         'darcs_command': '/usr/bin/darcs',
-                         'git_command': '/usr/bin/git',
-                         'mercurial_command': '/usr/bin/hg',
-                         'rsync_command': '/usr/bin/rsync',
-                         'svn_command': '/usr/bin/svn',
-                         'tar_command': '/bin/tar', }
+                         'http://www.gentoo.org/proj/en/gentoo-alt/prefix/layman-prefix.txt',
+                         'bzr_command': '@GENTOO_PORTAGE_EPREFIX@/usr/bin/bzr',
+                         'cvs_command': '@GENTOO_PORTAGE_EPREFIX@/usr/bin/cvs',
+                         'darcs_command': '@GENTOO_PORTAGE_EPREFIX@/usr/bin/darcs',
+                         'git_command': '@GENTOO_PORTAGE_EPREFIX@/usr/bin/git',
+                         'mercurial_command': '@GENTOO_PORTAGE_EPREFIX@/usr/bin/hg',
+                         'rsync_command': '@GENTOO_PORTAGE_EPREFIX@/usr/bin/rsync',
+                         'svn_command': '@GENTOO_PORTAGE_EPREFIX@/usr/bin/svn',
+                         'tar_command': '@GENTOO_PORTAGE_EPREFIX@/bin/tar', }
 
 
         self.parser = OptionParser(
