--- etc/layman.cfg
+++ etc/layman.cfg
@@ -3,7 +3,7 @@
 #-----------------------------------------------------------
 # Defines the directory where overlays should be installed
 
-storage   : /usr/local/portage/layman
+storage   : @GENTOO_PORTAGE_EPREFIX@/usr/local/portage/layman
 
 #-----------------------------------------------------------
 # Remote overlay lists will be stored here
@@ -31,7 +31,7 @@
 #            http://mydomain.org/my-layman-list.xml
 #            file:///usr/portage/local/layman/my-list.xml
 
-overlays  : http://www.gentoo.org/proj/en/overlays/repositories.xml
+overlays  : http://www.gentoo.org/proj/en/gentoo-alt/prefix/layman-prefix.txt
 
 #-----------------------------------------------------------
 # Proxy support
--- layman/config.py
+++ layman/config.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@GENTOO_PORTAGE_EPREFIX@/usr/bin/python
 # -*- coding: utf-8 -*-
 #################################################################################
 # LAYMAN CONFIGURATION
@@ -55,8 +55,8 @@
         ['cache', 'config', 'local_list', 'make_conf', 'nocheck', 'overlays', 'proxy', 'quietness', 'storage', 'umask']
         '''
 
-        self.defaults = {'config'    : '/etc/layman/layman.cfg',
-                         'storage'   : '/usr/local/portage/layman',
+        self.defaults = {'config'    : '@GENTOO_PORTAGE_EPREFIX@/etc/layman/layman.cfg',
+                         'storage'   : '@GENTOO_PORTAGE_EPREFIX@/usr/local/portage/layman',
                          'cache'     : '%(storage)s/cache',
                          'local_list': '%(storage)s/overlays.xml',
                          'make_conf' : '%(storage)s/make.conf',
@@ -64,7 +64,8 @@
                          'proxy'     : '',
                          'umask'     : '0022',
                          'overlays'  :
-                         'http://www.gentoo.org/proj/en/overlays/repositories.xml',}
+                         'http://www.gentoo.org/proj/en/gentoo-alt/prefix/'
+                         'layman-prefix.txt',}
 
 
         self.parser = OptionParser(
