diff -Naur hunspell-1.2.2b.orig/src/tools/example.cxx hunspell-1.2.2b/src/tools/example.cxx
--- hunspell-1.2.2b.orig/src/tools/example.cxx	2008-01-16 11:33:31.000000000 +0100
+++ hunspell-1.2.2b/src/tools/example.cxx	2008-02-10 19:20:39.000000000 +0100
@@ -17,8 +17,8 @@
     /* first parse the command line options */
 
     if (argc < 4) {
-        fprintf(stderr,"example (multiple dictionary version.:\n"); 
-        fprintf(stderr,"example affix_file dictionary_file(s) file_of_words_to_check\n");
+        fprintf(stderr,"hunspell-example (multiple dictionary version.:\n"); 
+        fprintf(stderr,"hunspell-example affix_file dictionary_file(s) file_of_words_to_check\n");
         exit(1);
     }
   
diff -Naur hunspell-1.2.2b.orig/src/tools/Makefile.am hunspell-1.2.2b/src/tools/Makefile.am
--- hunspell-1.2.2b.orig/src/tools/Makefile.am	2008-01-09 12:27:42.000000000 +0100
+++ hunspell-1.2.2b/src/tools/Makefile.am	2008-02-10 19:17:47.000000000 +0100
@@ -1,4 +1,4 @@
-bin_PROGRAMS=analyze chmorph example hunspell munch unmunch hzip hunzip
+bin_PROGRAMS=hunspell-analyze hunspell-chmorph hunspell-example hunspell hunspell-munch hunspell-unmunch hzip hunzip
 
 INCLUDES=-I${top_srcdir}/src/hunspell -I${top_srcdir}/src/parsers
 
@@ -6,21 +6,21 @@
 hunzip_SOURCES=hunzip.cxx
 hunzip_LDADD = ../hunspell/libhunspell-1.2.la
 
-munch_SOURCES=munch.c
-unmunch_SOURCES=unmunch.c
+hunspell_munch_SOURCES=munch.c
+hunspell_unmunch_SOURCES=unmunch.c
 include_HEADERS=munch.h unmunch.h
 
-example_SOURCES=example.cxx
-example_LDADD = ../hunspell/libhunspell-1.2.la
+hunspell_example_SOURCES=example.cxx
+hunspell_example_LDADD = ../hunspell/libhunspell-1.2.la
 
 hunspell_SOURCES=hunspell.cxx
 hunspell_LDADD = @LIBINTL@ ../hunspell/libhunspell-1.2.la \
 	../parsers/libparsers.a @CURSESLIB@ @READLINELIB@
 
-analyze_SOURCES=analyze.cxx
-analyze_LDADD = ../hunspell/libhunspell-1.2.la
+hunspell_analyze_SOURCES=analyze.cxx
+hunspell_analyze_LDADD = ../hunspell/libhunspell-1.2.la
 
-chmorph_SOURCES=chmorph.cxx
-chmorph_LDADD = ../hunspell/libhunspell-1.2.la ../parsers/libparsers.a
+hunspell_chmorph_SOURCES=chmorph.cxx
+hunspell_chmorph_LDADD = ../hunspell/libhunspell-1.2.la ../parsers/libparsers.a
 
 EXTRA_DIST=makealias affixcompress
diff -Naur hunspell-1.2.2b.orig/src/tools/munch.c hunspell-1.2.2b/src/tools/munch.c
--- hunspell-1.2.2b.orig/src/tools/munch.c	2007-04-06 10:05:28.000000000 +0200
+++ hunspell-1.2.2b/src/tools/munch.c	2008-02-10 19:24:19.000000000 +0100
@@ -42,14 +42,14 @@
        wf = mystrdup(argv[1]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"munch word_list_file affix_file\n");
+    fprintf(stderr,"hunspell-munch word_list_file affix_file\n");
     exit(1);
   }
   if (argv[2]) {
        af = mystrdup(argv[2]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"munch word_list_file affix_file\n");
+    fprintf(stderr,"hunspell-munch word_list_file affix_file\n");
     exit(1);
   }
 
diff -Naur hunspell-1.2.2b.orig/src/tools/unmunch.c hunspell-1.2.2b/src/tools/unmunch.c
--- hunspell-1.2.2b.orig/src/tools/unmunch.c	2007-04-06 10:05:28.000000000 +0200
+++ hunspell-1.2.2b/src/tools/unmunch.c	2008-02-10 19:24:57.000000000 +0100
@@ -39,14 +39,14 @@
        wf = mystrdup(argv[1]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"unmunch dic_file affix_file\n");
+    fprintf(stderr,"hunspell-unmunch dic_file affix_file\n");
     exit(1);
   }
   if (argv[2]) {
        af = mystrdup(argv[2]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"unmunch dic_file affix_file\n");
+    fprintf(stderr,"hunspell-unmunch dic_file affix_file\n");
     exit(1);
   }
 
diff -Naur hunspell-1.2.2b.orig/tests/test.sh hunspell-1.2.2b/tests/test.sh
--- hunspell-1.2.2b.orig/tests/test.sh	2008-02-10 19:29:21.000000000 +0100
+++ hunspell-1.2.2b/tests/test.sh	2008-02-10 19:34:13.000000000 +0100
@@ -34,7 +34,7 @@
 shopt -s expand_aliases
 
 alias hunspell='../src/tools/hunspell'
-alias analyze='../src/tools/analyze'
+alias analyze='../src/tools/hunspell-analyze'
 
 if [ "$VALGRIND" != "" ]; then
   rm -f $TEMPDIR/test.pid*
