--- Makefile.in~	2007-10-04 07:10:58 +0200
+++ Makefile.in	2007-10-04 07:22:01 +0200
@@ -226,14 +226,14 @@
 # --------------------
 
 $(VM): c/main.o $(OBJS) $(UNIX_OBJS) $(LIBOBJS) $(EXTERNAL_OBJECTS)
-	rm -f /tmp/s48_external_$$$$.c &&				\
-	$(srcdir)/build/build-external-modules /tmp/s48_external_$$$$.c	\
+	rm -f @GENTOO_PORTAGE_EPREFIX@/tmp/s48_external_$$$$.c &&				\
+	$(srcdir)/build/build-external-modules @GENTOO_PORTAGE_EPREFIX@/tmp/s48_external_$$$$.c	\
 		$(EXTERNAL_INITIALIZERS) &&				\
 	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ c/main.o $(OBJS) $(UNIX_OBJS)	\
-		/tmp/s48_external_$$$$.c				\
+		@GENTOO_PORTAGE_EPREFIX@/tmp/s48_external_$$$$.c				\
 		$(LIBOBJS) 					\
 		$(EXTERNAL_OBJECTS) $(EXTERNAL_LD_FLAGS) $(LIBS) &&		\
-	rm -f /tmp/s48_external_$$$$.c
+	rm -f @GENTOO_PORTAGE_EPREFIX@/tmp/s48_external_$$$$.c
 
 script-interpreter: c/script-interpreter.o
 	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ c/script-interpreter.o
@@ -291,7 +291,7 @@
 		scheme/cml/interfaces.scm scheme/cml/packages.scm \
 		scheme/posix/packages.scm scheme/srfi/packages.scm \
 		$(usual-files) build/initial.debug build/build-usual-image
-	$(srcdir)/build/build-usual-image $(srcdir) "/usr/lib/scheme48/" '$(IMAGE)' './$(VM)' \
+	$(srcdir)/build/build-usual-image $(srcdir) "@GENTOO_PORTAGE_EPREFIX@/usr/lib/scheme48/" '$(IMAGE)' './$(VM)' \
 		'$(INITIAL)'
 
 ### Fake targets:  all clean install man dist
@@ -304,16 +304,16 @@
 	$(INSTALL_PROGRAM) $(VM) $(LIB)
 
 inst-image:
-	rm -f '/tmp/$(IMAGE)' &&					\
-	$(srcdir)/build/build-usual-image $(srcdir) '/usr/lib/$(RUNNABLE)'            \
-	         '/tmp/$(IMAGE)' './$(VM)' '$(INITIAL)' &&		\
-	$(INSTALL_DATA) /tmp/$(IMAGE) $(LIB) &&				\
-	rm /tmp/$(IMAGE)
+	rm -f '@GENTOO_PORTAGE_EPREFIX@/tmp/$(IMAGE)' &&					\
+	$(srcdir)/build/build-usual-image $(srcdir) '@GENTOO_PORTAGE_EPREFIX@/usr/lib/$(RUNNABLE)'            \
+	         '@GENTOO_PORTAGE_EPREFIX@/tmp/$(IMAGE)' './$(VM)' '$(INITIAL)' &&		\
+	$(INSTALL_DATA) '@GENTOO_PORTAGE_EPREFIX@/tmp/$(IMAGE)' $(LIB) &&				\
+	rm '@GENTOO_PORTAGE_EPREFIX@/tmp/$(IMAGE)'
 
 inst-man:
 	$(srcdir)/mkinstalldirs $(mandir)
-	sed 's=LBIN=/usr/bin/=g' $(srcdir)/doc/scheme48.man |	   \
-		sed 's=LLIB=/usr/lib/scheme48=g' |				   \
+	sed 's=LBIN=@GENTOO_PORTAGE_EPREFIX@/usr/bin/=g' $(srcdir)/doc/scheme48.man |	   \
+		sed 's=LLIB=@GENTOO_PORTAGE_EPREFIX@/usr/lib/scheme48=g' |				   \
 		sed 's=LS48=$(RUNNABLE)=g' >$(MANPAGE) &&	   \
 	$(INSTALL_DATA) $(MANPAGE) $(mandir) &&			   \
 	rm $(MANPAGE);						   \
@@ -337,7 +337,7 @@
 	script=$(bindir)/$(RUNNABLE) &&					\
 	echo '#!/bin/sh'	>$$script &&				\
 	echo			>>$$script &&				\
-	echo 'lib=/usr/lib/scheme48'	>>$$script &&				\
+	echo 'lib=@GENTOO_PORTAGE_EPREFIX@/usr/lib/scheme48'	>>$$script &&				\
 	echo 'exec $$lib/$(VM) -o $$lib/$(VM) -i $$lib/$(IMAGE) "$$@"'	\
 				>>$$script &&				\
 	chmod +x $$script
@@ -351,7 +351,7 @@
 go:
 	echo '#!/bin/sh'	>$@ &&					\
 	echo			>>$@ &&					\
-	echo "lib=\"/usr/lib/scheme48\""	>>$@ &&					\
+	echo "lib=\"@GENTOO_PORTAGE_EPREFIX@/usr/lib/scheme48\""	>>$@ &&					\
 	echo 'exec "$$lib"/$(VM) -o "$$lib"/$(VM) -i "$$lib"/$(IMAGE) "$$@"'	\
 				>>$@ &&					\
 	chmod +x $@
