diff -ru TeXmacs-1.0.6.14-src.orig/Makefile.in TeXmacs-1.0.6.14-src/Makefile.in
--- TeXmacs-1.0.6.14-src.orig/Makefile.in	2008-04-15 10:19:35 +0200
+++ TeXmacs-1.0.6.14-src/Makefile.in	2008-04-15 10:57:50 +0200
@@ -24,6 +24,7 @@
 MKDIR = mkdir -p
 RM = rm -f
 CP = cp -r -f
+CPF= cp -f
 MV = mv -f
 LN = ln -f
 CHMOD = @CONFIG_CHMOD@
@@ -123,12 +124,12 @@
 	$(MKDIR) $(DESTDIR)$(tmbin)
 	$(MKDIR) $(DESTDIR)$(tmbin)/bin
 	$(MKDIR) $(DESTDIR)$(tmbin)/lib
-	$(CP) $(tmdir)/bin/texmacs.bin $(DESTDIR)$(tmbin)/bin
-	$(CP) $(tmdir)/bin/tm_gs $(DESTDIR)$(tmbin)/bin
-	$(CP) $(tmdir)/plugins/*/bin/* $(DESTDIR)$(tmbin)/bin
-	$(CP) $(tmdir)/plugins/*/lib/*.$(so) $(DESTDIR)$(tmbin)/lib 2>/dev/null || :
+	$(CPF) $(tmdir)/bin/texmacs.bin $(DESTDIR)$(tmbin)/bin
+	$(CPF) $(tmdir)/bin/tm_gs $(DESTDIR)$(tmbin)/bin
+	$(CPF) $(tmdir)/plugins/*/bin/* $(DESTDIR)$(tmbin)/bin
+	$(CPF) $(tmdir)/plugins/*/lib/*.$(so) $(DESTDIR)$(tmbin)/lib 2>/dev/null || :
 	$(STRIP) $(DESTDIR)$(tmbin)/bin/texmacs.bin
-	$(CP) $(tmdir)/lib/*.$(so) $(DESTDIR)$(tmbin)/lib 2>/dev/null || :
+	$(CPF) $(tmdir)/lib/*.$(so) $(DESTDIR)$(tmbin)/lib 2>/dev/null || :
 	$(STRIP) $(DESTDIR)$(tmbin)/lib/*.$(so) 2>/dev/null || :
 	$(CHMOD) 755 $(DESTDIR)$(tmbin)/bin/*
 	$(CHMOD) 755 $(DESTDIR)$(tmbin)/lib/*.$(so) 2>/dev/null || :
diff -ru TeXmacs-1.0.6.14-src.orig/plugins/maple/src/tm_maple_5.cpp TeXmacs-1.0.6.14-src/plugins/maple/src/tm_maple_5.cpp
--- TeXmacs-1.0.6.14-src.orig/plugins/maple/src/tm_maple_5.cpp	2008-04-15 10:19:36 +0200
+++ TeXmacs-1.0.6.14-src/plugins/maple/src/tm_maple_5.cpp	2008-04-15 10:46:44 +0200
@@ -10,6 +10,10 @@
 * 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 ******************************************************************************/
 
+#ifdef __INTERIX
+# define _ALL_SOURCE
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -272,7 +276,9 @@
   killpg (pid, sig);
   cout << DATA_END; // << DATA_END << DATA_END;
   signal (sig, maple_interrupt);
+  #ifndef __INTERIX
   siginterrupt (sig, 1);
+  #endif
 }
 
 /******************************************************************************
@@ -328,7 +334,9 @@
     in= tochild [OUT];
     close (tochild [IN]);
     signal (SIGINT, maple_interrupt);
+	#ifndef __INTERIX
     siginterrupt (SIGINT, 1);
+	#endif
     init_maple ();
     while (true) {
       maple_input ();
diff -ru TeXmacs-1.0.6.14-src.orig/src/System/Link/socket_link.cpp TeXmacs-1.0.6.14-src/src/System/Link/socket_link.cpp
--- TeXmacs-1.0.6.14-src.orig/src/System/Link/socket_link.cpp	2008-04-15 11:04:56 +0200
+++ TeXmacs-1.0.6.14-src/src/System/Link/socket_link.cpp	2008-04-15 09:34:14 +0200
@@ -31,6 +31,10 @@
 #include <sys/misc.h>
 #endif
 
+#ifdef __INTERIX
+# include <arpa/inet.h>
+#endif
+
 hashset<pointer> socket_link_set;
 
 /******************************************************************************
diff -ru TeXmacs-1.0.6.14-src.orig/src/System/Link/socket_server.cpp TeXmacs-1.0.6.14-src/src/System/Link/socket_server.cpp
--- TeXmacs-1.0.6.14-src.orig/src/System/Link/socket_server.cpp	2008-04-15 11:04:56 +0200
+++ TeXmacs-1.0.6.14-src/src/System/Link/socket_server.cpp	2008-04-15 09:34:14 +0200
@@ -22,6 +22,10 @@
 #include <arpa/inet.h>
 #include <sys/wait.h>
 
+#ifdef __INTERIX
+# include <sys/time.h>
+#endif
+
 hashset<pointer> socket_server_set;
 
 /******************************************************************************
