diff -r -U2 TeXmacs-1.0.7-src.orig/misc/autotroll/autotroll.m4 TeXmacs-1.0.7-src/misc/autotroll/autotroll.m4
--- TeXmacs-1.0.7-src.orig/misc/autotroll/autotroll.m4	2008-10-16 13:41:18.000000000 +0700
+++ TeXmacs-1.0.7-src/misc/autotroll/autotroll.m4	2008-11-02 18:03:26.000000000 +0600
@@ -241,5 +241,5 @@
   sed 's/^/| /' "$pro_file" >&AS_MESSAGE_LOG_FD
 
-  if $QMAKE -spec macx-g++ ; then :; else
+  if $QMAKE -spec linux-g++ ; then :; else
     AC_MSG_ERROR([Calling $QMAKE failed.])
   fi
