diff -ur transmission-1.42.orig/configure.ac transmission-1.42/configure.ac
--- transmission-1.42.orig/configure.ac	2008-12-24 19:48:45.000000000 +0200
+++ transmission-1.42/configure.ac	2009-01-01 21:24:02.000000000 +0200
@@ -42,10 +42,6 @@
 
 AC_PROG_CC
 AC_PROG_CXX
-if test "x$GCC" = "xyes" ; then
-    CFLAGS="-g -Wall -Wformat-security -W -Wmissing-prototypes -Wdeclaration-after-statement -O3 -funroll-loops"
-    CXXFLAGS="-g -Wall -Wformat-security -W -O3 -funroll-loops"
-fi
 
 AC_HEADER_STDC
 AC_HEADER_TIME
