--- pdfString.hpp.orig	2003-12-16 20:18:24.000000000 +0100
+++ pdfString.hpp	2006-04-08 12:42:54.000000000 +0200
@@ -13,17 +13,17 @@
    int maxLength;
    char *value;
 public:
-   pdfString::pdfString();
+   pdfString();
    pdfString(const char *src);
-   pdfString::~pdfString();
+   ~pdfString();
 
    /* copie une chaine dans value en utilisant le premier caractère [ ou ( pour déterminer si il s'agit
    d'un chaine hexa ou car*/
-   int pdfString::copy(const char *);
+   int copy(const char *);
    /* copie une chaine dans value en utilisant int comme longeuer à copier*/
-   int pdfString::copy(const char *, int, stringType);
+   int copy(const char *, int, stringType);
    /* copie la chaine org dans la chaine this */
-   int pdfString::copy(const pdfString *org);
+   int copy(const pdfString *org);
    int snprint(char *dest, int lg, bool printable); // printable = true si on veut une version affichable de la chaine cela n'et utile que dans le cas d'une chaine hexa
    bool compare(char *c, int lg);
 
