diff -ru -x '*.Po' -x '*.Plo' elinks-0.11.4.orig/src/network/socket.c elinks-0.11.4/src/network/socket.c
--- elinks-0.11.4.orig/src/network/socket.c	2008-08-01 13:14:54 +0200
+++ elinks-0.11.4/src/network/socket.c	2008-08-01 13:26:55 +0200
@@ -280,7 +280,7 @@
 	size_t addrlen;
 	int sock = -1;
 	int syspf; /* Protocol Family given to system, not EL_PF_... */
-	socklen_t len;
+	int len;
 #ifdef CONFIG_IPV6
 	struct sockaddr_in6 bind_addr6;
 
@@ -456,7 +456,7 @@
 connected(struct socket *socket)
 {
 	int err = 0;
-	socklen_t len = sizeof(err);
+	int len = sizeof(err);
 
 	assertm(socket->connect_info, "Lost connect_info!");
 	if_assert_failed return;
