Index: grep-2.5.1a/src/dfa.c
===================================================================
--- grep-2.5.1a.orig/src/dfa.c
+++ grep-2.5.1a/src/dfa.c
@@ -102,7 +102,7 @@ extern void free();
 /* If we (don't) have I18N.  */
 /* glibc defines _ */
 #ifndef _
-# ifdef HAVE_LIBINTL_H
+# if defined(ENABLE_NLS) && defined(HAVE_LIBINTL_H)
 #  include <libintl.h>
 #  ifndef _
 #   define _(Str) gettext (Str)
Index: grep-2.5.1a/lib/obstack.c
===================================================================
--- grep-2.5.1a.orig/lib/obstack.c
+++ grep-2.5.1a/lib/obstack.c
@@ -451,7 +451,7 @@ _obstack_memory_used (h)
 
 /* Define the error handler.  */
 #ifndef _
-# ifdef HAVE_LIBINTL_H
+# if defined(ENABLE_NLS) && defined(HAVE_LIBINTL_H)
 #  include <libintl.h>
 #  ifndef _
 #   define _(Str) gettext (Str)
Index: grep-2.5.1a/lib/regex.c
===================================================================
--- grep-2.5.1a.orig/lib/regex.c
+++ grep-2.5.1a/lib/regex.c
@@ -112,7 +112,7 @@
 #endif
 
 /* This is for other GNU distributions with internationalized messages.  */
-#if HAVE_LIBINTL_H || defined _LIBC
+#if ENABLE_NLS && (HAVE_LIBINTL_H || defined _LIBC)
 # include <libintl.h>
 # ifdef _LIBC
 #  undef gettext
