--- grep-2.5.1a.orig/lib/getopt.c	2007-06-08 17:23:51 +0200
+++ grep-2.5.1a/lib/getopt.c	2007-06-08 17:24:28 +0200
@@ -77,7 +77,7 @@
 #ifndef _
 /* This is for other GNU distributions with internationalized messages.
    When compiling libc, the _ macro is predefined.  */
-# ifdef HAVE_LIBINTL_H
+# if defined(ENABLE_NLS) && defined(HAVE_LIBINTL_H)
 #  include <libintl.h>
 #  define _(msgid)	gettext (msgid)
 # else

