--- lib/posix/regex.h
+++ lib/posix/regex.h
@@ -533,7 +533,8 @@ extern int re_exec _RE_ARGS ((const char
 /* For now unconditionally define __restrict_arr to expand to nothing.
    Ideally we would have a test for the compiler which allows defining
    it to restrict.  */
+#undef __restrict_arr
 #define __restrict_arr
 
 /* POSIX compatibility.  */
 extern int regcomp _RE_ARGS ((regex_t *__restrict __preg,
