--- configure
+++ configure
@@ -14,6 +14,19 @@
 progversion=0.4
 srctrigger=ed.h
 
+# NLS nuisances.
+for as_var in \
+  LANG LANGUAGE LC_ADDRESS LC_ALL LC_COLLATE LC_CTYPE LC_IDENTIFICATION \
+  LC_MEASUREMENT LC_MESSAGES LC_MONETARY LC_NAME LC_NUMERIC LC_PAPER \
+  LC_TELEPHONE LC_TIME
+do
+  if (set +x; test -z "`(eval $as_var=C; export $as_var) 2>&1`"); then
+    eval $as_var=C; export $as_var
+  else
+    unset $as_var
+  fi
+done
+
 # clear some things potentially inherited from environment.
 srcdir=
 prefix=/usr/local
