diff -Nurpa blassic-0.10.0.orig/cursor.h blassic-0.10.0/cursor.h
--- blassic-0.10.0.orig/cursor.h	2008-12-15 02:44:43.000000000 +0300
+++ blassic-0.10.0/cursor.h	2008-12-15 02:46:03.000000000 +0300
@@ -6,6 +6,7 @@
 
 
 #include <string>
+#include <cstring>
 
 
 namespace cursor {
diff -Nurpa blassic-0.10.0.orig/graphics.cpp blassic-0.10.0/graphics.cpp
--- blassic-0.10.0.orig/graphics.cpp	2008-12-15 02:44:43.000000000 +0300
+++ blassic-0.10.0/graphics.cpp	2008-12-15 02:46:03.000000000 +0300
@@ -24,6 +24,8 @@ using util::to_string;
 #include <sstream>
 #include <iomanip>
 
+#include <cstdlib>
+#include <cstring>
 #include <memory>
 using std::auto_ptr;
 
diff -Nurpa blassic-0.10.0.orig/memory.cpp blassic-0.10.0/memory.cpp
--- blassic-0.10.0.orig/memory.cpp	2008-12-15 02:44:43.000000000 +0300
+++ blassic-0.10.0/memory.cpp	2008-12-15 02:46:03.000000000 +0300
@@ -29,6 +29,7 @@
 #endif
 // HAVE_SYS_MMAN_H
 
+#include <cstring>
 #include <iostream>
 using std::cerr;
 using std::endl;
diff -Nurpa blassic-0.10.0.orig/program.cpp blassic-0.10.0/program.cpp
--- blassic-0.10.0.orig/program.cpp	2008-12-15 02:44:43.000000000 +0300
+++ blassic-0.10.0/program.cpp	2008-12-15 02:46:03.000000000 +0300
@@ -18,6 +18,7 @@ using util::to_string;
 #include <cctype>
 #include <sstream>
 
+#include <cstring>
 using std::string;
 // For debugging.
 using std::cerr;
diff -Nurpa blassic-0.10.0.orig/runnerline_instructions.cpp blassic-0.10.0/runnerline_instructions.cpp
--- blassic-0.10.0.orig/runnerline_instructions.cpp	2008-12-15 02:44:43.000000000 +0300
+++ blassic-0.10.0/runnerline_instructions.cpp	2008-12-15 02:46:03.000000000 +0300
@@ -16,6 +16,7 @@ using util::to_string;
 
 #include "trace.h"
 
+#include <cstring>
 #include <memory>
 using std::auto_ptr;
 
diff -Nurpa blassic-0.10.0.orig/socket.cpp blassic-0.10.0/socket.cpp
--- blassic-0.10.0.orig/socket.cpp	2008-12-15 02:44:43.000000000 +0300
+++ blassic-0.10.0/socket.cpp	2008-12-15 02:45:25.000000000 +0300
@@ -3,6 +3,7 @@
 
 #include "socket.h"
 
+#include <cstring>
 //------------------------------------------------
 // Changed this: now do not use winsock in Cygwin.
 //------------------------------------------------
diff -Nurpa blassic-0.10.0.orig/util.h blassic-0.10.0/util.h
--- blassic-0.10.0.orig/util.h	2008-12-15 02:44:43.000000000 +0300
+++ blassic-0.10.0/util.h	2008-12-15 02:46:03.000000000 +0300
@@ -8,6 +8,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <iostream>
+#include <cstdlib>
 
 #ifdef __BORLANDC__
 #pragma warn -8027
