diff -ur -x '*.pyc' -x '*.log' paramiko-1.7.2/test.py paramiko-1.7.2-fixed/test.py
--- paramiko-1.7.2/test.py	2007-02-13 04:52:19.000000000 +0200
+++ paramiko-1.7.2-fixed/test.py	2008-01-31 16:34:04.000000000 +0200
@@ -140,6 +140,13 @@
         filter = '|'.join(args)
         suite = filter_suite_by_re(suite, filter)
     runner.run(suite)
+    if options.use_sftp:
+        # Final cleanup
+        from test_sftp import get_sftp, get_tc
+        sftp = get_sftp()
+        tc = get_tc()
+        sftp.close()
+        tc.close()
 
 
 if __name__ == '__main__':
diff -ur -x '*.pyc' -x '*.log' paramiko-1.7.2/tests/test_sftp.py paramiko-1.7.2-fixed/tests/test_sftp.py
--- paramiko-1.7.2/tests/test_sftp.py	2007-02-13 05:01:32.000000000 +0200
+++ paramiko-1.7.2-fixed/tests/test_sftp.py	2008-01-31 16:32:52.000000000 +0200
@@ -75,6 +75,9 @@
     global sftp
     return sftp
 
+def get_tc():
+    global tc
+    return tc
 
 class SFTPTest (unittest.TestCase):
 
