--- pbzip2-1.0.5/Makefile.orig	2009-03-20 20:52:48.000000000 +0100
+++ pbzip2-1.0.5/Makefile	2009-03-20 20:54:30.000000000 +0100
@@ -21,11 +21,11 @@
 
 # Standard pbzip2 compile
 pbzip2: pbzip2.cpp
-	$(CC) $(CFLAGS) $^ -o pbzip2 -pthread -lpthread -lbz2
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o pbzip2 -pthread -lpthread -lbz2
 
 # Choose this if you want to compile in a static version of the libbz2 library
 pbzip2-static: pbzip2.cpp libbz2.a
-	$(CC) $(CFLAGS) $^ -o pbzip2 -pthread -lpthread -I. -L. -lbz2
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o pbzip2 -pthread -lpthread -I. -L. -lbz2
 
 # Install the binary pbzip2 program and man page
 install: pbzip2
