diff -urp pbzip2-1.0.2/pbzip2.cpp pbzip2-1.0.2-uclibc/pbzip2.cpp
--- pbzip2-1.0.2/pbzip2.cpp	2007-07-27 12:41:47.000000000 -0400
+++ pbzip2-1.0.2-uclibc/pbzip2.cpp	2008-11-14 22:26:24.000000000 -0500
@@ -128,7 +128,7 @@
 //#define PBZIP_NO_LOADAVG
 
 // detect systems that are known not to support load average code
-#if defined (WIN32) || defined (__CYGWIN32__) || defined (__MINGW32__) || defined (__BORLANDC__) || defined (__hpux) || defined (__osf__)
+#if defined (WIN32) || defined (__CYGWIN32__) || defined (__MINGW32__) || defined (__BORLANDC__) || defined (__hpux) || defined (__osf__) || defined(__UCLIBC__)
 	#define PBZIP_NO_LOADAVG
 #endif
 
@@ -1607,7 +1607,11 @@ void usage(char* progname)
 {
 	banner();
 	fprintf(stderr, "\nInvalid command line!  Aborting...\n\n");
+#if defined(PBZIP_NO_LOADAVG)
+	fprintf(stderr, "Usage: %s [-1 .. -9] [-b#cdfkp#qrtV] <filename> <filename2> <filenameN>\n", progname);
+#else
 	fprintf(stderr, "Usage: %s [-1 .. -9] [-b#cdfklp#qrtV] <filename> <filename2> <filenameN>\n", progname);
+#endif
 	fprintf(stderr, " -b#      : where # is the file block size in 100k (default 9 = 900k)\n");
 	fprintf(stderr, " -c       : output to standard out (stdout)\n");
 	fprintf(stderr, " -d       : decompress file\n");
