diff -ur cl-portable-aserve-1.2.42+cvs.2006.01.21.orig.orig/acl-compat/packages.lisp cl-portable-aserve-1.2.42+cvs.2006.01.21.orig/acl-compat/packages.lisp
--- cl-portable-aserve-1.2.42+cvs.2006.01.21.orig.orig/acl-compat/packages.lisp	2006-02-07 23:49:16.000000000 -0600
+++ cl-portable-aserve-1.2.42+cvs.2006.01.21.orig/acl-compat/packages.lisp	2006-05-11 15:05:31.000000000 -0500
@@ -52,7 +52,8 @@
    #+(or allegro mcl) #:fasl-write
    #+(or allegro cmu scl mcl lispworks) #:string-to-octets
    #+(or allegro cmu scl mcl lispworks) #:write-vector
-   ))
+   )
+  #+cmu (:shadow #:write-vector))
 
 
 ;; general
Only in cl-portable-aserve-1.2.42+cvs.2006.01.21.orig/acl-compat: packages.x86f
