Index: /trunk/gdal/ogr/ogrsf_frmts/mysql/ogr_mysql.h
===================================================================
--- /trunk/gdal/ogr/ogrsf_frmts/mysql/ogr_mysql.h (revision 15583)
+++ /trunk/gdal/ogr/ogrsf_frmts/mysql/ogr_mysql.h (revision 16921)
@@ -35,4 +35,14 @@
 #include <mysql.h>
 
+/* my_global.h from mysql 5.1 declares the min and max macros. */
+/* This conflicts with templates in g++-4.3.2 header files. Grrr */
+#ifdef min
+#undef min
+#endif
+
+#ifdef max
+#undef max
+#endif
+
 #ifdef bool
 #undef bool
