diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN shared-mime-info-0.22.orig/freedesktop.org.xml.in shared-mime-info-0.22/freedesktop.org.xml.in
--- shared-mime-info-0.22.orig/freedesktop.org.xml.in	2007-07-27 12:00:53.000000000 -0400
+++ shared-mime-info-0.22/freedesktop.org.xml.in	2007-11-05 11:49:02.000000000 -0500
@@ -1686,8 +1686,6 @@ command to generate the output files.
       <match type="string" value="&lt;!DOCTYPE NETSCAPE-Bookmark-file-1&gt;" offset="0:64"/>
     </magic>
     <alias type="application/x-netscape-bookmarks"/>
-    <glob pattern="*.html"/>
-    <glob pattern="*.htm"/>
   </mime-type>
   <mime-type type="application/x-ms-dos-executable">
     <_comment>DOS/Windows executable</_comment>
