--- TeX.orig/texk/web2c/mpware/Makefile.in	2007-05-26 20:40:14.000000000 +0200
+++ TeX/texk/web2c/mpware/Makefile.in	2007-05-26 20:40:57.000000000 +0200
@@ -49,7 +49,7 @@
 	$(INSTALL_PROGRAM) newer $(bindir)/newer
 	$(INSTALL_LIBTOOL_PROG) dmp $(bindir)
 	$(INSTALL_LIBTOOL_PROG) mpto $(bindir)
-	$(INSTALL_SCRIPT) makempx $(scriptdir)/makempx
+	$(INSTALL_LIBTOOL_PROG) makempx $(bindir)
 install-data:
 
 kpse_include ../make/tkpathsea.mk
