Index: work/libs/icu-xetex/tools/genccode/genccode.c
===================================================================
--- work.orig/libs/icu-xetex/tools/genccode/genccode.c
+++ work/libs/icu-xetex/tools/genccode/genccode.c
@@ -149,6 +149,7 @@ static const struct AssemblyType {
 } assemblyHeader[] = {
     {"gcc",
         ".globl %s\n"
+        "\t.section .note.GNU-stack,\"\",@progbits\n"
         "\t.section .rodata\n"
         "\t.align 8\n" /* Either align 8 bytes or 2^8 (256) bytes. 8 bytes is needed. */
         "\t.type %s,@object\n"
