--- setup.py
+++ setup.py
@@ -28,7 +28,7 @@
     before_install()
 
 dist = setup(
-    name="distribute",
+    name="setuptools",
     version=VERSION,
     description="Download, build, install, upgrade, and uninstall Python "
         "packages -- easily!",
--- setuptools/package_index.py
+++ setuptools/package_index.py
@@ -143,7 +143,7 @@
                 yield urlparse.urljoin(url, htmldecode(match.group(1)))
 
 user_agent = "Python-urllib/%s distribute/%s" % (
-    urllib2.__version__, require('distribute')[0].version
+    urllib2.__version__, require('setuptools')[0].version
 )
 
 
