diff -NubB ctags-5.5.4.orig/haskell.c ctags-5.5.4/haskell.c
--- ctags-5.5.4.orig/haskell.c	1970-01-01 01:00:00.000000000 +0100
+++ ctags-5.5.4/haskell.c	2005-11-13 16:11:17.000000000 +0100
@@ -0,0 +1,37 @@
+/***************************************************************************
+ * haskell.c
+ * Regex-based parser for Haskell
+ *
+ * Copyright (c) 2005 Fernando J. Pereda <ferdy@gentoo.org>
+ *
+ * Slightly based on haskell.c by Gaal Yahas <gaal@forum2.org>
+ *
+ **************************************************************************/
+/* INCLUDE FILES */
+#include "general.h"    /* always include first */
+#include "parse.h"      /* always include */
+
+/* FUNCTION DEFINITIONS */
+static void installHaskellRegex (const langType language)
+{
+	addTagRegex(language,"^([^,:-]*)[[:blank:]]*::[[:blank:]]*(.*)$","\\1","f,function",NULL);
+	addTagRegex(language,"^main[[:blank:]]*=.*$","main","f,function",NULL);
+	addTagRegex(language,
+		"^type[[:blank:]]*([^=]*)[[:blank:]]*=[[:blank:]]*(.*)$","\\1","t,type",NULL);
+	addTagRegex(language,
+		"^class[[:blank:]]*(\\([^)]*\\)[[:blank:]]*=>)?[[:blank:]]*(.*)[[:blank:]]*where.*$",
+		"\\2","c,class",NULL);
+	addTagRegex(language,"^module[[:blank:]]*([^[:blank:]]*).*$","\\1","m,module",NULL);
+	addTagRegex(language,"^data[[:blank:]]*([^[:blank:]]*).*$","\\1","d,data",NULL);
+}
+
+/* Create parser definition stucture */
+extern parserDefinition* HaskellParser (void)
+{
+    static const char *const extensions [] = { "hs", "lhs", NULL };
+    parserDefinition* const def = parserNew ("Haskell");
+    def->extensions = extensions;
+    def->initialize = installHaskellRegex;
+    def->regex      = TRUE;
+    return def;
+}
diff -NubB ctags-5.5.4.orig/parsers.h ctags-5.5.4/parsers.h
--- ctags-5.5.4.orig/parsers.h	2003-07-18 04:58:22.000000000 +0200
+++ ctags-5.5.4/parsers.h	2005-11-13 16:01:14.000000000 +0100
@@ -27,6 +27,7 @@
     EiffelParser, \
     ErlangParser, \
     FortranParser, \
+    HaskellParser, \
     HtmlParser, \
     JavaParser, \
     JavaScriptParser, \
diff -NubB ctags-5.5.4.orig/source.mak ctags-5.5.4/source.mak
--- ctags-5.5.4.orig/source.mak	2003-04-01 07:02:13.000000000 +0200
+++ ctags-5.5.4/source.mak	2005-11-13 16:01:45.000000000 +0100
@@ -20,6 +20,7 @@
 	erlang.c \
 	fortran.c \
 	get.c \
+	haskell.c \
 	html.c \
 	jscript.c \
 	keyword.c \
@@ -70,6 +71,7 @@
 	erlang.$(OBJEXT) \
 	fortran.$(OBJEXT) \
 	get.$(OBJEXT) \
+	haskell.$(OBJEXT) \
 	html.$(OBJEXT) \
 	jscript.$(OBJEXT) \
 	keyword.$(OBJEXT) \
