diff -NrU5 xmlrpc-c-1.16.06.orig/include/Makefile xmlrpc-c-1.16.06/include/Makefile
--- xmlrpc-c-1.16.06.orig/include/Makefile	2009-01-02 21:01:23.000000000 +0100
+++ xmlrpc-c-1.16.06/include/Makefile	2009-01-02 21:03:41.000000000 +0100
@@ -117,11 +117,11 @@
 default: all
 
 all:
 
 .PHONY: install-compat-hdr
-install-compat-hdr:
+install-compat-hdr: install-headers
 # Install old names of header files for backward compatibility
 	cd $(DESTDIR)$(HEADERINST_DIR); \
 	  rm -f xmlrpc.h xmlrpc_client.h xmlrpc_server.h xmlrpc_cgi.h \
                 xmlrpc_server_abyss.h xmlrpc_abyss.h \
 		xmlrpc_server_w32httpsys.h \
