diff -ru xmlrpc-c-1.06.09.orig/include/xmlrpc-c/abyss.h xmlrpc-c-1.06.09/include/xmlrpc-c/abyss.h
--- xmlrpc-c-1.06.09.orig/include/xmlrpc-c/abyss.h	2008-06-02 10:12:41 +0200
+++ xmlrpc-c-1.06.09/include/xmlrpc-c/abyss.h	2008-06-02 10:13:38 +0200
@@ -24,8 +24,12 @@
 #ifdef WIN32
 #include "xmlrpc_config.h"
 #else
+#ifdef __INTERIX
+#include <stdint.h>
+#else
 #include <inttypes.h>
 #endif
+#endif
 
 /****************************************************************************
   STUFF FOR THE OUTER CONTROL PROGRAM TO USE
diff -ru xmlrpc-c-1.06.09.orig/include/xmlrpc-c/base.hpp xmlrpc-c-1.06.09/include/xmlrpc-c/base.hpp
--- xmlrpc-c-1.06.09.orig/include/xmlrpc-c/base.hpp	2008-06-02 11:49:05 +0200
+++ xmlrpc-c-1.06.09/include/xmlrpc-c/base.hpp	2008-06-02 11:51:07 +0200
@@ -9,6 +9,9 @@
 #include <string>
 
 #include <xmlrpc-c/base.h>
+#ifdef __INTERIX
+#include <sys/time.h>
+#endif
 
 namespace xmlrpc_c {
 
