diff -pur --exclude='*~' xpdf-3.02-poppler.orig/config.h xpdf-3.02-poppler/config.h
--- xpdf-3.02-poppler.orig/config.h	2008-10-14 20:00:41.000000000 +0200
+++ xpdf-3.02-poppler/config.h	2008-10-14 21:08:15.000000000 +0200
@@ -28,6 +28,7 @@
 #define supportedPDFVersionNum 1.7
 
 // copyright notice
+#undef xpdfCopyright
 #define xpdfCopyright "Copyright 1996-2007 Glyph & Cog, LLC"
 
 // Windows resource file stuff
diff -pur --exclude='*~' xpdf-3.02-poppler.orig/XPDFCore.cc xpdf-3.02-poppler/XPDFCore.cc
--- xpdf-3.02-poppler.orig/XPDFCore.cc	2008-10-14 20:00:41.000000000 +0200
+++ xpdf-3.02-poppler/XPDFCore.cc	2008-10-14 21:13:48.000000000 +0200
@@ -22,6 +22,7 @@
 #include "GlobalParams.h"
 #include "PDFDoc.h"
 #include "Link.h"
+#include "FileSpec.h"
 #include "ErrorCodes.h"
 #include "GfxState.h"
 #include "CoreOutputDev.h"
@@ -465,7 +466,7 @@ void XPDFCore::doAction(LinkAction *acti
   GooString *fileName, *fileName2;
   GooString *cmd;
   GooString *actionName;
-  Object movieAnnot, obj1, obj2;
+  Object movieAnnot, obj1, obj2, obj3;
   GooString *msg;
   int i;
 
@@ -631,7 +632,9 @@ void XPDFCore::doAction(LinkAction *acti
     if (movieAnnot.isDict()) {
       if (movieAnnot.dictLookup("Movie", &obj1)->isDict()) {
 	if (obj1.dictLookup("F", &obj2)) {
-	  if ((fileName = LinkAction::getFileSpecName(&obj2))) {
+	  if (getFileSpecNameForPlatform(&obj2, &obj3)) {
+	    fileName = obj3.getString()->copy();
+	    obj3.free();
 	    if (!isAbsolutePath(fileName->getCString())) {
 	      fileName2 = appendToPath(
 			      grabPath(doc->getFileName()->getCString()),
