Index: xpdf-3.02-poppler/XPDFViewer.cc
===================================================================
--- xpdf-3.02-poppler.orig/XPDFViewer.cc
+++ xpdf-3.02-poppler/XPDFViewer.cc
@@ -3447,7 +3447,7 @@ void XPDFViewer::printPrintCbk(Widget wi
     lastPage = doc->getNumPages();
   }
 
-  psOut = new PSOutputDev(psFileName->getCString(), doc->getXRef(),
+  psOut = new PSOutputDev(psFileName->getCString(), doc, doc->getXRef(),
 			  doc->getCatalog(),
 			  NULL,
 			  firstPage, lastPage, psModePS);
