--- autounmask.orig	2009-04-15 20:33:23.000000000 +0200
+++ autounmask	2009-04-16 19:11:35.293590445 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!@GENTOO_PORTAGE_EPREFIX@/usr/bin/perl
 
 # -----------------------------------------------------------------------------
 #
@@ -34,9 +34,9 @@
 # - init vars & constants >
 my $VERSION		= '0.26';
 my $pxs			= PortageXS->new();
-my $keywordsFile	= '/etc/portage/package.keywords';
-my $unmaskFile		= '/etc/portage/package.unmask';
-my $useFile		= '/etc/portage/package.use';
+my $keywordsFile	= '@GENTOO_PORTAGE_EPREFIX@/etc/portage/package.keywords';
+my $unmaskFile		= '@GENTOO_PORTAGE_EPREFIX@/etc/portage/package.unmask';
+my $useFile		= '@GENTOO_PORTAGE_EPREFIX@/etc/portage/package.use';
 my ($p,$x,$r);
 my @l			= ();
 my $packageToUnmask	= '';
@@ -138,19 +138,19 @@
 
 # - Figure out if $keywordsFile is a directory >
 if (-d $keywordsFile) {
-	$keywordsFile='/etc/portage/package.keywords/autounmask-'.$pn;
+	$keywordsFile='@GENTOO_PORTAGE_EPREFIX@/etc/portage/package.keywords/autounmask-'.$pn;
 }
 $pxs->print_ok('Using package.keywords file: '.$keywordsFile."\n");
 
 # - Figure out if $unmaskFile is a directory >
 if (-d $unmaskFile) {
-	$unmaskFile='/etc/portage/package.unmask/autounmask-'.$pn;
+	$unmaskFile='@GENTOO_PORTAGE_EPREFIX@/etc/portage/package.unmask/autounmask-'.$pn;
 }
 $pxs->print_ok('Using package.unmask file: '.$unmaskFile."\n");
 
 # - Figure out if $useFile is a directory >
 if (-d $useFile) {
-	$useFile='/etc/portage/package.use/autounmask-'.$pn;
+	$useFile='@GENTOO_PORTAGE_EPREFIX@/etc/portage/package.use/autounmask-'.$pn;
 }
 $pxs->print_ok('Using package.use file: '.$useFile."\n");
 
