--- gcc_42-5564/gcc/fortran/Make-lang.in	2008-04-01 17:26:19.000000000 -0400
+++ gcc_42-5564.mod/gcc/fortran/Make-lang.in	2008-08-29 17:25:03.000000000 -0400
@@ -73,7 +73,7 @@
     fortran/trans.o fortran/trans-array.o fortran/trans-common.o \
     fortran/trans-const.o fortran/trans-decl.o fortran/trans-expr.o \
     fortran/trans-intrinsic.o fortran/trans-io.o fortran/trans-openmp.o \
-    fortran/trans-stmt.o fortran/trans-types.o
+    fortran/trans-stmt.o fortran/trans-types.o fortran/fake.o
 
 # GFORTRAN uses GMP for its internal arithmetics.
 F95_LIBS = $(GMPLIBS) $(LIBS)
@@ -106,7 +106,7 @@
 f951$(exeext): $(F95_OBJS) \
 		$(BACKEND) $(LIBDEPS)
 	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ \
-		$(F95_OBJS) $(BACKEND) $(F95_LIBS)
+		$(F95_OBJS) $(BACKEND) $(F95_LIBS) stub-objc.o
 
 gt-fortran-trans.h    : s-gtype; @true
 #
--- gcc_42-5564/gcc/fortran/fake.c	1969-12-31 19:00:00.000000000 -0500
+++ gcc_42-5564.mod/gcc/fortran/fake.c	2008-08-29 17:42:58.000000000 -0400
@@ -0,0 +1,27 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+#define MK_FAKE(X) void X (void); void X (void) { fprintf(stderr, "** FATAL: call to an unsupported function!\n"); exit(1); }
+
+int c_language = 0; /* 0 is C which is the default */
+int disable_typechecking_for_spec_flag = 0;
+
+MK_FAKE(build_function_call)
+MK_FAKE(build_modify_expr)
+MK_FAKE(build_stmt)
+MK_FAKE(decl_attributes)
+MK_FAKE(decl_constant_value)
+MK_FAKE(flag_iasm_blocks)
+MK_FAKE(iasm_addr)
+MK_FAKE(iasm_build_bracket)
+MK_FAKE(iasm_force_constraint)
+MK_FAKE(iasm_get_register_var)
+MK_FAKE(iasm_in_operands)
+MK_FAKE(iasm_is_pseudo)
+MK_FAKE(iasm_print_operand)
+MK_FAKE(iasm_state)
+MK_FAKE(lookup_name)
+MK_FAKE(parse_in)
+MK_FAKE(pointer_int_sum)
+MK_FAKE(store_init_value)
+
