--- gcc/cp/parser.c
+++ gcc/cp/parser.c
@@ -11845,8 +11845,7 @@
       || cp_lexer_next_token_is (parser->lexer, CPP_ATSIGN)
       || cp_lexer_next_token_is (parser->lexer, CPP_NAME)
       || cp_lexer_next_token_is_keyword (parser->lexer, RID_ASM)
-      || cp_lexer_next_token_is (parser->lexer, CPP_SEMICOLON)
-      || cp_lexer_iasm_bol (parser->lexer))
+      || cp_lexer_next_token_is (parser->lexer, CPP_SEMICOLON))
     {
       if (flag_iasm_blocks)
 	cp_parser_iasm_top_statement (parser);
--- gcc/c-parser.c
+++ gcc/c-parser.c
@@ -4505,8 +4505,7 @@
       || c_parser_next_token_is (parser, CPP_ATSIGN)
       || c_parser_next_token_is (parser, CPP_NAME)
       || c_parser_next_token_is_keyword (parser, RID_ASM)
-      || c_parser_next_token_is (parser, CPP_SEMICOLON)
-      || c_parser_iasm_bol (parser))
+      || c_parser_next_token_is (parser, CPP_SEMICOLON))
     {
       if (flag_iasm_blocks)
 	c_parser_iasm_top_statement (parser);
