diff -ur libxklavier-4.0.orig/tests/Makefile.am libxklavier-4.0/tests/Makefile.am
--- libxklavier-4.0.orig/tests/Makefile.am	2009-04-07 01:50:08.000000000 +0300
+++ libxklavier-4.0/tests/Makefile.am	2009-07-20 23:44:22.000000000 +0300
@@ -1,10 +1,10 @@
-noinst_PROGRAMS=test_config test_monitor
+check_PROGRAMS=test_config test_monitor
 
 test_config_SOURCES=test_config.c
 
 test_monitor_SOURCES=test_monitor.c
 
-AM_CFLAGS=-Wall -Werror -I$(top_srcdir) $(X_CFLAGS) $(GLIB_CFLAGS)
+AM_CFLAGS=-Wall -I$(top_srcdir) $(X_CFLAGS) $(GLIB_CFLAGS)
 
 LDADD=$(top_builddir)/libxklavier/libxklavier.la $(X_LIBS) $(GLIB_LIBS)
 
