diff -ru vte-0.16.13.orig/gnome-pty-helper/gnome-pty-helper.c vte-0.16.13/gnome-pty-helper/gnome-pty-helper.c
--- vte-0.16.13.orig/gnome-pty-helper/gnome-pty-helper.c	2008-04-01 12:04:57 +0200
+++ vte-0.16.13/gnome-pty-helper/gnome-pty-helper.c	2008-04-01 12:05:15 +0200
@@ -51,7 +51,11 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
-#include <utmp.h>
+
+#ifndef __INTERIX
+# include <utmp.h>
+#endif
+
 #include <grp.h>
 #include <glib/galloca.h>
 #include "gnome-pty.h"
diff -ru vte-0.16.13.orig/src/vteglyph.c vte-0.16.13/src/vteglyph.c
--- vte-0.16.13.orig/src/vteglyph.c	2008-04-01 12:04:57 +0200
+++ vte-0.16.13/src/vteglyph.c	2008-04-01 12:05:48 +0200
@@ -28,6 +28,10 @@
 #include "vtefc.h"
 #include "vteglyph.h"
 
+#ifndef howmany
+#define howmany(x, y) (((x) + ((y) - 1)) / (y))
+#endif
+
 #include <glib/gi18n-lib.h>
 
 #define FONT_INDEX_FUDGE 10
diff -ru vte-0.16.13.orig/src/vtepango.c vte-0.16.13/src/vtepango.c
--- vte-0.16.13.orig/src/vtepango.c	2008-04-01 12:04:57 +0200
+++ vte-0.16.13/src/vtepango.c	2008-04-01 12:05:15 +0200
@@ -30,6 +30,10 @@
 #include "vtedraw.h"
 #include "vtepango.h"
 
+#ifndef howmany
+#define howmany(x, y) (((x) + ((y) - 1)) / (y))
+#endif
+
 struct _vte_pango_data
 {
 	GdkColor color;
diff -ru vte-0.16.13.orig/src/vtepangox.c vte-0.16.13/src/vtepangox.c
--- vte-0.16.13.orig/src/vtepangox.c	2008-04-01 12:04:57 +0200
+++ vte-0.16.13/src/vtepangox.c	2008-04-01 12:05:15 +0200
@@ -34,6 +34,10 @@
 #include "vtedraw.h"
 #include "vtepangox.h"
 
+#ifndef howmany
+#define howmany(x, y) (((x) + ((y) - 1)) / (y))
+#endif
+
 struct _vte_pango_x_data
 {
 	GdkColor color;
diff -ru vte-0.16.13.orig/src/vtexft.c vte-0.16.13/src/vtexft.c
--- vte-0.16.13.orig/src/vtexft.c	2008-04-01 12:04:57 +0200
+++ vte-0.16.13/src/vtexft.c	2008-04-01 12:05:15 +0200
@@ -36,6 +36,10 @@
 
 #include <glib/gi18n-lib.h>
 
+#ifndef howmany
+#define howmany(x, y) (((x) + ((y) - 1)) / (y))
+#endif
+
 #define FONT_INDEX_FUDGE 1
 #define CHAR_WIDTH_FUDGE 1
 
