diff -ru -x '*.Po' -x '*.Plo' vte-0.20.1.orig/gnome-pty-helper/gnome-pty-helper.c vte-0.20.1/gnome-pty-helper/gnome-pty-helper.c
--- vte-0.20.1.orig/gnome-pty-helper/gnome-pty-helper.c	2009-05-15 09:40:35 +0200
+++ vte-0.20.1/gnome-pty-helper/gnome-pty-helper.c	2009-05-15 08:27:49 +0200
@@ -51,7 +51,11 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
-#include <utmp.h>
+
+#ifndef __INTERIX
+# include <utmp.h>
+#endif
+
 #include <grp.h>
 #include "gnome-pty.h"
 #include "gnome-login-support.h"
diff -ru -x '*.Po' -x '*.Plo' vte-0.20.1.orig/src/vtepangocairo.c vte-0.20.1/src/vtepangocairo.c
--- vte-0.20.1.orig/src/vtepangocairo.c	2009-05-15 09:18:52 +0200
+++ vte-0.20.1/src/vtepangocairo.c	2009-05-15 09:35:53 +0200
@@ -31,6 +31,9 @@
 
 #include <pango/pangocairo.h>
 
+#ifndef howmany
+#define howmany(x, y) (((x) + ((y) - 1)) / (y))
+#endif
 
 /* Overview:
  *
