diff -ru libffi-3.0.8.orig/configure.ac libffi-3.0.8/configure.ac
--- libffi-3.0.8.orig/configure.ac	2009-09-30 11:47:40 +0200
+++ libffi-3.0.8/configure.ac	2009-09-30 11:50:59 +0200
@@ -74,7 +74,7 @@
   i?86-*-freebsd* | i?86-*-openbsd*)
 	TARGET=X86_FREEBSD; TARGETDIR=x86
 	;;
-  i?86-win32* | i?86-*-cygwin* | i?86-*-mingw*)
+  i?86-win32* | i?86-*-cygwin* | i?86-*-mingw* | i?86-*-interix*)
 	TARGET=X86_WIN32; TARGETDIR=x86
 	;;
   i?86-*-darwin*)
