diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN pygtk-2.12.0.orig/gtk/Makefile.am pygtk-2.12.0/gtk/Makefile.am
--- pygtk-2.12.0.orig/gtk/Makefile.am	2007-09-15 11:00:42.000000000 -0400
+++ pygtk-2.12.0/gtk/Makefile.am	2007-12-22 11:17:35.000000000 -0500
@@ -138,7 +138,7 @@ nodist__gtk_la_SOURCES = gtk.c gdk.c
 if BUILD_GTK
 pygtkexec_LTLIBRARIES += _gtk.la
 defs_DATA += $(GDK_DEFS) $(GDK_TYPES_DEFS) $(GTK_DEFS) $(GTK_TYPES_DEFS) \
-	gdk.defs gtk.defs gdk-types.defs gtk-types.defs
+	gdk.defs gtk.defs gdk-types.defs gtk-types.defs gtk-extrafuncs.defs
 endif
 
 # libglade module
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN pygtk-2.12.0.orig/gtk/Makefile.in pygtk-2.12.0/gtk/Makefile.in
--- pygtk-2.12.0.orig/gtk/Makefile.in	2007-09-16 07:44:03.000000000 -0400
+++ pygtk-2.12.0/gtk/Makefile.in	2007-12-22 11:26:52.000000000 -0500
@@ -47,7 +47,7 @@ host_triplet = @host@
 @HAVE_GTK_2_12_TRUE@am__append_7 = gtk-2.12-types.defs
 @BUILD_GTK_TRUE@am__append_8 = _gtk.la
 @BUILD_GTK_TRUE@am__append_9 = $(GDK_DEFS) $(GDK_TYPES_DEFS) $(GTK_DEFS) $(GTK_TYPES_DEFS) \
-@BUILD_GTK_TRUE@	gdk.defs gtk.defs gdk-types.defs gtk-types.defs
+@BUILD_GTK_TRUE@	gdk.defs gtk.defs gdk-types.defs gtk-types.defs gtk-extrafuncs.defs
 
 @BUILD_LIBGLADE_TRUE@am__append_10 = glade.la
 @BUILD_LIBGLADE_TRUE@am__append_11 = $(LIBGLADE_DEFS)
