* http://bugs.gentoo.org/show_bug.cgi?id=111598

--- TinySVM-0.09/src/getopt.h.orig	2006-02-19 17:22:37.000000000 +0100
+++ TinySVM-0.09/src/getopt.h	2006-02-19 17:23:11.000000000 +0100
@@ -122,7 +122,7 @@
 #define optional_argument	2
 
 #if defined (__STDC__) && __STDC__
-#ifdef __GNU_LIBRARY__
+#if defined __GNU_LIBRARY__ || defined __APPLE__
 /* Many other libraries have conflicting prototypes for getopt, with
    differences in the consts, in stdlib.h.  To avoid compilation
    errors, only prototype getopt for the GNU C library.  */
