diff -Naur TinySVM-0.09.orig/perl/Makefile.PL.in TinySVM-0.09/perl/Makefile.PL.in
--- TinySVM-0.09.orig/perl/Makefile.PL.in	2001-08-29 23:56:31.000000000 +0900
+++ TinySVM-0.09/perl/Makefile.PL.in	2008-11-03 09:47:48.000000000 +0900
@@ -3,8 +3,9 @@
 # the contents of the Makefile that is written.
 WriteMakefile(
     'CC'		=> '@CXX@',
+    'LD'		=> '@CXX@',
     'NAME'		=> 'TinySVM',
-    'INC'		=> '-I@prefix@/include',
-    'LIBS'        	=> '-L@prefix@/lib -ltinysvm',
-    'OBJECT'            => 'TinySVM_wrap.o'
+    'INC'		=> '-I../src -I@includedir@',
+    'LIBS'		=> '-L../src/.libs -L@libdir@ -ltinysvm',
+    'OBJECT'		=> 'TinySVM_wrap.o'
 );
