Index: exo/exo-utils.h
===================================================================
--- exo/exo-utils.h	(revision 28047)
+++ exo/exo-utils.h	(revision 28048)
@@ -35,12 +35,13 @@
 gpointer                exo_noop_null   (void) G_GNUC_PURE;
 gboolean                exo_noop_true   (void) G_GNUC_PURE;
 gboolean                exo_noop_false  (void) G_GNUC_PURE;
-G_INLINE_FUNC void      exo_atomic_inc  (gint *value);
-G_INLINE_FUNC gboolean  exo_atomic_dec  (gint *value);
 
 /* inline function implementations */
 #if (defined(G_CAN_INLINE) && defined(__GNUC__) && defined(__i386__) && defined(__OPTIMIZE__)) || defined(__EXO_UTILS_C__)
 
+G_INLINE_FUNC void      exo_atomic_inc  (gint *value);
+G_INLINE_FUNC gboolean  exo_atomic_dec  (gint *value);
+
 G_INLINE_FUNC void
 exo_atomic_inc (gint *value)
 {
