Index: libsamplerate-0.1.2/examples/Makefile.am
===================================================================
--- libsamplerate-0.1.2.orig/examples/Makefile.am
+++ libsamplerate-0.1.2/examples/Makefile.am
@@ -3,7 +3,7 @@
 
 bin_PROGRAMS = sndfile-resample
 
-noinst_PROGRAMS = varispeed-play
+EXTRA_PROGRAMS = varispeed-play
 
 SAMPLERATEDIR =../src
 INCLUDES = -I$(srcdir)/$(SAMPLERATEDIR) @OS_SPECIFIC_INCLUDES@
Index: libsamplerate-0.1.2/tests/Makefile.am
===================================================================
--- libsamplerate-0.1.2.orig/tests/Makefile.am
+++ libsamplerate-0.1.2/tests/Makefile.am
@@ -1,4 +1,4 @@
-noinst_PROGRAMS = misc_test termination_test simple_test callback_test \
+EXTRA_PROGRAMS = misc_test termination_test simple_test callback_test \
 					reset_test multi_channel_test snr_bw_test \
 					float_short_test src-evaluate benchmark
 
@@ -47,7 +47,7 @@ benchmark_LDADD  = $(SAMPLRATEDIR)/libsa
 
 #===============================================================================
 
-check: $(noinst_PROGRAMS)
+check: $(EXTRA_PROGRAMS)
 	date
 	uname -a
 	./misc_test$(EXEEXT)
