--- xmlcopyeditor-1.2.0.4.orig/src/replace.cpp
+++ xmlcopyeditor-1.2.0.4/src/replace.cpp
@@ -18,6 +18,7 @@
  */
 
 #include <cstring>
+#include <cstdio>
 #include "replace.h"
 
 int Replace::run (
--- xmlcopyeditor-1.2.0.4.orig/src/stringset.h
+++ xmlcopyeditor-1.2.0.4/src/stringset.h
@@ -25,6 +25,7 @@
 #include <cmath>
 #include <climits>
 #include <cstring>
+#include <stdint.h>
 #ifdef __WXMSW__
 #include <mem.h>
 #endif
--- xmlcopyeditor-1.2.0.4.orig/src/wrapaspell.cpp
+++ xmlcopyeditor-1.2.0.4/src/wrapaspell.cpp
@@ -19,6 +19,7 @@
 
 #include <iostream>
 #include <stdexcept>
+#include <cstdio>
 #include "wrapaspell.h"
 #include "aspell.h"
 #include "casehandler.h"
--- xmlcopyeditor-1.2.0.4.orig/src/xmlassociatedtd.h
+++ xmlcopyeditor-1.2.0.4/src/xmlassociatedtd.h
@@ -24,6 +24,7 @@
 #include <expat.h>
 #include <string>
 #include <memory>
+#include <cstdio>
 #include "wrapexpat.h"
 
 struct DtdData : public ParserData
--- xmlcopyeditor-1.2.0.4.orig/src/xmlassociatexsd.h
+++ xmlcopyeditor-1.2.0.4/src/xmlassociatexsd.h
@@ -25,6 +25,7 @@
 #include <string>
 #include <map>
 #include <memory>
+#include <cstdio>
 #include "wrapexpat.h"
 
 struct AssociateXsdData : public ParserData
--- xmlcopyeditor-1.2.0.4.orig/src/xmlassociatexsl.h
+++ xmlcopyeditor-1.2.0.4/src/xmlassociatexsl.h
@@ -24,6 +24,7 @@
 #include <expat.h>
 #include <string>
 #include <memory>
+#include <cstdio>
 #include "wrapexpat.h"
 
 struct XslData : public ParserData
--- xmlcopyeditor-1.2.0.4.orig/src/xmlpromptgenerator.cpp
+++ xmlcopyeditor-1.2.0.4/src/xmlpromptgenerator.cpp
@@ -31,7 +31,7 @@
 #include "catalogresolver.h"
 
 // Xerces-C req'd for Schema parsing
-#define XERCES_TMPLSINC
+// #define XERCES_TMPLSINC
 
 #include <xercesc/util/NameIdPool.hpp>
 #include <xercesc/util/PlatformUtils.hpp>
--- xmlcopyeditor-1.2.0.4.orig/src/xmlutf8reader.cpp
+++ xmlcopyeditor-1.2.0.4/src/xmlutf8reader.cpp
@@ -20,6 +20,7 @@
 #include <string>
 #include <vector>
 #include <stdexcept>
+#include <cstdio>
 #include <expat.h>
 #include "xmlutf8reader.h"
 
--- xmlcopyeditor-1.2.0.4.orig/src/xsllocator.cpp
+++ xmlcopyeditor-1.2.0.4/src/xsllocator.cpp
@@ -46,7 +46,7 @@
 		return;
 
 	char *value, *iterator;
-	value = strstr ( ( const char * ) data, "href=" );
+	value = strstr ( ( char * ) data, "href=" );
 	if ( !value || strlen ( value ) < 7 )
 		return;
 	value += 6;
@@ -55,7 +55,7 @@
 	{
 		if ( *iterator == '"' || *iterator == '\'' )
 		{
-			*iterator = '\0';
+			*iterator = '\0'; //Danger, changing const char *data
 			break;
 		}
 	}

 	  	 
