diff -urN Gauche-0.8.6.orig/src/Makefile.in Gauche-0.8.6/src/Makefile.in
--- Gauche-0.8.6.orig/src/Makefile.in	2005-11-04 17:26:41.000000000 +0900
+++ Gauche-0.8.6/src/Makefile.in	2005-11-15 17:37:05.000000000 +0900
@@ -206,7 +206,7 @@
 	$(POSTBUILD)
 
 gauche-config$(EXEEXT) : gauche-config.$(OBJEXT)
-	$(LINK) -o gauche-config$(EXEEXT) gauche-config.$(OBJEXT) $(LIBS)
+	$(RELINK) -o gauche-config$(EXEEXT) gauche-config.$(OBJEXT) $(LIBS)
 
 gauche-config.c gauche/arch.h : genconfig
 	$(SHELL) ./genconfig
