Patch from upstream git (commit 099a98e9) to fix bug #247549.

--- lib/action_controller/mime_type.rb.1
+++ lib/action_controller/mime_type.rb
@@ -18,7 +18,7 @@
   #   end
   class Type
     @@html_types = Set.new [:html, :all]
-    @@unverifiable_types = Set.new [:text, :json, :csv, :xml, :rss, :atom, :yaml]
+    @@unverifiable_types = Set.new [:json, :csv, :xml, :rss, :atom, :yaml]
     cattr_reader :html_types, :unverifiable_types
 
     # A simple helper class used in parsing the accept header
