--- Makefile.am
+++ Makefile.am
@@ -17,13 +17,13 @@
 
 noinst_PROGRAMS = test1 test2 test3
 test1_SOURCES = test1.c
-test1_LDFLAGS = 
+test1_LDFLAGS = @LTLIBICONV@
 test1_LDADD = $(usrlib_LTLIBRARIES)
 test2_SOURCES = test2.c
-test2_LDFLAGS = 
+test2_LDFLAGS = @LTLIBICONV@
 test2_LDADD = $(usrlib_LTLIBRARIES)
 test3_SOURCES = test3.c
-test3_LDFLAGS = 
+test3_LDFLAGS = @LTLIBICONV@
 test3_LDADD = $(usrlib_LTLIBRARIES)
 
 noinst_SCRIPTS = testit.sh
--- configure.ac
+++ configure.ac
@@ -95,6 +95,7 @@
 AC_CHECK_FUNCS(getuid geteuid mtrace __secure_getenv setregid strerror iconv)
 
 AM_GNU_GETTEXT([external])
+AM_ICONV_LINK
 
 popt_sysconfdir="${sysconfdir}"
 eval "popt_sysconfdir=\"${popt_sysconfdir}\"" # expand contained ${prefix}
