That module doesn't compile on freebsd7.
--- libavdevice/Makefile.orig	2009-03-02 00:45:31 +0100
+++ libavdevice/Makefile	2009-03-02 00:45:59 +0100
@@ -10,7 +10,6 @@ OBJS    = alldevices.o
 # input/output devices
 OBJS-$(CONFIG_ALSA_DEMUXER)              += alsa-audio-common.o alsa-audio-dec.o
 OBJS-$(CONFIG_ALSA_MUXER)                += alsa-audio-common.o alsa-audio-enc.o
-OBJS-$(CONFIG_BKTR_DEMUXER)              += bktr.o
 OBJS-$(CONFIG_DV1394_DEMUXER)            += dv1394.o
 OBJS-$(CONFIG_OSS_DEMUXER)               += oss_audio.o
 OBJS-$(CONFIG_OSS_MUXER)                 += oss_audio.o
--- libavdevice/alldevices.c.orig	2009-03-02 00:45:39 +0100
+++ libavdevice/alldevices.c	2009-03-02 00:45:49 +0100
@@ -46,7 +46,6 @@ void avdevice_register_all(void)
     /* devices */
     REGISTER_MUXDEMUX (ALSA, alsa);
     REGISTER_MUXDEMUX (AUDIO_BEOS, audio_beos);
-    REGISTER_DEMUXER  (BKTR, bktr);
     REGISTER_DEMUXER  (DV1394, dv1394);
     REGISTER_MUXDEMUX (OSS, oss);
     REGISTER_DEMUXER  (V4L2, v4l2);
