--- configure.orig	2008-10-13 18:38:18.000000000 +0200
+++ configure	2008-10-13 18:49:15.000000000 +0200
@@ -1817,7 +1817,7 @@
     # LIBOBJFLAGS may have already been set in the OS configuration
     if test -z "$LIBOBJFLAGS" ; then
         case "$arch" in
-            x86_64|ia64|alpha|sparc*|power*) LIBOBJFLAGS='$(PIC)' ;;
+            x86_64|ia64|alpha|sparc*|power*|parisc) LIBOBJFLAGS='$(PIC)' ;;
         esac
     fi
 fi
