diff -ur /tmp/work/ffmpeg/libswscale/rgb2rgb.c ffmpeg/libswscale/rgb2rgb.c
--- ffmpeg/libswscale/rgb2rgb.c	2007-06-16 13:03:04 +0400
+++ ffmpeg/libswscale/rgb2rgb.c	2008-04-08 23:22:53 +0400
@@ -171,7 +171,7 @@
 #define RENAME(a) a ## _C
 #include "rgb2rgb_template.c"
 
-#if defined(ARCH_X86) && defined(CONFIG_GPL)
+#if defined(ARCH_X86) && defined(CONFIG_GPL) && !defined(__APPLE__)
 
 //MMX versions
 #undef RENAME
