--- cscope-15.6/contrib/webcscope/cscope.orig	2007-01-06 17:56:32.000000000 -0500
+++ cscope-15.6/contrib/webcscope/cscope	2007-01-06 17:57:41.000000000 -0500
@@ -1,4 +1,4 @@
-#!/bin/perl
+#!/usr/bin/perl
 
 # $Id: cscope-158831-warning_webscope.patch,v 1.3 2007/01/06 23:09:49 seemant Exp $
 #
@@ -375,6 +375,7 @@
 	print &HtmlTop ("Web-CScope");
 
 	print <<ENDOFTEXTA;
+<p style="color:red">Be aware that this web front-end is insecure and allows ALL apache readable files to be viewed -- including your configuration!</p>
 Select an operation below and enter a symbol, function or text to search in
 the database.  The active version is $version.  Input is case-sensitive,
 so if your search returns no results, check the case and the symbol name.<hr>
