diff -ru openssh-4.7p1.orig/uidswap.c openssh-4.7p1/uidswap.c
--- openssh-4.7p1.orig/uidswap.c	2009-02-10 16:04:56 +0100
+++ openssh-4.7p1/uidswap.c	2009-02-10 16:07:53 +0100
@@ -14,6 +14,10 @@
 
 #include "includes.h"
 
+#ifdef __INTERIX
+# include <interix/security.h>
+#endif
+
 #include <sys/param.h>
 #include <errno.h>
 #include <pwd.h>
@@ -225,6 +229,10 @@
 	debug("permanently_set_uid: %u/%u", (u_int)pw->pw_uid,
 	    (u_int)pw->pw_gid);
 
+#ifdef __INTERIX
+	if (setuser(pw->pw_name, NULL, SU_COMPLETE))
+		fatal("setuser %.100s: %.100s", pw->pw_name, strerror(errno));
+#else
 #if defined(HAVE_SETRESGID) && !defined(BROKEN_SETRESGID)
 	if (setresgid(pw->pw_gid, pw->pw_gid, pw->pw_gid) < 0)
 		fatal("setresgid %u: %.100s", (u_int)pw->pw_gid, strerror(errno));
@@ -252,6 +260,7 @@
 	if (setuid(pw->pw_uid) < 0)
 		fatal("setuid %u: %.100s", (u_int)pw->pw_uid, strerror(errno));
 #endif
+#endif /* __INTERIX */
 
 #if !defined(HAVE_CYGWIN) && !defined(__INTERIX)
 	/* Try restoration of GID if changed (test clearing of saved gid) */
