diff -ru openssh-5.2p1.orig/session.c openssh-5.2p1/session.c
--- openssh-5.2p1.orig/session.c	2009-08-25 09:27:05 +0200
+++ openssh-5.2p1/session.c	2009-08-25 09:55:05 +0200
@@ -1586,6 +1586,8 @@
 		if(InterixPwdToken) {
 			debug2("re-setting user password");
 			strcpy(pw->pw_passwd, InterixPwdToken);
+		} else {
+			INTERIX_PWD_WARNING
 		}
 #endif
 		/* Permanently switch to the desired uid. */
diff -ru openssh-5.2p1.orig/uidswap.c openssh-5.2p1/uidswap.c
--- openssh-5.2p1.orig/uidswap.c	2009-08-25 09:27:08 +0200
+++ openssh-5.2p1/uidswap.c	2009-08-25 09:57:12 +0200
@@ -16,20 +16,6 @@
 
 #ifdef __INTERIX
 # include <interix/security.h>
-#define INTERIX_PWD_WARNING \
-		fprintf(stderr,"@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\n" \
-					   "@ WARNING: Due to limitations in the POSIX Subsystem and Win32  @\n" \
-					   "@     a Password is required to acquire a full authentication.  @\n" \
-					   "@     Without such an authentication token, certain things will @\n" \
-					   "@     only be available in a very limited way (Visual Studio's  @\n" \
-					   "@     link.exe can only link without debug information, network @\n" \
-					   "@     shares that require user authentication don't fully work, @\n" \
-					   "@     etc.). However if you don't require those things to work, @\n" \
-					   "@     you may be just fine without password (public-key, etc.). @\n" \
-					   "@ To obtain a full authentication you need to use password      @\n" \
-					   "@ authentication at the moment. To do so, remove your public    @\n" \
-					   "@ key from your ~/.ssh/authorized_keys[2] file(s).              @\n" \
-					   "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\n");
 #endif
 
 #include <sys/param.h>
@@ -250,7 +236,6 @@
 	}
 
 	if (strcmp(pw->pw_passwd, "*") == 0) {
-		INTERIX_PWD_WARNING
 		if (setuser(pw->pw_name, NULL, SU_COMPLETE))
 			fatal("setuser %.100s: %.100s", pw->pw_name, strerror(errno));
 	} else {
