diff -u htop-0.8.1.orig/RichString.c htop-0.8.1/RichString.c
--- htop-0.8.1.orig/RichString.c	2008-03-09 03:23:49.000000000 +0100
+++ htop-0.8.1/RichString.c	2008-11-29 19:07:29.284860473 +0100
@@ -58,7 +58,7 @@
    int last = MIN(RICHSTRING_MAXLEN - 1, len + this->len);
    for (int i = this->len, j = 0; i < last; i++, j++) {
       memset(&this->chstr[i], 0, sizeof(this->chstr[i]));
-      this->chstr[i].chars[0] = data[j];
+      this->chstr[i].chars[0] = ((data[j] > 31) ? data[j] : '?' );
       this->chstr[i].attr = attrs;
    }
    this->chstr[last].chars[0] = 0;
