*** doc/refcard.tex	Mon Nov  6 19:59:00 2000
--- doc/refcard.tex.new	Tue Jun 24 03:07:15 2003
***************
*** 17,23 ****
--- 17,29 ----
  %% which I modified from his GNU emacs reference card
  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
  \def\TITLE{Pari-GP reference card}
+ % ignore parimacro.tex's \magnification setting
+ \let\oldmagnification\magnification
+ \catcode`@=11
+ \def\magnification{\afterassignment\relax\count@}%
+ \catcode`@=12
  \input parimacro.tex
+ \let\magnification\oldmagnification
  \ifPDF
    \input pdfmacs.tex
    \pdfpagewidth=11.69in
*** doc/users.tex	Mon Nov  6 19:59:01 2000
--- doc/users.tex.new	Tue Jun 24 03:07:15 2003
***************
*** 63,69 ****
    \ifPDF \writesecnumbers \fi
    \expandafter\end % stop here the first time (don't process index)
  \fi
- \newpage
  
  \ifPDF
  % Add a bookmark entry for the index.   CHB
--- 63,68 ----
