--- Configure
+++ Configure
@@ -184,39 +184,7 @@
 #
 # Proper PATH setting
 #
-pth="\
-  /bin\
-  /usr/bin\
-  /usr/locateal/bin\
-  /usr/ucb\
-  /usr/locateal\
-  /usr/lbin\
-  /usr/5bin\
-  /etc\
-  /usr/etc\
-  /usr/gnu/bin\
-  /usr/new\
-  /usr/new/bin\
-  /usr/nbin\
-  /sys5.3/bin\
-  /sys5.3/usr/bin\
-  /bsd4.3/bin\
-  /bsd4.3/usr/ucb\
-  /bsd4.3/usr/bin\
-  /usr/bsd\
-  /bsd43/bin\
-  /opt/ansic/bin\
-  /usr/ccs/bin\
-  /usr/lib\
-  /usr/ucblib\
-  /lib\
-  /usr/ccs/lib\
-  /sbin\
-  /usr/sbin\
-  /usr/libexec\
-  /usr/openwin/bin\
-  /usr/local/bin\
-"
+pth=${PATH}
 # /usr/openwin/bin added for xmkmf
 
 for p in $pth
@@ -418,21 +386,8 @@
 	fi;;
       os2) libpth=`echo $LIBRARY_PATH | sed 's/;/ /g' | sed 's,\\\\,/,g'`;;
  	*) libpth="\
- 	/usr/local/lib\
- 	/lib\
- 	/usr/lib\
- 	/opt/lib\
- 	/opt/local/lib\
- 	/opt/gnu/lib\
- 	/lib/pa1.1\
- 	/usr/lib/large\
- 	/lib/large\
- 	/usr/lib/small\
- 	/lib/small\
- 	/usr/ccs/lib\
- 	/usc/ucblib\
- 	/usr/shlib\
- 	.\
+    @GENTOO_PORTAGE_EPREFIX@/usr/lib\
+    @GENTOO_PORTAGE_EPREFIX@/lib\
      ";;
      esac
 
@@ -546,17 +501,7 @@
        $cygtop/H-${arch}-cygwin32/${arch}-cygwin32/include\
      ";;
       *) incpth="\
-      /usr/local/include/readline\
-      /usr/include/readline\
-      /opt/include/readline\
-      /opt/local/include/readline\
-      /opt/gnu/include/readline\
-      /usr/unsupported/include/readline\
-      /usr/local/include\
-      /usr/include\
-      /opt/include\
-      /opt/local/include\
-      /opt/gnu/include\
+      @GENTOO_PORTAGE_EPREFIX@/usr/include/readline\
     ";;
     esac
     pth="$readline_add $incpth"
