--- Configure
+++ Configure
@@ -904,7 +904,8 @@
 #
 LIBS=-lm
 case "$osname" in
-  darwin) LIBS= ;;
+  darwin*) LIBS=
+  LD=$CC; ld=$CC;;
 esac
 case "$osname-$arch" in
   linux-*|cygwin*|gnu-*)
@@ -954,7 +954,7 @@
 #    aix-*)  DLSUFFIX=a  ;; dynamic linking does not work!
     sunos-*) sodest=$VersionMajor$VersionMinor.$patch
              soname=$sodest;;
-    gnu-*|*-alpha|solaris-*|linux-*|freebsd-*)
+    gnu-*|*-alpha|solaris-*|linux-*|freebsd-*|darwin*-*)
       case $libpari_base in
         pari) sodest=$version.$patch;; # released versions
         *) sodest=$patch.0.0;; # unstable versions
@@ -988,6 +988,7 @@
     # not GNU ld
     case "$osname-$arch" in
       aix-*)      DLLDFLAGS="-r" ;;
+      darwin*-*)   DLLDFLAGS="-dynamiclib -install_name ${dfltlibdir}/\$(LIBPARI_SONAME)"; DLSUFFIX=dylib;;
       hpux-*)     DLLDFLAGS="-b -E" ;;
       freebsd-*)  DLLDFLAGS="-Bshareable -x" ;;
       gnu-*|linux-*)    DLLDFLAGS="-shared -soname \$(LIBPARI_SONAME)" ;;
@@ -1004,6 +1004,6 @@
   fi
   case "$osname" in 
-    freebsd|linux|sunos|solaris) DLLDFLAGS="$DLLDFLAGS -lc -lm";;
+    freebsd|linux|sunos|solaris|darwin*) DLLDFLAGS="$DLLDFLAGS -lc -lm";;
   esac
 
   if test "$fastread" != yes; then
--- config/Makefile.SH
+++ config/Makefile.SH
@@ -157,9 +157,15 @@
 LIB_PREFIX = lib
 LIBPARI    = \$(LIB_PREFIX)$libpari_base
 LIBPARI_STA= \$(LIBPARI)\$(_A)
+ifeq (\$(shell uname -s), Darwin)
+LIBPARI_SO = \$(LIBPARI).$DLSUFFIX
+LIBPARI_DYN   = \$(LIBPARI).$sodest.$DLSUFFIX
+LIBPARI_SONAME= \$(LIBPARI).$soname.$DLSUFFIX
+else
 LIBPARI_SO = \$(LIBPARI).$DLSUFFIX
 LIBPARI_DYN   = \$(LIBPARI_SO).$sodest
 LIBPARI_SONAME= \$(LIBPARI_SO).$soname
+endif
 
 RUNPTH       = $RUNPTH
 RUNPTH_FINAL = $RUNPTH_FINAL
