diff -urN numpy-1.0.4/numpy/distutils/cpuinfo.py numpy-1.0.4-patched/numpy/distutils/cpuinfo.py
--- numpy-1.0.4/numpy/distutils/cpuinfo.py	2007-11-07 17:05:15.000000000 -0500
+++ numpy-1.0.4-patched/numpy/distutils/cpuinfo.py	2008-01-11 11:44:23.000000000 -0500
@@ -222,7 +222,12 @@
         return self.is_PentiumIV() and self.has_sse3()
 
     def _is_Nocona(self):
-        return self.is_64bit() and self.is_PentiumIV()
+        return self.is_Intel() \
+            and ( self.info[0]['cpu family'] == '6' \
+                or self.info[0]['cpu family'] == '15' ) \
+            and self.has_sse3() \
+            and re.match(r'.*?\blm\b',self.info[0]['flags']) is not None
+
 
     def _is_Core2(self):
         return self.is_64bit() and self.is_Intel() and \
@@ -263,7 +268,7 @@
         return re.match(r'.*?\bsse2\b',self.info[0]['flags']) is not None
 
     def _has_sse3(self):
-        return re.match(r'.*?\bsse3\b',self.info[0]['flags']) is not None
+        return re.match(r'.*?\bsss?e3\b',self.info[0]['flags']) is not None
 
     def _has_3dnow(self):
         return re.match(r'.*?\b3dnow\b',self.info[0]['flags']) is not None
