*** plotutils-2.4.1/graph/plotter.c.orig	Thu Mar  2 22:05:24 2000
--- plotutils-2.4.1/graph/plotter.c	Tue Apr 29 01:48:09 2003
***************
*** 1045,1051 ****
    multigrapher->x_trans.output_min = 0.0;
    multigrapher->x_trans.output_max = (double)PLOT_SIZE;
    multigrapher->x_trans.output_range = multigrapher->x_trans.output_max - multigrapher->x_trans.output_min;
!   multigrapher->x_trans.output_min = 0.0;
    multigrapher->y_trans.output_max = (double)PLOT_SIZE;
    multigrapher->y_trans.output_range = multigrapher->y_trans.output_max - multigrapher->y_trans.output_min;
  
--- 1045,1051 ----
    multigrapher->x_trans.output_min = 0.0;
    multigrapher->x_trans.output_max = (double)PLOT_SIZE;
    multigrapher->x_trans.output_range = multigrapher->x_trans.output_max - multigrapher->x_trans.output_min;
!   multigrapher->y_trans.output_min = 0.0;
    multigrapher->y_trans.output_max = (double)PLOT_SIZE;
    multigrapher->y_trans.output_range = multigrapher->y_trans.output_max - multigrapher->y_trans.output_min;
  
*** plotutils-2.4.1/libplot/p_header.h.orig	Mon Sep  6 01:08:46 1999
--- plotutils-2.4.1/libplot/p_header.h	Wed Dec  3 23:43:54 2003
***************
*** 288,294 ****
  gsave\n\
  fgred fggreen fgblue setrgbcolor\n\
  /fontDict printFont findfont printSize scalefont dup setfont def\n\
! /descender fontDict begin 0 [FontBBox] 1 get FontMatrix end\n\
  transform exch pop def\n\
  /vertoffset 1 printSize sub descender sub def {\n\
  0 vertoffset moveto show\n\
--- 288,294 ----
  gsave\n\
  fgred fggreen fgblue setrgbcolor\n\
  /fontDict printFont findfont printSize scalefont dup setfont def\n\
! /descender fontDict begin 0 fontDict /FontBBox get 1 get FontMatrix end\n\
  transform exch pop def\n\
  /vertoffset 1 printSize sub descender sub def {\n\
  0 vertoffset moveto show\n\
