--- test/plot2ps.xout	2007-03-13 07:23:11.000000000 +0100
+++ /tmp/test/plot2ps.xout	2007-03-13 07:23:46.000000000 +0100
@@ -1,7 +1,6 @@
 %!PS-Adobe-3.0 EPSF-3.0
 %%Creator: GNU libplot drawing library 4.1
 %%Title: PostScript plot
-%%CreationDate: Fri Jun 16 21:08:33 2000
 %%DocumentData: Clean7Bit
 %%LanguageLevel: 1
 %%Pages: 1
@@ -294,7 +293,7 @@
 gsave
 fgred fggreen fgblue setrgbcolor
 /fontDict printFont findfont printSize scalefont dup setfont def
-/descender fontDict begin 0 [FontBBox] 1 get FontMatrix end
+/descender fontDict begin 0 fontDict /FontBBox get 1 get FontMatrix end
 transform exch pop def
 /vertoffset 1 printSize sub descender sub def {
 0 vertoffset moveto show
