--- plotutils-2.4.1/libplot/g_write.c~	Fri May 19 18:10:01 2000
+++ plotutils-2.4.1/libplot/g_write.c	Sun Jun 16 03:22:30 2002
@@ -40,7 +40,7 @@
     }
 #ifdef LIBPLOTTER
   else if (data->outstream)
-    data->outstream->write(c, n);
+    data->outstream->write ((const char*)c, n);
 #endif
 }
 
--- plotutils-2.4.1/libplot/i_rle.c~	Sun Jun 27 18:58:10 1999
+++ plotutils-2.4.1/libplot/i_rle.c	Sun Jun 16 03:22:30 2002
@@ -78,7 +78,7 @@
   else if (rle->outstream)
     {
       rle->outstream->put ((unsigned char)(rle->oblen));
-      rle->outstream->write (&(rle->oblock[0]), rle->oblen);
+      rle->outstream->write ((const char*)(&rle->oblock[0]), rle->oblen);
     }
 #endif  
 
--- plotutils-2.4.1/libplot/n_write.c~	Fri Jun 16 07:42:13 2000
+++ plotutils-2.4.1/libplot/n_write.c	Sun Jun 16 03:22:30 2002
@@ -208,7 +208,7 @@
 		  linebuf[pos++] = '0';
 		if (pos >= MAX_PBM_PIXELS_PER_LINE || i == (width - 1))
 		  {
-		    stream->write (linebuf, pos);
+		    stream->write ((const char*)linebuf, pos);
 		    stream->put ('\n');
 
 		    pos = 0;
@@ -253,7 +253,7 @@
 		  rowbuf[bytecount++] = outbyte;
 		}
 	      /* emit row of bytes */
-	      stream->write (rowbuf, bytecount);
+	      stream->write ((const char*)rowbuf, bytecount);
 	    }
 
 	  free (rowbuf);
@@ -366,7 +366,7 @@
 		num_pixels++;
 		if (num_pixels >= MAX_PGM_PIXELS_PER_LINE || i == (width - 1))
 		  {
-		    stream->write (linebuf, pos);
+		    stream->write ((const char*)linebuf, pos);
 		    stream->put ('\n');
 
 		    num_pixels = 0;
@@ -392,7 +392,7 @@
 	    {
 	      for (i = 0; i < width; i++)
 		rowbuf[i] = pixmap[j][i].u.rgb[0];
-	      stream->write (rowbuf, width);
+	      stream->write ((const char*)rowbuf, width);
 	    }
 	  free (rowbuf);
 	}
@@ -514,7 +514,7 @@
 		num_pixels++;
 		if (num_pixels >= MAX_PPM_PIXELS_PER_LINE || i == (width - 1))
 		  {
-		    stream->write (linebuf, pos);
+		    stream->write ((const char*)linebuf, pos);
 		    stream->put ('\n');
 
 		    num_pixels = 0;
@@ -542,7 +542,7 @@
 	      for (i = 0; i < width; i++)
 		for (component = 0; component < 3; component++)
 		  rowbuf[3 * i + component] = pixmap[j][i].u.rgb[component];
-	      stream->write (rowbuf, 3 * width);
+	      stream->write ((const char*)rowbuf, 3 * width);
 	    }
 	  free (rowbuf);
 	}
--- plotutils-2.4.1/libplot/z_write.c~	Tue Jun 20 06:34:42 2000
+++ plotutils-2.4.1/libplot/z_write.c	Sun Jun 16 03:22:30 2002
@@ -484,7 +484,7 @@
   ostream *stream;
 
   stream = (ostream *)png_get_io_ptr (png_ptr);
-  stream->write (data, length);
+  stream->write ((const char*)data, length);
 }
 
 static void 
