--- gcc-config-1.4.0	2008-12-17 21:08:31.000000000 +0100
+++ gcc-config-1.4.0	2008-12-17 21:08:51.000000000 +0100
@@ -329,11 +329,6 @@
 	#
 	# The funky move magic is required for proper updating of in-use files.
 	if ! is_cross_compiler ; then
-		# Need to cut out extra paths in multilib case and pray the first path
-		# is the "root" multilib path ... maybe some day change this to use
-		# `gcc -print-file-name` ...
-		LDPATH=${LDPATH%%:*}
-
 		# We can copy the other libs to /usr/lib without funky copying,
 		# don't use symlinks as on Darwin we need to fix the
 		# install_names of the objects.  Omit libgcc and unwind as we
@@ -525,6 +520,11 @@
 				echo "LDPATH='${LDPATH}'"
 			)
 
+			# Need to cut out extra paths in multilib case and pray the first path
+			# is the "root" multilib path ... maybe some day change this to use
+			# `gcc -print-file-name` ...
+			LDPATH=${LDPATH%%:*}
+
 			for multilib in $(
 				echo ".;" # ensure we always operate on base libdir first
 				"${ROOT}/${GCC_PATH}"/gcc -print-multi-lib
