--- gcc-config-1.3.13.orig	2006-09-07 23:21:31.000000000 +0200
+++ gcc-config-1.3.13	2006-09-07 23:22:16.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!GENTOO_PORTAGE_EPREFIX/bin/bash
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo-x86/sys-devel/gcc-config/files/gcc-config-1.3.13,v 1.12 2006/08/09 05:54:00 vapier Exp $
@@ -5,9 +5,11 @@
 
 trap ":" INT QUIT TSTP
 
+EPREFIX="GENTOO_PORTAGE_EPREFIX"
+
 argv0=${0##*/}
-source /sbin/functions.sh || {
-	echo "${argv0}: Could not source /sbin/functions.sh!"
+source ${EPREFIX}/sbin/functions.sh || {
+	echo "${argv0}: Could not source ${EPREFIX}/sbin/functions.sh!"
 	exit 1
 }
 umask 022
@@ -53,8 +55,8 @@
 	fi
 
 	for x in /bin /sbin /usr/bin /usr/sbin /usr/local/bin /usr/local/sbin ; do
-		if [[ -x ${x}/$1 ]] && [[ -r ${x}/$1 ]] ; then
-			echo "${x}/$1"
+		if [[ -x ${EPREFIX}/${x}/$1 ]] && [[ -r ${EPREFIX}/${x}/$1 ]] ; then
+			echo "${EPREFIX}/${x}/$1"
 			return 0
 		fi
 	done
@@ -153,10 +155,11 @@
 	local OLD_CC_COMP=
 	local GCC_BIN_PATH=
 
-	if [[ "$(id -u)" -ne 0 ]] ; then
-		eerror "${argv0}: Must be root."
-		exit 1
-	fi
+ 	# shouldn't be here in unprivileged
+#	if [[ "$(id -u)" -ne 0 ]] ; then
+#		eerror "${argv0}: Must be root."
+#		exit 1
+#	fi
 
 	if is_cross_compiler ; then
 		ebegin "Switching cross-compiler to ${CC_COMP}"
@@ -243,9 +246,9 @@
 	fi
 
 	# Save PATH
-	GCC_BIN_PATH=${PATH}
+	GCC_BIN_PATH=$(echo "${PATH}" | ${SED} -e "s|${ROOT}/*|/|")
 	# Fix environment
-	source /etc/profile
+	source "${EPREFIX}"/etc/profile
 	umask 022
 	# Find the bin wrapper
 	local wrapper
@@ -337,7 +340,7 @@
 			echo ""
 			${GREP} -h ^LDPATH= "${ROOT}"/etc/env.d/05gcc \
 				| ${SED} -e 's:^LDPATH=::' -e 's:"::g' -e 's|:|\n|g' \
-				>> /etc/ld.so.conf
+				>> ${EPREFIX}/etc/ld.so.conf
 			ldconfig
 		fi
 	fi
@@ -349,7 +352,7 @@
 		ewarn "If you intend to use the gcc from the new profile in an already"
 		ewarn "running shell, please remember to do:"
 		echo
-		ewarn "  # source /etc/profile"
+		ewarn "  # source ${EPREFIX}/etc/profile"
 		echo
 	fi
 
@@ -544,7 +547,7 @@
 FORCE="no"
 
 CC_COMP=
-[[ -z ${ROOT} ]] && ROOT="/"
+[[ -z ${ROOT} ]] && ROOT="${EPREFIX}"
 [[ ${ROOT:0-1} != "/" ]] && ROOT="${ROOT}/"
 ENV_D="${ROOT}etc/env.d"
 GCC_ENV_D="${ENV_D}/gcc"
@@ -679,7 +682,9 @@
 if [[ ${DOIT} != "get_current_profile" ]] ; then
 	GCC_LIB=$(
 		source "${GCC_ENV_D}/${CC_COMP}"
-		echo ${LDPATH} | ${AWK} -F/ '{ print  "/"$2"/"$3"/"$4"/" }'
+		echo ${LDPATH} | \
+ 			${SED} -e "s|${ROOT}/*|/|g" | \
+			${AWK} -F/ '{ print "/"$2"/"$3"/"$4"/" }'
 	)
 
 	CC_COMP_VERSION=$(chop_gcc_ver_spec ${CC_COMP})
