--- wrapper-1.4.7.c.orig	2006-09-07 21:48:23.000000000 +0200
+++ wrapper-1.4.7.c	2006-09-07 21:51:50.000000000 +0200
@@ -21,8 +21,8 @@
 #include <stdarg.h>
 #include <errno.h>
 
-#define GCC_CONFIG "/usr/bin/gcc-config"
-#define ENVD_BASE  "/etc/env.d/05gcc"
+#define GCC_CONFIG "@GENTOO_PORTAGE_EPREFIX@/usr/bin/gcc-config"
+#define ENVD_BASE  "@GENTOO_PORTAGE_EPREFIX@/etc/env.d/05gcc"
 
 struct wrapper_data {
 	char name[MAXPATHLEN + 1];
@@ -328,7 +328,7 @@
 
 	/* What is the full name of our wrapper? */
 	size = sizeof(data.fullname);
-	i = snprintf(data.fullname, size, "/usr/bin/%s", data.name);
+	i = snprintf(data.fullname, size, "@GENTOO_PORTAGE_EPREFIX@/usr/bin/%s", data.name);
 	if ((i == -1) || (i > (int)size))
 		wrapper_exit("invalid wrapper name: \"%s\"\n", data.name);
 
