http://bugs.gentoo.org/show_bug.cgi?id=268793

--- Makefile.in
+++ Makefile.in
@@ -424,16 +424,6 @@
 	    ${INSTALL_DATA} lib-src/$$f $(DESTDIR)${archlibdir}/$$f; \
 	  else true; fi ; \
 	done
-	if test "${ns_appdir}" != ""; then \
-	  ( cd ${ns_appresdir} ; \
-	    if test -d share/emacs ; then dir=share/emacs/*/*; $(MV_DIRS); fi;\
-	    if test -d share/info ; then dir=share/info; $(MV_DIRS) ; fi ; \
-	    rm -fr share ) ; \
-	  ( cd ${ns_appbindir}/libexec ; dir=emacs/*/*/* ; $(MV_DIRS); \
-	    rm -fr emacs ) ; \
-	  ( cd ${ns_appbindir}/bin ; rm -f emacs emacs-23* ; \
-	    ln -sf ../libexec/* .) ; \
-	else true ; fi
 
 ## http://lists.gnu.org/archive/html/emacs-devel/2007-10/msg01672.html
 ## Needs to be the user running install, so configure can't set it.
--- configure.in
+++ configure.in
@@ -1391,10 +1391,10 @@
   window_system=nextstep
   with_xft=no
   # set up packaging dirs
-  exec_prefix=${ns_appbindir}
-  libexecdir=${ns_appbindir}/libexec
   if test "${EN_NS_SELF_CONTAINED}" = yes; then
      prefix=${ns_appresdir}
+     exec_prefix=${ns_appbindir}
+     libexecdir=${ns_appbindir}/libexec
   fi
 fi
 CFLAGS="$tmp_CFLAGS"
--- leim/Makefile.in
+++ leim/Makefile.in
@@ -220,7 +220,6 @@
 	fi
 	sed -n '/^[^;]/ p' < ${srcdir}/leim-ext.el >> $@
 
-MV_DIRS = for i in $$dir; do rm -fr `basename "$$i"` ; mv "$$i" . ; done
 
 install: all
 	if [ ! -d ${INSTALLDIR} ] ; then \
@@ -263,11 +262,6 @@
 	  [ -n "$${installuser}" ] && break ; \
 	done ; \
 	find ${INSTALLDIR} -exec chown $${installuser} '{}' ';'
-	if [ "${ns_appresdir}" != "" ]; then \
-	  ( cd ${ns_appresdir} ; \
-	    if test -d share/emacs ; then dir=share/emacs/*/*; $(MV_DIRS); fi;\
-	    rm -fr share ) ; \
-	else true ; fi
 
 clean mostlyclean:
 	rm -f ${TIT-MISC} ${TIT-MISC:.elc=.el} \
