diff -ur ledger-2.6.1.old/gnucash.cc ledger-2.6.1/gnucash.cc
--- ledger-2.6.1.old/gnucash.cc	2009-07-07 09:10:35.461037758 +0300
+++ ledger-2.6.1/gnucash.cc	2009-07-07 09:29:17.462613252 +0300
@@ -201,7 +201,7 @@
 {
   const char * num = number.c_str();
 
-  if (char * p = std::strchr(num, '/')) {
+  if (const char * p = std::strchr(num, '/')) {
     std::string numer_str(num, p - num);
     std::string denom_str(p + 1);
 
diff -ur ledger-2.6.1.old/option.cc ledger-2.6.1/option.cc
--- ledger-2.6.1.old/option.cc	2009-07-07 09:10:35.459038062 +0300
+++ ledger-2.6.1/option.cc	2009-07-07 09:15:26.238434229 +0300
@@ -893,7 +893,7 @@
 namespace {
   void parse_price_setting(const char * optarg)
   {
-    char * equals = std::strchr(optarg, '=');
+    const char * equals = std::strchr(optarg, '=');
     if (! equals)
       return;
 
diff -ur ledger-2.6.1.old/textual.cc ledger-2.6.1/textual.cc
--- ledger-2.6.1.old/textual.cc	2009-07-07 09:10:35.458037934 +0300
+++ ledger-2.6.1/textual.cc	2009-07-07 09:26:23.627028972 +0300
@@ -298,8 +298,8 @@
       DEBUG_PRINT("ledger.textual.parse", "line " << linenum << ": " <<
 		  "Parsed a note '" << xact->note << "'");
 
-      if (char * b = std::strchr(xact->note.c_str(), '['))
-	if (char * e = std::strchr(xact->note.c_str(), ']')) {
+      if (const char * b = std::strchr(xact->note.c_str(), '['))
+	if (const char * e = std::strchr(xact->note.c_str(), ']')) {
 	  char buf[256];
 	  std::strncpy(buf, b + 1, e - b - 1);
 	  buf[e - b - 1] = '\0';
