diff -ur mc-4.7.0-pre1.orig/syntax/ebuild.syntax mc-4.7.0-pre1/syntax/ebuild.syntax
--- mc-4.7.0-pre1.orig/syntax/ebuild.syntax	2009-06-23 21:55:12.000000000 +0300
+++ mc-4.7.0-pre1/syntax/ebuild.syntax	2009-08-02 18:37:56.000000000 +0300
@@ -38,9 +38,9 @@
     keyword LINGUAS brightcyan
 
     keyword whole _elibtoolize yellow
-    keyword whole check_KV yellow
     keyword whole best_version yellow
     keyword whole built_with_use yellow
+    keyword whole check_KV yellow
     keyword whole die yellow
     keyword whole diropts yellow
     keyword whole dobin yellow
@@ -64,6 +64,7 @@
     keyword whole dosbin yellow
     keyword whole dosed yellow
     keyword whole dosym yellow
+    keyword whole eaclocal yellow
     keyword whole eautoconf yellow
     keyword whole eautoreconf yellow
     keyword whole eautomake yellow
@@ -110,6 +111,8 @@
     keyword whole use yellow
     keyword whole use_enable yellow
     keyword whole use_with yellow
+    #cvs
+    keyword whole cvs_src_unpack yellow
     #flag-o-matic
     keyword whole append-flags yellow
     keyword whole replace-flags yellow
@@ -118,6 +121,8 @@
     keyword whole git_src_unpack yellow
     #java-ant-2
     keyword whole EANT_BUILD_TARGET yellow
+    #mercurial
+    keyword whole mercurial_src_unpack yellow
     #svn
     keyword whole subversion_src_unpack yellow
 
