diff -ruN mc-4.6.1.orig/intl/Makefile.in mc-4.6.1/intl/Makefile.in
--- mc-4.6.1.orig/intl/Makefile.in	2007-03-06 22:31:02.000000000 +0300
+++ mc-4.6.1/intl/Makefile.in	2007-03-06 22:38:52.000000000 +0300
@@ -156,12 +156,7 @@
 	  test @GLIBC21@ != no || $(mkinstalldirs) $(DESTDIR)$(libdir); \
 	  temp=$(DESTDIR)$(libdir)/t-charset.alias; \
 	  dest=$(DESTDIR)$(libdir)/charset.alias; \
-	  if test -f $(DESTDIR)$(libdir)/charset.alias; then \
-	    orig=$(DESTDIR)$(libdir)/charset.alias; \
-	    sed -f ref-add.sed $$orig > $$temp; \
-	    $(INSTALL_DATA) $$temp $$dest; \
-	    rm -f $$temp; \
-	  else \
+	  if ! test -f $(libdir)/charset.alias; then \
 	    if test @GLIBC21@ = no; then \
 	      orig=charset.alias; \
 	      sed -f ref-add.sed $$orig > $$temp; \
@@ -170,14 +165,14 @@
 	    fi; \
 	  fi; \
 	  $(mkinstalldirs) $(DESTDIR)$(localedir); \
-	  test -f $(DESTDIR)$(localedir)/locale.alias \
-	    && orig=$(DESTDIR)$(localedir)/locale.alias \
-	    || orig=$(srcdir)/locale.alias; \
-	  temp=$(DESTDIR)$(localedir)/t-locale.alias; \
-	  dest=$(DESTDIR)$(localedir)/locale.alias; \
-	  sed -f ref-add.sed $$orig > $$temp; \
-	  $(INSTALL_DATA) $$temp $$dest; \
-	  rm -f $$temp; \
+	  if ! test -f $(localedir)/locale.alias; then \
+	    orig=$(srcdir)/locale.alias; \
+	    temp=$(DESTDIR)$(localedir)/t-locale.alias; \
+	    dest=$(DESTDIR)$(localedir)/locale.alias; \
+	    sed -f ref-add.sed $$orig > $$temp; \
+	    $(INSTALL_DATA) $$temp $$dest; \
+	    rm -f $$temp; \
+	  fi; \
 	else \
 	  : ; \
 	fi
