diff --exclude='*~' --exclude='.*' -I '$Id:' -urN bash_completion.orig/bash_completion bash_completion/bash_completion
--- bash_completion.orig/bash_completion	2005-05-18 23:59:24.000000000 -0400
+++ bash_completion/bash_completion	2005-05-19 00:00:21.000000000 -0400
@@ -1531,7 +1531,7 @@
 				for i in ${entries[@]:-}; do
 					files=( ${files[@]/#$i//} )
 				done
-				COMPREPLY=( $( compgen -W '${files[@]}' -- \
+				COMPREPLY=( $( compgen -X '*~' -W '${files[@]}' -- \
 					       $cur ) )
 			fi
 		else
@@ -1584,7 +1584,7 @@
 				COMPREPLY=( $( compgen -W '${changed[@]:-} \
 						   ${newremoved[@]:-}' -- $cur ) )
 			else
-				_filedir
+				COMPREPLY=( $(compgen $default -- "$cur") )
 			fi
 		else
 			COMPREPLY=( $( compgen -W '-n -R -l -f -F -m -r' -- \
