--- rpm-4.0.4/installplatform.uclibc~	Wed Dec  1 18:17:32 2004
+++ rpm-4.0.4/installplatform	Wed Dec  1 18:27:13 2004
@@ -92,6 +92,12 @@
     ;;
   esac
 
+  case $OS in
+  *uclibc)
+    UCLIBCSED='-e s,^@uclibc@,,'
+    ;;
+  esac
+
   cat $PLATFORM \
   | sed -e "s,@RPMRC_OPTFLAGS@,$RPMRC_OPTFLAGS," \
 	-e "s,@RPMRC_ARCH@,$base_arch," \
@@ -102,6 +108,7 @@
 	-e '/FLAGS:-/!s,\${,%{_,' \
 	$MULTILIBSED \
 	$VENDORSED \
+	$UCLIBCSED \
   | grep -v '^@' \
   > ${PPD}/macros
 
--- rpm-4.0.4/platform.in.uclibc~	Wed Dec  1 18:27:24 2004
+++ rpm-4.0.4/platform.in	Wed Dec  1 18:34:19 2004
@@ -310,3 +310,24 @@
 @gentoo@	--mandir=%{_mandir} \\\
 @gentoo@	--infodir=%{_infodir}
 #
+@uclibc@# uClibc specific settings
+@uclibc@#
+@uclibc@# busybox settings
+@uclibc@%__chown_Rhf %{__chown} -Rh
+@uclibc@%__chgrp_Rhf %{__chgrp} -R
+@uclibc@%_fixperms %{__chmod} -R a+rX,g-w,o-w
+@uclibc@
+@uclibc@# overwrite usage of getconf (only in glibc)
+@uclibc@%_smp_mflags		%(echo "-j2")
+@uclibc@#%_vendor		uclibc
+@uclibc@#%_os			linux
+@uclibc@#%_host			%{_arch}-%{_host_os}
+@uclibc@#%_host_os		linux-uclibc
+@uclibc@%_gnu			-uclibc
+@uclibc@#%_target_platform	%{_target_cpu}-%{_vendor}-%{_target_os}%{?_gnu}
+@uclibc@
+@uclibc@%_excludedocs 1
+@uclibc@
+@uclibc@#%_build_name_fmt	%%{ARCH}/%%{NAME}-%%{VERSION}-%%{RELEASE}uc.%%{ARCH}.rpm
+@uclibc@#%_repackage_name_fmt	%%{NAME}-%%{VERSION}-%%{RELEASE}uc.%%{ARCH}.rpm
+@uclibc@
