--- configure.ac
+++ configure.ac
@@ -625,6 +625,9 @@
 WITH_SQLITE3_INCLUDE=
 WITH_SQLITE3_LIB=
 
+withval=yes
+AC_ARG_WITH(sqlite,   [  --with-sqlite           use sqlite database library ])
+if test "$withval" != no; then
 AC_CHECK_HEADER([sqlite3.h], [
   AC_CHECK_LIB(sqlite3, sqlite3_open, [
     AC_DEFINE(HAVE_SQLITE3_H, 1, [Define if you have the <sqlite3.h> header file])
@@ -642,6 +645,7 @@
     DBLIBSRCS="$DBLIBSRCS sqlite.c"
   fi
 ])
+fi
 
 AC_SUBST(WITH_SQLITE3_SUBDIR)
 AC_SUBST(WITH_SQLITE3_INCLUDE)
