--- rpmio/url.c.orig	2007-01-20 16:29:35.471962977 +0100
+++ rpmio/url.c	2007-01-20 16:30:34.438967191 +0100
@@ -28,6 +28,8 @@
 #define	IPPORT_PGPKEYSERVER	11371
 #endif

+#include <neon/ne_session.h>
+
 /**
  */
 /*@unchecked@*/

--- lib/rpmrc.c.orig	2007-01-20 16:40:21.922342919 +0100
+++ lib/rpmrc.c	2007-01-20 16:42:11.185674112 +0100
@@ -13,7 +13,7 @@
 #else
 #define __power_pc() 0
 #endif
-
+#include <rpmio_internal.h>
 #include <rpmcli.h>
 #include <rpmmacro.h>
 #include <rpmlua.h>

--- lib/rpmfi.c.orig	2007-01-20 18:11:30.186124993 +0100
+++ lib/rpmfi.c	2007-01-20 18:13:26.938313715 +0100
@@ -24,6 +24,7 @@

 #include "misc.h"	/* XXX stripTrailingChar */
 #include "rpmmacro.h"	/* XXX rpmCleanPath */
+#include "legacy.h" /* XXX dodigest */

 #include "debug.h"
