diff -ur rpm-4.4.7.orig/configure.ac rpm-4.4.7/configure.ac
--- rpm-4.4.7.orig/configure.ac	2010-04-06 22:43:07.000000000 +0300
+++ rpm-4.4.7/configure.ac	2010-04-07 20:53:40.000000000 +0300
@@ -1120,6 +1120,7 @@
   LIBS="$LIBS -lefence"
 fi
 
+AC_CHECK_HEADER([locale.h], [AC_DEFINE(HAVE_LOCALE_H, 1, [locale.h isn't properly included without this])])
 AC_CHECK_FUNCS(setlocale)
 
 dnl XXX Solaris <= 2.6 only permits 8 chars in password.
