--- curl-7.16.2/libcurl.pc.in.orig	2007-04-14 10:11:14.000000000 +1000
+++ curl-7.16.2/libcurl.pc.in	2007-04-14 10:11:26.000000000 +1000
@@ -33,6 +33,6 @@
 URL: http://curl.haxx.se/
 Description: Library to transfer files with ftp, http, etc.
 Version: @VERSION@
-Libs: -L${libdir} -lcurl @LDFLAGS@ @LIBS@
+Libs: -L${libdir} -lcurl @LIBS@
 Libs.private: @LIBCURL_LIBS@ @LIBS@
 Cflags: -I${includedir}
--- curl-7.16.3/curl-config.in.orig	2007-06-28 22:22:51.000000000 +1000
+++ curl-7.16.3/curl-config.in	2007-06-28 22:23:10.000000000 +1000
@@ -187,9 +187,9 @@
 	   CURLLIBDIR=""
 	fi
 	if test "X@REQUIRE_LIB_DEPS@" = "Xyes"; then
-	  echo ${CURLLIBDIR}-lcurl @LDFLAGS@ @LIBCURL_LIBS@ @LIBS@
+	  echo ${CURLLIBDIR}-lcurl @LIBCURL_LIBS@ @LIBS@
 	else
-	  echo ${CURLLIBDIR}-lcurl @LDFLAGS@ @LIBS@
+	  echo ${CURLLIBDIR}-lcurl @LIBS@
 	fi
 	;;
 
