diff -urNp libksba-1.0.2.org/src/ber-decoder.c libksba-1.0.2/src/ber-decoder.c
--- libksba-1.0.2.org/src/ber-decoder.c	2007-07-04 14:27:14.000000000 +0300
+++ libksba-1.0.2/src/ber-decoder.c	2007-11-27 09:00:58.000000000 +0200
@@ -766,7 +766,7 @@ static gpg_error_t
 decoder_next (BerDecoder d)
 {
   struct tag_info ti;
-  AsnNode node;
+  AsnNode node = NULL;
   gpg_error_t err;
   DECODER_STATE ds = d->ds;
   int debug = d->debug;
