--- build/v2/tools/darwin.jam.orig	2009-11-10 18:26:59 +0100
+++ build/v2/tools/darwin.jam	2009-12-08 18:24:26 +0100
@@ -262,14 +262,14 @@
                 }
             }
             result += $(sdk-version) ;
-            flags darwin.compile OPTIONS $(condition)/<macosx-version>$(sdk-version)
-                : -isysroot $(sdk) ;
-            flags darwin.link OPTIONS $(condition)/<macosx-version>$(sdk-version)
-                : -isysroot $(sdk) ;
-            if $(.debug-configuration)
-            {
-                ECHO notice: available sdk :: $(condition)/<macosx-version>$(sdk-version) :: $(sdk) ;
-            }
+#            flags darwin.compile OPTIONS $(condition)/<macosx-version>$(sdk-version)
+#                : -isysroot $(sdk) ;
+#            flags darwin.link OPTIONS $(condition)/<macosx-version>$(sdk-version)
+#                : -isysroot $(sdk) ;
+#            if $(.debug-configuration)
+#            {
+#                ECHO notice: available sdk :: $(condition)/<macosx-version>$(sdk-version) :: $(sdk) ;
+#            }
         }
     }
     return $(result) ;
@@ -339,7 +339,7 @@
             support-ppc64 = ;
         }
     }
-    switch $(arch)
+    switch $(XXXdisabledXXXarch)
     {
         case combined : 
         {
@@ -443,14 +443,14 @@
     : -nodefaultlibs -shared-libgcc -lstdc++-static -lgcc_eh -lgcc -lSystem ;
 
 # Strip as much as possible when optimizing.
-flags darwin.link OPTIONS <optimization>speed : -Wl,-dead_strip -no_dead_strip_inits_and_terms ;
-flags darwin.link OPTIONS <optimization>space : -Wl,-dead_strip -no_dead_strip_inits_and_terms ;
+#flags darwin.link OPTIONS <optimization>speed : -Wl,-dead_strip -no_dead_strip_inits_and_terms ;
+#flags darwin.link OPTIONS <optimization>space : -Wl,-dead_strip -no_dead_strip_inits_and_terms ;
 
 # Dynamic/shared linking.
 flags darwin.compile OPTIONS <link>shared : -dynamic ;
 
 # Misc options.
-flags darwin.compile OPTIONS : -no-cpp-precomp -gdwarf-2 ;
+#flags darwin.compile OPTIONS : -no-cpp-precomp -gdwarf-2 ;
 
 # Add the framework names to use.
 flags darwin.link FRAMEWORK <framework> ;
