diff -ru xml-batik/sources/org/apache/batik/ext/awt/image/rendered/LRUCache.java xml-batik-patched/sources/org/apache/batik/ext/awt/image/rendered/LRUCache.java
--- xml-batik/sources/org/apache/batik/ext/awt/image/rendered/LRUCache.java	2004-08-18 03:14:08.000000000 -0400
+++ xml-batik-patched/sources/org/apache/batik/ext/awt/image/rendered/LRUCache.java	2005-08-09 19:18:20.000000000 -0400
@@ -33,12 +33,12 @@
          *            should be remembered so it can be returned by
          *            <tt>lruGet</tt>.  
          */
-		public  void    lruSet(LRUNode nde);
+		public  void    lruSet(LRUCache.LRUNode nde);
         /**
          * Called to get the LRU node for this object.  Should return the
          * node passed in to lruSet.
          */
-		public  LRUNode lruGet();
+		public  LRUCache.LRUNode lruGet();
         /**
          * Called to inform the object that it is no longer in the cache.
          */
diff -ru xml-batik/sources/org/apache/batik/script/rhino/RhinoInterpreter.java xml-batik-patched/sources/org/apache/batik/script/rhino/RhinoInterpreter.java
--- xml-batik/sources/org/apache/batik/script/rhino/RhinoInterpreter.java	2005-03-29 05:48:02.000000000 -0500
+++ xml-batik-patched/sources/org/apache/batik/script/rhino/RhinoInterpreter.java	2005-08-09 19:16:33.000000000 -0400
@@ -418,7 +418,7 @@
         public Object get(ScriptableObject so) {
             Context ctx = Context.getCurrentContext();
             if (ctx == null ) return null;
-            return ((ExtendedContext)ctx).getInterpreter().objects.get(name);
+            return ((RhinoInterpreter.ExtendedContext)ctx).getInterpreter().objects.get(name);
         }
     }
     // The method to use for getting the value from the
