diff -ru ImageMagick-6.4.3.orig/Makefile.in ImageMagick-6.4.3/Makefile.in
--- ImageMagick-6.4.3.orig/Makefile.in	2008-09-21 13:29:23.000000000 +0200
+++ ImageMagick-6.4.3/Makefile.in	2008-09-21 13:30:39.000000000 +0200
@@ -980,8 +980,8 @@
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(coders_sun_la_LDFLAGS) $(LDFLAGS) -o $@
 @WITH_MODULES_TRUE@am_coders_sun_la_rpath = -rpath $(codersdir)
-coders_svg_la_DEPENDENCIES = $(MAGICKCORE_LIBS) $(am__DEPENDENCIES_1) \
-	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
+coders_svg_la_DEPENDENCIES = $(MAGICKCORE_LIBS) $(MAGICKWAND_LIBS) \
+	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 am_coders_svg_la_OBJECTS = coders/coders_svg_la-svg.lo
 coders_svg_la_OBJECTS = $(am_coders_svg_la_OBJECTS)
