diff -ur ImageMagick-6.3.7-orig/configure ImageMagick-6.3.7/configure
--- ImageMagick-6.3.7-orig/configure	2008-01-21 00:51:35.000000000 +0100
+++ ImageMagick-6.3.7/configure	2008-01-21 00:52:02.000000000 +0100
@@ -40179,7 +40179,7 @@
 #
 # Path to ImageMagick documentation files
 DOCUMENTATION_RELATIVE_PATH="${PACKAGE_NAME}-${PACKAGE_VERSION}"
-DOCUMENTATION_PATH="${DATA_DIR}/doc/${DOCUMENTATION_RELATIVE_PATH}"
+DOCUMENTATION_PATH="/usr/share/doc/${PF}"
 DEFINE_DOCUMENTATION_PATH="${DATA_DIR}/doc/${DOCUMENTATION_RELATIVE_PATH}/"
 if test "$native_win32_build" = 'yes'; then
     DEFINE_DOCUMENTATION_PATH=`$WinPathScript "$DEFINE_DOCUMENTATION_PATH" 1`
