--- Python-2.2.3/setup.py	2003-05-22 18:36:54.000000000 +0100
+++ Python-2.2.3.new/setup.py	2003-07-03 19:42:29.000000000 +0100
@@ -429,8 +429,12 @@
         # Berkeley DB 3.x.)
 
         dblib = []
-        if self.compiler.find_library_file(lib_dirs, 'db-3.2'):
-            dblib = ['db-3.2']
+        if self.compiler.find_library_file(lib_dirs, 'db-4.1'):
+            dblib = ['db-4.1']
+        elif self.compiler.find_library_file(lib_dirs, 'db-4.0'):
+            dblib = ['db-4.0']
+        elif self.compiler.find_library_file(lib_dirs, 'db-3.2'):
+            dblib = ['db-3.2']	    
         elif self.compiler.find_library_file(lib_dirs, 'db-3.1'):
             dblib = ['db-3.1']
         elif self.compiler.find_library_file(lib_dirs, 'db3'):
@@ -443,7 +445,7 @@
             dblib = ['db']
 
         db185_incs = find_file('db_185.h', inc_dirs,
-                               ['/usr/include/db3', '/usr/include/db2'])
+                               ['/usr/include/db4.1','/usr/include/db4','/usr/include/db3', '/usr/include/db2'])
         db_inc = find_file('db.h', inc_dirs, ['/usr/include/db1'])
         if db185_incs is not None:
             exts.append( Extension('bsddb', ['bsddbmodule.c'],
