--- Python-3.4.3/Include/pyfpe.h
+++ Python-3.4.3/Include/pyfpe.h
@@ -129,9 +129,9 @@
 #include <signal.h>
 #include <setjmp.h>
 #include <math.h>
-extern jmp_buf PyFPE_jbuf;
-extern int PyFPE_counter;
-extern double PyFPE_dummy(void *);
+PyAPI_DATA(jmp_buf) PyFPE_jbuf;
+PyAPI_DATA(int) PyFPE_counter;
+PyAPI_FUNC(double) PyFPE_dummy(void *);
 
 #define PyFPE_START_PROTECT(err_string, leave_stmt) \
 if (!PyFPE_counter++ && setjmp(PyFPE_jbuf)) { \
--- Python-3.4.3/Python/pyfpe.c
+++ Python-3.4.3/Python/pyfpe.c
@@ -1,5 +1,4 @@
-#include "pyconfig.h"
-#include "pyfpe.h"
+#include "Python.h"
 /* 
  * The signal handler for SIGFPE is actually declared in an external
  * module fpectl, or as preferred by the user.  These variable
