http://bugs.python.org/issue3646

Index: Mac/Makefile.in
===================================================================
--- Mac/Makefile.in	(revision 71404)
+++ Mac/Makefile.in	(working copy)
@@ -19,7 +19,8 @@
 
 # These are normally glimpsed from the previous set
 bindir=$(prefix)/bin
-PYTHONAPPSDIR=/Applications/$(PYTHONFRAMEWORK) $(VERSION)
+PYTHONFRAMEWORKPREFIX=@PYTHONFRAMEWORKPREFIX@
+PYTHONAPPSDIR=$(PYTHONFRAMEWORKPREFIX)/Applications/$(PYTHONFRAMEWORK) $(VERSION)
 APPINSTALLDIR=$(prefix)/Resources/Python.app
 
 # Variables for installing the "normal" unix binaries
@@ -54,6 +55,9 @@
 
 
 install_pythonw: pythonw
+	if [ ! -d "$(DESTDIR)$(prefix)/bin" ]; then  \
+		$(INSTALL) -d -m $(DIRMODE) "$(DESTDIR)$(prefix)/bin" ;\
+	fi
 	$(INSTALL_PROGRAM) $(STRIPFLAG) pythonw "$(DESTDIR)$(prefix)/bin/pythonw$(VERSION)"
 	$(INSTALL_PROGRAM) $(STRIPFLAG) pythonw "$(DESTDIR)$(prefix)/bin/python$(VERSION)"
 	ln -sf python$(VERSION) "$(DESTDIR)$(prefix)/bin/python"
Index: Mac/PythonLauncher/Makefile.in
===================================================================
--- Mac/PythonLauncher/Makefile.in	(revision 71404)
+++ Mac/PythonLauncher/Makefile.in	(working copy)
@@ -21,7 +21,8 @@
 
 BUNDLEBULDER=$(srcdir)/../../Lib/plat-mac/bundlebuilder.py
 
-PYTHONAPPSDIR=/Applications/$(PYTHONFRAMEWORK) $(VERSION)
+PYTHONFRAMEWORKPREFIX=@PYTHONFRAMEWORKPREFIX@
+PYTHONAPPSDIR=$(PYTHONFRAMEWORKPREFIX)/Applications/$(PYTHONFRAMEWORK) $(VERSION)
 OBJECTS=FileSettings.o MyAppDelegate.o MyDocument.o PreferencesWindowController.o doscript.o main.o
 
 all: Python\ Launcher.app
Index: Mac/IDLE/Makefile.in
===================================================================
--- Mac/IDLE/Makefile.in	(revision 71404)
+++ Mac/IDLE/Makefile.in	(working copy)
@@ -22,7 +22,8 @@
 
 BUNDLEBULDER=$(srcdir)/../../Lib/plat-mac/bundlebuilder.py
 
-PYTHONAPPSDIR=/Applications/$(PYTHONFRAMEWORK) $(VERSION)
+PYTHONFRAMEWORKPREFIX=@PYTHONFRAMEWORKPREFIX@
+PYTHONAPPSDIR=$(PYTHONFRAMEWORKPREFIX)/Applications/$(PYTHONFRAMEWORK) $(VERSION)
 
 all: IDLE.app
 
Index: Makefile.pre.in
===================================================================
--- Makefile.pre.in	(revision 71404)
+++ Makefile.pre.in	(working copy)
@@ -1050,6 +1050,9 @@
 # Install a number of symlinks to keep software that expects a normal unix
 # install (which includes python-config) happy.
 frameworkinstallmaclib:
+	if [ ! -d "$(DESTDIR)$(prefix)/lib/python$(VERSION)/config" ]; then  \
+		$(INSTALL) -d -m $(DIRMODE) $(DESTDIR)$(prefix)/lib/python$(VERSION)/config; \
+	fi
 	ln -fs "../../../$(PYTHONFRAMEWORK)" "$(DESTDIR)$(prefix)/lib/python$(VERSION)/config/libpython$(VERSION).a"
 	cd Mac && $(MAKE) installmacsubtree DESTDIR="$(DESTDIR)"
 
