http://bugs.python.org/issue1471934

--- Python-2.7/setup.py
+++ Python-2.7/setup.py
@@ -611,11 +611,17 @@
             missing.append('readline')
 
         # crypt module.
+        libs = []
 
-        if self.compiler.find_library_file(lib_dirs, 'crypt'):
+        if platform in ['sunos5']:
+            # http://bugs.python.org/issue1471934 (64-bits Solaris 8-10)
+            # if normal libcrypt symlink is available, in theory the
+            # general check below could overwrite this hack, however,
+            # python just looks only at the 32-bits libs here :(
+            if self.compiler.find_library_file(lib_dirs, 'crypt_i'):
+                libs = ['crypt_i']
+        elif self.compiler.find_library_file(lib_dirs, 'crypt'):
             libs = ['crypt']
-        else:
-            libs = []
         exts.append( Extension('crypt', ['cryptmodule.c'], libraries=libs) )
 
         # CSV files
