--- Python-2.6.4/setup.py
+++ Python-2.6.4/setup.py
@@ -1053,7 +1053,7 @@
                 missing.append('resource')
 
             # Sun yellow pages. Some systems have the functions in libc.
-            if platform not in ['cygwin', 'atheos', 'qnx6'] and find_file('rpcsvc/yp_prot.h', inc_dirs, []) is not None:
+            if platform not in ['cygwin', 'atheos', 'qnx6'] and not platform.startswith('interix') and find_file('rpcsvc/yp_prot.h', inc_dirs, []) is not None:
                 if (self.compiler.find_library_file(lib_dirs, 'nsl')):
                     libs = ['nsl']
                 else:
