diff -ru Python-2.5.1.orig/Modules/timemodule.c Python-2.5.1/Modules/timemodule.c
--- Python-2.5.1.orig/Modules/timemodule.c	2008-03-31 17:02:30 +0200
+++ Python-2.5.1/Modules/timemodule.c	2008-04-01 14:52:00 +0200
@@ -907,7 +907,13 @@
 	t.tv_sec = (long)secs;
 	t.tv_usec = (long)(frac*1000000.0);
 	Py_BEGIN_ALLOW_THREADS
+#ifdef __INTERIX
+	/* may _not_ select zero fds. this returns immediatley on interix
+	 * with errno=14 -> Bad Address */
+	if (select(1, (fd_set *)0, (fd_set *)0, (fd_set *)0, &t) != 0) {
+#else
 	if (select(0, (fd_set *)0, (fd_set *)0, (fd_set *)0, &t) != 0) {
+#endif
 #ifdef EINTR
 		if (errno != EINTR) {
 #else
