--- setup.py.orig	2007-11-05 22:22:22 +0100
+++ setup.py	2007-11-05 22:27:05 +0100
@@ -252,8 +252,8 @@
     def detect_modules(self):
         global disable_ssl
         # Ensure that /usr/local is always used
-        add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
-        add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
+#        add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
+#        add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
 
         # Add paths specified in the environment variables LDFLAGS and
         # CPPFLAGS for header and library files.
@@ -528,8 +528,8 @@
                                depends = ['socketmodule.h']) )
         # Detect SSL support for the socket module (via _ssl)
         search_for_ssl_incs_in = [
-                              '/usr/local/ssl/include',
-                              '/usr/contrib/ssl/include/'
+#                              '/usr/local/ssl/include',
+#                              '/usr/contrib/ssl/include/'
                              ]
         ssl_incs = find_file('openssl/ssl.h', inc_dirs,
                              search_for_ssl_incs_in
@@ -540,8 +540,8 @@
             if krb5_h:
                 ssl_incs += krb5_h
         ssl_libs = find_library_file(self.compiler, 'ssl',lib_dirs,
-                                     ['/usr/local/ssl/lib',
-                                      '/usr/contrib/ssl/lib/'
+                                     [#'/usr/local/ssl/lib',
+                                      #'/usr/contrib/ssl/lib/'
                                      ] )
 
         if (ssl_incs is not None and
@@ -645,6 +645,7 @@
             db_inc_paths.append('/pkg/db-3.%d/include' % x)
             db_inc_paths.append('/opt/db-3.%d/include' % x)
 
+	db_inc_paths = []
         # Add some common subdirectories for Sleepycat DB to the list,
         # based on the standard include directories. This way DB3/4 gets
         # picked up when it is installed in a non-standard prefix and
@@ -754,11 +755,11 @@
         # We need to find >= sqlite version 3.0.8
         sqlite_incdir = sqlite_libdir = None
         sqlite_inc_paths = [ '/usr/include',
-                             '/usr/include/sqlite',
-                             '/usr/include/sqlite3',
-                             '/usr/local/include',
-                             '/usr/local/include/sqlite',
-                             '/usr/local/include/sqlite3',
+#                             '/usr/include/sqlite',
+#                             '/usr/include/sqlite3',
+#                             '/usr/local/include',
+#                             '/usr/local/include/sqlite',
+#                             '/usr/local/include/sqlite3',
                            ]
         MIN_SQLITE_VERSION_NUMBER = (3, 0, 8)
         MIN_SQLITE_VERSION = ".".join([str(x)
