diff -ru Python-2.6.2.orig/setup.py Python-2.6.2/setup.py
--- Python-2.6.2.orig/setup.py	2009-09-07 15:01:58 +0200
+++ Python-2.6.2/setup.py	2009-09-07 15:02:26 +0200
@@ -1053,7 +1053,7 @@
                 missing.append('resource')
 
             # Sun yellow pages. Some systems have the functions in libc.
-            if platform not in ['cygwin', 'atheos', 'qnx6']:
+            if platform not in ['cygwin', 'atheos', 'qnx6'] and not platform.startswith('interix'):
                 if (self.compiler.find_library_file(lib_dirs, 'nsl')):
                     libs = ['nsl']
                 else:
