--- setup.py.old	2004-02-10 22:22:12.028012888 +0000
+++ setup.py	2004-02-10 22:24:17.751899952 +0000
@@ -497,14 +497,8 @@
                                 '/opt/sfw',
                                 '/sw/lib',
                                 ),
-                    'incdirs': ('/usr/local/BerkeleyDB.4.2/include',
-                                '/usr/local/include/db42',
-                                '/usr/local/BerkeleyDB.4.1/include',
-                                '/usr/local/include/db41',
-                                '/usr/local/BerkeleyDB.4.0/include',
-                                '/usr/local/include/db4',
-                                '/opt/sfw/include/db4',
-                                '/sw/include/db4',
+                    'incdirs': ('/usr/include/db4.2',
+                                '/usr/include/db4.1',
                                 '/usr/include/db4',
                                 )},
             'db3': {'libs': ('db-3.3', 'db-3.2', 'db-3.1', 'db3',),
