http://bugs.python.org/issue6163

--- Lib/distutils/unixccompiler.py.orig	2009-07-16 18:39:42 +0200
+++ Lib/distutils/unixccompiler.py	2009-07-16 18:40:34 +0200
@@ -291,7 +291,9 @@
             # MacOSX's linker doesn't understand the -R flag at all
             return "-L" + dir
         elif sys.platform[:5] == "hp-ux":
-            return "+s -L" + dir
+            if compiler.find("gcc") >= 0 or compiler.find("g++") >= 0:
+              return ["-Wl,+s", "-L" + dir]
+            return ["+s", "-L" + dir]
         elif sys.platform[:7] == "irix646" or sys.platform[:6] == "osf1V5":
             return ["-rpath", dir]
         # This should recognize gcc, g++, gcc-x.y, ccache gcc, etc.
