--- setup.py
+++ setup.py
@@ -1352,11 +1352,9 @@
         else:
             missing.append('ossaudiodev')
 
-        if platform == 'sunos5':
-            # SunOS specific modules
-            exts.append( Extension('sunaudiodev', ['sunaudiodev.c']) )
-        else:
-            missing.append('sunaudiodev')
+        # this fails to compile on OpenSolaris, and frankly I don't
+        # care, who needs this?
+        missing.append('sunaudiodev')
 
         if platform == 'darwin' and ("--disable-toolbox-glue" not in
                 sysconfig.get_config_var("CONFIG_ARGS")):
