--- Python-2.7.2.orig/setup.py	2011-08-18 16:36:28 +0200
+++ Python-2.7.2/setup.py	2011-08-18 16:38:26 +0200
@@ -648,7 +648,7 @@
             # python just looks only at the 32-bits libs here :(
             if self.compiler.find_library_file(lib_dirs, 'crypt_i'):
                 libs = ['crypt_i']
-        elif self.compiler.find_library_file(lib_dirs, 'crypt'):
+        elif self.compiler.find_library_file(lib_dirs, 'crypt') or (platform.startswith('interix')):
             libs = ['crypt']
         exts.append( Extension('crypt', ['cryptmodule.c'], libraries=libs) )
 
