diff -ru Python-2.6.2.orig/setup.py Python-2.6.2/setup.py
--- Python-2.6.2.orig/setup.py	2009-07-09 10:29:00 +0200
+++ Python-2.6.2/setup.py	2009-07-09 10:52:45 +0200
@@ -1250,7 +1250,9 @@
         if sys.maxint == 0x7fffffff:
             # This requires sizeof(int) == sizeof(long) == sizeof(char*)
             dl_inc = find_file('dlfcn.h', [], inc_dirs)
-            if (dl_inc is not None) and (platform not in ['atheos']):
+            # interix note: dlfcn.h is in a compiler private dir, which this
+            # script doesn't seem to see - so overriding manually.
+            if ((dl_inc is not None) and (platform not in ['atheos'])) or (platform.startswith('interix')):
                 exts.append( Extension('dl', ['dlmodule.c']) )
             else:
                 missing.append('dl')
@@ -1317,7 +1319,13 @@
                 HAVE_FD_TRANSFER=1,
                 )
             libraries = []
-
+        elif platform.startswith('interix'):
+            macros = dict(
+                HAVE_SEM_OPEN=1,
+                HAVE_SEM_TIMEDWAIT=1,
+                HAVE_FD_TRANSFER=0,
+                )
+            libraries = []
         else:                                   # Linux and other unices
             macros = dict(
                 HAVE_SEM_OPEN=1,
