--- Lib/site.py	2003-07-18 18:45:33.000000000 +0100
+++ Lib/site.py	2003-09-30 00:03:54.000000000 +0100
@@ -171,6 +171,8 @@
                                      "python" + sys.version[:3],
                                      "site-packages"),
                         os.path.join(prefix, "lib", "site-python")]
+            # add Gentoo Linux specific search path - liquidx@gentoo.org (30/09/2003)
+            sitedirs = [os.path.join(prefix,"lib","portage","pym")] + sitedirs
         else:
             sitedirs = [prefix, os.path.join(prefix, "lib", "site-packages")]
         if sys.platform == 'darwin':
