diff -urpN john-1.7.3.1.orig/src/md5-mmx.S john-1.7.3.1/src/md5-mmx.S
--- john-1.7.3.1.orig/src/md5-mmx.S	2009-06-26 14:03:21.073960144 -0600
+++ john-1.7.3.1/src/md5-mmx.S	2009-06-26 14:01:28.889959256 -0600
@@ -306,3 +306,6 @@ mdfivemmx_noinit:
 	
 	ret
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
diff -urpN john-1.7.3.1.orig/src/sha1-mmx.S john-1.7.3.1/src/sha1-mmx.S
--- john-1.7.3.1.orig/src/sha1-mmx.S	2009-06-26 14:03:21.093949076 -0600
+++ john-1.7.3.1/src/sha1-mmx.S	2009-06-26 14:01:36.706958726 -0600
@@ -558,3 +558,7 @@ endianity:
 	por tmp2, ctxd
 	por tmp4, ctxe
 */
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
