Index: john-1.7.2/src/Makefile
===================================================================
--- john-1.7.2.orig/src/Makefile
+++ john-1.7.2/src/Makefile
@@ -718,23 +718,18 @@ john.com: john.asm
 
 ../run/john.exe: $(JOHN_OBJS)
 	$(LD) $(JOHN_OBJS) -lkernel32 -o ../run/john.exe
-	strip ../run/john.exe
 
 ../run/unshadow.exe: symlink.c
 	$(CC) symlink.c -o ../run/unshadow.exe
-	strip ../run/unshadow.exe
 
 ../run/unafs.exe: symlink.c
 	$(CC) symlink.c -o ../run/unafs.exe
-	strip ../run/unafs.exe
 
 ../run/undrop.exe: symlink.c
 	$(CC) symlink.c -o ../run/undrop.exe
-	strip ../run/undrop.exe
 
 ../run/unique.exe: symlink.c
 	$(CC) symlink.c -o ../run/unique.exe
-	strip ../run/unique.exe
 
 # Inlining the S-boxes produces faster code, as long as they fit in the cache
 # (that is, on RISC with at least 8 KB of L1 code cache).
