diff -urpN john-1.7.3.1.orig/src/alpha.S john-1.7.3.1/src/alpha.S
--- john-1.7.3.1.orig/src/alpha.S	2002-04-10 08:13:25.000000000 -0600
+++ john-1.7.3.1/src/alpha.S	2009-06-26 13:56:10.635952569 -0600
@@ -318,3 +318,7 @@ DES_KS_current:
 .space 128
 
 .comm DES_KS_table, (8 * 128 * 16 * 8), 128
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
diff -urpN john-1.7.3.1.orig/src/sparc.S john-1.7.3.1/src/sparc.S
--- john-1.7.3.1.orig/src/sparc.S	2002-04-10 08:13:25.000000000 -0600
+++ john-1.7.3.1/src/sparc.S	2009-06-26 13:54:31.388950767 -0600
@@ -410,3 +410,7 @@ DES_count:
 .common DES_KS_table, (8 * 128 * 16 * 8), 32
 
 #endif
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
diff -urpN john-1.7.3.1.orig/src/x86-64.S john-1.7.3.1/src/x86-64.S
--- john-1.7.3.1.orig/src/x86-64.S	2008-07-13 15:35:12.000000000 -0600
+++ john-1.7.3.1/src/x86-64.S	2009-06-26 13:54:31.392950649 -0600
@@ -1052,3 +1052,7 @@ DES_bs_crypt_LM_loop:
 	subl $1,rounds
 	jnz DES_bs_crypt_LM_loop
 	ret
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
diff -urpN john-1.7.3.1.orig/src/x86-mmx.S john-1.7.3.1/src/x86-mmx.S
--- john-1.7.3.1.orig/src/x86-mmx.S	2008-07-13 15:45:55.000000000 -0600
+++ john-1.7.3.1/src/x86-mmx.S	2009-06-26 13:54:31.396971552 -0600
@@ -1314,3 +1314,7 @@ DES_bs_crypt_LM_loop:
 	emms
 #endif
 	ret
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
diff -urpN john-1.7.3.1.orig/src/x86.S john-1.7.3.1/src/x86.S
--- john-1.7.3.1.orig/src/x86.S	2008-07-07 12:30:36.000000000 -0600
+++ john-1.7.3.1/src/x86.S	2009-06-26 13:54:31.404961468 -0600
@@ -1366,3 +1366,7 @@ CPU_detect_ret:
 	movzbl %al,%eax
 	popl %ebx
 	ret
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
diff -urpN john-1.7.3.1.orig/src/x86-sse.S john-1.7.3.1/src/x86-sse.S
--- john-1.7.3.1.orig/src/x86-sse.S	2008-07-13 15:44:58.000000000 -0600
+++ john-1.7.3.1/src/x86-sse.S	2009-06-26 13:54:31.399949655 -0600
@@ -1301,3 +1301,7 @@ DES_bs_crypt_LM_loop:
 	jnz DES_bs_crypt_LM_loop
 	popl %esi
 	ret
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
