Index: john-1.7.2/src/alpha.S
===================================================================
--- john-1.7.2.orig/src/alpha.S
+++ john-1.7.2/src/alpha.S
@@ -318,3 +318,7 @@ DES_KS_current:
 .space 128
 
 .comm DES_KS_table, (8 * 128 * 16 * 8), 128
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
Index: john-1.7.2/src/sparc.S
===================================================================
--- john-1.7.2.orig/src/sparc.S
+++ john-1.7.2/src/sparc.S
@@ -410,3 +410,7 @@ DES_count:
 .common DES_KS_table, (8 * 128 * 16 * 8), 32
 
 #endif
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
Index: john-1.7.2/src/x86-64.S
===================================================================
--- john-1.7.2.orig/src/x86-64.S
+++ john-1.7.2/src/x86-64.S
@@ -1040,3 +1040,7 @@ DES_bs_crypt_LM_loop:
 	subl $1,rounds
 	jnz DES_bs_crypt_LM_loop
 	ret
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
Index: john-1.7.2/src/x86-mmx.S
===================================================================
--- john-1.7.2.orig/src/x86-mmx.S
+++ john-1.7.2/src/x86-mmx.S
@@ -1302,3 +1302,7 @@ DES_bs_crypt_LM_loop:
 	emms
 #endif
 	ret
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
Index: john-1.7.2/src/x86-sse.S
===================================================================
--- john-1.7.2.orig/src/x86-sse.S
+++ john-1.7.2/src/x86-sse.S
@@ -1289,3 +1289,7 @@ DES_bs_crypt_LM_loop:
 	jnz DES_bs_crypt_LM_loop
 	popl %esi
 	ret
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
Index: john-1.7.2/src/x86.S
===================================================================
--- john-1.7.2.orig/src/x86.S
+++ john-1.7.2/src/x86.S
@@ -1362,3 +1362,7 @@ CPU_detect_ret:
 	movzbl %al,%eax
 	popl %ebx
 	ret
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
