*** wdiff-0.5-orig/wdiff.c	Sun Nov  6 09:57:23 1994
--- wdiff-0.5/wdiff.c	Thu Sep  2 12:49:07 2004
***************
*** 114,119 ****
--- 114,120 ----
    {"no-common"   , 0, NULL, '3'},
    {"help"        , 0, NULL, 'h'},
    {"ignore-case" , 0, NULL, 'i'},
+   {"avoid-wraps" , 0, NULL, 'n'},
    {"printer"     , 0, NULL, 'p'},
    {"statistics"  , 0, NULL, 's'},
    {"terminal"    , 0, NULL, 't'},
