diff -Naur dwww-1.9.28-orig/Makefile dwww-1.9.28/Makefile
--- dwww-1.9.28-orig/Makefile	2006-01-14 05:06:21.000000000 -0600
+++ dwww-1.9.28/Makefile	2007-02-03 19:45:32.000000000 -0600
@@ -6,25 +6,25 @@
 VERSION	= $(shell dpkg-parsechangelog | sed -ne 's/^Version: *//p')
 
 CC	= gcc
-CFLAGS	= -Wall -Wstrict-prototypes -Wmissing-prototypes -DVERSION='"$(VERSION)"' 
-LDFLAGS	=
+CFLAGS	+= -Wall -Wstrict-prototypes -Wmissing-prototypes -DVERSION='"$(VERSION)"' 
+LDFLAGS	+=
 LIBS	= -lpub
 PERL	= /usr/bin/perl
 
 
-ifeq (,$(findstring nodebug,$(DEB_BUILD_OPTIONS)))
-  CFLAGS += -g
-endif
-
-ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
-  CFLAGS += -O0
-else
-  CFLAGS += -O2
-endif
-
-ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
-  LDFLAGS += -s
-endif
+#ifeq (,$(findstring nodebug,$(DEB_BUILD_OPTIONS)))
+#  CFLAGS += -g
+#endif
+#
+#ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
+#  CFLAGS += -O0
+#else
+#  CFLAGS += -O2
+#endif
+#
+#ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
+#  LDFLAGS += -s
+#endif
 
 
 prefix		= debian/dwww
