diff -ru ufraw-0.14.orig/ufraw_ufraw.c ufraw-0.14/ufraw_ufraw.c
--- ufraw-0.14.orig/ufraw_ufraw.c	2008-10-28 09:50:11.047632499 +0100
+++ ufraw-0.14/ufraw_ufraw.c	2008-10-28 09:57:00.966131877 +0100
@@ -432,8 +432,10 @@
      * Some (all?) versions of Solaris followed a draft POSIX.1c standard
      * where ctime_r took a third length argument.
      */
-    ctime_r(&raw->timestamp, uf->conf->timestampText,
-	    sizeof(uf->conf->timestampText));
+    /*
+     * Solaris 10 witch CFLAG -D_POSIX_PTHREAD_SEMANTICS does not need a third length argument.
+     */
+    ctime_r(&raw->timestamp, uf->conf->timestampText);
 #else
     /* POSIX.1c version of ctime_r() */
     ctime_r(&raw->timestamp, uf->conf->timestampText);
