diff -ru guile-1.8.4.orig/guile-readline/configure.in guile-1.8.4/guile-readline/configure.in
--- guile-1.8.4.orig/guile-readline/configure.in	2008-04-28 09:47:20 +0200
+++ guile-1.8.4/guile-readline/configure.in	2008-04-28 10:13:55 +0200
@@ -1,7 +1,13 @@
 AC_PREREQ(2.50)
 
+dnl  `patsubst' here deletes the newline which "echo" prints.  We can't use
+dnl  "echo -n" since -n is not portable (see autoconf manual "Limitations of
+dnl  Builtins"), in particular on solaris it results in a literal "-n" in
+dnl  the output.
+dnl
 AC_INIT(guile-readline,
-        m4_esyscmd(. ../GUILE-VERSION && echo -n ${GUILE_VERSION}))
+        patsubst(m4_esyscmd(. ../GUILE-VERSION && echo ${GUILE_VERSION}),[
+]))
 AC_CONFIG_AUX_DIR([.])
 AC_CONFIG_SRCDIR(readline.c)
 AM_CONFIG_HEADER([guile-readline-config.h])
