--- libguile/Makefile.am	2006/11/18 18:14:55	1.209
+++ libguile/Makefile.am	2007/01/15 23:28:25	1.210
@@ -285,7 +285,7 @@
 .c.x:
 	./guile-snarf -o $@ $< $(snarfcppopts)
 .c.doc:
-	-(test -n "${AWK+set}" || AWK="@AWK@"; ${AWK} -f ./guile-func-name-check $<)
+	-$(AWK) -f ./guile-func-name-check $<
 	(./guile-snarf-docs $(snarfcppopts) $< | \
 	./guile_filter_doc_snarfage$(EXEEXT) --filter-snarfage) > $@ || { rm $@; false; }
 
--- test-suite/standalone/Makefile.am	2006/02/03 23:38:51	1.13
+++ test-suite/standalone/Makefile.am	2006/12/02 23:09:03	1.13.2.2
@@ -29,7 +29,7 @@
 
 TESTS_ENVIRONMENT = "${top_builddir}/pre-inst-guile-env"
 
-test_cflags := \
+test_cflags = \
   -I$(top_srcdir)/test-suite/standalone \
   -I$(top_srcdir) \
   -I$(top_srcdir)/libguile-ltdl $(EXTRA_DEFS) $(GUILE_CFLAGS)
@@ -38,7 +38,8 @@
 
 snarfcppopts = \
   $(DEFS) $(DEFAULT_INCLUDES) $(CPPFLAGS) $(CFLAGS) -I$(top_srcdir)
-%.x: %.c
+SUFFIXES = .x
+.c.x:
 	${top_builddir}/libguile/guile-snarf -o $@ $< $(snarfcppopts)
 
 CLEANFILES = *.x
--- configure.in	2006/11/29 23:30:43	1.268.2.22
+++ configure.in	2006/12/02 23:53:16	1.268.2.23
@@ -68,6 +68,8 @@
 AC_MINIX
 
 AM_PROG_CC_STDC
+# for per-target cflags in the libguile subdir
+AM_PROG_CC_C_O
 
 AC_LIBTOOL_DLOPEN
 AC_PROG_LIBTOOL
