diff -ru -x '*.Po' -x '*.Plo' guile-1.8.4.orig/libguile/iselect.h guile-1.8.4/libguile/iselect.h
--- guile-1.8.4.orig/libguile/iselect.h	2008-04-15 08:13:29 +0200
+++ guile-1.8.4/libguile/iselect.h	2008-04-15 08:47:44 +0200
@@ -38,7 +38,16 @@
 #ifdef FD_SET
 
 #define SELECT_TYPE fd_set
+#ifdef __INTERIX
+// interix' FD_SETSIZE should be == OPEN_MAX, but it seems
+// that there is an error in the headers, which makes it
+// four times as big, and thus causes errno 22 on select
+// (invalid argument).
+#include <limits.h>
+#define SELECT_SET_SIZE OPEN_MAX
+#else
 #define SELECT_SET_SIZE FD_SETSIZE
+#endif
 
 #else /* no FD_SET */
 
diff -ru -x '*.Po' -x '*.Plo' guile-1.8.4.orig/test-suite/standalone/test-conversion.c guile-1.8.4/test-suite/standalone/test-conversion.c
--- guile-1.8.4.orig/test-suite/standalone/test-conversion.c	2008-04-15 08:13:46 +0200
+++ guile-1.8.4/test-suite/standalone/test-conversion.c	2008-04-15 08:20:37 +0200
@@ -25,7 +25,8 @@
 
 #ifdef HAVE_INTTYPES_H
 # include <inttypes.h>
-#elif (!defined PRIiMAX)
+#endif
+#if !defined(PRIiMAX) && !defined(PRIuMAX)
 # if (defined SIZEOF_LONG_LONG) && (SIZEOF_LONG_LONG >= 8)
 #  define PRIiMAX "lli"
 #  define PRIuMAX "llu"
