--- configure.in.orig	2009-10-12 11:33:29 +0200
+++ configure.in	2009-10-12 11:33:46 +0200
@@ -363,7 +363,7 @@
 
 AC_CHECK_FUNCS(asprintf)
 
-AC_CHECK_FUNCS(setenv)
+AC_CHECK_FUNCS(setenv strtoll strtoull strtoq strtouq __strtoll __strtoull)
 
 dnl uclib doesn't have pselect
 AC_CHECK_FUNCS(pselect)
--- configure.orig	2009-10-12 11:33:33 +0200
+++ configure	2009-10-12 11:33:46 +0200
@@ -6087,7 +6087,7 @@
 
 
 
-for ac_func in setenv
+for ac_func in setenv strtoll strtoull strtoq strtouq __strtoll __strtoull
 do
 as_ac_var=`echo "ac_cv_func_$ac_func" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for $ac_func" >&5
--- xmlrpc_config.h.in.orig	2009-10-12 11:39:33 +0200
+++ xmlrpc_config.h.in	2009-10-12 11:47:26 +0200
@@ -159,4 +159,17 @@
 #pragma warning(disable:4996)
 #endif
 
+#if defined(HAVE_STRTOLL)
+#elif defined(HAVE_STRTOQ)
+# define strtoll strtoq /* Interix */
+#elif defined(HAVE___STRTOLL)
+# define strtoll __strtoll /* HP-UX <= 11.11 */
+#endif
+#if defined(HAVE_STRTOULL)
+#elif defined(HAVE_STRTOUQ)
+# define strtoull strtouq /* Interix */
+#elif defined(HAVE___STRTOULL)
+# define strtoull __strtoull /* HP-UX <= 11.11 */
+#endif
+
 #endif
--- xmlrpc_amconfig.h.in.orig	2009-10-12 14:12:39 +0200
+++ xmlrpc_amconfig.h.in	2009-10-12 14:12:03 +0200
@@ -22,6 +22,12 @@
 #undef HAVE_STRCASECMP
 #undef HAVE_STRICMP
 #undef HAVE__STRICMP
+#undef HAVE_STRTOLL
+#undef HAVE_STRTOQ
+#undef HAVE___STRTOLL
+#undef HAVE_STRTOULL
+#undef HAVE_STRTOUQ
+#undef HAVE___STRTOULL
 
 
 /* Define if you have the socket library (-lsocket).  */
