HP-UX 11.11 does not have <sys/select.h>

--- configure.in.orig	2009-10-12 09:59:10 +0200
+++ configure.in	2009-10-12 09:59:26 +0200
@@ -288,7 +288,7 @@
 
 # Needed by Abyss on Solaris:
 
-AC_CHECK_HEADERS(sys/filio.h)
+AC_CHECK_HEADERS(sys/filio.h sys/select.h)
 if test x"$ac_cv_header_sys_filio_h" = xyes; then
   HAVE_SYS_FILIO_H_DEFINE=1
 else
--- configure.orig	2009-10-12 09:59:13 +0200
+++ configure	2009-10-12 09:59:36 +0200
@@ -5017,7 +5017,7 @@
 # Needed by Abyss on Solaris:
 
 
-for ac_header in sys/filio.h
+for ac_header in sys/filio.h sys/select.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if eval "test \"\${$as_ac_Header+set}\" = set"; then
--- ./lib/libutil/select.c.orig	2009-10-12 10:01:45 +0200
+++ ./lib/libutil/select.c	2009-10-12 09:52:39 +0200
@@ -8,7 +8,9 @@
    in this order appears to work on all.
 */
 #include <sys/time.h>
+#ifdef HAVE_SYS_SELECT_H
 #include <sys/select.h>
+#endif
 #endif 
 #include <signal.h>
 
--- ./lib/curl_transport/curlmulti.c.orig	2009-10-12 10:01:51 +0200
+++ ./lib/curl_transport/curlmulti.c	2009-10-12 09:55:03 +0200
@@ -15,8 +15,10 @@
 
 #include <stdlib.h>
 #if !MSVCRT
+#ifdef HAVE_SYS_SELECT_H
 #include <sys/select.h>
 #endif
+#endif
 
 #include <curl/curl.h>
 #include <curl/types.h>
--- ./lib/curl_transport/xmlrpc_curl_transport.c.orig	2009-10-12 10:01:55 +0200
+++ ./lib/curl_transport/xmlrpc_curl_transport.c	2009-10-12 09:54:34 +0200
@@ -62,8 +62,10 @@
 #include <assert.h>
 #include <limits.h>
 #if !MSVCRT
+#ifdef HAVE_SYS_SELECT_H
 #include <sys/select.h>
 #endif
+#endif
 #include <signal.h>
 
 #ifdef WIN32
--- ./include/xmlrpc-c/select_int.h.orig	2009-10-12 10:01:58 +0200
+++ ./include/xmlrpc-c/select_int.h	2009-10-12 09:53:20 +0200
@@ -2,8 +2,10 @@
 #define SELECT_INT_H_INCLUDED
 
 #ifndef WIN32
+#ifdef HAVE_SYS_SELECT_H
 #include <sys/select.h>
 #endif
+#endif
 #include <signal.h>
 
 #include "xmlrpc-c/time_int.h"
--- xmlrpc_amconfig.h.in.orig	2009-10-12 14:12:39 +0200
+++ xmlrpc_amconfig.h.in	2009-10-12 15:06:14 +0200
@@ -11,6 +11,7 @@
    configure.in
 */
 
+#undef HAVE_SYS_SELECT_H
 #undef HAVE_SETGROUPS
 #undef HAVE_ASPRINTF
 #undef HAVE_SETENV
