diff -ru xmlrpc-c-1.18.02.orig/include/xmlrpc-c/inttypes.h xmlrpc-c-1.18.02/include/xmlrpc-c/inttypes.h
--- xmlrpc-c-1.18.02.orig/include/xmlrpc-c/inttypes.h	2009-09-15 11:01:45 +0200
+++ xmlrpc-c-1.18.02/include/xmlrpc-c/inttypes.h	2009-09-15 11:00:25 +0200
@@ -9,6 +9,9 @@
 
 #else
 #include <inttypes.h>
+#ifdef __INTERIX
+#  include <stdint.h>
+#endif
 
 typedef uint16_t xmlrpc_uint16_t;
 typedef uint32_t xmlrpc_uint32_t;
diff -ru xmlrpc-c-1.18.02.orig/lib/util/include/int.h xmlrpc-c-1.18.02/lib/util/include/int.h
--- xmlrpc-c-1.18.02.orig/lib/util/include/int.h	2009-09-15 11:01:45 +0200
+++ xmlrpc-c-1.18.02/lib/util/include/int.h	2009-09-15 11:00:25 +0200
@@ -49,6 +49,11 @@
 #define LL(x) x ## i64
 #define ULL(x) x ## u64
 
+#elif defined(__INTERIX)
+#  include <stdint.h>
+#  define PRId64 "I64d"
+#  define PRIu64 "I64u"
+
 #else
   /* Not Microsoft compiler */
   #include <inttypes.h>
diff -ru xmlrpc-c-1.18.02.orig/config.mk.in xmlrpc-c-1.18.02/config.mk.in
--- xmlrpc-c-1.18.02.orig/config.mk.in	2009-10-06 15:47:20 +0200
+++ xmlrpc-c-1.18.02/config.mk.in	2009-10-06 15:49:13 +0200
@@ -259,6 +259,18 @@
   NEED_WL_RPATH=yes
 endif
 
+ifeq ($(findstring interix,$(HOST_OS)),interix)
+  SHARED_LIB_TYPE = unix
+  SHLIB_SUFFIX = so
+  MUST_BUILD_SHLIB = Y
+  MUST_BUILD_SHLIBLE = Y
+  shlibfn = $(1:%=%.$(SHLIB_SUFFIX).$(MAJ).$(MIN))
+  shliblefn = $(1:%=%.$(SHLIB_SUFFIX))
+  CFLAGS_SHLIB =
+  LDFLAGS_SHLIB = -shared -Wl,-soname,$(SONAME) $(SHLIB_CLIB)
+  NEED_WL_RPATH=yes
+endif
+
 ifeq ($(HOST_OS),dragonfly)
   SHARED_LIB_TYPE = unix
   MUST_BUILD_SHLIB = Y
