diff -ru xmlrpc-c-1.18.02.orig/common.mk xmlrpc-c-1.18.02/common.mk
--- xmlrpc-c-1.18.02.orig/common.mk	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/common.mk	2009-05-19 10:41:55 +0200
@@ -72,17 +72,27 @@
 # To use this rule, the including make file must set a target-specific
 # variable LIBOBJECTS (and declare dependencies that include LIBOBJECTS).
 # Example:
-#   FOO_OBJECTS = foo1.o foo2.o
-#   libfoo.a: LIBOBJECTS = $(FOO_OBJECTS)
-#   libfoo.a: $(FOO_OBJECTS)
+#   FOO_OBJECTS = foo1.$(OEXT) foo2.$(OEXT)
+#   libfoo.$(LIBEXT): LIBOBJECTS = $(FOO_OBJECTS)
+#   libfoo.$(LIBEXT): $(FOO_OBJECTS)
 #   TARGET_LIBRARY_NAMES = libfoo
 
 TARGET_STATIC_LIBRARIES = \
-  $(TARGET_LIBRARY_NAMES:%=%.a) $(TARGET_LIB_NAMES_PP:%=%.a)
+  $(TARGET_LIBRARY_NAMES:%=%.$(LIBEXT)) $(TARGET_LIB_NAMES_PP:%=%.$(LIBEXT))
+
+ifeq ($(USE_LIBTOOL),)
 $(TARGET_STATIC_LIBRARIES):
 	-rm -f $@
 	$(AR) cru $@ $(LIBOBJECTS)
 	$(RANLIB) $@
+else
+LIBTOOL_NO_UNDEFINED = -no-undefined
+$(TARGET_LIBRARY_NAMES:%=%.$(LIBEXT)):
+	$(CCLD) -o $@ $(LIBOBJECTS) $(LIBTOOL_NO_UNDEFINED) -version-number $(MAJ):$(MIN) $(LIBDEP) $(LDFLAGS_SHLIB) -rpath $(LIBINST_DIR)
+
+$(TARGET_LIB_NAMES_PP:%=%.$(LIBEXT)):
+	$(CXXLD) -o $@ $(LIBOBJECTS) $(LIBTOOL_NO_UNDEFINED) -version-number $(MAJ):$(MIN) $(LIBDEP) $(LDFLAGS_SHLIB) -rpath $(LIBINST_DIR)
+endif
 
 
 ##############################################################################
@@ -147,7 +157,7 @@
 ifneq ($(OMIT_LIBXMLRPC_UTIL_RULE),Y)
 LIBXMLRPC_UTIL           = \
   $(call shliblefn, $(LIBXMLRPC_UTIL_DIR)/libxmlrpc_util)
-LIBXMLRPC_UTIL_A         = $(LIBXMLRPC_UTIL_DIR)/libxmlrpc_util.a
+LIBXMLRPC_UTIL_A         = $(LIBXMLRPC_UTIL_DIR)/libxmlrpc_util.$(LIBEXT)
 endif
 
 ifneq ($(OMIT_XMLRPC_LIB_RULE),Y)
@@ -163,11 +173,11 @@
 LIBXMLRPC_SERVER_CGI   = \
   $(call shliblefn, $(BLDDIR)/src/libxmlrpc_server_cgi)
 
-LIBXMLRPC_A              = $(BLDDIR)/src/libxmlrpc.a
-LIBXMLRPC_CLIENT_A       = $(BLDDIR)/src/libxmlrpc_client.a
-LIBXMLRPC_SERVER_A       = $(BLDDIR)/src/libxmlrpc_server.a
-LIBXMLRPC_SERVER_ABYSS_A = $(BLDDIR)/src/libxmlrpc_server_abyss.a
-LIBXMLRPC_SERVER_CGI_A   = $(BLDDIR)/src/libxmlrpc_server_cgi.a
+LIBXMLRPC_A              = $(BLDDIR)/src/libxmlrpc.$(LIBEXT)
+LIBXMLRPC_CLIENT_A       = $(BLDDIR)/src/libxmlrpc_client.$(LIBEXT)
+LIBXMLRPC_SERVER_A       = $(BLDDIR)/src/libxmlrpc_server.$(LIBEXT)
+LIBXMLRPC_SERVER_ABYSS_A = $(BLDDIR)/src/libxmlrpc_server_abyss.$(LIBEXT)
+LIBXMLRPC_SERVER_CGI_A   = $(BLDDIR)/src/libxmlrpc_server_cgi.$(LIBEXT)
 
 endif
 
@@ -176,7 +186,7 @@
 ifneq ($(OMIT_XMLTOK_LIB_RULE),Y)
 LIBXMLRPC_XMLTOK         = \
   $(call shliblefn, $(LIBXMLRPC_XMLTOK_DIR)/libxmlrpc_xmltok)
-LIBXMLRPC_XMLTOK_A       = $(LIBXMLRPC_XMLTOK_DIR)/libxmlrpc_xmltok.a
+LIBXMLRPC_XMLTOK_A       = $(LIBXMLRPC_XMLTOK_DIR)/libxmlrpc_xmltok.$(LIBEXT)
 endif
 
 LIBXMLRPC_XMLPARSE_DIR = $(BLDDIR)/lib/expat/xmlparse
@@ -184,7 +194,7 @@
 ifneq ($(OMIT_XMLPARSE_LIB_RULE),Y)
 LIBXMLRPC_XMLPARSE       = \
   $(call shliblefn, $(LIBXMLRPC_XMLPARSE_DIR)/libxmlrpc_xmlparse)
-LIBXMLRPC_XMLPARSE_A     = $(LIBXMLRPC_XMLPARSE_DIR)/libxmlrpc_xmlparse.a
+LIBXMLRPC_XMLPARSE_A     = $(LIBXMLRPC_XMLPARSE_DIR)/libxmlrpc_xmlparse.$(LIBEXT)
 endif
 
 LIBXMLRPC_ABYSS_DIR = $(BLDDIR)/lib/abyss/src
@@ -192,30 +202,30 @@
 ifneq ($(OMIT_ABYSS_LIB_RULE),Y)
 LIBXMLRPC_ABYSS          = \
   $(call shliblefn, $(LIBXMLRPC_ABYSS_DIR)/libxmlrpc_abyss)
-LIBXMLRPC_ABYSS_A        = $(LIBXMLRPC_ABYSS_DIR)/libxmlrpc_abyss.a
+LIBXMLRPC_ABYSS_A        = $(LIBXMLRPC_ABYSS_DIR)/libxmlrpc_abyss.$(LIBEXT)
 endif
 
 LIBXMLRPC_CPP              = \
   $(call shliblefn, $(BLDDIR)/src/cpp/libxmlrpc_cpp)
-LIBXMLRPC_CPP_A            = $(BLDDIR)/src/cpp/libxmlrpc_cpp.a
+LIBXMLRPC_CPP_A            = $(BLDDIR)/src/cpp/libxmlrpc_cpp.$(LIBEXT)
 LIBXMLRPCPP                = \
   $(call shliblefn, $(BLDDIR)/src/cpp/libxmlrpc++)
-LIBXMLRPCPP_A              = $(BLDDIR)/src/cpp/libxmlrpc++.a
+LIBXMLRPCPP_A              = $(BLDDIR)/src/cpp/libxmlrpc++.$(LIBEXT)
 LIBXMLRPC_PACKETSOCKET     = \
   $(call shliblefn, $(BLDDIR)/src/cpp/libxmlrpc_packetsocket)
-LIBXMLRPC_PACKETSOCKET_A   = $(BLDDIR)/src/cpp/libxmlrpc_packetsocket.a
+LIBXMLRPC_PACKETSOCKET_A   = $(BLDDIR)/src/cpp/libxmlrpc_packetsocket.$(LIBEXT)
 LIBXMLRPC_CLIENTPP         =  \
   $(call shliblefn, $(BLDDIR)/src/cpp/libxmlrpc_client++)
-LIBXMLRPC_CLIENTPP_A       = $(BLDDIR)/src/cpp/libxmlrpc_client++.a
+LIBXMLRPC_CLIENTPP_A       = $(BLDDIR)/src/cpp/libxmlrpc_client++.$(LIBEXT)
 LIBXMLRPC_SERVERPP         = \
   $(call shliblefn, $(BLDDIR)/src/cpp/libxmlrpc_server++)
-LIBXMLRPC_SERVERPP_A       = $(BLDDIR)/src/cpp/libxmlrpc_server++.a
+LIBXMLRPC_SERVERPP_A       = $(BLDDIR)/src/cpp/libxmlrpc_server++.$(LIBEXT)
 LIBXMLRPC_SERVER_ABYSSPP   = \
   $(call shliblefn, $(BLDDIR)/src/cpp/libxmlrpc_server_abyss++)
-LIBXMLRPC_SERVER_ABYSSPP_A = $(BLDDIR)/src/cpp/libxmlrpc_server_abyss++.a
+LIBXMLRPC_SERVER_ABYSSPP_A = $(BLDDIR)/src/cpp/libxmlrpc_server_abyss++.$(LIBEXT)
 LIBXMLRPC_SERVER_PSTREAMPP = \
   $(call shliblefn, $(BLDDIR)/src/cpp/libxmlrpc_server_pstream++)
-LIBXMLRPC_SERVER_PSTREAMPP_A = $(BLDDIR)/src/cpp/libxmlrpc_server_pstream++.a
+LIBXMLRPC_SERVER_PSTREAMPP_A = $(BLDDIR)/src/cpp/libxmlrpc_server_pstream++.$(LIBEXT)
 
 # LIBXMLRPC_XML is the list of Xmlrpc-c libraries we need to parse
 # XML.  If we're using an external library to parse XML, this is null.
@@ -243,12 +253,12 @@
 
 #   TARGET_MODS = foo bar
 #
-#   foo.o foo.osh: INCLUDES = -Iinclude -I/usr/include/foostuff
-#   bar.o bar.osh: INCLUDES = -Iinclude -I/usr/include/barstuff
+#   foo.$(OEXT) foo.osh: INCLUDES = -Iinclude -I/usr/include/foostuff
+#   bar.$(OEXT) bar.osh: INCLUDES = -Iinclude -I/usr/include/barstuff
 #
 #   include common.mk
 #
-# The above generates rules to build foo.o, bar.o, foo.osh, and bar.osh
+# The above generates rules to build foo.$(OEXT), bar.$(OEXT), foo.osh, and bar.osh
 #
 # For C++ source files, use TARGET_MODS_PP instead.
 
@@ -266,7 +276,7 @@
   $(INCLUDES) $(CFLAGS_PERSONAL) $(CADD)
 
 
-$(TARGET_MODS:%=%.o):%.o:%.c
+$(TARGET_MODS:%=%.$(OEXT)):%.$(OEXT):%.c
 	$(CC) -c -o $@ $(CFLAGS_ALL) $<
 
 $(TARGET_MODS:%=%.osh): CFLAGS_COMMON += $(CFLAGS_SHLIB)
@@ -274,7 +284,7 @@
 $(TARGET_MODS:%=%.osh):%.osh:%.c
 	$(CC) -c -o $@ $(INCLUDES) $(CFLAGS_ALL) $(CFLAGS_SHLIB) $<
 
-$(TARGET_MODS_PP:%=%.o):%.o:%.cpp
+$(TARGET_MODS_PP:%=%.$(OEXT)):%.$(OEXT):%.cpp
 	$(CXX) -c -o $@ $(INCLUDES) $(CXXFLAGS_ALL) $<
 
 $(TARGET_MODS_PP:%=%.osh): CXXFLAGS_COMMON += $(CFLAGS_SHLIB)
@@ -328,7 +338,7 @@
 ##############################################################################
 
 ifneq ($(OMIT_WININET_TRANSPORT_RULE),Y)
-$(BLDDIR)/lib/wininet_transport/xmlrpc_wininet_transport.o \
+$(BLDDIR)/lib/wininet_transport/xmlrpc_wininet_transport.$(OEXT) \
 $(BLDDIR)/lib/wininet_transport/xmlrpc_wininet_transport.osh \
 : FORCE
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/wininet_transport/Makefile \
@@ -336,13 +346,13 @@
 endif
 
 ifneq ($(OMIT_CURL_TRANSPORT_RULE),Y)
-$(BLDDIR)/lib/curl_transport/xmlrpc_curl_transport.o \
+$(BLDDIR)/lib/curl_transport/xmlrpc_curl_transport.$(OEXT) \
 $(BLDDIR)/lib/curl_transport/xmlrpc_curl_transport.osh \
-$(BLDDIR)/lib/curl_transport/curltransaction.o \
+$(BLDDIR)/lib/curl_transport/curltransaction.$(OEXT) \
 $(BLDDIR)/lib/curl_transport/curltransaction.osh \
-$(BLDDIR)/lib/curl_transport/curlmulti.o \
+$(BLDDIR)/lib/curl_transport/curlmulti.$(OEXT) \
 $(BLDDIR)/lib/curl_transport/curlmulti.osh \
-$(BLDDIR)/lib/curl_transport/lock_pthread.o \
+$(BLDDIR)/lib/curl_transport/lock_pthread.$(OEXT) \
 $(BLDDIR)/lib/curl_transport/lock_pthread.osh \
 : FORCE
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/curl_transport/Makefile \
@@ -350,7 +360,7 @@
 endif
 
 ifneq ($(OMIT_LIBWWW_TRANSPORT_RULE),Y)
-$(BLDDIR)/lib/libwww_transport/xmlrpc_libwww_transport.o \
+$(BLDDIR)/lib/libwww_transport/xmlrpc_libwww_transport.$(OEXT) \
 $(BLDDIR)/lib/libwww_transport/xmlrpc_libwww_transport.osh \
 : FORCE
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/libwww_transport/Makefile \
@@ -407,12 +417,12 @@
 UTIL_DIR = $(BLDDIR)/lib/util
 
 UTILS = \
-  casprintf.o \
-  cmdline_parser.o \
-  cmdline_parser_cpp.o \
-  getoptx.o \
-  stripcaseeq.o \
-  string_parser.o \
+  casprintf.$(OEXT) \
+  cmdline_parser.$(OEXT) \
+  cmdline_parser_cpp.$(OEXT) \
+  getoptx.$(OEXT) \
+  stripcaseeq.$(OEXT) \
+  string_parser.$(OEXT) \
 
 ifneq ($(OMIT_UTILS_RULE),Y)
 $(UTILS:%=$(UTIL_DIR)/%): FORCE
@@ -420,7 +430,7 @@
 	    $(notdir $@)
 endif
 
-CASPRINTF = $(UTIL_DIR)/casprintf.o
+CASPRINTF = $(UTIL_DIR)/casprintf.$(OEXT)
 
 
 # About version.h:  This is a built header file, which means it is a supreme
@@ -459,7 +469,7 @@
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/GNUmakefile $(notdir $@)
 endif
 
-$(TARGET_MODS:%=%.o) $(TARGET_MODS:%=%.osh): \
+$(TARGET_MODS:%=%.$(OEXT)) $(TARGET_MODS:%=%.osh): \
   $(BLDDIR)/include/xmlrpc-c/config.h
 
 # With a separate build directory, you have to make the directory itself
@@ -480,8 +490,13 @@
   install-static-libraries install-shared-libraries \
   install-headers install-bin install-man
 
+ifeq ($(USE_LIBTOOL),)
 INSTALL_LIB_CMD = $(INSTALL_DATA) $$p $(DESTDIR)$(LIBINST_DIR)/$$p
 RANLIB_CMD = $(RANLIB) $(DESTDIR)$(LIBINST_DIR)/$$p
+else
+INSTALL_LIB_CMD = $(LIBTOOL) --mode=install $(INSTALL_DATA) $$p $(DESTDIR)$(LIBINST_DIR)/$$p
+RANLIB_CMD = :
+endif
 
 install-static-libraries: $(STATIC_LIBRARIES_TO_INSTALL)
 	$(MKINSTALLDIRS) $(DESTDIR)$(LIBINST_DIR)
@@ -590,11 +605,11 @@
 
 DEP_SOURCES = $(wildcard *.c *.cpp)
 
-# This is a filter to turn "foo.o:" rules into "foo.o foo.lo foo.osh:"
+# This is a filter to turn "foo.$(OEXT):" rules into "foo.$(OEXT) foo.lo foo.osh:"
 # to make dependencies for all the various forms of object file out of
-# a file made by a depedency generator that knows only about .o.
+# a file made by a depedency generator that knows only about .$(OEXT).
 
-DEPEND_MASSAGER = perl -walnpe's{^(.*)\.o:}{$$1.o $$1.lo $$1.osh:}'
+DEPEND_MASSAGER = perl -walnpe's{^(.*)\.$(OEXT):}{$$1.o $$1.$(OEXT) $$1.lo $$1.osh:}'
 
 
 
diff -ru xmlrpc-c-1.18.02.orig/config.mk.in xmlrpc-c-1.18.02/config.mk.in
--- xmlrpc-c-1.18.02.orig/config.mk.in	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/config.mk.in	2009-05-19 10:46:06 +0200
@@ -43,16 +43,29 @@
 ABS_SRCDIR = @abs_srcdir@
 PREFIX = @prefix@
 
+USE_LIBTOOL =
 
 HAVE_WCHAR_H_DEFINE = @HAVE_WCHAR_H_DEFINE@
 
 # Stuff 'configure' figured out about our build platform:
 
 SHELL = @SHELL@
+ifeq ($(USE_LIBTOOL),)
 CC = @CC@
 CXX = @CXX@
 CCLD = $(CC)
 CXXLD = $(CXX)
+OEXT = o
+LIBEXT = a
+else
+LIBTOOL = libtool
+CC = $(LIBTOOL) --mode=compile --tag=CC @CC@
+CXX = $(LIBTOOL) --mode=compile --tag=CXX @CXX@
+CCLD = $(LIBTOOL) --mode=link --tag=CC @CC@
+CXXLD = $(LIBTOOL) --mode=link --tag=CXX @CXX@
+OEXT = lo
+LIBEXT = la
+endif
 AR = ar
 RANLIB = ranlib
 LN_S = ln -s
@@ -166,6 +179,8 @@
 shlibfn = $(1:%=%.shlibdummy)
 shliblefn = $(1:%=%.shlibledummy)
 
+ifeq ($(USE_LIBTOOL),)
+
 ifeq ($(HOST_OS),linux-gnu)
   # Assume linker is GNU Compiler (gcc)
   SHARED_LIB_TYPE = unix
@@ -303,6 +318,9 @@
   LDFLAGS_SHLIB = -shared -Wl,-soname,$(SONAME) $(SHLIB_CLIB)
 endif
 
+endif # not USE_LIBTOOL
+
+
 ##############################################################################
 #                     MISCELLANEOUS
 ##############################################################################
@@ -351,7 +369,7 @@
 # VPATH probably doesn't belong in this file, but it's a convenient
 # place to set it once.  VPATH is a special Make variable that tells
 # Make where to look for dependencies.  E.g. if a make file says bar.c
-# is a dependency of bar.o and VPATH is ".:/usr/src/mypkg", Make will
+# is a dependency of bar.$(OEXT) and VPATH is ".:/usr/src/mypkg", Make will
 # look for bar.c first in the current directory (.) (as it would with
 # no VPATH), then in /usr/src/mypkg.  The purpose of this is to allow
 # you to build in a fresh build directory, while your source stays in
diff -ru xmlrpc-c-1.18.02.orig/examples/cpp/Makefile xmlrpc-c-1.18.02/examples/cpp/Makefile
--- xmlrpc-c-1.18.02.orig/examples/cpp/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/examples/cpp/Makefile	2009-05-19 10:41:55 +0200
@@ -74,31 +74,31 @@
 
 all: $(PROGS)
 
-$(SERVERPROGS_CGI):%.cgi:%_cgi.o
+$(SERVERPROGS_CGI):%.cgi:%_cgi.$(OEXT)
 	$(CXXLD) -o $@ $^ $(LIBS_SERVER_CGI) $(LDFLAGS)
 
-$(SERVERPROGS_ABYSS):%:%.o
+$(SERVERPROGS_ABYSS):%:%.$(OEXT)
 	$(CXXLD) -o $@ $^ $(LIBS_SERVER_ABYSS) $(LDFLAGS)
 
-$(CLIENTPROGS):%:%.o
+$(CLIENTPROGS):%:%.$(OEXT)
 	$(CXXLD) -o $@ $^ $(LIBS_CLIENT) $(LDFLAGS)
 
 
 LIBS_PSTREAM_CLIENT = \
   $(shell $(XMLRPC_C_CONFIG) c++2 client --libs)
 
-pstream_client:%:%.o
+pstream_client:%:%.$(OEXT)
 	$(CXXLD) -o $@ $^ $(LIBS_PSTREAM_CLIENT) $(LDFLAGS)
 
 LIBS_PSTREAM_SERVER = \
   $(shell $(XMLRPC_C_CONFIG) c++2 pstream-server --libs)
 
-pstream_inetd_server pstream_serial_server:%:%.o
+pstream_inetd_server pstream_serial_server:%:%.$(OEXT)
 	$(CXXLD) -o $@ $^ $(LIBS_PSTREAM_SERVER) $(LDFLAGS)
 
-OBJECTS = $(patsubst %,%.o,$(patsubst %.cgi,%_cgi,$(PROGS)))
+OBJECTS = $(patsubst %,%.$(OEXT),$(patsubst %.cgi,%_cgi,$(PROGS)))
 
-$(OBJECTS):%.o:%.cpp
+$(OBJECTS):%.$(OEXT):%.cpp
 	$(CXX) -c $(INCLUDES) $(CXXFLAGS) $<
 
 # See example/Makefile for an explanation of config.h and xmlrpc_amconfig.h
@@ -112,7 +112,7 @@
 
 .PHONY: clean
 clean:
-	rm -f $(PROGS) *.o config.h xmlrpc_amconfig.h
+	rm -f $(PROGS) *.$(OEXT) config.h xmlrpc_amconfig.h
 
 .PHONY: distclean
 distclean: clean
diff -ru xmlrpc-c-1.18.02.orig/examples/Makefile xmlrpc-c-1.18.02/examples/Makefile
--- xmlrpc-c-1.18.02.orig/examples/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/examples/Makefile	2009-05-19 10:41:55 +0200
@@ -89,21 +89,21 @@
 $(BLDDIR)/examples/cpp:
 	mkdir $@
 
-$(CLIENTPROGS):%:%.o
+$(CLIENTPROGS):%:%.$(OEXT)
 	$(CCLD) -o $@ $^ $(LIBS_CLIENT) $(LDFLAGS)
 
-$(SERVERPROGS_CGI):%.cgi:%_cgi.o
+$(SERVERPROGS_CGI):%.cgi:%_cgi.$(OEXT)
 	$(CCLD) -o $@ $^ $(LIBS_SERVER_CGI) $(LDFLAGS)
 
-$(SERVERPROGS_ABYSS):%:%.o
+$(SERVERPROGS_ABYSS):%:%.$(OEXT)
 	$(CCLD) -o $@ $^ $(LIBS_SERVER_ABYSS) $(LDFLAGS)
 
-gen_sample_add_xml:%:%.o
+gen_sample_add_xml:%:%.$(OEXT)
 	$(CCLD) -o $@ $^ $(LIBS_BASE) $(LDFLAGS) 
 
-OBJECTS = $(patsubst %,%.o,$(patsubst %.cgi,%_cgi,$(PROGS)))
+OBJECTS = $(patsubst %,%.$(OEXT),$(patsubst %.cgi,%_cgi,$(PROGS)))
 
-$(OBJECTS):%.o:%.c
+$(OBJECTS):%.$(OEXT):%.c
 	$(CC) -c $(INCLUDES) $(CFLAGS) $<
 
 # config.h and xmlrpc_amconfig.h just describe the build environment.
@@ -121,7 +121,7 @@
 
 .PHONY: clean
 clean:
-	rm -f $(PROGS) *.o config.h xmlrpc_amconfig.h
+	rm -f $(PROGS) *.$(OEXT) config.h xmlrpc_amconfig.h
 	$(MAKE) -C cpp clean
 
 .PHONY: distclean
diff -ru xmlrpc-c-1.18.02.orig/lib/abyss/src/Makefile xmlrpc-c-1.18.02/lib/abyss/src/Makefile
--- xmlrpc-c-1.18.02.orig/lib/abyss/src/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/lib/abyss/src/Makefile	2009-05-19 10:41:55 +0200
@@ -13,7 +13,7 @@
 
 TARGET_LIBRARY_NAMES := libxmlrpc_abyss
 
-STATIC_LIBRARIES_TO_INSTALL = libxmlrpc_abyss.a
+STATIC_LIBRARIES_TO_INSTALL = libxmlrpc_abyss.$(LIBEXT)
 
 SHARED_LIBS_TO_BUILD := libxmlrpc_abyss
 SHARED_LIBS_TO_INSTALL := libxmlrpc_abyss
@@ -60,7 +60,7 @@
 #ABYSS_SHLIBLE is e.g. libxmlrpc_abyss.so
 
 .PHONY: all
-all: libxmlrpc_abyss.a $(TARGET_SHARED_LIBRARIES) $(TARGET_SHARED_LE_LIBS)
+all: libxmlrpc_abyss.$(LIBEXT) $(TARGET_SHARED_LIBRARIES) $(TARGET_SHARED_LE_LIBS)
 
 
 # Rule for this is in common.mk, courtesy of TARGET_LIBRARY_NAMES
@@ -74,8 +74,8 @@
 
 # Rule for this is in common.mk, courtesy of TARGET_STATIC_LIBRARIES:
 
-libxmlrpc_abyss.a: $(TARGET_MODS:%=%.o)
-libxmlrpc_abyss.a: LIBOBJECTS = $(TARGET_MODS:%=%.o)
+libxmlrpc_abyss.$(LIBEXT): $(TARGET_MODS:%=%.$(OEXT))
+libxmlrpc_abyss.$(LIBEXT): LIBOBJECTS = $(TARGET_MODS:%=%.$(OEXT))
 
 #-----------------------------------------------------------------------------
 #  RULES TO COMPILE OBJECT MODULES FOR LIBRARIES
@@ -87,7 +87,7 @@
 
 # Need this dependency for those who don't use depend.mk.
 # Without it, version.h doesn't get created.
-response.o response.osh handler.o handler.osh: version.h
+response.$(OEXT) response.osh handler.$(OEXT) handler.osh: version.h
 
 .PHONY: clean
 clean: clean-common
diff -ru xmlrpc-c-1.18.02.orig/lib/curl_transport/Makefile xmlrpc-c-1.18.02/lib/curl_transport/Makefile
--- xmlrpc-c-1.18.02.orig/lib/curl_transport/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/lib/curl_transport/Makefile	2009-05-19 10:41:55 +0200
@@ -13,7 +13,10 @@
 MODS := xmlrpc_curl_transport curltransaction curlmulti lock_pthread
 
 .PHONY: all
-all: $(MODS:%=%.o) $(MODS:%=%.osh)
+all: $(MODS:%=%.$(OEXT))
+ifeq ($USE_LIBTOOL,)
+all: $(MODS:%=%.osh)
+endif
 
 # Rules for the above dependencies are in common.mk,
 # courtesy of TARGET_MODS.
@@ -58,4 +61,4 @@
 
 # Need this dependency for those who don't use depend.mk.
 # Without it, version.h doesn't get created.
-xmlrpc_curl_transport.o xmlrpc_curl_transport.osh: version.h
+xmlrpc_curl_transport.$(OEXT) xmlrpc_curl_transport.osh: version.h
diff -ru xmlrpc-c-1.18.02.orig/lib/expat/gennmtab/Makefile xmlrpc-c-1.18.02/lib/expat/gennmtab/Makefile
--- xmlrpc-c-1.18.02.orig/lib/expat/gennmtab/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/lib/expat/gennmtab/Makefile	2009-05-19 10:41:55 +0200
@@ -39,10 +39,10 @@
 .PHONY: dep
 dep: dep-common
 
-gennmtab.o:%.o:%.c
+gennmtab.$(OEXT):%.$(OEXT):%.c
 	$(BUILDTOOL_CC) -c $< -o $@ $(CFLAGS_ALL) $(INCLUDES)
 
-gennmtab:%:%.o
+gennmtab:%:%.$(OEXT)
 	$(BUILDTOOL_CCLD) -o $@ $(LDFLAGS) $^
 
 include depend.mk
diff -ru xmlrpc-c-1.18.02.orig/lib/expat/xmlparse/Makefile xmlrpc-c-1.18.02/lib/expat/xmlparse/Makefile
--- xmlrpc-c-1.18.02.orig/lib/expat/xmlparse/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/lib/expat/xmlparse/Makefile	2009-05-19 10:41:55 +0200
@@ -26,7 +26,7 @@
 
 TARGET_LIBRARY_NAMES := libxmlrpc_xmlparse
 
-STATIC_LIBRARIES_TO_INSTALL = libxmlrpc_xmlparse.a
+STATIC_LIBRARIES_TO_INSTALL = libxmlrpc_xmlparse.$(LIBEXT)
 
 SHARED_LIBS_TO_BUILD := libxmlrpc_xmlparse
 SHARED_LIBS_TO_INSTALL := libxmlrpc_xmlparse
@@ -56,7 +56,7 @@
 #XMLPARSE_SHLIBLE is e.g. libxmlrpc_xmlparse.so
 
 .PHONY: all
-all: libxmlrpc_xmlparse.a $(TARGET_SHARED_LIBRARIES) $(TARGET_SHARED_LE_LIBS)
+all: libxmlrpc_xmlparse.$(LIBEXT) $(TARGET_SHARED_LIBRARIES) $(TARGET_SHARED_LE_LIBS)
 
 # Rule for this is in common.mk, courtesy of TARGET_LIBRARY_NAMES:
 $(XMLPARSE_SHLIB): $(TARGET_MODS:%=%.osh) $(LIBDEP)
@@ -64,8 +64,8 @@
 
 # Rule for this is in common.mk, courtesy of TARGET_STATIC_LIBRARIES:
 
-libxmlrpc_xmlparse.a: $(TARGET_MODS:%=%.o)
-libxmlrpc_xmlparse.a: LIBOBJECTS = $(TARGET_MODS:%=%.o)
+libxmlrpc_xmlparse.$(LIBEXT): $(TARGET_MODS:%=%.$(OEXT))
+libxmlrpc_xmlparse.$(LIBEXT): LIBOBJECTS = $(TARGET_MODS:%=%.$(OEXT))
 
 #-----------------------------------------------------------------------------
 #  RULES TO COMPILE OBJECT MODULES FOR LIBRARIES
diff -ru xmlrpc-c-1.18.02.orig/lib/expat/xmltok/Makefile xmlrpc-c-1.18.02/lib/expat/xmltok/Makefile
--- xmlrpc-c-1.18.02.orig/lib/expat/xmltok/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/lib/expat/xmltok/Makefile	2009-05-19 10:41:55 +0200
@@ -33,7 +33,7 @@
 
 TARGET_LIBRARY_NAMES := libxmlrpc_xmltok
 
-STATIC_LIBRARIES_TO_INSTALL = libxmlrpc_xmltok.a
+STATIC_LIBRARIES_TO_INSTALL = libxmlrpc_xmltok.$(LIBEXT)
 
 SHARED_LIBS_TO_BUILD := libxmlrpc_xmltok
 SHARED_LIBS_TO_INSTALL := libxmlrpc_xmltok
@@ -52,7 +52,7 @@
 #XMLTOK_SHLIBLE is e.g. libxmlrpc_xmltok.so
 
 .PHONY: all
-all: libxmlrpc_xmltok.a $(TARGET_SHARED_LIBRARIES) $(TARGET_SHARED_LE_LIBS)
+all: libxmlrpc_xmltok.$(LIBEXT) $(TARGET_SHARED_LIBRARIES) $(TARGET_SHARED_LE_LIBS)
 
 # Rule for this is in common.mk, courtesy of TARGET_LIBRARY_NAMES
 $(XMLTOK_SHLIB): $(TARGET_MODS:%=%.osh)
@@ -60,8 +60,8 @@
 
 # Rule for this is in common.mk, courtesy of TARGET_STATIC_LIBRARIES:
 
-libxmlrpc_xmltok.a: $(TARGET_MODS:%=%.o)
-libxmlrpc_xmltok.a: LIBOBJECTS = $(TARGET_MODS:%=%.o)
+libxmlrpc_xmltok.$(LIBEXT): $(TARGET_MODS:%=%.$(OEXT))
+libxmlrpc_xmltok.$(LIBEXT): LIBOBJECTS = $(TARGET_MODS:%=%.$(OEXT))
 
 #-----------------------------------------------------------------------------
 #  RULES TO COMPILE OBJECT MODULES FOR LIBRARIES
@@ -99,6 +99,6 @@
 $(GENNMTAB):
 	$(MAKE) -C $(dir $@) $(notdir $@)
 
-xmltok.o xmltok.osh: nametab.h
+xmltok.$(OEXT) xmltok.osh: nametab.h
 
 include depend.mk
diff -ru xmlrpc-c-1.18.02.orig/lib/libutil/Makefile xmlrpc-c-1.18.02/lib/libutil/Makefile
--- xmlrpc-c-1.18.02.orig/lib/libutil/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/lib/libutil/Makefile	2009-05-19 10:41:55 +0200
@@ -22,7 +22,7 @@
 
 TARGET_LIBRARY_NAMES := libxmlrpc_util
 
-STATIC_LIBRARIES_TO_INSTALL = libxmlrpc_util.a
+STATIC_LIBRARIES_TO_INSTALL = libxmlrpc_util.$(LIBEXT)
 
 SHARED_LIBS_TO_BUILD := libxmlrpc_util
 SHARED_LIBS_TO_INSTALL := libxmlrpc_util
@@ -52,7 +52,7 @@
 #UTIL_SHLIBLE is e.g. libxmlrpc_util.so
 
 .PHONY: all
-all: libxmlrpc_util.a $(TARGET_SHARED_LIBRARIES) $(TARGET_SHARED_LE_LIBS)
+all: libxmlrpc_util.$(LIBEXT) $(TARGET_SHARED_LIBRARIES) $(TARGET_SHARED_LE_LIBS)
 
 # Rule for this is in common.mk, courtesy of TARGET_LIBRARY_NAMES:
 $(UTIL_SHLIB): $(TARGET_MODS:%=%.osh)
@@ -60,8 +60,8 @@
 
 # Rule for this is in common.mk, courtesy of TARGET_LIBRARY_NAMES:
 
-libxmlrpc_util.a: $(TARGET_MODS:%=%.o)
-libxmlrpc_util.a: LIBOBJECTS = $(TARGET_MODS:%=%.o)
+libxmlrpc_util.$(LIBEXT): $(TARGET_MODS:%=%.$(OEXT))
+libxmlrpc_util.$(LIBEXT): LIBOBJECTS = $(TARGET_MODS:%=%.$(OEXT))
 
 #-----------------------------------------------------------------------------
 #  RULES TO COMPILE OBJECT MODULES FOR LIBRARIES
diff -ru xmlrpc-c-1.18.02.orig/lib/libwww_transport/Makefile xmlrpc-c-1.18.02/lib/libwww_transport/Makefile
--- xmlrpc-c-1.18.02.orig/lib/libwww_transport/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/lib/libwww_transport/Makefile	2009-05-19 10:41:55 +0200
@@ -11,7 +11,7 @@
 default: all
 
 .PHONY: all
-all: xmlrpc_libwww_transport.o xmlrpc_libwww_transport.osh
+all: xmlrpc_libwww_transport.$(OEXT) xmlrpc_libwww_transport.osh
 
 # Rules for the above dependencies are in common.mk,
 # courtesy of TARGET_MODS.
@@ -53,4 +53,4 @@
 
 # Need this dependency for those who don't use depend.mk.
 # Without it, version.h doesn't get created.
-xmlrpc_libwww_transport.o xmlrpc_libwww_transport.osh: version.h
+xmlrpc_libwww_transport.$(OEXT) xmlrpc_libwww_transport.osh: version.h
diff -ru xmlrpc-c-1.18.02.orig/lib/util/Makefile xmlrpc-c-1.18.02/lib/util/Makefile
--- xmlrpc-c-1.18.02.orig/lib/util/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/lib/util/Makefile	2009-05-19 10:41:55 +0200
@@ -34,22 +34,22 @@
 include $(SRCDIR)/common.mk
 
 LIBOBJS = \
-  casprintf.o \
-  cmdline_parser.o \
-  cmdline_parser_cpp.o \
-  getoptx.o \
-  string_parser.o \
-  stripcaseeq.o \
+  casprintf.$(OEXT) \
+  cmdline_parser.$(OEXT) \
+  cmdline_parser_cpp.$(OEXT) \
+  getoptx.$(OEXT) \
+  string_parser.$(OEXT) \
+  stripcaseeq.$(OEXT) \
 
 .PHONY: all
 all: $(LIBOBJS)
 
 INCLUDES = -I$(SRCDIR)/$(SUBDIR)/include -I$(BLDDIR)
 
-%.o:%.c
+%.$(OEXT):%.c
 	$(CC) -c $(CFLAGS_ALL) $<
 
-%.o:%.cpp
+%.$(OEXT):%.cpp
 	$(CXX) -c $(CXXFLAGS_ALL) $<
 
 include depend.mk
@@ -62,4 +62,4 @@
 .PHONY: dep
 dep: dep-common
 
-install:
\ No newline at end of file
+install:
diff -ru xmlrpc-c-1.18.02.orig/lib/wininet_transport/Makefile xmlrpc-c-1.18.02/lib/wininet_transport/Makefile
--- xmlrpc-c-1.18.02.orig/lib/wininet_transport/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/lib/wininet_transport/Makefile	2009-05-19 10:41:55 +0200
@@ -11,7 +11,7 @@
 default: all
 
 .PHONY: all
-all: xmlrpc_wininet_transport.o xmlrpc_wininet_transport.osh
+all: xmlrpc_wininet_transport.$(OEXT) xmlrpc_wininet_transport.osh
 
 # Rules for the above dependencies are in common.mk,
 # courtesy of TARGET_MODS.
@@ -50,4 +50,4 @@
 
 # Need this dependency for those who don't use depend.mk.
 # Without it, version.h doesn't get created.
-xmlrpc_wininet_transport.o xmlrpc_wininet_transport.osh: version.h
+xmlrpc_wininet_transport.$(OEXT) xmlrpc_wininet_transport.osh: version.h
diff -ru xmlrpc-c-1.18.02.orig/src/cpp/Makefile xmlrpc-c-1.18.02/src/cpp/Makefile
--- xmlrpc-c-1.18.02.orig/src/cpp/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/src/cpp/Makefile	2009-05-19 10:41:55 +0200
@@ -120,7 +120,7 @@
   $(LIBXMLRPC_SERVER) \
   $(LIBXMLRPC_UTIL)
 $(LIBXMLRPC_CPP_SH): LIBOBJECTS = XmlRpcCpp.osh
-$(LIBXMLRPC_CPP_SH): LIBDEP = \
+libxmlrpc_cpp.$(LIBEXT) $(LIBXMLRPC_CPP_SH): LIBDEP = \
   -L$(BLDDIR)/src  -lxmlrpc_server -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
@@ -131,7 +131,7 @@
   $(LIBXMLRPC) \
   $(LIBXMLRPC_UTIL)
 $(LIBXMLRPCPP_SH): LIBOBJECTS = $(LIBXMLRPCPP_MODS:%=%.osh)
-$(LIBXMLRPCPP_SH): LIBDEP = \
+lib$(LIBXMLRPCPP_NAME).$(LIBEXT) $(LIBXMLRPCPP_SH): LIBDEP = \
   -L$(BLDDIR)/src -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
@@ -145,7 +145,7 @@
   $(LIBXMLRPC_UTIL) \
   $(LIBXMLRPC)
 $(LIBXMLRPC_SERVERPP_SH): LIBOBJECTS = $(LIBXMLRPC_SERVERPP_MODS:%=%.osh)
-$(LIBXMLRPC_SERVERPP_SH): LIBDEP = \
+libxmlrpc_server++.$(LIBEXT) $(LIBXMLRPC_SERVERPP_SH): LIBDEP = \
   -L. -l$(LIBXMLRPCPP_NAME) \
   -L$(BLDDIR)/src -lxmlrpc_server -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
@@ -161,7 +161,7 @@
   $(LIBXMLRPC_UTIL) \
   $(LIBXMLRPC)
 $(LIBXMLRPC_SERVER_ABYSSPP_SH): LIBOBJECTS = $(LIBXMLRPC_SERVER_ABYSSPP_MODS:%=%.osh)
-$(LIBXMLRPC_SERVER_ABYSSPP_SH): LIBDEP = \
+libxmlrpc_server_abyss++.$(LIBEXT) $(LIBXMLRPC_SERVER_ABYSSPP_SH): LIBDEP = \
    -L. -lxmlrpc_server++  -l$(LIBXMLRPCPP_NAME) \
    -L$(BLDDIR)/src -lxmlrpc_server_abyss -lxmlrpc_server -lxmlrpc \
    $(XML_PARSER_LIBDEP) \
@@ -187,7 +187,7 @@
   $(call shliblefn, libxmlrpc_packetsocket) \
   $(call shliblefn, libxmlrpc_server++)
 $(LIBXMLRPC_SERVER_PSTREAMPP_SH): LIBOBJECTS = $(LIBXMLRPC_SERVER_PSTREAMPP_MODS:%=%.osh)
-$(LIBXMLRPC_SERVER_PSTREAMPP_SH): LIBDEP = \
+libxmlrpc_server_pstream++.$(LIBEXT) $(LIBXMLRPC_SERVER_PSTREAMPP_SH): LIBDEP = \
   -L. -lxmlrpc_server++ -l$(LIBXMLRPCPP_NAME) -lxmlrpc_packetsocket \
   -L$(BLDDIR)/src -lxmlrpc_server -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
@@ -202,7 +202,7 @@
   $(call shliblefn, lib$(LIBXMLRPCPP_NAME)) \
   $(LIBXMLRPC_CLIENT)
 $(LIBXMLRPC_CLIENTPP_SH): LIBOBJECTS = $(LIBXMLRPC_CLIENTPP_MODS:%=%.osh)
-$(LIBXMLRPC_CLIENTPP_SH): LIBDEP = \
+libxmlrpc_client++.$(LIBEXT) $(LIBXMLRPC_CLIENTPP_SH): LIBDEP = \
   -L. -l$(LIBXMLRPCPP_NAME) -lxmlrpc_packetsocket \
   -L$(BLDDIR)/src -lxmlrpc_client -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
@@ -213,35 +213,37 @@
 $(LIBXMLRPC_PACKETSOCKET_SH): $(LIBXMLRPC_PACKETSOCKET_MODS:%=%.osh) \
   $(call shliblefn, lib$(LIBXMLRPCPP_NAME))
 $(LIBXMLRPC_PACKETSOCKET_SH): LIBOBJECTS = $(LIBXMLRPC_PACKETSOCKET_MODS:%=%.osh)
-$(LIBXMLRPC_PACKETSOCKET_SH): LIBDEP = \
+libxmlrpc_packetsocket.$(LIBEXT) $(LIBXMLRPC_PACKETSOCKET_SH): LIBDEP = \
   -L. -l$(LIBXMLRPCPP_NAME) \
   -L$(BLDDIR)/src -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
 
-libxmlrpc_cpp.a: XmlRpcCpp.o
-libxmlrpc_cpp.a: LIBOBJECTS = XmlRpcCpp.o
+libxmlrpc_cpp.$(LIBEXT): XmlRpcCpp.$(OEXT)
+libxmlrpc_cpp.$(LIBEXT): LIBOBJECTS = XmlRpcCpp.$(OEXT)
 
-lib$(LIBXMLRPCPP_NAME).a: $(LIBXMLRPCPP_MODS:%=%.o)
-lib$(LIBXMLRPCPP_NAME).a: LIBOBJECTS = $(LIBXMLRPCPP_MODS:%=%.o)
+lib$(LIBXMLRPCPP_NAME).$(LIBEXT): $(LIBXMLRPCPP_MODS:%=%.$(OEXT))
+lib$(LIBXMLRPCPP_NAME).$(LIBEXT): LIBOBJECTS = $(LIBXMLRPCPP_MODS:%=%.$(OEXT))
 
-libxmlrpc_server++.a: $(LIBXMLRPC_SERVERPP_MODS:%=%.o)
-libxmlrpc_server++.a: LIBOBJECTS = $(LIBXMLRPC_SERVERPP_MODS:%=%.o)
+libxmlrpc_server++.$(LIBEXT): $(LIBXMLRPC_SERVERPP_MODS:%=%.$(OEXT))
+libxmlrpc_server++.$(LIBEXT): LIBOBJECTS = $(LIBXMLRPC_SERVERPP_MODS:%=%.$(OEXT))
 
-libxmlrpc_server_abyss++.a: $(LIBXMLRPC_SERVER_ABYSSPP_MODS:%=%.o)
-libxmlrpc_server_abyss++.a: LIBOBJECTS=$(LIBXMLRPC_SERVER_ABYSSPP_MODS:%=%.o)
+libxmlrpc_server_abyss++.$(LIBEXT): $(LIBXMLRPC_SERVER_ABYSSPP_MODS:%=%.$(OEXT))
+libxmlrpc_server_abyss++.$(LIBEXT): LIBOBJECTS=$(LIBXMLRPC_SERVER_ABYSSPP_MODS:%=%.$(OEXT))
 
-libxmlrpc_server_cgi++.a: $(LIBXMLRPC_SERVER_CGIPP_MODS:%=%.o)
-libxmlrpc_server_cgi++.a: LIBOBJECTS=$(LIBXMLRPC_SERVER_CGIPP_MODS:%=%.o)
+libxmlrpc_server_cgi++.$(LIBEXT): $(LIBXMLRPC_SERVER_CGIPP_MODS:%=%.$(OEXT))
+libxmlrpc_server_cgi++.$(LIBEXT): LIBOBJECTS=$(LIBXMLRPC_SERVER_CGIPP_MODS:%=%.$(OEXT))
+libxmlrpc_server_cgi++.$(LIBEXT): LIBDEP=-module
+libxmlrpc_server_cgi++.$(LIBEXT): LIBTOOL_NO_UNDEFINED=
 
-libxmlrpc_server_pstream++.a: $(LIBXMLRPC_SERVER_PSTREAMPP_MODS:%=%.o)
-libxmlrpc_server_pstream++.a: LIBOBJECTS=$(LIBXMLRPC_SERVER_PSTREAMPP_MODS:%=%.o)
+libxmlrpc_server_pstream++.$(LIBEXT): $(LIBXMLRPC_SERVER_PSTREAMPP_MODS:%=%.$(OEXT))
+libxmlrpc_server_pstream++.$(LIBEXT): LIBOBJECTS=$(LIBXMLRPC_SERVER_PSTREAMPP_MODS:%=%.$(OEXT))
 
-libxmlrpc_client++.a: $(LIBXMLRPC_CLIENTPP_MODS:%=%.o)
-libxmlrpc_client++.a: LIBOBJECTS = $(LIBXMLRPC_CLIENTPP_MODS:%=%.o)
+libxmlrpc_client++.$(LIBEXT): $(LIBXMLRPC_CLIENTPP_MODS:%=%.$(OEXT))
+libxmlrpc_client++.$(LIBEXT): LIBOBJECTS = $(LIBXMLRPC_CLIENTPP_MODS:%=%.$(OEXT))
 
-libxmlrpc_packetsocket.a: $(LIBXMLRPC_PACKETSOCKET_MODS:%=%.o)
-libxmlrpc_packetsocket.a: LIBOBJECTS = $(LIBXMLRPC_PACKETSOCKET_MODS:%=%.o)
+libxmlrpc_packetsocket.$(LIBEXT): $(LIBXMLRPC_PACKETSOCKET_MODS:%=%.$(OEXT))
+libxmlrpc_packetsocket.$(LIBEXT): LIBOBJECTS = $(LIBXMLRPC_PACKETSOCKET_MODS:%=%.$(OEXT))
 
 #-----------------------------------------------------------------------------
 #  RULES TO COMPILE OBJECT MODULES FOR LIBRARIES
@@ -249,47 +251,47 @@
 
 # Rules for these are in common.mk, courtesy of TARGET_MODS_PP:
 
-XmlRpcCpp.o% XmlrpcCpp.osh: \
+XmlRpcCpp.$(OEXT)% XmlrpcCpp.osh: \
   INCLUDES = $(BASIC_INCLUDES)
 
-$(LIBXMLRPCPP_MODS:%=%.o) \
+$(LIBXMLRPCPP_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPCPP_MODS:%=%.osh): \
   INCLUDES = $(BASIC_INCLUDES)
 
 SERVER_INCLUDES = $(BASIC_INCLUDES) $(LIBXML_INCLUDES)
 
-$(LIBXMLRPC_SERVERPP_MODS:%=%.o) \
+$(LIBXMLRPC_SERVERPP_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_SERVERPP_MODS:%=%.osh): \
   INCLUDES = $(SERVER_INCLUDES)
 
-$(LIBXMLRPC_SERVER_ABYSSPP_MODS:%=%.o) \
+$(LIBXMLRPC_SERVER_ABYSSPP_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_SERVER_ABYSSPP_MODS:%=%.osh): \
   INCLUDES = $(SERVER_INCLUDES)
 
-$(LIBXMLRPC_SERVER_CGIPP_MODS:%=%.o) \
+$(LIBXMLRPC_SERVER_CGIPP_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_SERVER_CGIPP_MODS:%=%.osh): \
   INCLUDES = $(SERVER_INCLUDES)
 
-$(LIBXMLRPC_SERVER_PSTREAMPP_MODS:%=%.o) \
+$(LIBXMLRPC_SERVER_PSTREAMPP_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_SERVER_PSTREAMPP_MODS:%=%.osh): \
   INCLUDES = $(SERVER_INCLUDES)
 
-$(LIBXMLRPC_CLIENTPP_MODS:%=%.o) \
+$(LIBXMLRPC_CLIENTPP_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_CLIENTPP_MODS:%=%.osh): \
   INCLUDES = $(BASIC_INCLUDES) $(LIBXML_INCLUDES) $(TRANSPORT_INCLUDES)
 
-$(LIBXMLRPC_PACKETSOCKET_MODS:%=%.o) \
+$(LIBXMLRPC_PACKETSOCKET_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_PACKETSOCKET_MODS:%=%.osh): \
   INCLUDES = $(BASIC_INCLUDES)
 
 # <sstream> in Glibc 2.2 has some failed inlines, so we disable that warning:
-$(LIBXMLRPC_PACKETSOCKET_MODS:%=%.o) \
+$(LIBXMLRPC_PACKETSOCKET_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_PACKETSOCKET_MODS:%=%.osh): \
   CFLAGS_LOCAL = -Wno-inline
 
 TRANSPORT_CONFIG_USERS = client curl libwww wininet
 
-$(TRANSPORT_CONFIG_USERS:%=%.o) $(TRANSPORT_CONFIG_USERS:%=%.osh): \
+$(TRANSPORT_CONFIG_USERS:%=%.$(OEXT)) $(TRANSPORT_CONFIG_USERS:%=%.osh): \
   $(BLDDIR)/transport_config.h
 
 #-----------------------------------------------------------------------------
diff -ru xmlrpc-c-1.18.02.orig/src/cpp/test/Makefile xmlrpc-c-1.18.02/src/cpp/test/Makefile
--- xmlrpc-c-1.18.02.orig/src/cpp/test/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/src/cpp/test/Makefile	2009-05-19 10:41:55 +0200
@@ -33,13 +33,13 @@
 include $(SRCDIR)/common.mk
 
 
-TEST_OBJS = test.o registry.o server_abyss.o server_pstream.o tools.o
+TEST_OBJS = test.$(OEXT) registry.$(OEXT) server_abyss.$(OEXT) server_pstream.$(OEXT) tools.$(OEXT)
 
 ifeq ($(MUST_BUILD_CLIENT),yes)
-  TEST_OBJS += testclient.o
+  TEST_OBJS += testclient.$(OEXT)
   CLIENT_LIBS = $(LIBXMLRPC_CLIENTPP_A) $(LIBXMLRPC_CLIENT_A)
 else
-  TEST_OBJS += testclient_dummy.o
+  TEST_OBJS += testclient_dummy.$(OEXT)
   CLIENT_LIBS =
 endif
 
@@ -61,7 +61,7 @@
   # We're using the internal Expat XML parser
   TEST_LIBS += $(LIBXMLRPC_XMLPARSE_A) $(LIBXMLRPC_XMLTOK_A)
   LIB_XML =
-  registry.o: D_INTERNAL_EXPAT=-DINTERNAL_EXPAT
+  registry.$(OEXT): D_INTERNAL_EXPAT=-DINTERNAL_EXPAT
 else
   LIB_XML = $(shell xml2-config --libs)
 endif
@@ -69,7 +69,7 @@
 test: $(TEST_OBJS) $(TEST_LIBS)
 	$(CXXLD) -o $@ $(LDFLAGS) $(LADD) $^ $(LIB_XML) $(LIBS)
 
-%.o:%.cpp
+%.$(OEXT):%.cpp
 	$(CXX) -c $(INCLUDES) $(CXXFLAGS_ALL) $(D_INTERNAL_EXPAT) $<
 
 # Note the difference between 'check' and 'runtests'.  'check' means to check
diff -ru xmlrpc-c-1.18.02.orig/src/Makefile xmlrpc-c-1.18.02/src/Makefile
--- xmlrpc-c-1.18.02.orig/src/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/src/Makefile	2009-05-19 10:41:55 +0200
@@ -152,7 +152,7 @@
   $(LIBXMLRPC_UTIL) \
   $(XML_PARSER_LIBDEP_DEP)
 $(LIBXMLRPC): LIBOBJECTS = $(LIBXMLRPC_MODS:%=%.osh)
-$(LIBXMLRPC): LIBDEP = \
+libxmlrpc.$(LIBEXT) $(LIBXMLRPC): LIBDEP = \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util \
   $(XML_PARSER_LIBDEP)
 
@@ -163,7 +163,7 @@
   $(LIBXMLRPC_UTIL) \
   $(call shliblefn, libxmlrpc)
 $(LIBXMLRPC_SERVER): LIBOBJECTS = $(LIBXMLRPC_SERVER_MODS:%=%.osh)
-$(LIBXMLRPC_SERVER): LIBDEP = \
+libxmlrpc_server.$(LIBEXT) $(LIBXMLRPC_SERVER): LIBDEP = \
    -L. -lxmlrpc $(XML_PARSER_LIBDEP) -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
 
 LIBXMLRPC_SERVER_ABYSS = $(call shlibfn, libxmlrpc_server_abyss)
@@ -175,7 +175,7 @@
   $(call shliblefn, libxmlrpc_server) \
   $(call shliblefn, libxmlrpc)
 $(LIBXMLRPC_SERVER_ABYSS): LIBOBJECTS = $(LIBXMLRPC_SERVER_ABYSS_MODS:%=%.osh)
-$(LIBXMLRPC_SERVER_ABYSS): LIBDEP = \
+libxmlrpc_server_abyss.$(LIBEXT) $(LIBXMLRPC_SERVER_ABYSS): LIBDEP = \
   -L. -lxmlrpc_server \
   -L$(LIBXMLRPC_ABYSS_DIR) -lxmlrpc_abyss \
   -L. -lxmlrpc $(XML_PARSER_LIBDEP) -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
@@ -188,7 +188,7 @@
   $(call shliblefn, libxmlrpc_server) \
   $(call shliblefn, libxmlrpc)
 $(LIBXMLRPC_SERVER_CGI): LIBOBJECTS = $(LIBXMLRPC_SERVER_CGI_MODS:%=%.osh)
-$(LIBXMLRPC_SERVER_CGI): LIBDEP = \
+libxmlrpc_server_cgi.$(LIBEXT) $(LIBXMLRPC_SERVER_CGI): LIBDEP = \
   -L. -lxmlrpc_server \
   -L. -lxmlrpc $(XML_PARSER_LIBDEP) -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
 
@@ -202,26 +202,26 @@
 $(LIBXMLRPC_CLIENT): LIBOBJECTS = \
   $(LIBXMLRPC_CLIENT_MODS:%=%.osh) \
   $(TRANSPORT_MODS:%=%.osh)
-$(LIBXMLRPC_CLIENT): LIBDEP = \
+libxmlrpc_client.$(LIBEXT) $(LIBXMLRPC_CLIENT): LIBDEP = \
   $(TRANSPORT_LIBDEP) \
   -L. -lxmlrpc $(XML_PARSER_LIBDEP) -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
 
-libxmlrpc.a: $(LIBXMLRPC_MODS:%=%.o)
-libxmlrpc.a: LIBOBJECTS = $(LIBXMLRPC_MODS:%=%.o)
+libxmlrpc.$(LIBEXT): $(LIBXMLRPC_MODS:%=%.$(OEXT))
+libxmlrpc.$(LIBEXT): LIBOBJECTS = $(LIBXMLRPC_MODS:%=%.$(OEXT))
 
-libxmlrpc_server.a: $(LIBXMLRPC_SERVER_MODS:%=%.o)
-libxmlrpc_server.a: LIBOBJECTS = $(LIBXMLRPC_SERVER_MODS:%=%.o)
+libxmlrpc_server.$(LIBEXT): $(LIBXMLRPC_SERVER_MODS:%=%.$(OEXT))
+libxmlrpc_server.$(LIBEXT): LIBOBJECTS = $(LIBXMLRPC_SERVER_MODS:%=%.$(OEXT))
 
-libxmlrpc_server_abyss.a: $(LIBXMLRPC_SERVER_ABYSS_MODS:%=%.o)
-libxmlrpc_server_abyss.a: LIBOBJECTS=$(LIBXMLRPC_SERVER_ABYSS_MODS:%=%.o)
+libxmlrpc_server_abyss.$(LIBEXT): $(LIBXMLRPC_SERVER_ABYSS_MODS:%=%.$(OEXT))
+libxmlrpc_server_abyss.$(LIBEXT): LIBOBJECTS=$(LIBXMLRPC_SERVER_ABYSS_MODS:%=%.$(OEXT))
 
-libxmlrpc_server_cgi.a: $(LIBXMLRPC_SERVER_CGI_MODS:%=%.o)
-libxmlrpc_server_cgi.a: LIBOBJECTS=$(LIBXMLRPC_SERVER_CGI_MODS:%=%.o)
-
-libxmlrpc_client.a: $(LIBXMLRPC_CLIENT_MODS:%=%.o) $(TRANSPORT_MODS:%=%.o)
-libxmlrpc_client.a: LIBOBJECTS = \
-  $(LIBXMLRPC_CLIENT_MODS:%=%.o) \
-  $(TRANSPORT_MODS:%=%.o)
+libxmlrpc_server_cgi.$(LIBEXT): $(LIBXMLRPC_SERVER_CGI_MODS:%=%.$(OEXT))
+libxmlrpc_server_cgi.$(LIBEXT): LIBOBJECTS=$(LIBXMLRPC_SERVER_CGI_MODS:%=%.$(OEXT))
+
+libxmlrpc_client.$(LIBEXT): $(LIBXMLRPC_CLIENT_MODS:%=%.$(OEXT)) $(TRANSPORT_MODS:%=%.$(OEXT))
+libxmlrpc_client.$(LIBEXT): LIBOBJECTS = \
+  $(LIBXMLRPC_CLIENT_MODS:%=%.$(OEXT)) \
+  $(TRANSPORT_MODS:%=%.$(OEXT))
 
 
 #-----------------------------------------------------------------------------
@@ -247,23 +247,23 @@
   LIBXML_INCLUDES = -I$(SRCDIR)/lib/expat/xmlparse
 endif
 
-$(LIBXMLRPC_MODS:%=%.o) \
+$(LIBXMLRPC_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_MODS:%=%.osh): \
   INCLUDES = $(BASIC_INCLUDES) $(LIBXML_INCLUDES)
 
-$(LIBXMLRPC_CLIENT_MODS:%=%.o) \
+$(LIBXMLRPC_CLIENT_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_CLIENT_MODS:%=%.osh): \
   INCLUDES = $(BASIC_INCLUDES) $(TRANSPORT_INCLUDES)
 
-$(LIBXMLRPC_SERVER_MODS:%=%.o) \
+$(LIBXMLRPC_SERVER_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_SERVER_MODS:%=%.osh): \
   INCLUDES = $(BASIC_INCLUDES)
 
-$(LIBXMLRPC_SERVER_ABYSS_MODS:%=%.o) \
+$(LIBXMLRPC_SERVER_ABYSS_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_SERVER_ABYSS_MODS:%=%.osh): \
   INCLUDES = $(BASIC_INCLUDES) -I$(SRCDIR)/lib/abyss/src
 
-$(LIBXMLRPC_SERVER_CGI_MODS:%=%.o) \
+$(LIBXMLRPC_SERVER_CGI_MODS:%=%.$(OEXT)) \
 $(LIBXMLRPC_SERVER_CGI_MODS:%=%.osh): \
   INCLUDES = $(BASIC_INCLUDES)
 
@@ -288,9 +288,9 @@
 dep-common:
   INCLUDES = $(BASIC_INCLUDES) $(TRANSPORT_INCLUDES) $(LIBXML_INCLUDES)
 
-xmlrpc_client.o xmlrpc_client.osh: $(BLDDIR)/transport_config.h
-xmlrpc_client.o xmlrpc_client.osh: $(BLDDIR)/version.h
-registry.o registry.osh: $(BLDDIR)/version.h
-version.o version.osh: $(BLDDIR)/version.h
+xmlrpc_client.$(OEXT) xmlrpc_client.osh: $(BLDDIR)/transport_config.h
+xmlrpc_client.$(OEXT) xmlrpc_client.osh: $(BLDDIR)/version.h
+registry.$(OEXT) registry.osh: $(BLDDIR)/version.h
+version.$(OEXT) version.osh: $(BLDDIR)/version.h
 
 include depend.mk
diff -ru xmlrpc-c-1.18.02.orig/src/test/Makefile xmlrpc-c-1.18.02/src/test/Makefile
--- xmlrpc-c-1.18.02.orig/src/test/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/src/test/Makefile	2009-05-19 10:41:55 +0200
@@ -28,23 +28,23 @@
 LDFLAGS = $(LADD)
 
 TEST_OBJS = \
-  test.o \
-  abyss.o \
-  cgi.o \
-  method_registry.o \
-  parse_xml.o \
-  serialize.o \
-  serialize_value.o \
-  server_abyss.o \
-  value.o \
-  value_datetime.o \
-  xml_data.o \
+  test.$(OEXT) \
+  abyss.$(OEXT) \
+  cgi.$(OEXT) \
+  method_registry.$(OEXT) \
+  parse_xml.$(OEXT) \
+  serialize.$(OEXT) \
+  serialize_value.$(OEXT) \
+  server_abyss.$(OEXT) \
+  value.$(OEXT) \
+  value_datetime.$(OEXT) \
+  xml_data.$(OEXT) \
 
 ifeq ($(MUST_BUILD_CLIENT),yes)
-  TEST_OBJS += client.o
+  TEST_OBJS += client.$(OEXT)
   LIBXMLRPC_CLIENT_DEP = $(LIBXMLRPC_CLIENT_A)
 else
-  TEST_OBJS += client_dummy.o
+  TEST_OBJS += client_dummy.$(OEXT)
   LIBXMLRPC_CLIENT_DEP =
 endif
 
@@ -60,13 +60,13 @@
 	$(CCLD) -o $@ $(LDFLAGS) \
 	    $(TEST_OBJS) $(LDADD_CLIENT) $(LDADD_ABYSS_SERVER) $(CASPRINTF)
 
-cgitest1:%:%.o $(LIBXMLRPC_SERVER_A) $(LIBXMLRPC_SERVER_CGI_A) \
+cgitest1:%:%.$(OEXT) $(LIBXMLRPC_SERVER_A) $(LIBXMLRPC_SERVER_CGI_A) \
   $(LIBXMLRPC_A) $(LIBXMLRPC_UTIL_A) $(LIBXMLRPC_XML)
 	$(CCLD) -o $@ $< $(LDFLAGS) $(LDADD_CGI_SERVER)
 
-OBJS = $(TEST_OBJS) cgitest1.o
+OBJS = $(TEST_OBJS) cgitest1.$(OEXT)
 
-$(OBJS):%.o:%.c
+$(OBJS):%.$(OEXT):%.c
 	$(CC) -c $(INCLUDES) $(CFLAGS_ALL) $<
 
 # Note the difference between 'check' and 'runtests'.  'check' means to check
diff -ru xmlrpc-c-1.18.02.orig/tools/common.mk xmlrpc-c-1.18.02/tools/common.mk
--- xmlrpc-c-1.18.02.orig/tools/common.mk	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/tools/common.mk	2009-05-19 10:41:55 +0200
@@ -29,7 +29,7 @@
 include $(SRCDIR)/common.mk
 
 ifneq ($(OMIT_LIB_RULE),Y)
-$(SRCDIR)/tools/lib/dumpvalue.o: FORCE
+$(SRCDIR)/tools/lib/dumpvalue.$(OEXT): FORCE
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/tools/lib/Makefile $(notdir $@) 
 endif
 
diff -ru xmlrpc-c-1.18.02.orig/tools/lib/Makefile xmlrpc-c-1.18.02/tools/lib/Makefile
--- xmlrpc-c-1.18.02.orig/tools/lib/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/tools/lib/Makefile	2009-05-19 10:41:55 +0200
@@ -21,12 +21,12 @@
   -I$(SRCDIR)/lib/util/include \
 
 LIBOBJS = \
-  dumpvalue.o \
+  dumpvalue.$(OEXT) \
 
 .PHONY: all
 all: $(LIBOBJS)
 
-%.o:%.c
+%.$(OEXT):%.c
 	$(CC) -c $(CFLAGS_ALL) $<
 
 include depend.mk
diff -ru xmlrpc-c-1.18.02.orig/tools/xmlrpc/Makefile xmlrpc-c-1.18.02/tools/xmlrpc/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xmlrpc/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/tools/xmlrpc/Makefile	2009-05-19 10:41:55 +0200
@@ -26,26 +26,26 @@
 all: xmlrpc
 
 UTIL_OBJS = \
-  casprintf.o \
-  cmdline_parser.o \
-  getoptx.o \
-  stripcaseeq.o \
-  string_parser.o \
+  casprintf.$(OEXT) \
+  cmdline_parser.$(OEXT) \
+  getoptx.$(OEXT) \
+  stripcaseeq.$(OEXT) \
+  string_parser.$(OEXT) \
 
 UTILS = $(UTIL_OBJS:%=$(UTIL_DIR)/%)
 
-DUMPVALUE = $(SRCDIR)/tools/lib/dumpvalue.o
+DUMPVALUE = $(SRCDIR)/tools/lib/dumpvalue.$(OEXT)
 
 # Can we just use $(LIBS) in the link?
 
 LIBS = $(LIBXMLRPC_CLIENT) $(LIBXMLRPC) $(LIBXMLRPC_XML) $(LIBXMLRPC_UTIL)
 
-XMLRPC_OBJS = xmlrpc.o $(DUMPVALUE)
+XMLRPC_OBJS = xmlrpc.$(OEXT) $(DUMPVALUE)
 
 xmlrpc: $(XMLRPC_OBJS) $(LIBS) $(UTILS)
 	$(CCLD) -o $@ $(LDFLAGS) $(XMLRPC_OBJS) $(UTILS) $(CLIENT_LDLIBS)
 
-%.o:%.c $(BLDDIR)/include/xmlrpc-c/config.h
+%.$(OEXT):%.c $(BLDDIR)/include/xmlrpc-c/config.h
 	$(CC) -c $(CFLAGS_ALL) $<
 
 include depend.mk
diff -ru xmlrpc-c-1.18.02.orig/tools/xml-rpc-api2cpp/Makefile xmlrpc-c-1.18.02/tools/xml-rpc-api2cpp/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xml-rpc-api2cpp/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/tools/xml-rpc-api2cpp/Makefile	2009-05-19 10:41:55 +0200
@@ -29,11 +29,11 @@
 LDLIBS = -L$(BLDDIR)/src/cpp -lxmlrpc_cpp -lxmlrpc_server $(CLIENT_LDLIBS)
 
 OBJECTS = \
-  xml-rpc-api2cpp.o \
-  DataType.o \
-  XmlRpcFunction.o \
-  XmlRpcClass.o \
-  SystemProxy.o \
+  xml-rpc-api2cpp.$(OEXT) \
+  DataType.$(OEXT) \
+  XmlRpcFunction.$(OEXT) \
+  XmlRpcClass.$(OEXT) \
+  SystemProxy.$(OEXT) \
 
 xml-rpc-api2cpp: \
   $(OBJECTS) \
@@ -45,7 +45,7 @@
   $(LIBXMLRPC_UTIL)
 	$(CXXLD) -o $@ $(LDFLAGS) $(OBJECTS) $(LDLIBS)
 
-%.o:%.cpp
+%.$(OEXT):%.cpp
 	$(CXX) -c $(CXXFLAGS_ALL) $<
 
 include depend.mk
diff -ru xmlrpc-c-1.18.02.orig/tools/xmlrpc_cpp_proxy/Makefile xmlrpc-c-1.18.02/tools/xmlrpc_cpp_proxy/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xmlrpc_cpp_proxy/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/tools/xmlrpc_cpp_proxy/Makefile	2009-05-19 10:41:55 +0200
@@ -25,11 +25,11 @@
 all: xmlrpc_cpp_proxy
 
 OBJECTS = \
-  xmlrpc_cpp_proxy.o \
-  xmlrpcMethod.o \
-  xmlrpcType.o \
-  proxyClass.o \
-  systemProxy.o \
+  xmlrpc_cpp_proxy.$(OEXT) \
+  xmlrpcMethod.$(OEXT) \
+  xmlrpcType.$(OEXT) \
+  proxyClass.$(OEXT) \
+  systemProxy.$(OEXT) \
 
 LIBS = \
   $(LIBXMLRPC_CLIENTPP) \
@@ -44,7 +44,7 @@
 xmlrpc_cpp_proxy:  $(OBJECTS) $(LIBS)
 	$(CXXLD) -o $@ $(LDFLAGS) $(OBJECTS) $(LDLIBS) $(LADD)
 
-%.o:%.cpp $(BLDDIR)/include/xmlrpc-c/config.h
+%.$(OEXT):%.cpp $(BLDDIR)/include/xmlrpc-c/config.h
 	$(CXX) -c $(CXXFLAGS_ALL) $<
 
 include depend.mk
diff -ru xmlrpc-c-1.18.02.orig/tools/xmlrpc_pstream/Makefile xmlrpc-c-1.18.02/tools/xmlrpc_pstream/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xmlrpc_pstream/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/tools/xmlrpc_pstream/Makefile	2009-05-19 10:41:55 +0200
@@ -28,8 +28,8 @@
 all: xmlrpc_pstream
 
 OBJECTS = \
-  xmlrpc_pstream.o \
-  $(SRCDIR)/tools/lib/dumpvalue.o \
+  xmlrpc_pstream.$(OEXT) \
+  $(SRCDIR)/tools/lib/dumpvalue.$(OEXT) \
 
 LIBS = \
   $(LIBXMLRPC_CLIENTPP) \
@@ -42,19 +42,19 @@
 LDLIBS = $(CLIENTPP_LDLIBS) $(CLIENT_LDLIBS) $(READLINE_LDLIBS)
 
 UTIL_OBJS = \
-  casprintf.o \
-  cmdline_parser_cpp.o \
-  cmdline_parser.o \
-  getoptx.o \
-  string_parser.o \
-  stripcaseeq.o \
+  casprintf.$(OEXT) \
+  cmdline_parser_cpp.$(OEXT) \
+  cmdline_parser.$(OEXT) \
+  getoptx.$(OEXT) \
+  string_parser.$(OEXT) \
+  stripcaseeq.$(OEXT) \
 
 UTILS = $(UTIL_OBJS:%=$(UTIL_DIR)/%)
 
 xmlrpc_pstream:  $(OBJECTS) $(LIBS) $(UTILS)
 	$(CXXLD) -o $@ $(LDFLAGS) $(OBJECTS) $(UTILS) $(LDLIBS) $(LADD)
 
-%.o:%.cpp $(BLDDIR)/include/xmlrpc-c/config.h
+%.$(OEXT):%.cpp $(BLDDIR)/include/xmlrpc-c/config.h
 	$(CXX) -c $(CXXFLAGS_ALL) $<
 
 include depend.mk
diff -ru xmlrpc-c-1.18.02.orig/tools/xmlrpc_transport/Makefile xmlrpc-c-1.18.02/tools/xmlrpc_transport/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xmlrpc_transport/Makefile	2009-05-19 10:41:35 +0200
+++ xmlrpc-c-1.18.02/tools/xmlrpc_transport/Makefile	2009-05-19 10:41:55 +0200
@@ -25,20 +25,20 @@
 all: xmlrpc_transport
 
 UTIL_OBJS = \
-  casprintf.o \
-  cmdline_parser.o \
-  getoptx.o \
-  stripcaseeq.o \
-  string_parser.o \
+  casprintf.$(OEXT) \
+  cmdline_parser.$(OEXT) \
+  getoptx.$(OEXT) \
+  stripcaseeq.$(OEXT) \
+  string_parser.$(OEXT) \
 
 UTILS = $(UTIL_OBJS:%=$(UTIL_DIR)/%)
 
 LIBS = $(LIBXMLRPC_CLIENT) $(LIBXMLRPC) $(LIBXMLRPC_XML) $(LIBXMLRPC_UTIL)
 
-xmlrpc_transport:%:%.o $(LIBS) $(UTILS)
+xmlrpc_transport:%:%.$(OEXT) $(LIBS) $(UTILS)
 	$(CCLD) -o $@ $(LDFLAGS) $< $(CLIENT_LDLIBS) $(UTILS)
 
-%.o:%.c
+%.$(OEXT):%.c
 	$(CC) -c $(CFLAGS_ALL) $<
 
 include depend.mk
