diff -Nru vanilla//scite/gtk/makefile patched//scite/gtk/makefile
--- vanilla//scite/gtk/makefile	2010-06-23 18:04:42.631333431 +0100
+++ patched//scite/gtk/makefile	2010-06-23 18:06:11.150336242 +0100
@@ -107,8 +107,10 @@
 	done
 
 ifdef gnomeprefix
-	$(INSTALL) -m 755 -D SciTE.desktop $(DESTDIR)$(datadir)/applications/SciTE.desktop
-	$(INSTALL) -m 644 -D Sci48M.png $(DESTDIR)$(pixmapdir)/Sci48M.png
+	$(INSTALL) -m 755 -d $(DESTDIR)$(datadir)/applications/
+	$(INSTALL) -m 755 SciTE.desktop $(DESTDIR)$(datadir)/applications/
+	$(INSTALL) -m 755 -d $(DESTDIR)$(pixmapdir)/
+	$(INSTALL) -m 644 Sci48M.png $(DESTDIR)$(pixmapdir)/Sci48M.png
 endif
 
 uninstall:
