--- scite/gtk/makefile.orig	2006-10-09 06:26:55 -0300
+++ scite/gtk/makefile	2006-10-09 06:36:09 -0300
@@ -123,11 +123,14 @@
 
 ifdef gnomeprefix
 ifdef GTK2
-	$(INSTALL) -m 755 -D SciTE.desktop $(DESTDIR)$(datadir)/applications/SciTE.desktop
+	$(INSTALL) -m 755 -d $(DESTDIR)$(datadir)/applications/
+	$(INSTALL) -m 755 SciTE.desktop $(DESTDIR)$(datadir)/applications/
 else
-	$(INSTALL) -m 755 -D SciTE.desktop $(DESTDIR)$(datadir)/applications/SciTE.desktop
+	$(INSTALL) -m 755 -d $(DESTDIR)$(datadir)/applications/
+	$(INSTALL) -m 755 SciTE.desktop $(DESTDIR)$(datadir)/applications/
 endif
-	$(INSTALL) -m 644 -D Sci48M.png $(DESTDIR)$(pixmapdir)/Sci48M.png
+	$(INSTALL) -m 755 -d $(DESTDIR)$(pixmapdir)/
+	$(INSTALL) -m 644 Sci48M.png $(DESTDIR)$(pixmapdir)/Sci48M.png
 endif
 
 uninstall:
