--- kaffe.env~	2007-09-24 18:51:38 +0200
+++ kaffe.env	2007-09-24 18:22:35 +0200
@@ -3,14 +3,14 @@
 # $Header: /var/cvsroot/gentoo-x86/dev-java/kaffe/files/kaffe.env,v 1.2 2006/12/01 18:44:57 betelgeuse Exp $
 
 VERSION="Kaffe @PV@"
-JAVA_HOME=/opt/@P@
-BOOTCLASSPATH="/opt/@P@/jre/lib/rt.jar:/opt/@P@/lib/tools.jar"
-JDK_HOME=/opt/@P@
+JAVA_HOME="@GENTOO_PORTAGE_EPREFIX@"/opt/@P@
+BOOTCLASSPATH="@GENTOO_PORTAGE_EPREFIX@"/opt/@P@/jre/lib/rt.jar:"@GENTOO_PORTAGE_EPREFIX@"/opt/@P@/lib/tools.jar
+JDK_HOME="@GENTOO_PORTAGE_EPREFIX@"/opt/@P@
 JAVAC=${JAVA_HOME}/bin/javac
-MANPATH=/opt/@P@/man
+MANPATH="@GENTOO_PORTAGE_EPREFIX@"/opt/@P@/man
 PATH="${JAVA_HOME}/bin:${JAVA_HOME}/jre/bin"
 ROOTPATH="${JAVA_HOME}/bin:${JAVA_HOME}/jre/bin"
-LDPATH="/opt/@P@/jre/lib/i386/"
+LDPATH="@GENTOO_PORTAGE_EPREFIX@"/opt/@P@/jre/lib/i386/
 PROVIDES_TYPE="JDK JRE"
 PROVIDES_VERSION="1.4"
 GENERATION="2"
