Index: src/tar.c
===================================================================
RCS file: /cvsroot/tar/tar/src/tar.c,v
retrieving revision 1.109
retrieving revision 1.110
diff -u -r1.109 -r1.110
--- src/tar.c	22 Dec 2004 10:53:48 -0000	1.109
+++ src/tar.c	5 Jan 2005 17:30:36 -0000	1.110
@@ -617,9 +617,9 @@
   
   switch (key)
     {
-      case 1:
+      case ARGP_KEY_ARG:
 	/* File name or non-parsed option, because of ARGP_IN_ORDER */
-	name_add (optarg);
+	name_add (arg);
 	args->input_files++;
 	break;
 
