diff -ru tar-1.22.orig/src/create.c tar-1.22/src/create.c
--- tar-1.22.orig/src/create.c	2009-08-26 00:39:38 -0700
+++ tar-1.22/src/create.c	2009-08-26 00:39:01 -0700
@@ -1654,6 +1654,11 @@
 	       /* Original ctime will change if the file is a directory and
 		  --remove-files is given */
 	       && !(remove_files_option && is_dir))
+		   #ifdef __INTERIX
+		   /* on some interix versions, this is triggered all the time
+			* (wrongly) for directories. i have no idea why?! */
+		   && !is_dir
+		   #endif
 	      || original_size < final_stat.st_size)
 	    {
 	      WARN ((0, 0, _("%s: file changed as we read it"),
