diff -ru tar-1.22.orig/src/create.c tar-1.22/src/create.c
--- tar-1.22.orig/src/create.c	2009-08-26 04:52:34 -0700
+++ tar-1.22/src/create.c	2009-08-26 04:53:17 -0700
@@ -1656,10 +1656,14 @@
 	       && !(remove_files_option && is_dir))
 	      || original_size < final_stat.st_size)
 	    {
+		   /* on interix, somehow this check fails to work correctly.
+			* it detects _lots_ of file/dir changes where there are none */
+		   #ifndef __INTERIX
 	      WARN ((0, 0, _("%s: file changed as we read it"),
 		     quotearg_colon (p)));
 	      if (exit_status == TAREXIT_SUCCESS)
 		exit_status = TAREXIT_DIFFERS;
+		   #endif
 	    }
 	  else if (atime_preserve_option == replace_atime_preserve
 		   && set_file_atime (fd, p, restore_times) != 0)
