Patch by Alan Hourihane for FreeMiNT

--- lib/fseeko.c.old	2008-05-23 13:40:45.000000000 +0100
+++ lib/fseeko.c	2008-05-23 10:29:15.000000000 +0100
@@ -101,6 +101,8 @@
   if ((fp->_Mode & _MWRITE ? fp->_Next == fp->_Buf : fp->_Next == fp->_Rend)
       && fp->_Rback == fp->_Back + sizeof (fp->_Back)
       && fp->_Rsave == NULL)
+#elif defined __MINT__
+  if (0)
 #else
   #error "Please port gnulib fseeko.c to your platform! Look at the code in fpurge.c, then report this to bug-gnulib."
 #endif
