--- mono/utils/freebsd-elf_common.h.orig	2009-02-12 10:56:14.604852206 +0100
+++ mono/utils/freebsd-elf_common.h	2009-02-12 10:56:52.034748775 +0100
@@ -42,10 +42,14 @@
  * not include the padding.
  */
 
+/*
+ * Patched acording to: http://lists.ximian.com/pipermail/mono-patches/2009-January/138782.html
+ */
+
 typedef struct {
-	u_int32_t	n_namesz;	/* Length of name. */
-	u_int32_t	n_descsz;	/* Length of descriptor. */
-	u_int32_t	n_type;		/* Type of this note. */
+	uint32_t	n_namesz;	/* Length of name. */
+	uint32_t	n_descsz;	/* Length of descriptor. */
+	uint32_t	n_type;		/* Type of this note. */
 } Elf_Note;
 
 /* Indexes into the e_ident array.  Keep synced with
