diff -ruN Confix-2.1.0.orig/libconfix/core/filesys/directory.py Confix-2.1.0/libconfix/core/filesys/directory.py
--- Confix-2.1.0.orig/libconfix/core/filesys/directory.py	2007-08-10 11:56:15.259323000 +0200
+++ Confix-2.1.0/libconfix/core/filesys/directory.py	2007-08-10 11:59:35.163484000 +0200
@@ -68,8 +68,8 @@
     def add(self, name, entry):
         assert isinstance(entry, DirectoryEntry)
         assert entry.filesystem() is None or entry.filesystem() is self.filesystem()
-        if self.entry_by_name_.has_key(name):
-            raise Directory.AlreadyMounted(name=name, dir=self)
+        #if self.entry_by_name_.has_key(name):
+            #raise Directory.AlreadyMounted(name=name, dir=self)
         self.entry_by_name_[name] = entry
         self.name_by_entry_[entry] = name
         entry.set_parent(self)
diff -ruN Confix-2.1.0.orig/libconfix/core/filesys/scan.py Confix-2.1.0/libconfix/core/filesys/scan.py
--- Confix-2.1.0.orig/libconfix/core/filesys/scan.py	2007-08-10 11:56:15.199323000 +0200
+++ Confix-2.1.0/libconfix/core/filesys/scan.py	2007-08-10 11:59:35.173484000 +0200
@@ -52,19 +52,6 @@
         if name in ['.', '..']:
             continue
         absname = os.path.join(abspath, name)
-        existing_entry = dir.get(name)
-        # <paranoia>
-        if existing_entry is not None:
-            if os.path.isfile(absname):
-                if not isinstance(existing_entry, File):
-                    raise Error('Cannot convert existing entry '+name+' to a file')
-                continue
-            if os.path.isdir(absname):
-                if not isinstance(existing_entry, Directory):
-                    raise Error('Cannot convert existing entry '+name+' to a directory')
-                continue
-            raise Error(absname+' has unknown type')
-        # </paranoia>
 
         # go add the new entry
         if os.path.isfile(absname):
