diff -rubB Confix-2.1.0.orig/libconfix/core/automake/makefile_am.py Confix-2.1.0/libconfix/core/automake/makefile_am.py
--- Confix-2.1.0.orig/libconfix/core/automake/makefile_am.py	Tue Oct  9 13:50:28 2007
+++ Confix-2.1.0/libconfix/core/automake/makefile_am.py	Tue Oct  9 14:09:47 2007
@@ -469,7 +469,15 @@
         
         # register automatic tests and set their environment
 
-        tests = self.dir_primary('check', 'PROGRAMS') + \
+        test_tmp = self.dir_primary('check', 'PROGRAMS')
+        count = 0
+        for prog in test_tmp:
+            prog += "$(EXEEXT)"
+            test_tmp[count] = prog
+            count += 1
+            pass
+
+        tests = test_tmp + \
                 self.dir_primary('check', 'SCRIPTS')
         if len(tests):
             lines.extend(List(name='TESTS', values=tests, mitigate=True).lines())
