diff -ru Confix-2.1.0.orig/libconfix/plugins/c/h.py Confix-2.1.0/libconfix/plugins/c/h.py
--- Confix-2.1.0.orig/libconfix/plugins/c/h.py	2008-10-17 10:37:57 +0200
+++ Confix-2.1.0/libconfix/plugins/c/h.py	2008-10-17 10:39:38 +0200
@@ -31,6 +31,7 @@
 
 class HeaderBuilder(CBaseBuilder):
     PROPERTY_INSTALLPATH = 'INSTALLPATH_CINCLUDE'
+    PROPERTY_PRIVATE = 'PRIVATE_CINCLUDE'
 
     class AmbiguousVisibility(Error):
         def __init__(self, header_builder, cur, prev):
@@ -56,6 +57,7 @@
         self.__namespace_install_path = None
         self.__namespace_error = None
         self.__property_install_path = None
+        self.__property_private = None
         self.__iface_install_path = None
         self.__external_install_path = None
 
@@ -75,6 +77,7 @@
 
         if self.file() is not None:
             self.__property_install_path = self.file().get_property(HeaderBuilder.PROPERTY_INSTALLPATH)
+            self.__property_private = self.file().get_property(HeaderBuilder.PROPERTY_PRIVATE)
             pass
         try:        
             self.__namespace_install_path = namespace.find_unique_namespace(self.file().lines())
@@ -129,7 +132,8 @@
     def output(self):
         super(HeaderBuilder, self).output()
         installdir = self.visible_in_directory()
-        self.parentbuilder().file_installer().add_public_header(filename=self.file().name(), dir=installdir)
+        if self.__property_private is None:
+            self.parentbuilder().file_installer().add_public_header(filename=self.file().name(), dir=installdir)
         self.parentbuilder().file_installer().add_private_header(filename=self.file().name(), dir=installdir)
         pass
 
