diff -NrU5 httpup-0.3.2.original/configparser.cpp httpup-0.3.2/configparser.cpp
--- httpup-0.3.2.original/configparser.cpp	2009-08-25 20:01:59.000000000 -0600
+++ httpup-0.3.2/configparser.cpp	2009-08-25 20:02:20.000000000 -0600
@@ -9,10 +9,11 @@
 //  (at your option) any later version.                                   
 ////////////////////////////////////////////////////////////////////////
 
 #include <iostream>
 #include <cstring>
+#include <cstdio>
 #include "configparser.h"
 
 using namespace std;
 
 int ConfigParser::parseConfig(const std::string& fileName,
diff -NrU5 httpup-0.3.2.original/fileutils.cpp httpup-0.3.2/fileutils.cpp
--- httpup-0.3.2.original/fileutils.cpp	2009-08-25 20:01:59.000000000 -0600
+++ httpup-0.3.2/fileutils.cpp	2009-08-25 20:02:29.000000000 -0600
@@ -8,10 +8,11 @@
 //  the Free Software Foundation; either version 2 of the License, or     
 //  (at your option) any later version.                                   
 ////////////////////////////////////////////////////////////////////////
 
 #include <sys/types.h>
+#include <cstdio>
 #include <sys/stat.h>
 #include <unistd.h>
 #include <dirent.h>
 
 #include "md5.h"
