diff -NrU5 httpup-0.3.2.orig/configparser.cpp httpup-0.3.2/configparser.cpp
--- httpup-0.3.2.orig/configparser.cpp	2008-06-20 15:59:35.000000000 +0200
+++ httpup-0.3.2/configparser.cpp	2008-06-20 16:00:07.000000000 +0200
@@ -8,10 +8,11 @@
 //  the Free Software Foundation; either version 2 of the License, or     
 //  (at your option) any later version.                                   
 ////////////////////////////////////////////////////////////////////////
 
 #include <iostream>
+#include <cstring>
 #include "configparser.h"
 
 using namespace std;
 
 int ConfigParser::parseConfig(const std::string& fileName,
diff -NrU5 httpup-0.3.2.orig/httpup.cpp httpup-0.3.2/httpup.cpp
--- httpup-0.3.2.orig/httpup.cpp	2008-06-20 15:59:35.000000000 +0200
+++ httpup-0.3.2/httpup.cpp	2008-06-20 15:59:43.000000000 +0200
@@ -13,10 +13,12 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
 #include <dirent.h>
+#include <cstring>
+#include <cstdlib>
 
 #include "fileutils.h"
 #include "httpup.h"
 #include "configparser.h"
 
diff -NrU5 httpup-0.3.2.orig/main.cpp httpup-0.3.2/main.cpp
--- httpup-0.3.2.orig/main.cpp	2008-06-20 15:59:35.000000000 +0200
+++ httpup-0.3.2/main.cpp	2008-06-20 15:59:43.000000000 +0200
@@ -8,11 +8,14 @@
 //  the Free Software Foundation; either version 2 of the License, or
 //  (at your option) any later version.
 ////////////////////////////////////////////////////////////////////////
 
 #include <iostream>
-#include <string>
+#include <cstring>
+#include <cstdlib>
+#include <algorithm>
+
 using namespace std;
 
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <dirent.h>
