Only in WindowMaker-0.92.0: nocliptitle.patch
diff -ur WindowMaker-0.92.0.orig/src/defaults.c WindowMaker-0.92.0/src/defaults.c
--- WindowMaker-0.92.0.orig/src/defaults.c	2009-10-28 13:07:57.594817000 +0100
+++ WindowMaker-0.92.0/src/defaults.c	2009-10-28 13:09:19.223404220 +0100
@@ -623,6 +623,9 @@
     {"ClipTitleFont",	DEF_CLIP_TITLE_FONT,	NULL,
     NULL,				getFont,	setClipTitleFont
     },
+    {"ShowClipTitle",	"YES",			NULL,
+    &wPreferences.show_clip_title,	getBool,	NULL
+    },
     {"LargeDisplayFont",DEF_WORKSPACE_NAME_FONT, NULL,
     NULL,				getFont,	setLargeDisplayFont
     },
Only in WindowMaker-0.92.0/src: defaults.c.orig
diff -ur WindowMaker-0.92.0.orig/src/dock.c WindowMaker-0.92.0/src/dock.c
--- WindowMaker-0.92.0.orig/src/dock.c	2009-10-28 13:07:56.325404000 +0100
+++ WindowMaker-0.92.0/src/dock.c	2009-10-28 13:09:19.221166695 +0100
@@ -1293,8 +1293,9 @@
 
     tx = CLIP_BUTTON_SIZE*ICON_SIZE/64;
 
-    WMDrawString(scr->wmscreen, win, color, scr->clip_title_font, tx,
-                 ty, ws_name, length);
+    if(wPreferences.show_clip_title)
+        WMDrawString(scr->wmscreen, win, color, scr->clip_title_font, tx,
+        	     ty, ws_name, length);
     /*WMDrawString(scr->wmscreen, win, color, scr->clip_title_font, 4,
      2, ws_name, length);*/
 
Only in WindowMaker-0.92.0/src: dock.c.orig
diff -ur WindowMaker-0.92.0.orig/src/WindowMaker.h WindowMaker-0.92.0/src/WindowMaker.h
--- WindowMaker-0.92.0.orig/src/WindowMaker.h	2009-10-28 13:07:57.595432000 +0100
+++ WindowMaker-0.92.0/src/WindowMaker.h	2009-10-28 13:09:56.616645224 +0100
@@ -481,6 +481,8 @@
     /* single click to lauch applications */
     char single_click;
 
+    int show_clip_title;
+
     RImage *swtileImage;
     RImage *swbackImage[9];
 
Only in WindowMaker-0.92.0/src: WindowMaker.h.orig
Only in WindowMaker-0.92.0/src: WindowMaker.h.rej
