windows doesn't have socklen_t

--- WindowMaker-0.92.0.orig/WINGs/connection.c	2008-08-18 10:17:56 +0200
+++ WindowMaker-0.92.0/WINGs/connection.c	2008-08-18 10:25:11 +0200
@@ -201,7 +201,7 @@
 
         if (cPtr->state == WCInProgress) {
             Bool failed;
-            socklen_t len = sizeof(result);
+            int len = sizeof(result);
 
             WCErrorCode = 0;
             if (getsockopt(cPtr->sock, SOL_SOCKET, SO_ERROR,
@@ -462,7 +462,7 @@
 {
     WMConnection *cPtr;
     struct sockaddr_in *socketaddr;
-    socklen_t size;
+    int size;
     int sock, on;
 
     WCErrorCode = 0;
@@ -678,7 +678,7 @@
 WMAcceptConnection(WMConnection *listener)
 {
     struct sockaddr_in clientname;
-    socklen_t size;
+    int size;
     int newSock;
     WMConnection *newConnection;
 
