
  Add a missing #include <math.h>.

--- WindowMaker-0.92.0/configure.ac
+++ WindowMaker-0.92.0/configure.ac
@@ -78,7 +78,8 @@
 [  --with-incs-from        pass compiler flags to look for header files],
 	[inc_search_path="$withval $inc_search_path"])
 
-
+AC_CHECK_LIBM
+AC_SUBST([LIBM])
 
 dnl Checks for library functions.
 dnl ============================
--- WindowMaker-0.92.0/src/placement.c
+++ WindowMaker-0.92.0/src/placement.c
@@ -27,6 +27,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <limits.h>
+#include <math.h>
 
 #include "WindowMaker.h"
 #include "wcore.h"
