--- file.c.orig 2005-04-03 19:48:57.000000000 +0000
+++ file.c      2005-04-03 19:50:12.000000000 +0000
@@ -576,7 +576,7 @@
      only one-second resolution.  The code below should work, but it's
      not worth the hassle of debugging it on hosts where it fails.  */
 #if FILE_TIMESTAMP_HI_RES
-# if HAVE_CLOCK_GETTIME && defined CLOCK_REALTIME
+# if HAVE_CLOCK_GETTIME && defined CLOCK_REALTIME && !defined(__PPC64__)
   {
     struct timespec timespec;
     if (clock_gettime (CLOCK_REALTIME, &timespec) == 0)

