--- make.h.orig	Wed Jan  3 14:40:33 2007
+++ make.h	Wed Jan  3 14:40:36 2007
@@ -606,6 +606,5 @@
    NULL at the end of the directory--and _doesn't_ reset errno.  So, we have
    to do it ourselves here.  */
 
-#define ENULLLOOP(_v,_c)   do{ errno = 0; \
-                               while (((_v)=_c)==0 && errno==EINTR); }while(0)
+#define ENULLLOOP(_v,_c)   do{ errno = 0; (_v)=_c; }while((_v)==0 && errno==EINTR)
 
