--- pcre-7.1/Makefile.am.old	2007-04-18 10:07:56.000000000 +0100
+++ pcre-7.1/Makefile.am	2007-06-20 10:49:01.932157547 +0100
@@ -194,6 +194,7 @@
 # pcre_chartables.c, used unless --enable-rebuild-chartables is specified.
 EXTRA_DIST += pcre_printint.src pcre_chartables.c.dist
 
+libpcre_la_CFLAGS = -fPIC -DPIC
 libpcre_la_LDFLAGS = $(EXTRA_LIBPCRE_LDFLAGS)
 
 CLEANFILES += pcre_chartables.c
@@ -202,6 +203,7 @@
 lib_LTLIBRARIES += libpcreposix.la
 libpcreposix_la_SOURCES = \
   pcreposix.c
+libpcreposix_la_CFLAGS = -fPIC -DPIC
 libpcreposix_la_LDFLAGS = $(EXTRA_LIBPCREPOSIX_LDFLAGS)
 libpcreposix_la_LIBADD = libpcre.la
 
@@ -213,6 +215,7 @@
   pcrecpp.cc \
   pcre_scanner.cc \
   pcre_stringpiece.cc
+libpcrecpp_la_CXXFLAGS = -fPIC -DPIC
 libpcrecpp_la_LDFLAGS = $(EXTRA_LIBPCRECPP_LDFLAGS)
 libpcrecpp_la_LIBADD = libpcre.la
 
