diff -r -U3 apache-ant-1.7.0/src/main/org/apache/tools/ant/taskdefs/Jar.java apache-ant-1.7.0.my/src/main/org/apache/tools/ant/taskdefs/Jar.java
--- apache-ant-1.7.0/src/main/org/apache/tools/ant/taskdefs/Jar.java	2006-12-13 13:16:21.000000000 +0100
+++ apache-ant-1.7.0.my/src/main/org/apache/tools/ant/taskdefs/Jar.java	2008-06-16 17:58:08.000000000 +0200
@@ -397,7 +397,7 @@
            service = (Service) serviceIterator.next();
            //stolen from writeManifest
            super.zipFile(service.getAsStream(), zOut,
-                         "META-INF/service/" + service.getType(),
+                         "META-INF/services/" + service.getType(),
                          System.currentTimeMillis(), null,
                          ZipFileSet.DEFAULT_FILE_MODE);
         }
