--- ant
+++ ant
@@ -1,4 +1,4 @@
-#! /bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 
 # Licensed to the Apache Software Foundation (ASF) under one or more
 # contributor license agreements.  See the NOTICE file distributed with
@@ -18,6 +18,8 @@
 
 #   Edited for Gentoo Linux
 
+EPREFIX="@GENTOO_PORTAGE_EPREFIX@"
+
 # Extract launch and ant arguments, (see details below).
 ant_exec_args=
 no_config=false
@@ -48,9 +50,9 @@
   usejikes=$use_jikes_default
 else
   # load system-wide ant configuration (ONLY if ANT_HOME has NOT been set)
-  if [ -z "$ANT_HOME" -o "$ANT_HOME" = "/usr/share/ant" ]; then
-      if [ -f "/etc/ant.conf" ] ; then
-          . /etc/ant.conf
+  if [ -z "$ANT_HOME" -o "$ANT_HOME" = "${EPREFIX}/usr/share/ant" ]; then
+      if [ -f "${EPREFIX}/etc/ant.conf" ] ; then
+          . "${EPREFIX}"/etc/ant.conf
       fi
   fi
 
@@ -86,7 +88,7 @@
   exit 1
 fi
 
-ANT_HOME=/usr/share/ant-core
+ANT_HOME="${EPREFIX}"/usr/share/ant-core
 
 # set ANT_LIB location
 ANT_LIB="${ANT_HOME}/lib"
@@ -121,11 +123,11 @@
 if [[ "${ANT_TASKS}" == "all" ]]; then
 	ANT_TASKS=""
 	# but only if it exists
-	if [[ -d /usr/share/ant/tasks ]]; then
-		ANT_TASKS="${ANT_TASKS} "/usr/share/ant/tasks/*
+	if [[ -d "${EPREFIX}"/usr/share/ant/tasks ]]; then
+		ANT_TASKS="${ANT_TASKS} ${EPREFIX}"/usr/share/ant/tasks/*
 	fi
-	if [[ -d /usr/share/ant/tasks-1.8.0 ]]; then
-		ANT_TASKS="${ANT_TASKS} "/usr/share/ant/tasks-1.8.0/*
+	if [[ -d "${EPREFIX}"/usr/share/ant/tasks-1.8.0 ]]; then
+		ANT_TASKS="${ANT_TASKS} ${EPREFIX}"/usr/share/ant/tasks-1.8.0/*
 	fi
 # if set to "none", make ANT_TASKS empty list
 elif [[ "${ANT_TASKS}" == "none" ]]; then
@@ -158,7 +160,7 @@
 
 if [[ -n "${TOOLS_JAR}" ]] ; then
   LOCALCLASSPATH="$LOCALCLASSPATH:${TOOLS_JAR}"
-else
+elif [[ $(java-config -f) != apple-jdk-bin* ]] ; then
   echo "Warning: Unable to determine tools.jar location."
   echo "  If build fails because sun.* classes could not be found,"
   echo "  Make sure you are using a JDK, not JRE as your user/system VM."
