Index: cairo-1.4.8/src/cairo-directfb-surface.c
===================================================================
--- cairo-1.4.8.orig/src/cairo-directfb-surface.c
+++ cairo-1.4.8/src/cairo-directfb-surface.c
@@ -1572,7 +1572,7 @@ static cairo_surface_backend_t cairo_dir
         NULL, /* show_glyphs */
 #endif
         NULL, /* snapshot */
-	_cairo_directfb_is_similar,
+	_cairo_directfb_surface_is_similar,
 	NULL /* reset */
 };
 
