--- src/cairo-path-fixed.c.orig	2007-12-27 00:01:13.000000000 +0000
+++ src/cairo-path-fixed.c	2007-12-27 00:02:10.000000000 +0000
@@ -415,8 +415,8 @@
 	buf->num_points = 0;
 	buf->buf_size = buf_size;
 
-	buf->op = (cairo_path_op_t *) (buf + 1);
-	buf->points = (cairo_point_t *) (buf->op + buf_size);
+	buf->points = (cairo_point_t *) (buf + 1);
+	buf->op = (cairo_path_op_t *) (buf->points + 2 * buf_size);
     }
 
     return buf;