--- converter/other/jpeg2000/jpeg2ktopam.c
+++ converter/other/jpeg2000/jpeg2ktopam.c
@@ -9,7 +9,9 @@
 *****************************************************************************/
 
 #define _BSD_SOURCE 1      /* Make sure strdup() is in string.h */
+#if 0
 #define _XOPEN_SOURCE 500 /* Make sure strdup() is in string.h */
+#endif
     /* In 2014.09, this was _XOPEN_SOURCE 600, with a comment saying it was
        necessary to make <inttypes.h> define int_fast32_t, etc. on AIX.
        <jasper/jasper.h> does use int_fast32_t and does include <inttypes.h>,
--- converter/other/jpeg2000/pamtojpeg2k.c
+++ converter/other/jpeg2000/pamtojpeg2k.c
@@ -9,7 +9,9 @@
 *****************************************************************************/
 
 #define _BSD_SOURCE 1    /* Make sure strdup() is in string.h */
+#if 0
 #define _XOPEN_SOURCE 500 /* Make sure strdup() is in string.h */
+#endif
     /* In 2014.09, this was _XOPEN_SOURCE 600, with a comment saying it was
        necessary to make <inttypes.h> define int_fast32_t, etc. on AIX.
        <jasper/jasper.h> does use int_fast32_t and does include <inttypes.h>,
