--- converter/other/xwdtopnm.c
+++ converter/other/xwdtopnm.c
@@ -852,7 +852,7 @@
 static void
 pixelReader_term(pixelReader * const pixelReaderP) {
 
-    uint remainingByteCount;
+    unsigned int remainingByteCount;
 
     if (pixelReaderP->nBitsLeft > 0)
         pm_message("Warning: %u unused bits left in the pixel reader "
--- generator/ppmrough.c
+++ generator/ppmrough.c
@@ -12,6 +12,7 @@
 
 #include <math.h>
 #include <sys/time.h>
+#include <stdlib.h>
 #include "ppm.h"
 #include "shhopt.h"
 
