http://bugs.gentoo.org/137871

--- converter/other/Makefile
+++ converter/other/Makefile
@@ -7,14 +7,6 @@
 
 include $(BUILDDIR)/Makefile.config
 
-ifeq ($(shell xml2-config --version),)
-  XML2_LIBS=NONE
-  XML2_CFLAGS=NONE
-else
-  XML2_LIBS=$(shell xml2-config --libs)
-  XML2_CFLAGS=$(shell xml2-config --cflags)
-endif
-
 SUBDIRS = jbig pnmtopalm jpeg2000 cameratopam pamtosvg
 ifneq ($(BUILD_FIASCO), N)
   SUBDIRS += fiasco
--- Makefile.config.in
+++ Makefile.config.in
@@ -419,6 +419,16 @@
 ZHDR_DIR = 
 #ZLIB = libz.so
 
+# The XML2 library
+
+ifeq ($(shell xml2-config --version),)
+  XML2_LIBS=NONE
+  XML2_CFLAGS=NONE
+else
+  XML2_LIBS=$(shell xml2-config --libs)
+  XML2_CFLAGS=$(shell xml2-config --cflags)
+endif
+
 # The JBIG lossless image compression library (aka JBIG-KIT):
 
 JBIGLIB = $(BUILDDIR)/converter/other/jbig/libjbig.a
