--- psycopg/config.h.orig       2008-09-16 09:54:26.000000000 -0600
+++ psycopg/config.h            2008-09-16 09:55:26.000000000 -0600
@@ -129,7 +129,7 @@
 #define inline
 #endif
 
-#if defined(__FreeBSD__) || (defined(_WIN32) && !defined(__GNUC__)) || defined(__sun__) || defined(sun)
+#if (defined(_WIN32) && !defined(__GNUC__)) || defined(__sun__) || defined(sun)
 /* what's this, we have no round function either? */
 static double round(double num)
 {
