--- libtool-1.3.5/ltmain.in.sopwith	Thu Jul 13 17:34:43 2000
+++ libtool-1.3.5/ltmain.in	Thu Jul 13 17:36:31 2000
@@ -1656,7 +1656,7 @@
 
 	# Check that each of the things are valid numbers.
 	case "$current" in
-	0 | [1-9] | [1-9][0-9]*) ;;
+	[0-9]*) ;;
 	*)
 	  $echo "$modename: CURRENT \`$current' is not a nonnegative integer" 1>&2
 	  $echo "$modename: \`$vinfo' is not valid version information" 1>&2
@@ -1665,7 +1665,7 @@
 	esac
 
 	case "$revision" in
-	0 | [1-9] | [1-9][0-9]*) ;;
+	[0-9]*) ;;
 	*)
 	  $echo "$modename: REVISION \`$revision' is not a nonnegative integer" 1>&2
 	  $echo "$modename: \`$vinfo' is not valid version information" 1>&2
@@ -1674,7 +1674,7 @@
 	esac
 
 	case "$age" in
-	0 | [1-9] | [1-9][0-9]*) ;;
+	[0-9]*) ;;
 	*)
 	  $echo "$modename: AGE \`$age' is not a nonnegative integer" 1>&2
 	  $echo "$modename: \`$vinfo' is not valid version information" 1>&2
