--- configure.ac.orig	2008-07-10 11:29:58.506230457 +0100
+++ configure.ac	2008-07-10 11:49:45.171417438 +0100
@@ -135,7 +135,7 @@
 AC_PROG_CC
 AX_COMPILER_VENDOR
 AC_PROG_CC_STDC
-AMX_PROG_AS     dnl AM_PROG_AS breaks at least one version of libtool
+AM_PROG_AS
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
@@ -229,6 +229,8 @@
 fi
 fi
 AC_SUBST(SIMD_CFLAGS)
+# pass simd flags to preprocessor/compiler for proper intrinsics detection.
+CPPFLAGS="$CPPFLAGS $SIMD_CFLAGS"
 
 dnl Checks for header files.
 AC_HEADER_STDC
@@ -362,10 +364,10 @@
 THREADLIBS=""
 if test "$enable_threads" = "yes"; then
 	if test "$with_openmp"x != nox; then
-		AX_OPENMP([THREADLIBS=" "
+		AX_OPENMP([THREADLIBS="$OPENMP_LIBS"
 		  AC_DEFINE(USING_OPENMP_THREADS, 1, [Define if we have and are using OpenMP multithreading directives])
                   CFLAGS="$CFLAGS $OPENMP_CFLAGS"],
-		  [AC_MSG_ERROR([don't know how to enable OpenMP])])
+		  [AC_MSG_WARN([don't know how to enable OpenMP, reverting to POSIX threads])])
 	fi
 
 	# POSIX threads, the default choice:
