diff -ru xfishtank-2.1tp.orig/Makefile xfishtank-2.1tp/Makefile
--- xfishtank-2.1tp.orig/Makefile	2009-02-02 14:52:24 +0100
+++ xfishtank-2.1tp/Makefile	2009-02-02 15:01:09 +0100
@@ -131,7 +131,7 @@
     CXXDEBUGFLAGS = -O2 -fno-strength-reduce -m486
  CXXEXTRA_DEFINES =
 CXXEXTRA_INCLUDES =
-   CXXSTD_DEFINES = -Dlinux -D__i386__ -D_POSIX_SOURCE -D_BSD_SOURCE -D_SVID_SOURCE -DX_LOCALE
+   CXXSTD_DEFINES = -DWIN32 -D__i386__ -D_POSIX_SOURCE -D_BSD_SOURCE -D_SVID_SOURCE -DX_LOCALE
        CXXOPTIONS =
       CXXINCLUDES = $(INCLUDES) $(TOP_INCLUDES) $(CXXEXTRA_INCLUDES)
        CXXDEFINES = $(CXXINCLUDES) $(CXXSTD_DEFINES) $(THREADS_CXXDEFINES) $(CXXEXTRA_DEFINES)
@@ -175,8 +175,8 @@
             LATEX = latex
 
      STD_INCLUDES =
-  STD_CPP_DEFINES = -traditional -Dlinux -D__i386__ -D_POSIX_SOURCE -D_BSD_SOURCE -D_SVID_SOURCE -DX_LOCALE
-      STD_DEFINES = -Dlinux -D__i386__ -D_POSIX_SOURCE -D_BSD_SOURCE -D_SVID_SOURCE -DX_LOCALE
+  STD_CPP_DEFINES = -traditional -DWIN32 -D__i386__ -D_POSIX_SOURCE -D_BSD_SOURCE -D_SVID_SOURCE -DX_LOCALE
+      STD_DEFINES = -DWIN32 -D__i386__ -D_POSIX_SOURCE -D_BSD_SOURCE -D_SVID_SOURCE -DX_LOCALE
  EXTRA_LOAD_FLAGS = -Wl,-rpath,$(SHLIBDIR)
   EXTRA_LDOPTIONS =
   EXTRA_LIBRARIES =
@@ -529,23 +529,23 @@
 LOCAL_LIBRARIES = $(XLIB)
         DEPLIBS = $(DEPXLIB)
   SYS_LIBRARIES = -lm
-    CDEBUGFLAGS = -g
+    CDEBUGFLAGS = 
 
            SRCS = xfish.c medcut.c read.c gifread.c
            OBJS = xfish.o medcut.o read.o gifread.o
 
- PROGRAM = xfishtank
+ PROGRAM = xfishtank.exe
 
-all:: xfishtank
+all:: xfishtank.exe
 
-xfishtank: $(OBJS) $(DEPLIBS)
+xfishtank.exe: $(OBJS) $(DEPLIBS)
 	$(RM) $@
 	$(CCLINK) -o $@ $(LDOPTIONS) $(OBJS) $(LOCAL_LIBRARIES) $(LDLIBS)  $(EXTRA_LOAD_FLAGS)
 
-install:: xfishtank
+install:: xfishtank.exe
 	@if [ -d $(DESTDIR)$(BINDIR) ]; then set +x; \
 	else (set -x; $(MKDIRHIER) $(DESTDIR)$(BINDIR)); fi
-	$(INSTALL) $(INSTALLFLAGS) $(INSTPGMFLAGS)  xfishtank $(DESTDIR)$(BINDIR)/xfishtank
+	$(INSTALL) $(INSTALLFLAGS) $(INSTPGMFLAGS)  xfishtank.exe $(DESTDIR)$(BINDIR)/xfishtank.exe
 
 install.man:: xfishtank.man
 	@if [ -d $(DESTDIR)$(MANDIR) ]; then set +x; \
@@ -561,7 +561,7 @@
 	$(LINT) $(LINTFLAGS) $(FILE) $(LINTLIBS)
 
 clean::
-	$(RM) xfishtank
+	$(RM) xfishtank.exe
 
 xfishy.h: makeh FishList
 	./makeh > xfishy.h
diff -ru xfishtank-2.1tp.orig/read.c xfishtank-2.1tp/read.c
--- xfishtank-2.1tp.orig/read.c	2009-02-02 14:52:24 +0100
+++ xfishtank-2.1tp/read.c	2009-02-02 14:59:34 +0100
@@ -7,6 +7,11 @@
 #define DEF_WHITE       WhitePixel(Dpy, DefaultScreen(Dpy))
 #define	MAX_LINE	81
 
+#ifdef WIN32
+# define index strchr
+# define rindex strrchr
+#endif
+
 
 extern unsigned char *ReadGifBitmap();
 
diff -ru xfishtank-2.1tp.orig/xfish.c xfishtank-2.1tp/xfish.c
--- xfishtank-2.1tp.orig/xfish.c	2009-02-02 14:52:24 +0100
+++ xfishtank-2.1tp/xfish.c	2009-02-02 14:57:26 +0100
@@ -1336,6 +1336,7 @@
 void 
 init_signals()
 {
+#ifndef WIN32
 	int ret;
 #ifdef linux
 	signal(SIGUSR1, toggle_secure);
@@ -1364,6 +1365,7 @@
 #endif
 #endif /* MOTOROLA */
 #endif /* LINUX */
+#endif /* WIN32 */
 }
 
 
