--- t1lib-5.1.1/examples/subset.c	2005-05-07 23:09:04.000000000 +0200
+++ t1lib-5.1.1-qa/examples/subset.c	2007-09-26 21:49:19.000000000 +0200
@@ -31,6 +31,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 /* Note: We include t1lib.h from lib/t1lib. That way the objectfile does only
    need to be rebuild when the header itself changes and not each time the
    library has been recompiled */
