diff -ur xscreensaver-5.06.orig/driver/screens.c xscreensaver-5.06/driver/screens.c
--- xscreensaver-5.06.orig/driver/screens.c	2008-07-07 12:03:00.000000000 +0300
+++ xscreensaver-5.06/driver/screens.c	2008-08-05 00:49:54.000000000 +0300
@@ -563,7 +563,7 @@
   if (! monitors) monitors = vidmode_scan_monitors (si->dpy);
 # endif
 
-# ifdef HAVE_XF86VMODE
+# ifdef HAVE_XINERAMA
   if (! monitors) monitors = xinerama_scan_monitors (si->dpy);
 # endif
 
