* grobian@gentoo.org -- also pick up libiconv correctly

--- configure.ac.orig	2007-10-01 19:22:19 +0200
+++ configure.ac	2007-10-01 19:24:58 +0200
@@ -458,8 +458,8 @@
 AC_SEARCH_LIBS( connect, socket, , [AC_MSG_ERROR([Cannot find the socket library, needed to build this package.] )] )
 AC_SEARCH_LIBS( nanosleep, rt, , [AC_MSG_ERROR([Cannot find the rt library, needed to build this package.] )] )
 AC_SEARCH_LIBS( setupterm, [ncurses curses], , [AC_MSG_ERROR([Could not find a curses implementation, needed to build fish])] )
-AC_SEARCH_LIBS( iconv_open, iconv, , [AC_MSG_ERROR([Could not find an iconv implementation, needed to build fish])] )
-
+AM_ICONV
+LIBS="$LIBS $LIBICONV"
 
 #
 # Check presense of various header files
