Index: configure.in
===================================================================
--- configure.in	(revision 845)
+++ configure.in	(working copy)
@@ -26,7 +26,11 @@
 AC_TYPE_SIZE_T
 AC_TYPE_UID_T
 
+AC_DEFUN(GTK_DOC_CHECK, [])
 GTK_DOC_CHECK
+AM_CONDITIONAL(ENABLE_GTK_DOC, 0)
+AM_CONDITIONAL(GTK_DOC_USE_LIBTOOL, 0)
+
 DISTCHECK_CONFIGURE_FLAGS="--enable-gtk-doc"
 AC_SUBST(DISTCHECK_CONFIGURE_FLAGS)
 
