diff -ru -x '*.Po' -x '*.Plo' gnome-keyring-2.22.1.orig/common/gkr-unix-credentials.c gnome-keyring-2.22.1/common/gkr-unix-credentials.c
--- gnome-keyring-2.22.1.orig/common/gkr-unix-credentials.c	2008-04-18 11:33:25 +0200
+++ gnome-keyring-2.22.1/common/gkr-unix-credentials.c	2008-04-18 11:54:31 +0200
@@ -39,8 +39,10 @@
 int
 gkr_unix_credentials_read (int sock, pid_t *pid, uid_t *uid)
 {
+#if defined(HAVE_CMSGCRED) || defined(LOCAL_CREDS)
 	struct msghdr msg;
 	struct iovec iov;
+#endif
 	char buf;
 	int ret;
 	
@@ -65,6 +67,7 @@
 	*pid = 0;
 	*uid = 0;
 	
+#if defined(HAVE_CMSGCRED) || defined(LOCAL_CREDS)
 	/* If LOCAL_CREDS are used in this platform, they have already been
 	 * initialized by init_connection prior to sending of the credentials
 	 * byte we receive below. */
@@ -76,14 +79,17 @@
 	msg.msg_iov = &iov;
 	msg.msg_iovlen = 1;
 	
-#if defined(HAVE_CMSGCRED) || defined(LOCAL_CREDS)
 	memset (&cmsg, 0, sizeof (cmsg));
 	msg.msg_control = (caddr_t) &cmsg;
 	msg.msg_controllen = cmsgspace;
 #endif
 
  again:
+#if defined(HAVE_CMSGCRED) || defined(LOCAL_CREDS)
 	ret = recvmsg (sock, &msg, 0);
+#else
+	ret = read(sock, &buf, 0);
+#endif
 
  	if (ret < 0) {
 		if (errno == EINTR)
diff -ru -x '*.Po' -x '*.Plo' gnome-keyring-2.22.1.orig/pkcs11/gkr-pkcs11-daemon.c gnome-keyring-2.22.1/pkcs11/gkr-pkcs11-daemon.c
--- gnome-keyring-2.22.1.orig/pkcs11/gkr-pkcs11-daemon.c	2008-04-18 11:33:42 +0200
+++ gnome-keyring-2.22.1/pkcs11/gkr-pkcs11-daemon.c	2008-04-18 12:02:45 +0200
@@ -96,6 +96,10 @@
 		g_hash_table_remove (session_workers, worker);
 }
 
+#ifdef __INTERIX
+typedef int socklen_t;
+#endif
+
 static gboolean
 handle_new_connection (GIOChannel *channel, GIOCondition cond, gpointer callback_data)
 {
