diff -ru -x '*.Plo' -x '*.Po' gnome-keyring-2.22.1.orig/common/gkr-secure-memory.c gnome-keyring-2.22.1/common/gkr-secure-memory.c
--- gnome-keyring-2.22.1.orig/common/gkr-secure-memory.c	2008-04-14 11:53:56 +0200
+++ gnome-keyring-2.22.1/common/gkr-secure-memory.c	2008-04-14 11:54:10 +0200
@@ -475,7 +475,7 @@
 	DEBUG_ALLOC ("gkr-secure-memory: freed block ", sz);
 	
 #else
-	ASSERT (FALSE);
+	ASSERT (0);
 #endif
 }
 
