--- cedet-1.0pre4/semantic/Makefile~	2007-06-06 04:00:10.000000000 +0200
+++ cedet-1.0pre4/semantic/Makefile	2007-06-27 00:45:06.000000000 +0200
@@ -114,7 +114,7 @@
 	"$(EMACS)" -batch --no-site-file -l $@-compile-script -f batch-byte-compile $^
 
 .PHONY:wisent
-wisent:
+wisent: bovinator
 	cd wisent; $(MAKE)
 
 .PHONY:Tests
--- cedet-1.0pre4/semantic/document.el~	2007-06-06 04:00:09.000000000 +0200
+++ cedet-1.0pre4/semantic/document.el	2007-09-07 00:40:24.000000000 +0200
@@ -41,9 +41,6 @@
 (require 'document-vars)
 
 ;; XEmacs change: needed to define macros at compile time.
-(condition-case nil
-    (require 'ede)
-  (error nil))
 (require 'semantic)
 (require 'semantic-util)
 
