--- mailwrapper.c
+++ mailwrapper.c
@@ -39,8 +39,8 @@
 #include <syslog.h>
 #include <unistd.h>
 
-#define _PATH_MAILERCONF	"/etc/mail/mailer.conf"
-#define _PATH_DEFAULTMTA	"/usr/sbin/sendmail.sendmail"
+#define _PATH_MAILERCONF	"@GENTOO_PORTAGE_EPREFIX@/etc/mail/mailer.conf"
+#define _PATH_DEFAULTMTA	"@GENTOO_PORTAGE_EPREFIX@/usr/sbin/sendmail.sendmail"
 
 char* fparseln(FILE*, size_t*, size_t*, const char[3], int);
 
