--- ncurses-5.3/misc/terminfo.src.orig	2002-10-05 20:40:53.000000000 +0200
+++ ncurses-5.3/misc/terminfo.src	2003-05-23 15:50:57.000000000 +0200
@@ -2922,8 +2922,9 @@
 
 # This is xterm for ncurses.
 xterm|xterm terminal emulator (X Window System), 
-	use=xterm-r6, 
+#	use=xterm-r6, 
 #	use=xterm-xfree86, 
+	use=xterm-xf86-v43, 
 
 # These entries allow access to the X titlebar and icon name as a status line. 
 # Note that twm (and possibly window managers descended from it such as tvtwm, 
