http://bugs.gentoo.org/show_bug.cgi?id=262118

--- test/Makefile.dist	2009-02-25 15:26:35.226306680 +0000
+++ test/Makefile	2009-02-25 16:33:38.145363000 +0000
@@ -241,7 +241,7 @@ $(destdir)demo_termcap$x: $(MODEL)/demo_
 	$(LINK) -o $@ $(MODEL)/demo_termcap$o $(LDFLAGS_TINFO)
 
 $(destdir)ditto$x: $(MODEL)/ditto$o $(LOCAL_LIBS)
-	$(LINK) -o $@ $(MODEL)/ditto$o $(LDFLAGS_THREADS)
+	$(LINK) -o $@ $(MODEL)/ditto$o $(MODEL)/use_screen$o $(LDFLAGS_THREADS)
 
 $(destdir)dots$x: $(MODEL)/dots$o $(LOCAL_LIBS)
 	$(LINK) -o $@ $(MODEL)/dots$o $(LDFLAGS_TINFO)
@@ -310,7 +310,7 @@ $(destdir)railroad$x: $(MODEL)/railroad$
 	$(LINK) -o $@ $(MODEL)/railroad$o $(LDFLAGS_TINFO)
 
 $(destdir)rain$x: $(MODEL)/rain$o $(LOCAL_LIBS)
-	$(LINK) -o $@ $(MODEL)/rain$o $(LDFLAGS_THREADS)
+	$(LINK) -o $@ $(MODEL)/rain$o $(MODEL)/use_window$o $(LDFLAGS_THREADS)
 
 $(destdir)redraw$x: $(MODEL)/redraw$o $(LOCAL_LIBS)
 	$(LINK) -o $@ $(MODEL)/redraw$o $(LDFLAGS_CURSES)
@@ -337,7 +337,7 @@ $(destdir)test_inwstr$x: $(MODEL)/test_i
 	$(LINK) -o $@ $(MODEL)/test_inwstr$o $(LDFLAGS_CURSES)
 
 $(destdir)test_opaque$x: $(MODEL)/test_opaque$o $(LOCAL_LIBS)
-	$(LINK) -o $@ $(MODEL)/test_opaque$o $(LDFLAGS_CURSES)
+	$(LINK) -o $@ $(MODEL)/test_opaque$o $(TEST_ARGS) $(LDFLAGS_CURSES)
 
 $(destdir)testaddch$x: $(MODEL)/testaddch$o $(LOCAL_LIBS)
 	$(LINK) -o $@ $(MODEL)/testaddch$o $(LDFLAGS_CURSES)
@@ -352,7 +352,7 @@ $(destdir)view$x: $(MODEL)/view$o $(LOCA
 	$(LINK) -o $@ $(MODEL)/view$o $(LDFLAGS_CURSES)
 
 $(destdir)worm$x: $(MODEL)/worm$o $(LOCAL_LIBS)
-	$(LINK) -o $@ $(MODEL)/worm$o $(LDFLAGS_THREADS)
+	$(LINK) -o $@ $(MODEL)/worm$o $(MODEL)/use_window$o $(LDFLAGS_THREADS)
 
 $(destdir)xmas$x: $(MODEL)/xmas$o $(LOCAL_LIBS)
 	$(LINK) -o $@ $(MODEL)/xmas$o $(LDFLAGS_CURSES)
