diff -urN ncurses-5.3/misc/run_tic.in ncurses-5.3.share/misc/run_tic.in
--- ncurses-5.3/misc/run_tic.in	2001-10-14 08:44:33.000000000 +0800
+++ ncurses-5.3.share/misc/run_tic.in	2004-02-21 15:46:09.000000000 +0800
@@ -81,7 +81,8 @@
 umask 022
 
 # Construct the name of the old (obsolete) pathname, e.g., /usr/lib/terminfo.
-TICDIR=`echo $TERMINFO | sed -e 's%/share/%/lib/%'`
+# make sure only last share directory in the $TERMINFO path is changed to lib
+TICDIR=`echo $TERMINFO | sed -e 's%/share/terminfo$%/lib/terminfo%'`
 
 # Remove the old terminfo stuff; we don't care if it existed before, and it
 # would generate a lot of confusing error messages if we tried to overwrite it.
