diff -Naurp linux_logo-5.10-orig//po/Makefile linux_logo-5.10/po/Makefile
--- linux_logo-5.10-orig//po/Makefile	2010-08-28 23:07:37.839925950 +0200
+++ linux_logo-5.10/po/Makefile	2010-08-28 23:12:49.077926707 +0200
@@ -32,9 +32,9 @@ install: $(CATALOGS)
 	msgfmt -o $@ $< 
 
 update: $(PACKAGE).pot
-       for n in $(CATALOGS); do \
-       l=basename $$n .mo; \
-       l=$$l".po";\
-       mv -f $$l $$l".bac";\
-       msgmerge -o $$l $$l".bac" $(PACKAGE).pot; \
-       done
+	for n in $(CATALOGS); do \
+	l=`basename $$n .mo`; \
+	l=$$l".po";\
+	mv -f $$l $$l".bac";\
+	msgmerge -o $$l $$l".bac" $(PACKAGE).pot; \
+	done
