diff -ru linux_logo-5.04.orig/configure linux_logo-5.04/configure
--- linux_logo-5.04.orig/configure	2009-01-20 11:50:30 +0100
+++ linux_logo-5.04/configure	2009-01-20 11:51:02 +0100
@@ -78,12 +78,12 @@
 echo "  Install prefix:  "$PREFIX
 echo "  Install program: "$INSTALL
 echo "  xgettext:   "$XGETTEXT
-if [ $INTL_MISSING -ne 0 ]; then
-   echo "  libintl.h not found, disabling i18n support"
+#if [ $INTL_MISSING -ne 0 ]; then
+#   echo "  libintl.h not found, disabling i18n support"
    USE_I18N=0
-else
-   USE_I18N=1
-fi
+#else
+#   USE_I18N=1
+#fi
 
 echo "  libsysinfo: "$LIBSYSINFO
 echo "  C Compiler : "$CC
