diff -ruN linux_logo-5.04.orig/libsysinfo-0.2.1/Interix/Makefile linux_logo-5.04/libsysinfo-0.2.1/Interix/Makefile
--- linux_logo-5.04.orig/libsysinfo-0.2.1/Interix/Makefile	1970-01-01 01:00:00 +0100
+++ linux_logo-5.04/libsysinfo-0.2.1/Interix/Makefile	2009-01-20 11:41:49 +0100
@@ -0,0 +1,6 @@
+all:
+	touch dummy.o
+
+clean:	
+	rm -f *~ *.o
+	
diff -ruN linux_logo-5.04.orig/linux_logo.c linux_logo-5.04/linux_logo.c
--- linux_logo-5.04.orig/linux_logo.c	2009-01-20 11:39:01 +0100
+++ linux_logo-5.04/linux_logo.c	2009-01-20 11:46:26 +0100
@@ -260,6 +260,7 @@
     int allocated_lines=10;
 
        /* Get the system info */
+	#if !defined(__INTERIX) && !defined(_WIN32)
     get_os_info(&os_info);
     get_cpu_info(&cpu_info);
 
@@ -431,6 +432,7 @@
        x++;
     }
     *string=temp_sysinfo;
+	#endif
     return line;
 }
 
