diff -urpN debianutils-1.16.7/run-parts.c debianutils-1.16.7.azarah/run-parts.c
--- debianutils-1.16.7/run-parts.c	2003-05-19 00:54:27.000000000 +0200
+++ debianutils-1.16.7.azarah/run-parts.c	2003-05-19 00:57:02.000000000 +0200
@@ -55,10 +55,10 @@ void error (char *format,...)
 void version ()
 {
   fprintf (stderr, "Debian GNU/Linux run-parts program, version " VERSION
-"\nCopyright (C) 1994 Ian Jackson, Copyright (C) 1996 Jeff Noxon.
-Copyright (C) 1996,1997,1998,1999 Guy Maor
-This is free software; see the GNU General Public License version 2
-or later for copying conditions.  There is NO warranty.
+"\nCopyright (C) 1994 Ian Jackson, Copyright (C) 1996 Jeff Noxon.\n\
+Copyright (C) 1996,1997,1998,1999 Guy Maor\n\
+This is free software; see the GNU General Public License version 2\n\
+or later for copying conditions.  There is NO warranty.\n\
 ");
   exit (0);
 }
@@ -66,14 +66,14 @@ or later for copying conditions.  There 
 
 void usage ()
 {
-  fprintf (stderr, "Usage: run-parts [OPTION]... DIRECTORY
-  --test          print script names which would run, but don't run them.
-  --verbose       print script names before running them.
-  --report        print script names if they produce output.
-  --umask=UMASK   sets umask to UMASK (octal), default is 022.
-  --arg=ARGUMENT  pass ARGUMENT to scripts, use once for each argument.
-  --version       output version information and exit.
-  --help          display this help and exit.
+  fprintf (stderr, "Usage: run-parts [OPTION]... DIRECTORY\n\
+  --test          print script names which would run, but don't run them.\n\
+  --verbose       print script names before running them.\n\
+  --report        print script names if they produce output.\n\
+  --umask=UMASK   sets umask to UMASK (octal), default is 022.\n\
+  --arg=ARGUMENT  pass ARGUMENT to scripts, use once for each argument.\n\
+  --version       output version information and exit.\n\
+  --help          display this help and exit.\n\
 ");
   exit(0);
 }
diff -urpN debianutils-1.16.7/tempfile.c debianutils-1.16.7.azarah/tempfile.c
--- debianutils-1.16.7/tempfile.c	2002-09-28 21:17:03.000000000 +0200
+++ debianutils-1.16.7.azarah/tempfile.c	2003-05-19 00:57:15.000000000 +0200
@@ -17,7 +17,7 @@ usage (int status)
     fprintf(stderr, "Try `%s --help' for more information.\n", progname);
   else
     printf("Usage: %s [OPTION]\n\
-Create a temporary file in a safe manner.
+Create a temporary file in a safe manner.\n\
 \n\
 -d, --directory=DIR  place temporary file in DIR\n\
 -p, --prefix=STRING  set temporary file's prefix to STRING\n\
