http://bugs.gentoo.org/show_bug.cgi?id=282086

--- unzpriv.h.dist	2009-08-20 11:24:17.386734120 +0100
+++ unzpriv.h	2009-08-20 11:29:37.592325480 +0100
@@ -1904,7 +1904,7 @@
 
 #ifdef ZIP64_SUPPORT
 # ifndef Z_UINT8_DEFINED
-#   if (defined(__GNUC__) || defined(__hpux) || defined(__SUNPRO_C))
+#   if (defined(__GNUC__) || defined(__hpux) || defined(__SUNPRO_C) || defined(__sgi))
   typedef unsigned long long    z_uint8;
 #   else
   typedef unsigned __int64      z_uint8;
--- unix/unxcfg.h.dist	2009-08-20 11:36:17.658689640 +0100
+++ unix/unxcfg.h	2009-08-20 11:38:31.458220280 +0100
@@ -42,8 +42,10 @@
   /* These have to be before any include that sets types so the large file
      versions of the types are set in the includes */
 
-# define _LARGEFILE_SOURCE      /* some OSes need this for fseeko */
-# define _LARGEFILE64_SOURCE
+# ifndef __sgi
+#  define _LARGEFILE_SOURCE      /* some OSes need this for fseeko */
+#  define _LARGEFILE64_SOURCE
+# endif
 # define _FILE_OFFSET_BITS 64   /* select default interface as 64 bit */
 # define _LARGE_FILES           /* some OSes need this for 64-bit off_t */
 # define __USE_LARGEFILE64
