diff -Naurp nmap-4.68-orig/nmap.cc nmap-4.68/nmap.cc
--- nmap-4.68-orig/nmap.cc	2008-07-02 22:37:00.000000000 +0200
+++ nmap-4.68/nmap.cc	2008-07-02 22:37:20.000000000 +0200
@@ -1899,7 +1899,9 @@ void nmap_free_mem() {
   if (o.dns_servers) free(o.dns_servers);
   if (o.extra_payload) free(o.extra_payload);
   if (o.ipoptions) free(o.ipoptions);
+#ifndef NOLUA
   free(o.scriptargs);
+#endif
 }
 
 /* Reads in a (normal or machine format) Nmap log file and gathers enough
diff -Naurp nmap-4.68-orig/NmapOps.cc nmap-4.68/NmapOps.cc
--- nmap-4.68-orig/NmapOps.cc	2008-07-02 22:37:00.000000000 +0200
+++ nmap-4.68/NmapOps.cc	2008-07-02 22:38:37.000000000 +0200
@@ -457,9 +457,11 @@ void NmapOps::ValidateOptions() {
   if(ipoptions && osscan)
     error("WARNING: Ip options are NOT used while OS scanning!");
 
+#ifndef NOLUA
   /* Make sure nmap.registry.args is available (even if it's empty) */
   if (!scriptargs)
     scriptargs = strdup("");
+#endif
 }
 
 void NmapOps::setMaxOSTries(int mot) {
diff -Naurp nmap-4.68-orig/nse_main.h nmap-4.68/nse_main.h
--- nmap-4.68-orig/nse_main.h	2008-07-02 22:37:00.000000000 +0200
+++ nmap-4.68/nse_main.h	2008-07-02 22:39:54.000000000 +0200
@@ -1,11 +1,13 @@
 #ifndef NMAP_LUA_H
 #define NMAP_LUA_H
 
+#ifndef NOLUA
 extern "C" {
   #include "lua.h"
   #include "lualib.h"
   #include "lauxlib.h"
 }
+#endif
 
 #include <vector>
 #include <list>
diff -Naurp nmap-4.68-orig/portlist.cc nmap-4.68/portlist.cc
--- nmap-4.68-orig/portlist.cc	2008-07-02 22:37:00.000000000 +0200
+++ nmap-4.68/portlist.cc	2008-07-02 22:38:16.000000000 +0200
@@ -150,11 +150,13 @@ Port::~Port() {
    free(serviceprobe_service);
  if (serviceprobe_fp)
    free(serviceprobe_fp);
+#ifndef NOLUA
  if (o.script) {
    ScriptResults::iterator sriter;
    for (sriter = scriptResults.begin(); sriter != scriptResults.end(); sriter++)
      free((*sriter).id);
  }
+#endif
 }
 
 // Uses the sd->{product,version,extrainfo} if available to fill
diff -Naurp nmap-4.68-orig/Target.cc nmap-4.68/Target.cc
--- nmap-4.68-orig/Target.cc	2008-07-02 22:37:00.000000000 +0200
+++ nmap-4.68/Target.cc	2008-07-02 22:39:02.000000000 +0200
@@ -179,11 +179,13 @@ void Target::FreeInternal() {
 
   if (FPR) delete FPR;
 
+#ifndef NOLUA
   if (o.script) {
     ScriptResults::iterator sriter;
     for (sriter = scriptResults.begin(); sriter != scriptResults.end(); sriter++)
       free((*sriter).id);
   }
+#endif
 }
 
 /*  Creates a "presentation" formatted string out of the IPv4/IPv6 address.
