diff -Naurp nmap-4.20-orig/osscan.cc nmap-4.20/osscan.cc
--- nmap-4.20-orig/osscan.cc	2007-05-06 17:02:42.000000000 +0200
+++ nmap-4.20/osscan.cc	2007-05-06 17:03:03.000000000 +0200
@@ -824,8 +824,7 @@ static FingerPrint *get_fingerprint(Targ
 	    if (si->seqs[seq_response_num] == 0) {
 	      /* New response found! */
 	      si->responses++;
-	      //	      si->seqs[seq_response_num] = ntohl(tcp->th_seq); /* TCP ISN */
-	      si->seqs[seq_response_num] = fake_seqs[seq_response_num]; /* TCP ISN */
+		  si->seqs[seq_response_num] = ntohl(tcp->th_seq); /* TCP ISN */
 	      si->ipids[seq_response_num] = ntohs(ip->ip_id);
 	      if ((gettcpopt_ts(tcp, &timestamp, NULL) == 0))
 		si->ts_seqclass = TS_SEQ_UNSUPPORTED;
