diff -Naurp nmap-4.68-orig/Makefile.in nmap-4.68/Makefile.in
--- nmap-4.68-orig/Makefile.in	2008-08-30 01:04:10.000000000 +0200
+++ nmap-4.68/Makefile.in	2008-08-30 01:04:45.000000000 +0200
@@ -101,7 +101,7 @@ nbase_build: $(NBASEDIR)/Makefile
 	@echo Compiling libnbase;
 	cd $(NBASEDIR) && $(MAKE)
 
-nsock_build: $(NSOCKDIR)/src/Makefile
+nsock_build: $(NSOCKDIR)/src/Makefile nbase_build
 	@echo Compiling libnsock;
 	cd $(NSOCKDIR)/src && $(MAKE)
 
