diff -Naur source-highlight-2.6.orig/configure.ac source-highlight-2.6/configure.ac
--- source-highlight-2.6.orig/configure.ac	2007-06-08 00:28:50.000000000 +0200
+++ source-highlight-2.6/configure.ac	2007-06-08 00:31:09.000000000 +0200
@@ -125,7 +125,8 @@
 AM_CONDITIONAL(NO_HELP2MAN, test -z "$ac_cv_path_HELP2MAN" )
 
 dnl this is for generating references
-AC_PATH_PROGS(CTAGS, ctags)
+AC_PATH_PROGS(CTAGS, exuberant-ctags)
+AC_SUBST([$CTAGS])
 AM_CONDITIONAL(NO_CTAGS, test -z "$ac_cv_path_CTAGS" )
 
 dnl *******************
@@ -229,4 +230,4 @@
 generate references (--gen-references).
 Furthermore, if you run make check, then some
 tests are going to fail.])
-fi
\ Kein Zeilenumbruch am Dateiende.
+fi
diff -Naur source-highlight-2.6.orig/doc/Makefile.am source-highlight-2.6/doc/Makefile.am
--- source-highlight-2.6.orig/doc/Makefile.am	2007-06-08 00:28:50.000000000 +0200
+++ source-highlight-2.6/doc/Makefile.am	2007-06-08 00:35:55.000000000 +0200
@@ -20,7 +20,7 @@
 
 CPP2HTML = $(SRCHILITEPROG) -s cpp -f html $(STD_STYLE)
 
-CPP2HTML_WREFS = $(CPP2HTML) --title="Contains references to tags" --gen-references=inline --ctags="ctags --excmd=n --tag-relative=yes -o my_test_ref.tags"
+CPP2HTML_WREFS = $(CPP2HTML) --title="Contains references to tags" --gen-references=inline --ctags="@CTAGS@ --excmd=n --tag-relative=yes -o my_test_ref.tags"
 
 CHANGELOG2HTML = $(SRCHILITEPROG) -s changelog -f html $(STD_STYLE)
 
@@ -47,80 +47,80 @@
 endif
 
 lang-list.texinfo: $(top_srcdir)/src/lang.map
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ --lang-list | sed -e 's/@/@@/' > $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ --lang-list | sed -e 's/@/@@/' > $@
 
 outlang-list.texinfo: $(top_srcdir)/src/outlang.map
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ --outlang-list | sed -e 's/@/@@/' > $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ --outlang-list | sed -e 's/@/@@/' > $@
 
 test.java.texinfo: $(top_srcdir)/tests/test.java
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test.java --style-file=$(top_srcdir)/src/texinfo.style -n > $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test.java --style-file=$(top_srcdir)/src/texinfo.style -n > $@
 
 debug.java.texinfo: $(top_srcdir)/tests/debug.java
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/debug.java --style-file=$(top_srcdir)/src/texinfo.style -n > $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/debug.java --style-file=$(top_srcdir)/src/texinfo.style -n > $@
 
 %.lang.texinfo: $(top_srcdir)/src/%.lang
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 %.langelems.texinfo: $(top_srcdir)/src/%.lang
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ --show-lang-elements=$< > $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ --show-lang-elements=$< > $@
 
 %.txt.texinfo: $(srcdir)/%.txt
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 %.txt.texinfo.texinfo: $(srcdir)/%.txt.texinfo
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=nohilite.lang
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=nohilite.lang
 
 %.outlang.texinfo: $(top_srcdir)/src/%.outlang
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 %.lang.texinfo: $(top_srcdir)/tests/%.lang
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 %.lang.texinfo: $(srcdir)/%.lang
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 %.lang.texinfo: $(top_srcdir)/src/%.lang
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 %.style.texinfo: $(top_srcdir)/src/%.style
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 %.style.texinfo: $(top_srcdir)/tests/%.style
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 %.diff.texinfo: %.diff
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 %.css.texinfo: %.css
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 %.css.texinfo: $(top_srcdir)/src/%.css
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 %.cpp.texinfo: $(top_srcdir)/tests/%.cpp
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $< --style-file=$(top_srcdir)/src/texinfo.style -o $@
 
 test_extreme_comment_wrong.texinfo: $(top_srcdir)/tests/test_extreme_comment.cpp
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test_extreme_comment.cpp --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=extreme_comment.lang
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test_extreme_comment.cpp --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=extreme_comment.lang
 
 test_extreme_comment_1.texinfo: $(top_srcdir)/tests/test_extreme_comment.cpp
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test_extreme_comment.cpp --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=extreme_comment2.lang
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test_extreme_comment.cpp --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=extreme_comment2.lang
 
 test_extreme_comment_wrong2.texinfo: $(top_srcdir)/tests/test_extreme_comment2.cpp
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test_extreme_comment2.cpp --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=extreme_comment2.lang
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test_extreme_comment2.cpp --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=extreme_comment2.lang
 
 test_extreme_comment_2.texinfo: $(top_srcdir)/tests/test_extreme_comment2.cpp
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test_extreme_comment2.cpp --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=extreme_comment3.lang
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test_extreme_comment2.cpp --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=extreme_comment3.lang
 
 test_extreme_comment_3.texinfo: $(top_srcdir)/tests/test_extreme_comment3.cpp
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test_extreme_comment3.cpp --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=extreme_comment3.lang
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test_extreme_comment3.cpp --style-file=$(top_srcdir)/src/texinfo.style -o $@ --lang-def=extreme_comment3.lang
 
 %.show.texinfo: %.lang
 	$(SRCHILITEEXE) --show-regex=$< > $@
 	sed -i -e 's/@/@@/g' $@
 
 test_ref.h.texinfo: $(top_srcdir)/tests/test.h
-	$(SRCHILITEEXE) --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test.h --style-file=$(top_srcdir)/src/texinfo.style -n --gen-references=postline > $@
+	$(SRCHILITEEXE) --ctags=@CTAGS@ --data-dir=$(top_srcdir)/src/ -f texinfo -i $(top_srcdir)/tests/test.h --style-file=$(top_srcdir)/src/texinfo.style -n --gen-references=postline > $@
 
 # the dependence on config.h is to ensure the correct version number
 # in the generated man file
@@ -189,7 +189,7 @@
 # note that $(srcdir) actually points to doc dir
 
 langdef.lang.html: $(top_srcdir)/src/langdef.lang
-	$(SRCHILITEPROG) $(STD_STYLE) -i $(top_srcdir)/src/langdef.lang -o langdef.lang.html
+	$(SRCHILITEPROG) --ctags=@CTAGS@ $(STD_STYLE) -i $(top_srcdir)/src/langdef.lang -o langdef.lang.html
 
 startapp.cc.html: $(top_srcdir)/src/startapp.cc
 	$(CPP2HTML) --doc $(OUTPUT_HERE) $(top_srcdir)/src/startapp.cc
@@ -310,4 +310,4 @@
 texinfo-clean:
 	rm -f $(GENERATEDINFOS)
 
-DISTCLEANFILES = $(HTML_GENERATED_FILES) source-highlight.1 $(GENERATEDINFOS) tags langdef.lang.html
\ Kein Zeilenumbruch am Dateiende.
+DISTCLEANFILES = $(HTML_GENERATED_FILES) source-highlight.1 $(GENERATEDINFOS) tags langdef.lang.html
diff -Naur source-highlight-2.6.orig/src/lib/Makefile.am source-highlight-2.6/src/lib/Makefile.am
--- source-highlight-2.6.orig/src/lib/Makefile.am	2007-06-08 00:28:50.000000000 +0200
+++ source-highlight-2.6/src/lib/Makefile.am	2007-06-08 00:38:29.000000000 +0200
@@ -127,7 +127,7 @@
 test_langinfer_LDADD = libcommon.la
 
 mytags: $(srcdir)/test_readtags_main.cpp
-	$(CTAGS) --excmd=n --fields=+n -o mytags $(srcdir)/test_readtags_main.cpp $(srcdir)/readtags.h
+	@CTAGS@ --excmd=n --fields=+n -o mytags $(srcdir)/test_readtags_main.cpp $(srcdir)/readtags.h
 
 outlangtest.outlang: $(srcdir)/outlangtest.outlang.tmpl
 	rm -f outlangtest.outlang
diff -Naur source-highlight-2.6.orig/tests/Makefile.am source-highlight-2.6/tests/Makefile.am
--- source-highlight-2.6.orig/tests/Makefile.am	2007-06-08 00:28:50.000000000 +0200
+++ source-highlight-2.6/tests/Makefile.am	2007-06-08 00:38:10.000000000 +0200
@@ -50,7 +50,7 @@
 
 JAVA2TEXINFO = $(PROGNAME) -s java -f texinfo --style-file $(top_srcdir)/src/texinfo.style
 
-CPP2TEXINFO_WREFS = $(PROGNAME) -s cpp -f texinfo --style-file $(top_srcdir)/src/texinfo.style --gen-references=postline --line-number --ctags=""
+CPP2TEXINFO_WREFS = $(PROGNAME) -s cpp -f texinfo --style-file $(top_srcdir)/src/texinfo.style --gen-references=postline --line-number --ctags="@CTAGS@"
 
 # this refers to an "absolute" path (../doc/) without using the automake
 # variables.  However this way the generated files will always be the
@@ -360,13 +360,13 @@
 
 CPP2HTML_WHEADERS = $(CPP2HTML) --header=$(srcdir)/header.html --footer=$(srcdir)/footer.html
 
-CPP2HTML_WREFS = $(CPP2HTML) --title="Contains references to tags" --gen-references=inline --ctags="$(CTAGS) --excmd=number --tag-relative=yes -o my_test_ref.tags"
+CPP2HTML_WREFS = $(CPP2HTML) --title="Contains references to tags" --gen-references=inline --ctags="@CTAGS@ --excmd=number --tag-relative=yes -o my_test_ref.tags"
 
-CPP2LATEX_WREFS = $(CPP2TEX) --title="Contains references to tags" --gen-references=postline --line-number --ctags=""
+CPP2LATEX_WREFS = $(CPP2TEX) --title="Contains references to tags" --gen-references=postline --line-number --ctags="@CTAGS@"
 
-CPP2LATEX_WREFSPOST = $(CPP2TEX) --title="Contains references to tags" --gen-references=postdoc --line-number --ctags=""
+CPP2LATEX_WREFSPOST = $(CPP2TEX) --title="Contains references to tags" --gen-references=postdoc --line-number --ctags="@CTAGS@"
 
-CTAGS_CMD=$(CTAGS) --excmd=number --tag-relative=yes 
+CTAGS_CMD=@CTAGS@ --excmd=number --tag-relative=yes 
 
 test_header.html: test.h
 	$(CPP2HTML_WHEADERS) -i test.h -o $@
@@ -847,4 +847,4 @@
 clean_html:
 	rm -f *.h.html *.java.html *.cc.html *.pas.html *.pl.html *.pm.html *.php3.html *.py.html *.chlog.html *.chlog_css.html *.l.html test_xhtml*.html test_notfixed.html test_css.html test_css_nodoc.html test_header.html *.rb.html *.js.html *.java.txt *.java.tex *.java.texinfo *.lua.html *.ml.html *.sml.html test_*_lines.html *.log.html *.y.html *.simple.html *.tex.html *.lgt.html *.diff.html *.lang.html *.htm.html *.style.html *.cs.html *.xml.html *.sh.html *.url.html *.txt.html *.unknown.html *.tcl.html *.bib.html *.makefile.html *.css.html 
 
-DISTCLEANFILES=test_ref.html test_ref.tags my_test_ref.tags tags test_ref.tex test_ref_post.tex test_ref.texinfo syslog.style.html
\ Kein Zeilenumbruch am Dateiende.
+DISTCLEANFILES=test_ref.html test_ref.tags my_test_ref.tags tags test_ref.tex test_ref_post.tex test_ref.texinfo syslog.style.html
