diff -ru Terminal-0.4.0.orig/terminal/Makefile.am Terminal-0.4.0/terminal/Makefile.am
--- Terminal-0.4.0.orig/terminal/Makefile.am	2009-09-07 14:47:06 +0200
+++ Terminal-0.4.0/terminal/Makefile.am	2009-09-07 14:54:48 +0200
@@ -92,7 +92,7 @@
 	$(TERMINAL_LIBS)
 
 # install symlink to 'terminal'
-install-data-local:
+install-data-local: install-binPROGRAMS
 	$(mkinstalldirs) $(DESTDIR)$(bindir)
 	-( cd $(DESTDIR)$(bindir) ; test -x terminal || ln -sf Terminal terminal )
 
