diff -ur xscreensaver-5.04.orig/driver/screensaver-properties.desktop.in xscreensaver-5.04/driver/screensaver-properties.desktop.in
--- xscreensaver-5.04.orig/driver/screensaver-properties.desktop.in	2003-12-15 13:16:12.000000000 +0200
+++ xscreensaver-5.04/driver/screensaver-properties.desktop.in	2007-12-30 21:00:15.000000000 +0200
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Exec=xscreensaver-demo
-Icon=xscreensaver.xpm
+Icon=xscreensaver
 Terminal=false
 _Name=Screensaver
 _Comment=Change screensaver properties
 Type=Application
-Categories=Application;Settings;Appearance;AdvancedSettings;
+Categories=Settings;DesktopSettings;Security;
