--- xscreensaver-5.08/aclocal.m4.po	2002-05-11 10:44:33.000000000 +0900
+++ xscreensaver-5.08/aclocal.m4	2008-12-30 04:33:01.000000000 +0900
@@ -252,7 +252,10 @@
 
     AC_OUTPUT_COMMANDS(
       [case "$CONFIG_FILES" in *po/Makefile.in*)
+         rm -f po/stamp-it po/stamp-it.tmp
+         > po/stamp-it.tmp
         sed -e "/POTFILES =/r po/POTFILES" po/Makefile.in > po/Makefile
+        mv po/stamp-it.tmp po/stamp-it
       esac])
 
     dnl These rules are solely for the distribution goal.  While doing this
--- xscreensaver-5.08/po/Makefile.in.in.po	2007-11-11 09:44:42.000000000 +0900
+++ xscreensaver-5.08/po/Makefile.in.in	2008-12-30 04:47:58.000000000 +0900
@@ -190,7 +190,7 @@
 all-yes: $(CATALOGS)
 all-no:
 
-$(srcdir)/$(GETTEXT_PACKAGE).pot: $(POTFILES)
+$(GETTEXT_PACKAGE).pot: $(POTFILES)
 	@INTLTOOL_EXTRACT=$(INTLTOOL_EXTRACT) ; \
 	export INTLTOOL_EXTRACT ; \
 	echo top_srcdir=$(top_srcdir) $(GENPOT) ; \
@@ -274,6 +274,7 @@
 
 distclean: clean
 	rm -f Makefile Makefile.in POTFILES *.mo *.msg *.cat *.cat.m *.gmo *.pot
+	rm -f stamp-it
 
 maintainer-clean: distclean
 	@echo "This command is intended for maintainers to use;"
@@ -282,8 +283,8 @@
 
 depend:
 # fuck off.  love, jwz.
-#distdepend: generate_potfiles_in update-po $(DISTFILES)
-distdepend::
+distdepend: generate_potfiles_in update-po $(DISTFILES)
+#distdepend::
 
 
 # jwz: Generates po/POTFILES.in by examining the source tree:
@@ -292,6 +293,8 @@
 generate_potfiles_in:
 	@tmp=po.$$$$ ;				\
 	 file=POTFILES.in ;			\
+	 echo "sleeping 3 seconds..." ;\
+	 sleep 3; \
 	 ( cd $(top_srcdir);			\
 	   echo "# Auto-generated: `LANG=C date`" ;	\
 	   ( grep -l '_("' driver/*.[ch] ;	\
@@ -362,16 +365,22 @@
 	    && (sed -e '/^#/d' 						\
 		    -e "s/^\[.*\] +//" 					\
 		    -e '/^[ 	]*$$/d' 				\
-		    -e "s@.*@	$$posrcprefix& \\\\@" < $(srcdir)/$@.in	\
+		    -e "s@.*@	$$posrcprefix& \\\\@" < $@.in	\
 		| sed -e '$$s/\\$$//') > $@-t \
 	    && chmod a-w $@-t \
 	    && mv $@-t $@ )
 
+Makefile: stamp-it
+	@if test ! -f $@; then \
+		rm -f stamp-it; \
+		$(MAKE) stamp-it; \
+	fi
+
 # Please to be fucking the hell off.   This breaks "make distdepend". -jwz
-#Makefile: Makefile.in.in ../config.status POTFILES
-#	cd .. \
-#	  && CONFIG_FILES=$(subdir)/$@.in CONFIG_HEADERS= \
-#	       $(SHELL) ./config.status
+stamp-it: Makefile.in.in ../config.status POTFILES
+	cd .. \
+	  && CONFIG_FILES=$(subdir)/Makefile.in CONFIG_HEADERS= \
+	       $(SHELL) ./config.status
 
 # Tell versions [3.59,3.63) of GNU make not to export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
