--- driver/screens.c.orig	2009-04-30 17:59:02.000000000 -0700
+++ driver/screens.c	2009-04-30 18:01:32.000000000 -0700
@@ -436,7 +436,7 @@
               XRROutputInfo *rroi = XRRGetOutputInfo (dpy, res, 
                                                       res->outputs[k]);
               RRCrtc crtc = (rroi->crtc  ? rroi->crtc :
-                             rroi->crtcs ? rroi->crtcs[0] : 0);
+                             rroi->ncrtc ? rroi->crtcs[0] : 0);
               XRRCrtcInfo *crtci = (crtc ? XRRGetCrtcInfo(dpy, res, crtc) : 0);
 
               monitors[j] = m;
