diff -ur xscreensaver-5.05.orig/Makefile.in xscreensaver-5.05/Makefile.in
--- xscreensaver-5.05.orig/Makefile.in	2008-03-01 06:32:52.000000000 +0200
+++ xscreensaver-5.05/Makefile.in	2008-03-02 16:27:25.000000000 +0200
@@ -326,7 +326,7 @@
   diff -U0 download.html $$TMP ;					    \
   echo '' ;								    \
 									    \
-  OLDEST=`ls xscreensaver*.tar.gz | head -1` ;				    \
+  OLDEST=`ls xscreensaver*.tar.gz | head -n 1` ;			    \
   /bin/echo -n "Delete $$DEST/$$OLDEST? ";				    \
   read line;								    \
   if [ "x$$line" = "xyes" -o "x$$line" = "xy" ]; then			    \
