diff -ur xscreensaver-5.02.orig/driver/XScreenSaver.ad.in xscreensaver-5.02/driver/XScreenSaver.ad.in
--- xscreensaver-5.02.orig/driver/XScreenSaver.ad.in	2007-04-21 06:33:17.000000000 +0300
+++ xscreensaver-5.02/driver/XScreenSaver.ad.in	2007-05-01 21:36:47.000000000 +0300
@@ -302,10 +302,6 @@
 				celtic -root				    \n\
 				cwaves -root				    \n\
 				m6502 -root				    \n\
-- default-n:			webcollage -root			    \n\
-- default-n:  "WebCollage (whacked)"					      \
-				webcollage -root -filter		      \
-				  'vidwhacker -stdin -stdout'		    \n\
 - default-n:			vidwhacker -root			    \n\
 									      \
 @GL_KLUDGE@ GL:			gears -root				    \n\
@@ -488,7 +484,6 @@
 *hacks.xanalogtv.name:      XAnalogTV
 *hacks.xspirograph.name:    XSpiroGraph
 *hacks.nerverot.name:       NerveRot
-*hacks.webcollage.name:     WebCollage
 *hacks.vidwhacker.name:     VidWhacker
 *hacks.morph3d.name:        Morph3D
 *hacks.bubble3d.name:       Bubble3D
diff -ur xscreensaver-5.02.orig/hacks/glx/glsnake.c xscreensaver-5.02/hacks/glx/glsnake.c
--- xscreensaver-5.02.orig/hacks/glx/glsnake.c	2006-05-03 22:39:21.000000000 +0300
+++ xscreensaver-5.02/hacks/glx/glsnake.c	2007-05-01 21:35:04.000000000 +0300
@@ -565,21 +565,6 @@
 	ZERO, PIN, ZERO, ZERO, ZERO, ZERO, PIN, ZERO, ZERO, ZERO, ZERO,
 	ZERO, PIN, ZERO }
     },
-    { "erect penis",     /* thanks benno */
-      { PIN, ZERO, PIN, PIN, ZERO, ZERO, PIN, ZERO, ZERO, ZERO, PIN,
-	PIN, ZERO, ZERO, ZERO, RIGHT, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO,
-	ZERO, ZERO }
-    },
-    { "flaccid penis",
-      { PIN, ZERO, PIN, PIN, ZERO, ZERO, PIN, ZERO, ZERO, ZERO, PIN,
-	PIN, ZERO, ZERO, ZERO, RIGHT, PIN, ZERO, ZERO, ZERO, ZERO, ZERO,
-	ZERO, ZERO }
-    },
-    { "vagina",
-      { RIGHT, ZERO, ZERO, ZERO, RIGHT, ZERO, ZERO, PIN, ZERO, ZERO,
-	LEFT, ZERO, ZERO, ZERO, LEFT, ZERO, LEFT, PIN, LEFT, PIN, RIGHT,
-	PIN, RIGHT, ZERO }
-    },
     { "mask",
       { ZERO, RIGHT, LEFT, PIN, RIGHT, RIGHT, PIN, ZERO, ZERO, PIN,
 	ZERO, ZERO, PIN, ZERO, PIN, ZERO, PIN, ZERO, ZERO, PIN, ZERO,
@@ -995,9 +980,6 @@
     { "Parrot",
         { ZERO, ZERO, ZERO, ZERO, RIGHT, RIGHT, ZERO, LEFT, PIN, RIGHT, ZERO, RIGHT, ZERO, RIGHT, ZERO, RIGHT, PIN, LEFT, ZERO, RIGHT, LEFT, ZERO, PIN, ZERO }
     },
-    { "Penis",
-        { PIN, PIN, RIGHT, ZERO, PIN, PIN, ZERO, PIN, ZERO, ZERO, RIGHT, PIN, LEFT, ZERO, ZERO, PIN, ZERO, PIN, PIN, ZERO, LEFT, PIN, PIN, ZERO }
-    },
     { "PictureComingSoon",
         { LEFT, LEFT, ZERO, RIGHT, LEFT, PIN, RIGHT, RIGHT, PIN, RIGHT, LEFT, PIN, LEFT, RIGHT, PIN, RIGHT, RIGHT, PIN, RIGHT, LEFT, ZERO, RIGHT, RIGHT, ZERO }
     },
diff -ur xscreensaver-5.02.orig/hacks/Makefile.in xscreensaver-5.02/hacks/Makefile.in
--- xscreensaver-5.02.orig/hacks/Makefile.in	2007-04-21 06:33:23.000000000 +0300
+++ xscreensaver-5.02/hacks/Makefile.in	2007-05-01 21:33:28.000000000 +0300
@@ -110,9 +110,8 @@
 		  fuzzyflakes.c anemotaxis.c memscroller.c substrate.c \
 		  intermomentary.c fireworkx.c fireworkx_mmx.S fiberlamp.c \
 		  boxfit.c interaggregate.c celtic.c cwaves.c m6502.c \
-		  asm6502.c \
-		  webcollage-cocoa.m webcollage-helper-cocoa.m
-SCRIPTS		= vidwhacker webcollage ljlatest
+		  asm6502.c
+SCRIPTS		= vidwhacker ljlatest
 
 # Programs that are mentioned in XScreenSaver.ad, and that have XML files,
 # but that are not shipped with xscreensaver itself.
@@ -140,7 +139,7 @@
 		  blaster.o bumps.o ripples.o xspirograph.o \
 		  nerverot.o xrayswarm.o hyperball.o zoom.o whirlwindwarp.o \
 		  rotzoomer.o whirlygig.o speedmine.o vermiculate.o \
-		  xpm-pixmap.o webcollage-helper.o twang.o apollonian.o \
+		  xpm-pixmap.o twang.o apollonian.o \
 		  euler2d.o juggle.o polyominoes.o thornbird.o fluidballs.o \
 		  anemone.o halftone.o metaballs.o eruption.o popsquares.o \
 		  barcode.o piecewise.o cloudlife.o fontglide.o apple2.o \
@@ -148,8 +147,7 @@
 		  mismunch.o pacman.o pacman_ai.o pacman_level.o \
 		  fuzzyflakes.o anemotaxis.o memscroller.o substrate.o \
 		  intermomentary.o fireworkx.o fiberlamp.o boxfit.o \
-		  interaggregate.o celtic.o cwaves.o webcollage-cocoa.o \
-		  webcollage-helper-cocoa.o m6502.0 asm6502.o
+		  interaggregate.o celtic.o cwaves.o m6502.0 asm6502.o
 
 NEXES		= attraction blitspin bouboule braid bubbles decayscreen deco \
 		  drift flag flame forest vines galaxy grav greynetic halo \
@@ -173,7 +171,7 @@
 		  celtic \
 		  @JPEG_EXES@
 SEXES		= sonar
-JPEG_EXES	= webcollage-helper
+JPEG_EXES	=
 EXES		= $(NEXES) $(SEXES)
 
 HACK_OBJS_1	= $(UTILS_BIN)/resources.o $(UTILS_BIN)/visual.o \
@@ -209,7 +207,7 @@
 	          spotlight.man squiral.man starfish.man strange.man \
 	          swirl.man t3d.man thornbird.man triangle.man truchet.man \
 	          twang.man vermiculate.man vidwhacker.man vines.man \
-	          wander.man webcollage.man whirlwindwarp.man whirlygig.man \
+	          wander.man whirlwindwarp.man whirlygig.man \
 	          worm.man xflame.man xjack.man xlyap.man xmatrix.man \
 	          xrayswarm.man xspirograph.man xsublim.man \
 	          zoom.man halftone.man eruption.man metaballs.man \
@@ -978,10 +976,6 @@
 xsublim:	xsublim.o	$(HACK_OBJS_1)
 	$(CC_HACK) -o $@ $@.o	$(HACK_OBJS_1) $(HACK_LIBS)
 
-webcollage-helper: webcollage-helper.o
-	$(CC_HACK) -o $@ $@.o	$(XPM_LIBS) $(JPEG_LIBS)
-
-
 ##############################################################################
 #
 # DO NOT DELETE: updated by make distdepend
@@ -2476,17 +2470,6 @@
 wander.o: $(UTILS_SRC)/usleep.h
 wander.o: $(UTILS_SRC)/visual.h
 wander.o: $(UTILS_SRC)/yarandom.h
-webcollage-cocoa.o: ../config.h
-webcollage-cocoa.o: $(srcdir)/screenhackI.h
-webcollage-cocoa.o: $(srcdir)/screenhack.h
-webcollage-cocoa.o: $(UTILS_SRC)/colors.h
-webcollage-cocoa.o: $(UTILS_SRC)/grabscreen.h
-webcollage-cocoa.o: $(UTILS_SRC)/hsv.h
-webcollage-cocoa.o: $(UTILS_SRC)/resources.h
-webcollage-cocoa.o: $(UTILS_SRC)/usleep.h
-webcollage-cocoa.o: $(UTILS_SRC)/visual.h
-webcollage-cocoa.o: $(UTILS_SRC)/yarandom.h
-webcollage-helper.o: ../config.h
 whirlwindwarp.o: ../config.h
 whirlwindwarp.o: $(srcdir)/screenhackI.h
 whirlwindwarp.o: $(srcdir)/screenhack.h
diff -ur xscreensaver-5.02.orig/po/POTFILES.in xscreensaver-5.02/po/POTFILES.in
--- xscreensaver-5.02.orig/po/POTFILES.in	2007-04-21 06:33:29.000000000 +0300
+++ xscreensaver-5.02/po/POTFILES.in	2007-05-01 21:35:33.000000000 +0300
@@ -203,7 +203,6 @@
 hacks/config/vidwhacker.xml
 hacks/config/vines.xml
 hacks/config/wander.xml
-hacks/config/webcollage.xml
 hacks/config/whirlwindwarp.xml
 hacks/config/whirlygig.xml
 hacks/config/worm.xml
