Only in java-config-wrapper-0.16: prefix
diff -ur java-config-wrapper-0.16.orig/src/shell/java-1.5-fixer java-config-wrapper-0.16/src/shell/java-1.5-fixer
--- java-config-wrapper-0.16.orig/src/shell/java-1.5-fixer	2007-07-27 23:24:18.000000000 +0200
+++ java-config-wrapper-0.16/src/shell/java-1.5-fixer	2009-04-19 13:34:44.032195616 +0200
@@ -1,6 +1,6 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 
-source "/etc/init.d/functions.sh"
+source "@GENTOO_PORTAGE_EPREFIX@/etc/init.d/functions.sh"
 
 EXCLUDED_JARS="
 /usr/share/hibernate-annotations-3.0/lib/hibernate-annotations.jar
@@ -35,7 +35,7 @@
 
 	local item
 	for item in ${list}; do
-		if [[ ${item} == ${target} ]]; then
+		if [[ "@GENTOO_PORTAGE_EPREFIX@"${item} == "${target}" ]]; then
 			return 0;
 		fi
 	done
@@ -203,17 +203,18 @@
 	
 }
 
-if [[ ${UID} != 0 ]]; then
-	ewarn "You aren't root, so we'll just show what packages we'd emerge"
-fi
+# disable this for the time being in prefix
+#if [[ ${UID} != 0 ]]; then
+#	ewarn "You aren't root, so we'll just show what packages we'd emerge"
+#fi
 
-QFILE="${ROOT}/usr/bin/qfile"
+QFILE="${EROOT}/usr/bin/qfile"
 if [[ ! -x ${QFILE} ]]; then
 	eerror "You need to emerge portage-utils."
 	exit 1
 fi
 
-VERIFIER="${ROOT}/usr/bin/class-version-verify.py"
+VERIFIER="${EROOT}/usr/bin/class-version-verify.py"
 if [[ ! -x "${VERIFIER}" ]]; then
 	eerror "You need to emerge >=javatoolkit-0.1.6"
 	exit 1
@@ -224,7 +225,7 @@
 ant_package_envs=""
 # build a list of package.env files
 for ant_dep in ${ant_packages}; do
-	package_env="${ROOT}/usr/share/${ant_dep}/package.env"
+	package_env="${EROOT}/usr/share/${ant_dep}/package.env"
 	ant_package_envs="${ant_package_envs} ${package_env}"
 done
 # get jars from these envs
@@ -236,16 +237,14 @@
 eend 0
 
 if [[ -n ${broken_ant_packages} ]]; then
-	if [[ ${UID} == 0 ]]; then
-		ebegin "Removing broken ant dependencies"
-		emerge -C ${broken_ant_packages}
-		result=$?
-		eend ${result}
-	fi
+	ebegin "Removing broken ant dependencies"
+	emerge -C ${broken_ant_packages}
+	result=$?
+	eend ${result}
 fi
 
 ebegin "Generating a list of jars in the system"
-all_jars=$(get_jars ${ROOT}/usr/share/*/package.env)
+all_jars=$(get_jars ${EROOT}/usr/share/*/package.env)
 eend 0
 
 ebegin "Checking for > 1.4 bytecode in system jars"
@@ -253,19 +252,18 @@
 eend 0
 
 if [[ -n ${broken_packages} || -n ${broken_ant_packages} ]]; then
-	if [[ ${UID} == 0 ]]; then
-		ebegin "Attempting to fix broken packages"
-		fix_packages ${broken_ant_packages} ${broken_packages}
-		result=$?
-		if [[ ${result} != 0 ]]; then
-			eerror "There was a problem fixing all the broken packages!"
-			exit 1
-		else
-			einfo "Congratulations! System jars seem to have been fixed."
-		fi
-		eend $result
-		exit $result
+   # warning: UID check removed for prefix
+	ebegin "Attempting to fix broken packages"
+	fix_packages ${broken_ant_packages} ${broken_packages}
+	result=$?
+	if [[ ${result} != 0 ]]; then
+		eerror "There was a problem fixing all the broken packages!"
+		exit 1
+	else
+		einfo "Congratulations! System jars seem to have been fixed."
 	fi
+	eend $result
+	exit $result
 else
 	einfo "Congratulations! All system jars seem to have been sanely built already."
 	exit 0
diff -ur java-config-wrapper-0.16.orig/src/shell/java-check-environment java-config-wrapper-0.16/src/shell/java-check-environment
--- java-config-wrapper-0.16.orig/src/shell/java-check-environment	2009-04-17 23:44:54.000000000 +0200
+++ java-config-wrapper-0.16/src/shell/java-check-environment	2009-04-19 13:37:59.489149532 +0200
@@ -1,6 +1,6 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 
-source /etc/init.d/functions.sh
+source "@GENTOO_PORTAGE_EPREFIX@"/etc/init.d/functions.sh
 
 # Used to update the global result.
 update_result() {
@@ -60,7 +60,7 @@
 	unset GENERATION
 
 	# Check /etc/env.d/java for old env files
-	for env in $(ls /etc/env.d/java/20* 2>/dev/null); do
+	for env in $(ls "@GENTOO_PORTAGE_EPREFIX@"/etc/env.d/java/20* 2>/dev/null); do
 		generation=$(source ${env}; echo $GENERATION)
 		version=$(source ${env}; echo $VERSION)
 		handle=${env#*20}
@@ -150,7 +150,7 @@
 # Checks that a system-vm for generation-1 is set
 check_generation_1_system_vm() {
 	local this_result=0
-	local env="/etc/env.d/20java"
+	local env="@GENTOO_PORTAGE_EPREFIX@/etc/env.d/20java"
 	# Make sure that no generation-1 VM is set
 	if [[ -f "${env}" ]]; then
 		qeerror "You have a generation-1 system VM set. (${env} exists)"
@@ -205,7 +205,7 @@
 
 		this_result=1
     fi
-	if grep "virtual/\(jre\|jdk\)" /etc/portage/profile/virtuals >/dev/null 2>&1; then
+	if grep "virtual/\(jre\|jdk\)" "@GENTOO_PORTAGE_EPREFIX@"/etc/portage/profile/virtuals >/dev/null 2>&1; then
 		qeerror "Found virtual/{jre,jdk} entries in /etc/portage/profile/virtuals"
 		qeerror "Please remove them"
 		
@@ -242,7 +242,7 @@
 check_package_env() {
 	local result=0
 	local pkg owner failed
-	for penv in /usr/share/*/package.env; do
+	for penv in "@GENTOO_PORTAGE_EPREFIX@"/usr/share/*/package.env; do
 		if grep DEPEND ${penv} | grep , > /dev/null; then
 			owner=$(qfile -qC ${penv})
 			qeerror "Syntax error in ${owner}:"
@@ -260,7 +260,7 @@
 					else
 						pkg=${atom}
 					fi
-					if [[ ! -e /usr/share/${pkg}/package.env && ! -e "/usr/share/java-config-2/virtuals/${pkg}" ]]; then
+					if [[ ! -e "@GENTOO_PORTAGE_EPREFIX@"/usr/share/${pkg}/package.env && ! -e "@GENTOO_PORTAGE_EPREFIX@/usr/share/java-config-2/virtuals/${pkg}" ]]; then
 						return 1
 					fi
 				done
@@ -279,7 +279,7 @@
 }
 
 check_fastjar() {
-	local fjar=/usr/bin/fastjar
+	local fjar="@GENTOO_PORTAGE_EPREFIX@"/usr/bin/fastjar
 	local result=0
 	if [[ -x ${fjar} ]]; then
 		local tmpdir="$(mktemp -d)"
diff -ur java-config-wrapper-0.16.orig/src/shell/java-config java-config-wrapper-0.16/src/shell/java-config
--- java-config-wrapper-0.16.orig/src/shell/java-config	2006-08-02 20:58:21.000000000 +0200
+++ java-config-wrapper-0.16/src/shell/java-config	2009-04-19 13:34:44.033273832 +0200
@@ -1,8 +1,8 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 
 WANT_JAVA_CONFIG=${WANT_JAVA_CONFIG:="2"}
 
-JAVA_CONFIG="/usr/bin/java-config-${WANT_JAVA_CONFIG}"
+JAVA_CONFIG="@GENTOO_PORTAGE_EPREFIX@/usr/bin/java-config-${WANT_JAVA_CONFIG}"
 
 if [[ -x ${JAVA_CONFIG} ]]; then
 	exec ${JAVA_CONFIG} "$@"
