Index: libtorrent/src/net/socket_datagram.cc
===================================================================
--- libtorrent/src/net/socket_datagram.cc	(revision 1060)
+++ libtorrent/src/net/socket_datagram.cc	(working copy)
@@ -73,7 +73,7 @@
   int r;
 
   if (sa != NULL) {
-    r = ::sendto(m_fileDesc, buffer, length, 0, sa->c_sockaddr(), sizeof(rak::socket_address));
+    r = ::sendto(m_fileDesc, buffer, length, 0, sa->sa_inet()->c_sockaddr(), sizeof(rak::socket_address_inet));
   } else {
     r = ::send(m_fileDesc, buffer, length, 0);
   }
