Index: libtorrent/configure.ac
===================================================================
--- libtorrent/configure.ac	(revision 1060)
+++ libtorrent/configure.ac	(working copy)
@@ -5,8 +5,8 @@
 AC_DEFINE(PEER_VERSION, "lt\x0C\x20", 4 byte client and version identifier for DHT)
 
 LIBTORRENT_CURRENT=11
-LIBTORRENT_REVISION=0
-LIBTORRENT_AGE=2
+LIBTORRENT_REVISION=2
+LIBTORRENT_AGE=0
 
 LIBTORRENT_INTERFACE_VERSION_INFO=$LIBTORRENT_CURRENT:$LIBTORRENT_REVISION:$LIBTORRENT_AGE
 LIBTORRENT_INTERFACE_VERSION_NO=$LIBTORRENT_CURRENT.$LIBTORRENT_AGE.$LIBTORRENT_REVISION
