committed upstream >0.1.19

--- ../pax-utils-0.1.18.orig/porting.h	2009-09-25 11:23:48 +0200
+++ porting.h	2009-09-25 11:31:41 +0200
@@ -141,7 +141,7 @@
 #  define ELF_DATA ELFDATA2MSB
 # elif defined(_LITTLE_ENDIAN)
 #  define ELF_DATA ELFDATA2LSB
-# elif defined(_BIG_ENDIAN)
+# elif defined(_BIG_ENDIAN) || defined(__BIG_ENDIAN__)
 #  define ELF_DATA ELFDATA2MSB
 # else
 #  error "no idea what the native byte order is"
