--- src/global.c.orig	2004-08-18 00:00:51.573889920 -0400
+++ src/global.c	2004-08-18 00:00:54.842393032 -0400
@@ -1043,8 +1043,6 @@
 		N_("No conversion from DOS/Mac format"), NO_CONVERT);
 	toggle_init_one(TOGGLE_DOS_KEY, N_("Writing file in DOS format"),
 		DOS_FILE);
-	toggle_init_one(TOGGLE_MAC_KEY, N_("Writing file in Mac format"),
-		MAC_FILE);
 	toggle_init_one(TOGGLE_BACKUP_KEY, N_("Backup files"), BACKUP_FILE);
     }
     toggle_init_one(TOGGLE_SMOOTH_KEY, N_("Smooth scrolling"), SMOOTHSCROLL);
