--- tcsh-6.14.00/Makefile.in
+++ tcsh-6.14.00/Makefile.in
@@ -135,9 +135,12 @@
 #DFLAGS=-DDELL
 #DFLAGS=
 #DFLAGS=-D_PATH_TCSHELL='"${DESTBIN}/tcsh"'
-## The following is set by autoconf.
-DFLAGS = -D_PATH_TCSHELL='"${bindir}/tcsh"' @DFLAGS@ @CPPFLAGS@
-
+## Set the path for various files, with help from autoconf
+DFLAGS = -D_PATH_TCSHELL='"${bindir}/tcsh"' \
+         -D_PATH_DOTCSHRC='"@sysconfdir@/csh.cshrc"' \
+         -D_PATH_DOTLOGIN='"@sysconfdir@/csh.login"' \
+         -D_PATH_DOTLOGOUT='"@sysconfdir@/csh.logout"' \
+		 @DFLAGS@ @CPPFLAGS@
 
 ################################################################
 ## LDFLAGS.  Define something here if you need to
--- gentoo/csh.cshrc
+++ gentoo/csh.cshrc
@@ -11,16 +11,16 @@
 ##
 ## Load the environment defaults.
 ##
-if ( $?prompt && -r /etc/csh.env ) then
-    source /etc/csh.env
+if ( $?prompt && -r @PREFIX@/etc/csh.env ) then
+    source @PREFIX@/etc/csh.env
 endif
 
 ##
 ## Make sure our path includes the basic stuff for root and normal users.
 ##
-set -f path = ( /usr/local/bin /usr/bin /bin /opt/bin $path )
+set -f path = ( @PREFIX@/usr/local/bin @PREFIX@/usr/bin @PREFIX@/bin @PREFIX@/opt/bin $path )
 if ( "$uid" == "0" ) then
-    set -f path = ( /sbin /usr/sbin /usr/local/sbin $path )
+    set -f path = ( @PREFIX@/sbin @PREFIX@/usr/sbin @PREFIX@/usr/local/sbin $path )
 endif
 
 if ( $?tcsh && $?prompt ) then
@@ -58,9 +58,9 @@
 ##
 ## Source extensions
 ##
-if ( $?prompt && -d /etc/profile.d ) then
+if ( $?prompt && -d @PREFIX@/etc/profile.d ) then
 	set nonomatch
-	foreach i ( /etc/profile.d/*.csh )
+	foreach i ( @PREFIX@/etc/profile.d/*.csh )
 		if ( -r $i ) then
 			source $i
 		endif
--- gentoo/csh.login
+++ gentoo/csh.login
@@ -29,13 +29,13 @@
 ##
 ## Set our SHELL variable.
 ##
-setenv SHELL /bin/tcsh
+setenv SHELL @PREFIX@/bin/tcsh
 
 ##
 ## Setup a default MAIL variable
 ##
-if ( -f /var/spool/mail/$USER ) then
-    setenv MAIL /var/spool/mail/$USER
+if ( -f @PREFIX@/var/spool/mail/$USER ) then
+    setenv MAIL @PREFIX@/var/spool/mail/$USER
     set mail = $MAIL
 endif
 
--- gentoo/tcsh-settings.csh
+++ gentoo/tcsh-settings.csh
@@ -22,8 +22,8 @@
 if ( ! $?HOST ) then
     setenv HOST `hostname -s`
 endif
-if ( -r /etc/inputrc && ! $?INPUTRC ) then
-    setenv INPUTRC /etc/inputrc
+if ( -r @PREFIX@/etc/inputrc && ! $?INPUTRC ) then
+    setenv INPUTRC @PREFIX@/etc/inputrc
 endif
 #setenv COLORTERM 1
 
