--- gentoo/csh.login
+++ gentoo/csh.login
@@ -1,6 +1,9 @@
 ## Gentoo csh.login
 ##
-## 2005-09-27  --  Fabian Groffen (grobian@gentoo.org)
+## 2006-12-02  --  Fabian Groffen (grobian@gentoo.org)
+##     Don't set SHELL variable, should be done by login (bug #91725)
+##     Properly call tset, such that it actually does something (bug #156422)
+## 2005-09-27  --  Fabian Groffen
 ##     Resolved reported bug on MOTD (disabled it)
 ## 2003-01-13  --  Alain Penders (alain@gentoo.org)
 ##     Initial version.  Inspired by the Suse version.
@@ -14,9 +17,7 @@
     # Console
     if ( ! ${?TERM} )           setenv TERM linux
     if ( "$TERM" == "unknown" ) setenv TERM linux
-    # No tset available on SlackWare
     if ( -x "`which stty`" ) stty sane cr0 pass8 dec
-    if ( -x "`which tset`" ) tset -I -Q
     unsetenv TERMCAP
     settc km yes
 endif
@@ -27,11 +30,6 @@
 umask 022
 
 ##
-## Set our SHELL variable.
-##
-setenv SHELL /bin/tcsh
-
-##
 ## Setup a default MAIL variable
 ##
 if ( -f /var/spool/mail/$USER ) then
