--- tcsh-6.14.00/Makefile.in
+++ tcsh-6.14.00/Makefile.in
@@ -135,9 +135,12 @@
 #DFLAGS=-DDELL
 #DFLAGS=
 #DFLAGS=-D_PATH_TCSHELL='"${DESTBIN}/tcsh"'
-## The following is set by autoconf.
-DFLAGS = -D_PATH_TCSHELL='"${bindir}/tcsh"' @DFLAGS@ @CPPFLAGS@
-
+## Set the path for various files, with help from autoconf
+DFLAGS = -D_PATH_TCSHELL='"${bindir}/tcsh"' \
+         -D_PATH_DOTCSHRC='"@sysconfdir@/csh.cshrc"' \
+         -D_PATH_DOTLOGIN='"@sysconfdir@/csh.login"' \
+         -D_PATH_DOTLOGOUT='"@sysconfdir@/csh.logout"' \
+		 @DFLAGS@ @CPPFLAGS@
 
 ################################################################
 ## LDFLAGS.  Define something here if you need to
