--- build/top.mk.orig	2008-02-12 00:24:07 +0100
+++ build/top.mk	2009-03-18 17:39:38 +0100
@@ -105,19 +105,19 @@
 
 LTSTATIC = @LTSTATIC@
 
-LTLINK   = $(LIBTOOL) --mode=link \
+LTLINK   = $(LIBTOOL) --mode=link --tag=CC \
 	$(CC) $(LTSTATIC) $(LT_CFLAGS) $(LDFLAGS) $(LTFLAGS)
 
-LTCOMPILE_LIB = $(LIBTOOL) $(LTONLY_LIB) --mode=compile \
+LTCOMPILE_LIB = $(LIBTOOL) $(LTONLY_LIB) --mode=compile --tag=CC \
 	$(CC) $(LT_CFLAGS) $(LT_CPPFLAGS) $(LIB_DEFS) -c
 
-LTLINK_LIB = $(LIBTOOL) $(LTONLY_LIB) --mode=link \
+LTLINK_LIB = $(LIBTOOL) $(LTONLY_LIB) --mode=link --tag=CC \
 	$(CC) $(LT_CFLAGS) $(LDFLAGS) $(LTFLAGS_LIB)
 
-LTCOMPILE_MOD = $(LIBTOOL) $(LTONLY_MOD) --mode=compile \
+LTCOMPILE_MOD = $(LIBTOOL) $(LTONLY_MOD) --mode=compile --tag=CC \
 	$(CC) $(LT_CFLAGS) $(LT_CPPFLAGS) $(MOD_DEFS) -c
 
-LTLINK_MOD = $(LIBTOOL) $(LTONLY_MOD) --mode=link \
+LTLINK_MOD = $(LIBTOOL) $(LTONLY_MOD) --mode=link --tag=CC \
 	$(CC) $(LT_CFLAGS) $(LDFLAGS) $(LTFLAGS_MOD)
 
 LTINSTALL = $(LIBTOOL) --mode=install $(INSTALL) 
