Index: mpf/set_str.c
===================================================================
RCS file: /home/cvsfiles/gmp42/mpf/set_str.c,v
retrieving revision 1.6
retrieving revision 1.7
diff -p -2 -r1.6 -r1.7
*** mpf/set_str.c	25 Aug 2008 14:13:11 -0000	1.6
--- mpf/set_str.c	9 Nov 2008 00:28:18 -0000	1.7
*************** mpf_set_str (mpf_ptr x, const char *str,
*** 138,142 ****
--- 138,147 ----
      }
  
+   /* Default base to decimal.  */
+   if (base == 0)
+     base = 10;
+ 
    exp_base = base;
+ 
    if (base < 0)
      {
*************** mpf_set_str (mpf_ptr x, const char *str,
*** 166,173 ****
      }
  
-   /* Default base to decimal.  */
-   if (base == 0)
-     base = 10;
- 
    /* Locate exponent part of the input.  Look from the right of the string,
       since the exponent is usually a lot shorter than the mantissa.  */
--- 171,174 ----
