--- gcc/config/rs6000/aix43.h.orig	2009-05-04 17:37:02 +0200
+++ gcc/config/rs6000/aix43.h	2009-05-04 17:38:16 +0200
@@ -143,7 +143,7 @@
    "
 
 #undef LINK_SPEC
-#define LINK_SPEC "-bpT:0x10000000 -bpD:0x20000000 %{!r:-btextro} -bnodelcsect\
+#define LINK_SPEC "-brtl -bpT:0x10000000 -bpD:0x20000000 %{!r:-btextro} -bnodelcsect\
    %{static:-bnso %(link_syscalls) } %{shared:-bM:SRE %{!e:-bnoentry}}\
    %{!maix64:%{!shared:%{g*: %(link_libg) }}} %{maix64:-b64}\
    %{mpe:-binitfini:poe_remote_main}"
--- gcc/config/rs6000/aix51.h.orig	2009-05-04 17:37:47 +0200
+++ gcc/config/rs6000/aix51.h	2009-05-04 17:38:19 +0200
@@ -138,7 +138,7 @@
    %{pthread:}-lpthreads -lc"
 
 #undef LINK_SPEC
-#define LINK_SPEC "-bpT:0x10000000 -bpD:0x20000000 %{!r:-btextro} -bnodelcsect\
+#define LINK_SPEC "-brtl -bpT:0x10000000 -bpD:0x20000000 %{!r:-btextro} -bnodelcsect\
    %{static:-bnso %(link_syscalls) } %{shared:-bM:SRE %{!e:-bnoentry}}\
    %{!maix64:%{!shared:%{g*: %(link_libg) }}} %{maix64:-b64}\
    %{mpe:-binitfini:poe_remote_main}"
--- gcc/config/rs6000/aix52.h.orig	2009-05-04 17:37:50 +0200
+++ gcc/config/rs6000/aix52.h	2009-05-04 17:38:22 +0200
@@ -140,7 +140,7 @@
    %{pthread:}-lpthreads -lc"
 
 #undef LINK_SPEC
-#define LINK_SPEC "-bpT:0x10000000 -bpD:0x20000000 %{!r:-btextro} -bnodelcsect\
+#define LINK_SPEC "-brtl -bpT:0x10000000 -bpD:0x20000000 %{!r:-btextro} -bnodelcsect\
    %{static:-bnso %(link_syscalls) } %{shared:-bM:SRE %{!e:-bnoentry}}\
    %{!maix64:%{!shared:%{g*: %(link_libg) }}} %{maix64:-b64}\
    %{mpe:-binitfini:poe_remote_main}"
--- gcc/config/rs6000/aix61.h.orig	2009-05-04 17:37:59 +0200
+++ gcc/config/rs6000/aix61.h	2009-05-04 17:38:24 +0200
@@ -143,7 +143,7 @@
    %{pthread:}-lpthreads -lc"
 
 #undef LINK_SPEC
-#define LINK_SPEC "-bpT:0x10000000 -bpD:0x20000000 %{!r:-btextro} -bnodelcsect\
+#define LINK_SPEC "-brtl -bpT:0x10000000 -bpD:0x20000000 %{!r:-btextro} -bnodelcsect\
    %{static:-bnso %(link_syscalls) } %{shared:-bM:SRE %{!e:-bnoentry}}\
    %{!maix64:%{!shared:%{g*: %(link_libg) }}} %{maix64:-b64}\
    %{mpe:-binitfini:poe_remote_main}"
