--- Makefile.in
+++ Makefile.in
@@ -457,7 +457,7 @@
 libtclsqlite3.la:	tclsqlite.lo libsqlite3.la
 	$(LTLINK) -o $@ tclsqlite.lo \
 		libsqlite3.la @TCL_STUB_LIB_SPEC@ $(TLIBS) \
-                -rpath "$(libdir)/sqlite" \
+		-rpath "$(libdir)/sqlite3" \
 		-module -avoid-version
 
 sqlite3$(TEXE):	$(TOP)/src/shell.c libsqlite3.la sqlite3.h
@@ -765,8 +765,13 @@
 	$(INSTALL) -d $(DESTDIR)$(pkgconfigdir)
 	$(INSTALL) -m 0644 sqlite3.pc $(DESTDIR)$(pkgconfigdir)
 
-tcl_install:	libtclsqlite3.la
-	$(TCLSH_CMD) $(TOP)/tclinstaller.tcl $(RELEASE)
+pkgIndex.tcl:
+	echo 'package ifneeded sqlite3 $(RELEASE) [list load $(libdir)/sqlite3/libtclsqlite3.so sqlite3]' > $@
+tcl_install:	libtclsqlite3.la pkgIndex.tcl
+	$(INSTALL) -d $(DESTDIR)$(libdir)/sqlite3
+	$(LTINSTALL) libtclsqlite3.la $(DESTDIR)$(libdir)/sqlite3
+	rm -f $(DESTDIR)$(libdir)/sqlite3/libtclsqlite3.a $(DESTDIR)$(libdir)/sqlite3/libtclsqlite3.la
+	$(INSTALL) -m 0644 pkgIndex.tcl $(DESTDIR)$(libdir)/sqlite3
 
 clean:	
 	rm -f *.lo *.la *.o sqlite3$(TEXE) libsqlite3.la
