diff -ru sqlite-3.6.11.orig/src/os_unix.c sqlite-3.6.11/src/os_unix.c
--- sqlite-3.6.11.orig/src/os_unix.c	2009-03-31 13:36:12 +0200
+++ sqlite-3.6.11/src/os_unix.c	2009-03-31 13:40:17 +0200
@@ -119,6 +119,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <time.h>
+#include <utime.h>
 #include <sys/time.h>
 #include <errno.h>
 
@@ -1751,7 +1752,7 @@
     pFile->locktype = locktype;
 #if !OS_VXWORKS
     /* Always update the timestamp on the old file */
-    utimes(zLockFile, NULL);
+    utime(zLockFile, NULL);
 #endif
     return SQLITE_OK;
   }
