Index: taglib-1.4/taglib/flac/flacfile.cpp
===================================================================
--- taglib-1.4.orig/taglib/flac/flacfile.cpp
+++ taglib-1.4/taglib/flac/flacfile.cpp
@@ -43,6 +43,7 @@ class FLAC::File::FilePrivate
 public:
   FilePrivate() :
     ID3v2FrameFactory(ID3v2::FrameFactory::instance()),
+    tag(0),
     ID3v2Tag(0),
     ID3v2Location(-1),
     ID3v2OriginalSize(0),
