diff -ru printproto-1.0.4.orig/Printstr.h printproto-1.0.4/Printstr.h
--- printproto-1.0.4.orig/Printstr.h	2008-09-26 07:59:00 +0200
+++ printproto-1.0.4/Printstr.h	2008-09-26 08:00:44 +0200
@@ -67,6 +67,15 @@
 #endif /* _XP_PRINT_SERVER_ */
 
 /*
+ * We need to include Xwindows.h, because the definition
+ * of PCONTEXT below breaks winnt.h, which, of course, could
+ * be included after this file otherwise ...
+ */
+#ifdef WIN32
+# include <X11/Xwindows.h>
+#endif
+
+/*
  * Pull in other definitions.  Print.h will hide some things if we're
  * doing server side work.
  */
