--- util/fluxbox-generate_menu.in	2006-06-10 19:04:11.000000000 +0200
+++ util/fluxbox-generate_menu.in.new	2006-06-14 14:39:04.000000000 +0200
@@ -1735,12 +1735,8 @@
 append_submenu "${FBSETTINGSMENU}"
     append "[config] (${CONFIGUREMENU})"
 
-    append_menu "[submenu] (${SYSTEMSTYLES}) {${STYLEMENUTITLE}}"
-        append "[stylesdir] (${PREFIX}/share/fluxbox/styles)"
-    append_menu_end
-
-    append_menu "[submenu] (${USERSTYLES}) {${STYLEMENUTITLE}}"
-        append "[stylesdir] (~/.fluxbox/styles)"
+    append_submenu "Styles"
+        append "[include] (/usr/share/fluxbox/menu.d/styles/)"
     append_menu_end
 
     # Backgroundmenu
