diff -NrU5 vigra1.6.0.orig/test/convolution/test.cxx vigra1.6.0/test/convolution/test.cxx
--- vigra1.6.0.orig/test/convolution/test.cxx	2009-05-17 14:30:10.674673754 +0200
+++ vigra1.6.0/test/convolution/test.cxx	2009-05-17 14:31:45.388542661 +0200
@@ -35,10 +35,11 @@
 /*                                                                      */
 /************************************************************************/
 
 #include <iostream>
 #include <cmath>
+#include <cstdio>
 #include "vigra/convolution.hxx"
 #include "unittest.hxx"
 #include "vigra/stdimage.hxx"
 #include "vigra/impex.hxx"
 #include "vigra/combineimages.hxx"
