Index: hoe-2.4.0/test/test_hoe_gemcutter.rb
===================================================================
--- hoe-2.4.0.orig/test/test_hoe_gemcutter.rb
+++ hoe-2.4.0/test/test_hoe_gemcutter.rb
@@ -1,4 +1,5 @@
 require "hoe/gemcutter" unless defined? Hoe::Gemcutter
+require 'minitest/autorun'
 
 class TestHoeGemcutter < MiniTest::Unit::TestCase
   include Hoe::Gemcutter
Index: hoe-2.4.0/test/test_hoe.rb
===================================================================
--- hoe-2.4.0.orig/test/test_hoe.rb
+++ hoe-2.4.0/test/test_hoe.rb
@@ -49,7 +49,7 @@ class TestHoe < MiniTest::Unit::TestCase
     assert_equal 'blah', spec.rubyforge_project
     assert_equal Gem::RubyGemsVersion, spec.rubygems_version
     assert_match(/^Hoe.*Rakefiles$/, spec.summary)
-    assert_equal files.grep(/^test/), spec.test_files
+    assert_equal files.grep(/^test/).sort, spec.test_files.sort
 
     deps = spec.dependencies
 
