https://savannah.gnu.org/patch/index.php?6691

--- libtool-1.5.26/libtool.m4	Fri Feb  1 17:26:46 2008
+++ libtool-1.5.26.mint/libtool.m4	Mon Apr  7 09:09:46 2008
@@ -820,6 +820,12 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  mint*)
+    # On MiNT this test can take minutes to run, and I'm not sure it's
+    # healthy for this platform. Let's restrict it to 8192.
+    lt_cv_sys_max_cmd_len=8192;
+    ;;
+
   amigaos*)
     # On AmigaOS with pdksh, this test takes hours, literally.
     # So we just punt and use a minimum line length of 8192.
@@ -5037,8 +5043,8 @@
       # Common symbols not allowed in MH_DYLIB files
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)='-fno-common'
       ;;
-    *djgpp*)
-      # DJGPP does not support shared libraries at all
+    *djgpp* | mint*)
+      # DJGPP/MiNT does not support shared libraries at all
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)=
       ;;
     interix[[3-9]]*)
