http://bugs.gentoo.org/show_bug.cgi?id=213277

--- libtool.m4.orig	2008-04-10 18:10:03 +0200
+++ libtool.m4	2008-04-10 18:10:38 +0200
@@ -1414,7 +1414,7 @@
 aix[[4-9]]*)
   version_type=linux
   need_lib_prefix=no
-  need_version=no
+  need_version=yes
   hardcode_into_libs=yes
   if test "$host_cpu" = ia64; then
     # AIX 5 supports IA64
@@ -3075,7 +3075,7 @@
 	for ld_flag in $LDFLAGS; do
 	  case $ld_flag in
 	  *-brtl*)
-	    aix_use_runtimelinking=yes
+#	    aix_use_runtimelinking=yes
 	    break
 	    ;;
 	  esac
@@ -3121,7 +3121,7 @@
 	;;
       esac
       shared_flag='-shared'
-      if test "$aix_use_runtimelinking" = yes; then
+      if test always -o "$aix_use_runtimelinking" = yes; then
 	shared_flag="$shared_flag "'${wl}-G'
       fi
     else
@@ -3131,7 +3131,7 @@
 	# chokes on -Wl,-G. The following line is correct:
 	shared_flag='-G'
       else
-	if test "$aix_use_runtimelinking" = yes; then
+	if test always -o "$aix_use_runtimelinking" = yes; then
 	  shared_flag='${wl}-G'
 	else
 	  shared_flag='${wl}-bM:SRE'
@@ -5968,7 +5968,7 @@
 	case $host_os in aix4.[[23]]|aix4.[[23]].*|aix[[5-9]]*)
 	  for ld_flag in $LDFLAGS; do
   	  if (test $ld_flag = "-brtl" || test $ld_flag = "-Wl,-brtl"); then
-  	    aix_use_runtimelinking=yes
+#  	    aix_use_runtimelinking=yes
   	    break
   	  fi
 	  done
@@ -6013,7 +6013,7 @@
 	  ;;
 	esac
 	shared_flag='-shared'
-	if test "$aix_use_runtimelinking" = yes; then
+	if test always -o "$aix_use_runtimelinking" = yes; then
 	  shared_flag="$shared_flag "'${wl}-G'
 	fi
       else
@@ -6023,7 +6023,7 @@
   	# chokes on -Wl,-G. The following line is correct:
 	  shared_flag='-G'
 	else
-	  if test "$aix_use_runtimelinking" = yes; then
+	  if test always -o "$aix_use_runtimelinking" = yes; then
 	    shared_flag='${wl}-G'
 	  else
 	    shared_flag='${wl}-bM:SRE'
