Index: gentoo-metadata.vim
===================================================================
--- gentoo-syntax/syntax/gentoo-metadata.vim	(revision 48)
+++ gentoo-syntax/syntax/gentoo-metadata.vim	(working copy)
@@ -20,8 +20,6 @@
 
 syn cluster xmlTagHook add=metadataElement
 syn match metadataElement contained 'packages'
-syn match metadataElement contained 'catmetadata'
-syn match metadataElement contained 'pkgmetadata'
 syn match metadataElement contained 'herd'
 syn match metadataElement contained 'maintainer'
 syn match metadataElement contained 'email'
@@ -36,6 +34,9 @@
 syn match metadataElement contained 'pkg'
 syn match metadataElement contained 'cat'
 
+syn match metadataElement contained 'catmetadata'
+syn match metadataElement contained 'pkgmetadata'
+
 hi def link metadataElement Keyword
 
 let b:current_syntax = "gentoo-metadata"

