Index: plugin/gentoo-common.vim
===================================================================
--- plugin/gentoo-common.vim	(revision 138)
+++ plugin/gentoo-common.vim	(working copy)
@@ -28,7 +28,7 @@
         return g:gentooarch
     else
         let l:a = system("portageq envvar ACCEPT_KEYWORDS 2>/dev/null")
-        let l:a = substitute(l:a, "\\n\|\\~", " ", "g")
+        let l:a = substitute(l:a, "[\\n~]", " ", "g")
         let l:a = substitute(l:a, "^\\s\\+", "", "")
         let l:a = substitute(l:a, "\\s.*", "", "")
         if l:a == ""
