--- setup.py~	2007-09-24 02:18:57 +0200
+++ setup.py	2007-09-24 02:19:35 +0200
@@ -50,11 +50,11 @@
 		('share/man/man1', ['man/java-config-2.1']),
 		('share/java-config-2/launcher', ['src/launcher.bash']),
 		('share/eselect/modules', glob('src/eselect/*.eselect')),
-		('/etc/java-config-2/', ['config/virtuals']),
-		('/etc/java-config-2/build/', ['config/jdk.conf','config/compilers.conf']),
-		('/etc/env.d/',['config/20java-config']),
-		('/etc/profile.d/', glob('src/profile.d/*')),
-		('/etc/revdep-rebuild/', ['src/revdep-rebuild/60-java'])
+		('@GENTOO_PORTAGE_EPREFIX@/etc/java-config-2/', ['config/virtuals']),
+		('@GENTOO_PORTAGE_EPREFIX@/etc/java-config-2/build/', ['config/jdk.conf','config/compilers.conf']),
+		('@GENTOO_PORTAGE_EPREFIX@/etc/env.d/',['config/20java-config']),
+		('@GENTOO_PORTAGE_EPREFIX@/etc/profile.d/', glob('src/profile.d/*')),
+		('@GENTOO_PORTAGE_EPREFIX@/etc/revdep-rebuild/', ['src/revdep-rebuild/60-java'])
 	]
 )
 
--- src/depend-java-query~	2007-09-24 04:28:47 +0200
+++ src/depend-java-query	2007-09-24 04:29:03 +0200
@@ -6,7 +6,7 @@
 # $Header: $
 
 import sys
-sys.path = ['/usr/share/java-config-2/pym']+sys.path
+sys.path = ['@GENTOO_PORTAGE_EPREFIX@/usr/share/java-config-2/pym']+sys.path
 
 from java_config import __version__
 from java_config.OutputFormatter import OutputFormatter
--- src/gjl~	2007-09-24 04:31:48 +0200
+++ src/gjl	2007-09-24 04:33:55 +0200
@@ -6,7 +6,7 @@
 # $Header: $
 
 import sys
-sys.path = ['/usr/share/java-config-2/pym']+sys.path
+sys.path = ['@GENTOO_PORTAGE_EPREFIX@/usr/share/java-config-2/pym']+sys.path
 
 from java_config import __version__
 from java_config.OutputFormatter import *
@@ -115,7 +115,7 @@
     envlp = os.getenv('LD_LIBRARY_PATH')
     envjlp = os.getenv('JAVA_LIBRARY_PATH')
 
-    newlibrary = '/lib:/usr/lib'
+    newlibrary = '@GENTOO_PORTAGE_EPREFIX@/lib:@GENTOO_PORTAGE_EPREFIX@/usr/lib'
     if library:
     	newlibrary = ':'.join((library, newlibrary))
     if envjlp:
--- src/java-config-2~	2007-09-24 04:34:13 +0200
+++ src/java-config-2	2007-09-24 04:35:03 +0200
@@ -6,7 +6,7 @@
 # $Header: $
 
 import sys
-sys.path = ['/usr/share/java-config-2/pym']+sys.path
+sys.path = ['@GENTOO_PORTAGE_EPREFIX@/usr/share/java-config-2/pym']+sys.path
 
 from java_config import __version__
 from java_config.OutputFormatter import OutputFormatter
@@ -171,7 +171,7 @@
 # Deprecated
 def system_classpath_target():
     # TODO: MAKE THIS MODULAR!! (compnerd)
-    return [{'file': '/etc/env.d/21java-classpath', 'format': '%s=%s\n' }]
+    return [{'file': '@GENTOO_PORTAGE_EPREFIX@/etc/env.d/21java-classpath', 'format': '%s=%s\n' }]
 
 def user_classpath_target():
     # TODO: MAKE THIS MODULAR!! (compnerd)
@@ -254,7 +254,7 @@
         fatalError("The vm could not be found")
 
 def update_env():
-    printer._print(getoutput("/usr/sbin/env-update"))
+    printer._print(getoutput("@GENTOO_PORTAGE_EPREFIX@/usr/sbin/env-update"))
     printer._printAlert("If you want the changes too take effect in your current session, you should update\n\
             your environment by running: source /etc/profile")
 
--- src/run-java-tool~	2006-09-30 19:10:43 +0200
+++ src/run-java-tool	2007-09-24 04:40:49 +0200
@@ -1,10 +1,10 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 
 user_vm="${HOME}/.gentoo/java-config-2/current-user-vm"
-system_vm="/etc/java-config-2/current-system-vm"
+system_vm="@GENTOO_PORTAGE_EPREFIX@/etc/java-config-2/current-system-vm"
 # Try GENTOO_VM
 if [[ -n ${GENTOO_VM} ]]; then
-	vmpath="/usr/lib/jvm/${GENTOO_VM}/"
+	vmpath="@GENTOO_PORTAGE_EPREFIX@/usr/lib/jvm/${GENTOO_VM}/"
 # Then user VM
 elif [[ -h ${user_vm} ]]; then
 	vmpath=$(readlink ${user_vm})
--- src/eselect/java-nsplugin.eselect~	2007-09-24 04:46:49 +0200
+++ src/eselect/java-nsplugin.eselect	2007-09-24 04:48:16 +0200
@@ -10,10 +10,10 @@
 VERSION=$(svn_date_to_version "${SVN_DATE}" )
 
 
-PLUGINS_HOME="${ROOT}/usr/share/java-config-2/nsplugin"
-SYSTEM_PLUGIN_DIR="${ROOT}/usr/lib/nsbrowser/plugins"
-SYSTEM_PLUGIN_DIR_32="${ROOT}/usr/lib32/nsbrowser/plugins"
-SYSTEM_PLUGIN_DIR_64="${ROOT}/usr/lib64/nsbrowser/plugins"
+PLUGINS_HOME="${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/share/java-config-2/nsplugin"
+SYSTEM_PLUGIN_DIR="${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/lib/nsbrowser/plugins"
+SYSTEM_PLUGIN_DIR_32="${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/lib32/nsbrowser/plugins"
+SYSTEM_PLUGIN_DIR_64="${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/lib64/nsbrowser/plugins"
 SYSTEM_PLUGIN="${SYSTEM_PLUGIN_DIR}/javaplugin.so"
 SYSTEM_PLUGIN_32="${SYSTEM_PLUGIN_DIR_32}/javaplugin.so"
 SYSTEM_PLUGIN_64="${SYSTEM_PLUGIN_DIR_64}/javaplugin.so"
--- src/eselect/java-vm.eselect~	2007-09-24 04:46:55 +0200
+++ src/eselect/java-vm.eselect	2007-09-24 04:49:00 +0200
@@ -9,8 +9,8 @@
 SVN_DATE='$Date: $'
 VERSION=$(svn_date_to_version "${SVN_DATE}" )
 
-VM_BASE=/usr/lib/jvm/
-VM_SYSTEM="/etc/java-config-2/current-system-vm"
+VM_BASE="@GENTOO_PORTAGE_EPREFIX@"/usr/lib/jvm/
+VM_SYSTEM="@GENTOO_PORTAGE_EPREFIX@"/etc/java-config-2/current-system-vm
 VM_USER="${HOME}/.gentoo/java-config-2/current-user-vm"
 
 find_targets() {
--- src/launcher.bash~	2007-09-24 04:54:27 +0200
+++ src/launcher.bash	2007-09-24 04:56:45 +0200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 # Not-so-elegant? patches more then welcome
 
 abort() {
@@ -9,7 +9,7 @@
 # Source package env
 # ---------------------
 gjl_user_env="${HOME}/.gentoo/java-config-2/launcher.d/${gjl_package}"
-gjl_system_env="/etc/java-config-2/launcher.d/${gjl_package}"
+gjl_system_env="@GENTOO_PORTAGE_EPREFIX@/etc/java-config-2/launcher.d/${gjl_package}"
 if [[ -f "${gjl_user_env}" ]]; then
 	source "${gjl_user_env}"
 elif [[ -f "${gjl_system_env}" ]]; then
--- src/profile.d/java-config-2.csh~	2007-09-24 05:03:55 +0200
+++ src/profile.d/java-config-2.csh	2007-09-24 05:06:00 +0200
@@ -3,7 +3,7 @@
 # $Header: /var/cvsroot/gentoo-x86/dev-java/java-config/files/java-config-2.profiled.csh,v 1.3 2007/01/13 17:17:08 grobian Exp $
 
 set gentoo_user_vm = "${HOME}/.gentoo/java-config-2/current-user-vm"
-set gentoo_system_vm = "/etc/java-config-2/current-system-vm"
+set gentoo_system_vm = "@GENTOO_PORTAGE_EPREFIX@"/etc/java-config-2/current-system-vm
 
 ## If we have a current-user-vm (and aren't root)... set it to JAVA_HOME
 ## Otherwise set to the current system vm
--- src/profile.d/java-config-2.sh~	2007-09-24 05:04:00 +0200
+++ src/profile.d/java-config-2.sh	2007-09-24 05:07:24 +0200
@@ -4,7 +4,7 @@
 
 # If we have a current-user-vm (and aren't root)... set it to JAVA_HOME
 gentoo_user_vm="${HOME}/.gentoo/java-config-2/current-user-vm"
-gentoo_system_vm="/etc/java-config-2/current-system-vm"
+gentoo_system_vm="@GENTOO_PORTAGE_EPREFIX@"/etc/java-config-2/current-system-vm
 
 # Please make sure that this script is POSIX compliant
 # See https://bugs.gentoo.org/show_bug.cgi?id=169925
@@ -23,7 +23,7 @@
 if [ "${user_id}" != 0 -a -L "${gentoo_user_vm}" ]; then
 	export JAVA_HOME=${gentoo_user_vm}
 # Otherwise set to the current system vm
-elif [ -L /etc/java-config-2/current-system-vm ]; then
+elif [ -L "@GENTOO_PORTAGE_EPREFIX@"/etc/java-config-2/current-system-vm ]; then
 	export JAVA_HOME=${gentoo_system_vm}
 fi
 
--- src/java_config/EnvironmentManager.py~	2007-09-24 05:11:48 +0200
+++ src/java_config/EnvironmentManager.py	2007-09-24 05:13:22 +0200
@@ -22,9 +22,9 @@
     active = None
 
     # Location of the vm ev files
-    vms_path = '/usr/share/java-config-2/vm'
+    vms_path = '@GENTOO_PORTAGE_EPREFIX@/usr/share/java-config-2/vm'
     # Location of the package env files to load
-    pkg_path = '/usr/share/*/package.env'
+    pkg_path = '@GENTOO_PORTAGE_EPREFIX@/usr/share/*/package.env'
 
     def __init__(self):
         pass
@@ -73,7 +73,7 @@
 
         virtual_prefs = {}
         try:
-            vprefs = EnvFileParser("/etc/java-config-2/virtuals")
+            vprefs = EnvFileParser("@GENTOO_PORTAGE_EPREFIX@/etc/java-config-2/virtuals")
             virtual_prefs = vprefs.get_config()
         except:
             pass
@@ -224,7 +224,7 @@
         elif os.path.exists(target):
             raise InvalidConfigError(target)
 
-        os.symlink('/usr/lib/jvm/'+vm.name(),target)
+        os.symlink('@GENTOO_PORTAGE_EPREFIX@/usr/lib/jvm/'+vm.name(),target)
 
     def vm_links(self):
         # Don't try to use user-vm if HOME is undefined
@@ -237,7 +237,7 @@
         return  os.path.join(os.environ.get('HOME'), '.gentoo/java-config-2/current-user-vm')
 
     def system_vm_link(self):
-        return '/etc/java-config-2/current-system-vm'
+        return '@GENTOO_PORTAGE_EPREFIX@/etc/java-config-2/current-system-vm'
 
     def clean_classpath(self, targets):
         for target in targets:
--- src/java_config/VM.py~	2007-09-24 05:11:51 +0200
+++ src/java_config/VM.py	2007-09-24 05:12:46 +0200
@@ -44,7 +44,7 @@
 
     def name(self):
         # TODO: MAKE THIS MODULAR!a (compnerd)
-        return self.file.split("/usr/share/java-config-2/vm/")[-1]
+        return self.file.split("@GENTOO_PORTAGE_EPREFIX@/usr/share/java-config-2/vm/")[-1]
 
     def is_jre(self):
         return self.is_type("JRE")
--- src/java_config/VersionManager.py~	2007-09-24 05:11:56 +0200
+++ src/java_config/VersionManager.py	2007-09-24 05:12:31 +0200
@@ -5,7 +5,7 @@
 # $Header: $
 
 import sys
-sys.path.append("/usr/lib/portage/pym")
+sys.path.append("@GENTOO_PORTAGE_EPREFIX@/usr/lib/portage/pym")
 
 import re
 from string import upper
@@ -26,7 +26,7 @@
     """
     #atom_parser = re.compile(r"([~!<>=]*)virtual/(jre|jdk)-([0-9\.]+)")
     atom_parser = re.compile(r"([<>=]+)virtual/(jre|jdk)-([0-9\.*]+)")
-    pref_files = ['/etc/java-config-2/build/jdk.conf', '/usr/share/java-config-2/config/jdk-defaults.conf']
+    pref_files = ['@GENTOO_PORTAGE_EPREFIX@/etc/java-config-2/build/jdk.conf', '@GENTOO_PORTAGE_EPREFIX@/usr/share/java-config-2/config/jdk-defaults.conf']
     _prefs = None
 
     def __init__(self):
--- config/20java-config~	2007-09-24 15:16:59 +0200
+++ config/20java-config	2007-09-24 15:20:37 +0200
@@ -1,2 +1,2 @@
-MANPATH=/etc/java-config/system-vm/man/
+MANPATH="@GENTOO_PORTAGE_EPREFIX@"/etc/java-config/system-vm/man/
 CONFIG_PROTECT_MASK="/etc/env.d/java/"
