diff -ru --exclude .svn tags/2.1.6/src/eselect/java-nsplugin.eselect trunk/src/eselect/java-nsplugin.eselect
--- tags/2.1.6/src/eselect/java-nsplugin.eselect	2008-04-28 02:58:44.000000000 +0300
+++ trunk/src/eselect/java-nsplugin.eselect	2008-06-16 22:00:26.000000000 +0300
@@ -68,7 +68,7 @@
 		system_name=$(get_system_plugin_vm)
 
 		for (( i = 0 ; i < ${#targets[@]} ; i = i + 1 )) ; do
-			twaek_list_item ${targets[${i}]} ${system_name}
+			tweak_list_item ${targets[${i}]} ${system_name}
 		done
 	}
 	
diff -ru --exclude .svn tags/2.1.6/src/java_config_2/VersionManager.py trunk/src/java_config_2/VersionManager.py
--- tags/2.1.6/src/java_config_2/VersionManager.py	2008-04-28 02:58:44.000000000 +0300
+++ trunk/src/java_config_2/VersionManager.py	2008-07-02 20:24:01.000000000 +0300
@@ -71,7 +71,10 @@
             use = os.environ["USE"]
 
             # Local import to avoid initializing portage elsewhere
-            from portage_dep import use_reduce,paren_reduce
+            try:
+                from portage.dep import use_reduce,paren_reduce
+            except ImportError:
+                from portage_dep import use_reduce,paren_reduce
             from portage import flatten
 
             # Normalize white space for Portage
