diff -ru -x '*.Po' -x '*.Plo' mpfr-2.4.1.orig/vasprintf.c mpfr-2.4.1/vasprintf.c
--- mpfr-2.4.1.orig/vasprintf.c	2009-09-24 10:32:14 +0200
+++ mpfr-2.4.1/vasprintf.c	2009-09-24 10:33:44 +0200
@@ -49,6 +49,12 @@
 # endif
 #endif
 
+#ifdef __INTERIX
+# include <stdint.h>
+# undef SIZE_MAX
+# define SIZE_MAX UINT32_MAX
+#endif
+
 #include <string.h>             /* for strlen, memcpy and others */
 
 #include "mpfr-impl.h"
