diff --git a/configure.dynamic.in b/configure.dynamic.in
index 6998825..21aeb3f 100644
--- a/configure.dynamic.in
+++ b/configure.dynamic.in
@@ -1023,7 +1023,7 @@ if test "$gww_archlist" != "" ; then
   LDFLAGS="$LDFLAGS $gww_archlist"
 fi
 AC_SUBST(CFLAGS_NOOPT)
-CFLAGS_NOOPT=`echo $CFLAGS | sed -e s/-O2// | sed -e s/-O//`
+CFLAGS_NOOPT=`echo $CFLAGS | sed -e s/-O.//g | sed -e s/-O//g`
 
 cp $ac_confdir/Makefile.dynamic.in $ac_confdir/Makefile.in
 cp $ac_confdir/Unicode/Makefile.dynamic.in $ac_confdir/Unicode/Makefile.in
diff --git a/configure.in b/configure.in
index 6998825..21aeb3f 100644
--- a/configure.in
+++ b/configure.in
@@ -1023,7 +1023,7 @@ if test "$gww_archlist" != "" ; then
   LDFLAGS="$LDFLAGS $gww_archlist"
 fi
 AC_SUBST(CFLAGS_NOOPT)
-CFLAGS_NOOPT=`echo $CFLAGS | sed -e s/-O2// | sed -e s/-O//`
+CFLAGS_NOOPT=`echo $CFLAGS | sed -e s/-O.//g | sed -e s/-O//g`
 
 cp $ac_confdir/Makefile.dynamic.in $ac_confdir/Makefile.in
 cp $ac_confdir/Unicode/Makefile.dynamic.in $ac_confdir/Unicode/Makefile.in
