diff -Naur fontforge-20071210-orig/fontforge/scripting.c fontforge-20071210/fontforge/scripting.c
--- fontforge-20071210-orig/fontforge/scripting.c	2007-12-05 13:27:51.000000000 -0600
+++ fontforge-20071210/fontforge/scripting.c	2007-12-27 16:45:55.000000000 -0600
@@ -9435,7 +9435,7 @@
 	verbose = getenv("FONTFORGE_VERBOSE")!=NULL;
 }
 
-static void ProcessNativeScript(int argc, char *argv[], FILE *script) {
+void ProcessNativeScript(int argc, char *argv[], FILE *script) {
     int i,j;
     Context c;
     enum token_type tok;
