--- slime-2.0_p20080731-old/slime.el	2008-08-01 15:48:07.000000000 +0200
+++ slime-2.0_p20080731/slime.el	2008-08-01 15:51:15.000000000 +0200
@@ -97,13 +97,7 @@
   (defun slime-changelog-date ()
     "Return the datestring of the latest entry in the ChangeLog file.
 Return nil if the ChangeLog file cannot be found."
-    (let ((changelog (concat slime-path "ChangeLog")))
-      (if (file-exists-p changelog)
-          (with-temp-buffer 
-            (insert-file-contents-literally changelog nil 0 100)
-            (goto-char (point-min))
-            (symbol-name (read (current-buffer))))
-        nil))))
+    "@SLIME-CHANGELOG-DATE@"))
 
 (defvar slime-protocol-version nil)
 (setq slime-protocol-version
