diff -ur slime.orig/slime.el slime/slime.el
--- slime.orig/slime.el	2008-02-18 14:16:13.000000000 +0100
+++ slime/slime.el	2008-02-18 14:19:55.000000000 +0100
@@ -1432,7 +1432,6 @@
     (format "%S\n\n"
             `(progn
                (load ,(expand-file-name loader) :verbose t)
-               (funcall (read-from-string "swank-loader:init"))
                (funcall (read-from-string "swank:start-server")
                         ,port-filename
                         :coding-system ,encoding)))))
Only in slime: slime.el.~1.905.~
