diff -up gnome-control-center-2.24.0.1/capplets/appearance/appearance-main.c.capplet-help gnome-control-center-2.24.0.1/capplets/appearance/appearance-main.c
--- gnome-control-center-2.24.0.1/capplets/appearance/appearance-main.c.capplet-help	2008-10-04 01:40:15.000000000 -0400
+++ gnome-control-center-2.24.0.1/capplets/appearance/appearance-main.c	2008-10-04 01:40:34.000000000 -0400
@@ -85,6 +85,38 @@ main_window_response (GtkWidget *widget,
     g_object_unref (data->client);
     g_object_unref (data->xml);
   }
+  else if (response_id == GTK_RESPONSE_HELP)
+  {
+      GtkNotebook *nb;
+      gint pindex;
+
+      nb = GTK_NOTEBOOK (glade_xml_get_widget (data->xml, "main_notebook"));
+      pindex = gtk_notebook_get_current_page (nb);
+
+      switch (pindex)
+      {
+        case 0: /* theme */
+          capplet_help (GTK_WINDOW (widget), "user-guide.xml", 
+                                    "goscustdesk-12"); 
+          break;
+        case 1: /* background */
+          capplet_help (GTK_WINDOW (widget), "user-guide.xml", 
+                                    "goscustdesk-7"); 
+          break;
+        case 2: /* fonts */
+          capplet_help (GTK_WINDOW (widget), "user-guide.xml", 
+                                    "goscustdesk-38"); 
+          break;
+        case 3: /* interface */
+          capplet_help (GTK_WINDOW (widget), "user-guide.xml", 
+                                    "goscustuserinter-2"); 
+          break;
+        default:
+          capplet_help (GTK_WINDOW (widget), "user-guide.xml", 
+                                    "prefs-look-and-feel"); 
+          break;
+       }
+  }
 }
 
 int
diff -up gnome-control-center-2.24.0.1/capplets/appearance/appearance-style.c.capplet-help gnome-control-center-2.24.0.1/capplets/appearance/appearance-style.c
--- gnome-control-center-2.24.0.1/capplets/appearance/appearance-style.c.capplet-help	2008-10-04 01:40:24.000000000 -0400
+++ gnome-control-center-2.24.0.1/capplets/appearance/appearance-style.c	2008-10-04 01:40:34.000000000 -0400
@@ -392,7 +392,7 @@ static void
 style_response_cb (GtkDialog *dialog, gint response_id)
 {
   if (response_id == GTK_RESPONSE_HELP) {
-    /* FIXME: help */
+    capplet_help (GTK_WINDOW (dialog), "user-guide.xml", "goscustdesk-61");
   } else {
     gtk_widget_hide (GTK_WIDGET (dialog));
   }
