--- capplets/default-applications/gnome-default-applications.xml.in~	2008-03-29 21:53:04.000000000 +0100
+++ capplets/default-applications/gnome-default-applications.xml.in	2008-03-29 21:53:04.000000000 +0100
@@ -62,24 +62,45 @@
       <win-command>mozilla-firebird -remote "openurl(%s,new-window)"</win-command>
     </web-browser>
     <web-browser>
-      <_name>Firefox</_name>
+      <_name>Mozilla Firefox</_name>
       <executable>firefox</executable>
       <command>firefox %s</command>
-      <icon-name>firefox</icon-name>
+      <icon-name>firefox-icon</icon-name>
       <run-in-terminal>false</run-in-terminal>
       <netscape-remote>true</netscape-remote>
       <tab-command>firefox -new-tab "%s"</tab-command>
       <win-command>firefox -new-window "%s"</win-command>
     </web-browser>
+<!-- Commenting because gentoo uses "firefox" executable for all of them causing them to be always shown
     <web-browser>
       <_name>Iceweasel</_name>
-      <executable>iceweasel</executable>
-      <command>iceweasel %s</command>
-      <icon-name>iceweasel</icon-name>
+      <executable>firefox</executable>
+      <command>firefox %s</command>
+      <icon-name>iceweasel-icon</icon-name>
       <run-in-terminal>false</run-in-terminal>
       <netscape-remote>true</netscape-remote>
-      <tab-command>iceweasel -new-tab "%s"</tab-command>
-      <win-command>iceweasel -new-window "%s"</win-command>
+      <tab-command>firefox -new-tab "%s"</tab-command>
+      <win-command>firefox -new-window "%s"</win-command>
+    </web-browser>
+    <web-browser>
+      <_name>Bon Echo</_name>
+      <executable>firefox</executable>
+      <command>firefox %s</command>   
+      <icon-name>firefox-icon-unbranded</icon-name>
+      <run-in-terminal>false</run-in-terminal>
+      <netscape-remote>true</netscape-remote> 
+      <tab-command>firefox -new-tab "%s"</tab-command>
+      <win-command>firefox -new-window "%s"</win-command>
+    </web-browser> -->
+    <web-browser> 
+      <_name>Mozilla Firefox (bin)</_name>
+      <executable>firefox-bin</executable>
+      <command>firefox-bin %s</command>   
+      <icon-name>mozilla-firefox-bin-icon</icon-name>
+      <run-in-terminal>false</run-in-terminal>
+      <netscape-remote>true</netscape-remote> 
+      <tab-command>firefox-bin -new-tab "%s"</tab-command>
+      <win-command>firefox-bin -new-window "%s"</win-command>
     </web-browser>
     <web-browser>
       <_name>Mozilla 1.6</_name>
@@ -112,6 +133,16 @@
       <win-command>seamonkey -remote "openurl(%s,new-window)"</win-command>
     </web-browser>
     <web-browser>
+      <_name>SeaMonkey (bin)</_name>
+      <executable>seamonkey-bin</executable>
+      <command>seamonkey-bin %s</command>
+      <icon-name>seamonkey-bin</icon-name>
+      <run-in-terminal>false</run-in-terminal>
+      <netscape-remote>true</netscape-remote>
+      <tab-command>seamonkey-bin -remote "openurl(%s,new-tab)"</tab-command>
+      <win-command>seamonkey-bin -remote "openurl(%s,new-window)"</win-command>
+    </web-browser>
+    <web-browser>
       <_name>Iceape</_name>
       <executable>iceape</executable>
       <command>iceape %s</command>
@@ -179,17 +210,24 @@
       <run-in-terminal>false</run-in-terminal>
     </mail-reader>
     <mail-reader>
-      <_name>Thunderbird</_name>
+      <_name>Mozilla Thunderbird</_name>
       <executable>thunderbird</executable>
       <command>thunderbird %s</command>
-      <icon-name>thunderbird</icon-name>
+      <icon-name>thunderbird-icon</icon-name>
       <run-in-terminal>false</run-in-terminal>
     </mail-reader>
     <mail-reader>
-      <_name>Mozilla Thunderbird</_name>
-      <executable>mozilla-thunderbird</executable>
-      <command>mozilla-thunderbird %s</command>
-      <icon-name>thunderbird</icon-name>
+      <_name>Mozilla Thunderbird (bin)</_name>
+      <executable>thunderbird-bin</executable>
+      <command>thunderbird-bin %s</command>
+      <icon-name>mozilla-thunderbird-bin-icon</icon-name>
+      <run-in-terminal>false</run-in-terminal>
+    </mail-reader>
+    <mail-reader>
+      <_name>SeaMonkey Mail (bin)</_name>
+      <executable>seamonkey-bin</executable>
+      <command>seamonkey-bin -mail %s</command>  
+      <icon-name>seamonkey-bin</icon-name>
       <run-in-terminal>false</run-in-terminal>
     </mail-reader>
     <mail-reader>
@@ -301,7 +339,7 @@
       <exec-flag>-e</exec-flag>
     </terminal>
     <terminal>
-       <_name>Konsole</_name>
+       <_name>KDE Konsole</_name>
        <executable>konsole</executable>
        <command>konsole</command>
        <icon-name>konsole</icon-name>
@@ -338,6 +376,55 @@
       <icon-name>totem</icon-name>
       <run-in-terminal>false</run-in-terminal>
     </media-player>
+    <media-player>
+      <_name>Banshee Music Player 1 (prerelease)</_name>
+      <executable>banshee-1</executable>
+      <command>banshee</command>
+      <icon-name>media-player-banshee</icon-name>
+      <run-in-terminal>false</run-in-terminal>
+    </media-player>
+    <media-player> 
+      <_name>Audacious Music Player</_name>
+      <executable>audacious</executable>   
+      <command>audacious</command>
+      <icon-name>audacious</icon-name>
+      <run-in-terminal>false</run-in-terminal>
+    </media-player>
+    <media-player> 
+      <_name>Gxine Movie Player</_name>
+      <executable>gxine</executable>
+      <command>gxine</command>
+      <icon-name>gxine</icon-name> 
+      <run-in-terminal>false</run-in-terminal>
+    </media-player>
+    <media-player> 
+      <_name>Kaffeine Movie Player</_name>    
+      <executable>kaffeine</executable>       
+      <command>kaffeine</command>    
+      <icon-name>kaffeine</icon-name>    
+      <run-in-terminal>false</run-in-terminal>
+    </media-player>
+    <media-player> 
+      <_name>Mplayer Movie Player</_name>
+      <executable>gmplayer</executable>    
+      <command>gmplayer</command>   
+      <icon-name>mplayer</icon-name> 
+      <run-in-terminal>false</run-in-terminal>
+    </media-player>
+    <media-player> 
+      <_name>RealPlayer Movie Player</_name>          
+      <executable>realplay</executable>    
+      <command>realplay</command>      
+      <icon-name>realplay</icon-name>  
+      <run-in-terminal>false</run-in-terminal>
+    </media-player>
+    <media-player> 
+      <_name>Xine Movie Player</_name>
+      <executable>xine</executable>    
+      <command>xine</command>      
+      <icon-name>xine</icon-name> 
+      <run-in-terminal>false</run-in-terminal>
+    </media-player>
   </media-players>
 
   <a11y-visual>
