=== modified file 'setup.py'
--- setup.py	2009-10-20 14:04:31 +0000
+++ setup.py	2009-10-20 14:05:01 +0000
@@ -166,26 +166,9 @@
 from distutils.errors import CCompilerError, DistutilsPlatformError
 from distutils.extension import Extension
 ext_modules = []
-try:
-    try:
-        from Pyrex.Distutils import build_ext
-        from Pyrex.Compiler.Version import version as pyrex_version
-    except ImportError:
-        print "No Pyrex, trying Cython..."
-        from Cython.Distutils import build_ext
-        from Cython.Compiler.Version import version as pyrex_version
-except ImportError:
-    have_pyrex = False
-    # try to build the extension from the prior generated source.
-    print
-    print ("The python package 'Pyrex' is not available."
-           " If the .c files are available,")
-    print ("they will be built,"
-           " but modifying the .pyx files will not rebuild them.")
-    print
-    from distutils.command.build_ext import build_ext
-else:
-    have_pyrex = True
+
+have_pyrex = False
+from distutils.command.build_ext import build_ext
 
 
 class build_ext_if_possible(build_ext):

