--- setup.py	2007-11-09 15:19:32.000000000 +0200
+++ setup.py	2007-11-14 18:26:00.000000000 +0200
@@ -154,21 +154,8 @@
 from distutils.errors import CCompilerError, DistutilsPlatformError
 from distutils.extension import Extension
 ext_modules = []
-try:
-    from Pyrex.Distutils import build_ext
-except ImportError:
-    have_pyrex = False
-    # try to build the extension from the prior generated source.
-    print
-    print ("The python package 'Pyrex' is not available."
-           " If the .c files are available,")
-    print ("they will be built,"
-           " but modifying the .pyx files will not rebuild them.")
-    print
-    from distutils.command.build_ext import build_ext
-else:
-    have_pyrex = True
-
+have_pyrex = False
+from distutils.command.build_ext import build_ext
 
 class build_ext_if_possible(build_ext):
 
