https://bugs.launchpad.net/bzr/+bug/109124
--- bzrlib/tests/__init__.py	2007-08-28 22:50:57.000000000 +0300
+++ bzrlib/tests/__init__.py	2007-08-31 14:28:44.000000000 +0300
@@ -933,6 +933,9 @@
             return
         path_stat = transport.stat(path)
         actual_mode = stat.S_IMODE(path_stat.st_mode)
+        # test if running run beneath a directory with the setgid bit set.
+        if actual_mode & stat.S_ISGID and not mode & stat.S_ISGID:
+            mode |= stat.S_ISGID
         self.assertEqual(mode, actual_mode,
             'mode of %r incorrect (%o != %o)' % (path, mode, actual_mode))
 
