diff --exclude-from=/home/dang/bin/scripts/diffrc -up -ruN control-center-2.18.1.orig/libslab/directory-tile.c control-center-2.18.1/libslab/directory-tile.c
--- control-center-2.18.1.orig/libslab/directory-tile.c	2007-03-18 13:04:44.000000000 -0400
+++ control-center-2.18.1/libslab/directory-tile.c	2007-05-01 11:53:39.000000000 -0400
@@ -644,5 +644,5 @@ static void
 disown_spawned_child (gpointer user_data)
 {
 	setsid  ();
-	setpgrp ();
+	setpgid (0, 0);
 }
