diff -ru gv-3.6.1/src/Makefile.am src.orig/Makefile.am
--- src/Makefile.am	2006-04-13 16:24:11.000000000 -0700
+++ src.orig/Makefile.am	2006-04-13 16:25:49.000000000 -0700
@@ -195,7 +195,7 @@
 	@echo "GV.gsX11AlphaDevice:	-sDEVICE=x11 -dTextAlphaBits=4 -dGraphicsAlphaBits=2 -dMaxBitmap=10000000 -dNOPLATFONTS" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.gsSafer:		True" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.gsQuiet:		True" >> $(srcdir)/gv_intern_res_unix.dat
-	@echo "GV.gsArguments:" >> $(srcdir)/gv_intern_res_unix.dat
+	@echo "GV.gsArguments:		-dFIXEDMEDIA" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.uncompressCommand:	gzip -d -c %s > %s" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.printCommand:	$(PRINT_COMMAND)" >> $(srcdir)/gv_intern_res_unix.dat
 
diff -ru gv-3.6.1/src/Makefile.in src.orig/Makefile.in
--- src/Makefile.in	2006-04-13 16:25:04.000000000 -0700
+++ src.orig/Makefile.in	2006-04-13 16:25:49.000000000 -0700
@@ -769,7 +769,7 @@
 	@echo "GV.gsX11AlphaDevice:	-sDEVICE=x11 -dTextAlphaBits=4 -dGraphicsAlphaBits=2 -dMaxBitmap=10000000 -dNOPLATFONTS" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.gsSafer:		True" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.gsQuiet:		True" >> $(srcdir)/gv_intern_res_unix.dat
-	@echo "GV.gsArguments:" >> $(srcdir)/gv_intern_res_unix.dat
+	@echo "GV.gsArguments:		-dFIXEDMEDIA" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.uncompressCommand:	gzip -d -c %s > %s" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.printCommand:	$(PRINT_COMMAND)" >> $(srcdir)/gv_intern_res_unix.dat
 
