diff -ur a/src/output.c b/src/output.c
--- a/src/output.c	2007-07-05 04:56:06 +0100
+++ b/src/output.c	2007-10-15 09:56:23 +0100
@@ -252,6 +252,9 @@
   else if (set_cloexec_flag (fileno (file), true) != 0)
     M4ERROR ((warning_status, errno,
 	      "Warning: cannot protect diversion across forks"));
+  else if (fseeko (file, 0, SEEK_SET) != 0)
+    M4ERROR ((EXIT_FAILURE, errno,
+              "cannot seek to beginning of diversion"));
   return file;
 }
 
