diff -ru -x '*.Po' -x '*.Plo' m4-1.4.13.orig/lib/spawni.c m4-1.4.13/lib/spawni.c
--- m4-1.4.13.orig/lib/spawni.c	2009-04-16 10:21:50 +0200
+++ m4-1.4.13/lib/spawni.c	2009-04-16 10:26:07 +0200
@@ -351,7 +351,9 @@
 	{
 	case EACCES:
 	case ENOENT:
+#ifdef ESTALE
 	case ESTALE:
+#endif
 	case ENOTDIR:
 	  /* Those errors indicate the file is missing or not executable
 	     by us, in which case we want to just try the next path
