diff -ur xfdesktop-4.6.1.orig/modules/menu/desktop-menu.c xfdesktop-4.6.1/modules/menu/desktop-menu.c
--- xfdesktop-4.6.1.orig/modules/menu/desktop-menu.c	2009-02-26 00:32:50.000000000 +0200
+++ xfdesktop-4.6.1/modules/menu/desktop-menu.c	2009-09-23 14:10:16.000000000 +0300
@@ -63,7 +63,7 @@
 #include <libxfcegui4/libxfcegui4.h>
 #include <libxfce4menu/libxfce4menu.h>
 
-#ifdef HAVE_THUNAR_VFS
+#ifdef ENABLE_FILE_ICONS
 #include <thunar-vfs/thunar-vfs.h>
 #endif
 
@@ -82,7 +82,7 @@
 	
     gint idle_id;  /* source id for idled generation */
     
-#ifdef HAVE_THUNAR_VFS
+#ifdef ENABLE_FILE_ICONS
     GList *monitors;
 #endif
 } XfceDesktopMenu;
@@ -134,7 +134,7 @@
         desktop_menu->idle_id = g_idle_add(_generate_menu_idled, desktop_menu);
 }
 
-#ifdef HAVE_THUNAR_VFS
+#ifdef ENABLE_FILE_ICONS
 
 static void
 desktop_menu_something_changed(ThunarVfsMonitor *monitor,
@@ -443,7 +443,7 @@
 xfce_desktop_menu_new_impl(const gchar *menu_file,
                            gboolean deferred)
 {
-#ifdef HAVE_THUNAR_VFS
+#ifdef ENABLE_FILE_ICONS
     static XfceMenuMonitorVTable monitor_vtable = {
         NULL, /*desktop_menu_xfce_menu_monitor_file,*/
         desktop_menu_xfce_menu_monitor_directory,
@@ -460,7 +460,7 @@
     else
         desktop_menu->filename = xfce_desktop_get_menufile();
     
-#ifdef HAVE_THUNAR_VFS
+#ifdef ENABLE_FILE_ICONS
     thunar_vfs_init();
     xfce_menu_monitor_set_vtable(&monitor_vtable, desktop_menu);
 #endif
@@ -469,7 +469,7 @@
         desktop_menu->idle_id = g_idle_add(_generate_menu_idled, desktop_menu);
     else {
         if(!_generate_menu(desktop_menu)) {
-#ifdef HAVE_THUNAR_VFS
+#ifdef ENABLE_FILE_ICONS
             xfce_menu_monitor_set_vtable(NULL, NULL);
 #endif
             g_free(desktop_menu);
@@ -595,7 +595,7 @@
     g_return_if_fail(desktop_menu != NULL);
     TRACE("dummy");
     
-#ifdef HAVE_THUNAR_VFS
+#ifdef ENABLE_FILE_ICONS
     xfce_menu_monitor_set_vtable(NULL, NULL);
 #endif
     
diff -ur xfdesktop-4.6.1.orig/modules/menu/Makefile.am xfdesktop-4.6.1/modules/menu/Makefile.am
--- xfdesktop-4.6.1.orig/modules/menu/Makefile.am	2009-01-13 00:03:21.000000000 +0200
+++ xfdesktop-4.6.1/modules/menu/Makefile.am	2009-09-23 14:04:57.000000000 +0300
@@ -15,11 +15,15 @@
 	$(LIBXFCE4MENU_CFLAGS) \
 	$(LIBX11_CFLAGS) \
 	$(LIBXFCEGUI4_CFLAGS) \
-	$(THUNAR_VFS_CFLAGS) \
 	-DSYSCONFDIR=\"$(sysconfdir)\" \
 	-DDATADIR=\"$(datadir)\" \
 	-DBINDIR=\"$(bindir)\"
 
+if ENABLE_FILE_ICONS
+xfce4_desktop_menu_la_CFLAGS += \
+	$(THUNAR_VFS_CFLAGS)
+endif
+
 xfce4_desktop_menu_la_DEPENDENCIES = \
 	$(top_builddir)/common/libxfdesktop-menu-utils.la
 
@@ -40,5 +44,9 @@
 	$(LIBXFCE4MENU_LIBS) \
 	$(LIBX11_LDFLAGS) \
 	$(LIBX11_LIBS) \
-	$(LIBXFCEGUI4_LIBS) \
+	$(LIBXFCEGUI4_LIBS)
+
+if ENABLE_FILE_ICONS
+xfce4_desktop_menu_la_LIBADD += \
 	$(THUNAR_VFS_LIBS)
+endif
diff -ur xfdesktop-4.6.1.orig/panel-plugin/desktop-menu-plugin.c xfdesktop-4.6.1/panel-plugin/desktop-menu-plugin.c
--- xfdesktop-4.6.1.orig/panel-plugin/desktop-menu-plugin.c	2009-02-26 00:33:00.000000000 +0200
+++ xfdesktop-4.6.1/panel-plugin/desktop-menu-plugin.c	2009-09-23 14:09:30.000000000 +0300
@@ -53,7 +53,7 @@
 #include <libxfce4panel/xfce-panel-plugin.h>
 #include <libxfce4panel/xfce-panel-convenience.h>
 
-#ifdef HAVE_THUNAR_VFS
+#ifdef ENABLE_FILE_ICONS
 #include <thunar-vfs/thunar-vfs.h>
 #endif
 
@@ -896,7 +896,7 @@
 desktop_menu_plugin_preinit(int argc,
                             char **argv)
 {
-#ifdef HAVE_THUNAR_VFS
+#ifdef ENABLE_FILE_ICONS
     g_thread_init(NULL);
 #endif
     return TRUE;
@@ -1018,7 +1018,7 @@
     
     xfce_textdomain(GETTEXT_PACKAGE, LOCALEDIR, "UTF-8");
     
-#ifdef HAVE_THUNAR_VFS
+#ifdef ENABLE_FILE_ICONS
     thunar_vfs_init();
 #endif
     
diff -ur xfdesktop-4.6.1.orig/panel-plugin/Makefile.am xfdesktop-4.6.1/panel-plugin/Makefile.am
--- xfdesktop-4.6.1.orig/panel-plugin/Makefile.am	2009-01-13 00:03:12.000000000 +0200
+++ xfdesktop-4.6.1/panel-plugin/Makefile.am	2009-09-23 14:04:38.000000000 +0300
@@ -9,11 +9,15 @@
 	$(LIBX11_CFLAGS) \
 	$(LIBXFCEGUI4_CFLAGS) \
 	$(LIBXFCE4PANEL_CFLAGS) \
-	$(THUNAR_VFS_CFLAGS) \
 	-DDATADIR=\"$(datadir)\" \
 	-DLOCALEDIR=\"$(localedir)\" \
 	-DBINDIR=\"$(bindir)\"
 
+if ENABLE_FILE_ICONS
+xfce4_menu_plugin_CFLAGS += \
+	$(THUNAR_VFS_CFLAGS)
+endif
+
 xfce4_menu_plugin_DEPENDENCIES = \
 	$(top_builddir)/common/libxfdesktop-menu.la \
 	$(top_builddir)/common/libxfdesktop.la
@@ -23,8 +27,12 @@
 	$(top_builddir)/common/libxfdesktop.la \
 	$(LIBX11_LIBS) \
 	$(LIBXFCEGUI4_LIBS) \
-	$(LIBXFCE4PANEL_LIBS) \
+	$(LIBXFCE4PANEL_LIBS)
+
+if ENABLE_FILE_ICONS
+xfce4_menu_plugin_LDADD += \
 	$(THUNAR_VFS_LIBS)
+endif
 
 if HAVE_CYGWIN
 xfce4_menu_plugin_LDFLAGS = \
