--- src/man-getopt.c
+++ src/man-getopt.c
@@ -233,7 +233,7 @@ get_options_from_string(const char *s) {

 static void 
 mysetenv(const char *name, const char *value) {
-#if defined(__sgi__) || defined(__sun__) || defined(sun) || defined(__hpux)
+#if defined(__sgi) || defined(__sun__) || defined(sun) || defined(__hpux)
     int len = strlen(value)+1+strlen(value)+1;
     char *str = my_malloc(len);
     sprintf(str, "%s=%s", name, value);
--- catopen/catopen.c
+++ catopen/catopen.c
@@ -49,7 +49,7 @@ my_catopen(char *name, int oflag) {
      * "If `name' contains a `/', then `name' specifies a pathname"
      */
     if (index(name, '/')) {
-#ifdef __GLIBC__
+#if defined __GLIBC__ || defined __sgi
       /* glibc uses a pointer type for nl_catd, not a fd */
       return fd;
 #else
