--- man-1.5o2/src/makewhatis.sh.orig	2005-01-07 21:19:37.812163552 -0500
+++ man-1.5o2/src/makewhatis.sh	2005-01-07 21:20:43.023249960 -0500
@@ -41,7 +41,7 @@
 # and should be first.
 # It is a bug to add /var/cache/man to DEFCATPATH.
 dm=
-for d in /usr/man /usr/share/man /usr/X11R6/man /usr/local/man
+for d in /usr/local/man /usr/share/man
 do
     if [ -d $d ]; then
 	if [ x$dm = x ]; then dm=$d; else dm=$dm:$d; fi
@@ -49,7 +49,7 @@
 done
 DEFMANPATH=$dm
 dc=
-for d in /usr/man/preformat /usr/man /usr/share/man/preformat /usr/share/man
+for d in /usr/local/man /usr/share/man
 do
     if [ -d $d ]; then
 	if [ x$dc = x ]; then dc=$d; else dc=$dc:$d; fi
