--- src/man-getopt.c.orig	2007-11-20 14:52:24 +0100
+++ src/man-getopt.c	2007-11-20 14:52:38 +0100
@@ -233,7 +233,7 @@
 
 static void 
 mysetenv(const char *name, const char *value) {
-#if defined(__sgi__) || defined(__sun__) || defined(sun)
+#if defined(__sgi__) || defined(__sun__) || defined(sun) || defined(__hpux)
     int len = strlen(value)+1+strlen(value)+1;
     char *str = my_malloc(len);
     sprintf(str, "%s=%s", name, value);
