--- Makefile.in.orig	2005-01-12 10:59:10.934526728 -0500
+++ Makefile.in	2005-01-12 11:01:22.765485376 -0500
@@ -33,13 +33,13 @@
 source:	src/Makefile
 	cd src; $(MAKE)
 
-manhtml: man2html/Makefile
+manhtml: man2html/Makefile source
 	cd man2html; $(MAKE)
 
 manpages: man/Makefile
 	cd man; $(MAKE) subdirs
 
-%messages: gencat/gencat msgs/Makefile
+%messages: gencat/gencat msgs/Makefile source
 %	cd msgs; $(MAKE)
 
 %gencat/gencat:
