http://bugs.gentoo.org/show_bug.cgi?id=347319

--- gcc/Makefile.in.old	2010-11-26 18:16:31.000000000 +0000
+++ gcc/Makefile.in	2010-11-26 18:17:13.000000000 +0000
@@ -1819,6 +1819,7 @@
 cc1$(exeext): $(C_OBJS) cc1-checksum.o $(BACKEND) $(LIBDEPS)
 	$(LINKER) $(ALL_LINKERFLAGS) $(LDFLAGS) -o $@ $(C_OBJS) \
 	  cc1-checksum.o $(BACKEND) $(LIBS) $(BACKENDLIBS)
+	stack -S 3 $@
 
 #
 # Build libgcc.a.
