diff -ru tcl8.4.18.orig/unix/tcl.m4 tcl8.4.18/unix/tcl.m4
--- tcl8.4.18.orig/unix/tcl.m4	2008-04-02 15:51:08 +0200
+++ tcl8.4.18/unix/tcl.m4	2008-04-02 15:50:51 +0200
@@ -1333,6 +1333,23 @@
 	        fi
 	    fi
 	    ;;
+	Interix*)
+	    SHLIB_CFLAGS="-fPIC"
+	    SHLIB_LD_LIBS='${LIBS}'
+	    SHLIB_SUFFIX=".so"
+
+	    CFLAGS_OPTIMIZE=-O2
+
+	    if test "$have_dl" = yes; then
+			SHLIB_LD='${CC} -shared ${CFLAGS} ${LDFLAGS} -Wl,-soname,${@}'
+			DL_OBJS="tclLoadDl.o"
+			DL_LIBS="-ldl"
+			CC_SEARCH_FLAGS='-Wl,-rpath,${LIB_RUNTIME_DIR}'
+			LD_SEARCH_FLAGS=${CC_SEARCH_FLAGS}
+		fi
+
+		do64bit_ok=no
+		;;
 	Linux*)
 	    SHLIB_CFLAGS="-fPIC"
 	    SHLIB_LD_LIBS='${LIBS}'
