--- subversion-1.4.2/subversion/libsvn_subr/prompt.c.orig	Wed Aug 29 17:16:29 2007
+++ subversion-1.4.2/subversion/libsvn_subr/prompt.c	Wed Aug 29 17:16:32 2007
@@ -50,9 +50,9 @@
  * on Windows it returns APR_EBADF instead.  So just return APR_ENOTIMPL
  * ourselves here.
  */
-#ifdef WIN32
+#if defined(WIN32) || defined(__INTERIX)
   return APR_ENOTIMPL;
-#endif /* WIN32 */
+#endif /* WIN32 || __INTERIX */
   
   pollset.desc_type = APR_POLL_FILE;
   pollset.desc.f = f;
