--- configure.in.orig	2007-05-15 20:22:18.000000000 +0200
+++ configure.in	2007-05-15 20:22:55.000000000 +0200
@@ -69,7 +69,7 @@
 
 # Either a space-separated list of allowable Neon versions, or "any" to
 # mean allow anything.
-NEON_ALLOWED_LIST="0.24.7 0.25.0 0.25.1 0.25.2 0.25.3 0.25.4 0.25.5 0.26.0 0.26.1 0.26.2"
+NEON_ALLOWED_LIST="0.24.7 0.25.0 0.25.1 0.25.2 0.25.3 0.25.4 0.25.5 0.26.0 0.26.1 0.26.2 0.26.3"
 NEON_LATEST_WORKING_VER=0.25.5
 NEON_URL="http://www.webdav.org/neon/neon-${NEON_LATEST_WORKING_VER}.tar.gz"
 dnl You can skip the neon version check only if you know what you are doing 
