--- configure.in.orig	2007-02-27 21:58:01.000000000 +0100
+++ configure.in	2007-02-27 22:00:07.000000000 +0100
@@ -433,7 +433,7 @@
 
 AC_ARG_ENABLE(debug,
 AC_HELP_STRING([--enable-debug],
-	       [Turn on debugging and compile time warnings]),
+	       [(gentoo) disabled, just observe CFLAGS]),
 [
     if test "$enableval" = "yes" ; then
       enable_debugging="yes"
@@ -441,17 +441,6 @@
       enable_debugging="no"
     fi
 ])
-if test "$enable_debugging" = "yes" ; then
-  dnl At the moment, we don't want optimization, because we're
-  dnl debugging.
-  dnl ### actually, debugging should be fine with the default -O2
-  CFLAGS=["`echo $CFLAGS' ' | sed -e 's/-O[^ ]* //g'`"]
-else
-  if test "$enable_debugging" = "no" ; then
-    CFLAGS=["`echo $CFLAGS' ' | sed -e 's/-g[0-9] //g' | sed -e 's/-g//g'`"]
-  fi
-fi
-
 
 AC_ARG_WITH(editor,
 AC_HELP_STRING([--with-editor=PATH],
