--- PDFlib-Lite-7.0.4p4.orig/configure.in	2010-04-02 20:25:18.000000000 +0200
+++ PDFlib-Lite-7.0.4p4/configure.in	2010-04-02 20:26:10.000000000 +0200
@@ -849,9 +849,8 @@
 	    WITH_PYTHON=no
 	else
 	    PYTHONLIBDIR=`cat << EOF | python
-import sys
-import string
-print string.join(filter(lambda x: -1 != string.find(x, 'dynload'), sys.path))
+import distutils.sysconfig
+print distutils.sysconfig.get_python_lib()
 EOF
 `
 	    if test "$PYTHONLIBDIR" = "" ; then
