--- libs/tiff/tiff.h
+++ libs/tiff/tiff.h
@@ -64,6 +64,18 @@
     typedef u_int8	uint8;
     typedef u_int16	uint16;
     typedef u_int32	uint32;
+#elif defined(__APPLE_CC__)
+    #include <stdint.h>
+    typedef uint32_t uint32;
+    typedef uint16_t uint16;
+    typedef uint8_t uint8;
+    typedef int int32;
+    typedef short int16;
+    #ifdef __STDC__
+    typedef	signed char int8; /* NB: non-ANSI compilers may not grok */
+    #else
+    typedef	char int8;
+    #endif
 #else
     #ifdef __STDC__
     typedef	signed char int8; /* NB: non-ANSI compilers may not grok */
