diff -Naur PDFlib-Lite-7.0.2p8-orig/bind/pdflib/cpp/image.cpp PDFlib-Lite-7.0.2p8/bind/pdflib/cpp/image.cpp
--- PDFlib-Lite-7.0.2p8-orig/bind/pdflib/cpp/image.cpp	2008-02-28 03:43:00.000000000 -0600
+++ PDFlib-Lite-7.0.2p8/bind/pdflib/cpp/image.cpp	2008-03-05 20:53:52.000000000 -0600
@@ -4,6 +4,7 @@
 //
 
 #include <iostream>
+#include <cstdlib>
 
 #include "pdflib.hpp"
 
diff -Naur PDFlib-Lite-7.0.2p8-orig/bind/pdflib/cpp/invoice.cpp PDFlib-Lite-7.0.2p8/bind/pdflib/cpp/invoice.cpp
--- PDFlib-Lite-7.0.2p8-orig/bind/pdflib/cpp/invoice.cpp	2008-02-28 03:43:00.000000000 -0600
+++ PDFlib-Lite-7.0.2p8/bind/pdflib/cpp/invoice.cpp	2008-03-05 20:55:56.000000000 -0600
@@ -4,6 +4,7 @@
 //
 
 #include <iostream>
+#include <cstring>
 
 #include <time.h>
 
diff -Naur PDFlib-Lite-7.0.2p8-orig/bind/pdflib/cpp/starter_block.cpp PDFlib-Lite-7.0.2p8/bind/pdflib/cpp/starter_block.cpp
--- PDFlib-Lite-7.0.2p8-orig/bind/pdflib/cpp/starter_block.cpp	2008-02-28 03:43:00.000000000 -0600
+++ PDFlib-Lite-7.0.2p8/bind/pdflib/cpp/starter_block.cpp	2008-03-05 20:57:39.000000000 -0600
@@ -12,6 +12,7 @@
  */
 
 #include <iostream>
+#include <cstring>
 
 #include "pdflib.hpp"
 
diff -Naur PDFlib-Lite-7.0.2p8-orig/bind/pdflib/cpp/starter_basic.cpp PDFlib-Lite-7.0.2p8/bind/pdflib/cpp/starter_basic.cpp
--- PDFlib-Lite-7.0.2p8-orig/bind/pdflib/cpp/starter_basic.cpp	2008-02-28 03:43:00.000000000 -0600
+++ PDFlib-Lite-7.0.2p8/bind/pdflib/cpp/starter_basic.cpp	2008-03-05 20:58:56.000000000 -0600
@@ -7,6 +7,7 @@
 // required data: none
 
 #include <iostream>
+#include <cstring>
 
 #include "pdflib.hpp"
 
diff -Naur PDFlib-Lite-7.0.2p8-orig/bind/pdflib/cpp/starter_pvf.cpp PDFlib-Lite-7.0.2p8/bind/pdflib/cpp/starter_pvf.cpp
--- PDFlib-Lite-7.0.2p8-orig/bind/pdflib/cpp/starter_pvf.cpp	2008-02-28 03:43:00.000000000 -0600
+++ PDFlib-Lite-7.0.2p8/bind/pdflib/cpp/starter_pvf.cpp	2008-03-05 21:01:17.000000000 -0600
@@ -13,6 +13,7 @@
  * Required data: image file
  */
 #include <iostream>
+#include <cstdlib>
 
 #include "pdflib.hpp"
 
