--- configure.in.orig	2008-02-28 10:42:52.000000000 +0100
+++ configure.in	2008-02-28 15:22:12.000000000 +0100
@@ -867,7 +867,7 @@
 RUBYINCLUDE=nope
 RUBYBIN=nope
 
-AC_ARG_WITH(py,[  --with-ruby=path        x, or no],[RUBYBIN="$withval"], [RUBYBIN=nope])
+AC_ARG_WITH(ruby,[  --with-ruby=path        x, or no],[RUBYBIN="$withval"], [RUBYBIN=nope])
 
 if test "$RUBYBIN" = "no" ; then
     WITH_RUBY=no
