--- proj-4.6.1.orig/nad/Makefile.am	2008-08-22 04:58:20.000000000 +0200
+++ proj-4.6.1/nad/Makefile.am	2009-09-14 14:15:02.061120311 +0200
@@ -21,7 +21,16 @@
 		IGNF testIGNF proj_outIGNF.dist \
 		makefile.vc
 
-install-data-local:
+process-nad2bin:
+	@if [ -f conus.lla -a ! -f conus ] ; then \
+          for x in *.lla ; do \
+            DEST=`basename $$x .lla`; \
+            echo "../src/nad2bin$(EXEEXT) $$DEST < $$x"; \
+            ../src/nad2bin$(EXEEXT) $$DEST < $$x ; \
+          done; \
+    	fi 
+
+install-data-local: process-nad2bin
 	$(mkinstalldirs) $(DESTDIR)$(pkgdatadir)
 	@if [ -f conus.lla ] ; then \
 	  for x in *.lla ; do \
@@ -43,10 +52,13 @@
           fi; \
 	done
 
-check-local:
+check-local: process-nad2bin
 	$(TEST27) $(PROJEXE)
 	$(TEST83) $(PROJEXE)
-	$(TESTVARIOUS) $(CS2CSEXE)
+	@if [ -f conus ] ; then \
+          export PROJ_LIB=. ; \
+          $(TESTVARIOUS) $(CS2CSEXE) ; \
+        fi
 
 #	$(TESTNTV2) $(CS2CSEXE)
 #	$(TESTIGNF) $(CS2CSEXE)
--- proj-4.6.1.orig/nad/Makefile.in	2008-08-22 04:58:46.000000000 +0200
+++ proj-4.6.1/nad/Makefile.in	2009-09-14 14:18:19.282650172 +0200
@@ -374,7 +374,16 @@
 	uninstall-pkgdataDATA
 
 
-install-data-local:
+process-nad2bin:
+	@if [ -f conus.lla -a ! -f conus ] ; then \
+          for x in *.lla ; do \
+            DEST=`basename $$x .lla`; \
+            echo "../src/nad2bin$(EXEEXT) $$DEST < $$x"; \
+            ../src/nad2bin$(EXEEXT) $$DEST < $$x ; \
+          done; \
+        fi 
+
+install-data-local: process-nad2bin
 	$(mkinstalldirs) $(DESTDIR)$(pkgdatadir)
 	@if [ -f conus.lla ] ; then \
 	  for x in *.lla ; do \
@@ -396,10 +405,13 @@
           fi; \
 	done
 
-check-local:
+check-local: process-nad2bin
 	$(TEST27) $(PROJEXE)
 	$(TEST83) $(PROJEXE)
-	$(TESTVARIOUS) $(CS2CSEXE)
+	@if [ -f conus ] ; then \
+          export PROJ_LIB=. ; \
+          $(TESTVARIOUS) $(CS2CSEXE) ; \
+        fi
 
 #	$(TESTNTV2) $(CS2CSEXE)
 #	$(TESTIGNF) $(CS2CSEXE)
