--- configure.in.old	2008-01-26 11:43:17.423830001 +0100
+++ configure.in	2008-01-26 11:44:23.137155036 +0100
@@ -129,7 +129,12 @@
   fi
 fi
 
-GTK_DOC_CHECK([1.0])
+dnl Crazy hack for dropping the dep on gtk-doc
+AC_DEFUN(GTK_DOC_CHECK, [])
+
+dnl Crazy hack for dropping the dep on gtk-doc
+AM_CONDITIONAL(ENABLE_GTK_DOC, 0, [])
+AM_CONDITIONAL(GTK_DOC_USE_LIBTOOL, 0, [])
 
 AC_ARG_ENABLE(gtk,
 	      AC_HELP_STRING([--enable-gtk],
Index: Makefile.am
===================================================================
--- Makefile.am	(révision 2495)
+++ Makefile.am	(copie de travail)
@@ -1,7 +1,11 @@
 
-SUBDIRS = gconf backends po doc examples
+SUBDIRS = gconf backends po examples
 DIST_SUBDIRS=tests $(SUBDIRS)
 
+if ENABLE_GTK_DOC
+SUBDIRS += doc
+endif
+
 EXTRA_DIST =                \
 	TODO                \
 	gtk-doc.make        \
