diff -u -ru exo-0.3.4.orig/configure.in exo-0.3.4/configure.in
--- exo-0.3.4.orig/configure.in	2008-11-10 20:29:01.000000000 +0100
+++ exo-0.3.4/configure.in	2008-11-10 20:30:12.000000000 +0100
@@ -106,7 +106,7 @@
 dnl ***************************************
 AC_HEADER_STDC()
 AC_CHECK_HEADERS([assert.h errno.h fcntl.h fnmatch.h fstab.h libintl.h \
-                  locale.h math.h md5.h md5global.h mmintrin.h mntent.h \
+                  locale.h math.h mmintrin.h mntent.h \
                   paths.h regex.h signal.h stdarg.h string.h sys/mman.h \
                   sys/mnttab.h sys/mount.h sys/param.h sys/resource.h \
                   sys/statvfs.h sys/stat.h sys/time.h sys/types.h sys/ucred.h \
@@ -118,12 +118,6 @@
 AC_FUNC_MMAP()
 AC_CHECK_FUNCS([getfsstat getmntent getvfsstat regexec setfsent setmntent])
 
-dnl ******************************************
-dnl *** Check for Message Digest functions ***
-dnl ******************************************
-AC_SEARCH_LIBS([MD5Init], [md md5 c], \
-               [AC_DEFINE([HAVE_MD5INIT], [1], [Define if MD5Init present])])
-
 dnl ***************************************
 dnl *** Check for strftime() extensions ***
 dnl ***************************************
diff -u -ru exo-0.3.4.orig/exo/exo-md5.c exo-0.3.4/exo/exo-md5.c
--- exo-0.3.4.orig/exo/exo-md5.c	2008-11-10 20:29:01.000000000 +0100
+++ exo-0.3.4/exo/exo-md5.c	2008-11-10 20:29:40.000000000 +0100
@@ -43,8 +43,6 @@
 
 
 
-#if !defined(HAVE_MD5INIT) || !defined(HAVE_MD5_H)
-
 /*
  * This code implements the MD5 message-digest algorithm.
  * The algorithm is due to Ron Rivest. The original code was
@@ -274,19 +272,6 @@
   memcpy (digest, ctx->buf, 16);
 }
 
-#else
-
-/* RedHat AS, Fedora */
-#ifdef HAVE_MD5GLOBAL_H
-#include <md5global.h>
-#endif
-
-#include <md5.h>
-
-#endif  /* !defined (HAVE_MD5INIT) */
-
-
-
 static void
 get_md5 (const gchar *contents,
          guchar       digest[16])
