diff -ru libXt-1.0.6.orig/include/X11/Intrinsic.h libXt-1.0.6/include/X11/Intrinsic.h
--- libXt-1.0.6.orig/include/X11/Intrinsic.h	2009-09-25 09:20:37 +0200
+++ libXt-1.0.6/include/X11/Intrinsic.h	2009-09-25 09:21:07 +0200
@@ -96,7 +96,11 @@
 #define externalref globalref
 #define externaldef(psect) globaldef {"psect"} noshare
 #else
+#if defined(__PARITY__) && !defined(BUILDING_XT)
+#define externalref extern __declspec(dllimport)
+#else
 #define externalref extern
+#endif
 #define externaldef(psect)
 #endif /* VMS */
 
diff -ru libXt-1.0.6.orig/src/Makefile.am libXt-1.0.6/src/Makefile.am
--- libXt-1.0.6.orig/src/Makefile.am	2009-09-25 09:20:38 +0200
+++ libXt-1.0.6/src/Makefile.am	2009-09-25 09:21:45 +0200
@@ -7,7 +7,7 @@
 ERRORDB_DEFINES = -DERRORDB=\"$(libdir)/X11/XtErrorDB\"
 
 AM_CFLAGS = $(XT_CFLAGS) $(SRCH_DEFINES) $(ERRORDB_DEFINES) \
-	$(XTMALLOC_ZERO_CFLAGS)
+	$(XTMALLOC_ZERO_CFLAGS) -DBUILDING_XT
 
 INCLUDES = -I$(top_srcdir)/include/ -I$(top_srcdir)/include/X11 \
 	   -I$(top_builddir)/include -I$(top_builddir)/include/X11
@@ -97,7 +97,7 @@
 $(BUILT_HEADER): $(BUILT_SOURCE)
 
 $(BUILT_SOURCE):
-	$(top_builddir)/util/makestrs -i $(top_srcdir) $(STRINGSABIOPTIONS) < $(STRING_LIST) > StringDefs.c
+	$(top_builddir)/util/makestrs$(EXEEXT) -i $(top_srcdir) $(STRINGSABIOPTIONS) < $(STRING_LIST) > StringDefs.c
 	mkdir -p $(top_builddir)/src
 	mkdir -p $(top_builddir)/include/X11
 	cp StringDefs.h $(top_builddir)/include/X11
diff -ru libXt-1.0.6.orig/util/makestrs.c libXt-1.0.6/util/makestrs.c
--- libXt-1.0.6.orig/util/makestrs.c	2009-09-25 09:20:38 +0200
+++ libXt-1.0.6/util/makestrs.c	2009-09-25 09:21:06 +0200
@@ -215,7 +215,9 @@
 
     WriteHeaderProlog (f, phile);
 
-    (void) fprintf (f, "%s %sConst char %s[];\n", 
+	(void) fprintf (f, "\n#if defined(_WIN32) && !defined(BUILDING_XT)\n#define XT_STR_IMPORT __declspec(dllimport)\n#else\n#define XT_STR_IMPORT\n#endif\n\n");
+
+    (void) fprintf (f, "%s XT_STR_IMPORT %sConst char %s[];\n", 
 		    externrefstr, conststr ? conststr : fileprotstr, 
 		    phile->table->name);
 
