diff -ru -x '*.Po' -x '*.Plo' libXt-1.0.6.orig/src/Initialize.c libXt-1.0.6/src/Initialize.c
--- libXt-1.0.6.orig/src/Initialize.c	2009-09-25 09:41:16 +0200
+++ libXt-1.0.6/src/Initialize.c	2009-09-25 10:00:49 +0200
@@ -237,11 +237,25 @@
  * cygwin-xfree: http://www.cygwin.com/ml/cygwin-xfree/2003-10/msg00000.html
  */
 
+#ifndef __PARITY__
 asm (".data\n\
  .globl __XtInherit        \n\
  __XtInherit:      jmp *_y \n\
   _y: .long ___XtInherit   \n\
     .text                 \n");
+#else
+/* we're using the native win32 compiler here as a backend, so we
+ * cannot use gcc style inline assembly. we'll do the same thing
+ * again, but differently. */
+#pragma code_seg(push, ".data")
+void __XtInherit(void);
+__declspec(naked) void _XtInherit(void) {
+	__asm {
+		jmp [__XtInherit]
+	}
+}
+#pragma code_seg(pop)
+#endif
 
 #define _XtInherit __XtInherit
 #endif
