--- a/Configure
+++ b/Configure
@@ -4083,6 +4083,11 @@
 	LSOF_TSTLFLG="-lsocket -lnsl"
 	LSOF_VERS="100000"
 	;;
+      5.11*)
+	LSOF_TSTBIGF=" "
+	LSOF_TSTLFLG="-lsocket -lnsl"
+	LSOF_VERS="110000"
+	;;
       *)
 	echo Unknown Solaris version: $LSOF_VSTR
 	rm -f $LSOF_HLP
@@ -4164,7 +4164,7 @@
       ;;
     20500|20501)
       ;;
-    20600|70000|80000|90000|100000)
+    20600|70000|80000|90000|100000|110000)
       if test "X$SOLARIS_26PR_GWINDOWS" = "X"	# {
       then
 	rm -f ${LSOF_TMPC}.*
@@ -4267,8 +4267,8 @@
 	    rm -f $LSOF_HLP
 	    exit 1
           fi	# }
-          LSOF_TMP4=`$LSOF_TMP3 $LSOF_TMP2`
-	  if test $? -eq 0	# {
+          LSOF_TMP4=''
+	  if true	# {
 	  then
 
 	  # ZFS support has been requested.
