diff -ru xlsfonts-1.0.2.orig/dsimple.c xlsfonts-1.0.2/dsimple.c
--- xlsfonts-1.0.2.orig/dsimple.c	2009-01-19 08:31:12 +0100
+++ xlsfonts-1.0.2/dsimple.c	2009-01-19 08:32:39 +0100
@@ -28,6 +28,12 @@
 */
 /* $XFree86: xc/programs/xlsfonts/dsimple.c,v 3.6 2001/12/14 20:02:09 dawes Exp $ */
 
+#ifdef _WIN32
+# define Beep wBeep
+# include <X11/Xwindows.h>
+# undef Beep
+#endif
+
 #include <X11/Xos.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
diff -ru xlsfonts-1.0.2.orig/xlsfonts.c xlsfonts-1.0.2/xlsfonts.c
--- xlsfonts-1.0.2.orig/xlsfonts.c	2009-01-19 08:31:12 +0100
+++ xlsfonts-1.0.2/xlsfonts.c	2009-01-19 08:33:31 +0100
@@ -26,6 +26,14 @@
  * */
 /* $XFree86: xc/programs/xlsfonts/xlsfonts.c,v 1.9 2003/09/08 14:25:33 eich Exp $ */
 
+#ifdef _WIN32
+# define Beep wBeep
+# define IgnoreError wIgnoreError
+# include <X11/Xwindows.h>
+# undef Beep
+# undef IgnoreError
+#endif
+
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <X11/Xos.h>
