diff -ru bash-4.0.orig/shell.c bash-4.0/shell.c
--- bash-4.0.orig/shell.c	2009-10-06 09:39:27 +0200
+++ bash-4.0/shell.c	2009-10-06 10:17:47 +0200
@@ -357,6 +357,36 @@
 #ifdef __OPENNT
   char **env;
 
+#ifdef __INTERIX
+  /* on x64 interix, microsoft introduced freaky environment names
+   * like COMMONPROGRAMFILES(X86) - (yes, _with_ those braces) ...
+   * simply handle them, by transforming all '(' and ')' in the existing
+   * environment to '_'s ... */
+#define _ENV_RPL(var, end, srch, rpl)   \
+  while(var) {                          \
+    char * _p_fnd = strchr(var, srch);  \
+                                        \
+    if(!_p_fnd || _p_fnd >= end)        \
+      break;                            \
+                                        \
+    *_p_fnd = rpl;                      \
+  }
+
+  {
+    char ** _pe = environ;
+    
+    while(_pe && *_pe) {
+      char * _p_eq = strchr(*_pe, '=');
+
+      if(_p_eq) {
+		_ENV_RPL(*_pe, _p_eq, '(', '_');
+		_ENV_RPL(*_pe, _p_eq, ')', '_');
+      }
+	  ++_pe;
+    }
+  }
+#endif
+
   env = environ;
 #endif /* __OPENNT */
 
