diff -ru -x '*.Po' -x '*.Plo' bash-4.0.orig/findcmd.c bash-4.0/findcmd.c
--- bash-4.0.orig/findcmd.c	2009-08-25 10:57:38 +0200
+++ bash-4.0/findcmd.c	2009-08-25 11:12:37 +0200
@@ -111,7 +111,7 @@
 
   /* Root only requires execute permission for any of owner, group or
      others to be able to exec a file, and can read any file. */
-  if (current_user.euid == (uid_t)0)
+  if (current_user.euid == (uid_t)ROOT_UID)
     {
       r |= FS_READABLE;
       if (finfo.st_mode & S_IXUGO)
diff -ru -x '*.Po' -x '*.Plo' bash-4.0.orig/lib/readline/complete.c bash-4.0/lib/readline/complete.c
--- bash-4.0.orig/lib/readline/complete.c	2009-08-25 10:57:39 +0200
+++ bash-4.0/lib/readline/complete.c	2009-08-25 11:35:09 +0200
@@ -52,6 +52,10 @@
 #include <pwd.h>
 #endif
 
+/* can't include externs.h here - so directly define this one. this
+ * is required to fix interix access() which behaves totally stupid */
+extern int sh_eaccess(char*,int);
+
 #include "posixdir.h"
 #include "posixstat.h"
 
@@ -554,7 +558,7 @@
 #endif
   else if (S_ISREG (finfo.st_mode))
     {
-      if (access (filename, X_OK) == 0)
+      if (sh_eaccess (filename, X_OK) == 0)
 	character = '*';
     }
   return (character);
diff -ru -x '*.Po' -x '*.Plo' bash-4.0.orig/lib/sh/eaccess.c bash-4.0/lib/sh/eaccess.c
--- bash-4.0.orig/lib/sh/eaccess.c	2009-08-25 10:57:40 +0200
+++ bash-4.0/lib/sh/eaccess.c	2009-08-25 16:34:38 +0200
@@ -141,7 +141,7 @@
   if (sh_stat (path, &st) < 0)
     return (-1);
 
-  if (current_user.euid == 0)
+  if (current_user.euid == ROOT_UID)
     {
       /* Root can read or write any file. */
       if ((mode & X_OK) == 0)
@@ -203,7 +203,7 @@
 
 #if defined (HAVE_EACCESS)		/* FreeBSD */
   return (eaccess (path, mode));
-#elif defined (EFF_ONLY_OK)		/* SVR4(?), SVR4.2 */
+#elif defined (EFF_ONLY_OK) && !defined(__INTERIX)	/* SVR4(?), SVR4.2 */
   return access (path, mode|EFF_ONLY_OK);
 #else
   if (mode == F_OK)
@@ -214,8 +214,13 @@
     return (sh_euidaccess (path, mode));
 #  endif
 
+/* on interix, use stataccess always, since access on interix behaves bad:
+ * whenever a process has appropriate priviledges (speaking in windows terms),
+ * success is reported, even if the bits are not set. */
+#ifndef __INTERIX
   if (current_user.uid == current_user.euid && current_user.gid == current_user.egid)
     return (access (path, mode));  
+#endif
 
   return (sh_stataccess (path, mode));
 #endif
diff -ru -x '*.Po' -x '*.Plo' bash-4.0.orig/parse.y bash-4.0/parse.y
--- bash-4.0.orig/parse.y	2009-08-25 10:57:41 +0200
+++ bash-4.0/parse.y	2009-08-25 11:27:04 +0200
@@ -4990,9 +4990,9 @@
 
 	    case '$':
 	      t = temp = (char *)xmalloc (3);
-	      if ((promptvars || posixly_correct) && (current_user.euid != 0))
+	      if ((promptvars || posixly_correct) && (current_user.euid != ROOT_UID))
 		*t++ = '\\';
-	      *t++ = current_user.euid == 0 ? '#' : '$';
+	      *t++ = current_user.euid == ROOT_UID ? '#' : '$';
 	      *t = '\0';
 	      goto add_string;
 
diff -ru -x '*.Po' -x '*.Plo' bash-4.0.orig/shell.h bash-4.0/shell.h
--- bash-4.0.orig/shell.h	2009-08-25 10:57:42 +0200
+++ bash-4.0/shell.h	2009-08-25 11:11:39 +0200
@@ -41,6 +41,14 @@
 #include "pathnames.h"
 #include "externs.h"
 
+#ifndef __INTERIX
+# define ROOT_UID 0
+# define ROOT_GID 0
+#else
+# define ROOT_UID 197108
+# define ROOT_GID 131616
+#endif
+
 extern int EOF_Reached;
 
 #define NO_PIPE -1
