--- Makefile.in	2002-05-31 19:44:23.000000000 +0200
+++ Makefile.in-parallelbuild	2004-02-09 21:25:28.000000000 +0100
@@ -604,7 +604,7 @@
 
 config.h:	stamp-h 
 
-stamp-h:	config.status $(srcdir)/config.h.in $(srcdir)/config-top.h $(srcdir)/config-bot.h
+stamp-h:	version.h config.status $(srcdir)/config.h.in $(srcdir)/config-top.h $(srcdir)/config-bot.h
 	CONFIG_FILES= CONFIG_HEADERS=config.h $(SHELL) ./config.status
 
 config.status:	$(srcdir)/configure
