diff -ur highlight-2.6.9.orig/src/makefile highlight-2.6.9/src/makefile
--- highlight-2.6.9.orig/src/makefile	2008-03-26 23:19:37.000000000 +0200
+++ highlight-2.6.9/src/makefile	2008-05-04 23:26:13.000000000 +0300
@@ -49,11 +49,11 @@
 	ifeq ($(WX29),true)
 		WX_FLAGS=`wx-config --cxxflags` -I.
 		OBJECTFILES:=${OBJECTFILES} gui_wx29.o gui_failure_dlg.o gui_showfile_dlg.o
-		CXX_LINK=${CXX} ${LDFLAGS} `wx-config --libs core,base,html,adv` -o highlight-gui ${OBJECTFILES}
+		CXX_LINK=${CXX} ${LDFLAGS} ${OBJECTFILES} `wx-config --libs core,base,html,adv` -o highlight-gui
 	else
 		WX_FLAGS=`wx-config --cxxflags` -I.
 		OBJECTFILES:=${OBJECTFILES} gui.o gui_failure_dlg.o gui_showfile_dlg.o
-		CXX_LINK=${CXX} ${LDFLAGS} `wx-config --libs core,base,html,adv` -o highlight-gui ${OBJECTFILES}
+		CXX_LINK=${CXX} ${LDFLAGS} ${OBJECTFILES} `wx-config --libs core,base,html,adv` -o highlight-gui
 	endif
 else
 	OBJECTFILES:=${OBJECTFILES} main.o
