--- makefile.orig	2009-06-23 17:08:18.017847256 +0300
+++ makefile	2009-06-23 17:10:32.682853270 +0300
@@ -3,7 +3,6 @@
 #--------------------------------
 OBJ=.
 SRC=.
-CFLAGS= -O3 -Wall
 
 all: jhead
 
@@ -11,10 +10,10 @@
 	$(OBJ)/exif.o $(OBJ)/iptc.o $(OBJ)/gpsinfo.o $(OBJ)/makernote.o 
 
 $(OBJ)/%.o:$(SRC)/%.c
-	${CC} $(CFLAGS) -c $< -o $@
+	${CC} ${CFLAGS} -c $< -o $@
 
 jhead: $(objs) jhead.h
-	${CC} -o jhead $(objs) -lm
+	${CC} ${LDFLAGS} -o jhead $(objs) -lm
 
 clean:
 	rm -f $(objs) jhead
