--- pango-1.1.13/pango/pangox.aliases.xfonts	Mon Jun 24 20:19:34 2002
+++ pango-1.1.13/pango/pangox.aliases	Mon Jun 24 20:26:34 2002
@@ -4,6 +4,7 @@
 
 sans normal normal normal normal \
   "-*-helvetica-medium-r-normal--*-*-*-*-*-*-*-*,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-gulim-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
@@ -22,6 +23,7 @@
 
 sans italic normal normal normal \
   "-*-helvetica-medium-o-normal--*-*-*-*-*-*-*-*,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-gulim-medium-o-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
@@ -40,6 +42,7 @@
 
 sans normal normal bold normal \
   "-*-helvetica-bold-r-normal--*-*-*-*-*-*-*-*,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-gulim-bold-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
@@ -58,6 +61,7 @@
 
 sans italic normal bold normal \
   "-*-helvetica-bold-o-normal--*-*-*-*-*-*-*-*,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-gulim-bold-o-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
@@ -78,6 +82,7 @@
 serif normal normal normal normal \
   "-*-times-medium-r-normal--*-*-*-*-*-*-*-*,\
    -*-batang-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
    -kaist-iyagi-bold-r-normal--*-*-*-*-*-*-johab-1,\
@@ -96,6 +101,7 @@
 serif italic normal normal normal \
   "-*-times-medium-i-normal--*-*-*-*-*-*-*-*,\
    -*-batang-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
    -kaist-iyagi-bold-r-normal--*-*-*-*-*-*-johab-1,\
@@ -114,6 +120,7 @@
 serif normal normal bold normal \
   "-*-times-bold-r-normal--*-*-*-*-*-*-*-*,\
    -*-batang-bold-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
    -kaist-iyagi-bold-r-normal--*-*-*-*-*-*-johab-1,\
@@ -132,6 +139,7 @@
 serif italic normal bold normal \
   "-*-times-bold-i-normal--*-*-*-*-*-*-*-*,\
    -*-batang-bold-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
    -kaist-iyagi-bold-r-normal--*-*-*-*-*-*-johab-1,\
@@ -150,6 +158,7 @@
 monospace normal normal normal normal \
   "-*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
    -*-dotum-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
    -kaist-iyagi-bold-r-normal--*-*-*-*-*-*-johab-1,\
@@ -168,6 +177,7 @@
 monospace italic normal normal normal \
   "-*-fixed-medium-i-normal--*-*-*-*-*-*-*-*,\
    -*-dotum-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
    -kaist-iyagi-bold-r-normal--*-*-*-*-*-*-johab-1,\
@@ -186,6 +196,7 @@
 monospace normal normal bold normal \
   "-*-fixed-bold-r-normal--*-*-*-*-*-*-*-*,\
    -*-dotum-bold-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
    -kaist-iyagi-bold-r-normal--*-*-*-*-*-*-johab-1,\
@@ -204,6 +215,7 @@
 monospace italic normal bold normal \
   "-*-fixed-bold-i-normal--*-*-*-*-*-*-*-*,\
    -*-dotum-bold-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
+   -*-gulimbdf-medium-r-normal--*-*-*-*-*-*-ksc5601.1987-0,\
    -*-clearlyu-medium-r-normal--*-*-*-*-*-*-iso10646-1,\
    -*-fixed-medium-r-normal--*-*-*-*-*-*-*-*,\
    -kaist-iyagi-bold-r-normal--*-*-*-*-*-*-johab-1,\
