Only in pango-1.26.2: darwin8.patch
diff -ur pango-1.26.2.orig/modules/basic/basic-atsui.c pango-1.26.2/modules/basic/basic-atsui.c
--- pango-1.26.2.orig/modules/basic/basic-atsui.c	2009-11-26 01:44:17 +0100
+++ pango-1.26.2/modules/basic/basic-atsui.c	2009-12-29 13:10:48 +0100
@@ -86,10 +86,10 @@
   const char *p;
   PangoATSUIFont *afont = PANGO_ATSUI_FONT (font);
   ATSUStyle style;
-  CGFontRef fontID;
+  ATSUFontID fontID;
   ATSUAttributeTag styleTags[] = { kATSUFontTag };
   ATSUAttributeValuePtr styleValues[] = { &fontID };
-  ByteCount styleSizes[] = { sizeof (CGFontRef) };
+  ByteCount styleSizes[] = { sizeof (ATSUFontID) };
 
   utf16 = g_utf8_to_utf16 (text, length, NULL, &n16, NULL);
 
@@ -97,7 +97,7 @@
   err = ATSUSetTextPointerLocation (text_layout, utf16, 0, n16, n16);
 
   err = ATSUCreateStyle(&style);
-  fontID = pango_atsui_font_get_cgfont (afont);
+  fontID = pango_atsui_font_get_atsu_font_id (afont);
 
   err = ATSUSetAttributes(style,
 			  sizeof(styleTags) / sizeof(styleTags[0]),
Only in pango-1.26.2/modules/basic: basic-atsui.c.rej
diff -ur pango-1.26.2.orig/pango/pangoatsui-private.h pango-1.26.2/pango/pangoatsui-private.h
--- pango-1.26.2.orig/pango/pangoatsui-private.h	2009-11-26 01:44:17 +0100
+++ pango-1.26.2/pango/pangoatsui-private.h	2009-12-29 13:10:48 +0100
@@ -93,8 +93,8 @@
 gpointer              _pango_atsui_font_get_context_key      (PangoATSUIFont    *afont);
 void                  _pango_atsui_font_set_context_key      (PangoATSUIFont    *afont,
                                                               gpointer           context_key);
-void                  _pango_atsui_font_set_cgfont           (PangoATSUIFont    *font,
-                                                              CGFontRef         font_id);
+void                  _pango_atsui_font_set_atsu_font_id     (PangoATSUIFont    *font,
+                                                              ATSUFontID         font_id);
 
 G_END_DECLS
 
Only in pango-1.26.2/pango: pangoatsui-private.h.rej
diff -ur pango-1.26.2.orig/pango/pangoatsui.c pango-1.26.2/pango/pangoatsui.c
--- pango-1.26.2.orig/pango/pangoatsui.c	2009-11-26 01:44:17 +0100
+++ pango-1.26.2/pango/pangoatsui.c	2009-12-29 13:10:48 +0100
@@ -32,7 +32,7 @@
   PangoFontDescription *desc;
   gpointer context_key;
 
-  CGFontRef font_id;
+  ATSUFontID font_id;
 
   PangoFontMap *fontmap;
 };
@@ -176,8 +176,8 @@
 }
 
 void
-_pango_atsui_font_set_cgfont (PangoATSUIFont *font,
-                                    CGFontRef      font_id)
+_pango_atsui_font_set_atsu_font_id (PangoATSUIFont *font,
+                                    ATSUFontID      font_id)
 {
   PangoATSUIFontPrivate *priv = font->priv;
 
@@ -185,17 +185,17 @@
 }
 
 /**
- * pango_atsui_font_get_cgfont:
+ * pango_atsui_font_get_atsu_font_id:
  * @font: A #PangoATSUIFont
  *
- * Returns the CGFontRef of a font.
+ * Returns the ATSUFontID of a font.
  *
- * Return value: the CGFontRef associated to @font.
+ * Return value: the ATSUFontID associated to @font.
  *
  * Since: 1.18
  */
-CGFontRef
-pango_atsui_font_get_cgfont (PangoATSUIFont *font)
+ATSUFontID
+pango_atsui_font_get_atsu_font_id (PangoATSUIFont *font)
 {
   PangoATSUIFontPrivate *priv = font->priv;
 
Only in pango-1.26.2/pango: pangoatsui.c.rej
diff -ur pango-1.26.2.orig/pango/pangoatsui.h pango-1.26.2/pango/pangoatsui.h
--- pango-1.26.2.orig/pango/pangoatsui.h	2009-11-26 01:44:17 +0100
+++ pango-1.26.2/pango/pangoatsui.h	2009-12-29 13:10:48 +0100
@@ -68,7 +68,7 @@
 
 #endif /* PANGO_ENABLE_BACKEND */
 
-CGFontRef pango_atsui_font_get_cgfont (PangoATSUIFont *font);
+ATSUFontID pango_atsui_font_get_atsu_font_id (PangoATSUIFont *font);
 
 #endif /* PANGO_ENABLE_ENGINE || PANGO_ENABLE_BACKEND */
 
Only in pango-1.26.2/pango: pangoatsui.h.rej
diff -ur pango-1.26.2.orig/pango/pangocairo-atsuifont.c pango-1.26.2/pango/pangocairo-atsuifont.c
--- pango-1.26.2.orig/pango/pangocairo-atsuifont.c	2009-12-14 22:15:04 +0100
+++ pango-1.26.2/pango/pangocairo-atsuifont.c	2009-12-29 13:14:20 +0100
@@ -83,10 +83,10 @@
 pango_cairo_atsui_font_create_font_face (PangoCairoFont *font)
 {
   PangoATSUIFont *afont = (PangoATSUIFont *) (font);
-  CGFontRef font_id;
+  ATSUFontID font_id;
 
-  font_id = pango_atsui_font_get_cgfont (afont);
-  return cairo_quartz_font_face_create_for_cgfont (font_id);
+  font_id = pango_atsui_font_get_atsu_font_id (afont);
+  return cairo_quartz_font_face_create_for_atsu_font_id (font_id);
 }
 
 static int
@@ -119,8 +119,8 @@
 {
   PangoCairoATSUIFont *cafont = (PangoCairoATSUIFont *) font;
   PangoATSUIFont *afont = (PangoATSUIFont *) font;
-  CGFontRef cg_font;
-  CTFontRef ct_font;
+  ATSFontRef ats_font;
+  ATSFontMetrics ats_metrics;
   PangoFontMetrics *metrics;
   PangoFontDescription *font_desc;
   PangoLayout *layout;
@@ -128,19 +128,20 @@
   PangoLanguage *language = pango_context_get_language (context);
   const char *sample_str = pango_language_get_sample_string (language);
 
-  cg_font = pango_atsui_font_get_cgfont (afont);
-  ct_font = CTFontCreateWithGraphicsFont(cg_font, cafont->size, NULL, NULL);
+  ats_font = FMGetATSFontRefFromFont (pango_atsui_font_get_atsu_font_id (afont));
+
+  ATSFontGetHorizontalMetrics (ats_font, kATSOptionFlagsDefault, &ats_metrics);
 
   metrics = pango_font_metrics_new ();
 
-  metrics->ascent = CTFontGetAscent(ct_font) * PANGO_SCALE;
-  metrics->descent = CTFontGetDescent(ct_font) * PANGO_SCALE;
+  metrics->ascent = ats_metrics.ascent * cafont->size * PANGO_SCALE;
+  metrics->descent = ats_metrics.descent * cafont->size * PANGO_SCALE;
 
-  metrics->underline_position = CTFontGetUnderlinePosition(ct_font) * PANGO_SCALE;
-  metrics->underline_thickness = CTFontGetUnderlineThickness(ct_font) * PANGO_SCALE;
+  metrics->underline_position = ats_metrics.underlinePosition * cafont->size * PANGO_SCALE;
+  metrics->underline_thickness = ats_metrics.underlineThickness * cafont->size * PANGO_SCALE;
 
   metrics->strikethrough_position = metrics->ascent / 3;
-  metrics->strikethrough_thickness = metrics->underline_thickness * PANGO_SCALE;
+  metrics->strikethrough_thickness = ats_metrics.underlineThickness * cafont->size * PANGO_SCALE;
 
   layout = pango_layout_new (context);
   font_desc = pango_font_describe_with_absolute_size ((PangoFont *) font);
@@ -155,8 +156,6 @@
 
   pango_font_description_free (font_desc);
   g_object_unref (layout);
-  
-  
 
   return metrics;
 }
@@ -213,7 +212,7 @@
   PangoATSUIFont *afont;
   CFStringRef cfstr;
   ATSFontRef font_ref;
-  CGFontRef font_id;
+  ATSUFontID font_id;
   double size;
   double dpi;
   double m;
@@ -258,7 +257,7 @@
   if (font_ref == kATSFontRefUnspecified)
     return NULL;
 
-  font_id = CGFontCreateWithPlatformFont (&font_ref);
+  font_id = FMGetFontFromATSFontRef (font_ref);
   if (!font_id)
     return NULL;
 
@@ -269,7 +268,7 @@
   _pango_atsui_font_set_face (afont, face);
 
   size = (double) pango_font_description_get_size (desc) / PANGO_SCALE;
-  _pango_atsui_font_set_cgfont (afont, font_id);
+  _pango_atsui_font_set_atsu_font_id (afont, font_id);
 
   if (context)
     {
Only in pango-1.26.2/pango: pangocairo-atsuifont.c.orig
Only in pango-1.26.2/pango: pangocairo-atsuifont.c.rej
