diff -Naur puppet-0.24.2.orig/lib/puppet/provider/service/gentoo.rb puppet-0.24.2/lib/puppet/provider/service/gentoo.rb
--- puppet-0.24.2.orig/lib/puppet/provider/service/gentoo.rb	2008-03-04 01:32:37.000000000 +0900
+++ puppet-0.24.2/lib/puppet/provider/service/gentoo.rb	2008-03-06 00:53:20.000000000 +0900
@@ -33,7 +33,7 @@
         return :false unless line
 
         # If it's enabled then it will print output showing service | runlevel
-        if output =~ /#{@resource[:name]}\s*|\s*default/
+        if output =~ /#{@resource[:name]}\s*\|\s*default/
             return :true
         else
             return :false
