Gentoo bug #389347
--- configure.ac.orig	2011-08-21 13:35:05.000000000 +0000
+++ configure.ac	2011-11-06 19:46:07.000000000 +0000
@@ -893,11 +893,12 @@
     CPPFLAGS="$CPPFLAGS -I$with_gssapi/include"
   fi
   AC_CHECK_HEADERS(gss.h gssapi.h gssapi/gssapi.h gssapi/gssapi_generic.h)
-  if test "$ac_cv_header_gssapi_h" = "yes"; then
-    AC_EGREP_HEADER(GSS_C_NT_HOSTBASED_SERVICE, gssapi.h, AC_DEFINE(HAVE_GSS_C_NT_HOSTBASED_SERVICE,1,Define if you have MIT kerberos))
-  else
-    AC_EGREP_HEADER(GSS_C_NT_HOSTBASED_SERVICE, gssapi/gssapi.h, AC_DEFINE(HAVE_GSS_C_NT_HOSTBASED_SERVICE))
-  fi
+  AC_EGREP_CPP(hostbased_service_gss_nt_yes, gssapi.h,
+  [#include <gssapi.h>
+   #ifdef GSS_C_NT_HOSTBASED_SERVICE
+     hostbased_service_gss_nt_yes
+   #endif],
+   AC_DEFINE(HAVE_GSS_C_NT_HOSTBASED_SERVICE,1,Define if your GSSAPI implemantation defines GSS_C_NT_HOSTBASED_SERVICE))
 fi])
 
 dnl ,------------------------------------------------------------------
