--- fetchmail~	2007-10-03 17:13:22 +0200
+++ fetchmail	2007-10-03 17:15:12 +0200
@@ -1,4 +1,4 @@
-#!/sbin/runscript
+#!@GENTOO_PORTAGE_EPREFIX@/sbin/runscript
 
 depend() {
 	need net
@@ -6,7 +6,7 @@
 }
 
 checkconfig() {
-        if [ ! -f /etc/fetchmailrc ]; then
+        if [[ ! -f "@GENTOO_PORTAGE_EPREFIX@"/etc/fetchmailrc ]]; then
                 eerror "Configuration file /etc/fetchmailrc not found"
                 return 1
         fi
@@ -16,14 +16,14 @@
         checkconfig || return 1
 
         ebegin "Starting fetchmail"
-        start-stop-daemon --start --quiet --exec /usr/bin/fetchmail \
-                -- -d ${polling_period} -f /etc/fetchmailrc
+        start-stop-daemon --start --quiet --exec "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/fetchmail \
+                -- -d ${polling_period} -f "@GENTOO_PORTAGE_EPREFIX@"/etc/fetchmailrc
         eend ${?}
 }
 
 stop() {
         ebegin "Stopping fetchmail"
-        start-stop-daemon --stop --quiet --pidfile /var/run/fetchmail.pid
+        start-stop-daemon --stop --quiet --pidfile "@GENTOO_PORTAGE_EPREFIX@"/var/run/fetchmail.pid
         eend ${?}
 }
 
