diff -ru multitail-5.2.2.orig/Makefile multitail-5.2.2/Makefile
--- multitail-5.2.2.orig/Makefile	2008-10-31 15:11:52 +0100
+++ multitail-5.2.2/Makefile	2008-10-31 15:17:32 +0100
@@ -4,7 +4,7 @@
 CONFIG_FILE=$(DESTDIR)/etc/multitail.conf
 
 DEBUG=-g # -D_DEBUG # -pg #  -D_DEBUG  #-pg -W -pedantic # -pg #-fprofile-arcs
-LDFLAGS+=-lpanel -lncurses -lutil -lm $(DEBUG) -rdynamic
+LDFLAGS+=-lpanel -lncurses -lm $(DEBUG)
 # LDFLAGS+=-lc_p -lpanel_g -lncurses_g -lutil -lm -pg -g -rdynamic
 CFLAGS+=-funsigned-char -D`uname` -O2 -Wall -DVERSION=\"$(VERSION)\" $(DEBUG) -DCONFIG_FILE=\"$(CONFIG_FILE)\"
 
diff -ru multitail-5.2.2.orig/mt.h multitail-5.2.2/mt.h
--- multitail-5.2.2.orig/mt.h	2008-10-31 15:11:52 +0100
+++ multitail-5.2.2/mt.h	2008-10-31 15:18:12 +0100
@@ -72,6 +72,12 @@
       #define atoll(a) strtoll((a), (char **)NULL, 10)
 #endif
 
+#ifdef __INTERIX
+typedef unsigned long off64_t;
+# define stat64 stat
+# define open64 open
+#endif
+
 /* Tru64 workaround */
 #if defined(OSF1)
 	#undef getmaxyx
diff -ru multitail-5.2.2.orig/my_pty.c multitail-5.2.2/my_pty.c
--- multitail-5.2.2.orig/my_pty.c	2008-10-31 15:11:52 +0100
+++ multitail-5.2.2/my_pty.c	2008-10-31 15:17:07 +0100
@@ -29,7 +29,7 @@
 #ifdef __OpenBSD__
 #include <util.h>
 #endif
-#if defined(linux) || defined(__CYGWIN__) || defined(__GNU__) || defined(__GLIBC__)
+#if defined(linux) || defined(__CYGWIN__) || defined(__GNU__) || defined(__GLIBC__) || defined(__INTERIX)
 #include <pty.h>
 #endif
 #ifdef __FreeBSD__
@@ -53,6 +53,7 @@
 #include <sys/strtio.h>
 #include <sys/eucioctl.h>
 #endif
+
 #include <sys/socket.h>
 
 #include "error.h"
@@ -78,7 +79,7 @@
 
 int get_pty_and_fork(int *fd_master, int *fd_slave)
 {
-#if defined(__FreeBSD__) || defined(linux) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__APPLE__) || defined(__CYGWIN__) || defined(__GNU__) || defined(__GLIBC__)
+#if defined(__FreeBSD__) || defined(linux) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__APPLE__) || defined(__CYGWIN__) || defined(__GNU__) || defined(__GLIBC__) || defined(__INTERIX)
 
 	if (openpty(fd_master, fd_slave, NULL, NULL, NULL) == -1)
 	{
