diff --exclude-from=/home/dang/bin/scripts/diffrc -up -ruN esound-0.2.37.orig/players.c esound-0.2.37/players.c
--- esound-0.2.37.orig/players.c	2007-01-01 18:56:06.000000000 -0500
+++ esound-0.2.37/players.c	2007-03-19 14:47:06.000000000 -0400
@@ -303,14 +303,13 @@ int read_player( esd_player_t *player )
 			   &rd_fds, NULL, NULL, &timeout ) ;
 	if ( can_read > 0 )
 	{
-	    int bytes_read;
 	    player->actual_length = 0;
 	    do {
-		bytes_read = ESD_READ_BIN( player->source_id,
+		ESD_READ_BIN( player->source_id,
 			      player->data_buffer + player->actual_length, 
 			      player->buffer_length - player->actual_length,
 			      actual, "str rd" );
-		if (bytes_read < player->buffer_length - player->actual_length)
+		if (actual < player->buffer_length - player->actual_length)
 			break;
 		
 		/* check for end of stream */
