diff -ru -x '*.Po' -x '*.Plo' esound-0.2.38.orig/clients.c esound-0.2.38/clients.c
--- esound-0.2.38.orig/clients.c	2008-04-11 11:21:40 +0200
+++ esound-0.2.38/clients.c	2008-04-11 11:22:16 +0200
@@ -132,6 +132,9 @@
     return;
 }
 
+#ifdef __INTERIX
+typedef int socklen_t;
+#endif
 
 /*******************************************************************/
 /* checks for new connections at listener - zero when done */
