diff -Naur astyle.orig/build/Makefile astyle/build/Makefile
--- astyle.orig/build/Makefile	2007-06-14 17:28:24.000000000 +0200
+++ astyle/build/Makefile	2007-10-06 17:43:53.000000000 +0200
@@ -83,7 +83,7 @@
 release:  astyle
 astyle:  $(OBJ)
 	@ mkdir -p $(bindir)
-	$(CXX) -s -o $(bindir)/$@ $^
+	$(CXX) -o $(bindir)/$@ $^
 	@ echo
 
 debug:  astyled
@@ -95,7 +95,7 @@
 shared:  libastyle.so
 libastyle.so:  $(OBJs)
 	@ mkdir -p $(bindir)
-	$(CXX) -shared -s -o $(bindir)/$@ $^
+	$(CXX) -shared -o $(bindir)/$@ $^
 	@ echo
 
 shareddebug:  libastyled.so
