--- php-toolkit-1.0.2/php-select
+++ php-toolkit-1.0.2/php-select
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 #
 # /usr/sbin/php-select
 #		Tool for managing PHP symlinks on Gentoo Linux
@@ -40,8 +40,8 @@
 G_COPYRIGHT[1]="Released under v2 of the GNU General Public License"
 G_EXITSTATUS=0
 G_LIBDIR_NAME="@@GENTOO_LIBDIR@@"
-G_LIBDIR="/usr/${G_LIBDIR_NAME}"
-G_MODULE_PATH="/usr/share/php-select"
+G_LIBDIR="@GENTOO_PORTAGE_EPREFIX@/usr/${G_LIBDIR_NAME}"
+G_MODULE_PATH="@GENTOO_PORTAGE_EPREFIX@/usr/share/php-select"
 G_NEED_COMPONENT=0
 G_NEED_PHPVERSION=0
 G_PN="$0"
@@ -105,8 +105,8 @@
 	echo
 	echo "    apache1   - Apache webserver v1.x"
 	echo "    apache2   - Apache webserver v2.x"
-	echo "    php       - /usr/bin/php (the CLI SAPI)"
-	echo "    php-cgi   - /usr/bin/php-cgi (the CGI SAPI)"
+	echo "    php       - @GENTOO_PORTAGE_EPREFIX@/usr/bin/php (the CLI SAPI)"
+	echo "    php-cgi   - @GENTOO_PORTAGE_EPREFIX@/usr/bin/php-cgi (the CGI SAPI)"
 	echo "    php-devel - the scripts used to build PECL extensions"
 	echo
 	echo "and where <php-version> is one of:"
--- php-toolkit-1.0.2/php-select-modules/apache1.sh
+++ php-toolkit-1.0.2/php-select-modules/apache1.sh
@@ -12,9 +12,9 @@
 #
 # ========================================================================
 
-G_APACHE_CONF=/etc/conf.d/apache
-G_APACHE_INIT=/etc/init.d/apache
+G_APACHE_CONF="@GENTOO_PORTAGE_EPREFIX@/etc/conf.d/apache"
+G_APACHE_INIT="@GENTOO_PORTAGE_EPREFIX@/etc/init.d/apache"
 G_OPTS_VAR="APACHE_OPTS"
-G_APACHE_MOD_DIR="/usr/lib/apache/modules"
+G_APACHE_MOD_DIR="@GENTOO_PORTAGE_EPREFIX@/usr/lib/apache/modules"
 
 . $G_MODULE_PATH/libapache.sh
--- php-toolkit-1.0.2/php-select-modules/apache2.sh
+++ php-toolkit-1.0.2/php-select-modules/apache2.sh
@@ -12,9 +12,9 @@
 #
 # ========================================================================
 
-G_APACHE_CONF=/etc/conf.d/apache2
-G_APACHE_INIT=/etc/init.d/apache2
+G_APACHE_CONF="@GENTOO_PORTAGE_EPREFIX@/etc/conf.d/apache2"
+G_APACHE_INIT="@GENTOO_PORTAGE_EPREFIX@/etc/init.d/apache2"
 G_OPTS_VAR="APACHE2_OPTS"
-G_APACHE_MOD_DIR="/usr/lib/apache2/modules"
+G_APACHE_MOD_DIR="@GENTOO_PORTAGE_EPREFIX@/usr/lib/apache2/modules"
 
 . $G_MODULE_PATH/libapache.sh
--- php-toolkit-1.0.2/php-select-modules/php-cgi.sh
+++ php-toolkit-1.0.2/php-select-modules/php-cgi.sh
@@ -13,6 +13,6 @@
 # ========================================================================
 
 G_SYMLINK_SOURCE[0]="bin/php-cgi"
-G_SYMLINK_TARGET[0]="/usr/bin/php-cgi"
+G_SYMLINK_TARGET[0]="@GENTOO_PORTAGE_EPREFIX@/usr/bin/php-cgi"
 
 . $G_MODULE_PATH/libsymlink.sh
--- php-toolkit-1.0.2/php-select-modules/php-devel.sh
+++ php-toolkit-1.0.2/php-select-modules/php-devel.sh
@@ -13,10 +13,10 @@
 # ========================================================================
 
 G_SYMLINK_SOURCE[0]="bin/php-config"
-G_SYMLINK_TARGET[0]="/usr/bin/php-config"
+G_SYMLINK_TARGET[0]="@GENTOO_PORTAGE_EPREFIX@/usr/bin/php-config"
 
 G_SYMLINK_SOURCE[1]="bin/phpize"
-G_SYMLINK_TARGET[1]="/usr/bin/phpize"
+G_SYMLINK_TARGET[1]="@GENTOO_PORTAGE_EPREFIX@/usr/bin/phpize"
 
 G_SYMLINK_SOURCE[2]="${G_LIBDIR_NAME}/libphp5.so"
 G_SYMLINK_TARGET[2]="${G_LIBDIR}/libphp5.so"
--- php-toolkit-1.0.2/php-select-modules/php.sh
+++ php-toolkit-1.0.2/php-select-modules/php.sh
@@ -13,6 +13,6 @@
 # ========================================================================
 
 G_SYMLINK_SOURCE[0]="bin/php"
-G_SYMLINK_TARGET[0]="/usr/bin/php"
+G_SYMLINK_TARGET[0]="@GENTOO_PORTAGE_EPREFIX@/usr/bin/php"
 
 . $G_MODULE_PATH/libsymlink.sh
