diff -u php-toolkit-1.0.1.orig/php-select php-toolkit-1.0.1/php-select
--- php-toolkit-1.0.1.orig/php-select	2008-04-06 12:04:04 +0200
+++ php-toolkit-1.0.1/php-select	2009-02-22 22:18:19 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 #
 # /usr/sbin/php-select
 #		Tool for managing PHP symlinks on Gentoo Linux
@@ -39,10 +39,10 @@
 G_COPYRIGHT[0]="Copyright (c) 2005 Gentoo Foundation, Inc."
 G_COPYRIGHT[1]="Released under v2 of the GNU General Public License"
 G_EXITSTATUS=0
-G_LIBDIR[0]="/usr/lib64"
-G_LIBDIR[1]="/usr/lib32"
-G_LIBDIR[2]="/usr/lib"
-G_MODULE_PATH="/usr/share/php-select"
+G_LIBDIR[0]="@GENTOO_PORTAGE_EPREFIX@/usr/lib64"
+G_LIBDIR[1]="@GENTOO_PORTAGE_EPREFIX@/usr/lib32"
+G_LIBDIR[2]="@GENTOO_PORTAGE_EPREFIX@/usr/lib"
+G_MODULE_PATH="@GENTOO_PORTAGE_EPREFIX@/usr/share/php-select"
 G_NEED_COMPONENT=0
 G_NEED_PHPVERSION=0
 G_PN="$0"
@@ -106,8 +106,8 @@
 	echo
 	echo "    apache1   - Apache webserver v1.x"
 	echo "    apache2   - Apache webserver v2.x"
-	echo "    php       - /usr/bin/php (the CLI SAPI)"
-	echo "    php-cgi   - /usr/bin/php-cgi (the CGI SAPI)"
+	echo "    php       - @GENTOO_PORTAGE_EPREFIX@/usr/bin/php (the CLI SAPI)"
+	echo "    php-cgi   - @GENTOO_PORTAGE_EPREFIX@/usr/bin/php-cgi (the CGI SAPI)"
 	echo "    php-devel - the scripts used to build PECL extensions"
 	echo
 	echo "and where <php-version> is one of:"
Common subdirectories: php-toolkit-1.0.1.orig/php-select-modules and php-toolkit-1.0.1/php-select-modules
Only in php-toolkit-1.0.1: prefix.patch
diff -Naur old/php-select-modules/apache1.sh new/php-select-modules/apache1.sh
--- old/php-select-modules/apache1.sh	2009-02-23 01:54:29 +0100
+++ new/php-select-modules/apache1.sh	2009-02-23 01:56:52 +0100
@@ -12,9 +12,9 @@
 #
 # ========================================================================
 
-G_APACHE_CONF=/etc/conf.d/apache
-G_APACHE_INIT=/etc/init.d/apache
+G_APACHE_CONF="@GENTOO_PORTAGE_EPREFIX@/etc/conf.d/apache"
+G_APACHE_INIT="@GENTOO_PORTAGE_EPREFIX@/etc/init.d/apache"
 G_OPTS_VAR="APACHE_OPTS"
-G_APACHE_MOD_DIR="/usr/lib/apache/modules"
+G_APACHE_MOD_DIR="@GENTOO_PORTAGE_EPREFIX@/usr/lib/apache/modules"
 
 . $G_MODULE_PATH/libapache.sh
diff -Naur old/php-select-modules/apache2.sh new/php-select-modules/apache2.sh
--- old/php-select-modules/apache2.sh	2009-02-23 01:54:29 +0100
+++ new/php-select-modules/apache2.sh	2009-02-23 01:57:07 +0100
@@ -12,9 +12,9 @@
 #
 # ========================================================================
 
-G_APACHE_CONF=/etc/conf.d/apache2
-G_APACHE_INIT=/etc/init.d/apache2
+G_APACHE_CONF="@GENTOO_PORTAGE_EPREFIX@/etc/conf.d/apache2"
+G_APACHE_INIT="@GENTOO_PORTAGE_EPREFIX@/etc/init.d/apache2"
 G_OPTS_VAR="APACHE2_OPTS"
-G_APACHE_MOD_DIR="/usr/lib/apache2/modules"
+G_APACHE_MOD_DIR="@GENTOO_PORTAGE_EPREFIX@/usr/lib/apache2/modules"
 
 . $G_MODULE_PATH/libapache.sh
diff -Naur old/php-select-modules/php-cgi.sh new/php-select-modules/php-cgi.sh
--- old/php-select-modules/php-cgi.sh	2009-02-23 01:54:29 +0100
+++ new/php-select-modules/php-cgi.sh	2009-02-23 01:57:22 +0100
@@ -13,6 +13,6 @@
 # ========================================================================
 
 G_SYMLINK_SOURCE[0]="bin/php-cgi"
-G_SYMLINK_TARGET[0]="/usr/bin/php-cgi"
+G_SYMLINK_TARGET[0]="@GENTOO_PORTAGE_EPREFIX@/usr/bin/php-cgi"
 
 . $G_MODULE_PATH/libsymlink.sh
diff -Naur old/php-select-modules/php-devel.sh new/php-select-modules/php-devel.sh
--- old/php-select-modules/php-devel.sh	2009-02-23 01:54:29 +0100
+++ new/php-select-modules/php-devel.sh	2009-02-23 01:57:33 +0100
@@ -13,9 +13,9 @@
 # ========================================================================
 
 G_SYMLINK_SOURCE[0]="bin/php-config"
-G_SYMLINK_TARGET[0]="/usr/bin/php-config"
+G_SYMLINK_TARGET[0]="@GENTOO_PORTAGE_EPREFIX@/usr/bin/php-config"
 
 G_SYMLINK_SOURCE[1]="bin/phpize"
-G_SYMLINK_TARGET[1]="/usr/bin/phpize"
+G_SYMLINK_TARGET[1]="@GENTOO_PORTAGE_EPREFIX@/usr/bin/phpize"
 
 . $G_MODULE_PATH/libsymlink.sh
diff -Naur old/php-select-modules/php.sh new/php-select-modules/php.sh
--- old/php-select-modules/php.sh	2009-02-23 01:54:29 +0100
+++ new/php-select-modules/php.sh	2009-02-23 01:57:43 +0100
@@ -13,6 +13,6 @@
 # ========================================================================
 
 G_SYMLINK_SOURCE[0]="bin/php"
-G_SYMLINK_TARGET[0]="/usr/bin/php"
+G_SYMLINK_TARGET[0]="@GENTOO_PORTAGE_EPREFIX@/usr/bin/php"
 
 . $G_MODULE_PATH/libsymlink.sh
