--- configure.ac
+++ configure.ac
@@ -890,7 +890,7 @@
 
     darwin*) # check for special options added by Apple
              FLAGS_DYN="$FLAGS_DYN$DASH_bundle$DASH_flat_namespace_undefined_suppress"
-             FLAGS_LIB="$FLAGS_LIB$DASH_dynamiclib$DASH_flat_namespace_undefined_suppress"
+             FLAGS_LIB="$FLAGS_LIB$DASH_dynamiclib$DASH_flat_namespace_undefined_suppress -install_name \$(libdir)/\$(LIBRARY)"
              ;;
 
           *) # for other systems
