diff -ruBb coreutils-6.9.orig/lib/savewd.c coreutils-6.9/lib/savewd.c
--- coreutils-6.9.orig/lib/savewd.c	Wed Jan 16 13:48:39 2008
+++ coreutils-6.9/lib/savewd.c	Wed Jan 16 13:49:08 2008
@@ -35,6 +35,9 @@
 #include "dirname.h"
 #include "fcntl-safer.h"
 
+#ifndef ESTALE
+# define ESTALE -1
+#endif
 
 /* Save the working directory into *WD, if it hasn't been saved
    already.  Return true if a child has been forked to do the real
