--- lib/wctype_.h.orig	2007-06-29 11:01:48.155230000 +0200
+++ lib/wctype_.h	2007-06-29 11:39:39.658842000 +0200
@@ -29,7 +29,7 @@
 #ifndef _GL_WCTYPE_H
 #define _GL_WCTYPE_H
 
-#if @HAVE_WINT_T@
+#if HAVE_WINT_T
 /* Solaris 2.5 has a bug: <wchar.h> must be included before <wctype.h>.
    Tru64 with Desktop Toolkit C has a bug: <stdio.h> must be included before
    <wchar.h>.
@@ -46,12 +46,15 @@
 
 /* Include the original <wctype.h> if it exists.
    BeOS 5 has the functions but no <wctype.h>.  */
-#if @HAVE_WCTYPE_H@
-# include @ABSOLUTE_WCTYPE_H@
+#if HAVE_WCTYPE_H
+# include <wctype.h>
 #endif
 
 /* FreeBSD 4.4 to 4.11 has <wctype.h> but lacks the functions.
    Assume all 12 functions are implemented the same way, or not at all.  */
+/* With one exception: iswblank is missing on Solaris 2.9.
+   It was first specified in "The Open Group Base Specifications Issue 6",
+   while others where available in "The Open Group Base Specifications Issue 4" */
 #if ! HAVE_ISWCNTRL
 
 /* IRIX 5.3 has macros but no functions, its isw* macros refer to an
@@ -61,7 +64,6 @@
    implementations like this to work, just disable them.  */
 #  undef iswalnum
 #  undef iswalpha
-#  undef iswblank
 #  undef iswcntrl
 #  undef iswdigit
 #  undef iswgraph
@@ -86,12 +88,6 @@
 }
 
 static inline int
-iswblank (__wctype_wint_t wc)
-{
-  return wc == ' ' || wc == '\t';
-}
-
-static inline int
 iswcntrl (__wctype_wint_t wc)
 {
   return (wc & ~0x1f) == 0 || wc == 0x7f;
@@ -151,4 +147,16 @@
 
 # endif /* ! HAVE_ISWCNTRL */
 
+#if ! HAVE_ISWBLANK
+/* Solaris in principal has isw*, but lacks iswblank */
+
+#  undef iswblank
+
+static inline int
+iswblank (__wctype_wint_t wc)
+{
+  return wc == ' ' || wc == '\t';
+}
+#endif /* ! HAVE_ISWBLANK */
+
 #endif /* _GL_WCTYPE_H */
--- lib/wchar_.h.orig	2007-06-29 11:28:33.908300000 +0200
+++ lib/wchar_.h	2007-06-29 11:29:38.526424000 +0200
@@ -37,6 +37,8 @@
 #include <time.h>
 
 /* Include the original <wchar.h>.  */
-#include @ABSOLUTE_WCHAR_H@
+#if HAVE_WCHAR_H
+# include <wchar.h>
+#endif
 
 #endif /* _GL_WCHAR_H */
--- src/join.c.orig	2007-06-29 11:28:10.129002000 +0200
+++ src/join.c	2007-06-29 11:28:20.168712000 +0200
@@ -29,9 +29,7 @@
 #endif
 
 /* Get iswblank(), towupper.  */
-#if HAVE_WCTYPE_H
-# include <wctype.h>
-#endif
+#include <wctype_.h>
 
 #include "system.h"
 #include "error.h"
--- src/sort.c.orig	2007-06-29 11:31:00.134019000 +0200
+++ src/sort.c	2007-06-29 11:31:23.123345000 +0200
@@ -28,13 +28,9 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <signal.h>
-#if HAVE_WCHAR_H
-# include <wchar.h>
-#endif
+#include <wchar_.h>
 /* Get isw* functions. */
-#if HAVE_WCTYPE_H
-# include <wctype.h>
-#endif
+#include <wctype_.h>
 
 #include "system.h"
 #include "argmatch.h"
--- src/uniq.c.orig	2007-06-29 11:31:52.972469000 +0200
+++ src/uniq.c	2007-06-29 11:32:05.962088000 +0200
@@ -24,14 +24,10 @@
 #include <sys/types.h>
 
 /* Get mbstate_t, mbrtowc(). */
-#if HAVE_WCHAR_H
-# include <wchar.h>
-#endif
+#include <wchar_.h>
 
 /* Get isw* functions. */
-#if HAVE_WCTYPE_H
-# include <wctype.h>
-#endif
+#include <wctype_.h>
 
 #include "system.h"
 #include "argmatch.h"
--- m4/jm-macros.m4.orig	2007-06-29 11:01:11.916342000 +0200
+++ m4/jm-macros.m4	2007-06-29 11:01:21.196078000 +0200
@@ -62,6 +62,7 @@
     fchmod \
     ftruncate \
     iswspace \
+    iswblank \
     mkfifo \
     mbrlen \
     setgroups \
--- src/fold.c.orig	2007-06-29 11:59:50.674323000 +0200
+++ src/fold.c	2007-06-29 12:00:07.223872000 +0200
@@ -29,9 +29,7 @@
 #endif
 
 /* Get iswprint(), iswblank(), wcwidth().  */
-#if HAVE_WCTYPE_H
-# include <wctype.h>
-#endif
+#include <wctype_.h>
 
 #include "system.h"
 #include "error.h"
