--- coreutils-8.8.orig/lib/striconv.c
+++ coreutils-8.8/lib/striconv.c
@@ -53,7 +53,7 @@ mem_cd_iconv (const char *src, size_t sr
   /* Avoid glibc-2.1 bug and Solaris 2.7-2.9 bug.  */
 # if defined _LIBICONV_VERSION \
      || !(((__GLIBC__ == 2 && __GLIBC_MINOR__ <= 1) && !defined __UCLIBC__) \
-          || defined __sun)
+          || defined __sun || (defined __hppa && defined __hpux))
   /* Set to the initial state.  */
   iconv (cd, NULL, NULL, NULL, NULL);
 # endif
