--- src/wheel-gen.pl
+++ src/wheel-gen.pl
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!@GENTOO_PORTAGE_EPREFIX@/usr/bin/perl -w
 # Generate the spokes of a wheel, for wheel factorization.
 
 # Copyright (C) 2001, 2005 Free Software Foundation, Inc.
@@ -17,7 +17,7 @@
 # along with this program; if not, write to the Free Software Foundation,
 # Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 
-eval 'exec /usr/bin/perl -S $0 ${1+"$@"}'
+eval 'exec @GENTOO_PORTAGE_EPREFIX@/usr/bin/perl -S $0 ${1+"$@"}'
   if 0;
 
 use strict;
