--- src/remove.c
+++ src/remove.c
@@ -59,9 +59,12 @@
    we could conceivably increase the maximum value to 254.  */
 enum
   {
-    CONSECUTIVE_READDIR_UNLINK_THRESHOLD = 200
+    CONSECUTIVE_READDIR_UNLINK_THRESHOLD = 10
   };
 
+#define NEED_REWIND(readdir_unlink_count) \
+  (CONSECUTIVE_READDIR_UNLINK_THRESHOLD <= (readdir_unlink_count))
+
 enum Ternary
   {
     T_UNKNOWN = 2,
@@ -899,8 +902,7 @@
 	      /* Arrange to give a diagnostic after exiting this loop.  */
 	      dirp = NULL;
 	    }
-	  else if (CONSECUTIVE_READDIR_UNLINK_THRESHOLD
-		   < n_unlinked_since_opendir_or_last_rewind)
+	  else if (NEED_REWIND (n_unlinked_since_opendir_or_last_rewind))
 	    {
 	      /* Call rewinddir if we've called unlink or rmdir so many times
 		 (since the opendir or the previous rewinddir) that this
