http://bugs.gentoo.org/show_bug.cgi?id=290236

--- src/libssh2_priv.h.dist	2009-10-23 13:11:07.159736240 +0100
+++ src/libssh2_priv.h	2009-10-23 13:13:28.335412880 +0100
@@ -679,7 +679,7 @@ struct _LIBSSH2_SESSION
 
     /* Agreed Key Exchange Method */
     const LIBSSH2_KEX_METHOD *kex;
-    int burn_optimistic_kexinit:1;
+    unsigned int burn_optimistic_kexinit:1;
 
     unsigned char *session_id;
     unsigned long session_id_len;
@@ -1026,9 +1026,11 @@ struct _LIBSSH2_MAC_METHOD
 void _libssh2_debug(LIBSSH2_SESSION * session, int context, const char *format,
                     ...);
 #else
-#if (defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)) || defined(__GNUC__)
+#if (defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L) && !defined(__sgi)) || defined(__GNUC__)
 /* C99 supported and also by older GCC */
 #define _libssh2_debug(x,y,z,...) do {} while (0)
+#elif defined(__sgi)
+#define _libssh2_debug(x,y,...) do {} while (0)
 #else
 /* no gcc and not C99, do static and hopefully inline */
 static inline void
