diff -ru -x '*.Po' -x '*.Plo' libXaw-1.0.5.orig/configure.ac libXaw-1.0.5/configure.ac
--- libXaw-1.0.5.orig/configure.ac	2009-01-20 17:14:37 +0100
+++ libXaw-1.0.5/configure.ac	2009-01-21 08:06:31 +0100
@@ -23,17 +23,6 @@
 # fix libtool to set SONAME to libXaw.so.$major
 #
 AC_PROG_SED
-AC_CONFIG_COMMANDS([libtool_hack], [
-	cp -f libtool libtool_
-	test -z "$SED" && SED=sed
-	$SED '1,/^soname_spec/{
-/^soname_spec/i\
-# X.Org hack to match monolithic Xaw SONAME\
-xorglibxawname="libXaw"
-/^soname_spec/s/libname/xorglibxawname/
-}' libtool_ > libtool
-	rm -f libtool_
-])
 
 # Win32 DLL rules are different.
 case $host_os in
