diff -ur libXaw-1.0.5.orig/Makefile.am libXaw-1.0.5/Makefile.am
--- libXaw-1.0.5.orig/Makefile.am	2008-11-07 18:15:56 +0100
+++ libXaw-1.0.5/Makefile.am	2009-01-18 18:06:56 +0100
@@ -6,14 +6,22 @@
 if BUILD_XAW6
 
 pkgconfig_DATA += xaw6.pc
+if PLATFORM_DARWIN
+DEFAULT_LIB = libXaw6.dylib
+else
 DEFAULT_LIB = libXaw6.so
+endif
 
 endif
 
 if BUILD_XAW7
 
 pkgconfig_DATA += xaw7.pc
+if PLATFORM_DARWIN
+DEFAULT_LIB = libXaw7.dylib
+else
 DEFAULT_LIB = libXaw7.so
+endif
 
 endif
 
@@ -25,8 +33,13 @@
 	ChangeLog
 
 if !PLATFORM_WIN32
+if PLATFORM_DARWIN
 install-exec-hook:
-	cd $(DESTDIR)$(libdir) && rm -f libXaw.so && ln -s $(DEFAULT_LIB) libXaw.so
+	cd $(DESTDIR)$(libdir) && rm -f libXaw.dylib && $(LN_S) $(DEFAULT_LIB) libXaw.dylib
+else
+install-exec-hook:
+	cd $(DESTDIR)$(libdir) && rm -f libXaw.so && $(LN_S) $(DEFAULT_LIB) libXaw.so
+endif
 endif
 
 MAINTAINERCLEANFILES=ChangeLog
diff -ur libXaw-1.0.5.orig/configure.ac libXaw-1.0.5/configure.ac
--- libXaw-1.0.5.orig/configure.ac	2008-11-07 18:43:06 +0100
+++ libXaw-1.0.5/configure.ac	2009-01-18 18:04:31 +0100
@@ -39,12 +39,19 @@
 case $host_os in
     cygwin*|mingw*)
 	platform_win32=yes
+	platform_darwin=no
+	;;
+	*darwin*)
+	platform_win32=no
+	platform_darwin=yes
 	;;
     *)
 	platform_win32=no
+	platform_darwin=no
 	;;
 esac
 AM_CONDITIONAL(PLATFORM_WIN32, test "x$platform_win32" = "xyes")
+AM_CONDITIONAL(PLATFORM_DARWIN, test "x$platform_darwin" = "xyes")
 
 # Whether to build Xaw6
 
Only in libXaw-1.0.5: configure.ac.orig
diff -ur libXaw-1.0.5.orig/src/Makefile.am libXaw-1.0.5/src/Makefile.am
--- libXaw-1.0.5.orig/src/Makefile.am	2008-09-25 21:50:19 +0200
+++ libXaw-1.0.5/src/Makefile.am	2009-01-18 18:05:44 +0100
@@ -76,9 +76,15 @@
 libXaw6_la_LIBADD = $(XAW6_LIBS)
 
 if !PLATFORM_WIN32
+if PLATFORM_DARWIN
+install-exec-hook::
+	-rm -f $(DESTDIR)$(libdir)/libXaw.6.dylib
+	(cd $(DESTDIR)$(libdir) && $(LN_S) libXaw6.6.dylib libXaw.6.dylib)
+else
 install-exec-hook::
 	-rm -f $(DESTDIR)$(libdir)/libXaw.so.6
-	(cd $(DESTDIR)$(libdir) && ln -s libXaw6.so.6 libXaw.so.6)
+	(cd $(DESTDIR)$(libdir) && $(LN_S) libXaw6.so.6 libXaw.so.6)
+endif
 endif
 
 endif
@@ -100,9 +106,15 @@
 libXaw7_la_LIBADD = $(XAW7_LIBS)
 
 if !PLATFORM_WIN32
+if PLATFORM_DARWIN
+install-exec-hook::
+	-rm -f $(DESTDIR)$(libdir)/libXaw.7.dylib
+	(cd $(DESTDIR)$(libdir) && $(LN_S) libXaw7.7.dylib libXaw.7.dylib)
+else
 install-exec-hook::
 	-rm -f $(DESTDIR)$(libdir)/libXaw.so.7
-	(cd $(DESTDIR)$(libdir) && ln -s libXaw7.so.7 libXaw.so.7)
+	(cd $(DESTDIR)$(libdir) && $(LN_S) libXaw7.so.7 libXaw.so.7)
+endif
 endif
 
 endif
