diff -ru libXaw-1.0.5.orig/include/X11/Xaw/AllWidgets.h libXaw-1.0.5/include/X11/Xaw/AllWidgets.h
--- libXaw-1.0.5.orig/include/X11/Xaw/AllWidgets.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/AllWidgets.h	2009-01-26 11:34:32 +0100
@@ -28,11 +28,12 @@
 #define _XawAllWidgets_h
 
 #include <X11/Xmu/WidgetNode.h>
+#include <X11/Intrinsic.h>
 
 /*
  * This file matches the generated AllWidgets.c
  */
-extern XmuWidgetNode XawWidgetArray[];
-extern int           XawWidgetCount;
+externalref XmuWidgetNode XawWidgetArray[];
+externalref int           XawWidgetCount;
 
 #endif /* _XawAllWidgets_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/AsciiSink.h libXaw-1.0.5/include/X11/Xaw/AsciiSink.h
--- libXaw-1.0.5.orig/include/X11/Xaw/AsciiSink.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/AsciiSink.h	2009-01-26 11:34:37 +0100
@@ -58,6 +58,7 @@
  */
 
 #include <X11/Xaw/TextSink.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -74,7 +75,7 @@
 #define XtNecho "echo"
 
 /* Class record constants */
-extern WidgetClass asciiSinkObjectClass;
+externalref WidgetClass asciiSinkObjectClass;
 
 typedef struct _AsciiSinkClassRec *AsciiSinkObjectClass;
 typedef struct _AsciiSinkRec      *AsciiSinkObject;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/AsciiSinkP.h libXaw-1.0.5/include/X11/Xaw/AsciiSinkP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/AsciiSinkP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/AsciiSinkP.h	2009-01-26 11:34:41 +0100
@@ -59,6 +59,7 @@
  */
 #include <X11/Xaw/TextSinkP.h> 
 #include <X11/Xaw/AsciiSink.h> 
+#include <X11/Intrinsic.h>
 
 /* New fields for the AsciiSink object class record */
 typedef struct _AsciiSinkClassPart {
@@ -72,7 +73,7 @@
     AsciiSinkClassPart	ascii_sink_class;
 } AsciiSinkClassRec;
 
-extern AsciiSinkClassRec asciiSinkClassRec;
+externalref AsciiSinkClassRec asciiSinkClassRec;
 
 /* New fields for the AsciiSink object record */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/AsciiSrc.h libXaw-1.0.5/include/X11/Xaw/AsciiSrc.h
--- libXaw-1.0.5.orig/include/X11/Xaw/AsciiSrc.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/AsciiSrc.h	2009-01-26 11:34:46 +0100
@@ -47,6 +47,7 @@
 #define _XawAsciiSrc_h
 
 #include <X11/Xaw/TextSrc.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -61,7 +62,7 @@
 
 */
  
-extern WidgetClass asciiSrcObjectClass;
+externalref WidgetClass asciiSrcObjectClass;
 
 typedef struct _AsciiSrcClassRec *AsciiSrcObjectClass;
 typedef struct _AsciiSrcRec      *AsciiSrcObject;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/AsciiSrcP.h libXaw-1.0.5/include/X11/Xaw/AsciiSrcP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/AsciiSrcP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/AsciiSrcP.h	2009-01-26 11:34:52 +0100
@@ -70,6 +70,7 @@
 
 #include <X11/Xaw/TextSrcP.h>
 #include <X11/Xaw/AsciiSrc.h>
+#include <X11/Intrinsic.h>
 
 #ifdef L_tmpnam
 #define TMPSIZ L_tmpnam
@@ -100,7 +101,7 @@
     AsciiSrcClassPart	ascii_src_class;
 } AsciiSrcClassRec;
 
-extern AsciiSrcClassRec asciiSrcClassRec;
+externalref AsciiSrcClassRec asciiSrcClassRec;
 
 /* New fields for the AsciiSrc object */
 typedef struct _AsciiSrcPart {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/AsciiText.h libXaw-1.0.5/include/X11/Xaw/AsciiText.h
--- libXaw-1.0.5.orig/include/X11/Xaw/AsciiText.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/AsciiText.h	2009-01-26 11:34:58 +0100
@@ -68,6 +68,7 @@
 #include <X11/Xaw/Text.h>
 #include <X11/Xaw/AsciiSrc.h>
 #include <X11/Xaw/MultiSrc.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -112,17 +113,17 @@
 typedef struct _AsciiTextClassRec	*AsciiTextWidgetClass;
 typedef struct _AsciiRec	        *AsciiWidget;
 
-extern WidgetClass asciiTextWidgetClass;
+externalref WidgetClass asciiTextWidgetClass;
 
 /*
  * Disk and String Emulation Info
  */
 #ifdef ASCII_STRING
-extern WidgetClass asciiStringWidgetClass;
+externalref WidgetClass asciiStringWidgetClass;
 #endif
 
 #ifdef ASCII_DISK
-extern WidgetClass asciiDiskWidgetClass;
+externalref WidgetClass asciiDiskWidgetClass;
 #endif
 
 #endif /* _AsciiText_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/AsciiTextP.h libXaw-1.0.5/include/X11/Xaw/AsciiTextP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/AsciiTextP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/AsciiTextP.h	2009-01-26 11:35:04 +0100
@@ -70,6 +70,7 @@
 #include <X11/Xaw/AsciiText.h>
 #include <X11/Xaw/AsciiSrc.h>
 #include <X11/Xaw/MultiSrc.h>
+#include <X11/Intrinsic.h>
 
 typedef struct {
   XtPointer extension;
@@ -82,7 +83,7 @@
     AsciiClassPart	ascii_class;
 } AsciiTextClassRec;
 
-extern AsciiTextClassRec asciiTextClassRec;
+externalref AsciiTextClassRec asciiTextClassRec;
 
 typedef struct {
     int resource;
@@ -114,7 +115,7 @@
     AsciiStringClassPart string_class;
 } AsciiStringClassRec;
 
-extern AsciiStringClassRec asciiStringClassRec;
+externalref AsciiStringClassRec asciiStringClassRec;
 
 typedef struct {
     int resource;
@@ -148,7 +149,7 @@
     AsciiDiskClassPart	disk_class;
 } AsciiDiskClassRec;
 
-extern AsciiDiskClassRec asciiDiskClassRec;
+externalref AsciiDiskClassRec asciiDiskClassRec;
 
 typedef struct {
     char resource;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Box.h libXaw-1.0.5/include/X11/Xaw/Box.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Box.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Box.h	2009-01-26 11:35:08 +0100
@@ -55,6 +55,7 @@
 #define _XawBox_h
 
 #include <X11/Xmu/Converters.h>
+#include <X11/Intrinsic.h>
 
 /*
  * Box Widget (subclass of CompositeClass)
@@ -100,7 +101,7 @@
 #endif /* OLDXAW */
 
 /* Class record constants */
-extern WidgetClass boxWidgetClass;
+externalref WidgetClass boxWidgetClass;
 
 typedef struct _BoxClassRec *BoxWidgetClass;
 typedef struct _BoxRec      *BoxWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/BoxP.h libXaw-1.0.5/include/X11/Xaw/BoxP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/BoxP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/BoxP.h	2009-01-26 11:35:12 +0100
@@ -60,6 +60,7 @@
 #include <X11/Xaw/Box.h>
 #include <X11/Xmu/Converters.h>
 #include <X11/Xaw/XawInit.h>
+#include <X11/Intrinsic.h>
 
 /* New fields for the Box widget class record */
 typedef struct {
@@ -73,7 +74,7 @@
     BoxClassPart	box_class;
 } BoxClassRec;
 
-extern BoxClassRec boxClassRec;
+externalref BoxClassRec boxClassRec;
 
 /* New fields for the Box widget record */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Command.h libXaw-1.0.5/include/X11/Xaw/Command.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Command.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Command.h	2009-01-26 11:35:17 +0100
@@ -52,6 +52,7 @@
 #define _XawCommand_h
 
 #include <X11/Xaw/Label.h>
+#include <X11/Intrinsic.h>
 
 /* Command widget resources:
 
@@ -111,7 +112,7 @@
 #define XawShapeEllipse XmuShapeEllipse
 #define XawShapeRoundedRectangle XmuShapeRoundedRectangle
 
-extern WidgetClass     commandWidgetClass;
+externalref WidgetClass     commandWidgetClass;
 
 typedef struct _CommandClassRec   *CommandWidgetClass;
 typedef struct _CommandRec        *CommandWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/CommandP.h libXaw-1.0.5/include/X11/Xaw/CommandP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/CommandP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/CommandP.h	2009-01-26 11:35:22 +0100
@@ -59,6 +59,7 @@
  */
 #include <X11/Xaw/Command.h>
 #include <X11/Xaw/LabelP.h>
+#include <X11/Intrinsic.h>
 
 typedef enum {
     HighlightNone,		/* Do not highlight */
@@ -82,7 +83,7 @@
     CommandClassPart    command_class;
 } CommandClassRec;
 
-extern CommandClassRec commandClassRec;
+externalref CommandClassRec commandClassRec;
 
 /* New fields for the Command widget record */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Dialog.h libXaw-1.0.5/include/X11/Xaw/Dialog.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Dialog.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Dialog.h	2009-01-26 11:35:25 +0100
@@ -52,6 +52,7 @@
 #define _XawDialog_h
 
 #include <X11/Xaw/Form.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -79,7 +80,7 @@
 typedef struct _DialogClassRec	*DialogWidgetClass;
 typedef struct _DialogRec	*DialogWidget;
 
-extern WidgetClass dialogWidgetClass;
+externalref WidgetClass dialogWidgetClass;
 
 _XFUNCPROTOBEGIN
 
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/DialogP.h libXaw-1.0.5/include/X11/Xaw/DialogP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/DialogP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/DialogP.h	2009-01-26 11:35:28 +0100
@@ -54,6 +54,7 @@
 
 #include <X11/Xaw/Dialog.h>
 #include <X11/Xaw/FormP.h>
+#include <X11/Intrinsic.h>
 
 typedef struct {
     XtPointer extension;
@@ -67,7 +68,7 @@
     DialogClassPart	dialog_class;
 } DialogClassRec;
 
-extern DialogClassRec dialogClassRec;
+externalref DialogClassRec dialogClassRec;
 
 typedef struct _DialogPart {
     /* resources */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Form.h libXaw-1.0.5/include/X11/Xaw/Form.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Form.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Form.h	2009-01-26 11:35:33 +0100
@@ -152,7 +152,7 @@
 typedef struct _FormClassRec	*FormWidgetClass;
 typedef struct _FormRec		*FormWidget;
 
-extern WidgetClass formWidgetClass;
+externalref WidgetClass formWidgetClass;
 
 _XFUNCPROTOBEGIN
 
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/FormP.h libXaw-1.0.5/include/X11/Xaw/FormP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/FormP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/FormP.h	2009-01-26 11:35:39 +0100
@@ -57,6 +57,7 @@
 
 #include <X11/Xaw/Form.h>
 #include <X11/Xaw/XawInit.h>
+#include <X11/Intrinsic.h>
 
 _XFUNCPROTOBEGIN
 
@@ -85,7 +86,7 @@
     FormClassPart	form_class;
 } FormClassRec;
 
-extern FormClassRec formClassRec;
+externalref FormClassRec formClassRec;
 
 typedef struct _FormPart {
     /* resources */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Grip.h libXaw-1.0.5/include/X11/Xaw/Grip.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Grip.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Grip.h	2009-01-26 11:35:44 +0100
@@ -57,6 +57,7 @@
 #define _XawGrip_h
 
 #include <X11/Xaw/Simple.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -91,7 +92,7 @@
 
 /* Class Record Constant */
 
-extern WidgetClass gripWidgetClass;
+externalref WidgetClass gripWidgetClass;
 
 typedef struct _GripClassRec *GripWidgetClass;
 typedef struct _GripRec      *GripWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/GripP.h libXaw-1.0.5/include/X11/Xaw/GripP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/GripP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/GripP.h	2009-01-26 11:35:47 +0100
@@ -56,6 +56,7 @@
 
 #include <X11/Xaw/Grip.h>
 #include <X11/Xaw/SimpleP.h>
+#include <X11/Intrinsic.h>
 
 #define DEFAULT_GRIP_SIZE 8
 
@@ -71,7 +72,7 @@
     GripClassPart    grip_class;
 } GripClassRec;
 
-extern GripClassRec gripClassRec;
+externalref GripClassRec gripClassRec;
 
 /* New fields for the Grip widget */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Label.h libXaw-1.0.5/include/X11/Xaw/Label.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Label.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Label.h	2009-01-26 11:35:52 +0100
@@ -56,6 +56,7 @@
  */
 
 #include <X11/Xaw/Simple.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -127,7 +128,7 @@
 
 /* Class record constants */
 
-extern WidgetClass labelWidgetClass;
+externalref WidgetClass labelWidgetClass;
 
 typedef struct _LabelClassRec *LabelWidgetClass;
 typedef struct _LabelRec      *LabelWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/LabelP.h libXaw-1.0.5/include/X11/Xaw/LabelP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/LabelP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/LabelP.h	2009-01-26 11:35:57 +0100
@@ -59,6 +59,7 @@
  */
 #include <X11/Xaw/Label.h>
 #include <X11/Xaw/SimpleP.h>
+#include <X11/Intrinsic.h>
 
 /* New fields for the Label widget class record */
 typedef struct {
@@ -72,7 +73,7 @@
     LabelClassPart	label_class;
 } LabelClassRec;
 
-extern LabelClassRec labelClassRec;
+externalref LabelClassRec labelClassRec;
 
 /* New fields for the Label widget record */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/List.h libXaw-1.0.5/include/X11/Xaw/List.h
--- libXaw-1.0.5.orig/include/X11/Xaw/List.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/List.h	2009-01-26 11:36:00 +0100
@@ -39,6 +39,7 @@
 #define _XawList_h
 
 #include <X11/Xaw/Simple.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -136,7 +137,7 @@
 #define XtCFontSet "FontSet"
 #endif
 
-extern WidgetClass listWidgetClass;
+externalref WidgetClass listWidgetClass;
 
 typedef struct _ListClassRec *ListWidgetClass;
 typedef struct _ListRec      *ListWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/ListP.h libXaw-1.0.5/include/X11/Xaw/ListP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/ListP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/ListP.h	2009-01-26 11:36:03 +0100
@@ -43,6 +43,7 @@
  */
 #include <X11/Xaw/SimpleP.h>
 #include <X11/Xaw/List.h>
+#include <X11/Intrinsic.h>
 
 #define NO_HIGHLIGHT            XAW_LIST_NONE
 #define OUT_OF_RANGE            -1
@@ -60,7 +61,7 @@
     ListClassPart	list_class;
 } ListClassRec;
 
-extern ListClassRec listClassRec;
+externalref ListClassRec listClassRec;
 
 /* New fields for the List widget */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/MenuButtoP.h libXaw-1.0.5/include/X11/Xaw/MenuButtoP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/MenuButtoP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/MenuButtoP.h	2009-01-26 11:36:05 +0100
@@ -42,6 +42,7 @@
 
 #include <X11/Xaw/MenuButton.h>
 #include <X11/Xaw/CommandP.h>
+#include <X11/Intrinsic.h>
 
 /* New fields for the MenuButton widget class */
 typedef struct _MenuButtonClass {
@@ -57,7 +58,7 @@
     MenuButtonClassPart	    menuButton_class;
 } MenuButtonClassRec;
 
-extern MenuButtonClassRec menuButtonClassRec;
+externalref MenuButtonClassRec menuButtonClassRec;
 
 /* New fields for the MenuButton widget */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/MenuButton.h libXaw-1.0.5/include/X11/Xaw/MenuButton.h
--- libXaw-1.0.5.orig/include/X11/Xaw/MenuButton.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/MenuButton.h	2009-01-26 11:36:08 +0100
@@ -48,6 +48,7 @@
 #define _XawMenuButton_h
 
 #include <X11/Xaw/Command.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -83,7 +84,7 @@
 #define XtNmenuName "menuName"
 #define XtCMenuName "MenuName"
 
-extern WidgetClass     menuButtonWidgetClass;
+externalref WidgetClass     menuButtonWidgetClass;
 
 typedef struct _MenuButtonClassRec   *MenuButtonWidgetClass;
 typedef struct _MenuButtonRec        *MenuButtonWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/MultiSink.h libXaw-1.0.5/include/X11/Xaw/MultiSink.h
--- libXaw-1.0.5.orig/include/X11/Xaw/MultiSink.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/MultiSink.h	2009-01-26 11:36:11 +0100
@@ -80,6 +80,7 @@
  */
 
 #include <X11/Xaw/TextSink.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -105,7 +106,7 @@
 #endif
 
 /* Class record constants */
-extern WidgetClass multiSinkObjectClass;
+externalref WidgetClass multiSinkObjectClass;
 
 typedef struct _MultiSinkClassRec *MultiSinkObjectClass;
 typedef struct _MultiSinkRec      *MultiSinkObject;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/MultiSinkP.h libXaw-1.0.5/include/X11/Xaw/MultiSinkP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/MultiSinkP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/MultiSinkP.h	2009-01-26 11:36:14 +0100
@@ -83,6 +83,7 @@
  */
 #include <X11/Xaw/TextSinkP.h> 
 #include <X11/Xaw/MultiSink.h> 
+#include <X11/Intrinsic.h>
 
 /* new fields for the MultiSink object class */
 typedef struct _MultiSinkClassPart {
@@ -96,7 +97,7 @@
     MultiSinkClassPart	multi_sink_class;
 } MultiSinkClassRec;
 
-extern MultiSinkClassRec multiSinkClassRec;
+externalref MultiSinkClassRec multiSinkClassRec;
 
 /* New fields for the MultiSink object record */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/MultiSrc.h libXaw-1.0.5/include/X11/Xaw/MultiSrc.h
--- libXaw-1.0.5.orig/include/X11/Xaw/MultiSrc.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/MultiSrc.h	2009-01-26 11:36:18 +0100
@@ -61,6 +61,7 @@
 #define _XawMultiSrc_h
 
 #include <X11/Xaw/TextSrc.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -75,7 +76,7 @@
 
 */
  
-extern WidgetClass multiSrcObjectClass;
+externalref WidgetClass multiSrcObjectClass;
 
 typedef struct _MultiSrcClassRec *MultiSrcObjectClass;
 typedef struct _MultiSrcRec      *MultiSrcObject;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/MultiSrcP.h libXaw-1.0.5/include/X11/Xaw/MultiSrcP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/MultiSrcP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/MultiSrcP.h	2009-01-26 11:36:22 +0100
@@ -99,6 +99,7 @@
 #include <X11/Xfuncproto.h>
 #include <X11/Xaw/TextSrcP.h>
 #include <X11/Xaw/MultiSrc.h>
+#include <X11/Intrinsic.h>
 
 #ifdef L_tmpnam
 #define TMPSIZ L_tmpnam
@@ -130,7 +131,7 @@
     MultiSrcClassPart	multi_src_class;
 } MultiSrcClassRec;
 
-extern MultiSrcClassRec multiSrcClassRec;
+externalref MultiSrcClassRec multiSrcClassRec;
 
 /* New fields for the MultiSrc object */
 typedef struct _MultiSrcPart {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Paned.h libXaw-1.0.5/include/X11/Xaw/Paned.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Paned.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Paned.h	2009-01-26 11:36:33 +0100
@@ -152,7 +152,7 @@
 #define XtCShowGrip "ShowGrip"
 
 /* Class record constant */
-extern WidgetClass panedWidgetClass;
+externalref WidgetClass panedWidgetClass;
 
 typedef struct _PanedClassRec	*PanedWidgetClass;
 typedef struct _PanedRec	*PanedWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/PanedP.h libXaw-1.0.5/include/X11/Xaw/PanedP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/PanedP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/PanedP.h	2009-01-26 11:36:40 +0100
@@ -63,6 +63,7 @@
 #define _XawPanedP_h
 
 #include <X11/Xaw/Paned.h>
+#include <X11/Intrinsic.h>
 
 /* New fields for the Paned widget class record */
 typedef struct _PanedClassPart {
@@ -77,7 +78,7 @@
     PanedClassPart      paned_class;
 } PanedClassRec;
 
-extern PanedClassRec panedClassRec;
+externalref PanedClassRec panedClassRec;
 
 /* Paned constraint record */
 typedef struct _PanedConstraintsPart {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Panner.h libXaw-1.0.5/include/X11/Xaw/Panner.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Panner.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Panner.h	2009-01-26 11:36:45 +0100
@@ -99,7 +99,7 @@
 #define XtNsliderHeight "sliderHeight"
 #define XtCSliderHeight "SliderHeight"
 
-extern WidgetClass pannerWidgetClass;
+externalref WidgetClass pannerWidgetClass;
 
 typedef struct _PannerClassRec *PannerWidgetClass;
 typedef struct _PannerRec      *PannerWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/PannerP.h libXaw-1.0.5/include/X11/Xaw/PannerP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/PannerP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/PannerP.h	2009-01-26 11:36:49 +0100
@@ -32,6 +32,7 @@
 
 #include <X11/Xaw/Panner.h>
 #include <X11/Xaw/SimpleP.h>
+#include <X11/Intrinsic.h>
 
 /* new fields in widget class */
 typedef struct {
@@ -103,6 +104,6 @@
 #define PANNER_DEFAULT_SCALE	8		/* percent */
 #define PANNER_OUTOFRANGE -30000
 
-extern PannerClassRec pannerClassRec;
+externalref PannerClassRec pannerClassRec;
 
 #endif /* _XawPannerP_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Porthole.h libXaw-1.0.5/include/X11/Xaw/Porthole.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Porthole.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Porthole.h	2009-01-26 11:36:51 +0100
@@ -56,7 +56,7 @@
  * 
  *****************************************************************************/
 
-extern WidgetClass portholeWidgetClass;
+externalref WidgetClass portholeWidgetClass;
 typedef struct _PortholeClassRec *PortholeWidgetClass;
 typedef struct _PortholeRec      *PortholeWidget;
 
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/PortholeP.h libXaw-1.0.5/include/X11/Xaw/PortholeP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/PortholeP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/PortholeP.h	2009-01-26 11:36:54 +0100
@@ -31,6 +31,7 @@
 #define _XawPortholeP_h
 
 #include <X11/Xaw/Porthole.h>
+#include <X11/Intrinsic.h>
 
 /* new fields in widget class */
 typedef struct {
@@ -59,6 +60,6 @@
     PortholePart porthole;
 } PortholeRec;
 
-extern PortholeClassRec portholeClassRec;
+externalref PortholeClassRec portholeClassRec;
 
 #endif /* _XawPortholeP_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Repeater.h libXaw-1.0.5/include/X11/Xaw/Repeater.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Repeater.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Repeater.h	2009-01-26 11:36:56 +0100
@@ -31,6 +31,7 @@
 #define _XawRepeater_h
 
 #include <X11/Xaw/Command.h>
+#include <X11/Intrinsic.h>
 
 /*****************************************************************************
  * 
@@ -67,7 +68,7 @@
 #define XtCStopCallback "StopCallback"
 
 
-extern WidgetClass repeaterWidgetClass;
+externalref WidgetClass repeaterWidgetClass;
 
 typedef struct _RepeaterClassRec *RepeaterWidgetClass;
 typedef struct _RepeaterRec      *RepeaterWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/RepeaterP.h libXaw-1.0.5/include/X11/Xaw/RepeaterP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/RepeaterP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/RepeaterP.h	2009-01-26 11:36:59 +0100
@@ -32,6 +32,7 @@
 
 #include <X11/Xaw/CommandP.h>
 #include <X11/Xaw/Repeater.h>
+#include <X11/Intrinsic.h>
 
 /* new fields in widget class */
 typedef struct {
@@ -79,6 +80,6 @@
 #define REP_DEF_MINIMUM_DELAY		10
 #define REP_DEF_REPEAT_DELAY		50
 
-extern RepeaterClassRec repeaterClassRec;
+externalref RepeaterClassRec repeaterClassRec;
 
 #endif /* _XawRepeaterP_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Scrollbar.h libXaw-1.0.5/include/X11/Xaw/Scrollbar.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Scrollbar.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Scrollbar.h	2009-01-26 11:37:06 +0100
@@ -58,6 +58,7 @@
 
 #include <X11/Xmu/Converters.h>
 #include <X11/Xfuncproto.h>
+#include <X11/Intrinsic.h>
 
 /* Scrollbar resources:
 
@@ -116,7 +117,7 @@
 typedef struct _ScrollbarRec	  *ScrollbarWidget;
 typedef struct _ScrollbarClassRec *ScrollbarWidgetClass;
 
-extern WidgetClass scrollbarWidgetClass;
+externalref WidgetClass scrollbarWidgetClass;
 
 _XFUNCPROTOBEGIN
 
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/ScrollbarP.h libXaw-1.0.5/include/X11/Xaw/ScrollbarP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/ScrollbarP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/ScrollbarP.h	2009-01-26 11:37:18 +0100
@@ -56,6 +56,7 @@
 
 #include <X11/Xaw/Scrollbar.h>
 #include <X11/Xaw/SimpleP.h>
+#include <X11/Intrinsic.h>
 
 typedef struct {
     /* resources */
@@ -104,6 +105,6 @@
     ScrollbarClassPart		scrollbar_class;
 } ScrollbarClassRec;
 
-extern ScrollbarClassRec scrollbarClassRec;
+externalref ScrollbarClassRec scrollbarClassRec;
 
 #endif /* _ScrollbarP_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Simple.h libXaw-1.0.5/include/X11/Xaw/Simple.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Simple.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Simple.h	2009-01-26 11:37:23 +0100
@@ -52,6 +52,7 @@
 #define _Simple_h
 
 #include <X11/Xmu/Converters.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -111,6 +112,6 @@
 typedef struct _SimpleClassRec	*SimpleWidgetClass;
 typedef struct _SimpleRec	*SimpleWidget;
 
-extern WidgetClass simpleWidgetClass;
+externalref WidgetClass simpleWidgetClass;
 
 #endif /* _Simple_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/SimpleMenP.h libXaw-1.0.5/include/X11/Xaw/SimpleMenP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/SimpleMenP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/SimpleMenP.h	2009-01-26 11:37:27 +0100
@@ -43,6 +43,7 @@
 #include <X11/Xaw/SmeP.h>
 #include <X11/ShellP.h>
 #include <X11/Xaw/XawInit.h>
+#include <X11/Intrinsic.h>
 
 typedef struct {
     XtPointer extension;		/* For future needs */
@@ -56,7 +57,7 @@
     SimpleMenuClassPart	    simpleMenu_class;
 } SimpleMenuClassRec;
 
-extern SimpleMenuClassRec simpleMenuClassRec;
+externalref SimpleMenuClassRec simpleMenuClassRec;
 
 typedef struct _SimpleMenuPart {
   /* resources */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/SimpleMenu.h libXaw-1.0.5/include/X11/Xaw/SimpleMenu.h
--- libXaw-1.0.5.orig/include/X11/Xaw/SimpleMenu.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/SimpleMenu.h	2009-01-26 11:37:31 +0100
@@ -49,6 +49,7 @@
 
 #include <X11/Shell.h>
 #include <X11/Xmu/Converters.h>
+#include <X11/Intrinsic.h>
 
 /*
  * SimpleMenu widget
@@ -86,7 +87,7 @@
 typedef struct _SimpleMenuClassRec*	SimpleMenuWidgetClass;
 typedef struct _SimpleMenuRec*		SimpleMenuWidget;
 
-extern WidgetClass simpleMenuWidgetClass;
+externalref WidgetClass simpleMenuWidgetClass;
 
 #define XtNcursor "cursor"
 #define XtNbottomMargin "bottomMargin"
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/SimpleP.h libXaw-1.0.5/include/X11/Xaw/SimpleP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/SimpleP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/SimpleP.h	2009-01-26 11:37:35 +0100
@@ -59,6 +59,7 @@
 _XFUNCPROTOBEGIN
 
 #include <X11/Xaw/XawInit.h>
+#include <X11/Intrinsic.h>
 
 typedef struct {
     Bool (*change_sensitive)(Widget);
@@ -74,7 +75,7 @@
     SimpleClassPart	simple_class;
 } SimpleClassRec;
 
-extern SimpleClassRec simpleClassRec;
+externalref SimpleClassRec simpleClassRec;
 
 typedef struct {
     /* resources */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Sme.h libXaw-1.0.5/include/X11/Xaw/Sme.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Sme.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Sme.h	2009-01-26 11:37:38 +0100
@@ -68,6 +68,6 @@
 typedef struct _SmeClassRec *SmeObjectClass;
 typedef struct _SmeRec *SmeObject;
 
-extern WidgetClass smeObjectClass;
+externalref WidgetClass smeObjectClass;
 
 #endif /* _Sme_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/SmeBSB.h libXaw-1.0.5/include/X11/Xaw/SmeBSB.h
--- libXaw-1.0.5.orig/include/X11/Xaw/SmeBSB.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/SmeBSB.h	2009-01-26 11:37:41 +0100
@@ -44,6 +44,7 @@
 
 #include <X11/Xmu/Converters.h>
 #include <X11/Xaw/Sme.h>
+#include <X11/Intrinsic.h>
 
 /* BSB Menu Entry Resources:
 
@@ -71,7 +72,7 @@
 typedef struct _SmeBSBClassRec    *SmeBSBObjectClass;
 typedef struct _SmeBSBRec         *SmeBSBObject;
 
-extern WidgetClass smeBSBObjectClass;
+externalref WidgetClass smeBSBObjectClass;
 
 #define XtNleftBitmap "leftBitmap"
 #define XtNleftMargin "leftMargin"
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/SmeBSBP.h libXaw-1.0.5/include/X11/Xaw/SmeBSBP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/SmeBSBP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/SmeBSBP.h	2009-01-26 11:37:45 +0100
@@ -35,6 +35,7 @@
  */
 #include <X11/Xaw/SmeP.h>
 #include <X11/Xaw/SmeBSB.h>
+#include <X11/Intrinsic.h>
 
 typedef struct _SmeBSBClassPart {
     XtPointer extension;
@@ -47,7 +48,7 @@
     SmeBSBClassPart	sme_bsb_class;
 } SmeBSBClassRec;
 
-extern SmeBSBClassRec smeBSBClassRec;
+externalref SmeBSBClassRec smeBSBClassRec;
 
 /* New fields for the Sme Object record */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/SmeLine.h libXaw-1.0.5/include/X11/Xaw/SmeLine.h
--- libXaw-1.0.5.orig/include/X11/Xaw/SmeLine.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/SmeLine.h	2009-01-26 11:37:48 +0100
@@ -42,6 +42,7 @@
 
 #include <X11/Xaw/Sme.h>
 #include <X11/Xmu/Converters.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -66,6 +67,6 @@
 typedef struct _SmeLineClassRec *SmeLineObjectClass;
 typedef struct _SmeLineRec *SmeLineObject;
 
-extern WidgetClass smeLineObjectClass;
+externalref WidgetClass smeLineObjectClass;
 
 #endif /* _SmeLine_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/SmeLineP.h libXaw-1.0.5/include/X11/Xaw/SmeLineP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/SmeLineP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/SmeLineP.h	2009-01-26 11:37:51 +0100
@@ -35,6 +35,7 @@
  */
 #include <X11/Xaw/SmeP.h>
 #include <X11/Xaw/SmeLine.h>
+#include <X11/Intrinsic.h>
 
 /* New fields for the SmeLine widget class */
 typedef struct _SmeLineClassPart {
@@ -48,7 +49,7 @@
     SmeLineClassPart	sme_line_class;
 } SmeLineClassRec;
 
-extern SmeLineClassRec smeLineClassRec;
+externalref SmeLineClassRec smeLineClassRec;
 
 /* New fields for the SmeLine widget */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/SmeP.h libXaw-1.0.5/include/X11/Xaw/SmeP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/SmeP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/SmeP.h	2009-01-26 11:37:53 +0100
@@ -45,6 +45,7 @@
 #include <X11/Xfuncproto.h>
 
 #include <X11/Xaw/Sme.h>
+#include <X11/Intrinsic.h>
 
 _XFUNCPROTOBEGIN
 
@@ -62,7 +63,7 @@
     SmeClassPart	sme_class;
 } SmeClassRec;
 
-extern SmeClassRec smeClassRec;
+externalref SmeClassRec smeClassRec;
 
 /* New fields for the Sme widget */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/StripCharP.h libXaw-1.0.5/include/X11/Xaw/StripCharP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/StripCharP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/StripCharP.h	2009-01-26 11:37:57 +0100
@@ -56,6 +56,7 @@
 
 #include <X11/Xaw/StripChart.h>
 #include <X11/Xaw/SimpleP.h>
+#include <X11/Intrinsic.h>
 
 #define NO_GCS 0
 #define FOREGROUND	(1 << 0)
@@ -105,6 +106,6 @@
     StripChartClassPart strip_chart_class;
 } StripChartClassRec;
 
-extern StripChartClassRec stripChartClassRec;
+externalref StripChartClassRec stripChartClassRec;
 
 #endif /* _XawStripChartP_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/StripChart.h libXaw-1.0.5/include/X11/Xaw/StripChart.h
--- libXaw-1.0.5.orig/include/X11/Xaw/StripChart.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/StripChart.h	2009-01-26 11:38:01 +0100
@@ -114,6 +114,6 @@
 typedef struct _StripChartRec *StripChartWidget;
 typedef struct _StripChartClassRec *StripChartWidgetClass;
 
-extern WidgetClass stripChartWidgetClass;
+externalref WidgetClass stripChartWidgetClass;
 
 #endif /* _XawStripChart_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Template.h libXaw-1.0.5/include/X11/Xaw/Template.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Template.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Template.h	2009-01-26 11:38:04 +0100
@@ -65,6 +65,6 @@
 typedef struct _TemplateRec *TemplateWidget;
 
 /* declare the class constant */
-extern WidgetClass templateWidgetClass;
+externalref WidgetClass templateWidgetClass;
 
 #endif /* _Template_h */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/TemplateP.h libXaw-1.0.5/include/X11/Xaw/TemplateP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/TemplateP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/TemplateP.h	2009-01-26 11:38:08 +0100
@@ -34,6 +34,7 @@
 
 /* include superclass private header file */
 #include <X11/CoreP.h>
+#include <X11/Intrinsic.h>
 
 /* define unique representation types not found in <X11/StringDefs.h> */
 #define XtRTemplateResource "TemplateResource"
@@ -47,7 +48,7 @@
     TemplateClassPart	template_class;
 } TemplateClassRec;
 
-extern TemplateClassRec templateClassRec;
+externalref TemplateClassRec templateClassRec;
 
 typedef struct {
     /* resources */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Text.h libXaw-1.0.5/include/X11/Xaw/Text.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Text.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Text.h	2009-01-26 11:38:40 +0100
@@ -52,6 +52,7 @@
 #define _XawText_h
 
 #include <X11/Xaw/Simple.h>
+#include <X11/Intrinsic.h>
 
 /*
 
@@ -232,11 +233,11 @@
 /* XrmQuark is not unsigned long */
 #define XawTextFormat(tw,fmt) ((unsigned long)_XawTextFormat(tw) == (fmt))
 
-extern unsigned long FMT8BIT;
-extern unsigned long XawFmt8Bit;
-extern unsigned long XawFmtWide;
+externalref unsigned long FMT8BIT;
+externalref unsigned long XawFmt8Bit;
+externalref unsigned long XawFmtWide;
 
-extern WidgetClass textWidgetClass;
+externalref WidgetClass textWidgetClass;
 
 typedef struct _TextClassRec *TextWidgetClass;
 typedef struct _TextRec      *TextWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/TextP.h libXaw-1.0.5/include/X11/Xaw/TextP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/TextP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/TextP.h	2009-01-26 11:38:48 +0100
@@ -58,6 +58,7 @@
 
 #include <X11/Xaw/Text.h>
 #include <X11/Xaw/SimpleP.h>
+#include <X11/Intrinsic.h>
 
 _XFUNCPROTOBEGIN
 
@@ -70,8 +71,8 @@
 
 #define zeroPosition ((XawTextPosition)0)
 
-extern XtActionsRec _XawTextActionsTable[];
-extern Cardinal _XawTextActionsTableCount;
+externalref XtActionsRec _XawTextActionsTable[];
+externalref Cardinal _XawTextActionsTableCount;
 
 #define XawLF	  0x0a
 #define XawCR	  0x0d
@@ -125,7 +126,7 @@
     unsigned long format;
 } XawTextKillRing;
 
-extern XawTextKillRing *xaw_text_kill_ring;
+externalref XawTextKillRing *xaw_text_kill_ring;
 #endif
 
 /* Line Tables are n+1 long - last position displayed is in last lt entry */
@@ -186,7 +187,7 @@
     TextClassPart	text_class;
 } TextClassRec;
 
-extern TextClassRec textClassRec;
+externalref TextClassRec textClassRec;
 
 /* New fields for the Text widget record */
 typedef struct _TextPart {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/TextSink.h libXaw-1.0.5/include/X11/Xaw/TextSink.h
--- libXaw-1.0.5.orig/include/X11/Xaw/TextSink.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/TextSink.h	2009-01-26 11:38:52 +0100
@@ -54,6 +54,7 @@
 #define _XawTextSink_h
 
 #include <X11/Xaw/Text.h>
+#include <X11/Intrinsic.h>
 
 /***********************************************************************
  *
@@ -73,7 +74,7 @@
  
 /* Class record constants */
 
-extern WidgetClass textSinkObjectClass;
+externalref WidgetClass textSinkObjectClass;
 
 typedef struct _TextSinkClassRec *TextSinkObjectClass;
 typedef struct _TextSinkRec      *TextSinkObject;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/TextSinkP.h libXaw-1.0.5/include/X11/Xaw/TextSinkP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/TextSinkP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/TextSinkP.h	2009-01-26 11:39:02 +0100
@@ -61,6 +61,7 @@
 #include <X11/Xaw/TextP.h>	/* This sink works with the Text widget */
 #include <X11/Xaw/TextSrcP.h>	/* This sink works with the Text Source */
 #include <X11/Xmu/Xmu.h>
+#include <X11/Intrinsic.h>
 
 #ifndef OLDXAW
 /* font/fontset defined? */
@@ -201,7 +202,7 @@
     TextSinkClassPart	text_sink_class;
 } TextSinkClassRec;
 
-extern TextSinkClassRec textSinkClassRec;
+externalref TextSinkClassRec textSinkClassRec;
 
 /* New fields for the TextSink object */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/TextSrc.h libXaw-1.0.5/include/X11/Xaw/TextSrc.h
--- libXaw-1.0.5.orig/include/X11/Xaw/TextSrc.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/TextSrc.h	2009-01-26 11:39:07 +0100
@@ -54,6 +54,7 @@
 #define _XawTextSrc_h
 
 #include <X11/Xaw/Text.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -67,7 +68,7 @@
 */
  
 /* Class record constants */
-extern WidgetClass textSrcObjectClass;
+externalref WidgetClass textSrcObjectClass;
 
 typedef struct _TextSrcClassRec *TextSrcObjectClass;
 typedef struct _TextSrcRec      *TextSrcObject;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/TextSrcP.h libXaw-1.0.5/include/X11/Xaw/TextSrcP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/TextSrcP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/TextSrcP.h	2009-01-26 11:39:11 +0100
@@ -61,6 +61,7 @@
 
 #include <X11/Xaw/TextSrc.h>
 #include <X11/Xaw/TextP.h>	/* This source works with the Text widget */
+#include <X11/Intrinsic.h>
 
 _XFUNCPROTOBEGIN
 
@@ -132,7 +133,7 @@
     TextSrcClassPart	textSrc_class;
 } TextSrcClassRec;
 
-extern TextSrcClassRec textSrcClassRec;
+externalref TextSrcClassRec textSrcClassRec;
 
 #ifndef OLDXAW
 typedef struct _XawTextUndo XawTextUndo;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Tip.h libXaw-1.0.5/include/X11/Xaw/Tip.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Tip.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Tip.h	2009-01-26 11:39:48 +0100
@@ -37,6 +37,7 @@
  */
 
 #include <X11/Xaw/Simple.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -65,7 +66,7 @@
 typedef struct _TipClassRec *TipWidgetClass;
 typedef struct _TipRec *TipWidget;
 
-extern WidgetClass tipWidgetClass;
+externalref WidgetClass tipWidgetClass;
 
 #define XtNbottomMargin		"bottomMargin"
 #define XawNdisplayList		"displayList"
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/TipP.h libXaw-1.0.5/include/X11/Xaw/TipP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/TipP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/TipP.h	2009-01-26 11:39:52 +0100
@@ -35,6 +35,7 @@
 #include <X11/Xaw/Tip.h>
 #include <X11/CoreP.h>
 #include <X11/Xaw/XawInit.h>
+#include <X11/Intrinsic.h>
 
 typedef struct {
     XtPointer extension;
@@ -45,7 +46,7 @@
     TipClassPart tip_class;
 } TipClassRec;
 
-extern TipClassRec tipClassRec;
+externalref TipClassRec tipClassRec;
 
 typedef struct _TipPart {
     /* resources */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Toggle.h libXaw-1.0.5/include/X11/Xaw/Toggle.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Toggle.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Toggle.h	2009-01-26 11:39:55 +0100
@@ -39,6 +39,7 @@
 #define _XawToggle_h
 
 #include <X11/Xaw/Command.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -97,7 +98,7 @@
 #define XtNradioGroup "radioGroup"
 #define XtNradioData "radioData"
 
-extern WidgetClass               toggleWidgetClass;
+externalref WidgetClass               toggleWidgetClass;
 
 typedef struct _ToggleClassRec   *ToggleWidgetClass;
 typedef struct _ToggleRec        *ToggleWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/ToggleP.h libXaw-1.0.5/include/X11/Xaw/ToggleP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/ToggleP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/ToggleP.h	2009-01-26 11:39:58 +0100
@@ -41,6 +41,7 @@
 
 #include <X11/Xaw/Toggle.h>
 #include <X11/Xaw/CommandP.h>
+#include <X11/Intrinsic.h>
 
 /*
  * Toggle Widget Private Data
@@ -68,7 +69,7 @@
     ToggleClassPart	toggle_class;
 } ToggleClassRec;
 
-extern ToggleClassRec toggleClassRec;
+externalref ToggleClassRec toggleClassRec;
 
 /* New fields for the Toggle widget */
 typedef struct {
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Tree.h libXaw-1.0.5/include/X11/Xaw/Tree.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Tree.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Tree.h	2009-01-26 11:40:02 +0100
@@ -55,6 +55,7 @@
 #define _XawTree_h
 
 #include <X11/Xmu/Converters.h>
+#include <X11/Intrinsic.h>
 
 /******************************************************************************
  * 
@@ -121,7 +122,7 @@
 #endif
 #endif
                                         /* external declarations */
-extern WidgetClass treeWidgetClass;
+externalref WidgetClass treeWidgetClass;
 
 typedef struct _TreeClassRec *TreeWidgetClass;
 typedef struct _TreeRec      *TreeWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/TreeP.h libXaw-1.0.5/include/X11/Xaw/TreeP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/TreeP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/TreeP.h	2009-01-26 11:40:05 +0100
@@ -55,6 +55,7 @@
 #define _XawTreeP_h
 
 #include <X11/Xaw/Tree.h>
+#include <X11/Intrinsic.h>
 
 typedef struct _TreeClassPart {
     XtPointer extension;
@@ -67,7 +68,7 @@
     TreeClassPart tree_class;
 } TreeClassRec;
 
-extern TreeClassRec treeClassRec;
+externalref TreeClassRec treeClassRec;
 
 typedef struct {
     /* fields available through resources */
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/Viewport.h libXaw-1.0.5/include/X11/Xaw/Viewport.h
--- libXaw-1.0.5.orig/include/X11/Xaw/Viewport.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/Viewport.h	2009-01-26 11:40:08 +0100
@@ -53,6 +53,7 @@
 
 #include <X11/Xaw/Form.h>
 #include <X11/Xaw/Reports.h>
+#include <X11/Intrinsic.h>
 
 /* Resources:
 
@@ -85,7 +86,7 @@
 #define XtNuseRight "useRight"
 #endif
 
-extern WidgetClass viewportWidgetClass;
+externalref WidgetClass viewportWidgetClass;
 
 typedef struct _ViewportClassRec *ViewportWidgetClass;
 typedef struct _ViewportRec  *ViewportWidget;
diff -ru libXaw-1.0.5.orig/include/X11/Xaw/ViewportP.h libXaw-1.0.5/include/X11/Xaw/ViewportP.h
--- libXaw-1.0.5.orig/include/X11/Xaw/ViewportP.h	2009-01-26 11:32:20 +0100
+++ libXaw-1.0.5/include/X11/Xaw/ViewportP.h	2009-01-26 11:40:18 +0100
@@ -56,6 +56,7 @@
 
 #include <X11/Xaw/Viewport.h>
 #include <X11/Xaw/FormP.h>
+#include <X11/Intrinsic.h>
 
 typedef struct {
     XtPointer extension;
@@ -69,7 +70,7 @@
     ViewportClassPart	viewport_class;
 } ViewportClassRec;
 
-extern ViewportClassRec viewportClassRec;
+externalref ViewportClassRec viewportClassRec;
 
 typedef struct _ViewportPart {
     /* resources */
