https://bugzilla.gnome.org/show_bug.cgi?id=616754

--- glib/gthread.h.old	2010-03-16 02:38:47.000000000 +0000
+++ glib/gthread.h	2010-06-08 20:01:55.000000000 +0000
@@ -359,12 +359,11 @@
  */
 extern void glib_dummy_decl (void);
 #define G_LOCK_NAME(name)               g__ ## name ## _lock
-#ifdef  G_THREADS_ENABLED
-#  define G_LOCK_DEFINE_STATIC(name)    static G_LOCK_DEFINE (name)
-#  define G_LOCK_DEFINE(name)           \
+#define G_LOCK_DEFINE(name)           \
     GStaticMutex G_LOCK_NAME (name) = G_STATIC_MUTEX_INIT
-#  define G_LOCK_EXTERN(name)           extern GStaticMutex G_LOCK_NAME (name)
-
+#define G_LOCK_DEFINE_STATIC(name)    static G_LOCK_DEFINE (name)
+#define G_LOCK_EXTERN(name)           extern GStaticMutex G_LOCK_NAME (name)
+#ifdef  G_THREADS_ENABLED
 #  ifdef G_DEBUG_LOCKS
 #    define G_LOCK(name)                G_STMT_START{             \
         g_log (G_LOG_DOMAIN, G_LOG_LEVEL_DEBUG,                   \
@@ -391,9 +390,6 @@
 #    define G_TRYLOCK(name) g_static_mutex_trylock (&G_LOCK_NAME (name))
 #  endif /* !G_DEBUG_LOCKS */
 #else   /* !G_THREADS_ENABLED */
-#  define G_LOCK_DEFINE_STATIC(name)    extern void glib_dummy_decl (void)
-#  define G_LOCK_DEFINE(name)           extern void glib_dummy_decl (void)
-#  define G_LOCK_EXTERN(name)           extern void glib_dummy_decl (void)
 #  define G_LOCK(name)
 #  define G_UNLOCK(name)
 #  define G_TRYLOCK(name)               (TRUE)
