--- pinentry-0.7.2/gtk+-2/pinentry-gtk-2.c	2007-06-15 15:53:28.000000000 +0200
+++ pinentry-0.7.2/gtk+-2/pinentry-gtk-2.c	2007-06-15 16:01:57.000000000 +0200
@@ -207,10 +207,10 @@
   if (!confirm_mode)
     {
       g_signal_connect (G_OBJECT (win),
-			pinentry->grab ? "map-event" : "focus-in-event",
+			pinentry->grab ? "expose-event" : "focus-in-event",
 			G_CALLBACK (grab_keyboard), NULL);
       g_signal_connect (G_OBJECT (win),
-			pinentry->grab ? "unmap-event" : "focus-out-event",
+			pinentry->grab ? "no-expose-event" : "focus-out-event",
 			G_CALLBACK (ungrab_keyboard), NULL);
     }
   gtk_window_add_accel_group (GTK_WINDOW (win), acc);
