--- src/xercesc/util/MsgLoaders/ICU/resources/Makefile.in.orig	2009-09-01 20:29:37.000000000 +0200
+++ src/xercesc/util/MsgLoaders/ICU/resources/Makefile.in	2009-09-01 20:30:21.000000000 +0200
@@ -50,8 +50,8 @@ PKGDATA=$(ICU_PREFIX)/bin/pkgdata
 
 # Add -v if you want to see more verbose output.
 #
-PKGDATAOPTS=-d . -M '"CC=$(CC)" "CXX=$(CXX)" "AR=$(AR)" "RANLIB=$(RANLIB)" \
-"CPPFLAGS=$(CPPFLAGS)" "CFLAGS=$(CFLAGS)" "CXXFLAGS=$(CXXFLAGS)" "LDFLAGS=$(LDFLAGS)"'
+PKGDATAOPTSSHARED=-d . -r 3.0.1
+PKGDATAOPTSSTATIC=-d .
 
 # Resource shortname
 PKGNAME=xercesc_messages_$(INTERFACE_VER_U)
@@ -78,13 +78,13 @@ all: xercesc_messages.lo
 xercesc_messages.lo: $(RESFILES)
 ifeq ($(BUILD_SHARED),yes)
 	@rm -f $(PKGNAME)_dat.o root_res.o
-	$(PKGDATA) --name $(PKGNAME) --mode dll $(PKGDATAOPTS) $(RESLIST)
+	$(PKGDATA) --name $(PKGNAME) --mode dll $(PKGDATAOPTSSHARED) $(RESLIST)
 	mv $(PKGNAME)_dat.o $(PKGNAME)_dat-pic.o
 	@if test -f root_res.o; then mv root_res.o root_res-pic.o; fi
 endif
 ifeq ($(BUILD_STATIC),yes)
 	@rm -f $(PKGNAME)_dat.o $(PKGNAME)_dat.ao root_res.o root_res.ao
-	$(PKGDATA) --name $(PKGNAME) --mode static $(PKGDATAOPTS) $(RESLIST)
+	$(PKGDATA) --name $(PKGNAME) --mode static $(PKGDATAOPTSSTATIC) $(RESLIST)
 	@if test -f $(PKGNAME)_dat.ao; then mv $(PKGNAME)_dat.ao $(PKGNAME)_dat.o; fi
 	@if test -f root_res.ao; then mv root_res.ao root_res.o; fi
 endif
