--- ./src/xercesc/util/MsgLoaders/MsgCatalog/MsgCatalogLoader.cpp.orig	2007-12-19 12:11:57.000000000 +0100
+++ ./src/xercesc/util/MsgLoaders/MsgCatalog/MsgCatalogLoader.cpp	2007-12-19 12:30:12.000000000 +0100
@@ -100,8 +100,8 @@
     * To open user-specified locale specific cat file
     * and default cat file if necessary
     */
-    if ( ((int)(fCatalogHandle=catopen(catuser, 0)) == -1) &&
-         ((int)(fCatalogHandle=catopen(catdefault, 0)) == -1)   )
+    if ( ((long)(fCatalogHandle=catopen(catuser, 0)) == -1) &&
+         ((long)(fCatalogHandle=catopen(catdefault, 0)) == -1)   )
     {
         // Probably have to call panic here
         printf("Could not open catalog:\n %s\n or %s\n", catuser, catdefault);
