Index: rails-1.0.0/lib/tasks/documentation.rake
===================================================================
--- rails-1.0.0.orig/lib/tasks/documentation.rake
+++ rails-1.0.0/lib/tasks/documentation.rake
@@ -2,7 +2,7 @@ desc "Generate documentation for the app
 Rake::RDocTask.new("appdoc") { |rdoc|
   rdoc.rdoc_dir = 'doc/app'
   rdoc.title    = "Rails Application Documentation"
-  rdoc.options << '--line-numbers --inline-source'
+  rdoc.options << '--line-numbers' << '--inline-source'
   rdoc.rdoc_files.include('doc/README_FOR_APP')
   rdoc.rdoc_files.include('app/**/*.rb')
 }
