--- lighttpd-1.4.19/src/mod_userdir.c	2008-03-10 22:45:29.000000000 +1100
+++ lighty-dev/src/mod_userdir.c	2008-03-12 17:44:43.000000000 +1100
@@ -286,6 +286,11 @@
 
 	BUFFER_APPEND_SLASH(p->temp_path);
 	buffer_append_string(p->temp_path, rel_url + 1); /* skip the / */
+
+    if (con->conf.force_lowercase_filenames) {
+        buffer_to_lower(p->temp_path);
+    }
+
 	buffer_copy_string_buffer(con->physical.path, p->temp_path);
 
 	buffer_reset(p->temp_path);
