diff -Naur octave-3.2.0/scripts/io/dlmwrite.m octave-3.2.0.new/scripts/io/dlmwrite.m
--- octave-3.2.0/scripts/io/dlmwrite.m	2009-05-25 02:04:59.000000000 -0400
+++ octave-3.2.0.new/scripts/io/dlmwrite.m	2009-09-08 22:08:40.000000000 -0400
@@ -85,7 +85,7 @@
 function dlmwrite (file, a, varargin)
 
   if (nargin < 2 || ! ischar (file))
-    ptint_usage ();
+    print_usage ();
   endif
 
   ## set defaults
