diff -ru OpenSP-1.5.2.orig/configure.in OpenSP-1.5.2/configure.in
--- OpenSP-1.5.2.orig/configure.in	2008-03-04 14:37:30 +0100
+++ OpenSP-1.5.2/configure.in	2008-03-04 14:38:26 +0100
@@ -31,7 +31,7 @@
 AH_TEMPLATE([sig_atomic_t],[define this appropriately if the type sig_atomic_t is not available])
 AH_TEMPLATE([SP_NEW_H_MISSING],[define this if your system misses new.h])
 AH_TEMPLATE([SP_SET_NEW_HANDLER_EXTERN_C],[define this if  set_new_handler() has to be declared extern "C"])
-AC_DEFINE(SP_MULTI_BYTE,[1],[define this to build a multi-byte version])
+dnl AC_DEFINE(SP_MULTI_BYTE,[1],[define this to build a multi-byte version])
 AH_TEMPLATE([SP_DEFINE_TEMPLATES],[define this to include template definitions in the headers])
 AH_TEMPLATE([SP_MANUAL_INST],[define this to compile explicit template instantiations])
 AH_TEMPLATE([SP_FANCY_NEW_HANDLER],[define this if the new handler takes size_t and returns int.])
