diff -NrU5 levee-3.5.old/configure.inc levee-3.5/configure.inc
--- levee-3.5.old/configure.inc	2008-04-12 08:43:13.000000000 +0200
+++ levee-3.5/configure.inc	2008-05-10 17:24:51.000000000 +0200
@@ -1073,11 +1073,11 @@
     else
 	PROG_INSTALL="$DEST"
     fi
 
     AC_SUB 'INSTALL' "$PROG_INSTALL"
-    AC_SUB 'INSTALL_PROGRAM' "$PROG_INSTALL -s -m 755"
+    AC_SUB 'INSTALL_PROGRAM' "$PROG_INSTALL -m 755"
     AC_SUB 'INSTALL_DATA' "$PROG_INSTALL -m 444"
 
     # finally build a little directory installer
     # if mkdir -p works, use that, otherwise use install -d,
     # otherwise build a script to do it by hand.
diff -NrU5 levee-3.5.old/editcor.c levee-3.5/editcor.c
--- levee-3.5.old/editcor.c	2008-04-12 08:43:13.000000000 +0200
+++ levee-3.5/editcor.c	2008-05-10 18:32:33.000000000 +0200
@@ -17,10 +17,11 @@
  * WARRANTIES OF MERCHANTIBILITY AND FITNESS FOR A PARTICULAR
  * PURPOSE.
  */
 #include "levee.h"
 #include "extern.h"
+#include <stdlib.h>
     
 /* do some undoable modification */
 
 /* These variables make docommand nonrecursive */
 
@@ -172,10 +173,12 @@
     return(putback(curr, &newend));
 }
 
 bool PROC
 execute(start, end)
+int start;
+int end;
 {
     int tf;
     FILE *f;
     char scratch[20];
     bool ret = FALSE;
diff -NrU5 levee-3.5.old/globals.c levee-3.5/globals.c
--- levee-3.5.old/globals.c	2008-04-12 08:43:13.000000000 +0200
+++ levee-3.5/globals.c	2008-05-10 18:14:21.000000000 +0200
@@ -250,11 +250,11 @@
     {"ignorecase","ic",	VBOOL,	0,		(void*)&ignorecase },
     {"bell",      "",	VBOOL,	0,		(void*)&bell       },
 #if OS_ATARI
     {"mapslash",  "ms", VBOOL,	0,		(void*)&mapslash   },
 #endif
-    {NULL}
+    {0, 0, 0, 0, 0}
 };
 
 /* For movement routines */
 int setstep[2] = {-1,1};
 
diff -NrU5 levee-3.5.old/main.c levee-3.5/main.c
--- levee-3.5.old/main.c	2008-04-12 08:43:13.000000000 +0200
+++ levee-3.5/main.c	2008-05-10 18:08:14.000000000 +0200
@@ -277,11 +277,11 @@
 #if OS_ATARI
 long _STKSIZ = 4096;
 long _BLKSIZ = 4096;
 #endif
 
-void /* should be union { void a; int b; float c; } to annoy the purists */
+int
 main(argc,argv)
 int argc;
 char **argv;
 {
     initialize(argc, argv);
