--- mkvtoolnix-5.6.0/rake.d/target.rb
+++ mkvtoolnix-5.6.0/rake.d/target.rb
@@ -101,7 +101,6 @@
       when :mtxoutput  then "src/output/libmtxoutput.a"
       when :avi        then "lib/avilib-0.6.10/libavi.a"
       when :rmff       then "lib/librmff/librmff.a"
-      when :pugixml    then "lib/pugixml/src/libpugixml.a"
       when :mpegparser then "src/mpegparser/libmpegparser.a"
       when :ebml       then c?("EBML_MATROSKA_INTERNAL") ? "lib/libebml/src/libebml.a"         : nil
       when :matroska   then c?("EBML_MATROSKA_INTERNAL") ? "lib/libmatroska/src/libmatroska.a" : nil
--- mkvtoolnix-5.6.0/Rakefile
+++ mkvtoolnix-5.6.0/Rakefile
@@ -48,8 +48,8 @@
   $applications            =  $programs.collect { |name| "src/#{$application_subdirs[name]}#{name}" + c(:EXEEXT) }
   $manpages                =  $programs.collect { |name| "doc/man/#{name}.1" }
 
-  $system_includes         = "-I. -Ilib -Ilib/avilib-0.6.10 -Ilib/utf8-cpp/source -Ilib/pugixml/src -Isrc"
-  $system_libdirs          = "-Llib/avilib-0.6.10 -Llib/librmff -Llib/pugixml/src -Lsrc/common -Lsrc/input -Lsrc/output -Lsrc/mpegparser"
+  $system_includes         = "-I. -Ilib -Ilib/avilib-0.6.10 -Ilib/utf8-cpp/source -Isrc"
+  $system_libdirs          = "-Llib/avilib-0.6.10 -Llib/librmff -Lsrc/common -Lsrc/input -Lsrc/output -Lsrc/mpegparser"
 
   $source_directories      =  %w{lib/avilib-0.6.10 lib/librmff src src/input src/output src/common src/common/chapters src/common/compression src/common/strings src/common/tags src/common/xml
                                  src/mmg src/mmg/header_editor src/mmg/options src/mmg/tabs src/extract src/propedit src/merge src/info src/mpegparser}
@@ -579,7 +579,6 @@
 
 [ { :name => 'avi',        :dir => 'lib/avilib-0.6.10'                                                                             },
   { :name => 'rmff',       :dir => 'lib/librmff'                                                                                   },
-  { :name => 'pugixml',    :dir => 'lib/pugixml/src'                                                                               },
   { :name => 'mpegparser', :dir => 'src/mpegparser'                                                                                },
   { :name => 'mtxcommon',  :dir => [ 'src/common' ] + %w{chapters compression strings tags xml }.collect { |e| "src/common/#{e}" } },
   { :name => 'mtxinput',   :dir => 'src/input'                                                                                     },
