--- vile-9.5/filters/sql-filt.l.orig	2006-05-16 14:35:46.000000000 -0400
+++ vile-9.5/filters/sql-filt.l	2006-05-16 14:36:01.000000000 -0400
@@ -29,7 +29,6 @@
 static char *String_attr;
 
 static int next_line;		/* state after one-line SQL*Plus */
-static int saved_state = LEAD;
 
 /******************************************************************************/
 
@@ -55,6 +54,8 @@
 
 %%
 
+static int saved_state = LEAD;
+
 <LEAD>{IDENT}		{   char *text = lowercase_of(yytext);
 			    char *attr = ci_keyword_attr(text);
 			    DPRINTF(("{%s->%s}", text, NONNULL(attr)));
