--- src/MacVim/gui_macvim.m
+++ src/MacVim/gui_macvim.m
@@ -57,12 +57,8 @@
     // - ensure this is called first to avoid above problem
     // - encoding
 
-    NSString *path = [[[NSBundle mainBundle] resourcePath]
-        stringByAppendingPathComponent:@"vim"];
-    vim_setenv((char_u*)"VIM", (char_u*)[path UTF8String]);
-
-    path = [path stringByAppendingPathComponent:@"runtime"];
-    vim_setenv((char_u*)"VIMRUNTIME", (char_u*)[path UTF8String]);
+    vim_setenv((char_u*)"VIM", (char_u*)"@GENTOO_PORTAGE_EPREFIX@/usr/share/vim");
+    vim_setenv((char_u*)"VIMRUNTIME", (char_u*)"@GENTOO_PORTAGE_EPREFIX@/usr/share/vim/vim71");
 }
 
 
