http://bugs.gentoo.org/show_bug.cgi?id=302144

--- config/gen/makefiles/root.in.orig	2010-03-20 11:41:35.000000000 +0100
+++ config/gen/makefiles/root.in	2010-03-20 11:41:55.000000000 +0100
@@ -538,7 +538,7 @@
 
 # Libraries
 LIBPARROT_STATIC    = @blib_dir@/@libparrot_static@
-#IF(darwin):export DYLD_LIBRARY_PATH := @blib_dir@:$(DYLD_LIBRARY_PATH)
+#IF(darwin):export DYLD_LIBRARY_PATH := @build_dir@/@blib_dir@:$(DYLD_LIBRARY_PATH)
 #IF(win32):LIBPARROT_SHARED  = @libparrot_shared@
 #ELSE:LIBPARROT_SHARED  = @blib_dir@/@libparrot_shared@
 
--- config/init/hints/darwin.pm
+++ config/init/hints/darwin.pm
@@ -51,14 +51,11 @@ sub runstep {
         $flagsref->{ccflags} .= ' -pipe -fno-common ';
     }
     else {
-        $flagsref->{ccflags} .= ' -pipe -fno-common -Wno-long-double ';
+        $flagsref->{ccflags} .= ' -pipe -fno-common ';
     }
 
     $flagsref->{linkflags} .= " -undefined dynamic_lookup";
 
-    _probe_for_libraries($conf, $flagsref, 'fink');
-    _probe_for_libraries($conf, $flagsref, 'macports');
-
     for my $flag ( keys %$flagsref ) {
         $flagsref->{$flag} =~ s/^\s+//;
     }
@@ -89,7 +92,7 @@ sub runstep {
         libparrot_shared_alias => "libparrot$share_ext",
         rpath                  => "-L",
         libparrot_soname       => "-install_name "
-            . $lib_dir
+            . $conf->data->get('libdir')
             . $conf->data->get('slash')
             . "libparrot"
             . $conf->data->get('share_ext')
