--- config/init/hints/linux.pm.old	2006-05-15 20:45:00.000000000 +0000
+++ config/init/hints/linux.pm	2006-05-15 20:45:21.000000000 +0000
@@ -63,9 +63,9 @@
 
         has_dynamic_linking     => 1,
         parrot_is_shared        => 1,
-        libparrot_shared        => 'libparrot$(SHARE_EXT).$(SOVERSION)',
-        libparrot_shared_alias  => 'libparrot$(SHARE_EXT)',
-        libparrot_soname        => '-Wl,-soname=libparrot$(SHARE_EXT).$(SOVERSION)',
+        libparrot_shared        => 'libparrot$(SHARE_EXT)',
+        libparrot_shared_alias  => 'libparrot$(SHARE_EXT).$(SOVERSION)',
+        libparrot_soname        => '-Wl,-soname=libparrot$(SHARE_EXT)',
     );
 
     if ((split('-', $Config{archname}))[0] eq 'ia64') {
