--- xmldocs.make	2007-01-06 23:39:36.000000000 +0100
+++ xmldocs.make	2008-02-12 12:45:21.000000000 +0100
@@ -36,11 +36,7 @@
 # **********  Begin of section some packagers may need to modify  **********
 # This variable (docdir) specifies where the documents should be installed.
 # This default value should work for most packages.
-if HAVE_GNOME
 docdir = $(datadir)/gnome/help/$(docname)/$(lang)
-else
-docdir = $(datadir)/help/$(docname)/$(lang)
-endif
 
 # **********  You should not have to edit below this line  **********
 xml_files = $(entities) $(docname).xml
--- doc/Makefile.am	2008-02-12 12:51:13.000000000 +0100
+++ doc/Makefile.am	2008-02-12 12:45:59.000000000 +0100
@@ -11,6 +11,7 @@
 helpdir = $(datadir)/gnome/help/dia
 else
 helpdir = $(pkgdatadir)/help
+docdir = $(datadir)/help/$(docname)/$(lang)
 endif
 
 # a simple rule to symlink en to C
--- doc/en/Makefile.am	2008-02-12 12:51:13.000000000 +0100
+++ doc/en/Makefile.am	2008-02-12 12:47:12.000000000 +0100
@@ -5,6 +5,11 @@
 entities=
 
 include $(top_srcdir)/xmldocs.make
+
+if ! HAVE_GNOME
+docdir = $(datadir)/help/$(docname)/$(lang)
+endif
+
 dist-hook: app-dist-hook
 
 nodist_include_EXTRAS = dia.1
--- doc/eu/Makefile.am	2008-02-12 12:51:13.000000000 +0100
+++ doc/eu/Makefile.am	2008-02-12 12:47:23.000000000 +0100
@@ -5,6 +5,11 @@
 entities=
 
 include $(top_srcdir)/xmldocs.make
+
+if ! HAVE_GNOME
+docdir = $(datadir)/help/$(docname)/$(lang)
+endif
+
 dist-hook: app-dist-hook
 
 EXTRA_DIST += \
--- doc/pl/Makefile.am	2008-02-12 12:51:13.000000000 +0100
+++ doc/pl/Makefile.am	2008-02-12 12:47:33.000000000 +0100
@@ -5,6 +5,11 @@
 entities=
 
 include $(top_srcdir)/xmldocs.make
+
+if ! HAVE_GNOME
+docdir = $(datadir)/help/$(docname)/$(lang)
+endif
+
 dist-hook: app-dist-hook
 
 EXTRA_DIST += \
--- configure.in	2008-02-12 12:13:12.000000000 +0100
+++ configure.in	2008-02-12 12:40:16.000000000 +0100
@@ -111,17 +111,19 @@
 fi
 
 AC_MSG_CHECKING([whether GNOME specific code should be used])
-AC_ARG_ENABLE(gnome,[  --enable-gnome          enable gnome code],
-              GNOME=$enableval, GNOME=no)
-if test "$GNOME" = "yes" ; then
+AC_ARG_ENABLE(gnome,
+	      AC_HELP_STRING([--enable-gnome],[enable gnome code]),,
+	      enable_gnome=no)
+if test "x$enable_gnome" = "xyes" ; then
     GTK_MODULES="$GTK_MODULES libgnome-2.0 libgnomeui-2.0"
     AC_DEFINE(GNOME,1,[Define if building with GNOME support])
+    AC_DEFINE(HAVE_GNOME,1,[Define if building with GNOME support])
     AC_DEFINE_UNQUOTED(GNOME_ICONDIR, "${prefix}/share/pixmaps", [GNOME icon directory])
-    have_gnome=true
+    have_gnome=yes
 else
-    have_gnome=false
+    have_gnome=no
 fi
-AM_CONDITIONAL(HAVE_GNOME, $have_gnome)
+AM_CONDITIONAL(HAVE_GNOME, [test "x$have_gnome" != "xno"])
 AC_MSG_RESULT($GNOME)
 
 dnl GTK2:     libgnomecanvas-2.0 ?
@@ -130,7 +132,8 @@
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
-AM_CONDITIONAL(HAVE_BONOBO,false)
+#AM_CONDITIONAL(HAVE_BONOBO,false)
+AM_CONDITIONAL(HAVE_BONOBO,[test "true" = "false"])
 
 dnl Look for Gnome Print
 AC_ARG_WITH(gnomeprint,
@@ -534,7 +537,7 @@
 	Source code location:	${srcdir}
 	Compiler:		${CC}
 
-	Gnome support:				${GNOME}
+	Gnome support:				${have_gnome}
 	Python support:				${with_python}
 	Libart support (PNG export):		${have_libart}
 	Cairo support (experimental):		${with_cairo}
