applied in 0.2.1+

--- libq/hash_fd.c	2009-03-27 22:42:25 +0200
+++ libq/hash_fd.c	2009-09-25 15:21:29 +0300
@@ -23,7 +23,10 @@
 #include <fcntl.h>
 #include <limits.h>
 #include <stdio.h>
+#ifndef __INTERIX
 #include <inttypes.h>
+#endif
+
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
--- libq/md5_sha1_sum.c	2009-03-27 22:42:25 +0200
+++ libq/md5_sha1_sum.c	2009-09-25 15:21:44 +0300
@@ -20,7 +20,9 @@
 #include <fcntl.h>
 #include <limits.h>
 #include <stdio.h>
+#ifndef __INTERIX
 #include <inttypes.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
--- tests/atom_explode/test.c	2009-03-27 22:42:25 +0200
+++ tests/atom_explode/test.c	2009-09-25 15:22:01 +0300
@@ -11,7 +11,9 @@
 
 #include <stdio.h>
 #include <stdbool.h>
+#ifndef __INTERIX
 #include <inttypes.h>
+#endif
 #include <stdlib.h>
 #include <sys/types.h>
 #include <string.h>
