Committed in 0.2+

--- qsize.c
+++ qsize.c
@@ -40,6 +40,11 @@
 # define S_BLKSIZE DK_DEVID_BLKSIZE
 #endif
 
+#ifdef _AIX
+# include <sys/stat.h>
+# define S_BLKSIZE DEV_BSIZE
+#endif
+
 int qsize_main(int argc, char **argv)
 {
 	DIR *dir, *dirp;
--- main.c
+++ main.c
@@ -8,6 +8,9 @@
  */
 
 #define _GNU_SOURCE
+#ifdef _AIX
+#define _LINUX_SOURCE_COMPAT
+#endif
 
 #include <stdio.h>
 #include <stdlib.h>
