--- qmerge.c.orig	2007-06-06 22:11:53 +0200
+++ qmerge.c	2007-06-06 22:10:14 +0200
@@ -11,6 +11,7 @@
 
 #include <fnmatch.h>
 #include <glob.h>
+#include <unistd.h>
 
 #ifndef GLOB_BRACE
 # define GLOB_BRACE     (1 << 10)	/* Expand "{a,b}" to "a" "b".  */
@@ -952,7 +953,7 @@
 
 	if (pkg->MD5[0]) {
 		if ((hash = (char*) hash_file(fname, HASH_MD5)) == NULL) {
-			errf("hash is NULL for %s %s", fname, get_current_dir_name());
+			errf("hash is NULL for %s %s", fname, getwd(NULL));
 		}
 		if (strcmp(hash, pkg->MD5) == 0) {
 			if (display)
