--- qlop.c
+++ qlop.c
@@ -21,6 +21,17 @@
 # include <sys/time.h>
 #endif
 
+#ifdef __MACH__
+# include <fcntl.h>
+# include <kvm.h>
+# include <sys/sysctl.h>
+/* Darwin doesn't have KERN_PROC_PROC, KERN_PROC_ALL comes closest */
+# define KERN_PROC_PROC KERN_PROC_ALL
+/* Fugly FreeBSD immitation */
+# define ki_comm kp_proc.p_comm
+# define ki_start kp_proc.p_starttime
+#endif
+
 #define QLOP_DEFAULT_LOGFILE "/var/log/emerge.log"
 
 #define QLOP_FLAGS "gtHluscf:" COMMON_FLAGS
@@ -369,7 +380,7 @@
 	if (start_time == 0 && verbose)
 		puts("No emerge processes located");
 }
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD__) || defined(__MACH__)
 void show_current_emerge(void)
 {
 	kvm_t *kd = NULL;
@@ -378,7 +389,7 @@
 	char *p, *q;
 	time_t start_date = 0;
 
-	if (! (kd = kvm_open("/dev/null", "/dev/null", "/dev/null", O_RDONLY, "kvm_open"))) {
+	if (! (kd = kvm_open(NULL, NULL, NULL, O_RDONLY, "kvm_open"))) {
 		warnp("Could not open kvm: %s", kvm_geterr(kd));
 		return;
 	}
