Prefix patch as commited in 0.2+

Index: Makefile
===================================================================
RCS file: /var/cvsroot/gentoo-projects/portage-utils/Makefile,v
retrieving revision 1.63
retrieving revision 1.64
diff -u -r1.63 -r1.64
--- Makefile	12 Apr 2008 17:25:59 -0000	1.63
+++ Makefile	24 Mar 2009 20:53:24 -0000	1.64
@@ -21,11 +21,14 @@
 #LDFLAGS  := -pie
 LIBADD    += $(shell echo | $(CC) -dM -E - | grep -q ' __FreeBSD__' && echo '-lkvm')
 DESTDIR   :=
-PREFIX    := $(DESTDIR)/usr
+EPREFIX   := 
+PREFIX    := $(DESTDIR)$(EPREFIX)/usr
 STRIP     := strip
 MKDIR     := mkdir -p
 CP        := cp
 
+HFLAGS    += -DEPREFIX=\"$(EPREFIX)\"
+
 ifdef PV
 HFLAGS    += -DVERSION=\"$(PV)\"
 else
@@ -51,7 +54,7 @@
 
 debug:
 	$(MAKE) CFLAGS="$(CFLAGS) -DEBUG -g3 -ggdb -fno-pie" clean symlinks
-	@-/usr/bin/scanelf -o /dev/null -BXxz permsx q
+	@-$(EPREFIX)/usr/bin/scanelf -o /dev/null -BXxz permsx q
 
 q: $(SRC) libq/*.c *.h libq/*.h
 ifeq ($(subst s,,$(MAKEFLAGS)),$(MAKEFLAGS))
Index: main.c
===================================================================
RCS file: /var/cvsroot/gentoo-projects/portage-utils/main.c,v
retrieving revision 1.161
retrieving revision 1.162
diff -u -r1.161 -r1.162
--- main.c	24 Mar 2009 18:53:33 -0000	1.161
+++ main.c	24 Mar 2009 20:53:24 -0000	1.162
@@ -56,15 +56,15 @@
 char pretend = 0;
 char reinitialize = 0;
 char reinitialize_metacache = 0;
-char portdir[_Q_PATH_MAX] = "/usr/portage";
-char portarch[20] = "";
+char portdir[_Q_PATH_MAX] = EPREFIX "/usr/portage";
+char portarch[20] = ""; 
 char portvdb[_Q_PATH_MAX] = "var/db/pkg";
 char portcachedir[] = "metadata/cache";
 char portroot[_Q_PATH_MAX] = "/";
-char config_protect[_Q_PATH_MAX] = "/etc/";
+char config_protect[_Q_PATH_MAX] = EPREFIX "/etc/";
 
-char pkgdir[512] = "/usr/portage/packages/";
-char port_tmpdir[512] = "/var/tmp/portage/";
+char pkgdir[512] = EPREFIX "/usr/portage/packages/";
+char port_tmpdir[512] = EPREFIX "/var/tmp/portage/";
 
 char binhost[1024] = PORTAGE_BINHOST;
 char features[2048] = "noman noinfo nodoc";
@@ -475,9 +475,10 @@
 	struct stat st;
 	FILE *fp;
 	char buf[BUFSIZE], *s, *p;
+	char *e = EPREFIX;
 
 	char profile[_Q_PATH_MAX], portage_file[_Q_PATH_MAX];
-	const char *files[] = {portage_file, "/etc/make.globals", "/etc/make.conf"};
+	const char *files[] = {portage_file, EPREFIX "/etc/make.globals", EPREFIX "/etc/make.conf"};
 	typedef enum { _Q_BOOL, _Q_STR, _Q_ISTR } var_types;
 	struct {
 		const char *name;
@@ -503,6 +504,10 @@
 	if (s) {
 		strncpy(portvdb, s, sizeof(portvdb));
 		portvdb[sizeof(portvdb) - 1] = '\0';
+	} else if ((i = strlen(e)) > 1) {
+		memmove(portvdb + i, portvdb, strlen(portvdb));
+		memcpy(portvdb, e + 1, i - 1);
+		portvdb[i - 1] = '/';
 	}
 
 	if ((p = strchr(portroot, '/')) != NULL)
@@ -510,15 +515,15 @@
 			strncat(portroot, "/", sizeof(portroot));
 
 	f = 0;
-	profilelen = readlink("/etc/make.profile", profile, sizeof(profile) - 1);
+	profilelen = readlink(EPREFIX "/etc/make.profile", profile, sizeof(profile) - 1);
 	if (profilelen == -1)
-		strcpy(profile, "/etc/make.profile");
+		strcpy(profile, EPREFIX "/etc/make.profile");
 	else
 		profile[profilelen]='\0';
 
 	if (profile[0] != '/') {
-		memmove(profile+5, profile, strlen(profile)+1);
-		memcpy(profile, "/etc/", 5);
+		memmove(profile+5+strlen(EPREFIX), profile, strlen(profile)+1);
+		memcpy(profile, EPREFIX "/etc/", strlen(EPREFIX) + 5);
 	}
 	do {
 		if (f == 0)
@@ -1057,7 +1062,7 @@
 
 #ifdef ENABLE_NLS	/* never tested */
 	setlocale(LC_ALL, "");
-	bindtextdomain(argv0, "/usr/share/locale");
+	bindtextdomain(argv0, EPREFIX "/usr/share/locale");
 	textdomain(argv0);
 #endif
 #if 1
Index: qlop.c
===================================================================
RCS file: /var/cvsroot/gentoo-projects/portage-utils/qlop.c,v
retrieving revision 1.44
retrieving revision 1.45
diff -u -r1.44 -r1.45
--- qlop.c	24 Mar 2009 18:52:09 -0000	1.44
+++ qlop.c	24 Mar 2009 20:53:24 -0000	1.45
@@ -27,7 +27,7 @@
 # include <sys/sysctl.h>
 #endif
 
-#define QLOP_DEFAULT_LOGFILE "/var/log/emerge.log"
+#define QLOP_DEFAULT_LOGFILE EPREFIX "/var/log/emerge.log"
 
 #define QLOP_FLAGS "gtHluscf:" COMMON_FLAGS
 static struct option const qlop_long_opts[] = {
@@ -554,7 +554,7 @@
 #else
 void show_current_emerge(void)
 {
-	errf("not supported on your crapbox OS");
+	errf("not supported on your OS");
 }
 #endif
 
