--- qsize.c.orig	2008-07-25 11:24:43 +0200
+++ qsize.c	2008-07-25 11:24:45 +0200
@@ -39,6 +39,10 @@
 #include <sys/dklabel.h>
 # define S_BLKSIZE DK_DEVID_BLKSIZE
 #endif
+#ifdef _AIX
+# include <sys/stat.h>
+# define S_BLKSIZE DEV_BSIZE
+#endif
 
 int qsize_main(int argc, char **argv)
 {
--- main.c.orig	2008-07-25 14:04:25 +0200
+++ main.c	2008-07-25 14:07:59 +0200
@@ -8,6 +8,9 @@
  */
 
 #define _GNU_SOURCE
+#ifdef _AIX
+#define _LINUX_SOURCE_COMPAT
+#endif
 
 #include <stdio.h>
 #include <stdlib.h>
