diff -ur portage-utils-0.1.28.orig/libq/xasprintf.c portage-utils-0.1.28/libq/xasprintf.c
--- portage-utils-0.1.28.orig/libq/xasprintf.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/libq/xasprintf.c	2007-06-07 22:02:16.108215295 +0200
@@ -22,8 +22,20 @@
 #include <stdio.h>
 
 /* asprintf(char **strp, const char *fmt, ...); */
+#ifndef __sun__
 #define xasprintf(strp, fmt, args...) \
 	do { \
 		if (asprintf(strp, fmt , ## args) == -1) \
 			err("Out of memory"); \
 	} while (0)
+#else
+#include <string.h>
+#define xasprintf(strp, fmt, args...) \
+	do { \
+		char str[8096 + 1]; \
+		if ((snprintf(str, 8096, fmt , ## args)) == -1) \
+			err("Out of memory"); \
+		str[8096] = '\0'; \
+		*strp = strdup(str); \
+	} while (0)
+#endif
diff -ur portage-utils-0.1.28.orig/main.c portage-utils-0.1.28/main.c
--- portage-utils-0.1.28.orig/main.c	2007-06-06 22:17:00.969709000 +0200
+++ portage-utils-0.1.28/main.c	2007-06-07 21:46:33.166045926 +0200
@@ -97,15 +97,23 @@
 /* include common applet defs */
 #include "applets.h"
 
+/* unfortunately Solaris defines the name field in the option struct as
+ * char* not as const char* */
+#ifdef __sun__
+# define QOPS (char*)
+#else
+# define QOPS
+#endif
+
 /* Common usage for all applets */
 #define COMMON_FLAGS "vqChV"
 #define COMMON_LONG_OPTS \
-	{"verbose",   no_argument, NULL, 'v'}, \
-	{"quiet",     no_argument, NULL, 'q'}, \
-	{"nocolor",   no_argument, NULL, 'C'}, \
-	{"help",      no_argument, NULL, 'h'}, \
-	{"version",   no_argument, NULL, 'V'}, \
-	{NULL,        no_argument, NULL, 0x0}
+	{QOPS"verbose",   no_argument, NULL, 'v'}, \
+	{QOPS"quiet",     no_argument, NULL, 'q'}, \
+	{QOPS"nocolor",   no_argument, NULL, 'C'}, \
+	{QOPS"help",      no_argument, NULL, 'h'}, \
+	{QOPS"version",   no_argument, NULL, 'V'}, \
+	{    NULL,        no_argument, NULL, 0x0}
 #define COMMON_OPTS_HELP \
 	"Make a lot of noise", \
 	"Tighter output; suppress warnings", \
@@ -201,15 +209,15 @@
 	return ret;
 }
 
-int rematch(const char *regex, const char *match, int cflags)
+int rematch(const char *qregex, const char *match, int cflags)
 {
 	regex_t preg;
 	int ret;
 
-	if ((match == NULL) || (regex == NULL))
+	if ((match == NULL) || (qregex == NULL))
 		return EXIT_FAILURE;
 
-	if ((ret = regcomp(&preg, regex, cflags))) {
+	if ((ret = regcomp(&preg, qregex, cflags))) {
 		char err[256];
 		if (regerror(ret, &preg, err, sizeof(err)))
 			warnf("regcomp failed: %s", err);
diff -ur portage-utils-0.1.28.orig/q.c portage-utils-0.1.28/q.c
--- portage-utils-0.1.28.orig/q.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/q.c	2007-06-07 21:43:59.074920606 +0200
@@ -9,10 +9,10 @@
 
 #define Q_FLAGS "irmM:" COMMON_FLAGS
 static struct option const q_long_opts[] = {
-	{"install",      no_argument, NULL, 'i'},
-	{"reinitialize", no_argument, NULL, 'r'},
-	{"metacache",    no_argument, NULL, 'm'},
-	{"modpath",       a_argument, NULL, 'M'},
+	{QOPS"install",      no_argument, NULL, 'i'},
+	{QOPS"reinitialize", no_argument, NULL, 'r'},
+	{QOPS"metacache",    no_argument, NULL, 'm'},
+	{QOPS"modpath",       a_argument, NULL, 'M'},
 	COMMON_LONG_OPTS
 };
 static const char *q_opts_help[] = {
diff -ur portage-utils-0.1.28.orig/qatom.c portage-utils-0.1.28/qatom.c
--- portage-utils-0.1.28.orig/qatom.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qatom.c	2007-06-07 22:34:49.293884744 +0200
@@ -11,7 +11,7 @@
 
 #define QATOM_FLAGS "c" COMMON_FLAGS
 static struct option const qatom_long_opts[] = {
-	{"compare",   no_argument, NULL, 'c'},
+	{QOPS"compare",   no_argument, NULL, 'c'},
 	COMMON_LONG_OPTS
 };
 static const char *qatom_opts_help[] = {
diff -ur portage-utils-0.1.28.orig/qcache.c portage-utils-0.1.28/qcache.c
--- portage-utils-0.1.28.orig/qcache.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qcache.c	2007-06-07 22:56:44.973453171 +0200
@@ -13,7 +13,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifndef __sun__
 #include <sys/dir.h>
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <time.h>
@@ -25,14 +27,14 @@
 
 #define QCACHE_FLAGS "p:c:idtans" COMMON_FLAGS
 static struct option const qcache_long_opts[] = {
-	{"matchpkg", a_argument, NULL, 'p'},
-	{"matchcat", a_argument, NULL, 'c'},
-	{"imlate",  no_argument, NULL, 'i'},
-	{"dropped", no_argument, NULL, 'd'},
-	{"testing", no_argument, NULL, 't'},
-	{"stats",   no_argument, NULL, 's'},
-	{"all",     no_argument, NULL, 'a'},
-	{"not",     no_argument, NULL, 'n'},
+	{QOPS"matchpkg", a_argument, NULL, 'p'},
+	{QOPS"matchcat", a_argument, NULL, 'c'},
+	{QOPS"imlate",  no_argument, NULL, 'i'},
+	{QOPS"dropped", no_argument, NULL, 'd'},
+	{QOPS"testing", no_argument, NULL, 't'},
+	{QOPS"stats",   no_argument, NULL, 's'},
+	{QOPS"all",     no_argument, NULL, 'a'},
+	{QOPS"not",     no_argument, NULL, 'n'},
 	COMMON_LONG_OPTS
 };
 
@@ -441,10 +443,10 @@
  *   1 (OLDER)
  *   0 (SAME)
  */
-int qcache_vercmp(const void *x, const void *y);
-int qcache_vercmp(const void *x, const void *y)
+int qcache_vercmp(const struct dirent **x, const struct dirent **y);
+int qcache_vercmp(const struct dirent **x, const struct dirent **y)
 {
-	switch (atom_compare_str((*((const struct dirent **)x))->d_name, (*((const struct dirent **)y))->d_name)) {
+	switch (atom_compare_str((*x)->d_name, (*y)->d_name)) {
 		case NEWER: return -1;
 		case OLDER: return  1;
 		default:    return  0;
@@ -510,7 +512,7 @@
 	qcache_data data;
 	char *catpath, *pkgpath, *ebuildpath, *cachepath;
 	int i, j, k, len, num_cat, num_pkg, num_ebuild;
-	struct direct **categories, **packages, **ebuilds;
+	struct dirent **categories, **packages, **ebuilds;
 
 	xasprintf(&catpath, "%s%s", QCACHE_EDB, portdir);
 
@@ -762,7 +764,7 @@
 	int i;
 
 	if (!numpkg) {
-		struct direct **categories;
+		struct dirent **categories;
 		char *catpath;
 		int len;
 
diff -ur portage-utils-0.1.28.orig/qcheck.c portage-utils-0.1.28/qcheck.c
--- portage-utils-0.1.28.orig/qcheck.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qcheck.c	2007-06-07 22:05:25.562591599 +0200
@@ -11,12 +11,12 @@
 
 #define QCHECK_FLAGS "eauAHT" COMMON_FLAGS
 static struct option const qcheck_long_opts[] = {
-	{"exact",   no_argument, NULL, 'e'},
-	{"all",     no_argument, NULL, 'a'},
-	{"update",  no_argument, NULL, 'u'},
-	{"noafk",   no_argument, NULL, 'A'},
-	{"nohash",  no_argument, NULL, 'H'},
-	{"nomtime", no_argument, NULL, 'T'},
+	{QOPS"exact",   no_argument, NULL, 'e'},
+	{QOPS"all",     no_argument, NULL, 'a'},
+	{QOPS"update",  no_argument, NULL, 'u'},
+	{QOPS"noafk",   no_argument, NULL, 'A'},
+	{QOPS"nohash",  no_argument, NULL, 'H'},
+	{QOPS"nomtime", no_argument, NULL, 'T'},
 	COMMON_LONG_OPTS
 };
 static const char *qcheck_opts_help[] = {
@@ -184,7 +184,7 @@
 							if (!verbose)
 								continue;
 						}
-						printf(" %sPERM %4o%s: %s\n", RED, (st.st_mode & 07777), NORM, e->name);
+						printf(" %sPERM %4o%s: %s\n", RED, (unsigned int)(st.st_mode & 07777), NORM, e->name);
 						continue;
 					} else if (strcmp(e->digest, hashed_file)) {
 						if (chk_hash) {
diff -ur portage-utils-0.1.28.orig/qdepends.c portage-utils-0.1.28/qdepends.c
--- portage-utils-0.1.28.orig/qdepends.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qdepends.c	2007-06-07 22:06:08.308791885 +0200
@@ -11,13 +11,13 @@
 
 #define QDEPENDS_FLAGS "drpaNk:Q:" COMMON_FLAGS
 static struct option const qdepends_long_opts[] = {
-	{"depend",    no_argument, NULL, 'd'},
-	{"rdepend",   no_argument, NULL, 'r'},
-	{"pdepend",   no_argument, NULL, 'p'},
-	{"key",        a_argument, NULL, 'k'},
-	{"query",      a_argument, NULL, 'Q'},
-	{"name-only", no_argument, NULL, 'N'},
-	{"all",       no_argument, NULL, 'a'},
+	{QOPS"depend",    no_argument, NULL, 'd'},
+	{QOPS"rdepend",   no_argument, NULL, 'r'},
+	{QOPS"pdepend",   no_argument, NULL, 'p'},
+	{QOPS"key",        a_argument, NULL, 'k'},
+	{QOPS"query",      a_argument, NULL, 'Q'},
+	{QOPS"name-only", no_argument, NULL, 'N'},
+	{QOPS"all",       no_argument, NULL, 'a'},
 	COMMON_LONG_OPTS
 };
 static const char *qdepends_opts_help[] = {
diff -ur portage-utils-0.1.28.orig/qfile.c portage-utils-0.1.28/qfile.c
--- portage-utils-0.1.28.orig/qfile.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qfile.c	2007-06-07 22:06:34.978532132 +0200
@@ -15,12 +15,12 @@
 
 #define QFILE_FLAGS "ef:m:oRx:" COMMON_FLAGS
 static struct option const qfile_long_opts[] = {
-	{"exact",       no_argument, NULL, 'e'},
-	{"from",        a_argument,  NULL, 'f'},
-	{"max-args",    a_argument,  NULL, 'm'},
-	{"orphans",     no_argument, NULL, 'o'},
-	{"root-prefix", no_argument, NULL, 'R'},
-	{"exclude",     a_argument,  NULL, 'x'},
+	{QOPS"exact",       no_argument, NULL, 'e'},
+	{QOPS"from",        a_argument,  NULL, 'f'},
+	{QOPS"max-args",    a_argument,  NULL, 'm'},
+	{QOPS"orphans",     no_argument, NULL, 'o'},
+	{QOPS"root-prefix", no_argument, NULL, 'R'},
+	{QOPS"exclude",     a_argument,  NULL, 'x'},
 	COMMON_LONG_OPTS
 };
 static const char *qfile_opts_help[] = {
diff -ur portage-utils-0.1.28.orig/qgrep.c portage-utils-0.1.28/qgrep.c
--- portage-utils-0.1.28.orig/qgrep.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qgrep.c	2007-06-07 22:34:33.645913151 +0200
@@ -12,21 +12,21 @@
 
 #define QGREP_FLAGS "IiHNclLexJEsS:B:A:" COMMON_FLAGS
 static struct option const qgrep_long_opts[] = {
-	{"invert-match",  no_argument, NULL, 'I'},
-	{"ignore-case",   no_argument, NULL, 'i'},
-	{"with-filename", no_argument, NULL, 'H'},
-	{"with-name",     no_argument, NULL, 'N'},
-	{"count",         no_argument, NULL, 'c'},
-	{"list",          no_argument, NULL, 'l'},
-	{"invert-list",   no_argument, NULL, 'L'},
-	{"regexp",        no_argument, NULL, 'e'},
-	{"extended",      no_argument, NULL, 'x'},
-	{"installed",     no_argument, NULL, 'J'},
-	{"eclass",        no_argument, NULL, 'E'},
-	{"skip-comments", no_argument, NULL, 's'},
-	{"skip",           a_argument, NULL, 'S'},
-	{"before",         a_argument, NULL, 'B'},
-	{"after",          a_argument, NULL, 'A'},
+	{QOPS"invert-match",  no_argument, NULL, 'I'},
+	{QOPS"ignore-case",   no_argument, NULL, 'i'},
+	{QOPS"with-filename", no_argument, NULL, 'H'},
+	{QOPS"with-name",     no_argument, NULL, 'N'},
+	{QOPS"count",         no_argument, NULL, 'c'},
+	{QOPS"list",          no_argument, NULL, 'l'},
+	{QOPS"invert-list",   no_argument, NULL, 'L'},
+	{QOPS"regexp",        no_argument, NULL, 'e'},
+	{QOPS"extended",      no_argument, NULL, 'x'},
+	{QOPS"installed",     no_argument, NULL, 'J'},
+	{QOPS"eclass",        no_argument, NULL, 'E'},
+	{QOPS"skip-comments", no_argument, NULL, 's'},
+	{QOPS"skip",           a_argument, NULL, 'S'},
+	{QOPS"before",         a_argument, NULL, 'B'},
+	{QOPS"after",          a_argument, NULL, 'A'},
 	COMMON_LONG_OPTS
 };
 static const char *qgrep_opts_help[] = {
@@ -233,6 +233,22 @@
 	goto get_next_ebuild_from_category;
 }
 
+#ifdef __sun__
+#include <ctype.h>
+#include <alloca.h>
+/* strcasestr is a GNU extention, and should actually not be used... */
+static char* strcasestr(const char *big, const char *little) {
+	char* b = alloca((strlen(big) + 1) * sizeof(char));
+	char* l = alloca((strlen(little) + 1) * sizeof(char));
+	char* off;
+	size_t i;
+	for (i = 0; big[i]; i++) b[i] = (char)tolower(big[i]);
+	for (i = 0; little[i]; i++) l[i] = (char)tolower(little[i]);
+	off = strstr(b, l);
+	return(off == NULL ? off : (char*)(big + (off - b)));
+}
+#endif
+
 int qgrep_main(int argc, char **argv)
 {
 	int i;
diff -ur portage-utils-0.1.28.orig/qlist.c portage-utils-0.1.28/qlist.c
--- portage-utils-0.1.28.orig/qlist.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qlist.c	2007-06-07 22:06:51.658341852 +0200
@@ -12,16 +12,16 @@
 
 #define QLIST_FLAGS "ISUDeados" COMMON_FLAGS
 static struct option const qlist_long_opts[] = {
-	{"installed", no_argument, NULL, 'I'},
-	{"slots",     no_argument, NULL, 'S'},
-	{"umap",      no_argument, NULL, 'U'},
-	{"dups",      no_argument, NULL, 'D'},
-	{"exact",     no_argument, NULL, 'e'},
-	{"all",       no_argument, NULL, 'a'},
-	{"dir",       no_argument, NULL, 'd'},
-	{"obj",       no_argument, NULL, 'o'},
-	{"sym",       no_argument, NULL, 's'},
-	/* {"file",       a_argument, NULL, 'f'}, */
+	{QOPS"installed", no_argument, NULL, 'I'},
+	{QOPS"slots",     no_argument, NULL, 'S'},
+	{QOPS"umap",      no_argument, NULL, 'U'},
+	{QOPS"dups",      no_argument, NULL, 'D'},
+	{QOPS"exact",     no_argument, NULL, 'e'},
+	{QOPS"all",       no_argument, NULL, 'a'},
+	{QOPS"dir",       no_argument, NULL, 'd'},
+	{QOPS"obj",       no_argument, NULL, 'o'},
+	{QOPS"sym",       no_argument, NULL, 's'},
+	/* {QOPS"file",       a_argument, NULL, 'f'}, */
 	COMMON_LONG_OPTS
 };
 static const char *qlist_opts_help[] = {
diff -ur portage-utils-0.1.28.orig/qlop.c portage-utils-0.1.28/qlop.c
--- portage-utils-0.1.28.orig/qlop.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qlop.c	2007-06-07 22:07:18.643008184 +0200
@@ -25,14 +25,14 @@
 
 #define QLOP_FLAGS "gtHluscf:" COMMON_FLAGS
 static struct option const qlop_long_opts[] = {
-	{"gauge",     no_argument, NULL, 'g'},
-	{"time",      no_argument, NULL, 't'},
-	{"human",     no_argument, NULL, 'H'},
-	{"list",      no_argument, NULL, 'l'},
-	{"unlist",    no_argument, NULL, 'u'},
-	{"sync",      no_argument, NULL, 's'},
-	{"current",   no_argument, NULL, 'c'},
-	{"logfile",    a_argument, NULL, 'f'},
+	{QOPS"gauge",     no_argument, NULL, 'g'},
+	{QOPS"time",      no_argument, NULL, 't'},
+	{QOPS"human",     no_argument, NULL, 'H'},
+	{QOPS"list",      no_argument, NULL, 'l'},
+	{QOPS"unlist",    no_argument, NULL, 'u'},
+	{QOPS"sync",      no_argument, NULL, 's'},
+	{QOPS"current",   no_argument, NULL, 'c'},
+	{QOPS"logfile",    a_argument, NULL, 'f'},
 	COMMON_LONG_OPTS
 };
 static const char *qlop_opts_help[] = {
diff -ur portage-utils-0.1.28.orig/qmerge.c portage-utils-0.1.28/qmerge.c
--- portage-utils-0.1.28.orig/qmerge.c	2007-06-06 22:17:00.819094000 +0200
+++ portage-utils-0.1.28/qmerge.c	2007-06-07 22:35:07.374050394 +0200
@@ -27,16 +27,16 @@
 
 #define QMERGE_FLAGS "fFsKUpuyO5" COMMON_FLAGS
 static struct option const qmerge_long_opts[] = {
-	{"fetch",   no_argument, NULL, 'f'},
-	{"force",   no_argument, NULL, 'F'},
-	{"search",  no_argument, NULL, 's'},
-	{"install", no_argument, NULL, 'K'},
-	{"unmerge", no_argument, NULL, 'U'},
-	{"pretend", no_argument, NULL, 'p'},
-	{"update",  no_argument, NULL, 'u'},
-	{"yes",     no_argument, NULL, 'y'},
-	{"nodeps",  no_argument, NULL, 'O'},
-	{"nomd5",   no_argument, NULL, '5'},
+	{QOPS"fetch",   no_argument, NULL, 'f'},
+	{QOPS"force",   no_argument, NULL, 'F'},
+	{QOPS"search",  no_argument, NULL, 's'},
+	{QOPS"install", no_argument, NULL, 'K'},
+	{QOPS"unmerge", no_argument, NULL, 'U'},
+	{QOPS"pretend", no_argument, NULL, 'p'},
+	{QOPS"update",  no_argument, NULL, 'u'},
+	{QOPS"yes",     no_argument, NULL, 'y'},
+	{QOPS"nodeps",  no_argument, NULL, 'O'},
+	{QOPS"nomd5",   no_argument, NULL, '5'},
 	COMMON_LONG_OPTS
 };
 
diff -ur portage-utils-0.1.28.orig/qpkg.c portage-utils-0.1.28/qpkg.c
--- portage-utils-0.1.28.orig/qpkg.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qpkg.c	2007-06-07 22:19:31.915897231 +0200
@@ -11,9 +11,9 @@
 
 #define QPKG_FLAGS "cpP:" COMMON_FLAGS
 static struct option const qpkg_long_opts[] = {
-	{"clean",    no_argument, NULL, 'c'},
-	{"pretend",  no_argument, NULL, 'p'},
-	{"pkgdir",    a_argument, NULL, 'P'},
+	{QOPS"clean",    no_argument, NULL, 'c'},
+	{QOPS"pretend",  no_argument, NULL, 'p'},
+	{QOPS"pkgdir",    a_argument, NULL, 'P'},
 	COMMON_LONG_OPTS
 };
 static const char *qpkg_opts_help[] = {
diff -ur portage-utils-0.1.28.orig/qsearch.c portage-utils-0.1.28/qsearch.c
--- portage-utils-0.1.28.orig/qsearch.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qsearch.c	2007-06-07 22:07:54.826151270 +0200
@@ -11,12 +11,12 @@
 
 #define QSEARCH_FLAGS "acsSNH" COMMON_FLAGS
 static struct option const qsearch_long_opts[] = {
-	{"all",       no_argument, NULL, 'a'},
-	{"cache",     no_argument, NULL, 'c'},
-	{"search",    no_argument, NULL, 's'},
-	{"desc",       a_argument, NULL, 'S'},
-	{"name-only", no_argument, NULL, 'N'},
-	{"homepage",  no_argument, NULL, 'H'},
+	{QOPS"all",       no_argument, NULL, 'a'},
+	{QOPS"cache",     no_argument, NULL, 'c'},
+	{QOPS"search",    no_argument, NULL, 's'},
+	{QOPS"desc",       a_argument, NULL, 'S'},
+	{QOPS"name-only", no_argument, NULL, 'N'},
+	{QOPS"homepage",  no_argument, NULL, 'H'},
 	COMMON_LONG_OPTS
 };
 static const char *qsearch_opts_help[] = {
diff -ur portage-utils-0.1.28.orig/qsize.c portage-utils-0.1.28/qsize.c
--- portage-utils-0.1.28.orig/qsize.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qsize.c	2007-06-07 22:17:43.287497755 +0200
@@ -11,13 +11,13 @@
 
 #define QSIZE_FLAGS "fasSmkb" COMMON_FLAGS
 static struct option const qsize_long_opts[] = {
-	{"filesystem", no_argument, NULL, 'f'},
-	{"all",        no_argument, NULL, 'a'},
-	{"sum",        no_argument, NULL, 's'},
-	{"sum-only",   no_argument, NULL, 'S'},
-	{"megabytes",  no_argument, NULL, 'm'},
-	{"kilobytes",  no_argument, NULL, 'k'},
-	{"bytes",      no_argument, NULL, 'b'},
+	{QOPS"filesystem", no_argument, NULL, 'f'},
+	{QOPS"all",        no_argument, NULL, 'a'},
+	{QOPS"sum",        no_argument, NULL, 's'},
+	{QOPS"sum-only",   no_argument, NULL, 'S'},
+	{QOPS"megabytes",  no_argument, NULL, 'm'},
+	{QOPS"kilobytes",  no_argument, NULL, 'k'},
+	{QOPS"bytes",      no_argument, NULL, 'b'},
 	COMMON_LONG_OPTS
 };
 static const char *qsize_opts_help[] = {
@@ -33,6 +33,11 @@
 static const char qsize_rcsid[] = "$Id: qsize.c,v 1.25 2007/05/24 14:47:18 solar Exp $";
 #define qsize_usage(ret) usage(ret, QSIZE_FLAGS, qsize_long_opts, qsize_opts_help, lookup_applet_idx("qsize"))
 
+#ifdef __sun__
+#include <sys/dklabel.h>
+# define S_BLKSIZE DK_DEVID_BLKSIZE
+#endif
+
 int qsize_main(int argc, char **argv)
 {
 	DIR *dir, *dirp;
diff -ur portage-utils-0.1.28.orig/qtbz2.c portage-utils-0.1.28/qtbz2.c
--- portage-utils-0.1.28.orig/qtbz2.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qtbz2.c	2007-06-07 22:18:38.042734328 +0200
@@ -29,11 +29,11 @@
 
 #define QTBZ2_FLAGS "jstxO" COMMON_FLAGS
 static struct option const qtbz2_long_opts[] = {
-	{"join",      no_argument, NULL, 'j'},
-	{"split",     no_argument, NULL, 's'},
-	{"tarbz2",    no_argument, NULL, 't'},
-	{"xpak",      no_argument, NULL, 'x'},
-	{"stdout",    no_argument, NULL, 'O'},
+	{QOPS"join",      no_argument, NULL, 'j'},
+	{QOPS"split",     no_argument, NULL, 's'},
+	{QOPS"tarbz2",    no_argument, NULL, 't'},
+	{QOPS"xpak",      no_argument, NULL, 'x'},
+	{QOPS"stdout",    no_argument, NULL, 'O'},
 	COMMON_LONG_OPTS
 };
 static const char *qtbz2_opts_help[] = {
diff -ur portage-utils-0.1.28.orig/quse.c portage-utils-0.1.28/quse.c
--- portage-utils-0.1.28.orig/quse.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/quse.c	2007-06-07 22:18:52.282052620 +0200
@@ -16,13 +16,13 @@
 
 #define QUSE_FLAGS "eavKLDF:N" COMMON_FLAGS
 static struct option const quse_long_opts[] = {
-	{"exact",     no_argument, NULL, 'e'},
-	{"all",       no_argument, NULL, 'a'},
-	{"keywords",  no_argument, NULL, 'K'},
-	{"license",   no_argument, NULL, 'L'},
-	{"describe",  no_argument, NULL, 'D'},
-	{"format",     a_argument, NULL, 'F'},
-	{"name-only", no_argument, NULL, 'N'},
+	{QOPS"exact",     no_argument, NULL, 'e'},
+	{QOPS"all",       no_argument, NULL, 'a'},
+	{QOPS"keywords",  no_argument, NULL, 'K'},
+	{QOPS"license",   no_argument, NULL, 'L'},
+	{QOPS"describe",  no_argument, NULL, 'D'},
+	{QOPS"format",     a_argument, NULL, 'F'},
+	{QOPS"name-only", no_argument, NULL, 'N'},
 	COMMON_LONG_OPTS
 };
 static const char *quse_opts_help[] = {
diff -ur portage-utils-0.1.28.orig/qxpak.c portage-utils-0.1.28/qxpak.c
--- portage-utils-0.1.28.orig/qxpak.c	2007-06-03 21:20:28.000000000 +0200
+++ portage-utils-0.1.28/qxpak.c	2007-06-07 22:19:12.570432138 +0200
@@ -31,11 +31,11 @@
 
 #define QXPAK_FLAGS "lxcd:O" COMMON_FLAGS
 static struct option const qxpak_long_opts[] = {
-	{"list",      no_argument, NULL, 'l'},
-	{"extract",   no_argument, NULL, 'x'},
-	{"create",    no_argument, NULL, 'c'},
-	{"dir",        a_argument, NULL, 'd'},
-	{"stdout",    no_argument, NULL, 'O'},
+	{QOPS"list",      no_argument, NULL, 'l'},
+	{QOPS"extract",   no_argument, NULL, 'x'},
+	{QOPS"create",    no_argument, NULL, 'c'},
+	{QOPS"dir",        a_argument, NULL, 'd'},
+	{QOPS"stdout",    no_argument, NULL, 'O'},
 	COMMON_LONG_OPTS
 };
 static const char *qxpak_opts_help[] = {
