committed in 0.3+

--- qsize.c
+++ qsize.c
@@ -44,6 +44,10 @@
 # define S_BLKSIZE S_BLOCK_SIZE
 #endif
 
+#ifdef __hpux
+# define S_BLKSIZE st.st_blksize
+#endif
+
 int qsize_main(int argc, char **argv)
 {
 	DIR *dir, *dirp;
--- libq/compat.c.orig	2009-07-28 16:52:13 +0200
+++ libq/compat.c	2009-07-28 16:52:40 +0200
@@ -30,6 +30,8 @@
 		*strp = xstrdup(str); \
 	} while (0)
 
-#else /* __sun__ */
+#elif defined(__hpux__) || defined(__MINT__)
+  /* must not include both dir.h and dirent.h on hpux11.11 & FreeMiNT */
+#else
 # include <sys/dir.h>
 #endif
