Index: qmerge.c
===================================================================
RCS file: /var/cvsroot/gentoo-projects/portage-utils/qmerge.c,v
retrieving revision 1.75
diff -u -b -B -r1.75 qmerge.c
--- qmerge.c	3 Jun 2007 19:17:39 -0000	1.75
+++ qmerge.c	4 Jun 2007 16:50:32 -0000
@@ -952,7 +952,7 @@
 
 	if (pkg->MD5[0]) {
 		if ((hash = (char*) hash_file(fname, HASH_MD5)) == NULL) {
-			errf("hash is NULL for %s %s", fname, get_current_dir_name());
+			errf("hash is NULL for %s", fname);
 		}
 		if (strcmp(hash, pkg->MD5) == 0) {
 			if (display)
