http://bugs.gentoo.org/show_bug.cgi?id=277916

--- lib/iconv_open2.h
+++ lib/iconv_open2.h
@@ -79,9 +79,11 @@
   cd->hooks.wc_hook = NULL;
   cd->hooks.data = NULL;
   #endif
+  #if HAVE_WCRTOMB || HAVE_MBRTOWC
   /* Initialize additional fields. */
   if (from_wchar != to_wchar) {
     struct wchar_conv_struct * wcd = (struct wchar_conv_struct *) cd;
     memset(&wcd->state,'\0',sizeof(mbstate_t));
   }
+  #endif
   /* Done. */
