Index: gcc-4.3/gcc/Makefile.in
===================================================================
--- gcc-4.3/gcc/Makefile.in	(revision 131913)
+++ gcc-4.3/gcc/Makefile.in	(working copy)
@@ -3617,7 +3622,11 @@ doc: $(BUILD_INFO) $(GENERATED_MANPAGES)
 INFOFILES = doc/cpp.info doc/gcc.info doc/gccint.info \
             doc/gccinstall.info doc/cppinternals.info
 
+ifneq ($(BUILD_INFO),)
 info: $(INFOFILES) lang.info @GENINSRC@ srcinfo lang.srcinfo
+else
+info:
+endif
 
 srcinfo: $(INFOFILES)
 	-cp -p $^ $(srcdir)/doc
@@ -3983,6 +3992,7 @@ install-driver: installdirs xgcc$(exeext
 	    mv -f $(target_noncanonical)-gcc-tmp$(exeext) $(GCC_TARGET_INSTALL_NAME)$(exeext) ); \
 	fi
 
+ifneq ($(BUILD_INFO),)
 # Install the info files.
 # $(INSTALL_DATA) might be a relative pathname, so we can't cd into srcdir
 # to do the install.
@@ -3993,6 +4003,9 @@ install-info:: doc installdirs \
 	$(DESTDIR)$(infodir)/gccinstall.info \
 	$(DESTDIR)$(infodir)/gccint.info \
 	lang.install-info
+else
+install-info:: doc installdirs
+endif
 
 $(DESTDIR)$(infodir)/%.info: doc/%.info installdirs
 	rm -f $@
