diff -ru libopenraw-0.0.5.orig/lib/ljpegdecompressor.cpp libopenraw-0.0.5/lib/ljpegdecompressor.cpp
--- libopenraw-0.0.5.orig/lib/ljpegdecompressor.cpp	2008-10-22 12:55:51.531884824 +0200
+++ libopenraw-0.0.5/lib/ljpegdecompressor.cpp	2008-10-22 12:56:12.307787951 +0200
@@ -55,6 +55,7 @@
 #include <boost/scoped_array.hpp>
 #include <boost/format.hpp>
 
+#include <cstring>
 #include <libopenraw++/rawdata.h>
 #include "io/memstream.h"
 #include "debug.h"
