--- aclocal.m4.old	2007-09-18 19:07:48.000000000 +0200
+++ aclocal.m4	2007-09-18 19:15:29.000000000 +0200
@@ -102,49 +102,49 @@
         [ac_cpu=$withval])
 
    AC_ARG_VAR(OPTS, [Optimization flags for all compilers.])
-   if test x${OPTS+set} = xset ; then
-     saved_CFLAGS="$CFLAGS"
-     CFLAGS=
-     for opt in $saved_CFLAGS ; do
-       case $opt in
-         -O*|-g*) ;;
-         *) CFLAGS="$CFLAGS $opt" ;;
-       esac
-     done
-     AC_MSG_CHECKING([user provided debugging flags])
+#   if test x${OPTS+set} = xset ; then
+#     saved_CFLAGS="$CFLAGS"
+#     CFLAGS=
+#     for opt in $saved_CFLAGS ; do
+#       case $opt in
+#         -O*|-g*) ;;
+#         *) CFLAGS="$CFLAGS $opt" ;;
+#       esac
+#     done
+#     AC_MSG_CHECKING([user provided debugging flags])
      AC_MSG_RESULT($OPTS)
-   else 
-     saved_CFLAGS="$CFLAGS"
-     CFLAGS=
-     for opt in $saved_CFLAGS ; do
-       case $opt in
-         -O*) ;;
-         -g*) OPTS="$OPTS $opt" ;;
-         *) CFLAGS="$CFLAGS $opt" ;;
-       esac
-     done
+#   else 
+#     saved_CFLAGS="$CFLAGS"
+#     CFLAGS=
+#     for opt in $saved_CFLAGS ; do
+#       case $opt in
+#         -O*) ;;
+#         -g*) OPTS="$OPTS $opt" ;;
+#         *) CFLAGS="$CFLAGS $opt" ;;
+#       esac
+#     done
      if test x$ac_debug = xno ; then
        OPTS=-DNO_DEBUG
-       AC_CHECK_CC_OPT([-Wall],[OPTS="$OPTS -Wall"])
-       AC_CHECK_CC_OPT([-O3],[OPTS="$OPTS -O3"],
-         [ AC_CHECK_CC_OPT([-O2], [OPTS="$OPTS -O2"] ) ] )
-       if test -z "$ac_cpu" ; then
-        AC_MSG_WARN([guessing cpu type (use --with-cpu=cpuname to override.)])
-       fi
-       opt="-march=${ac_cpu-${host_cpu}}"
-       AC_CHECK_CC_OPT([$opt], [OPTS="$OPTS $opt"],
-	  [ opt="-mcpu=${ac_cpu-${host_cpu}}"
-            AC_CHECK_CC_OPT([$opt], [OPTS="$OPTS $opt"]) ] )
-       if test -z "$ac_cpu" -a "$host_cpu" = "i686" ; then
-            AC_CHECK_CC_OPT([-mmmx],[OPTS="$OPTS -mmmx"
-              AC_MSG_WARN([use --with-cpu=cpuname to avoid assuming that MMX works.])])
-            if test -r /proc/cpuinfo && grep -q sse /proc/cpuinfo ; then
-              AC_CHECK_CC_OPT([-msse],[OPTS="$OPTS -msse"
-                AC_MSG_WARN([use --with-cpu=cpuname to avoid assuming that SSE works.])])
-            fi
-       fi
+#       AC_CHECK_CC_OPT([-Wall],[OPTS="$OPTS -Wall"])
+#       AC_CHECK_CC_OPT([-O3],[OPTS="$OPTS -O3"],
+#         [ AC_CHECK_CC_OPT([-O2], [OPTS="$OPTS -O2"] ) ] )
+#       if test -z "$ac_cpu" ; then
+#        AC_MSG_WARN([guessing cpu type (use --with-cpu=cpuname to override.)])
+#       fi
+#       opt="-march=${ac_cpu-${host_cpu}}"
+#       AC_CHECK_CC_OPT([$opt], [OPTS="$OPTS $opt"],
+#	  [ opt="-mcpu=${ac_cpu-${host_cpu}}"
+#            AC_CHECK_CC_OPT([$opt], [OPTS="$OPTS $opt"]) ] )
+#       if test -z "$ac_cpu" -a "$host_cpu" = "i686" ; then
+#            AC_CHECK_CC_OPT([-mmmx],[OPTS="$OPTS -mmmx"
+#              AC_MSG_WARN([use --with-cpu=cpuname to avoid assuming that MMX works.])])
+#            if test -r /proc/cpuinfo && grep -q sse /proc/cpuinfo ; then
+#              AC_CHECK_CC_OPT([-msse],[OPTS="$OPTS -msse"
+#                AC_MSG_WARN([use --with-cpu=cpuname to avoid assuming that SSE works.])])
+#            fi
+#       fi
      fi
-   fi
+#   fi
 ])
 
 
