--- include/define.h
+++ include/define.h
@@ -180,8 +180,12 @@
 # ifdef tolower
 #  undef tolower
 # endif
-# define NEED_TOUPPER
-# define NEED_TOLOWER
+# ifndef HAVE_TOUPPER
+#  define NEED_TOUPPER
+# endif
+# ifndef HAVE_TOLOWER
+#  define NEED_TOLOWER
+# endif
 #endif
 
 #ifdef HAVE_WCHAR_H
--- configure.ac
+++ configure.ac
@@ -296,6 +296,7 @@
 AC_CHECK_FUNCS(setenv putenv fseeko ftello)
 AC_FUNC_MBRTOWC
 AC_CHECK_FUNCS(setlocale nl_langinfo iconv)
+AC_CHECK_FUNCS(tolower toupper)
 
 
 
