--- assuan/assuan-socket.c
+++ assuan/assuan-socket.c
@@ -35,6 +35,7 @@
 #include <assert.h>
 
 #include "assuan-defs.h"
+#include "priv-io.h"
 
 /* Hacks for Slowaris.  */
 #ifndef PF_LOCAL
--- src/priv-io.h
+++ src/priv-io.h
@@ -22,6 +22,7 @@
 #ifndef IO_H
 #define IO_H
 
+#include <sys/socket.h>
 
 /* A single file descriptor passed to spawn.  For child fds, dup_to
    specifies the fd it should become in the child, but only 0, 1 and 2
@@ -50,6 +51,7 @@
 
 /* These function are either defined in posix-io.c or w32-io.c.  */
 void _gpgme_io_subsystem_init (void);
+int _gpgme_io_connect (int fd, struct sockaddr *addr, int addrlen);
 int _gpgme_io_read (int fd, void *buffer, size_t count);
 int _gpgme_io_write (int fd, const void *buffer, size_t count);
 int _gpgme_io_pipe (int filedes[2], int inherit_idx);
