https://savannah.gnu.org/bugs/?28612

--- gl/lib/quotearg.c.old	2010-03-01 23:05:13.000000000 +0000
+++ gl/lib/quotearg.c	2010-03-01 23:05:35.000000000 +0000
@@ -31,7 +31,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <wchar.h>
+#ifndef __MINT__
 #include <wctype.h>
+#endif
 
 #include "gettext.h"
 #define _(msgid) gettext (msgid)
@@ -447,7 +449,7 @@
 	      }
 	    else
 	      {
-#ifdef __INTERIX
+#if defined(__INTERIX) || defined(__MINT__)
 		/* broken wchar support :( */
 		printable = false;
 #else // !__INTERIX
