diff -Naur gimp-2.6.0-old/plug-ins/Makefile.am gimp-2.6.0/plug-ins/Makefile.am
--- gimp-2.6.0-old/plug-ins/Makefile.am	2008-08-21 13:35:22.000000000 +0200
+++ gimp-2.6.0/plug-ins/Makefile.am	2008-10-09 17:03:04.000000000 +0200
@@ -10,6 +10,7 @@
 
 if BUILD_JPEG
 file_jpeg = file-jpeg
+file_psd = file-psd
 endif
 
 if BUILD_PRINT
@@ -57,7 +58,7 @@
 	file-fli		\
 	$(file_ico)		\
 	$(file_jpeg)		\
-	file-psd		\
+	$(file_psd)		\
 	file-sgi		\
 	$(file_uri)		\
 	$(file_xjt)		\
diff -Naur gimp-2.6.0-old/plug-ins/Makefile.in gimp-2.6.0/plug-ins/Makefile.in
--- gimp-2.6.0-old/plug-ins/Makefile.in	2008-09-29 19:23:20.000000000 +0200
+++ gimp-2.6.0/plug-ins/Makefile.in	2008-10-09 17:05:56.000000000 +0200
@@ -385,6 +385,7 @@
 
 @BUILD_HELP_BROWSER_TRUE@help_browser = help-browser
 @BUILD_JPEG_TRUE@file_jpeg = file-jpeg
+@BUILD_JPEG_TRUE@file_psd = file-psd
 @BUILD_PRINT_TRUE@print = print
 @BUILD_PYTHON_TRUE@pygimp = pygimp
 @BUILD_SCRIPT_FU_TRUE@script_fu = script-fu
@@ -405,7 +406,7 @@
 	file-fli		\
 	$(file_ico)		\
 	$(file_jpeg)		\
-	file-psd		\
+	$(file_psd)		\
 	file-sgi		\
 	$(file_uri)		\
 	$(file_xjt)		\
