diff -ru -x '*.Po' -x '*.Plo' gimp-2.4.5.orig/app/Makefile.am gimp-2.4.5/app/Makefile.am
--- gimp-2.4.5.orig/app/Makefile.am	2008-03-28 10:51:09 +0100
+++ gimp-2.4.5/app/Makefile.am	2008-03-28 10:49:57 +0100
@@ -236,3 +236,42 @@
 
 gimprc.o:
 	$(WINDRES) gimp.rc gimprc.o
+
+gimp-2.4$(EXEEXT): $(gimp_2_4_OBJECTS) $(gimp_2_4_DEPENDENCIES) 
+	@rm -f gimp-2.4$(EXEEXT)
+	ldadd_objects=; \
+	for x in $(gimp_2_4_LDADD); do \
+		case "$$x" in \
+		*.a) \
+			case "$$x" in /* ) ;; * ) x="./$$x" ;; esac; \
+			target_libdir=temp/$${x##*/}; \
+			abs_lib=$$(cd $${x%/*} && pwd)/$${x##*/}; \
+			test -d $${target_libdir} || ( mkdir -p $${target_libdir}; cd $${target_libdir}; ar x $${abs_lib}; ); \
+			for obj in $${target_libdir}/*; do \
+				ldadd_objects="$${ldadd_objects} $${obj}"; \
+			done \
+		;; \
+		* ) ldadd_objects="$${ldadd_objects} $$x" ;; \
+		esac; \
+	done; \
+	$(LINK) $(gimp_2_4_OBJECTS) $${ldadd_objects} $(LIBS)
+
+gimp-console-2.4$(EXEEXT): $(gimp_console_2_4_OBJECTS) $(gimp_console_2_4_DEPENDENCIES) 
+	@rm -f gimp-console-2.4$(EXEEXT)
+	ldadd_objects=; \
+	for x in $(gimp_console_2_4_LDADD); do \
+		case "$$x" in \
+		*.a) \
+			case "$$x" in /* ) ;; * ) x="./$$x" ;; esac; \
+			target_libdir=temp/$${x##*/}; \
+			abs_lib=$$(cd $${x%/*} && pwd)/$${x##*/}; \
+			test -d $${target_libdir} || ( mkdir -p $${target_libdir}; cd $${target_libdir}; ar x $${abs_lib}; ); \
+			for obj in $${target_libdir}/*; do \
+				ldadd_objects="$${ldadd_objects} $${obj}"; \
+			done \
+		;; \
+		* ) ldadd_objects="$${ldadd_objects} $$x" ;; \
+		esac; \
+	done; \
+	$(LINK) $(gimp_console_2_4_OBJECTS) $${ldadd_objects} $(LIBS)
+
diff -ru -x '*.Po' -x '*.Plo' gimp-2.4.5.orig/app/plug-in/gimpplugin.c gimp-2.4.5/app/plug-in/gimpplugin.c
--- gimp-2.4.5.orig/app/plug-in/gimpplugin.c	2008-03-28 10:51:09 +0100
+++ gimp-2.4.5/app/plug-in/gimpplugin.c	2008-03-28 10:09:05 +0100
@@ -452,9 +452,11 @@
           /*  If the plug-in opened a process group, kill the group instead
            *  of only the plug-in, so we kill the plug-in's children too
            */
+		  #if !defined(__INTERIX)
           if (getpgid (0) != getpgid (plug_in->pid))
             status = kill (- plug_in->pid, SIGKILL);
           else
+		  #endif
             status = kill (plug_in->pid, SIGKILL);
         }
 
