diff -Naur gimp-2.6.0.orig/plug-ins/file-uri/uri-backend-libcurl.c gimp-2.6.0/plug-ins/file-uri/uri-backend-libcurl.c
--- gimp-2.6.0.orig/plug-ins/file-uri/uri-backend-libcurl.c     2008-10-02 10:34:43.000000000 +0300
+++ gimp-2.6.0/plug-ins/file-uri/uri-backend-libcurl.c          2008-10-02 10:34:57.000000000 +0300
@@ -21,6 +21,8 @@
 
 #include "config.h"
 
+#include <errno.h>
+
 #include <curl/curl.h>
 #include <curl/types.h>
 #include <curl/easy.h>
@@ -161,6 +161,6 @@
     {
       g_set_error (error, G_FILE_ERROR, g_file_error_from_errno (errno),
                    _("Could not open '%s' for writing: %s"),
-                   gimp_filename_to_utf8 (filename),  g_strerror (errno));
+                   gimp_filename_to_utf8 (tmpname),  g_strerror (errno));
       return FALSE;
     }
diff -Naur gimp-2.6.0.orig/plug-ins/file-uri/uri-backend-wget.c gimp-2.6.0/plug-ins/file-uri/uri-backend-wget.c
--- gimp-2.6.0.orig/plug-ins/file-uri/uri-backend-wget.c	2008-10-03 15:06:32.000000000 +0300
+++ gimp-2.6.0/plug-ins/file-uri/uri-backend-wget.c	2008-10-03 15:06:42.000000000 +0300
@@ -91,7 +91,7 @@
 
   if (pipe (p) != 0)
     {
-      g_set_error (error, G_FILE_ERROR, G_FILE_ERROR_FAILED,,
+      g_set_error (error, G_FILE_ERROR, G_FILE_ERROR_FAILED,
                    "pipe() failed: %s", g_strerror (errno));
       return FALSE;
     }
@@ -101,7 +101,7 @@
 
   if ((pid = fork()) < 0)
     {
-      g_set_error (error, G_FILE_ERROR, G_FILE_ERROR_FAILED,,
+      g_set_error (error, G_FILE_ERROR, G_FILE_ERROR_FAILED,
                    "fork() failed: %s", g_strerror (errno));
       return FALSE;
     }
