--- CPP/7zip/Bundles/Format7zFree/makefile	2009-02-09 21:47:24.000000000 +0100
+++ CPP/7zip/Bundles/Format7zFree/makefile.new	2009-02-25 14:18:31.000000000 +0100
@@ -271,8 +271,6 @@
   MyAes.o \
   Pbkdf2HmacSha1.o \
   RandGen.o \
-  Rar20Crypto.o \
-  RarAes.o \
   Sha1.o \
   WzAes.o \
   ZipCrypto.o \
@@ -313,7 +313,6 @@
   $(LZH_OBJS) \
   $(LZM_OBJS) \
   $(NSIS_OBJS) \
-  $(RAR_OBJS) \
   $(SPLIT_OBJS) \
   $(TAR_OBJS) \
   $(UDF_OBJS) \

