diff -ru cdrkit-1.1.11.orig/genisoimage/tree.c cdrkit-1.1.11/genisoimage/tree.c
--- cdrkit-1.1.11.orig/genisoimage/tree.c	2011-08-24 11:21:07 +0200
+++ cdrkit-1.1.11/genisoimage/tree.c	2011-08-24 11:28:02 +0200
@@ -1551,7 +1551,11 @@
 		return (0);
 	}
 	/* print a warning but don't spam too much */
+	#ifndef __INTERIX
 	if (S_ISREG(lstatbuf.st_mode) && (lstatbuf.st_size >= (off_t)0xFFFFFFFF)) {
+	#else
+	if (S_ISREG(lstatbuf.st_mode) && (lstatbuf.st_size >= LONG_MAX)) {
+	#endif
 		static int udf_warned;
 
 		if( !allow_limited_size || verbose>1)
diff -ru cdrkit-1.1.11.orig/include/AddScgBits.cmake cdrkit-1.1.11/include/AddScgBits.cmake
--- cdrkit-1.1.11.orig/include/AddScgBits.cmake	2011-08-24 11:34:08 +0200
+++ cdrkit-1.1.11/include/AddScgBits.cmake	2011-08-24 11:11:42 +0200
@@ -61,8 +61,11 @@
    LIST(APPEND EXTRA_LIBS -pthread "-framework CoreFoundation" "-framework IOKit")
 ELSE(APPLE)
    IF(NOT LIBC_SCHED)
-      LIST(APPEND EXTRA_LIBS -lrt)
-      #MESSAGE("Using librt for realtime functions")
+   	FIND_LIBRARY(HAVE_LIBRT rt)
+   	IF(HAVE_LIBRT)
+		LIST(APPEND EXTRA_LIBS rt)
+	   	#MESSAGE("Using librt for realtime functions")
+	ENDIF(HAVE_LIBRT)
    ENDIF(NOT LIBC_SCHED)
 ENDIF(APPLE)
 
diff -ru cdrkit-1.1.11.orig/include/CMakeLists.txt cdrkit-1.1.11/include/CMakeLists.txt
--- cdrkit-1.1.11.orig/include/CMakeLists.txt	2011-08-24 11:34:08 +0200
+++ cdrkit-1.1.11/include/CMakeLists.txt	2011-08-24 11:11:50 +0200
@@ -166,5 +166,13 @@
 ")
 CHECK_C_SOURCE_COMPILES("${TESTSRC}" HAVE_DYN_ARRAYS)
 
+SET(TESTSRC "
+int main() {
+	(void)valloc(0x10);
+	return 0;
+}
+")
+CHECK_C_SOURCE_COMPILES("${TESTSRC}" HAVE_VALLOC)
+
 CONFIGURE_FILE( xconfig.h.in xconfig.h)
 
diff -ru cdrkit-1.1.11.orig/include/xconfig.h.in cdrkit-1.1.11/include/xconfig.h.in
--- cdrkit-1.1.11.orig/include/xconfig.h.in	2011-08-24 11:34:09 +0200
+++ cdrkit-1.1.11/include/xconfig.h.in	2011-08-24 11:28:23 +0200
@@ -88,7 +88,11 @@
 
 #cmakedefine HAVE_SYS_MMAN_H
 		/* to use definitions for mmap()/madvise()... from sys/mman.h */
+#ifdef __INTERIX
+#define MAJOR_IN_MKDEV 1		/* interix has those in sys/mkdev.h */
+#else
 #define MAJOR_IN_SYSMACROS 1	/* if we should include sys/sysmacros.h to get major()/minor()/makedev() */
+#endif
 #cmakedefine HAVE_SYS_DKIO_H
 #cmakedefine HAVE_SUN_DKIO_H
 #cmakedefine HAVE_POLL_H
@@ -174,7 +178,7 @@
 #define HAVE_USLEEP 1		/* usleep() is present in libc */
 #define HAVE_FORK 1		/* fork() is present in libc */
 #define HAVE_ALLOCA 1		/* alloca() is present (else use malloc())*/
-#define HAVE_VALLOC 1		/* valloc() is present in libc (else use malloc())*/
+#cmakedefine HAVE_VALLOC 1		/* valloc() is present in libc (else use malloc())*/
 
 /*
  * Important:	This must be a result from a check _before_ the Large File test
@@ -270,10 +274,12 @@
  */
 #define RETSIGTYPE void
 
+#ifndef __INTERIX
 #define	HAVE_LARGEFILES 1
 #define _FILE_OFFSET_BITS 64 /* glibc style */
 #define _LARGEFILE_SOURCE 1	/* To make ftello() visible (HP-UX 10.20). */
 #define _LARGE_FILES 1 	/* Large file defined on AIX-style hosts.  */
+#endif
 
 /*
  * Misc CC / LD related stuff
