diff -ru cdrkit-1.1.9.orig/genisoimage/genisoimage.h cdrkit-1.1.9/genisoimage/genisoimage.h
--- cdrkit-1.1.9.orig/genisoimage/genisoimage.h	2009-09-15 12:56:07 +0200
+++ cdrkit-1.1.9/genisoimage/genisoimage.h	2009-09-15 12:55:24 +0200
@@ -38,6 +38,9 @@
 
 #define APPID_DEFAULT "GENISOIMAGE ISO 9660/HFS FILESYSTEM CREATOR (C) 1993 E.YOUNGDALE (C) 1997-2006 J.PEARSON/J.SCHILLING (C) 2006-2007 CDRKIT TEAM"
 
+#ifdef __INTERIX
+#  include <stdint.h>
+#endif
 
 #include <mconfig.h>	/* Must be before stdio.h for LARGEFILE support */
 #include <stdio.h>
diff -ru cdrkit-1.1.9.orig/genisoimage/tree.c cdrkit-1.1.9/genisoimage/tree.c
--- cdrkit-1.1.9.orig/genisoimage/tree.c	2009-09-15 12:55:42 +0200
+++ cdrkit-1.1.9/genisoimage/tree.c	2009-09-15 12:58:07 +0200
@@ -1550,8 +1550,13 @@
 #endif
 		return (0);
 	}
+#ifndef __INTERIX
 	/* print a warning but don't spam too much */
 	if (S_ISREG(lstatbuf.st_mode) && (lstatbuf.st_size >= (off_t)0xFFFFFFFF)) {
+#else
+	/* on interix, off_t is a (signed!) long */
+	if (S_ISREG(lstatbuf.st_mode) && (lstatbuf.st_size >= LONG_MAX)) {
+#endif
 		static int udf_warned;
 
 		if( !allow_limited_size || verbose>1)
@@ -1568,6 +1573,7 @@
 					"ISO9660, Joliet, RockRidge, HFS will display incorrect size.\n");
 		}
 	}
+
 	/*
 	 * Add this so that we can detect directory loops with hard links.
 	 * If we are set up to follow symlinks, then we skip this checking.
diff -ru cdrkit-1.1.9.orig/include/AddScgBits.cmake cdrkit-1.1.9/include/AddScgBits.cmake
--- cdrkit-1.1.9.orig/include/AddScgBits.cmake	2009-09-15 12:56:07 +0200
+++ cdrkit-1.1.9/include/AddScgBits.cmake	2009-09-15 12:55:24 +0200
@@ -61,7 +61,7 @@
    LIST(APPEND EXTRA_LIBS -pthread "-framework CoreFoundation" "-framework IOKit")
 ELSE(APPLE)
    IF(NOT LIBC_SCHED)
-      LIST(APPEND EXTRA_LIBS -lrt)
+      #LIST(APPEND EXTRA_LIBS -lrt)
       #MESSAGE("Using librt for realtime functions")
    ENDIF(NOT LIBC_SCHED)
 ENDIF(APPLE)
diff -ru cdrkit-1.1.9.orig/include/xconfig.h.in cdrkit-1.1.9/include/xconfig.h.in
--- cdrkit-1.1.9.orig/include/xconfig.h.in	2009-09-15 12:56:07 +0200
+++ cdrkit-1.1.9/include/xconfig.h.in	2009-09-15 12:55:24 +0200
@@ -62,7 +62,9 @@
 #define TIME_WITH_SYS_TIME 1   /* may include both time.h and sys/time.h */
 
 #define HAVE_UTIME 1		/* to use AT&T utime() and utimbuf */
-#define HAVE_UTIMES 1		/* to use BSD utimes() and sys/time.h */
+#ifndef __INTERIX
+#  define HAVE_UTIMES 1		/* to use BSD utimes() and sys/time.h */
+#endif
 
 #cmakedefine HAVE_UTIME_H
 		/* to use utime.h for the utimbuf structure declaration, else declare struct utimbuf yourself */
@@ -88,7 +90,9 @@
 
 #cmakedefine HAVE_SYS_MMAN_H
 		/* to use definitions for mmap()/madvise()... from sys/mman.h */
-#define MAJOR_IN_SYSMACROS 1	/* if we should include sys/sysmacros.h to get major()/minor()/makedev() */
+#ifndef __INTERIX
+#  define MAJOR_IN_SYSMACROS 1	/* if we should include sys/sysmacros.h to get major()/minor()/makedev() */
+#endif
 #cmakedefine HAVE_SYS_DKIO_H
 #cmakedefine HAVE_SUN_DKIO_H
 #cmakedefine HAVE_POLL_H
@@ -128,30 +132,44 @@
 #define HAVE_STRERROR 1		/* strerror() is present in libc */
 #define HAVE_MEMMOVE 1		/* memmove() is present in libc */
 
-#define HAVE_MLOCK 1		/* mlock() is present in libc */
+#ifndef __INTERIX
+#  define HAVE_MLOCK 1		/* mlock() is present in libc */
+#endif
 #cmakedefine HAVE_MLOCKALL
 	/* working mlockall() is present in libc */
 #define HAVE_FLOCK 1		/* *BSD flock() is present in libc */
 #define HAVE_SETREUID 1		/* setreuid() is present in libc */
-#define HAVE_SETEUID 1		/* seteuid() is present in libc */
+#ifndef __INTERIX
+#  define HAVE_SETEUID 1		/* seteuid() is present in libc */
+#endif
 #define HAVE_SETUID 1		/* setuid() is present in libc */
 #define HAVE_SETREGID 1		/* setregid() is present in libc */
-#define HAVE_SETEGID 1		/* setegid() is present in libc */
+#ifndef __INTERIX
+#  define HAVE_SETEGID 1		/* setegid() is present in libc */
+#endif
 #define HAVE_SETGID 1		/* setgid() is present in libc */
 #define HAVE_TCGETATTR 1		/* tcgetattr() is present in libc */
 #define HAVE_TCSETATTR 1		/* tcsetattr() is present in libc */
-#define HAVE_GETHOSTID 1		/* gethostid() is present in libc */
+#ifndef __INTERIX
+#  define HAVE_GETHOSTID 1		/* gethostid() is present in libc */
+#endif
 #define HAVE_GETHOSTNAME 1		/* gethostname() is present in libc */
-#define HAVE_GETDOMAINNAME 1	/* getdomainname() is present in libc */
+#ifndef __INTERIX
+#  define HAVE_GETDOMAINNAME 1	/* getdomainname() is present in libc */
+#endif
 #define HAVE_GETPAGESIZE 1		/* getpagesize() is present in libc */
-#define HAVE_POLL 1		/* poll() is present in libc */
+#ifndef __INTERIX
+#  define HAVE_POLL 1		/* poll() is present in libc */
+#endif
 #define HAVE_SELECT 1		/* select() is present in libc */
 #define HAVE_LCHOWN 1		/* lchown() is present in libc */
 #define HAVE_SBRK 1		/* sbrk() is present in libc */
 #define HAVE_VA_COPY 1		/* va_copy() is present in varargs.h/stdarg.h */
 #define HAVE_DUP2 1		/* dup2() is present in libc */
 #define HAVE_GETCWD 1		/* POSIX getcwd() is present in libc */
-#define HAVE_SMMAP 1		/* may map anonymous memory to get shared mem */
+#ifndef __INTERIX
+#  define HAVE_SMMAP 1		/* may map anonymous memory to get shared mem */
+#endif
 #define HAVE_SHMAT 1		/* shmat() is present in libc */
 #define HAVE_SEMGET 1		/* semget() is present in libc */
 #define HAVE_LSTAT 1		/* lstat() is present in libc */
@@ -170,11 +188,15 @@
 #define HAVE_STRDUP 1		/* strdup() is present in libc */
 #define HAVE_SIGPROCMASK 1		/* sigprocmask() is present in libc (POSIX) */
 #define HAVE_SIGSET 1		/* sigset() is present in libc (POSIX) */
-#define HAVE_NANOSLEEP 1		/* nanosleep() is present in libc */
+#ifndef __INTERIX
+#  define HAVE_NANOSLEEP 1		/* nanosleep() is present in libc */
+#endif
 #define HAVE_USLEEP 1		/* usleep() is present in libc */
 #define HAVE_FORK 1		/* fork() is present in libc */
 #define HAVE_ALLOCA 1		/* alloca() is present (else use malloc())*/
-#define HAVE_VALLOC 1		/* valloc() is present in libc (else use malloc())*/
+#ifndef __INTERIX
+#  define HAVE_VALLOC 1		/* valloc() is present in libc (else use malloc())*/
+#endif
 
 /*
  * Important:	This must be a result from a check _before_ the Large File test
@@ -223,9 +245,11 @@
 /* have union wait in wait.h */
 #cmakedefine HAVE_UNION_WAIT
 
-#define HAVE_ST_NSEC 1		/* if struct stat contains st_atim.tv_nsec (nanosecs) */
 #define HAVE_ST_RDEV 1		/* if struct stat contains st_rdev */
-#define HAVE_SOCKADDR_STORAGE 1	/* if socket.h defines struct sockaddr_storage */
+#ifndef __INTERIX
+#  define HAVE_ST_NSEC 1		/* if struct stat contains st_atim.tv_nsec (nanosecs) */
+#  define HAVE_SOCKADDR_STORAGE 1	/* if socket.h defines struct sockaddr_storage */
+#endif
 
 /*
  * Byteorder/Bitorder
@@ -270,10 +294,12 @@
  */
 #define RETSIGTYPE void
 
-#define	HAVE_LARGEFILES 1
-#define _FILE_OFFSET_BITS 64 /* glibc style */
-#define _LARGEFILE_SOURCE 1	/* To make ftello() visible (HP-UX 10.20). */
-#define _LARGE_FILES 1 	/* Large file defined on AIX-style hosts.  */
+#ifndef __INTERIX
+#  define	HAVE_LARGEFILES 1
+#  define _FILE_OFFSET_BITS 64 /* glibc style */
+#  define _LARGEFILE_SOURCE 1	/* To make ftello() visible (HP-UX 10.20). */
+#  define _LARGE_FILES 1 	/* Large file defined on AIX-style hosts.  */
+#endif
 
 /*
  * Misc CC / LD related stuff
