diff -ru -x 'Makefile*' -x '*.make' -x link.txt cdrkit-1.1.7.1.orig/genisoimage/genisoimage.h cdrkit-1.1.7.1/genisoimage/genisoimage.h
--- cdrkit-1.1.7.1.orig/genisoimage/genisoimage.h	2008-06-03 10:20:12 +0200
+++ cdrkit-1.1.7.1/genisoimage/genisoimage.h	2008-06-03 10:22:15 +0200
@@ -54,6 +54,10 @@
 #include "jte.h"
 #endif
 
+#ifdef __INTERIX
+#include <stdint.h>
+#endif
+
 #ifdef	DVD_VIDEO
 #ifndef	UDF
 #define	UDF
diff -ru -x 'Makefile*' -x '*.make' -x link.txt cdrkit-1.1.7.1.orig/include/AddScgBits.cmake cdrkit-1.1.7.1/include/AddScgBits.cmake
--- cdrkit-1.1.7.1.orig/include/AddScgBits.cmake	2008-06-03 10:20:12 +0200
+++ cdrkit-1.1.7.1/include/AddScgBits.cmake	2008-06-03 10:52:39 +0200
@@ -57,9 +57,9 @@
 SET(CMAKE_REQUIRED_LIBRARIES )
    CHECK_C_SOURCE_COMPILES("${TESTSRC}" LIBC_SCHED)
 
-IF(NOT LIBC_SCHED)
-   LIST(APPEND EXTRA_LIBS -lrt)
-   #MESSAGE("Using librt for realtime functions")
-ENDIF(NOT LIBC_SCHED)
+#IF(NOT LIBC_SCHED)
+#   LIST(APPEND EXTRA_LIBS -lrt)
+#   MESSAGE("Using librt for realtime functions")
+#ENDIF(NOT LIBC_SCHED)
 
 ENDIF(NOT GUARD_SCGCONFIG)
diff -ru -x 'Makefile*' -x '*.make' -x link.txt cdrkit-1.1.7.1.orig/include/xconfig.h.in cdrkit-1.1.7.1/include/xconfig.h.in
--- cdrkit-1.1.7.1.orig/include/xconfig.h.in	2008-06-03 10:20:13 +0200
+++ cdrkit-1.1.7.1/include/xconfig.h.in	2008-06-03 11:01:27 +0200
@@ -88,7 +88,9 @@
 
 #cmakedefine HAVE_SYS_MMAN_H
 		/* to use definitions for mmap()/madvise()... from sys/mman.h */
+#ifndef __INTERIX
 #define MAJOR_IN_SYSMACROS 1	/* if we should include sys/sysmacros.h to get major()/minor()/makedev() */
+#endif
 #cmakedefine HAVE_SYS_DKIO_H
 #cmakedefine HAVE_SUN_DKIO_H
 #cmakedefine HAVE_POLL_H
@@ -174,7 +176,9 @@
 #define HAVE_USLEEP 1		/* usleep() is present in libc */
 #define HAVE_FORK 1		/* fork() is present in libc */
 #define HAVE_ALLOCA 1		/* alloca() is present (else use malloc())*/
+#ifndef __INTERIX
 #define HAVE_VALLOC 1		/* valloc() is present in libc (else use malloc())*/
+#endif
 
 /*
  * Important:	This must be a result from a check _before_ the Large File test
diff -ru cdrkit-1.1.7.1.orig/genisoimage/tree.c cdrkit-1.1.7.1/genisoimage/tree.c
--- cdrkit-1.1.7.1.orig/genisoimage/tree.c	2008-06-03 14:29:45 +0200
+++ cdrkit-1.1.7.1/genisoimage/tree.c	2008-06-03 15:56:23 +0200
@@ -1549,6 +1549,8 @@
 		return (0);
 	}
 	/* print a warning but don't spam too much */
+	/* interix cannot handle files bigger than 2 GiB (off_t -> signed long) */
+	#ifndef __INTERIX
 	if (S_ISREG(lstatbuf.st_mode) && (lstatbuf.st_size >= (off_t)0xFFFFFFFF)) {
 		static int udf_warned;
 
@@ -1566,6 +1568,7 @@
 					"ISO9660, Joliet, RockRidge, HFS will display incorrect size.\n");
 		}
 	}
+	#endif
 	/*
 	 * Add this so that we can detect directory loops with hard links.
 	 * If we are set up to follow symlinks, then we skip this checking.
