diff -Naurp wxPython-src-2.8.9.2-orig/wxPython/setup.py wxPython-src-2.8.9.2/wxPython/setup.py
--- wxPython-src-2.8.9.2-orig/wxPython/setup.py	2009-02-21 13:13:31.000000000 -0600
+++ wxPython-src-2.8.9.2/wxPython/setup.py	2009-02-21 13:14:12.000000000 -0600
@@ -878,12 +878,6 @@ WX_PKGLIST =      [ 'wx',
                     'wx.tools',
                     'wx.tools.XRCed',
                     'wx.tools.XRCed.plugins',
-                    'wx.tools.Editra',
-                    'wx.tools.Editra.src',
-                    'wx.tools.Editra.src.autocomp',
-                    'wx.tools.Editra.src.eclib',
-                    'wx.tools.Editra.src.extern',
-                    'wx.tools.Editra.src.syntax',
                     ]
 
 if not EGGing:
@@ -910,7 +904,6 @@ else:
                opj('scripts/pywrap'),
                opj('scripts/pywxrc'),
                opj('scripts/xrced'),
-               opj('scripts/editra'),
                ]
 
 
@@ -922,16 +915,6 @@ DATA_FILES += find_data_files('wx/tools/
 DATA_FILES += find_data_files('wx/tools/XRCed/plugins', '*.crx')
 DATA_FILES += find_data_files('wx/tools/XRCed/plugins/bitmaps', '*.png')
 
-DATA_FILES += find_data_files('wx/tools/Editra/docs', '*.txt')
-DATA_FILES += find_data_files('wx/tools/Editra/locale', '*.mo')
-DATA_FILES += find_data_files('wx/tools/Editra/pixmaps',
-                              '*.png', '*.icns', '*.ico', 'README', 'AUTHORS', 'COPYING')
-DATA_FILES += find_data_files('wx/tools/Editra/plugins', '*.egg')
-DATA_FILES += find_data_files('wx/tools/Editra/src', 'README')
-DATA_FILES += find_data_files('wx/tools/Editra/styles', '*.ess')
-DATA_FILES += find_data_files('wx/tools/Editra/tests/syntax', '*')
-DATA_FILES += find_data_files('wx/tools/Editra', '[A-Z]*', recursive=False)
-
 
 ## import pprint
 ## pprint.pprint(DATA_FILES)
@@ -981,7 +964,6 @@ if EGGing:
                                   'pyshell = wx.py.PyShell:main',
                                   'pywrap = wx.py.PyWrap:main',
                                   'helpviewer = wx.tools.helpviewer:main',
-                                  'editra = wx.tools.Editra.launcher:main',
                                   'xrced = wx.tools.XRCed.xrced:main',
                                   ], 
             },
