diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/editra wxPython-src-2.8.8.0/wxPython/scripts/editra
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/editra	2008-02-29 14:32:17.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/editra	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.tools.Editra.launcher import main
 main()
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/genaxmodule wxPython-src-2.8.8.0/wxPython/scripts/genaxmodule
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/genaxmodule	2008-02-29 14:32:17.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/genaxmodule	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.tools.genaxmodule import main
 main()
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/helpviewer wxPython-src-2.8.8.0/wxPython/scripts/helpviewer
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/helpviewer	2007-07-18 16:33:55.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/helpviewer	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.tools.helpviewer import main
 main()
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/img2png wxPython-src-2.8.8.0/wxPython/scripts/img2png
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/img2png	2007-07-18 16:33:55.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/img2png	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.tools.img2png import main
 main()
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/img2py wxPython-src-2.8.8.0/wxPython/scripts/img2py
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/img2py	2008-02-29 14:32:17.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/img2py	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.tools.img2py import main
 main()
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/img2xpm wxPython-src-2.8.8.0/wxPython/scripts/img2xpm
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/img2xpm	2007-07-18 16:33:55.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/img2xpm	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.tools.img2xpm import main
 main()
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/pyalacarte wxPython-src-2.8.8.0/wxPython/scripts/pyalacarte
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/pyalacarte	2007-07-18 16:33:55.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/pyalacarte	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.py.PyAlaCarte import main
 main()
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/pyalamode wxPython-src-2.8.8.0/wxPython/scripts/pyalamode
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/pyalamode	2007-07-18 16:33:55.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/pyalamode	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.py.PyAlaMode import main
 main()
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/pycrust wxPython-src-2.8.8.0/wxPython/scripts/pycrust
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/pycrust	2007-07-18 16:33:55.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/pycrust	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.py.PyCrust import main
 main()
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/pyshell wxPython-src-2.8.8.0/wxPython/scripts/pyshell
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/pyshell	2007-07-18 16:33:55.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/pyshell	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.py.PyShell import main
 main()
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/pywrap wxPython-src-2.8.8.0/wxPython/scripts/pywrap
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/pywrap	2007-07-18 16:33:55.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/pywrap	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.py.PyWrap import main
 main()
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/pywxrc wxPython-src-2.8.8.0/wxPython/scripts/pywxrc
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/pywxrc	2008-02-29 14:32:17.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/pywxrc	2008-06-26 17:39:49.000000000 -0600
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
-
+import wxversion
+wxversion.select("2.8")
 from wx.tools.pywxrc import main
 main()
-
diff -Naur wxPython-src-2.8.8.0-orig/wxPython/scripts/xrced wxPython-src-2.8.8.0/wxPython/scripts/xrced
--- wxPython-src-2.8.8.0-orig/wxPython/scripts/xrced	2007-07-18 16:33:55.000000000 -0600
+++ wxPython-src-2.8.8.0/wxPython/scripts/xrced	2008-06-26 17:38:39.000000000 -0600
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+import wxversion
+wxversion.select("2.8")
 
 from wx.tools.XRCed.xrced import main
 main()
