diff -ruN valgrind-3.3.1/configure.in valgrind-3.3.1-fixed/configure.in
--- valgrind-3.3.1/configure.in	2008-12-13 11:41:07.000000000 +0100
+++ valgrind-3.3.1-fixed/configure.in	2008-12-13 11:43:06.000000000 +0100
@@ -479,6 +479,16 @@
 ],
 libc="2.8")
 
+AC_EGREP_CPP([GLIBC_29], [
+#include <features.h>
+#ifdef __GNU_LIBRARY__
+ #if (__GLIBC__ == 2 && __GLIBC_MINOR__ == 9)
+  GLIBC_29
+ #endif
+#endif
+],
+libc="2.9")
+
 AC_EGREP_CPP([AIX5_LIBC], [
 #include <standards.h>
 #if defined(_AIXVERSION_510) || defined(_AIXVERSION_520) || defined(_AIXVERSION_530)
@@ -535,6 +545,12 @@
 	DEFAULT_SUPP="glibc-2.8.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	;;
+     2.9)
+	AC_MSG_RESULT(2.9 family)
+	AC_DEFINE([GLIBC_2_9], 1, [Define to 1 if you're using glibc 2.9.x])
+	DEFAULT_SUPP="glibc-2.9.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	;;
      aix5)
 	AC_MSG_RESULT(AIX 5.1 or 5.2 or 5.3)
 	AC_DEFINE([AIX5_LIBC], 1, [Define to 1 if you're using AIX 5.1 or 5.2 or 5.3])
@@ -543,7 +559,7 @@
 
      *)
 	AC_MSG_RESULT(unsupported version)
-	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.7])
+	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.9])
 	AC_MSG_ERROR([or AIX 5.1 or 5.2 or 5.3 libc])
 	;;
 esac
diff -ruN valgrind-3.3.1/glibc-2.9.supp valgrind-3.3.1-fixed/glibc-2.9.supp
--- valgrind-3.3.1/glibc-2.9.supp	1970-01-01 01:00:00.000000000 +0100
+++ valgrind-3.3.1-fixed/glibc-2.9.supp	2008-12-13 11:38:58.000000000 +0100
@@ -0,0 +1,95 @@
+
+# Errors to suppress by default with glibc 2.9.x
+
+# Format of this file is:
+# {
+#     name_of_suppression
+#     tool_name:supp_kind
+#     (optional extra info for some suppression types)
+#     caller0 name, or /name/of/so/file.so
+#     caller1 name, or ditto
+#     (optionally: caller2 name)
+#     (optionally: caller3 name)
+#  }
+#
+# For Memcheck, the supp_kinds are:
+#
+#     Param Value1 Value2 Value4 Value8 Value16 Jump
+#     Free Addr1 Addr2 Addr4 Addr8 Addr16
+#     Cond (previously known as Value0)
+#
+# and the optional extra info is:
+#     if Param: name of system call param
+
+{
+   dl-hack3-cond-1
+   Memcheck:Cond
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/ld-2.9*.so*
+}
+{
+   dl-hack3-cond-2
+   Memcheck:Cond
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/libc-2.9*.so*
+}
+{
+   dl-hack3-cond-3
+   Memcheck:Cond
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/libc-2.9*.so*
+   obj:/lib*/libc-2.9*.so*
+}
+{
+   dl-hack3-cond-4
+   Memcheck:Cond
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/libdl-2.9*.so*
+}
+
+{
+   dl-hack4-64bit-addr-1
+   Memcheck:Addr8
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/ld-2.9*.so*
+}
+{
+   dl-hack4-64bit-addr-2
+   Memcheck:Addr8
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/libc-2.9*.so*
+}
+{
+   dl-hack4-64bit-addr-3
+   Memcheck:Addr8
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/ld-2.9*.so*
+   obj:/lib*/libdl-2.9*.so*
+}
+
+{
+   dl-hack5-32bit-addr-1
+   Memcheck:Addr4
+   obj:/lib/ld-2.9*.so
+   obj:/lib/ld-2.9*.so
+   obj:/lib/ld-2.9*.so
+}
+{
+   dl-hack5-32bit-addr-3
+   Memcheck:Addr4
+   obj:/lib/ld-2.9*.so
+   obj:/lib/ld-2.9*.so
+   obj:/lib/libdl-2.9*.so*
+}
+{
+   dl-hack5-32bit-addr-4
+   Memcheck:Addr4
+   obj:/lib/ld-2.9*.so
+   obj:/lib/libdl-2.9*.so*
+   obj:/lib/ld-2.9*.so
+}
