diff -ruN valgrind-3.2.1/coregrind/launcher.c valgrind-3.2.1-patched/coregrind/launcher.c
--- valgrind-3.2.1/coregrind/launcher.c	2006-08-28 15:39:04.000000000 +0200
+++ valgrind-3.2.1-patched/coregrind/launcher.c	2006-09-17 18:22:41.458977000 +0200
@@ -244,7 +244,7 @@
    else if (0==strcmp(VG_PLATFORM,"ppc32-linux"))
       default_platform = "ppc32-linux";
    else if (0==strcmp(VG_PLATFORM,"ppc64-linux"))
-      default_platform = "ppc32-linux";
+      default_platform = "ppc64-linux";
    else
       barf("Unknown VG_PLATFORM '%s'", VG_PLATFORM);
 
