diff -ru xprop-1.0.4.orig/xprop.c xprop-1.0.4/xprop.c
--- xprop-1.0.4.orig/xprop.c	2009-01-16 16:01:46 +0100
+++ xprop-1.0.4/xprop.c	2009-01-16 16:03:42 +0100
@@ -31,6 +31,10 @@
 
 #include "config.h"
 
+#ifdef _WIN32
+# define GetAtomName wGetAtomName
+#endif
+
 #include <X11/Xlib.h>
 #include <X11/Xos.h>
 #include <X11/Xfuncs.h>
@@ -54,6 +58,10 @@
 
 #include "dsimple.h"
 
+#ifdef _WIN32
+# undef GetAtomName
+#endif
+
 #define MAXSTR 500000
 #define MAXELEMENTS 64
 
