--- configure.in
+++ configure.in
@@ -26,4 +26,6 @@
 AC_TYPE_MODE_T
 AC_FUNC_ALLOCA
 
+AC_CHECK_HEADER([features.h])
+
 AC_OUTPUT(include/builddefs)
--- include/xattr.h
+++ include/xattr.h
@@ -18,7 +18,13 @@
 #ifndef __XATTR_H__
 #define __XATTR_H__
 
+#if defined (HAVE_CONFIG_H)
+#include "config.h"
+#endif
+
+#ifdef HAVE_FEATURES_H
 #include <features.h>
+#endif
 
 #include <errno.h>
 #ifndef ENOATTR
