--- rpmoffset.c
+++ rpmoffset.c
@@ -16,8 +15,17 @@
 {
         char *buff = malloc(RPMBUFSIZ),*eb,*p;
         for (p = buff, eb = buff + read(0,buff,RPMBUFSIZ); p < eb; p++)
-                if (*p == '\037' && p[1] == '\213' && p[2] == '\010')
-                        printf("%d\n",p - buff),
+	{
+                if (*p == '\037' && p[1] == '\213' && p[2] == '\010') 
+		{
+                        printf("%d\n",p - buff);
                         exit(0);
+		}
+		else if (*p == 'B' && p[1] == 'Z' && p[2] == 'h' )
+		{
+		 	printf("%d\n",p - buff);
+			exit(0);
+		}
+        }
         exit(1);
 }
