Index: work/rpmoffset.c
===================================================================
--- work.orig/rpmoffset.c
+++ work/rpmoffset.c
@@ -5,6 +5,7 @@
 /* Wouldn't it be a lot more sane if we could just untar these things? */
 
 #include <stdlib.h>
+#include <stdio.h>
 
 /* These offsets keep getting bigger, so we're going to just bite a 2MB */
 /* chunk of RAM right away so that we have enough.  Yeah, horrible */
@@ -19,12 +20,12 @@ main()
 	{
                 if (*p == '\037' && p[1] == '\213' && p[2] == '\010') 
 		{
-                        printf("%d\n",p - buff);
+                        printf("%ld\n",p - buff);
                         exit(0);
 		}
 		else if (*p == 'B' && p[1] == 'Z' && p[2] == 'h' )
 		{
-		 	printf("%d\n",p - buff);
+		 	printf("%ld\n",p - buff);
 			exit(0);
 		}
         }
