diff -pur --exclude='*~' inkscape-0.46.p1/src/style.cpp inkscape-0.46/src/style.cpp
--- inkscape-0.46.p1/src/style.cpp	2008-04-08 09:23:25.000000000 +0200
+++ inkscape-0.46/src/style.cpp	2008-04-08 12:30:35.000000000 +0200
@@ -3614,7 +3614,7 @@ sp_style_write_ipaint(gchar *b, gint con
                 css << "currentColor";
             }
 
-            if ( paint->colorSet ) {
+            if ( paint->colorSet && !paint->currentcolor ) {
                 if ( !css.str().empty() ) {
                     css << " ";
                 }
@@ -3623,7 +3623,7 @@ sp_style_write_ipaint(gchar *b, gint con
                 css << color_buf;
             }
 
-            if (paint->value.color.icc) {
+            if (paint->value.color.icc && !paint->currentcolor) {
                 if ( !css.str().empty() ) {
                     css << " ";
                 }
