diff -pur --exclude='*~' inkscape-0.46.p2/src/style.cpp inkscape-0.46/src/style.cpp
--- inkscape-0.46.p2/src/style.cpp	2008-04-08 22:46:34.000000000 +0200
+++ inkscape-0.46/src/style.cpp	2008-04-08 23:29:29.000000000 +0200
@@ -1582,7 +1582,7 @@ sp_style_merge_paint_prop_from_dying_par
      * I haven't given this much attention.  See comments below about
      * currentColor, colorProfile, and relative URIs.
      */
-    if (!child.set || child.inherit || child.currentcolor) {
+    if (!child.set || child.inherit) {
         sp_style_merge_ipaint(style, &child, &parent);
         child.set = parent.set;
         child.inherit = parent.inherit;
@@ -2142,6 +2142,8 @@ sp_style_merge_ipaint(SPStyle *style, SP
         }
     } else if ( parent->isColor() ) {
         paint->setColor( parent->value.color );
+    } else if ( parent->isNoneSet() ) {
+        paint->noneSet = TRUE;
     } else if ( parent->isNone() ) {
         //
     } else {
