diff -ru libxcb-1.4.orig/src/xcb_util.c libxcb-1.4/src/xcb_util.c
--- libxcb-1.4.orig/src/xcb_util.c	2009-08-21 05:27:29 -0700
+++ libxcb-1.4/src/xcb_util.c	2009-08-21 05:28:12 -0700
@@ -246,7 +246,7 @@
     hints.ai_socktype = SOCK_STREAM;
 #endif
 
-#ifdef AF_INET6
+#if defined(AF_INET6) && !defined(__INTERIX)
     /* Allow IPv6 addresses enclosed in brackets. */
     if(host[0] == '[' && (bracket = strrchr(host, ']')) && bracket[1] == '\0')
     {
