diff -Nur gtkspell-2.0.15/gtkspell/gtkspell.c gtkspell-2.0.15_patched/gtkspell/gtkspell.c
--- gtkspell-2.0.15/gtkspell/gtkspell.c	2008-12-01 10:38:40.000000000 +0200
+++ gtkspell-2.0.15_patched/gtkspell/gtkspell.c	2009-05-17 14:50:09.000000000 +0300
@@ -387,10 +387,9 @@
 			if (inside_more_submenu) gtk_menu_shell_append(GTK_MENU_SHELL(menu), mi);
 			else gtk_menu_shell_insert(GTK_MENU_SHELL(menu), mi, menu_position++);
 		}
+		enchant_dict_free_string_list(spell->speller, suggestions);
 	}
 
-	enchant_dict_free_suggestions(spell->speller, suggestions);
-
 	/* + Add to Dictionary */
 	label = g_strdup_printf(_("Add \"%s\" to Dictionary"), word);
 	mi = gtk_image_menu_item_new_with_label(label);
