diff -ru wxPython-src-2.8.9.2.orig/acinclude.m4 wxPython-src-2.8.9.2/acinclude.m4
--- wxPython-src-2.8.9.2.orig/acinclude.m4	2009-02-26 15:58:27 +0100
+++ wxPython-src-2.8.9.2/acinclude.m4	2009-02-26 15:50:38 +0100
@@ -540,7 +540,7 @@
 AC_DEFUN([WX_VERSIONED_SYMBOLS],
 [
     case "${host}" in
-        *-*-cygwin* | *-*-mingw* )
+        *-*-cygwin* | *-*-mingw* | *-interix* )
             dnl although ld does support version script option on these
             dnl platforms, it doesn't make much sense to use it under Win32
             dnl and, moreover, this breaks linking because of a bug in handling
diff -ru wxPython-src-2.8.9.2.orig/build/aclocal/bakefile.m4 wxPython-src-2.8.9.2/build/aclocal/bakefile.m4
--- wxPython-src-2.8.9.2.orig/build/aclocal/bakefile.m4	2009-02-26 15:56:45 +0100
+++ wxPython-src-2.8.9.2/build/aclocal/bakefile.m4	2009-02-26 15:16:49 +0100
@@ -429,6 +429,9 @@
         WINDOWS_IMPLIB=1
       ;;
 
+	  *-interix* )
+	  ;;
+
       *-pc-os2_emx | *-pc-os2-emx )
         SHARED_LD_CC="`pwd`/dllar.sh -libf INITINSTANCE -libf TERMINSTANCE -o"
         SHARED_LD_CXX="`pwd`/dllar.sh -libf INITINSTANCE -libf TERMINSTANCE -o"
diff -ru wxPython-src-2.8.9.2.orig/configure.in wxPython-src-2.8.9.2/configure.in
--- wxPython-src-2.8.9.2.orig/configure.in	2009-02-26 15:58:46 +0100
+++ wxPython-src-2.8.9.2/configure.in	2009-02-26 15:53:04 +0100
@@ -271,6 +271,12 @@
     DEFAULT_DEFAULT_wxUSE_MSW=1
   ;;
 
+  *-interix*)
+    USE_BSD=1
+    DEFAULT_DEFAULT_wxUSE_GTK=1
+	NEEDS_D_REENTRANT_FOR_R_FUNCS=1
+    ;;
+
   *-pc-msdosdjgpp )
     USE_UNIX=0
     USE_DOS=1
@@ -4385,11 +4391,11 @@
     WX_VERSIONED_SYMBOLS([\$(wx_top_builddir)/version-script])
 
     case "${host}" in
-      *-*-linux* | *-*-gnu* )
+      *-*-linux* | *-*-gnu* | *-winnt* )
         SAMPLES_RPATH_FLAG="-Wl,-rpath,\$(wx_top_builddir)/lib"
         WXCONFIG_RPATH="-Wl,-rpath,\$libdir"
         ;;
-
+	
       *-*-solaris2* )
         if test "$GCC" = yes ; then
             dnl newer versions of gcc need -isystem to compile X headers on
@@ -5552,7 +5558,7 @@
                     flag="$flag -D_RWSTD_MULTI_THREAD"
                 fi
                 ;;
-            *solaris* | alpha*-osf*)
+            *solaris* | alpha*-osf* | *-interix* )
                 flag="-D_REENTRANT"
                 ;;
           esac
