diff -urN ./wxPython-src-2.8.10.1.orig/build/bakefiles/wxpresets/presets/wx.bkl ./wxPython-src-2.8.10.1/build/bakefiles/wxpresets/presets/wx.bkl
--- ./wxPython-src-2.8.10.1.orig/build/bakefiles/wxpresets/presets/wx.bkl	2007-07-04 21:24:53.000000000 +0200
+++ ./wxPython-src-2.8.10.1/build/bakefiles/wxpresets/presets/wx.bkl	2009-07-21 22:25:53.558360790 +0200
@@ -51,12 +51,12 @@
     <set var="__wx_included_impl">0</set>
 
     <if cond="FORMAT in ['autoconf','gnu']">
-        <include file="wx_unix.bkl"/>
+        <include file="wx28_unix.bkl"/>
         <set var="__wx_included_impl">1</set>
     </if>
 
     <if cond="FORMAT!='autoconf' and PLATFORM_WIN32=='1'">
-        <include file="wx_win32.bkl"/>
+        <include file="wx28_win32.bkl"/>
         <set var="__wx_included_impl">1</set>
     </if>
 
