diff -Naur wxPython-src-2.6.3.3-orig/Makefile.in wxPython-src-2.6.3.3/Makefile.in
--- wxPython-src-2.6.3.3-orig/Makefile.in	2007-02-25 17:08:11.000000000 -0600
+++ wxPython-src-2.6.3.3/Makefile.in	2007-02-25 17:20:46.000000000 -0600
@@ -1081,8 +1081,8 @@
 	$(__WXUNIV_DEFINE_p) $(__EXCEPTIONS_DEFINE_p) $(__RTTI_DEFINE_p) \
 	$(__THREAD_DEFINE_p) $(__INC_TIFF_p) $(__INC_JPEG_p) $(__INC_PNG_p) \
 	$(__INC_ZLIB_p) $(__INC_ODBC_p) $(__INC_REGEX_p) $(__INC_EXPAT_p) \
-	-DwxUSE_GUI=0 -DWXUSINGDLL -DWXMAKINGDLL_ODBC $(PIC_FLAG) $(CPPFLAGS) \
-	$(CXXFLAGS)
+	-DwxUSE_GUI=0 -DWXUSINGDLL -DWXMAKINGDLL_ODBC -DSQL_WCHART_CONVERT \
+	$(PIC_FLAG) $(CPPFLAGS) $(CXXFLAGS)
 ODBCDLL_OBJECTS =  \
 	$(__odbcdll___win32rc) \
 	odbcdll_db.o \
