Index: qtjambi-src-gpl-4.4.3_01/com/trolltech/tools/ant/InitializeTask.java
===================================================================
--- qtjambi-src-gpl-4.4.3_01.orig/com/trolltech/tools/ant/InitializeTask.java
+++ qtjambi-src-gpl-4.4.3_01/com/trolltech/tools/ant/InitializeTask.java
@@ -120,10 +120,34 @@ public class InitializeTask extends Task
         this.configuration = configuration;
     }
 
+    public void setQtdir(String qtdir) {
+	this.qtdir = qtdir;
+    }
+
     public String getConfiguration() {
         return configuration;
     }
 
+    public void setSqlite(String sqlite) {
+	this.sqlite = sqlite;
+    }
+
+    public void setPhonon(String phonon) {
+	this.phonon = phonon;
+    }
+
+    public void setOpengl(String opengl) {
+	this.opengl = opengl;
+    }
+
+    public void setWebkit(String webkit) {
+	this.webkit = webkit;
+    }
+
+    public void setXmlpatterns(String xmlpatterns) {
+	this.xmlpatterns = xmlpatterns;
+    }
+
     public void execute() throws BuildException {
         props = PropertyHelper.getPropertyHelper(getProject());
         props.setNewProperty(null, OSNAME, decideOSName());
@@ -323,7 +347,7 @@ public class InitializeTask extends Task
     }
 
     private String decideQtDir() {
-        qtdir = System.getenv("QTDIR");
+        //qtdir = System.getenv("QTDIR");
         if (qtdir == null)
             throw new BuildException("QTDIR environment variable missing");
         if (!new File(qtdir).exists())
@@ -346,8 +370,8 @@ public class InitializeTask extends Task
         StringBuilder path = new StringBuilder();
         path.append(props.getProperty(null, QTDIR));
         path.append("/");
-        path.append(props.getProperty(null, LIBSUBDIR));
-        path.append("/");
+        //path.append(props.getProperty(null, LIBSUBDIR));
+        //path.append("/");
         path.append(LibraryEntry.formatQtName(name, debug, version));
         return new File(path.toString()).exists();
     }
@@ -363,31 +387,36 @@ public class InitializeTask extends Task
     }
 
     private String decideSqlite() {
-        String result = String.valueOf(doesQtPluginExist("qsqlite", "sqldrivers"));
+        //String result = String.valueOf(doesQtPluginExist("qsqlite", "sqldrivers"));
+	String result = sqlite;
         if (verbose) System.out.println(SQLITE + ": " + result);
         return result;
     }
 
     private String decidePhonon() {
-        String result = String.valueOf(doesQtLibExist("phonon", 4));
+        //String result = String.valueOf(doesQtLibExist("phonon", 4));
+	String result = phonon;
         if (verbose) System.out.println(PHONON + ": " + result);
         return result;
     }
 
     private String decideWebkit() {
-        String result = String.valueOf(doesQtLibExist("QtWebKit", 4));
+        //String result = String.valueOf(doesQtLibExist("QtWebKit", 4));
+	String result = webkit;
         if (verbose) System.out.println(WEBKIT + ": " + result);
         return result;
     }
 
     private String decideXMLPatterns() {
-        String result = String.valueOf(doesQtLibExist("QtXmlPatterns", 4));
+        //String result = String.valueOf(doesQtLibExist("QtXmlPatterns", 4));
+	String result = xmlpatterns;
         if (verbose) System.out.println(XMLPATTERNS + ": " + result);
         return result;
     }
 
     private String decideOpenGL() {
-        String result = String.valueOf(doesQtLibExist("QtOpenGL", 4));
+        //String result = String.valueOf(doesQtLibExist("QtOpenGL", 4));
+	String result = opengl;
         if (verbose) System.out.println(OPENGL + ": " + result);
         return result;
     }
@@ -398,6 +427,11 @@ public class InitializeTask extends Task
     private PropertyHelper props;
     private String configuration;
     private boolean debug;
+    private String sqlite;
+    private String phonon;
+    private String xmlpatterns;
+    private String webkit;
+    private String opengl;
     private String qmakespec;
     private String qtdir;
     private String libSubDir;
Index: qtjambi-src-gpl-4.4.3_01/build.xml
===================================================================
--- qtjambi-src-gpl-4.4.3_01.orig/build.xml
+++ qtjambi-src-gpl-4.4.3_01/build.xml
@@ -425,7 +425,9 @@
     <target name="init"
         depends="init.ant, init.taskdef"
         description="Initializes the Qt Jambi build environment." >
-      <qtjambi-initialize verbose="true" configuration="${qtjambi.config}" />
+      <qtjambi-initialize verbose="true" configuration="${qtjambi.config}" qtdir="${gentoo.qtdir}" phonon="${gentoo.phonon}" webkit="${gentoo.webkit}" 
+opengl="${gentoo.opengl}" 
+sqlite="${gentoo.sqlite}" xmlpatterns="${gentoo.xmlpatterns}" />
     </target>
 
     <target name="generator"
