--- generator/main.h.orig
+++ generator/main.h
@@ -91,7 +91,6 @@ struct Preprocess
         if (qtdir.isEmpty()) {
             qWarning("QTDIR environment variable not set. This may cause problems with finding the necessary include files.");
         } else {
-            qtdir += "/include";
             includes << (qtdir + "/QtXml");
             includes << (qtdir + "/QtNetwork");
             includes << (qtdir + "/QtCore");
