--- java_files.orig	2007-06-07 11:30:22.000000000 -0400
+++ java_files	2007-06-07 11:30:30.000000000 -0400
@@ -186,17 +186,6 @@
 ./com/trolltech/qt/gui/QAbstractTextDocumentLayout.java
 ./com/trolltech/qt/gui/QAbstractTextDocumentLayout_PaintContext.java
 ./com/trolltech/qt/gui/QAbstractTextDocumentLayout_Selection.java
-./com/trolltech/qt/gui/QAccessible.java
-./com/trolltech/qt/gui/QAccessible2Interface.java
-./com/trolltech/qt/gui/QAccessibleBridge.java
-./com/trolltech/qt/gui/QAccessibleEvent.java
-./com/trolltech/qt/gui/QAccessibleInterface.java
-./com/trolltech/qt/gui/QAccessibleInterfaceEx.java
-./com/trolltech/qt/gui/QAccessibleObject.java
-./com/trolltech/qt/gui/QAccessibleObjectEx.java
-./com/trolltech/qt/gui/QAccessibleTableInterface.java
-./com/trolltech/qt/gui/QAccessibleWidget.java
-./com/trolltech/qt/gui/QAccessibleWidgetEx.java
 ./com/trolltech/qt/gui/QAction.java
 ./com/trolltech/qt/gui/QActionEvent.java
 ./com/trolltech/qt/gui/QActionGroup.java
