--- designer-integration/pri/jambi.pri
+++ designer-integration/pri/jambi.pri
@@ -33,7 +33,6 @@
 win32:CONFIG += precompile_header
 
 macx{
-    QMAKE_MAC_SDK=/Developer/SDKs/MacOSX10.4u.sdk
     contains(QT_CONFIG, x86):contains(QT_CONFIG, ppc):CONFIG += x86 ppc
     CONFIG -= precompile_header
 }
--- generator_example/generator_example.pro
+++ generator_example/generator_example.pro
@@ -40,7 +40,6 @@
 win32:CONFIG += precompile_header
 
 macx{
-    QMAKE_MAC_SDK=/Developer/SDKs/MacOSX10.4u.sdk
     CONFIG += ppc x86
     CONFIG -= precompile_header
 }
@@ -88,4 +87,4 @@
 mac { 
     CONFIG += ppc x86
     CONFIG -= precompile_header
-}
\ No newline at end of file
+}
--- qtjambi/qtjambi_base.pri
+++ qtjambi/qtjambi_base.pri
@@ -19,7 +19,6 @@
 DEPENDPATH += $$PWD/../qtjambi $$PWD/../common
 
 macx:{
-    QMAKE_MAC_SDK=/Developer/SDKs/MacOSX10.4u.sdk
     LIBS += -framework JavaVM
     QMAKE_EXTENSION_SHLIB = jnilib
 } else {
