--- designer-integration/pri/jambi.pri
+++ designer-integration/pri/jambi.pri
@@ -9,7 +9,7 @@
 }
 
 macx:{
-    LIBS += -framework JavaVm
+    LIBS += -framework JavaVM
 } else {
     INCLUDEPATH += $$(JAVADIR)/include
     win32 {
--- generator_example/generator_example.pro
+++ generator_example/generator_example.pro
@@ -20,7 +20,7 @@
 INCLUDEPATH += ../qtjambi ../common ../include
 
 macx:{
-    LIBS += -framework JavaVm
+    LIBS += -framework JavaVM
     QMAKE_EXTENSION_SHLIB = jnilib
 } else {
     INCLUDEPATH += $(JAVADIR)/include
