--- /var/folders/Pd/Pdnfrt7lFNuAynJB73STUk+++TI/-Tmp-/tmp.HIVIn3acKS	2009-09-05 07:36:05 -0400
+++ a/bin/revdep-rebuild	2009-09-05 07:36:19 -0400
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 # Copyright 1999-2008 Gentoo Foundation
 
 # revdep-rebuild: Reverse dependency rebuilder.
--- /var/folders/Pd/Pdnfrt7lFNuAynJB73STUk+++TI/-Tmp-/tmp.JJY8h9joFj	2009-09-05 07:36:57 -0400
+++ a/data/revdep-rebuild/99revdep-rebuild	2009-09-05 07:37:24 -0400
@@ -16,6 +16,5 @@
 # Note: This file is sourced using bash by the revdep-rebuild script
 
 LD_LIBRARY_MASK="libodbcinst.so libodbc.so libjava.so libjvm.so"
-SEARCH_DIRS="/bin /sbin /usr/bin /usr/sbin /lib* /usr/lib*"
-SEARCH_DIRS_MASK="/lib*/modules"
-
+SEARCH_DIRS="@GENTOO_PORTAGE_EPREFIX@/bin @GENTOO_PORTAGE_EPREFIX@/sbin @GENTOO_PORTAGE_EPREFIX@/usr/bin @GENTOO_PORTAGE_EPREFIX@/usr/sbin @GENTOO_PORTAGE_EPREFIX@/lib* @GENTOO_PORTAGE_EPREFIX@/usr/lib*"
+SEARCH_DIRS_MASK="@GENTOO_PORTAGE_EPREFIX@/lib*/modules"
