--- src/eclean/eclean.old	2008-09-18 18:27:27 +0900
+++ src/eclean/eclean	2008-10-12 18:29:04 +0900
@@ -775,6 +775,8 @@
 	# parse the exclusion file
 	if not 'exclude-file' in myoptions:
 		my_exclude_file = "/etc/%s/%s.exclude" % (__productname__ , myaction)
+		if 'EPREFIX' in os.environ:
+			my_exclude_file = os.environ['EPREFIX'] + my_exclude_file
 		if os.path.isfile(my_exclude_file):
 			myoptions['exclude-file'] = my_exclude_file
 	if 'exclude-file' in myoptions:
