--- /var/folders/Pd/Pdnfrt7lFNuAynJB73STUk+++TI/-Tmp-/tmp.eTwFt57igy	2009-09-05 07:56:13 -0400
+++ a/setup.py	2009-09-05 07:56:26 -0400
@@ -31,13 +31,13 @@
 		'bin/revdep-rebuild'
 	),
 	data_files=(
-		('/etc/env.d', ['data/99gentoolkit-env']),
-		('/etc/revdep-rebuild', ['data/revdep-rebuild/99revdep-rebuild']),
-		('/etc/eclean', [
+		('@GENTOO_PORTAGE_EPREFIX@/etc/env.d', ['data/99gentoolkit-env']),
+		('@GENTOO_PORTAGE_EPREFIX@/etc/revdep-rebuild', ['data/revdep-rebuild/99revdep-rebuild']),
+		('@GENTOO_PORTAGE_EPREFIX@/etc/eclean', [
 			'data/eclean/distfiles.exclude',
 			'data/eclean/packages.exclude'
 		]),
-		('/usr/share/man/man1', [
+		('@GENTOO_PORTAGE_EPREFIX@/usr/share/man/man1', [
 			'man/eclean.1',
 			'man/epkginfo.1',
 			'man/equery.1',
