--- /var/folders/Pd/Pdnfrt7lFNuAynJB73STUk+++TI/-Tmp-/tmp.tISo7uyL3p	2009-09-05 08:56:05 -0400
+++ a/bin/eclean	2009-09-05 08:56:56 -0400
@@ -801,7 +801,7 @@
 			sys.exit(2)
 	# parse the exclusion file
 	if not 'exclude-file' in myoptions:
-		my_exclude_file = "/etc/%s/%s.exclude" % (__productname__ , myaction)
+		my_exclude_file = "@GENTOO_PORTAGE_EPREFIX@/etc/%s/%s.exclude" % (__productname__ , myaction)
 		if os.path.isfile(my_exclude_file):
 			myoptions['exclude-file'] = my_exclude_file
 	if 'exclude-file' in myoptions:
