--- gentoolkit-0.2.3/src/revdep-rebuild/revdep-rebuild
+++ gentoolkit-0.2.3/src/revdep-rebuild/revdep-rebuild
@@ -103,6 +103,21 @@
 			ldd "$1" 2>/dev/null | sort -u | $2 "$3" | awk '{print $ 1}' | tr '\n' ' ' | sed 's/ $//'
 		}
 	;;
+	AIX)
+		SONAME_SUFFIX=".so"
+		function all_missing_libs_f () {
+			echo $(ldd "$1" 2>/dev/null | sort -u | sed -n 's/	\(.*\) => not found/\1/p')
+		}
+		function ldd_f () {
+			ldd "$1"
+		}
+		function required_libs_f () {
+			echo $(ldd --shallow "$1" | sed -n 's/^	//; s/ => .*//; p')
+		}
+		function search_lib_f () {
+			echo $(ldd "$1" 2>/dev/null | sort -u | $2 "$3" | awk '{print $ 1}')
+		}
+	;;
 esac
 
 # Update the incremental variables using /etc/profile.env, /etc/ld.so.conf,
