--- bin/revdep-rebuild.orig	2010-03-16 10:27:35.000000000 -0500
+++ bin/revdep-rebuild	2010-03-16 10:33:58.000000000 -0500
@@ -1063,25 +1063,25 @@
 #	LD_LIBRARY_MASK+=" "$(unset LD_LIBRARY_MASK; portageq envvar LD_LIBRARY_MASK)
 
 	# Add the defaults
-	if [[ -d /etc/revdep-rebuild ]]; then
-		for configfile in /etc/revdep-rebuild/*; do
+	if [[ -d @GENTOO_PORTAGE_EPREFIX@/etc/revdep-rebuild ]]; then
+		for configfile in @GENTOO_PORTAGE_EPREFIX@/etc/revdep-rebuild/*; do
 			SEARCH_DIRS+=" "$(. $configfile; echo $SEARCH_DIRS)
 			SEARCH_DIRS_MASK+=" "$(. $configfile; echo $SEARCH_DIRS_MASK)
 			LD_LIBRARY_MASK+=" "$(. $configfile; echo $LD_LIBRARY_MASK)
 		done
 	else
-		SEARCH_DIRS+=" /bin /sbin /usr/bin /usr/sbin /lib* /usr/lib*"
-		SEARCH_DIRS_MASK+=" /opt/OpenOffice /usr/lib/openoffice"
+		SEARCH_DIRS+=" @GENTOO_PORTAGE_EPREFIX@/bin	@GENTOO_PORTAGE_EPREFIX@/sbin @GENTOO_PORTAGE_EPREFIX@/usr/bin @GENTOO_PORTAGE_EPREFIX@/usr/sbin @GENTOO_PORTAGE_EPREFIX@/lib* @GENTOO_PORTAGE_EPREFIX@/usr/lib*"
+		SEARCH_DIRS_MASK+=" @GENTOO_PORTAGE_EPREFIX@/opt/OpenOffice	@GENTOO_PORTAGE_EPREFIX@/usr/lib/openoffice"
 		LD_LIBRARY_MASK+=" libodbcinst.so libodbc.so libjava.so libjvm.so"
 	fi
 
 	# Get the ROOTPATH and PATH from /etc/profile.env
-	if [[ -r "/etc/profile.env" && -s "/etc/profile.env" ]]; then
-		SEARCH_DIRS+=" "$(. /etc/profile.env; /usr/bin/tr ':' ' ' <<< "$ROOTPATH $PATH")
+	if [[ -r "@GENTOO_PORTAGE_EPREFIX@/etc/profile.env" && -s "@GENTOO_PORTAGE_EPREFIX@/etc/profile.env" ]]; then
+		SEARCH_DIRS+=" "$(. @GENTOO_PORTAGE_EPREFIX@/etc/profile.env; @GENTOO_PORTAGE_EPREFIX@/usr/bin/tr ':' ' ' <<< "$ROOTPATH $PATH")
 	fi
 
 	# Get the directories from /etc/ld.so.conf
-	if [[ -r /etc/ld.so.conf && -s /etc/ld.so.conf ]]; then
+	if [[ -r @GENTOO_PORTAGE_EPREFIX@/etc/ld.so.conf && -s @GENTOO_PORTAGE_EPREFIX@/etc/ld.so.conf ]]; then
 		SEARCH_DIRS+=" "$(parse_ld_so_conf)
 	fi
 
