--- src/equery/equery~	2007-08-09 12:52:06 +0200
+++ src/equery/equery	2007-08-09 12:53:39 +0200
@@ -28,8 +28,12 @@
 except ImportError:
 	sys.path.insert(0, "/Users/pipping/Gentoo/usr/lib/portage/pym")
 	import portage
-import portage_checksum
-from portage_util import unique_array
+try:
+	import portage.checksum as portage_checksum
+	from portage.util import unique_array
+except ImportError:
+	import portage_checksum
+	from portage_util import unique_array
 
 import gentoolkit.pprinter as pp
 from gentoolkit.pprinter import print_info, print_error, print_warn, die
--- src/gentoolkit/helpers.py~	2007-08-09 12:46:26 +0200
+++ src/gentoolkit/helpers.py	2007-08-09 12:47:31 +0200
@@ -10,7 +10,10 @@
 import portage
 from gentoolkit import *
 from package import *
-from portage_util import unique_array
+try:
+	from portage.util import unique_array
+except ImportError:
+	from portage_util import unique_array
 
 def find_packages(search_key, masked=False):
 	"""Returns a list of Package objects that matched the search key."""
--- src/glsa-check/glsa-check~	2007-08-09 12:49:36 +0200
+++ src/glsa-check/glsa-check	2007-08-09 12:51:35 +0200
@@ -306,7 +306,11 @@
 
 # mail mode as requested by solar
 if mode == "mail":
-	import portage_mail, socket
+	try:
+		import portage.mail as portage_mail
+	except ImportError:
+		import portage_mail
+	import socket
 	from StringIO import StringIO
 	
 	# color doesn't make any sense for mail
