Only in gentoolkit-0.2.4.1: prefix.patch
diff -ur gentoolkit-0.2.4.1.orig/src/revdep-rebuild/99revdep-rebuild gentoolkit-0.2.4.1/src/revdep-rebuild/99revdep-rebuild
--- gentoolkit-0.2.4.1.orig/src/revdep-rebuild/99revdep-rebuild	2008-09-03 22:32:33 +0200
+++ gentoolkit-0.2.4.1/src/revdep-rebuild/99revdep-rebuild	2008-09-16 22:34:11 +0200
@@ -16,6 +16,6 @@
 # Note: This file is sourced using bash by the revdep-rebuild script
 
 LD_LIBRARY_MASK="libodbcinst.so libodbc.so libjava.so libjvm.so"
-SEARCH_DIRS="/bin /sbin /usr/bin /usr/sbin /lib* /usr/lib*"
-SEARCH_DIRS_MASK="/lib*/modules"
+SEARCH_DIRS="@GENTOO_PORTAGE_EPREFIX@/bin @GENTOO_PORTAGE_EPREFIX@/sbin @GENTOO_PORTAGE_EPREFIX@/usr/bin @GENTOO_PORTAGE_EPREFIX@/usr/sbin @GENTOO_PORTAGE_EPREFIX@/lib* @GENTOO_PORTAGE_EPREFIX@/usr/lib*"
+SEARCH_DIRS_MASK="@GENTOO_PORTAGE_EPREFIX@/lib*/modules"
 
Only in gentoolkit-0.2.4.1/src/revdep-rebuild: Makefile.orig
Only in gentoolkit-0.2.4.1/src/revdep-rebuild: Makefile.rej
diff -ur gentoolkit-0.2.4.1.orig/src/revdep-rebuild/revdep-rebuild gentoolkit-0.2.4.1/src/revdep-rebuild/revdep-rebuild
--- gentoolkit-0.2.4.1.orig/src/revdep-rebuild/revdep-rebuild	2008-09-03 22:32:33 +0200
+++ gentoolkit-0.2.4.1/src/revdep-rebuild/revdep-rebuild	2008-09-16 22:38:47 +0200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 # Copyright 1999-2008 Gentoo Foundation
 
 # revdep-rebuild: Reverse dependency rebuilder.
Only in gentoolkit-0.2.4.1/src/revdep-rebuild: revdep-rebuild.orig
Only in gentoolkit-0.2.4.1/src/revdep-rebuild: revdep-rebuild.rej
