--- trunk/gtk-sharp/gtk/Gtk.metadata	2008/11/15 20:36:23	118943
+++ trunk/gtk-sharp/gtk/Gtk.metadata	2008/11/15 22:38:07	118944
@@ -559,6 +559,7 @@
   <attr path="/api/namespace/object[@cname='GtkSeparatorToolItem']/property[@name='Draw']" name="new_flag">1</attr>
   <attr path="/api/namespace/object[@cname='GtkSizeGroup']/method[@name='GetWidgets']/return-type" name="element_type">GtkWidget*</attr>
   <attr path="/api/namespace/object[@cname='GtkSpinButton']/constructor[@cname='gtk_spin_button_new_with_range']" name="hidden">1</attr>
+  <attr path="/api/namespace/object[@cname='GtkSpinButton']/signal[@name='Output']" name="manual">true</attr>
   <attr path="/api/namespace/object[@cname='GtkStatusIcon']/method[@name='GetGeometry']/*/*[@name='screen']" name="pass_as">out</attr>
   <attr path="/api/namespace/object[@cname='GtkStatusIcon']/method[@name='GetGeometry']/*/*[@name='area']" name="pass_as">out</attr>
   <attr path="/api/namespace/object[@cname='GtkStatusIcon']/method[@name='SetFromStock']" name="name">SetStock</attr>
