--- tclpkg/gv/Makefile.am~	2007-10-10 03:41:27 +0200
+++ tclpkg/gv/Makefile.am	2007-10-10 03:44:27 +0200
@@ -209,16 +209,16 @@
 if WITH_LUA
 endif
 if WITH_PERL
-	(cd $(DESTDIR)$(pkgperldir);   rm -f gv.so libgv_perl.so libgv_perl.so.0 libgv_perl.la;  mv libgv_perl.so.0.0.0 gv.so;)
+	(cd $(DESTDIR)$(pkgperldir);   rm -f gv.so; $(LN_S) libgv_perl.dylib gv.dylib;)
 endif
 if WITH_PYTHON
-	(cd $(DESTDIR)$(pkgpythondir); rm -f _gv.so libgv_python.so libgv_python.so.0 libgv_python.la; mv libgv_python.so.0.0.0 _gv.so;)
+	(cd $(DESTDIR)$(pkgpythondir); rm -f _gv.so; $(LN_S) libgv_python.dylib _gv.dylib;)
 endif
 if WITH_RUBY
-	(cd $(DESTDIR)$(pkgrubydir);   rm -f gv.so libgv_ruby.so libgv_ruby.so.0 libgv_ruby.la;  mv libgv_ruby.so.0.0.0 gv.so;)
+	(cd $(DESTDIR)$(pkgrubydir);   rm -f gv.so; $(LN_S) libgv_ruby.dylib gv.dylib;)
 endif
 if WITH_TCL
-	(cd $(DESTDIR)$(pkgtcldir);    rm -f gv.so;  $(LN_S) libgv_tcl.so gv.so;)
+	(cd $(DESTDIR)$(pkgtcldir);    rm -f gv.so;  $(LN_S) libgv_tcl.dylib gv.dylib;)
 endif
 
 uninstall-hook:
