diff -Naur graphviz-2.16.1.orig/configure.ac graphviz-2.16.1/configure.ac
--- graphviz-2.16.1.orig/configure.ac	2007-12-16 21:05:21.000000000 +0100
+++ graphviz-2.16.1/configure.ac	2007-12-16 21:08:59.000000000 +0100
@@ -760,6 +760,7 @@
   PERL_ARCHLIB=`$PERL -e 'use Config; print $Config{archlib};'`
   PERL_INCLUDES=-I$PERL_ARCHLIB/CORE
   PERL_LIBS="-L$PERL_ARCHLIB/CORE -lperl"
+  PERL_INSTALL_DIR="`$PERL -e 'use Config; print $Config{vendorarch};'`"
   save_CPPFLAGS=$CPPFLAGS
   CPPFLAGS="$CPPFLAGS $PERL_INCLUDES"
   AC_CHECK_HEADER(EXTERN.h,,[
@@ -770,6 +771,7 @@
 fi
 AC_SUBST(PERL_INCLUDES)
 AC_SUBST(PERL_LIBS)
+AC_SUBST(PERL_INSTALL_DIR)
 fi
 fi
 AM_CONDITIONAL(WITH_PERL, [test "x$PERL" != "x"])
@@ -844,6 +846,7 @@
     PYTHON_PREFIX=`$PYTHON -c "import sys; print sys.prefix"`
     PYTHON_INCLUDES=-I$PYTHON_PREFIX/include/python$PYTHON_VERSION_SHORT
     PYTHON_LIBS="-L$PYTHON_PREFIX/lib$LIBPOSTFIX/python$PYTHON_VERSION_SHORT/config"
+    PYTHON_INSTALL_DIR="`$PYTHON $srcdir/config/config_python.py archsitelib`"
     save_CPPFLAGS=$CPPFLAGS
     CPPFLAGS="$CPPFLAGS $PYTHON_INCLUDES"
     AC_CHECK_HEADER(Python.h,,[
@@ -855,6 +858,7 @@
 fi
 AC_SUBST(PYTHON_INCLUDES)
 AC_SUBST(PYTHON_LIBS)
+AC_SUBST(PYTHON_INSTALL_DIR)
 fi
 fi
 fi
@@ -1059,6 +1063,7 @@
 if test "x$RUBY" != "x"; then
   RUBY_INCLUDES="-I`$RUBY $srcdir/config/config_ruby.rb INCLUDES`"
   RUBY_LIBS="-L`$RUBY $srcdir/config/config_ruby.rb lib` `$RUBY -rrbconfig -e \"puts Config::CONFIG[['LIBRUBYARG_SHARED']]\"`"
+  RUBY_INSTALL_DIR="`$RUBY $srcdir/config/config_ruby.rb archsitelib`"
   save_CPPFLAGS=$CPPFLAGS
   CPPFLAGS="$CPPFLAGS $RUBY_INCLUDES"
   AC_CHECK_HEADER(ruby.h,,[
@@ -1069,6 +1074,7 @@
 fi
 AC_SUBST(RUBY_INCLUDES)
 AC_SUBST(RUBY_LIBS)
+AC_SUBST(RUBY_INSTALL_DIR)
 fi
 fi
 fi
diff -Naur graphviz-2.16.1.orig/tclpkg/gv/Makefile.am graphviz-2.16.1/tclpkg/gv/Makefile.am
--- graphviz-2.16.1.orig/tclpkg/gv/Makefile.am	2007-12-16 21:05:21.000000000 +0100
+++ graphviz-2.16.1/tclpkg/gv/Makefile.am	2007-12-16 21:08:13.000000000 +0100
@@ -98,7 +98,7 @@
 gv.cmxa gv.a: gv.cma
 	ocamlopt -a -o gv.cmxa
 
-pkgperldir = $(pkglibdir)/perl
+pkgperldir = @PERL_INSTALL_DIR@
 PERL_data = gv.pm
 nodist_libgv_perl_la_SOURCES = gv_perl.cpp $(PERL_data)
 libgv_perl_la_SOURCES = $(BASESOURCES) gv_dummy_init.cpp
@@ -118,7 +118,7 @@
 gv_php.cpp: gv.i
 	$(SWIG) -c++ -php5 -o gv_php.cpp $(srcdir)/gv.i
 
-pkgpythondir = $(pkglibdir)/python
+pkgpythondir = @PYTHON_INSTALL_DIR@
 PYTHON_data = gv.py
 nodist_libgv_python_la_SOURCES = gv_python.cpp $(PYTHON_data)
 libgv_python_la_SOURCES = $(BASESOURCES) gv_dummy_init.cpp
@@ -155,7 +155,7 @@
 gv_r.cpp: gv.i
 	$(SWIG) -c++ -r -o gv_r.cpp $(srcdir)/gv.i
 
-pkgrubydir = $(pkglibdir)/ruby
+pkgrubydir = @RUBY_INSTALL_DIR@
 RUBY_data = 
 nodist_libgv_ruby_la_SOURCES = gv_ruby.cpp $(RUBY_data)
 libgv_ruby_la_SOURCES = $(BASESOURCES) gv_dummy_init.cpp
@@ -306,15 +306,24 @@
 	-groff -Tps -man $< | ps2pdf - - >$@
 
 install-data-hook:
+if WITH_LUA
 	(cd $(DESTDIR)$(pkgluadir);    rm -f gv.so;  $(LN_S) libgv_lua.so gv.so;)
+endif
+if WITH_PERL
 	(cd $(DESTDIR)$(pkgperldir);   rm -f gv.so;  $(LN_S) libgv_perl.so gv.so;)
+endif
+if WITH_PYTHON
 	(cd $(DESTDIR)$(pkgpythondir); rm -f _gv.so; $(LN_S) libgv_python.so _gv.so;)
 	(cd $(DESTDIR)$(pkgpython23dir); rm -f _gv.so; $(LN_S) libgv_python23.so _gv.so;)
 	(cd $(DESTDIR)$(pkgpython24dir); rm -f _gv.so; $(LN_S) libgv_python24.so _gv.so;)
 	(cd $(DESTDIR)$(pkgpython25dir); rm -f _gv.so; $(LN_S) libgv_python25.so _gv.so;)
-	(cd $(DESTDIR)$(pkgrdir);   rm -f gv.so;  $(LN_S) libgv_r.so gv.so;)
+endif
+if WITH_RUBY
 	(cd $(DESTDIR)$(pkgrubydir);   rm -f gv.so;  $(LN_S) libgv_ruby.so gv.so;)
+endif
+if WITH_TCL
 	(cd $(DESTDIR)$(pkgtcldir);    rm -f gv.so;  $(LN_S) libgv_tcl.so gv.so;)
+endif
 
 uninstall-hook:
 	(cd $(DESTDIR)$(pkgluadir);    rm -f gv.so;)
