diff -ru graphviz-2.12.orig/contrib/diffimg/diffimg.c graphviz-2.12/contrib/diffimg/diffimg.c
--- graphviz-2.12.orig/contrib/diffimg/diffimg.c	2008-03-07 16:00:06 +0100
+++ graphviz-2.12/contrib/diffimg/diffimg.c	2008-03-07 16:04:40 +0100
@@ -36,7 +36,16 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
-#include <sysexits.h>
+
+#ifndef __INTERIX
+# include <sysexits.h>
+#else
+# define EX_USAGE 1
+# define EX_NOINPUT 1
+# define EX_UNAVAILABLE 1
+# define EX_DATAERR 1
+#endif
+
 #include <gd.h>
 #if defined HAVE_STDBOOL_H && ! defined __cplusplus
 #include <stdbool.h>
