--- configure.ac.orig	2007-07-16 21:34:31 +0200
+++ configure.ac	2007-07-16 21:35:33 +0200
@@ -617,11 +617,7 @@
 dnl Checks for libwww and OpenSSL:
 AC_MSG_CHECKING(whether we can find OpenSSL)
 dnl find the ssl library dir (empirical)
-if test -d '/usr/local/ssl/lib'; then
-  ssllib="-L/usr/local/ssl/lib -lssl -lcrypto"
-else
-  ssllib="-L/usr/lib -lssl -lcrypto"
-fi
+ssllib=`pkg-config --libs openssl`
 dnl find the ssl include dir (empirical)
 ssldir=`pkg-config --cflags openssl`
 WWWSSL=""
