diff -ru -x '*.Po' -x '*.Plo' w3c-libwww-5.4.0.orig/Library/src/HTEvtLst.c w3c-libwww-5.4.0/Library/src/HTEvtLst.c
--- w3c-libwww-5.4.0.orig/Library/src/HTEvtLst.c	2008-04-17 08:39:19 +0200
+++ w3c-libwww-5.4.0/Library/src/HTEvtLst.c	2008-04-17 09:04:33 +0200
@@ -550,9 +550,11 @@
 #ifndef WWW_WIN_ASYNC
     MaxSock = 0 ;
     HTTRACE(THD_TRACE, "Event....... New value for MaxSock is %d\n" _ MaxSock);
-    FD_ZERO(FdArray+HTEvent_INDEX(HTEvent_READ));
-    FD_ZERO(FdArray+HTEvent_INDEX(HTEvent_WRITE));
-    FD_ZERO(FdArray+HTEvent_INDEX(HTEvent_OOB));
+	/* WARNING: Interix' FD_ZERO is broken before version 6.0
+	 * and needs extra braces to not produce vompile failures! */
+    FD_ZERO((FdArray+HTEvent_INDEX(HTEvent_READ)));
+    FD_ZERO((FdArray+HTEvent_INDEX(HTEvent_WRITE)));
+    FD_ZERO((FdArray+HTEvent_INDEX(HTEvent_OOB)));
 #endif /* !WWW_WIN_ASYNC */
 
     EventOrder_deleteAll();
