Index: w3c-libwww-5.4.0/Library/src/Makefile.am
===================================================================
--- w3c-libwww-5.4.0.orig/Library/src/Makefile.am
+++ w3c-libwww-5.4.0/Library/src/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with Automake to create Makefile.in.
 
-LDFLAGS = -version-info "1:0:1"
+LDFLAGS := $(LDFLAGS) -version-info "1:0:1"
 
 lib_LTLIBRARIES = \
 	libwwwutils.la \
Index: w3c-libwww-5.4.0/Library/src/SSL/Makefile.am
===================================================================
--- w3c-libwww-5.4.0.orig/Library/src/SSL/Makefile.am
+++ w3c-libwww-5.4.0/Library/src/SSL/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with Automake to create Makefile.in.
 
-LDFLAGS = -version-info "1:0:1"
+LDFLAGS := $(LDFLAGS) -version-info "1:0:1"

 lib_LTLIBRARIES = @WWWSSL@
 
Index: w3c-libwww-5.4.0/modules/expat/xmlparse/Makefile.am
===================================================================
--- w3c-libwww-5.4.0.orig/modules/expat/xmlparse/Makefile.am
+++ w3c-libwww-5.4.0/modules/expat/xmlparse/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with Automake to create Makefile.in.
 
-LDFLAGS = -version-info "1:0:1"
+LDFLAGS := $(LDFLAGS) -version-info "1:0:1"
 
 CFLAGS = @CFLAGS@ -DXML_NS
 
Index: w3c-libwww-5.4.0/modules/expat/xmltok/Makefile.am
===================================================================
--- w3c-libwww-5.4.0.orig/modules/expat/xmltok/Makefile.am
+++ w3c-libwww-5.4.0/modules/expat/xmltok/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with Automake to create Makefile.in.
 
-LDFLAGS = -version-info "1:0:1"
+LDFLAGS := $(LDFLAGS) -version-info "1:0:1"
 
 CFLAGS = @CFLAGS@ -DXML_NS
 
