--- screen-4.0.3_p20070403/config.h.in
+++ screen-4.0.3_p20070403/config.h.in
@@ -549,6 +549,16 @@
 #undef USESETENV
 
 /*
+ * If setenv() takes 3 arguments define HAVE_SETENV_3
+ */
+#undef HAVE_SETENV_3
+
+/*
+ * If setenv() takes 2 arguments define HAVE_SETENV_2
+ */
+#undef HAVE_SETENV_2
+
+/*
  * If your system does not come with a setenv()/putenv()/getenv()
  * functions, you may bring in our own code by defining NEEDPUTENV.
  */
--- screen-4.0.3_p20070403/configure.in
+++ screen-4.0.3_p20070403/configure.in
@@ -1201,11 +1201,32 @@
 AC_HEADER_DIRENT
 
 AC_MSG_CHECKING(for setenv)
-AC_TRY_LINK(,[setenv((char *)0,(char *)0);unsetenv((char *)0);], AC_MSG_RESULT(yes);AC_DEFINE(USESETENV),
-AC_MSG_RESULT(no)
-AC_MSG_CHECKING(for putenv)
-AC_TRY_LINK(,[putenv((char *)0);unsetenv((char *)0);], AC_MSG_RESULT(yes) ,  AC_MSG_RESULT(no);AC_DEFINE(NEEDPUTENV)
-))
+if test -z "$ac_setenv_args"; then
+    AC_TRY_LINK(
+    [#include <stdlib.h>],
+    [
+        setenv((char *) 0, (char *) 0, 0);
+    ], ac_setenv_args=3)
+fi
+if test -z "$ac_setenv_args"; then
+    AC_TRY_LINK(
+    [#include <stdlib.h>],
+    [
+        setenv((char *) 0, (char *) 0);
+    ], ac_setenv_args=2)
+fi
+if test -n "$ac_setenv_args"; then
+    AC_DEFINE(USESETENV)
+    if test "$ac_setenv_args" = 3; then
+        AC_DEFINE(HAVE_SETENV_3)
+    elif test "$ac_setenv_args" = 2; then
+        AC_DEFINE(HAVE_SETENV_2)
+    fi
+else
+    AC_MSG_RESULT(no)
+    AC_MSG_CHECKING(for putenv)
+    AC_TRY_LINK(,[putenv((char *)0);unsetenv((char *)0);], AC_MSG_RESULT(yes) ,  AC_MSG_RESULT(no);AC_DEFINE(NEEDPUTENV))
+fi
 AC_MSG_CHECKING([for nl_langinfo(CODESET)])
 AC_TRY_LINK([
 #include <langinfo.h>
--- screen-4.0.3_p20070403/misc.c
+++ screen-4.0.3_p20070403/misc.c
@@ -613,11 +613,13 @@
    */
 # endif /* NEEDSETENV */
 #else /* USESETENV */
-# if defined(linux) || defined(__convex__) || (BSD >= 199103) || defined(_AIX) || defined(__GLIBC__)
+# if HAVE_SETENV_3
   setenv(var, value, 1);
-# else
+# elif HAVE_SETENV_2
   setenv(var, value);
-# endif /* linux || convex || BSD >= 199103 */
+# else
+#  error have setenv but not 2 or 3 args?
+# endif /* HAVE_SETENV_3 */
 #endif /* USESETENV */
 }
 
