--- src/eixTk/ptr_list.h
+++ src/eixTk/ptr_list.h
@@ -47,7 +47,7 @@
 	/// A list that only stores pointers to type.
 	template<typename type>
 	class ptr_list
-		: virtual public std::list<type*>
+		: public std::list<type*>
 	{
 		public:
 			using std::list<type*>::begin;
--- src/output/formatstring.cc
+++ src/output/formatstring.cc
@@ -278,7 +278,7 @@
 FormatParser::ParserState
 FormatParser::state_COLOR()
 {
-	char *q = strchr(band_position, ')');
+	const char *q = strchr(band_position, ')');
 	if(q == NULL) {
 		last_error = _("'(' without closing ')'");
 		return ERROR;
@@ -299,7 +299,7 @@
 FormatParser::ParserState
 FormatParser::state_PROPERTY()
 {
-	char *q = strchr(band_position, '>');
+	const char *q = strchr(band_position, '>');
 	if(q == NULL) {
 		last_error = _("'<' without closing '>'");
 		return ERROR;
