commit 0e33a46a1a15c60b6e736122d36c44bbe0a15789
Author: Marti Maria <info@littlecms.com>
Date:   Tue Jun 28 18:26:59 2011 +0200

    Some architecturees does need aligned bus adresses (sparc) Fixed

diff --git a/testbed/testcms2.c b/testbed/testcms2.c
index 9a96d17..13fd7fc 100644
--- a/testbed/testcms2.c
+++ b/testbed/testcms2.c
@@ -73,10 +73,10 @@ static cmsUInt32Number SingleHit, MaxAllocated=0, TotalMemory=0;
 // I'm hidding the size before the block. This is a well-known technique and probably the blocks coming from
 // malloc are built in a way similar to that, but I do on my own to be portable.
 typedef struct { 
-    cmsUInt32Number KeepSize;
-    cmsUInt32Number Align8;
-    cmsContext WhoAllocated; // Some systems do need pointers aligned to 8-byte boundaries.
-
+    cmsUInt32Number KeepSize;    
+    cmsContext      WhoAllocated; 
+    cmsUInt32Number Align8_1;             // Some systems do need pointers aligned to 8-byte boundaries.
+    cmsUInt32Number Align8_2;        
 } _cmsMemoryBlock;
 
 #define SIZE_OF_MEM_HEADER (sizeof(_cmsMemoryBlock))
