diff -ru parity-1.2.2.orig/parity.utils/Context.cpp parity-1.2.2/parity.utils/Context.cpp
--- parity-1.2.2.orig/parity.utils/Context.cpp	2009-06-09 17:26:22 +0200
+++ parity-1.2.2/parity.utils/Context.cpp	2009-06-09 17:27:14 +0200
@@ -239,7 +239,7 @@
 			} else {
 				Path direct(name);
 				direct.toNative();
-				if(direct.exists())
+				if(direct.exists() && direct.isFile())
 					return direct;
 
 				names.push_back(name);
@@ -270,7 +270,7 @@
 #if 0
 						if(pth.exists() && !isBadLinkerPath(pth))
 #else
-						if(pth.exists())
+						if(pth.exists() && pth.isFile())
 #endif
 							return pth;
 					}
