diff -ru parity-1.1.0.orig/parity.tasks/LoaderGenerator.cpp parity-1.1.0/parity.tasks/LoaderGenerator.cpp
--- parity-1.1.0.orig/parity.tasks/LoaderGenerator.cpp	2008-08-27 15:10:45 +0200
+++ parity-1.1.0/parity.tasks/LoaderGenerator.cpp	2008-08-27 15:11:22 +0200
@@ -619,15 +619,15 @@
 						{
 						case utils::SubsystemWindowsCui:
 							if(ctx.getBackendType() == utils::ToolchainInterixMS) {
-								entry = "___DllMainCRTStartup@12";
+								entry = "__DllMainCRTStartup@12";
 								break;
 							}
 						case utils::SubsystemWindowsGui:
 						case utils::SubsystemWindowsCeGui:
-							entry = "__DllMainCRTStartup@12";
+							entry = "_DllMainCRTStartup@12";
 							break;
 						case utils::SubsystemPosixCui:
-							entry = "___DllMainCRTStartup@12";
+							entry = "__DllMainCRTStartup@12";
 							break;
 						default:
 							throw utils::Exception("subsystem default entry point not implemented!");
@@ -640,16 +640,16 @@
 						{
 						case utils::SubsystemWindowsCui:
 							if(ctx.getBackendType() == utils::ToolchainInterixMS)
-								entry = "___MixedModeProcessStartup";
+								entry = "__MixedModeProcessStartup";
 							else
-								entry = "_mainCRTStartup";
+								entry = "mainCRTStartup";
 							break;
 						case utils::SubsystemWindowsGui:
 						case utils::SubsystemWindowsCeGui:
-							entry = "_WinMainCRTStartup";
+							entry = "WinMainCRTStartup";
 							break;
 						case utils::SubsystemPosixCui:
-							entry = "___PosixProcessStartup";
+							entry = "__PosixProcessStartup";
 							break;
 						default:
 							throw utils::Exception("subsystem default entry point not implemented!");
@@ -711,6 +711,7 @@
 				}
 
 				ctx.setEntryPoint("ParityLoaderGeneratedEntry");
+				ctx.setSharedEntryPoint("ParityLoaderGeneratedEntry");
 			}
 
 			//
