Index: parity.tasks/MsCompiler.cpp
===================================================================
--- parity.tasks/MsCompiler.cpp	(revision 197)
+++ parity.tasks/MsCompiler.cpp	(working copy)
@@ -114,6 +114,11 @@
 			if(ctx.getOptimizeLevel() <= 0 && ctx.getInlineFunctions())
 				vec.push_back("/Ob2");
 
+			if(!ctx.getOmitFramePointer())
+				vec.push_back("/Oy-");
+			else if(ctx.getOmitFramePointer() && ctx.getOptimizeLevel() < 1)
+				vec.push_back("/Oy");
+
 			//
 			// Decide about microsoft extensions.
 			//
Index: parity.loader/LoaderLog.c
===================================================================
--- parity.loader/LoaderLog.c	(revision 197)
+++ parity.loader/LoaderLog.c	(working copy)
@@ -97,7 +97,10 @@
 			case 's':
 				{
 					const char* tmp = va_arg(args, const char*);
-					lstrcat(buffer, tmp);
+					if(tmp)
+						lstrcat(buffer, tmp);
+					else
+						lstrcat(buffer, "(null)");
 				}
 				break;
 			case 'c':
@@ -109,8 +112,7 @@
 				break;
 			default:
 				{
-					const char msgErrUnsupported[] = "<unsupported specifier>";
-					lstrcat(buffer, msgErrUnsupported);
+					lstrcat(buffer, "<unsupported specifier>");
 				}
 			}
 		}
