Index: parity.loader/loaderlog.c
===================================================================
--- parity.loader/loaderlog.c	(revision 111)
+++ parity.loader/loaderlog.c	(working copy)
@@ -182,7 +182,7 @@
 		{
 			LogWarning("cannot read LD_DEBUG_OUTPUT variable!\n");
 		} else {
-			HANDLE handle = CreateFile(ptrOut, FILE_WRITE_ACCESS, FILE_SHARE_READ | FILE_SHARE_WRITE, 0, OPEN_ALWAYS, FILE_ATTRIBUTE_NORMAL, 0);
+			HANDLE handle = CreateFile(ptrOut, GENERIC_WRITE, FILE_SHARE_READ | FILE_SHARE_WRITE, 0, OPEN_ALWAYS, FILE_ATTRIBUTE_NORMAL, 0);
 			
 			if(handle == INVALID_HANDLE_VALUE) {
 				LogWarning("cannot open debug output file %s!\n", ptrOut);
