--- gettext-0.14.2/gettext-tools/tests/Makefile.am
+++ gettext-0.14.2/gettext-tools/tests/Makefile.am
@@ -53,7 +53,6 @@
 	msgmerge-update-1 msgmerge-update-2 msgmerge-update-3 \
 	msgunfmt-1 \
 	msgunfmt-csharp-1 \
-	msgunfmt-java-1 \
 	msgunfmt-properties-1 \
 	msgunfmt-tcl-1 \
 	msguniq-1 msguniq-2 msguniq-3 msguniq-4 \
@@ -66,8 +65,6 @@
 	xgettext-csharp-4 xgettext-csharp-5 \
 	xgettext-elisp-1 \
 	xgettext-glade-1 xgettext-glade-2 xgettext-glade-3 \
-	xgettext-java-1 xgettext-java-2 xgettext-java-3 xgettext-java-4 \
-	xgettext-java-5 \
 	xgettext-librep-1 \
 	xgettext-lisp-1 \
 	xgettext-objc-1 xgettext-objc-2 \
@@ -88,7 +85,6 @@
 	format-csharp-1 format-csharp-2 \
 	format-elisp-1 format-elisp-2 \
 	format-gcc-internal-1 format-gcc-internal-2 \
-	format-java-1 format-java-2 \
 	format-librep-1 format-librep-2 \
 	format-lisp-1 format-lisp-2 \
 	format-php-1 format-php-2 \
@@ -105,7 +101,7 @@
 	plural-1 plural-2 \
 	lang-c lang-c++ lang-objc lang-sh lang-bash lang-python-1 \
 	lang-python-2 lang-clisp lang-elisp lang-librep lang-guile \
-	lang-smalltalk lang-java lang-csharp lang-gawk lang-pascal \
+	lang-smalltalk lang-csharp lang-gawk lang-pascal \
 	lang-ycp lang-tcl lang-perl-1 lang-perl-2 lang-php lang-po lang-rst
 
 EXTRA_DIST += $(TESTS) \
--- gettext-0.14.2/gettext-tools/tests/Makefile.in
+++ gettext-0.14.2/gettext-tools/tests/Makefile.in
@@ -397,7 +397,6 @@
 	msgmerge-update-1 msgmerge-update-2 msgmerge-update-3 \
 	msgunfmt-1 \
 	msgunfmt-csharp-1 \
-	msgunfmt-java-1 \
 	msgunfmt-properties-1 \
 	msgunfmt-tcl-1 \
 	msguniq-1 msguniq-2 msguniq-3 msguniq-4 \
@@ -410,8 +409,6 @@
 	xgettext-csharp-4 xgettext-csharp-5 \
 	xgettext-elisp-1 \
 	xgettext-glade-1 xgettext-glade-2 xgettext-glade-3 \
-	xgettext-java-1 xgettext-java-2 xgettext-java-3 xgettext-java-4 \
-	xgettext-java-5 \
 	xgettext-librep-1 \
 	xgettext-lisp-1 \
 	xgettext-objc-1 xgettext-objc-2 \
@@ -432,7 +429,6 @@
 	format-csharp-1 format-csharp-2 \
 	format-elisp-1 format-elisp-2 \
 	format-gcc-internal-1 format-gcc-internal-2 \
-	format-java-1 format-java-2 \
 	format-librep-1 format-librep-2 \
 	format-lisp-1 format-lisp-2 \
 	format-php-1 format-php-2 \
@@ -449,7 +445,7 @@
 	plural-1 plural-2 \
 	lang-c lang-c++ lang-objc lang-sh lang-bash lang-python-1 \
 	lang-python-2 lang-clisp lang-elisp lang-librep lang-guile \
-	lang-smalltalk lang-java lang-csharp lang-gawk lang-pascal \
+	lang-smalltalk lang-csharp lang-gawk lang-pascal \
 	lang-ycp lang-tcl lang-perl-1 lang-perl-2 lang-php lang-po lang-rst
 
 
