diff -Nura gettext-0.14.1/gettext-tools/tests/Makefile.am gettext-0.14.1.sparc/gettext-tools/tests/Makefile.am
--- gettext-0.14.1/gettext-tools/tests/Makefile.am	2004-01-15 08:30:24.000000000 -0300
+++ gettext-0.14.1.sparc/gettext-tools/tests/Makefile.am	2004-10-05 23:10:19.000000000 -0300
@@ -52,7 +52,6 @@
 	msgmerge-update-1 msgmerge-update-2 msgmerge-update-3 \
 	msgunfmt-1 \
 	msgunfmt-csharp-1 \
-	msgunfmt-java-1 \
 	msgunfmt-properties-1 \
 	msgunfmt-tcl-1 \
 	msguniq-1 msguniq-2 msguniq-3 msguniq-4 \
@@ -65,8 +64,6 @@
 	xgettext-csharp-4 xgettext-csharp-5 \
 	xgettext-elisp-1 \
 	xgettext-glade-1 xgettext-glade-2 xgettext-glade-3 \
-	xgettext-java-1 xgettext-java-2 xgettext-java-3 xgettext-java-4 \
-	xgettext-java-5 \
 	xgettext-librep-1 \
 	xgettext-lisp-1 \
 	xgettext-objc-1 xgettext-objc-2 \
@@ -86,7 +83,6 @@
 	format-csharp-1 format-csharp-2 \
 	format-elisp-1 format-elisp-2 \
 	format-gcc-internal-1 format-gcc-internal-2 \
-	format-java-1 format-java-2 \
 	format-librep-1 format-librep-2 \
 	format-lisp-1 format-lisp-2 \
 	format-php-1 format-php-2 \
@@ -102,7 +98,7 @@
 	plural-1 plural-2 \
 	lang-c lang-c++ lang-objc lang-sh lang-bash lang-python-1 \
 	lang-python-2 lang-clisp lang-elisp lang-librep lang-smalltalk \
-	lang-java lang-csharp lang-gawk lang-pascal lang-ycp lang-tcl \
+	lang-csharp lang-gawk lang-pascal lang-ycp lang-tcl \
 	lang-perl-1 lang-perl-2 lang-php lang-po lang-rst
 
 EXTRA_DIST += $(TESTS) \
diff -Nura gettext-0.14.1/gettext-tools/tests/Makefile.in gettext-0.14.1.sparc/gettext-tools/tests/Makefile.in
--- gettext-0.14.1/gettext-tools/tests/Makefile.in	2004-01-29 14:13:39.000000000 -0300
+++ gettext-0.14.1.sparc/gettext-tools/tests/Makefile.in	2004-10-05 23:10:06.000000000 -0300
@@ -381,7 +381,6 @@
 	msgmerge-update-1 msgmerge-update-2 msgmerge-update-3 \
 	msgunfmt-1 \
 	msgunfmt-csharp-1 \
-	msgunfmt-java-1 \
 	msgunfmt-properties-1 \
 	msgunfmt-tcl-1 \
 	msguniq-1 msguniq-2 msguniq-3 msguniq-4 \
@@ -394,8 +393,6 @@
 	xgettext-csharp-4 xgettext-csharp-5 \
 	xgettext-elisp-1 \
 	xgettext-glade-1 xgettext-glade-2 xgettext-glade-3 \
-	xgettext-java-1 xgettext-java-2 xgettext-java-3 xgettext-java-4 \
-	xgettext-java-5 \
 	xgettext-librep-1 \
 	xgettext-lisp-1 \
 	xgettext-objc-1 xgettext-objc-2 \
@@ -415,7 +412,6 @@
 	format-csharp-1 format-csharp-2 \
 	format-elisp-1 format-elisp-2 \
 	format-gcc-internal-1 format-gcc-internal-2 \
-	format-java-1 format-java-2 \
 	format-librep-1 format-librep-2 \
 	format-lisp-1 format-lisp-2 \
 	format-php-1 format-php-2 \
@@ -431,7 +427,7 @@
 	plural-1 plural-2 \
 	lang-c lang-c++ lang-objc lang-sh lang-bash lang-python-1 \
 	lang-python-2 lang-clisp lang-elisp lang-librep lang-smalltalk \
-	lang-java lang-csharp lang-gawk lang-pascal lang-ycp lang-tcl \
+	lang-csharp lang-gawk lang-pascal lang-ycp lang-tcl \
 	lang-perl-1 lang-perl-2 lang-php lang-po lang-rst
 
 
