diff -ru gettext-0.17.orig/gettext-runtime/gnulib-m4/include_next.m4 gettext-0.17/gettext-runtime/gnulib-m4/include_next.m4
--- gettext-0.17.orig/gettext-runtime/gnulib-m4/include_next.m4	2009-09-14 11:59:18 +0200
+++ gettext-0.17/gettext-runtime/gnulib-m4/include_next.m4	2009-09-14 11:36:06 +0200
@@ -89,12 +89,12 @@
 	    dnl so use subshell.
 	    AS_VAR_SET([gl_next_header],
 	      ['"'`(eval "$ac_cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD |
-	       sed -n '\#[\\/]]m4_quote(m4_defn([gl_HEADER_NAME]))[#{
+	       ( sed -n '\#[\\/]]m4_quote(m4_defn([gl_HEADER_NAME]))[#{
 		 s#.*"\(.*[\\/]]m4_quote(m4_defn([gl_HEADER_NAME]))[\)".*#\1#
 		 s#^/[^/]#//&#
 		 p
 		 q
-	       }'`'"'])
+	       }'; cat > /dev/null; );`'"'])
 	  else
 	    AS_VAR_SET([gl_next_header], ['<'gl_HEADER_NAME'>'])
 	  fi
diff -ru gettext-0.17.orig/gettext-tools/gnulib-m4/include_next.m4 gettext-0.17/gettext-tools/gnulib-m4/include_next.m4
--- gettext-0.17.orig/gettext-tools/gnulib-m4/include_next.m4	2009-09-14 11:58:33 +0200
+++ gettext-0.17/gettext-tools/gnulib-m4/include_next.m4	2009-09-14 11:59:53 +0200
@@ -89,12 +89,12 @@
 	    dnl so use subshell.
 	    AS_VAR_SET([gl_next_header],
 	      ['"'`(eval "$ac_cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD |
-	       sed -n '\#[\\/]]m4_quote(m4_defn([gl_HEADER_NAME]))[#{
+	       ( sed -n '\#[\\/]]m4_quote(m4_defn([gl_HEADER_NAME]))[#{
 		 s#.*"\(.*[\\/]]m4_quote(m4_defn([gl_HEADER_NAME]))[\)".*#\1#
 		 s#^/[^/]#//&#
 		 p
 		 q
-	       }'`'"'])
+	       }'; cat > /dev/null; );`'"'])
 	  else
 	    AS_VAR_SET([gl_next_header], ['<'gl_HEADER_NAME'>'])
 	  fi
