--- configure.in
+++ configure.in
@@ -322,14 +322,14 @@
 AC_MSG_CHECKING([for zlib installation])
 AC_ARG_WITH([zlib],
 [  --with-zlib=DIR	  path to directory containing zlib library (default=
-			  /usr/local or /usr if not found in /usr/local)],
+			  @GENTOO_PORTAGE_EPREFIX@/usr or /usr if not found in @GENTOO_PORTAGE_EPREFIX@/usr)],
 [
 if test "$withval"; then
   ZLIB_HOME="$withval"
   AC_MSG_RESULT([using $ZLIB_HOME])
 fi
 ], [
-ZLIB_HOME=/usr/local
+ZLIB_HOME=@GENTOO_PORTAGE_EPREFIX@/usr
 if test ! -f "$ZLIB_HOME/include/zlib.h"
 then
   ZLIB_HOME=/usr
