diff -ru clamav-0.92.1.orig/configure.in clamav-0.92.1/configure.in
--- clamav-0.92.1.orig/configure.in	2008-03-10 11:07:57 +0100
+++ clamav-0.92.1/configure.in	2008-03-10 12:48:59 +0100
@@ -239,14 +239,14 @@
 AC_MSG_CHECKING(for zlib installation)
 AC_ARG_WITH(zlib,
 [  --with-zlib=DIR	  path to directory containing zlib library (default=
-			  /usr/local or /usr if not found in /usr/local)],
+			  @GENTOO_PORTAGE_EPREFIX@/usr or /usr if not found in @GENTOO_PORTAGE_EPREFIX@/usr)],
 [
 if test "$withval"; then
   ZLIB_HOME="$withval"
   AC_MSG_RESULT(using $ZLIB_HOME)
 fi
 ], [
-ZLIB_HOME=/usr/local
+ZLIB_HOME=@GENTOO_PORTAGE_EPREFIX@/usr
 if test ! -f "$ZLIB_HOME/include/zlib.h"
 then
   ZLIB_HOME=/usr
