diff -uNr clamav-0.93.orig/configure.in clamav-0.93/configure.in
--- clamav-0.93.orig/configure.in	2008-04-17 18:40:31.041851000 +0200
+++ clamav-0.93/configure.in	2008-04-17 18:42:42.922065687 +0200
@@ -387,7 +387,7 @@
     LDFLAGS="$save_LDFLAGS"
     if test "$HAVE_LIBBZ2" = "yes"; then
 	AC_CHECK_HEADER([bzlib.h],
-			[LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS $LTLIBBZ2";
+			[LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS -lbz2";
 			 AC_DEFINE([HAVE_BZLIB_H],1,[have bzip2])],
 			[AC_MSG_WARN([****** bzip2 support disabled])])
     else
@@ -498,7 +498,7 @@
     dnl we don't need to check for __gmpz_init and mpz_init,
     dnl since we are not checking directly for symbols
     if test "$HAVE_LIBGMP" = "yes"; then
-	    LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS $LTLIBGMP";
+	    LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS -lgmp";
     else
 	    AC_MSG_WARN([****** GNU MP 2 or newer NOT FOUND - digital signature support will be disabled !]);
 	    want_dsig="no";
