diff -bur clamav-0.90.1-orig/clamav-milter/clamav-milter.c clamav-0.90.1/clamav-milter/clamav-milter.c
--- clamav-0.90.1-orig/clamav-milter/clamav-milter.c	2007-03-07 17:56:53.000000000 +0100
+++ clamav-0.90.1/clamav-milter/clamav-milter.c	2007-03-07 18:00:05.000000000 +0100
@@ -108,7 +108,7 @@
 #endif
 #endif
 
-#ifdef	C_LINUX
+#if defined(C_LINUX) && defined(ENABLE_NLS)
 #include <sys/sendfile.h>	/* FIXME: use sendfile on BSD not Linux */
 #include <libintl.h>
 #include <locale.h>
diff -bur clamav-0.90.1-orig/configure.in clamav-0.90.1/configure.in
--- clamav-0.90.1-orig/configure.in	2007-03-07 17:56:53.000000000 +0100
+++ clamav-0.90.1/configure.in	2007-03-07 17:57:30.000000000 +0100
@@ -160,6 +160,15 @@
     AC_CHECK_HEADER(dlfcn.h,[LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS -ldl" ; AC_DEFINE(HAVE_NCORE,1,Support for NodalCore acceleration)], AC_MSG_WARN([****** NodalCore support disabled (no support for dlopen)]))
 fi
 
+AC_ARG_ENABLE(nls,
+    AC_HELP_STRING([--disable-nls], [disable nls support]),
+    [want_nls=$enableval], [want_nls=yes]
+)
+
+if test $want_nls = yes; then
+   CPPFLAGS="$CPPFLAGS -DENABLE_NLS"
+fi
+
 AC_ARG_ENABLE(dns,
     AC_HELP_STRING([--disable-dns], [disable support for database verification through DNS]),
     [want_dns=$enableval], [want_dns=yes]
diff -bur clamav-0.90.1-orig/shared/getopt.c clamav-0.90.1/shared/getopt.c
--- clamav-0.90.1-orig/shared/getopt.c	2007-03-07 17:56:53.000000000 +0100
+++ clamav-0.90.1/shared/getopt.c	2007-03-07 18:00:25.000000000 +0100
@@ -82,7 +82,7 @@
 
 #ifndef _
 /* This is for other GNU distributions with internationalized messages.  */
-# if defined HAVE_LIBINTL_H || defined _LIBC
+# if (defined(HAVE_LIBINTL_H) || defined(_LIBC)) && defined(ENABLE_NLS)
 #  include <libintl.h>
 #  ifndef _
 #   define _(msgid)	gettext (msgid)
diff -bur clamav-0.90.1-orig/shared/output.c clamav-0.90.1/shared/output.c
--- clamav-0.90.1-orig/shared/output.c	2007-03-07 17:56:53.000000000 +0100
+++ clamav-0.90.1/shared/output.c	2007-03-07 17:57:30.000000000 +0100
@@ -61,7 +61,7 @@
 pthread_mutex_t logg_mutex = PTHREAD_MUTEX_INITIALIZER;
 #endif
 
-#ifdef  C_LINUX
+#if defined(C_LINUX) && defined(ENABLE_NLS)
 #include <libintl.h>
 #include <locale.h>
 
