diff -Naur clamav-0.94.orig/configure.in clamav-0.94/configure.in
--- clamav-0.94.orig/configure.in	2008-09-01 20:41:06.000000000 +0200
+++ clamav-0.94/configure.in	2008-09-05 21:09:41.201851168 +0200
@@ -473,7 +473,7 @@
 
 	case "$ac_cv_c_cve_2008_1372" in
 	ok|bugged)
-		LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS $LTLIBBZ2"
+		LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS  -lbz2"
 		AC_DEFINE([HAVE_BZLIB_H],1,[have bzip2])
 		;;
 	*)
@@ -566,7 +566,7 @@
     dnl we don't need to check for __gmpz_init and mpz_init,
     dnl since we are not checking directly for symbols
     if test "$HAVE_LIBGMP" = "yes"; then
-	    LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS $LTLIBGMP";
+	    LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS -lgmp";
     else
 	    AC_MSG_WARN([****** GNU MP 2 or newer NOT FOUND - digital signature support will be disabled !]);
 	    want_dsig="no";
