Index: echangelog
===================================================================
--- src/echangelog/echangelog	(revision 589)
+++ src/echangelog/echangelog	(revision 617)
@@ -686,6 +686,12 @@
 		# Move things around and show the diff
 		system "diff -U 0 $e $e.new";
 		rename "$e.new", $e or warn("Can't rename $e.new: $!\n");
+
+		# git requires to re-add this file else it wouln't be included in the commit.
+		if ($vcs eq "git")
+		{
+			system("$vcs{$vcs}{add} ${e}");
+		}
 	}
 }
 
