--- debian/cron.daily
+++ debian/cron.daily
@@ -1,10 +1,10 @@
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/sh
 
-if [ -x /usr/bin/updatedb ]
+if [ -x "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/updatedb ]
 then
-	if [ -f /etc/updatedb.conf ]
+	if [ -f "@GENTOO_PORTAGE_EPREFIX@"/etc/updatedb.conf ]
 	then
-		. /etc/updatedb.conf
+		. "@GENTOO_PORTAGE_EPREFIX@"/etc/updatedb.conf
 		args=""
 	else
 		args="-f proc"
@@ -14,5 +14,5 @@
 	# system is really old and ionice doesnt work ...
 	ionice -c ${IONICE_CLASS:-2} -n ${IONICE_PRIORITY:-7} -p $$ 2>/dev/null
 
-	nice -n ${NICE:-10} /usr/bin/updatedb ${args}
+	nice -n ${NICE:-10} "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/updatedb ${args}
 fi
