--- debian/cron.daily
+++ debian/cron.daily
@@ -1,11 +1,11 @@
-#! /bin/sh
+#! @GENTOO_PORTAGE_EPREFIX@/bin/bash
 
-if [ -x /usr/bin/updatedb ]
+if [ -x "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/updatedb ]
 then
-	if [ -f /etc/updatedb.conf ]
+	if [ -f "@GENTOO_PORTAGE_EPREFIX@"/etc/updatedb.conf ]
 	then
-		nice /usr/bin/updatedb
+		nice "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/updatedb
 	else
-		nice /usr/bin/updatedb -f proc
+		nice "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/updatedb -f proc
 	fi
 fi
