--- src/slocate.h
+++ src/slocate.h
@@ -32,11 +32,11 @@
 /* Printable version of WARN_SECONDS.  */
 #define WARN_MESSAGE "8 days"
 
-#define MTAB_FILE "/etc/mtab"
-#define UPDATEDB_FILE "/etc/updatedb.conf"
+#define MTAB_FILE "@GENTOO_PORTAGE_EPREFIX@/etc/mtab"
+#define UPDATEDB_FILE "@GENTOO_PORTAGE_EPREFIX@/etc/updatedb.conf"
 
-#define DEFAULT_DB "/var/lib/slocate/slocate.db"
-#define DEFAULT_DB_DIR "/var/lib/slocate/"
+#define DEFAULT_DB "@GENTOO_PORTAGE_EPREFIX@/var/lib/slocate/slocate.db"
+#define DEFAULT_DB_DIR "@GENTOO_PORTAGE_EPREFIX@/var/lib/slocate/"
 
 #define DB_UID 0
 #define DB_GROUP "locate"
