--- Makefile.SH.orig	2007-03-26 17:55:48.000000000 +0200
+++ Makefile.SH	2007-03-26 17:55:28.000000000 +0200
@@ -549,15 +549,17 @@
 !NO!SUBS!
 	case "$useshrplib" in
 	true)
-		$spitshell >>Makefile <<'!NO!SUBS!'
-	$(LD) -o $@ $(SHRPLDFLAGS) perl$(OBJ_EXT) $(obj) $(libs)
-!NO!SUBS!
 		case "$osname" in
 		aix)
 			$spitshell >>Makefile <<'!NO!SUBS!'
 	rm -f libperl$(OBJ_EXT)
 	mv $@ libperl$(OBJ_EXT)
 	$(AR) qv $(LIBPERL) libperl$(OBJ_EXT)
+!NO!SUBS!
+		;;
+		*)
+		$spitshell >>Makefile <<'!NO!SUBS!'
+	$(LD) -o $@ $(SHRPLDFLAGS) perl$(OBJ_EXT) $(obj) $(libs) -Wl,-soname -Wl,libperl.so.`echo $(LIBPERL) | cut -d. -f3`
 !NO!SUBS!
 			;;
 		esac
