diff --git a/ragel/main.cpp b/ragel/main.cpp
index a16a50a..13d0152 100644
--- a/ragel/main.cpp
+++ b/ragel/main.cpp
@@ -336,7 +336,7 @@ void processArgs( int argc, const char **argv, InputData &id )
 				printStatistics = true;
 				break;
 			case '-': {
-				char *eq = strchr( pc.paramArg, '=' );
+				char *eq = const_cast<char *>(strchr( pc.paramArg, '=' ));
 
 				if ( eq != 0 )
 					*eq++ = 0;
@@ -507,7 +507,7 @@ char *makeIntermedTemplate( const char *baseFileName )
 {
 	char *result = 0;
 	const char *templ = "ragel-XXXXXX.xml";
-	char *lastSlash = strrchr( baseFileName, '/' );
+	const char *lastSlash = strrchr( baseFileName, '/' );
 	if ( lastSlash == 0 ) {
 		result = new char[strlen(templ)+1];
 		strcpy( result, templ );
