--- Makefile-libbz2_so.orig	2009-06-23 12:35:06.000000000 +0200
+++ Makefile-libbz2_so	2009-06-23 12:36:39.000000000 +0200
@@ -43,8 +43,17 @@
 	rm -f libbz2.${SOEXT}.1.0
 	ln -s libbz2.${SOEXT}.1.0.4 libbz2.${SOEXT}.1.0
 
+# AIX has dynamic libraries packed into archive library.
+.PHONY: all-aix
+all-aix: libbz2.a
+libbz2.a: $(OBJS)
+	$(CC) $(LDFLAGS) -shared -Wl,-G -o $(SONAME) $(OBJS)
+	$(AR) q libbz2.a $(SONAME)
+	$(CC) $(LDFLAGS) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.a
+
 clean: 
 	rm -f $(OBJS) bzip2.o libbz2.${SOEXT}.1.0.4 libbz2.${SOEXT}.1.0 bzip2-shared
+	rm -f $(SONAME) libbz2.a
 
 blocksort.o: blocksort.c
 	$(CC) $(CFLAGS) -c blocksort.c
