--- bzdiff
+++ bzdiff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
 # sh is buggy on RS/6000 AIX 3.2. Replace above line with #!/bin/ksh
 
 # Bzcmp/diff wrapped for bzip2, 
@@ -12,7 +12,7 @@
 # necessary) and fed to cmp or diff.  The exit status from cmp
 # or diff is preserved.
 
-PATH="/usr/bin:/bin:$PATH"; export PATH
+PATH="@GENTOO_PORTAGE_EPREFIX@/usr/bin:@GENTOO_PORTAGE_EPREFIX@/bin:$PATH"; export PATH
 prog=`echo $0 | sed 's|.*/||'`
 case "$prog" in
   *cmp) comp=${CMP-cmp}   ;;
@@ -56,7 +56,7 @@
                         bzip2 -cdfq "$2" > $tmp
                         bzip2 -cdfq "$1" | $comp $OPTIONS - $tmp
                         STAT="$?"
-			/bin/rm -f $tmp;;
+			@GENTOO_PORTAGE_EPREFIX@/bin/rm -f $tmp;;
 
                 *)      bzip2 -cdfq "$1" | $comp $OPTIONS - "$2"
                         STAT="$?";;
--- bzgrep
+++ bzgrep
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 
 # Bzgrep wrapped for bzip2, 
 # adapted from zgrep by Philippe Troin <phil@fifi.org> for Debian GNU/Linux.
@@ -6,7 +6,7 @@
 ## zgrep -- a wrapper around a grep program that decompresses files as needed
 ## Adapted from a version sent by Charles Levert <charles@comm.polymtl.ca>
 
-PATH="/usr/bin:$PATH"; export PATH
+PATH="@GENTOO_PORTAGE_EPREFIX@/usr/bin:$PATH"; export PATH
 
 prog=`echo $0 | sed 's|.*/||'`
 case "$prog" in
--- bzmore
+++ bzmore
@@ -1,9 +1,9 @@
-#!/bin/sh
+#!@GENTOO_PORTAGE_EPREFIX@/bin/sh
 
 # Bzmore wrapped for bzip2, 
 # adapted from zmore by Philippe Troin <phil@fifi.org> for Debian GNU/Linux.
 
-PATH="/usr/bin:$PATH"; export PATH
+PATH="@GENTOO_PORTAGE_EPREFIX@/usr/bin:$PATH"; export PATH
 
 prog=`echo $0 | sed 's|.*/||'`
 case "$prog" in
