Fix shared makefile to also add $(BIGFILES), bug #98802

--- bzip2-1.0.3/Makefile-libbz2_so	2005-07-13 12:11:19.000000000 +0200
+++ bzip2-1.0.3.az/Makefile-libbz2_so	2005-07-13 12:09:00.000000000 +0200
@@ -9,7 +9,7 @@
 SHELL=/bin/sh
 CC=gcc
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS+=-fpic -fPIC -Wall -Winline
+CFLAGS+=-fpic -fPIC -Wall -Winline $(BIGFILES)
 
 OBJS= blocksort.o  \
       huffman.o    \
