--- Makefile.orig	2004-12-26 16:30:06.354116536 -0500
+++ Makefile	2004-12-26 16:32:13.617769520 -0500
@@ -9,7 +9,7 @@
 
 # Suitably paranoid flags to avoid bugs in gcc-2.7
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-Wall -Winline -O2 -fomit-frame-pointer -fno-strength-reduce $(BIGFILES)
+CFLAGS+=-Wall -Winline -fno-strength-reduce $(BIGFILES)
 
 # Where you want it installed when you do 'make install'
 PREFIX=/usr
--- Makefile-libbz2_so.orig	2004-12-26 16:32:23.972195408 -0500
+++ Makefile-libbz2_so	2004-12-26 16:32:34.835543928 -0500
@@ -9,7 +9,7 @@
 SHELL=/bin/sh
 CC=gcc
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-fpic -fPIC -Wall -Winline -O2 -fomit-frame-pointer -fno-strength-reduce $(BIGFILES)
+CFLAGS+=-fpic -fPIC -Wall -Winline -fomit-frame-pointer -fno-strength-reduce $(BIGFILES)
 
 OBJS= blocksort.o  \
       huffman.o    \
