--- Makefile
+++ Makefile
@@ -10,4 +10,4 @@
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-Wall -Winline -O -g $(BIGFILES)
+CFLAGS+=-Wall -Winline $(BIGFILES)
 
 # Where you want it installed when you do 'make install'
--- Makefile-libbz2_so
+++ Makefile-libbz2_so
@@ -11,4 +11,4 @@
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-fpic -fPIC -Wall -Winline -O -g
+CFLAGS+=-fpic -fPIC -Wall -Winline
 
 OBJS= blocksort.o  \
