--- bzip2.c	2005-02-15 10:25:35.000000000 -0600
+++ bzip2.c-new	2005-10-12 21:19:21.000000000 -0500
@@ -133,6 +133,9 @@
 #define BZ_UNIX 0
 #endif
 
+#ifdef __APPLE__
+#include <copyfile.h>
+#endif
 
 /*---------------------------------------------*/
 /*--
@@ -1148,6 +1151,11 @@
    /* chown() will in many cases return with EPERM, which can
       be safely ignored.
    */
+
+   #if __APPLE__
+   	copyfile(inName, outName, 0, COPYFILE_ACL | COPYFILE_XATTR);
+   #endif
+
 #  endif
 }
 
