--- Makefile-libbz2_so.orig	2009-06-23 12:30:55.000000000 +0200
+++ Makefile-libbz2_so	2009-06-23 12:31:38.000000000 +0200
@@ -25,6 +25,9 @@
 CC=gcc
 BIGFILES=-D_FILE_OFFSET_BITS=64
 CFLAGS+=-fpic -fPIC -Wall -Winline $(BIGFILES)
+SOEXT=so
+SONAME=libbz2.${SOEXT}.1
+SOLDFLAGS=-shared -Wl,-soname -Wl,$(SONAME)
 
 OBJS= blocksort.o  \
       huffman.o    \
@@ -35,13 +38,13 @@
       bzlib.o
 
 all: $(OBJS)
-	$(CC) $(LDFLAGS) -shared -Wl,-soname -Wl,libbz2.so.1 -o libbz2.so.1.0.4 $(OBJS)
-	$(CC) $(LDFLAGS) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.4
-	rm -f libbz2.so.1.0
-	ln -s libbz2.so.1.0.4 libbz2.so.1.0
+	$(CC) $(LDFLAGS) $(SOLDFLAGS) -o libbz2.${SOEXT}.1.0.4 $(OBJS)
+	$(CC) $(LDFLAGS) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.${SOEXT}.1.0.4
+	rm -f libbz2.${SOEXT}.1.0
+	ln -s libbz2.${SOEXT}.1.0.4 libbz2.${SOEXT}.1.0
 
 clean: 
-	rm -f $(OBJS) bzip2.o libbz2.so.1.0.4 libbz2.so.1.0 bzip2-shared
+	rm -f $(OBJS) bzip2.o libbz2.${SOEXT}.1.0.4 libbz2.${SOEXT}.1.0 bzip2-shared
 
 blocksort.o: blocksort.c
 	$(CC) $(CFLAGS) -c blocksort.c
