http://bugs.gentoo.org/show_bug.cgi?id=257653

--- libpurple/plugins/perl/perl.c.dist	2009-02-26 11:59:33.632774600 +0000
+++ libpurple/plugins/perl/perl.c	2009-02-26 12:00:13.207914920 +0000
@@ -682,7 +682,7 @@ init_plugin(PurplePlugin *plugin)
 	loader_info.exts = g_list_append(loader_info.exts, "pl");
 }
 
-#ifdef __SUNPRO_C
+#if defined(__SUNPRO_C) || defined(__sgi)
 #pragma init (my_init)
 #else
 void __attribute__ ((constructor)) my_init(void);
