--- src/portable.cpp.orig	2007-07-12 21:10:39 +0200
+++ src/portable.cpp	2007-07-12 21:13:18 +0200
@@ -373,16 +373,16 @@
 size_t portable_iconv (void *cd, const char** inbuf,  size_t *inbytesleft, 
                                        char** outbuf, size_t *outbytesleft)
 {
-#if (defined(_LIBICONV_VERSION) && (_LIBICONV_VERSION==0x0109))
+#if ((defined(_LIBICONV_VERSION) && (_LIBICONV_VERSION>=0x0109)) || defined(_OS_SOLARIS_))
 #define CASTNEEDED(x) (x)
 #else
 #define CASTNEEDED(x) (char **)(x)
 #endif
-  return iconv(cd,CASTNEEDED(inbuf),inbytesleft,outbuf,outbytesleft);
+  return iconv((iconv_t)cd,CASTNEEDED(inbuf),inbytesleft,outbuf,outbytesleft);
 }
 
 int portable_iconv_close (void *cd)
 {
-  return iconv_close(cd);
+  return iconv_close((iconv_t)cd);
 }
 
