--- doxygen-1.5.7.1-orig/src/util.cpp	2008-10-04 14:28:00.000000000 +0200
+++ doxygen-1.5.7.1/src/util.cpp	2008-12-18 20:55:15.000000000 +0100
@@ -4447,10 +4447,10 @@
 
 QCString substitute(const char *s,const char *src,const char *dst)
 {
-  if (s==0 || src==0 || dst==0) return s;
+  if (s==0 || src==0) return s;
   const char *p, *q;
   int srcLen = strlen(src);
-  int dstLen = strlen(dst);
+  int dstLen = dst ? strlen(dst) : 0;
   int resLen;
   if (srcLen!=dstLen)
   {
@@ -4469,7 +4469,7 @@
     int l = (int)(q-p);
     memcpy(r,p,l);
     r+=l;
-    memcpy(r,dst,dstLen);
+    if (dst) memcpy(r,dst,dstLen);
     r+=dstLen;
   }
   strcpy(r,p);
