Index: src/pv/file.c
===================================================================
--- src/pv/file.c	(revision 162)
+++ src/pv/file.c	(working copy)
@@ -147,7 +147,8 @@
 
 	/*
 	 * Check that this new input file is not the same as stdout's
-	 * destination. This restriction is ignored for tty devices.
+	 * destination. This restriction is ignored for anything other
+	 * than a regular file or block device.
 	 */
 	if (isb.st_dev != osb.st_dev)
 		return fd;
@@ -155,6 +156,8 @@
 		return fd;
 	if (isatty(fd))
 		return fd;
+	if ((!S_ISREG(isb.st_mode)) && (!S_ISBLK(isb.st_mode)))
+		return fd;
 
 	fprintf(stderr, "%s: %s: %s\n",
 		opts->program_name,
