--- src/preload.conf.in~	2008-10-18 17:12:39.000000000 +0200
+++ src/preload.conf.in	2008-10-18 17:12:39.000000000 +0200
@@ -132,7 +132,7 @@
 # preload doesn't special-handle device files internally.
 #
 # default: (empty list, accept all)
-mapprefix = /usr;/lib;/lib64;/var/cache;!/
+mapprefix = /usr;/lib;/lib64;/lib32;/opt;/var/cache;!/
 
 # exeprefix:
 #
@@ -141,7 +141,7 @@
 # files instead of maps.
 #
 # default: (empty list, accept all)
-exeprefix = !/usr/sbin;!/usr/local/sbin;/usr;!/
+exeprefix = !/usr/sbin;!/usr/local/sbin;/usr;/opt;!/
 
 # maxprocs
 #
