diff -ru postgresql-8.3.7/src/include/pg_config_manual.h postgresql-new/src/include/pg_config_manual.h
--- postgresql-8.3.7/src/include/pg_config_manual.h	2009-04-21 14:47:54 -0700
+++ postgresql-new/src/include/pg_config_manual.h	2009-04-21 15:08:20 -0700
@@ -184,7 +184,7 @@
  * here's where to twiddle it.  You can also override this at runtime
  * with the postmaster's -k switch.
  */
-#define DEFAULT_PGSOCKET_DIR  "/var/run/postgresql"
+#define DEFAULT_PGSOCKET_DIR  "@GENTOO_PORTAGE_EPREFIX@/var/run/postgresql"
 
 /*
  * The random() function is expected to yield values between 0 and
