--- ruby.eselect-20081211.orig	2008-12-26 14:11:59 +0100
+++ ruby.eselect-20081211	2008-12-26 14:20:12 +0100
@@ -7,8 +7,11 @@
 SVN_DATE='$Date: $'
 VERSION="20081211"
 
+bindir=/usr/bin
+man1dir=/usr/share/man/man1
+
 find_targets() {
-	for t in ${ROOT}/usr/bin/${1:-ruby}{18,19,20,ee} ; do
+	for t in ${ROOT}${bindir}/${1:-ruby}{18,19,20,ee} ; do
 		[[ -e $t ]] || continue
 		echo ${t}
 	done
@@ -30,7 +33,7 @@
 		fi
 	# `rubyXX' string
 	else
-	[[ -e "${ROOT}/usr/bin/${target}" ]] || \
+	[[ -e "${ROOT}${bindir}/${target}" ]] || \
 			die -q "Can't use that profile. No suitable Ruby interpreter found."
 	fi
 
@@ -38,14 +41,14 @@
 }
 
 remove_symlinks() {
-	rm -f ${ROOT}/usr/bin/{ruby,gem,irb,erb,ri,rdoc,testrb} && \
-	rm -f ${ROOT}/usr/share/man/man1/ruby.{1,1.gz,1.bz2,1.lzma}
+	rm -f ${ROOT}${bindir}/{ruby,gem,irb,erb,ri,rdoc,testrb} && \
+	rm -f ${ROOT}${man1dir}/ruby.{1,1.gz,1.bz2,1.lzma}
 }
 
 create_man_symlinks() {
 	local version=${1} 
 
-	for m in ${ROOT}/usr/share/man/man1/ruby${version}.{1,1.gz,1.bz2,1.lzma}
+	for m in ${ROOT}${man1dir}/ruby${version}.{1,1.gz,1.bz2,1.lzma}
 	do
 		if [[ -e ${m} ]] ; then
 			break
@@ -62,7 +65,7 @@
 	target=$(check_target ${target})
 	version=${target##*ruby}
 	
-	local path="${ROOT}/usr/bin/"
+	local path="${ROOT}${bindir}/"
 	# these have to work
 	for f in "ruby" "irb" "erb" "ri" "testrb" "rdoc" ; do
 		ln -s "${path}${f}${version}" "${path}${f}" || \
@@ -97,23 +100,23 @@
 	local rb=""
 
 	write_list_start "Current Ruby version:"
-	if [[ -L "${ROOT}/usr/bin/ruby" ]] ; then
-		rb=$(basename $(canonicalise ${ROOT}/usr/bin/ruby ) )
+	if [[ -L "${ROOT}${bindir}/ruby" ]] ; then
+		rb=$(basename $(canonicalise ${ROOT}${bindir}/ruby ) )
 		write_kv_list_entry $rb ""
-	elif [[ -e "${ROOT}/usr/bin/ruby" ]] ; then
-		write_warning_msg "/usr/bin/ruby is a SLOT incompatible version."
-		write_kv_list_entry "$(basename $(canonicalise ${ROOT}/usr/bin/ruby ) )" ""
+	elif [[ -e "${ROOT}${bindir}/ruby" ]] ; then
+		write_warning_msg "${bindir}/ruby is a SLOT incompatible version."
+		write_kv_list_entry "$(basename $(canonicalise ${ROOT}${bindir}/ruby ) )" ""
 	else
 		write_kv_list_entry "(none)"
 		return 1
 	fi
 	
 	write_list_start "Current Rubygems version:"
-	if [[ -L "${ROOT}/usr/bin/gem" ]] ; then
-		write_kv_list_entry "$(basename $(canonicalise ${ROOT}/usr/bin/gem ) )" ""
-	elif [[ -e "${ROOT}/usr/bin/gem" ]] ; then
-		write_warning_msg "/usr/bin/gem is a SLOT incompatible version."
-		write_kv_list_entry "$(basename $(canonicalise ${ROOT}/usr/bin/gem) )" ""
+	if [[ -L "${ROOT}${bindir}/gem" ]] ; then
+		write_kv_list_entry "$(basename $(canonicalise ${ROOT}${bindir}/gem ) )" ""
+	elif [[ -e "${ROOT}${bindir}/gem" ]] ; then
+		write_warning_msg "${bindir}/gem is a SLOT incompatible version."
+		write_kv_list_entry "$(basename $(canonicalise ${ROOT}${bindir}/gem) )" ""
 	else
 		write_kv_list_entry "gem??" "(not found)"
 	fi
@@ -140,7 +143,7 @@
 
 			
 			# find out the current version
-			if [[ ${targets[${i}]} = $(canonicalise ${ROOT}/usr/bin/ruby) ]] ; then
+			if [[ ${targets[${i}]} = $(canonicalise ${ROOT}${bindir}/ruby) ]] ; then
 				targets[${i}]="${line} $(highlight '*' )"
 			else
 				targets[${i}]=${line}
