diff -ru ghostscript-8.62.orig/src/configure.ac ghostscript-8.62/src/configure.ac
--- ghostscript-8.62.orig/src/configure.ac	2008-05-05 16:12:02 +0200
+++ ghostscript-8.62/src/configure.ac	2008-05-05 16:19:40 +0200
@@ -923,6 +923,13 @@
 		DYNAMIC_LIBS=""
 		X_DEVS=""
 		;;
+		Interix)
+		INSTALL_SHARED="install-shared"
+		DYNAMIC_FLAGS="-DGS_DEVS_SHARED -DGS_DEVS_SHARED_DIR=\\\"\$(gssharedir)\\\""
+		DYNAMIC_LDFLAGS="-shared"
+		DYNAMIC_LIBS="-ldl"
+		X_DEVS=""
+		;;
 		*)
 		AC_MSG_ERROR([Sorry, dynamic driver support not available on this platform!])
 		;;
diff -ru ghostscript-8.62.orig/contrib/gomni.c ghostscript-8.62/contrib/gomni.c
--- ghostscript-8.62.orig/contrib/gomni.c	2008-05-05 16:51:20 +0200
+++ ghostscript-8.62/contrib/gomni.c	2008-05-06 08:20:30 +0200
@@ -437,8 +437,14 @@
    {
       if (fDebugOutput) dprintf("Remapping color pointers\n");
 
-      set_dev_proc(pdev, map_rgb_color, bmp_map_16m_rgb_color);
-      set_dev_proc(pdev, map_color_rgb, bmp_map_16m_color_rgb);
+      eprintf ("\n<<<<<<<<<<<<<<<<<<<<<< ERROR >>>>>>>>>>>>>>>>>>>>>>>\n\n");
+      eprintf ("Error: bmp device needs to be dynamically accessed here!\n");
+
+	  // this cannot be done like this, because it introduces a hard
+	  // dependency to the bmp device. this would need to be done
+	  // dynamically.
+      //set_dev_proc(pdev, map_rgb_color, bmp_map_16m_rgb_color);
+      //set_dev_proc(pdev, map_color_rgb, bmp_map_16m_color_rgb);
    }
 
    if (pDev->iSync)

