diff -ru orage-4.5.14.0.orig/configure.in orage-4.5.14.0/configure.in
--- orage-4.5.14.0.orig/configure.in	2008-10-28 19:04:37.489180921 +0100
+++ orage-4.5.14.0/configure.in	2008-10-28 19:10:43.725708992 +0100
@@ -9,7 +9,7 @@
 dnl
 
 dnl Version information
-m4_define([orage_version], [4.5.14.0])
+m4_define([orage_version], [4.5.14.1-svn])
 
 m4_define([gtk_minimum_version], [2.6.0])
 m4_define([xfce_minimum_version], [4.4.0])
@@ -172,6 +172,21 @@
 esac
 AC_SUBST([PTHREAD_LIBS])
 
+dnl **********************************
+dnl *** check if we have _NL_TIME_FIRST_WEEKDAY
+dnl *** note that it is an enum and not a define
+dnl **********************************
+AC_MSG_CHECKING([for _NL_TIME_FIRST_WEEKDAY])
+AC_TRY_LINK([#include <langinfo.h>], [
+char c;
+c = *((unsigned char *)  nl_langinfo(_NL_TIME_FIRST_WEEKDAY));
+], nl_ok=yes, nl_ok=no)
+AC_MSG_RESULT($nl_ok)
+if test "$nl_ok" = "yes"; then
+  AC_DEFINE([HAVE__NL_TIME_FIRST_WEEKDAY], [1],
+      [Define if _NL_TIME_FIRST_WEEKDAY is available])
+fi
+
 AM_CONDITIONAL([INCLUDED_LIBICAL], [test x"$ac_INCLUDED_LIBICAL" = x"yes"])
 AM_CONDITIONAL([HAVE_PTHREAD], [test x"$have_pthread" = x"yes"])
 AM_CONDITIONAL([WITH_BDB4], [test x"$WITH_BDB4" = x"yes"])
