diff -ru -x '*.Po' -x '*.Plo' orage-4.5.14.0.orig/src/parameters.c orage-4.5.14.0/src/parameters.c
--- orage-4.5.14.0.orig/src/parameters.c	2008-11-21 08:56:57 +0100
+++ orage-4.5.14.0/src/parameters.c	2008-11-21 09:33:21 +0100
@@ -129,6 +129,7 @@
  *     to return 0..6 mon..sun, which is what libical uses */
 int get_first_weekday_from_locale()
 {
+#ifndef __INTERIX
     union { unsigned int word; char *string; } langinfo;
     int week_1stday = 0;
     int first_weekday = 1;
@@ -147,6 +148,10 @@
         orage_message(150, "get_first_weekday: unknown value of _NL_TIME_WEEK_1STDAY.");
 
     return((week_1stday + first_weekday - 2 + 7) % 7);
+#else
+	/* ARGH: what can we do? */
+	return 1;
+#endif
 }
 
 static void dialog_response(GtkWidget *dialog, gint response_id
