=== modified file 'configure.in'
--- configure.in	2009-05-24 15:45:58 +0000
+++ configure.in	2009-05-24 16:02:19 +0000
@@ -107,8 +107,13 @@
 AC_MSG_CHECKING(for C flags to get more warnings)
 ac_save_CFLAGS="$CFLAGS"
 if test "x$ac_cv_c_compiler_gnu" = "xyes" ; then
-  dnl gcc is the easiest C compiler
-  warning_CFLAGS="-Wall -Wno-pointer-sign"
+  warning_CFLAGS="-Wall"
+  # Check if compiler supports -Wno-pointer-sign and add it if supports
+  CFLAGS_saved="$CFLAGS"
+  CFLAGS="$CFLAGS -Wno-pointer-sign"
+  AC_COMPILE_IFELSE([ AC_LANG_SOURCE([[ int foo; ]])],
+                  [ warning_CFLAGS="${warning_CFLAGS} -Wno-pointer-sign" ],)
+  CFLAGS="$CFLAGS_saved"
 else
   dnl Vendor supplied C compilers are a bit tricky
   case "$host_os" in

