--- emacs-updater.orig	2008-02-27 16:10:42.000000000 +0100
+++ emacs-updater	2008-02-27 16:11:51.000000000 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 # Licensed under GPL version 2
 # Authors:
@@ -6,8 +6,8 @@
 # Ulrich Mueller <ulm@gentoo.org>
 
 VERSION=0.4
-SITELISP=/usr/share/emacs/site-lisp
-TMPFILE="$(mktemp /tmp/emacs-updater.XXXXXX)"
+SITELISP="@GENTOO_PORTAGE_EPREFIX@"/usr/share/emacs/site-lisp
+TMPFILE="$(mktemp /var/tmp/emacs-updater.XXXXXX)"
 
 cat <<-EOF
 
@@ -60,7 +60,7 @@
     echo "${RED}*${NORMAL}${BOLD} ${OUTPUT}${NORMAL}" 
 }
 
-if ! [ -x /usr/bin/qfile ]; then
+if ! [ -x "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/qfile ]; then
     echo
     failure "Please emerge app-portage/portage-utils to use this tool"
     exit 1
