--- emacs.eselect
+++ emacs.eselect
@@ -21,7 +21,7 @@
 # do_update(): Set the target to the highest version available
 #	(optionally: only if not set)
 
-DESCRIPTION="Manage /usr/bin/emacs version"
+DESCRIPTION="Manage @GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs version"
 MAINTAINER="emacs@gentoo.org"
 SVN_DATE='$Date: 2007-06-17 17:12:48 +0200 (Sun, 17 Jun 2007) $'
 VERSION=$(svn_date_to_version "${SVN_DATE}" )
@@ -32,32 +32,32 @@
 find_targets() {
 	# Return the list of available Emacs binaries
 	local j
-	for j in "${ROOT}"/usr/bin/emacs-[0-9]*; do
+	for j in "${ROOT}@GENTOO_PORTAGE_EPREFIX@"/usr/bin/emacs-[0-9]*; do
 		[[ -f ${j} ]] && basename ${j}
 	done
 }
 
 remove_infopath() {
 	# When cleaning symlinks this takes care of the info documentation settings
-	rm -f "${ROOT}/etc/env.d/50emacs"
+	rm -f "${ROOT}@GENTOO_PORTAGE_EPREFIX@/etc/env.d/50emacs"
 }
 
 # Define INFOPATH environment variable in env file
 set_infopath() {
-	[[ -d "${ROOT}"/usr/share/info/${1} ]] || return 1
-	echo "INFOPATH=/usr/share/info/${1}" > "${ROOT}/etc/env.d/50emacs"
+	[[ -d "${ROOT}@GENTOO_PORTAGE_EPREFIX@"/usr/share/info/${1} ]] || return 1
+	echo "INFOPATH=@GENTOO_PORTAGE_EPREFIX@/usr/share/info/${1}" > "${ROOT}@GENTOO_PORTAGE_EPREFIX@/etc/env.d/50emacs"
 }
 
 remove_symlinks() {
 	# Remove existing symlinks to binaries, man pages,
 	# and the env file (own function)
 	local f
-	rm -f "${ROOT}"/usr/bin/emacs
+	rm -f "${ROOT}@GENTOO_PORTAGE_EPREFIX@"/usr/bin/emacs
 	for f in ${BINARYLIST}; do
-		rm -f "${ROOT}"/usr/bin/${f}
+		rm -f "${ROOT}@GENTOO_PORTAGE_EPREFIX@"/usr/bin/${f}
 	done
 	for f in ${MANPAGELIST}; do
-		rm -f "${ROOT}"/usr/share/man/man1/${f}.1*
+		rm -f "${ROOT}@GENTOO_PORTAGE_EPREFIX@"/usr/share/man/man1/${f}.1*
 	done
 	remove_infopath
 }
@@ -67,9 +67,9 @@
 	local target=${1} f
 	for f in ${BINARYLIST}; do
 		# set symlink only if target binary actually exists
-		if [[ -f "${ROOT}/usr/bin/${f}-${target}" ]]; then
-			ln -s "${f}-${target}" "${ROOT}/usr/bin/${f}" ||
-				die "Couldn't set ${f}-${target} ${ROOT}/usr/bin/${f} symlink"
+		if [[ -f "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/${f}-${target}" ]]; then
+			ln -s "${f}-${target}" "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/${f}" ||
+				die "Couldn't set ${f}-${target} ${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/${f} symlink"
 		fi
 	done
 }
@@ -78,12 +78,12 @@
 	# Set symlinks to man pages
 	local target=${1} extension f i
 	for f in ${MANPAGELIST}; do
-		for i in "${ROOT}"/usr/share/man/man1/${f}-${target}.1*; do
+		for i in "${ROOT}@GENTOO_PORTAGE_EPREFIX@"/usr/share/man/man1/${f}-${target}.1*; do
 			if [[ -f ${i} ]]; then
 				# target file exists; determine compress extension
 				extension=${i##*/${f}-${target}.1}
 				ln -s "${f}-${target}.1${extension}" \
-					"${ROOT}/usr/share/man/man1/${f}.1${extension}"
+					"${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/share/man/man1/${f}.1${extension}"
 			fi
 		done
 	done
@@ -102,7 +102,7 @@
 	fi
 
 	# is the target valid, i.e. does an Emacs binary with this name exist?
-	[[ -f "${ROOT}/usr/bin/${target}" ]] \
+	[[ -f "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/${target}" ]] \
 		|| die -q "Target \"${1}\" doesn't appear to be valid!"
 
 	echo "Switching to ${target} ..."
@@ -110,8 +110,8 @@
 	# the main /usr/bin/emacs symlink is only set for Emacs binaries
 	# (but not for other providers of auxiliary programs, e.g., XEmacs)
 	if [[ ${target} == emacs-* ]]; then
-		ln -s "${target}" "${ROOT}/usr/bin/emacs" \
-			|| die "Couldn't set ${target} ${ROOT}/usr/bin/emacs symlink"
+		ln -s "${target}" "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs" \
+			|| die "Couldn't set ${target} ${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs symlink"
 	fi
 	set_bin_symlinks "${target}"
 	set_man_symlinks "${target}"
@@ -127,7 +127,7 @@
 xemacs_info() {
 	# This is just informational to tell that XEmacs has been found,
 	# a wish from graaff
-	if [[ -e "${ROOT}/usr/bin/xemacs" ]]; then
+	if [[ -e "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/xemacs" ]]; then
 		echo
 		write_list_start "XEmacs is also installed"
 	fi
@@ -135,7 +135,7 @@
 
 test_for_root() {
 	# checks if the user has rights to modify /usr/bin/
-	[[ -w "${ROOT}/usr/bin" ]] || die -q "You need to be root!"
+	[[ -w "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin" ]] || die -q "You need to be root!"
 }
 
 ### show action ###
@@ -148,11 +148,11 @@
 	[[ -z "${@}" ]] || die -q "Too many parameters"
 
 	write_list_start "Current target of GNU Emacs symlink:"
-	if [[ -L "${ROOT}/usr/bin/emacs" && \
-		  -e $(canonicalise "${ROOT}/usr/bin/emacs") ]]; then
+	if [[ -L "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs" && \
+		  -e $(canonicalise "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs") ]]; then
 		write_kv_list_entry \
-			"$(basename $(canonicalise ${ROOT}/usr/bin/emacs) )" ""
-	elif [[ -e "${ROOT}/usr/bin/emacs" ]]; then
+			"$(basename $(canonicalise ${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs) )" ""
+	elif [[ -e "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs" ]]; then
 		write_kv_list_entry \
 			"(not a symlink or target of symlink does not exist)" ""
 	else
@@ -177,7 +177,7 @@
 		for (( i = 0; i < ${#targets[@]}; i = i + 1 )); do
 			# Display a star to indicate the currently chosen version
 			[[ ${targets[${i}]} = \
-				$(basename $(canonicalise "${ROOT}/usr/bin/emacs") ) ]] \
+				$(basename $(canonicalise "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs") ) ]] \
 				&& targets[${i}]="${targets[${i}]} $(highlight '*')"
 		done
 		write_list_start "Available GNU Emacs symlink targets:"
@@ -207,9 +207,9 @@
 	[[ -n "${2}" ]] && die -q "Too many parameters"
 	test_for_root
 
-	if [[ -e "${ROOT}/usr/bin/emacs" ]] \
-		&& ! [[ -L "${ROOT}/usr/bin/emacs" ]]; then
-		die -q "Sorry, ${ROOT}/usr/bin/emacs exists but is not a symlink"
+	if [[ -e "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs" ]] \
+		&& ! [[ -L "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs" ]]; then
+		die -q "Sorry, ${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs exists but is not a symlink"
 	fi
 
 	set_symlinks "${1}" || die -q "Couldn't set a new symlink"
@@ -230,20 +230,20 @@
 		|| die -q "Usage error"
 	test_for_root
 
-	if [[ -L "${ROOT}/usr/bin/emacs" ]]; then
+	if [[ -L "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs" ]]; then
 		[[ ${1} == "--if-unset" \
-			&& -e "$(canonicalise ${ROOT}/usr/bin/emacs)" ]] && return
+			&& -e "$(canonicalise ${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs)" ]] && return
 		# this is not redundant: "update" is called in pkg_postrm() of emacs
 		# and should clean up any dead symlinks if no valid target exists
 		remove_symlinks || die -q "Couldn't remove existing symlink"
-	elif [[ -e "${ROOT}/usr/bin/emacs" ]]; then
-		die -q "Sorry, ${ROOT}/usr/bin/emacs exists but is not a symlink"
+	elif [[ -e "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs" ]]; then
+		die -q "Sorry, ${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/emacs exists but is not a symlink"
 	fi
 
 	local targets=( $(find_targets) )
 	if [[ ${#targets[@]} -gt 0 ]]; then
 		set_symlinks ${#targets[@]} || die -q "Couldn't set a new symlink"
-	elif [[ -f "${ROOT}/usr/bin/xemacs" ]]; then
+	elif [[ -f "${ROOT}@GENTOO_PORTAGE_EPREFIX@/usr/bin/xemacs" ]]; then
 		# no Emacs target found -- link ctags, etags etc. to XEmacs versions
 		set_symlinks xemacs
 	fi
