--- lxpanel/src/plugins/Makefile.am	2009-01-02 00:52:08.000000000 +0100
+++ lxpanel-fixed/src/plugins/Makefile.am	2009-01-02 01:15:48.000000000 +0100
@@ -46,6 +46,4 @@
 	icon.xpm
 
 install-exec-hook:
-	rm -f $(DESTDIR)$(libdir)/lxpanel/plugins/*.la
-	rm -f $(DESTDIR)$(libdir)/lxpanel/plugins/*.a
-	rm -f $(DESTDIR)$(libdir)/lxpanel/plugins/*.
+
--- lxpanel/src/plugins/netstat/Makefile.am	2009-01-02 00:52:07.000000000 +0100
+++ lxpanel-fixed/src/plugins/netstat/Makefile.am	2009-01-02 01:18:50.000000000 +0100
@@ -33,6 +33,5 @@
     @LXPANEL_MODULE@
 
 install-exec-hook:
-	rm -f "$(moduledir)\*.la" *.a
 
 endif
--- lxpanel/src/plugins/netstatus/Makefile.am	2009-01-02 00:52:08.000000000 +0100
+++ lxpanel-fixed/src/plugins/netstatus/Makefile.am	2009-01-02 01:19:08.000000000 +0100
@@ -36,4 +36,4 @@
 	$(LXPANEL_MODULE)
 
 install-exec-hook:
-	rm -f "$(moduledir)\*.la" *.a
+
