=== modified file 'buildbot/test/test_vc.py'
--- buildbot/test/test_vc.py	2007-03-08 15:34:45 +0000
+++ buildbot/test/test_vc.py	2007-03-08 15:38:45 +0000
@@ -1083,6 +1083,8 @@
     try_branchname = "branch"
 
     def capable(self):
+        if not os.geteuid():
+            return (False, "CVS tests fail as root")
         cvspaths = which('cvs')
         if not cvspaths:
             return (False, "CVS is not installed")

