Index: asymptote-1.59/settings.cc
===================================================================
--- asymptote-1.59.orig/settings.cc
+++ asymptote-1.59/settings.cc
@@ -80,14 +80,14 @@ const bool haveglut=false;
 bool msdos=false;
 const char *HOME="HOME";
 const char pathSeparator=':';
-string defaultPSViewer="gv";
+string defaultPSViewer="xdg-open";
 #ifdef __APPLE__
 string defaultPDFViewer="open";
 #else  
-string defaultPDFViewer="acroread";
+string defaultPDFViewer="xdg-open";
 #endif  
 string defaultGhostscript="gs";
-string defaultDisplay="display";
+string defaultDisplay="xdg-open";
 string defaultPython;
 const string docdir=ASYMPTOTE_DOCDIR;
 void queryRegistry() {}
