diff -U2 -r asymptote-1.46.orig/settings.cc asymptote-1.46/settings.cc
--- asymptote-1.46.orig/settings.cc	2008-10-19 01:38:49.000000000 +0700
+++ asymptote-1.46/settings.cc	2008-10-21 01:07:49.000000000 +0700
@@ -81,12 +81,12 @@
 const char *HOME="HOME";
 const char pathSeparator=':';
-string defaultPSViewer="gv";
+string defaultPSViewer="xdg-open";
 #ifdef __APPLE__
 string defaultPDFViewer="open";
 #else  
-string defaultPDFViewer="acroread";
+string defaultPDFViewer="xdg-open";
 #endif  
 string defaultGhostscript="gs";
-string defaultDisplay="display";
+string defaultDisplay="xdg-open";
 string defaultPython;
 const string docdir=ASYMPTOTE_DOCDIR;
