diff -r -U2 asymptote-1.53.orig/settings.cc asymptote-1.53/settings.cc
--- asymptote-1.53.orig/settings.cc	2008-11-28 00:43:50.000000000 +0600
+++ asymptote-1.53/settings.cc	2008-11-29 03:15:52.000000000 +0600
@@ -81,12 +81,12 @@
 const char *HOME="HOME";
 const char pathSeparator=':';
-string defaultPSViewer="gv";
+string defaultPSViewer="xdg-open";
 #ifdef __APPLE__
 string defaultPDFViewer="open";
 #else  
-string defaultPDFViewer="acroread";
+string defaultPDFViewer="xdg-open";
 #endif  
 string defaultGhostscript="gs";
-string defaultDisplay="display";
+string defaultDisplay="xdg-open";
 string defaultPython;
 const string docdir=ASYMPTOTE_DOCDIR;
