Index: asymptote-1.78/settings.cc
===================================================================
--- asymptote-1.78.orig/settings.cc
+++ asymptote-1.78/settings.cc
@@ -83,15 +83,15 @@ const bool haveglut=false;
 bool msdos=false;
 const char *HOME="HOME";
 const char pathSeparator=':';
-string defaultPSViewer="gv";
+string defaultPSViewer="xdg-open";
 #ifdef __APPLE__
 string defaultPDFViewer="open";
 #else  
-string defaultPDFViewer="acroread";
+string defaultPDFViewer="xdg-open";
 #endif  
 string defaultGhostscript="gs";
 string defaultPython;
-string defaultDisplay="display";
+string defaultDisplay="xdg-open";
 string systemDir=ASYMPTOTE_SYSDIR;
 string docdir=ASYMPTOTE_DOCDIR;
 void queryRegistry() {}
