diff -U2 --recursive asymptote-1.41.orig/Makefile.in asymptote-1.41/Makefile.in
--- asymptote-1.41.orig/Makefile.in	2008-02-02 07:13:09.000000000 +0600
+++ asymptote-1.41/Makefile.in	2008-02-04 06:23:35.000000000 +0600
@@ -51,5 +51,5 @@
 GUIdir = $(asydir)/GUI
 docdir = $(DESTDIR)@docdir@
-exampledir = $(docdir)/examples
+exampledir = $(asydir)/examples
 animationsdir = $(exampledir)/animations
 latexdir = $(DESTDIR)@latexdir@
@@ -136,9 +136,6 @@
 	${INSTALL} -p -m 644 examples/animations/*.asy $(animationsdir)
 	-${INSTALL} -p -m 644 $(addprefix doc/,$(LATEXFILES)) $(latexdir)
-	-if test -z "$(DESTDIR)"; then \
-	  texhash; \
-	fi
 
-install-doc: html
+install-doc:
 	cd doc; $(MAKE) install-all
 
diff -U2 --recursive asymptote-1.41.orig/doc/Makefile asymptote-1.41/doc/Makefile
--- asymptote-1.41.orig/doc/Makefile	2008-02-02 07:14:11.000000000 +0600
+++ asymptote-1.41/doc/Makefile	2008-02-04 06:23:50.000000000 +0600
@@ -17,10 +17,10 @@
 html:	asymptote/index.html
 
-man:	$(DOCFILES) asy.1
+man:	$(MANFILES)
 
 faq:	
 	cd FAQ; $(MAKE) faq
 
-doc:	asymptote.dvi asymptote.info html man faq
+doc:	$(DOCFILES) asymptote.info html man faq
 
 %.png:	%.asy
@@ -85,13 +85,12 @@
 	rm -rf asymptote
 
-install: man faq
-	cd FAQ; $(MAKE) install
-	${INSTALL} -d -m 755 $(docdir) $(mandir)/man1
-	${INSTALL} -p -m 644 $(DOCFILES) $(docdir)
-	${INSTALL} -p -m 644 $(MANFILES) $(mandir)/man1
+install: man
+	${INSTALL} -d -m 755 $(mandir)/man1
 
-install-all: doc install
-	cd FAQ; $(MAKE) install-info
-	${INSTALL} -d -m 755 $(infodir)
+install-all: doc faq install
+	cd FAQ; $(MAKE) install-all
+	${INSTALL} -d -m 755 $(infodir) $(docdir)/html/asymptote
+	${INSTALL} -p -m 644 $(DOCFILES) $(docdir)
+	${INSTALL} -p -m 644 asymptote/* $(docdir)/html/asymptote
 	${INSTALL} -p -m 644 asymptote.info $(infodir)
 	-${INSTALL} -p -m 644 asymptote.info-* $(infodir)
