Index: asymptote-1.81/settings.cc
===================================================================
--- asymptote-1.81.orig/settings.cc
+++ asymptote-1.81/settings.cc
@@ -82,14 +82,14 @@ mode_t mask;
 bool msdos=false;
 string HOME="HOME";
 const char pathSeparator=':';
-string defaultPSViewer="gv";
+string defaultPSViewer="xdg-open";
 #ifdef __APPLE__
 string defaultPDFViewer="open";
 #else  
-string defaultPDFViewer="acroread";
+string defaultPDFViewer="xdg-open";
 #endif  
 string defaultGhostscript="gs";
-string defaultDisplay="display";
+string defaultDisplay="xdg-open";
 string defaultAnimate="animate";
 string systemDir=ASYMPTOTE_SYSDIR;
 string docdir=ASYMPTOTE_DOCDIR;
