--- xfprint/print_dialog.c.orig	2008-01-24 19:59:21.000000000 +0100
+++ xfprint/print_dialog.c	2008-01-24 19:59:53.000000000 +0100
@@ -406,7 +406,7 @@
                         PRINTER_NAME_COLUMN, printer_data->name, 
                         PRINTER_ALIAS_COLUMN, printer_data->alias ? printer_data->alias : "", -1);
     
-    if (!default_printer_found && g_ascii_strcasecmp (default_printer->name, printer_data->name) == 0) {
+    if (default_printer && !default_printer_found && g_ascii_strcasecmp (default_printer->name, printer_data->name) == 0) {
       default_printer_found = TRUE;
       gtk_combo_box_set_active_iter (GTK_COMBO_BOX (priv->combobox_printer), &iter);
     }
