diff -ru djvulibre-3.5.21.orig/libdjvu/atomic.cpp djvulibre-3.5.21/libdjvu/atomic.cpp
--- djvulibre-3.5.21.orig/libdjvu/atomic.cpp	2008-11-04 09:24:25 +0100
+++ djvulibre-3.5.21/libdjvu/atomic.cpp	2008-11-04 09:32:06 +0100
@@ -144,7 +144,7 @@
 #  define USE_INTEL_ATOMIC_BUILTINS 1
 # endif
 #endif
-#if HAVE_INTEL_ATOMIC_BUILTINS
+#if HAVE_INTEL_ATOMIC_BUILTINS && !defined(__INTERIX)
 # define USE_INTEL_ATOMIC_BUILTINS 1
 #elif defined(WIN32)
 # define USE_WIN32_INTERLOCKED 1
