diff -ur libspiff-0.8.3.orig/test/ChunkCallbackSuite.cpp libspiff-0.8.3/test/ChunkCallbackSuite.cpp
--- libspiff-0.8.3.orig/test/ChunkCallbackSuite.cpp	2007-09-17 02:43:41.000000000 +0300
+++ libspiff-0.8.3/test/ChunkCallbackSuite.cpp	2008-04-14 21:15:18.000000000 +0300
@@ -23,6 +23,7 @@
 #include "ChunkCallbackSuite.h"
 #include <spiff/SpiffReader.h>
 #include <memory>
+#include <cstring>
 using namespace Spiff;
 using namespace std;
 
