--- binutils-config-1.9	2008-04-08 14:53:19.000000000 +0200
+++ binutils-config	2008-04-08 15:04:43.000000000 +0200
@@ -119,8 +119,15 @@
 			ld|ld64)
 				# remove symlink first if exists, to avoid overwriting
 				# the target in the cp that follows
-				rm -f "${EROOT}/${BINPATH_LINKS}/${x}" > /dev/null
-				cp -f "${wrapper}" "${EROOT}/${BINPATH_LINKS}/${x}"
+				"${wrapper}"/create-ldwrapper \
+					--wrapper-name="${x}" \
+					--wrapper-source="${wrapper}" \
+					--output-file="${EPREFIX}/${BINPATH_LINKS}/${x}" \
+					--host="${HOST}" \
+					--target="${TARGET}" \
+					--binpath="${BINPATH}" \
+					--debug="${DEBUG}" \
+				|| return 1
 				touch -r "${EPREFIX}/${BINPATH}/${x}" "${EROOT}/${BINPATH_LINKS}/${x}"
 			;;
 			*)
