diff -ru toolchain-prefix-wrapper-0.3.1660.orig/ld/gnuplugin.c toolchain-prefix-wrapper-0.3.1660/ld/gnuplugin.c
--- toolchain-prefix-wrapper-0.3.1660.orig/ld/gnuplugin.c	2009-11-09 12:11:24.552590166 +0100
+++ toolchain-prefix-wrapper-0.3.1660/ld/gnuplugin.c	2009-11-09 14:42:56.056096154 +0100
@@ -29,6 +29,7 @@
 	int argc = 0;
 	int err;
 	StringList *runpathList = NULL;
+	StringList *tmpArgList = NULL;
 	String const *argString;
 	String *newString = NULL;
 	char const *argBuffer;
@@ -44,6 +45,10 @@
 		if (runpathList == NULL)
 			break;
 
+		tmpArgList = StringListCreate(NULL, 0, 0);
+		if (tmpArgList == NULL)
+			break;
+
 		err = 0;
 		for(argc = 1; !err && argc < StringListGetSize(data->in->argList); argc++) {
 			err = -1;
@@ -105,7 +110,7 @@
 						 * entire argument in that case, and not just
 						 * everything up to the first : */
 						if (rv != 0 || (sbuf.st_mode & S_IFDIR) == 0) {
-							if (StringListAppendConcat(data->out->argList
+							if (StringListAppendConcat(tmpArgList
 									, _R, _R_len
 									, curr, -1
 									, NULL) < 0)
@@ -150,17 +155,37 @@
 				continue;
 			}
 
+			if (strncmp(argBuffer, "-L", 2) == 0) {
+				/* drop -L arguments, as they are in userLibpath list already,
+				 * which is added below before sysLibpath list. */
+				err = 0;
+				continue;
+			}
+
 			/* keep other arguments on commandline */
-			if (StringListAppendString(data->out->argList, argString) < 0)
+			if (StringListAppendString(tmpArgList, argString) < 0)
 				break;
 
 			err = 0;
 		}
+
 		if (argc < StringListGetSize(data->in->argList)) {
 			/* error during argument parsing */
 			break;
 		}
 
+		/* append user libpath's with "-L" */
+		if (StringListAppendListModify(data->out->argList, data->in->userLibpath, 0, -1, "-L", 2, NULL, 0) < 0)
+			break;
+
+		/* append sys libpath's with "-L" */
+		if (StringListAppendListModify(data->out->argList, data->in->sysLibpath, 0, -1, "-L", 2, NULL, 0) < 0)
+			break;
+
+		/* append other arguments */
+		if (StringListAppendList(data->out->argList, tmpArgList, 0, -1) < 0)
+			break;
+
 		/* add sys libpath's to runpath list */
 		if (StringListAppendList(runpathList, data->in->sysRunpath, 0, -1) < 0)
 			break;
@@ -179,15 +204,12 @@
 			if (StringListAppendString(data->out->argList, newString) < 0) break;
 		}
 
-		/* append sys libpath's with "-L" */
-		if (StringListAppendListModify(data->out->argList, data->in->sysLibpath, 0, -1, "-L", 2, NULL, 0) < 0)
-			break;
-
 		err = 0;
 	} while(0);	/* end dummy loop */
 
 	if (newString != NULL) newString = StringDestroy(newString);
 	if (runpathList != NULL) runpathList = StringListDestroy(runpathList);
+	if (tmpArgList != NULL) tmpArgList = StringListDestroy(tmpArgList);
 
 	return err;
 }
diff -ru toolchain-prefix-wrapper-0.3.1660.orig/ld/ldwrapper.c toolchain-prefix-wrapper-0.3.1660/ld/ldwrapper.c
--- toolchain-prefix-wrapper-0.3.1660.orig/ld/ldwrapper.c	2009-11-09 12:11:24.554791134 +0100
+++ toolchain-prefix-wrapper-0.3.1660/ld/ldwrapper.c	2009-11-09 14:20:30.719214769 +0100
@@ -734,11 +734,11 @@
 
 	if ((*plugins)(&pluginData) < 0) {
 		if (pluginData.out->diagnostics == NULL) {
-			wrapper_exit("%s wrapper: %s: %s\n",
-				StringGetBuffer(data->name), pluginData.out->diagnostics, strerror(errno));
-		} else {
 			wrapper_exit("%s wrapper could not run plugin: %s\n",
 				StringGetBuffer(data->name), strerror(errno));
+		} else {
+			wrapper_exit("%s wrapper: %s: %s\n",
+				StringGetBuffer(data->name), pluginData.out->diagnostics, strerror(errno));
 		}
 	}
 	if (pluginOut.argList != NULL) {
