--- nautilus-2.20.0/configure.in
+++ nautilus-2.20.0/configure.in
@@ -42,7 +42,7 @@
 AC_SUBST(BONOBO_REQUIRED, [bonobo_minver])
 AC_SUBST(EEL_REQUIRED, [eel_minver])
 AC_SUBST(ESOUND_REQUIRED, [esound_minver])
-AC_SUBST(GLIB_REQUIRED [glib_minver])
+AC_SUBST(GLIB_REQUIRED, [glib_minver])
 AC_SUBST(GNOME_DESKTOP_REQUIRED, [gnome_desktop_minver])
 AC_SUBST(GNOME_REQUIRED, [gnome_minver])
 AC_SUBST(GNOME_UI_REQUIRED, [gnome_ui_minver])
