Fix crash with small fonts (#454884)
--- src/nautilus-sidebar-title.c	2007/10/03 06:38:48	13255
+++ src/nautilus-sidebar-title.c	2007/10/03 06:39:37	13256
@@ -364,7 +364,7 @@
 	title_font = pango_font_description_copy (style->font_desc);
 
 	max_style_font_size = pango_font_description_get_size (title_font) * 1.8 / PANGO_SCALE;
-	if (max_style_font_size < MIN_TITLE_FONT_SIZE) {
+	if (max_style_font_size < MIN_TITLE_FONT_SIZE + 1) {
 		max_style_font_size = MIN_TITLE_FONT_SIZE + 1;
 	}
 
