So we can check for statfs in FreeBSD

diff -ur a/configure.in b/configure.in
--- a/configure.in	2007-10-19 10:41:53 +0100
+++ b/configure.in	2007-10-19 10:42:53 +0100
@@ -149,7 +149,7 @@
 
 dnl ==========================================================================
 
-AC_CHECK_HEADERS(sys/mount.h sys/vfs.h)
+AC_CHECK_HEADERS(sys/param.h sys/mount.h sys/stat.h sys/vfs.h)
 			      
 dnl ==========================================================================
 
