--- nautilus-2.22.2/libnautilus-private/apps_nautilus_preferences.schemas.in.orig	2008-05-02 14:05:28.000000000 -0400
+++ nautilus-2.22.2/libnautilus-private/apps_nautilus_preferences.schemas.in	2008-05-02 14:08:30.000000000 -0400
@@ -122,7 +122,7 @@
       <owner>nautilus</owner>
       <type>list</type>
       <list_type>string</list_type>
-      <default>[]</default>
+      <default>[x-content/audio-cdda,x-content/video-dvd,x-content/audio-player,x-content/image-dcf,x-content/video-vcd,x-content/video-svcd,x-content/image-picturecd,x-content/video-hddvd,x-content/audio-dvd,x-content/blank-bd,x-content/blank-cd,x-content/blank-dvd,x-content/blank-hddvd,x-content/video-bluray]</default>
       <locale name="C">
          <short>List of x-content/* types to ask the user what to do on insertion</short>
          <long>
