From bf6bfde36bcadf43a39fb095650b7fc78741d62a Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Thu, 27 Aug 2009 00:05:28 +0000
Subject: Fix a duplicate id in a .ui file

This caused nautilus-file-management-properties to crash.
---
diff --git a/src/nautilus-file-management-properties.ui b/src/nautilus-file-management-properties.ui
index eae16d1..c4fb6ee 100644
--- a/src/nautilus-file-management-properties.ui
+++ b/src/nautilus-file-management-properties.ui
@@ -1128,7 +1128,7 @@
                               </packing>
                             </child>
                             <child>
-                              <object class="GtkLabel" id="label">
+                              <object class="GtkLabel" id="label99">
                                 <property name="height_request">6</property>
                                 <property name="visible">True</property>
                                 <property name="label" translatable="yes"/>
--
cgit v0.8.2
