*** makeinfo.c.~1.60.~  Sun Jun  6 07:04:12 2004
--- makeinfo.c  Fri Jul  2 08:42:28 2004
***************
*** 3934,3940 ****
  
    /* If insertion stack level changes during execution, that means a multiline
       command is used inside braces or @section ... kind of commands.  */
!   if (insertion_level_at_start != insertion_level)
      {
        line_error (_("Multiline command %c%s used improperly"),
            COMMAND_PREFIX,
--- 3934,3940 ----
  
    /* If insertion stack level changes during execution, that means a multiline
       command is used inside braces or @section ... kind of commands.  */
!   if (insertion_level_at_start != insertion_level && !executing_macro)
      {
        line_error (_("Multiline command %c%s used improperly"),
            COMMAND_PREFIX,
