--- gcc/config/rs6000/aix52.h.orig	2009-04-09 13:51:21 +0200
+++ gcc/config/rs6000/aix52.h	2009-04-09 15:49:14 +0200
@@ -150,6 +150,7 @@
 
 #undef LINK_SPEC
 #define LINK_SPEC "-bpT:0x10000000 -bpD:0x20000000 %{!r:-btextro} -bnodelcsect\
+   %{!shared:-bexpfull}\
    %{static:-bnso %(link_syscalls) } %{shared:-bM:SRE %{!e:-bnoentry}}\
    %{!maix64:%{!shared:%{g*: %(link_libg) }}} %{maix64:-b64}\
    %{mpe:-binitfini:poe_remote_main}"
