--- emacs-orig/src/dispextern.h	2007-09-13 10:50:51.000000000 +0000
+++ emacs/src/dispextern.h	2007-10-12 21:57:44.000000000 +0000
@@ -853,9 +853,11 @@
   /* Continuation lines width at the start of the row.  */
   int continuation_lines_width;
 
+#ifdef HAVE_WINDOW_SYSTEM
   /* Non-NULL means the current clipping area.  This is temporarily
      set while exposing a region.  Coordinates are frame-relative.  */
   XRectangle *clip;
+#endif
 };
 
 
