--- configure.in	12 Sep 2007 07:03:30 -0000	1.474
+++ configure.in	15 Sep 2007 19:45:32 -0000	1.475
@@ -1347,7 +1347,7 @@
 dnl By this stage, configure has already checked for egrep and set EGREP, 
 dnl or exited with an error if no egrep was found.
 if test "$MAKEINFO" != "no" && \
-  test x"`$MAKEINFO --version 2> /dev/null | $EGREP 'texinfo[[^0-9]]*([[5-9]]|4\.[[6-9]])'`" = x; then
+  test x"`$MAKEINFO --version 2> /dev/null | $EGREP 'texinfo[[^0-9]]*([[1-4]][[0-9]]+|[[5-9]]|4\.[[6-9]]|4\.[[1-5]][[0-9]]+)'`" = x; then
    MAKEINFO=no    
 fi
 
