--- emacs-orig/src/alloc.c	2008-07-09 19:33:18.000000000 +0200
+++ emacs/src/alloc.c	2008-07-09 19:39:48.000000000 +0200
@@ -4487,7 +4487,7 @@
   /* Fixme: Code in the Boehm GC suggests flushing (with `flushrs') is
      needed on ia64 too.  See mach_dep.c, where it also says inline
      assembler doesn't work with relevant proprietary compilers.  */
-#ifdef __sparc__
+#if defined(__sparc__) && !defined(__FreeBSD__)
   asm ("ta 3");
 #endif
 
