--- configure.in.orig	Wed Sep 22 02:03:18 2004
+++ configure.in	Wed Sep 22 02:11:16 2004
@@ -1126,12 +1126,12 @@
     machine=powermac opsys=darwin
     # Define CPP as follows to make autoconf work correctly.
     CPP="${CC-cc} -E -no-cpp-precomp"
-    # Use fink packages if available.
-    if test -d /sw/include && test -d /sw/lib; then
-	GCC_TEST_OPTIONS="-I/sw/include -L/sw/lib"
-        CPP="${CPP} ${GCC_TEST_OPTIONS}"
-	NON_GCC_TEST_OPTIONS=${GCC_TEST_OPTIONS}
-    fi
+#    # Use fink packages if available.
+#    if test -d /sw/include && test -d /sw/lib; then
+#	GCC_TEST_OPTIONS="-I/sw/include -L/sw/lib"
+#        CPP="${CPP} ${GCC_TEST_OPTIONS}"
+#	NON_GCC_TEST_OPTIONS=${GCC_TEST_OPTIONS}
+#    fi
   ;;
 
   ## AMD x86-64 Linux-based GNU system
--- configure.orig	Wed Sep 22 02:18:49 2004
+++ configure	Wed Sep 22 02:20:03 2004
@@ -2601,12 +2601,12 @@
     machine=powermac opsys=darwin
     # Define CPP as follows to make autoconf work correctly.
     CPP="${CC-cc} -E -no-cpp-precomp"
-    # Use fink packages if available.
-    if test -d /sw/include && test -d /sw/lib; then
-	GCC_TEST_OPTIONS="-I/sw/include -L/sw/lib"
-        CPP="${CPP} ${GCC_TEST_OPTIONS}"
-	NON_GCC_TEST_OPTIONS=${GCC_TEST_OPTIONS}
-    fi
+#    # Use fink packages if available.
+#    if test -d /sw/include && test -d /sw/lib; then
+#	GCC_TEST_OPTIONS="-I/sw/include -L/sw/lib"
+#        CPP="${CPP} ${GCC_TEST_OPTIONS}"
+#	NON_GCC_TEST_OPTIONS=${GCC_TEST_OPTIONS}
+#    fi
   ;;
 
   ## AMD x86-64 Linux-based GNU system
