diff -ru -x '*.Po' -x '*.Plo' gstreamer-0.10.20.orig/libs/gst/net/gstnettimepacket.c gstreamer-0.10.20/libs/gst/net/gstnettimepacket.c
--- gstreamer-0.10.20.orig/libs/gst/net/gstnettimepacket.c	2008-07-29 08:56:37 +0200
+++ gstreamer-0.10.20/libs/gst/net/gstnettimepacket.c	2008-07-29 12:02:15 +0200
@@ -190,8 +190,12 @@
   flags = 1;
   send_flags = 0;
 #else
+#ifdef __INTERIX
+  send_flags = 0;
+#else
   send_flags = MSG_DONTWAIT;
 #endif
+#endif
 
   buffer = gst_net_time_packet_serialize (packet);
 
diff -ru -x '*.Po' -x '*.Plo' gstreamer-0.10.20.orig/libs/gst/net/gstnettimepacket.h gstreamer-0.10.20/libs/gst/net/gstnettimepacket.h
--- gstreamer-0.10.20.orig/libs/gst/net/gstnettimepacket.h	2008-07-29 08:56:37 +0200
+++ gstreamer-0.10.20/libs/gst/net/gstnettimepacket.h	2008-07-29 11:59:16 +0200
@@ -36,6 +36,11 @@
 #define socklen_t int
 #endif
 #else
+#ifdef __INTERIX
+#ifndef socklen_t
+#define socklen_t int
+#endif
+#endif
 #include <netdb.h>
 #include <sys/socket.h>
 #include <netinet/in.h>

