diff -ru -x '*.Po' -x '*.Plo' gstreamer-0.10.19.orig/gst/gstpoll.c gstreamer-0.10.19/gst/gstpoll.c
--- gstreamer-0.10.19.orig/gst/gstpoll.c	2008-04-15 10:36:33 +0200
+++ gstreamer-0.10.19/gst/gstpoll.c	2008-04-15 10:41:43 +0200
@@ -58,7 +58,14 @@
 
 #ifndef _MSC_VER
 #define _GNU_SOURCE 1
+#ifndef __INTERIX
 #include <sys/poll.h>
+#else
+// just for definitions, since these are used for
+// select mode too. poll() is broken, so this mode
+// is disabled on interix.
+#include <poll.h>
+#endif
 #include <sys/time.h>
 #endif
 
