diff -ru -x '*.Po' -x '*.Plo' gstreamer-0.10.17.orig/libs/gst/net/gstnetclientclock.c gstreamer-0.10.17/libs/gst/net/gstnetclientclock.c
--- gstreamer-0.10.17.orig/libs/gst/net/gstnetclientclock.c	2008-03-17 13:14:32 +0100
+++ gstreamer-0.10.17/libs/gst/net/gstnetclientclock.c	2008-03-17 13:15:41 +0100
@@ -61,6 +61,10 @@
 #include <io.h>
 #endif
 
+#if defined(__INTERIX)
+# include <sys/time.h>
+#endif
+
 GST_DEBUG_CATEGORY_STATIC (ncc_debug);
 #define GST_CAT_DEFAULT (ncc_debug)
 
diff -ru -x '*.Po' -x '*.Plo' gstreamer-0.10.17.orig/libs/gst/net/gstnettimeprovider.c gstreamer-0.10.17/libs/gst/net/gstnettimeprovider.c
--- gstreamer-0.10.17.orig/libs/gst/net/gstnettimeprovider.c	2008-03-17 13:14:32 +0100
+++ gstreamer-0.10.17/libs/gst/net/gstnettimeprovider.c	2008-03-17 13:16:51 +0100
@@ -60,6 +60,10 @@
 #include <sys/filio.h>
 #endif
 
+#if defined(__INTERIX)
+# include <sys/time.h>
+#endif
+
 GST_DEBUG_CATEGORY_STATIC (ntp_debug);
 #define GST_CAT_DEFAULT (ntp_debug)
 
diff -ru -x '*.Po' -x '*.Plo' gstreamer-0.10.17.orig/plugins/elements/gstfdsink.c gstreamer-0.10.17/plugins/elements/gstfdsink.c
--- gstreamer-0.10.17.orig/plugins/elements/gstfdsink.c	2008-03-17 13:14:32 +0100
+++ gstreamer-0.10.17/plugins/elements/gstfdsink.c	2008-03-17 13:17:51 +0100
@@ -54,6 +54,10 @@
 #include <errno.h>
 #include <string.h>
 
+#if defined(__INTERIX)
+# include <sys/time.h>
+#endif
+
 #include "gstfdsink.h"
 
 /* We add a control socket as in fdsrc to make it shutdown quickly when it's blocking on the fd.
diff -ru -x '*.Po' -x '*.Plo' gstreamer-0.10.17.orig/plugins/elements/gstfdsrc.c gstreamer-0.10.17/plugins/elements/gstfdsrc.c
--- gstreamer-0.10.17.orig/plugins/elements/gstfdsrc.c	2008-03-17 13:14:32 +0100
+++ gstreamer-0.10.17/plugins/elements/gstfdsrc.c	2008-03-17 13:17:29 +0100
@@ -48,6 +48,10 @@
 #include <stdlib.h>
 #include <errno.h>
 
+#if defined(__INTERIX)
+# include <sys/time.h>
+#endif
+
 #include "gstfdsrc.h"
 
 /* the select call is also performed on the control sockets, that way
