Index: subversion-1.4.6/configure.in
===================================================================
--- subversion-1.4.6.orig/configure.in
+++ subversion-1.4.6/configure.in
@@ -245,7 +245,7 @@ powerpc-apple-darwin*)
 esac
 # Look for libdb4.so first:
 SVN_LIB_BERKELEY_DB($SVN_FS_WANT_DB_MAJOR, $SVN_FS_WANT_DB_MINOR,
-                    $SVN_FS_WANT_DB_PATCH, [db4 db])
+                    $SVN_FS_WANT_DB_PATCH, [db-4.3 db-4.2 db-4.0 db-4 db-4.1 db4 db])
 
 AC_DEFINE_UNQUOTED(SVN_FS_WANT_DB_MAJOR, $SVN_FS_WANT_DB_MAJOR,
                    [The desired major version for the Berkeley DB])
