Index: subversion-1.4.6/tools/backup/hot-backup.py.in
===================================================================
--- subversion-1.4.6.orig/tools/backup/hot-backup.py.in
+++ subversion-1.4.6/tools/backup/hot-backup.py.in
@@ -34,7 +34,14 @@ svnlook = "@SVN_BINDIR@/svnlook"
 svnadmin = "@SVN_BINDIR@/svnadmin"
 
 # Number of backups to keep around (0 for "keep them all")
-num_backups = 64
+# Configurable in /etc/env.d/80hotbackup
+envvar_hb_num='SVN_HOTBACKUP_NUM_BACKUPS'
+if os.environ.has_key(envvar_hb_num):
+	num_backups = int(os.environ[envvar_hb_num])
+else:
+	num_backups = 64
+print 'Keeping up to', num_backups, 'around.'			
+
 
 # Archive types/extensions
 archive_map = {
