https://svn.collab.net/viewvc/svn?view=revision&revision=37710

--- subversion/libsvn_fs_base/bdb/changes-table.c
+++ subversion/libsvn_fs_base/bdb/changes-table.c
@@ -33,7 +33,7 @@
 #include "dbt.h"
 #include "changes-table.h"
 
-
+#include "private/svn_fs_util.h"
 #include "svn_private_config.h"
 
 
@@ -217,7 +217,7 @@
       /* This change is new to the hash, so make a new public change
          structure from the internal one (in the hash's pool), and dup
          the path into the hash's pool, too. */
-      new_change = svn_fs_path_change2_create(
+      new_change = svn_fs__path_change2_create(
                        svn_fs_base__id_copy(change->noderev_id, pool),
                        change->kind,
                        pool);
--- subversion/libsvn_fs_fs/fs_fs.c
+++ subversion/libsvn_fs_fs/fs_fs.c
@@ -4780,7 +4780,7 @@
                            APR_APPEND | APR_WRITE | APR_CREATE
                            | APR_BUFFERED, APR_OS_DEFAULT, pool));
 
-  change = svn_fs_path_change2_create(id, change_kind, pool);
+  change = svn_fs__path_change2_create(id, change_kind, pool);
   change->text_mod = text_mod;
   change->prop_mod = prop_mod;
   change->node_kind = node_kind;
