https://svn.collab.net/viewvc/svn?view=revision&revision=36720

--- subversion/mod_dav_svn/liveprops.c
+++ subversion/mod_dav_svn/liveprops.c
@@ -466,7 +466,7 @@
                      && resource->info->r->content_type)
               mime_type = resource->info->r->content_type;
             else
-              mime_type = ap_default_type(resource->info->r);
+              mime_type = "application/octet-stream";
 
             if ((serr = svn_mime_type_validate(mime_type, p)))
               {
--- subversion/mod_dav_svn/repos.c
+++ subversion/mod_dav_svn/repos.c
@@ -2690,7 +2690,7 @@
                && r->content_type)
         mimetype = r->content_type;
       else
-        mimetype = ap_default_type(r);
+        mimetype = "application/octet-stream";
 
       serr = svn_mime_type_validate(mimetype, resource->pool);
       if (serr)
