diff -ru subversion-1.4.6.orig/build/ac-macros/neon.m4 subversion-1.4.6/build/ac-macros/neon.m4
--- subversion-1.4.6.orig/build/ac-macros/neon.m4	Tue Jan 22 10:15:51 2008
+++ subversion-1.4.6/build/ac-macros/neon.m4	Tue Jan 22 10:16:47 2008
@@ -50,11 +50,11 @@
       NEON_VERSION=`cat $abs_srcdir/neon/.version`
       AC_MSG_RESULT([$NEON_VERSION])
 
-      if test -n "`echo \"$NEON_VERSION\" | grep '^0\.2[[56]]\.'`" ; then
+      if test -n "`echo $NEON_VERSION | grep '^0\.2[[56]]\.'`" ; then
         AC_DEFINE_UNQUOTED([SVN_NEON_0_25], [1],
                            [Define to 1 if you have Neon 0.25 or later.])
       fi
-      if test -n "`echo \"$NEON_VERSION\" | grep '^0\.26\.'`" ; then
+      if test -n "`echo $NEON_VERSION | grep '^0\.26\.'`" ; then
         AC_DEFINE_UNQUOTED([SVN_NEON_0_26], [1],
                            [Define to 1 if you have Neon 0.26 or later.])
       fi
@@ -127,11 +127,11 @@
       NEON_VERSION=`$neon_config --version | sed -e 's/^neon //'`
       AC_MSG_RESULT([$NEON_VERSION])
 
-      if test -n "`echo \"$NEON_VERSION\" | grep '^0\.2[[56]]\.'`" ; then
+      if test -n "`echo $NEON_VERSION | grep '^0\.2[[56]]\.'`" ; then
         AC_DEFINE_UNQUOTED([SVN_NEON_0_25], [1],
                            [Define to 1 if you have Neon 0.25 or later.])
       fi
-      if test -n "`echo \"$NEON_VERSION\" | grep '^0\.26\.'`" ; then
+      if test -n "`echo $NEON_VERSION | grep '^0\.26\.'`" ; then
         AC_DEFINE_UNQUOTED([SVN_NEON_0_26], [1],
                            [Define to 1 if you have Neon 0.26 or later.])
       fi
