diff -ru lv451.orig/src/console.c lv451/src/console.c
--- lv451.orig/src/console.c	2008-04-12 15:18:21 +0200
+++ lv451/src/console.c	2008-04-12 15:17:44 +0200
@@ -422,7 +422,9 @@
 #ifdef VDISCRD /* IBM AIX */ 
 #define VDISCARD VDISCRD 
 #endif 
+#ifdef VDISCARD
   ttyNew.c_cc[ VDISCARD ] = -1;
+#endif
   ttyNew.c_cc[ VMIN ] = 1;
   ttyNew.c_cc[ VTIME ] = 0;
   tcsetattr( 0, TCSADRAIN, &ttyNew );
