## 05_font_override.dpatch by  <tmancill@debian.org>

diff -urN clusterssh-3.21/src/cssh.pl clusterssh-3.21-patched/src/cssh.pl
--- clusterssh-3.21/src/cssh.pl	2007-11-28 17:11:14.000000000 -0500
+++ clusterssh-3.21-patched/src/cssh.pl	2008-01-10 19:33:15.000000000 -0500
@@ -603,7 +603,7 @@
 sub update_display_text($) {
   my $char = shift;
 
-warn("config{show_history}=$config{show_history}");
+#warn("config{show_history}=$config{show_history}");
 
   return if(!$config{show_history});
 
