diff -ru clusterssh-3.21.orig/src/cssh.pl clusterssh-3.21/src/cssh.pl
--- clusterssh-3.21.orig/src/cssh.pl	2008-08-01 14:45:09 +0200
+++ clusterssh-3.21/src/cssh.pl	2008-08-01 14:45:23 +0200
@@ -433,19 +433,19 @@
       logmsg( 2, "Unknown keycode ", $keyboard[$_][1] )
         if ( $keyboard[$_][1] != 0 );
     }
-    if ( defined( $keycodetosym{ $keyboard[$_][2] } ) ) {
+    if (defined( $keyboard[$_][2] ) && defined( $keycodetosym{ $keyboard[$_][2] } ) ) {
       $keyboardmap{ $keycodetosym{ $keyboard[$_][2] } } = 'a' . ( $_ + $min );
     }
     else {
       logmsg( 2, "Unknown keycode ", $keyboard[$_][2] )
-        if ( $keyboard[$_][2] != 0 );
+        if ( defined( $keyboard[$_][2] ) && $keyboard[$_][2] != 0 );
     }
-    if ( defined( $keycodetosym{ $keyboard[$_][3] } ) ) {
+    if ( defined( $keyboard[$_][3] ) && defined( $keycodetosym{ $keyboard[$_][3] } ) ) {
       $keyboardmap{ $keycodetosym{ $keyboard[$_][3] } } = 'sa' . ( $_ + $min );
     }
     else {
       logmsg( 2, "Unknown keycode ", $keyboard[$_][3] )
-        if ( $keyboard[$_][3] != 0 );
+        if ( defined( $keyboard[$_][3] ) && $keyboard[$_][3] != 0 );
     }
 
     # dont know these two key combs yet...
@@ -1442,11 +1442,14 @@
     -buttons    => ['Close'],
   );
 
-  my $manpage = `pod2text -l -q=\"\" $0`;
-  $windows{mantext} =
-    $windows{manpage}->Scrolled( "Text", )->pack( -fill => 'both' );
-  $windows{mantext}->insert( 'end', $manpage );
-  $windows{mantext}->configure( -state => 'disabled' );
+  # this causes cssh to be unstartable on interix somehow...
+  # commenting this out seems to disable documentation in the
+  # main windows. use "man cssh" instead for now...
+  #my $manpage = `pod2text -l -q=\"\" $0`;
+  #$windows{mantext} =
+  #  $windows{manpage}->Scrolled( "Text", )->pack( -fill => 'both' );
+  #$windows{mantext}->insert( 'end', $manpage );
+  #$windows{mantext}->configure( -state => 'disabled' );
 
   $windows{addhost} = $windows{main_window}->DialogBox(
     -popover        => $windows{main_window},
