--- rules/72_active.cf.orig	2010-01-01 23:09:53.794608796 +0100
+++ rules/72_active.cf	2010-01-01 23:12:14.622599894 +0100
@@ -562,7 +562,7 @@
 ##} FH_DATE_IS_19XX
 
 ##{ FH_DATE_PAST_20XX
-header   FH_DATE_PAST_20XX	Date =~ /20[1-9][0-9]/ [if-unset: 2006]
+header   FH_DATE_PAST_20XX	Date =~ /20[2-9][0-9]/ [if-unset: 2006]
 describe FH_DATE_PAST_20XX	The date is grossly in the future.
 ##} FH_DATE_PAST_20XX
 
