Index: liboil-0.3.12/examples/audioresample/Makefile.am
===================================================================
--- liboil-0.3.12.orig/examples/audioresample/Makefile.am
+++ liboil-0.3.12/examples/audioresample/Makefile.am
@@ -1,8 +1,8 @@
 
-noinst_PROGRAMS = test_functable1
+EXTRA_PROGRAMS = test_functable1
 
-noinst_LTLIBRARIES = libaudioresample.la
-#noinst_PROGRAMS = audioresample_test
+EXTRA_LTLIBRARIES = libaudioresample.la
+#EXTRA_PROGRAMS = audioresample_test
 
 libaudioresample_la_SOURCES = functable.c resample.c
 libaudioresample_la_CFLAGS = $(LIBOIL_CFLAGS) $(GLIB_CFLAGS)
Index: liboil-0.3.12/examples/huffman/Makefile.am
===================================================================
--- liboil-0.3.12.orig/examples/huffman/Makefile.am
+++ liboil-0.3.12/examples/huffman/Makefile.am
@@ -1,5 +1,5 @@
 
-noinst_PROGRAMS = huffman_test
+EXTRA_PROGRAMS = huffman_test
 
 huffman_test_SOURCES = \
 	huffman.c \
Index: liboil-0.3.12/examples/jpeg/Makefile.am
===================================================================
--- liboil-0.3.12.orig/examples/jpeg/Makefile.am
+++ liboil-0.3.12/examples/jpeg/Makefile.am
@@ -1,7 +1,7 @@
 
 
-noinst_LTLIBRARIES = libjpeg.la
-noinst_PROGRAMS = jpeg_test jpeg_rgb_test
+EXTRA_LTLIBRARIES = libjpeg.la
+EXTRA_PROGRAMS = jpeg_test jpeg_rgb_test
 
 libjpeg_la_SOURCES = \
 	jpeg.c \
Index: liboil-0.3.12/examples/Makefile.am
===================================================================
--- liboil-0.3.12.orig/examples/Makefile.am
+++ liboil-0.3.12/examples/Makefile.am
@@ -3,7 +3,7 @@ SUBDIRS = jpeg md5 uberopt work huffman 
 
 bin_PROGRAMS = oil-bugreport
 
-noinst_PROGRAMS = example1 oil-graph oil-inspect oil-test memcpy-speed \
+EXTRA_PROGRAMS = example1 oil-graph oil-inspect oil-test memcpy-speed \
 	oil-suggest printcpu oil-mt19937 report oil-random
 
 
Index: liboil-0.3.12/examples/md5/Makefile.am
===================================================================
--- liboil-0.3.12.orig/examples/md5/Makefile.am
+++ liboil-0.3.12/examples/md5/Makefile.am
@@ -1,6 +1,6 @@
 
 
-noinst_PROGRAMS = md5sum
+EXTRA_PROGRAMS = md5sum
 
 md5sum_SOURCES = md5sum.c
 md5sum_CFLAGS = $(LIBOIL_CFLAGS)
Index: liboil-0.3.12/examples/taylor/Makefile.am
===================================================================
--- liboil-0.3.12.orig/examples/taylor/Makefile.am
+++ liboil-0.3.12/examples/taylor/Makefile.am
@@ -1,5 +1,5 @@
 
-noinst_PROGRAMS = example1
+EXTRA_PROGRAMS = example1
 
 
 example1_SOURCES = example1.c
Index: liboil-0.3.12/examples/uberopt/Makefile.am
===================================================================
--- liboil-0.3.12.orig/examples/uberopt/Makefile.am
+++ liboil-0.3.12/examples/uberopt/Makefile.am
@@ -3,9 +3,9 @@ EXTRA_DIST = example.c
 
 
 if HAVE_GLIB
-noinst_PROGRAMS = uberopt
+EXTRA_PROGRAMS = uberopt
 else
-noinst_PROGRAMS = 
+EXTRA_PROGRAMS = 
 endif
 
 uberopt_SOURCES = uberopt.c
Index: liboil-0.3.12/examples/videoscale/Makefile.am
===================================================================
--- liboil-0.3.12.orig/examples/videoscale/Makefile.am
+++ liboil-0.3.12/examples/videoscale/Makefile.am
@@ -1,7 +1,7 @@
 
 
-noinst_LTLIBRARIES = libvideoscale.la
-#noinst_PROGRAMS = videoscale_test
+EXTRA_LTLIBRARIES = libvideoscale.la
+#EXTRA_PROGRAMS = videoscale_test
 
 libvideoscale_la_SOURCES = vs_4tap.c vs_image.c vs_scanline.c
 libvideoscale_la_CFLAGS = $(LIBOIL_CFLAGS)
Index: liboil-0.3.12/examples/work/Makefile.am
===================================================================
--- liboil-0.3.12.orig/examples/work/Makefile.am
+++ liboil-0.3.12/examples/work/Makefile.am
@@ -1,8 +1,8 @@
 
 if HAVE_GLIB
-noinst_PROGRAMS = work
+EXTRA_PROGRAMS = work
 else
-noinst_PROGRAMS = 
+EXTRA_PROGRAMS = 
 endif
 
 work_SOURCES = work.c
