--- configure.in.orig	2009-09-30 15:48:24.338782000 +0200
+++ configure.in	2009-09-30 15:49:29.679511000 +0200
@@ -72,10 +72,11 @@
 dnl then add it
 dnl
 VERSION_SCRIPT_FLAGS=
-$(/usr/bin/ld --help 2>&1 | grep -- --version-script >/dev/null) && \
+if test "x$with_gnu_ld" = "xyes"; then
     VERSION_SCRIPT_FLAGS=-Wl,--version-script=
-test "`uname`" == "SunOS" && \
+elif test "`uname`" == "SunOS"; then
     VERSION_SCRIPT_FLAGS="-Wl,-M -Wl,"
+fi
 AC_SUBST(VERSION_SCRIPT_FLAGS)
 AM_CONDITIONAL([USE_VERSION_SCRIPT], [test -n "$VERSION_SCRIPT_FLAGS"])
 
