diff -ru e2fsprogs-libs-1.41.7.orig/lib/uuid/gen_uuid.c e2fsprogs-libs-1.41.7/lib/uuid/gen_uuid.c
--- e2fsprogs-libs-1.41.7.orig/lib/uuid/gen_uuid.c	2009-08-26 02:17:02 -0700
+++ e2fsprogs-libs-1.41.7/lib/uuid/gen_uuid.c	2009-08-26 02:18:34 -0700
@@ -219,7 +219,7 @@
  */
 static int get_node_id(unsigned char *node_id)
 {
-#ifdef HAVE_NET_IF_H
+#if defined(HAVE_NET_IF_H) && !defined(__INTERIX)
 	int 		sd;
 	struct ifreq 	ifr, *ifrp;
 	struct ifconf 	ifc;
