--- openmotif-2.3.1-orig/lib/Xm/TextF.c	2008-04-22 18:34:22.000000000 +0200
+++ openmotif-2.3.1/lib/Xm/TextF.c	2008-07-25 23:31:47.000000000 +0200
@@ -1959,6 +1959,7 @@
   
   if (change_stipple) {
     valueMask |= GCFillStyle;
+    values.fill_style = FillStippled;
     if (stipple) {
 #ifdef FIX_1381
 		  /*added for gray insensitive foreground (instead stipple)*/
--- openmotif-2.3.1-orig/lib/Xm/TextOut.c	2008-02-11 17:19:52.000000000 +0100
+++ openmotif-2.3.1/lib/Xm/TextOut.c	2008-07-25 23:31:47.000000000 +0200
@@ -623,6 +623,7 @@
   values.background = tw->core.background_pixel;
   if (change_stipple) {
     valueMask |= GCFillStyle;
+    values.fill_style = FillStippled;
     if (stipple) {
 #ifdef FIX_1381
 		/*added for gray insensitive foreground (instead stipple)*/
