--- openmotif-2.3.1-orig/lib/Xm/XmRenderT.c	2008-04-23 16:58:27.000000000 +0200
+++ openmotif-2.3.1/lib/Xm/XmRenderT.c	2008-06-28 10:05:58.000000000 +0200
@@ -1979,6 +1979,7 @@
 }
 
 #ifdef FIX_1414
+#ifdef USE_XFT
 static int
 GetSameRenditions(XmRendition *rend_cache, XmRendition rend, int count_rend)
 {
@@ -2009,6 +2010,7 @@
 
 }
 #endif
+#endif
 
 /* Make sure all the font related resources make sense together and */
 /* then load the font specified by fontName if necessary. */
