diff --exclude='*~' --exclude='.*' -I '$Id:' -urN most-4.10.2.orig/src/keym.c most-4.10.2/src/keym.c
--- most-4.10.2.orig/src/keym.c	2005-08-13 23:11:23.000000000 -0400
+++ most-4.10.2/src/keym.c	2005-08-13 23:37:11.000000000 -0400
@@ -199,7 +199,7 @@
 {
    Most_Mini_Buf[0] = 0;
 
-   if (!most_read_from_minibuffer(prompt, (char *) Most_Mini_Buf, MOST_MINI_BUF_LEN))
+   if (most_read_from_minibuffer(prompt, (char *) Most_Mini_Buf, MOST_MINI_BUF_LEN))
      return -1;
 
    if (1 != sscanf((char *) Most_Mini_Buf, "%d", n))
