diff -NurdB mlview-0.8/configure mlview-0.8-patched/configure
--- mlview-0.8/configure	2005-07-03 12:20:48.000000000 -0500
+++ mlview-0.8-patched/configure	2005-08-07 05:34:49.000000000 -0500
@@ -2218,7 +2218,7 @@
 \* 10000 + $MLVIEW_MINOR_VERSION \
 \* 100 + $MLVIEW_MICRO_VERSION`
 
-MLVIEW_EXE=$prefix/bin/mlv
+MLVIEW_EXE=$prefix/bin/mlview
 
 
 
diff -NurdB mlview-0.8/mlview.desktop.in mlview-0.8-patched/mlview.desktop.in
--- mlview-0.8/mlview.desktop.in	2005-06-25 07:57:05.000000000 -0500
+++ mlview-0.8-patched/mlview.desktop.in	2005-08-07 05:35:08.000000000 -0500
@@ -6,8 +6,8 @@
 GenericName=MlView Generic XML Editor
 Categories=Application;Development;
 Comment=A generic simple XML editor for GNOME
-Icon=mlview-app-icon.xpm
-TryExec=mlv
+Icon=mlview-app-icon.png
+TryExec=mlview
 Exec=@MLVIEW_EXE@ %F
 Terminal=false
 
