--- pastemecli.c~	2004-10-15 03:40:15 +0200
+++ pastemecli.c	2007-10-08 23:30:41 +0200
@@ -295,7 +295,7 @@
 	/*
 	 * Populate config from pastemecli.conf in multiple places
 	 */
-	file2config("/etc/pastemecli.conf", &config);
+	file2config("@GENTOO_PORTAGE_EPREFIX@/etc/pastemecli.conf", &config);
 	file2config("~/.pastemecli.conf", &config);
 	file2config("~/.pastemecli", &config);
 	file2config("pastemecli.conf", &config);
