diff -ru ORBit2-2.14.16.orig/acinclude.m4 ORBit2-2.14.16/acinclude.m4
--- ORBit2-2.14.16.orig/acinclude.m4	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/acinclude.m4	2008-12-04 09:41:41 +0100
@@ -25,7 +25,7 @@
 			struct test {char s1; $1 s2;};
 			main()
 			{
-			FILE *f=fopen("conftestval", "w");
+			FILE *f=fopen("conftestval", "wb");
 			if (!f) exit(1);
 			fprintf(f, "%d\n", &(((struct test*)0)->s2));
 			exit(0);
diff -ru ORBit2-2.14.16.orig/configure.in ORBit2-2.14.16/configure.in
--- ORBit2-2.14.16.orig/configure.in	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/configure.in	2008-12-04 09:41:41 +0100
@@ -53,6 +53,11 @@
     LIBM=
     MINGW_LDFLAGS="-Wl,--enable-runtime-pseudo-reloc"
     ;;
+  *-winnt*)
+  	os_win32=yes
+	LIBM=
+	MINGW_LDFLAGS=
+   ;;
   *)
     os_win32=no
     LIBM=-lm
diff -ru ORBit2-2.14.16.orig/include/orbit/GIOP/giop-types.h ORBit2-2.14.16/include/orbit/GIOP/giop-types.h
--- ORBit2-2.14.16.orig/include/orbit/GIOP/giop-types.h	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/include/orbit/GIOP/giop-types.h	2008-12-04 09:41:41 +0100
@@ -41,7 +41,7 @@
 	GIOP_CONNECTION_SSL
 } GIOPConnectionOptions;
 
-extern const char giop_version_ids[GIOP_NUM_VERSIONS][2];
+extern ORBIT2_IMPORT const char giop_version_ids[GIOP_NUM_VERSIONS][2];
 
 typedef struct {
 	CORBA_char magic[4];
diff -ru ORBit2-2.14.16.orig/include/orbit/dynamic/Makefile.am ORBit2-2.14.16/include/orbit/dynamic/Makefile.am
--- ORBit2-2.14.16.orig/include/orbit/dynamic/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/include/orbit/dynamic/Makefile.am	2008-12-04 09:41:41 +0100
@@ -11,7 +11,7 @@
 	-I$(top_srcdir)/src/idl/misc				\
 	--define=Object=OObject --define=TypeCode=TTypeCode	\
 	--noskels --nodefskels --nostubs --nocommon --noidata	\
-        --showcpperrors 
+        --showcpperrors --dll-define=ORBIT2
 IDL_DIR=$(top_srcdir)/src/orb/dynamic/
 IDL_FILES=dynamic-defs.idl
 
diff -ru ORBit2-2.14.16.orig/include/orbit/orb-core/Makefile.am ORBit2-2.14.16/include/orbit/orb-core/Makefile.am
--- ORBit2-2.14.16.orig/include/orbit/orb-core/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/include/orbit/orb-core/Makefile.am	2008-12-04 09:41:41 +0100
@@ -35,7 +35,7 @@
 	-I$(top_srcdir)/src/orb/orb-core		\
 	--noskels --nodefskels --nostubs --noidata	\
 	--nocommon					\
-        --showcpperrors
+        --showcpperrors --dll-define=ORBIT2
 IDL_FLAGS = $(IDL_FLAGS_NO_DEFS)		\
 	 --define=Object=OObject		\
 	--define=TypeCode=TTypeCode
diff -ru ORBit2-2.14.16.orig/include/orbit/orb-core/corba-object.h ORBit2-2.14.16/include/orbit/orb-core/corba-object.h
--- ORBit2-2.14.16.orig/include/orbit/orb-core/corba-object.h	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/include/orbit/orb-core/corba-object.h	2008-12-04 09:41:41 +0100
@@ -51,8 +51,8 @@
 #define CORBA_OBJECT_SMALL_GET_TYPE_ID    12
 #define CORBA_OBJECT_SMALL_GET_IINTERFACE 13
 
-extern ORBit_IInterface CORBA_Object__iinterface;
-extern ORBit_IMethod    CORBA_Object__imethods[];
+extern ORBIT2_IMPORT ORBit_IInterface CORBA_Object__iinterface;
+extern ORBIT2_IMPORT ORBit_IMethod    CORBA_Object__imethods[];
 
 #define CORBA_Object_IMETHODS_LEN 12
 
diff -ru ORBit2-2.14.16.orig/include/orbit/orb-core/corba-orb-type.h ORBit2-2.14.16/include/orbit/orb-core/corba-orb-type.h
--- ORBit2-2.14.16.orig/include/orbit/orb-core/corba-orb-type.h	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/include/orbit/orb-core/corba-orb-type.h	2008-12-04 09:41:41 +0100
@@ -28,7 +28,7 @@
 #define TC_IMPL_TC_CORBA_ORB_ObjectId_7 'e'
 #define TC_IMPL_TC_CORBA_ORB_ObjectId_8 'f'
 #define TC_IMPL_TC_CORBA_ORB_ObjectId_9 's'
-   extern ORBIT2_MAYBE_CONST struct CORBA_TypeCode_struct TC_CORBA_ORB_ObjectId_struct;
+   extern ORBIT2_IMPORT ORBIT2_MAYBE_CONST struct CORBA_TypeCode_struct TC_CORBA_ORB_ObjectId_struct;
 #define TC_CORBA_ORB_ObjectId ((CORBA_TypeCode)&TC_CORBA_ORB_ObjectId_struct)
 #endif
 #define CORBA_ORB_ObjectId__freekids CORBA_string__freekids
@@ -38,7 +38,7 @@
 #define _CORBA_InterfaceDef_defined 1
 #define CORBA_InterfaceDef__freekids CORBA_Object__freekids
    typedef CORBA_Object CORBA_InterfaceDef;
-   extern CORBA_unsigned_long CORBA_InterfaceDef__classid;
+   extern ORBIT2_IMPORT CORBA_unsigned_long CORBA_InterfaceDef__classid;
 #if !defined(TC_IMPL_TC_CORBA_InterfaceDef_0)
 #define TC_IMPL_TC_CORBA_InterfaceDef_0 'c'
 #define TC_IMPL_TC_CORBA_InterfaceDef_1 'o'
@@ -50,7 +50,7 @@
 #define TC_IMPL_TC_CORBA_InterfaceDef_7 'e'
 #define TC_IMPL_TC_CORBA_InterfaceDef_8 'f'
 #define TC_IMPL_TC_CORBA_InterfaceDef_9 's'
-   extern ORBIT2_MAYBE_CONST struct CORBA_TypeCode_struct TC_CORBA_InterfaceDef_struct;
+   extern ORBIT2_IMPORT ORBIT2_MAYBE_CONST struct CORBA_TypeCode_struct TC_CORBA_InterfaceDef_struct;
 #define TC_CORBA_InterfaceDef ((CORBA_TypeCode)&TC_CORBA_InterfaceDef_struct)
 #endif
 #endif
diff -ru ORBit2-2.14.16.orig/include/orbit/orb-core/corba-typecode.h ORBit2-2.14.16/include/orbit/orb-core/corba-typecode.h
--- ORBit2-2.14.16.orig/include/orbit/orb-core/corba-typecode.h	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/include/orbit/orb-core/corba-typecode.h	2008-12-04 09:41:41 +0100
@@ -30,6 +30,10 @@
 	CORBA_short scale;	            /* for fixed */
 };
 
+#ifdef _WIN32
+# define extern extern ORBIT2_IMPORT
+#endif
+
 extern ORBIT2_MAYBE_CONST ORBit_RootObject_Interface ORBit_TypeCode_epv;
 
 #define TC_null ((CORBA_TypeCode)&TC_null_struct)
@@ -76,6 +80,10 @@
 extern ORBIT2_MAYBE_CONST struct CORBA_TypeCode_struct TC_CORBA_long_long_struct;
 extern ORBIT2_MAYBE_CONST struct CORBA_TypeCode_struct TC_CORBA_unsigned_long_long_struct;
 
+#if defined(_WIN32) && defined(extern)
+# undef extern
+#endif
+
 #ifdef ORBIT2_INTERNAL_API
 
 #include <orbit/GIOP/giop-basics.h>
diff -ru ORBit2-2.14.16.orig/include/orbit/orb-core/orbit-object.h ORBit2-2.14.16/include/orbit/orb-core/orbit-object.h
--- ORBit2-2.14.16.orig/include/orbit/orb-core/orbit-object.h	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/include/orbit/orb-core/orbit-object.h	2008-12-04 09:41:41 +0100
@@ -63,7 +63,7 @@
 void     ORBit_RootObject_release     (gpointer obj);
 void     ORBit_RootObject_release_T   (gpointer obj);
 
-extern GMutex *ORBit_RootObject_lifecycle_lock;
+extern ORBIT2_IMPORT GMutex *ORBit_RootObject_lifecycle_lock;
 
 #endif /* ORBIT2_INTERNAL_API */
 
diff -ru ORBit2-2.14.16.orig/include/orbit/orb-core/orbit-small.h ORBit2-2.14.16/include/orbit/orb-core/orbit-small.h
--- ORBit2-2.14.16.orig/include/orbit/orb-core/orbit-small.h	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/include/orbit/orb-core/orbit-small.h	2008-12-04 09:41:41 +0100
@@ -70,7 +70,7 @@
 #define ORBIT_SMALL_FAST_LOCALS           1
 #define ORBIT_SMALL_FORCE_GENERIC_MARSHAL 2
 
-extern int     ORBit_small_flags;
+extern ORBIT2_IMPORT int     ORBit_small_flags;
 
 /* Deprecated - only for bin-compat with pre 2.4 stubs */
 void           ORBit_small_invoke_stub (CORBA_Object        object,
diff -ru ORBit2-2.14.16.orig/include/orbit/orbit-config.h.in ORBit2-2.14.16/include/orbit/orbit-config.h.in
--- ORBit2-2.14.16.orig/include/orbit/orbit-config.h.in	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/include/orbit/orbit-config.h.in	2008-12-04 09:41:41 +0100
@@ -27,4 +27,10 @@
 #define ORBIT_ALIGNOF_CORBA_STRUCT       @ORBIT_ALIGNOF_CORBA_STRUCT@
 #define ORBIT_ALIGNOF_CORBA_POINTER      @ORBIT_ALIGNOF_CORBA_POINTER@
 
+#if defined(_WIN32) && !defined(ORBIT2_COMPILATION)
+# define ORBIT2_IMPORT __declspec(dllimport)
+#else
+# define ORBIT2_IMPORT
+#endif
+
 #endif
diff -ru ORBit2-2.14.16.orig/include/orbit/orbit.h ORBit2-2.14.16/include/orbit/orbit.h
--- ORBit2-2.14.16.orig/include/orbit/orbit.h	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/include/orbit/orbit.h	2008-12-04 09:41:41 +0100
@@ -20,10 +20,10 @@
 
 G_BEGIN_DECLS
 
-extern const char  *orbit_version;
-extern unsigned int orbit_major_version;
-extern unsigned int orbit_minor_version;
-extern unsigned int orbit_micro_version;
+extern ORBIT2_IMPORT const char  *orbit_version;
+extern ORBIT2_IMPORT unsigned int orbit_major_version;
+extern ORBIT2_IMPORT unsigned int orbit_minor_version;
+extern ORBIT2_IMPORT unsigned int orbit_micro_version;
 
 G_END_DECLS
 
diff -ru ORBit2-2.14.16.orig/include/orbit/poa/Makefile.am ORBit2-2.14.16/include/orbit/poa/Makefile.am
--- ORBit2-2.14.16.orig/include/orbit/poa/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/include/orbit/poa/Makefile.am	2008-12-04 09:41:41 +0100
@@ -18,7 +18,7 @@
 	-I$(top_srcdir)/src/idl/misc					\
 	--define=Object=OObject --define=TypeCode=TTypeCode		\
 	--noskels --nodefskels --nostubs --nocommon --noidata		\
-        --showcpperrors
+        --showcpperrors --dll-define=ORBIT2
 IDL_FILES=poa-defs.idl
 IDL_DIR=$(top_srcdir)/src/orb/poa/
 include $(top_srcdir)/Makefile.shared
diff -ru ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-c-common.c ORBit2-2.14.16/src/idl-compiler/orbit-idl-c-common.c
--- ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-c-common.c	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/idl-compiler/orbit-idl-c-common.c	2008-12-04 09:41:41 +0100
@@ -631,6 +631,7 @@
 	fprintf (ci->fh, "#define ORBIT_IDL_C_COMMON\n");
 	fprintf (ci->fh, "#define %s_COMMON\n", ci->c_base_name);
 	fprintf (ci->fh, "#include \"%s.h\"\n\n", ci->base_name);
+	fprintf (ci->fh, "#ifdef __cplusplus\nextern \"C\" {\n#endif\n");
 	fprintf (ci->fh, "static const CORBA_unsigned_long ORBit_zero_int = 0;\n");
 
 	/* FIXME: this is slightly nasty, but we need these in common,
@@ -654,4 +655,6 @@
 		list = cc_build_interfaces (list, tree);
 		cc_output_itypes (list, ci);
 	}
+
+	fprintf (ci->fh, "#ifdef __cplusplus\n} /* extern \"C\" */\n#endif\n");
 }
diff -ru ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-c-headers.c ORBit2-2.14.16/src/idl-compiler/orbit-idl-c-headers.c
--- ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-c-headers.c	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/idl-compiler/orbit-idl-c-headers.c	2008-12-04 09:41:41 +0100
@@ -7,7 +7,7 @@
 /* ch = C header */
 static void ch_output_types(IDL_tree tree, OIDL_Run_Info *rinfo, OIDL_C_Info *ci);
 static void ch_output_poa(IDL_tree tree, OIDL_Run_Info *rinfo, OIDL_C_Info *ci);
-static void ch_output_itypes (IDL_tree tree, OIDL_C_Info *ci);
+static void ch_output_itypes (IDL_tree tree, OIDL_Run_Info* rinfo, OIDL_C_Info *ci);
 static void ch_output_stub_protos(IDL_tree tree, OIDL_Run_Info *rinfo, OIDL_C_Info *ci);
 static void ch_output_skel_protos(IDL_tree tree, OIDL_Run_Info *rinfo, OIDL_C_Info *ci);
 static GSList *ch_build_interfaces (GSList *list, IDL_tree tree);
@@ -153,6 +153,12 @@
   fprintf(ci->fh, "extern \"C\" {\n");
   fprintf(ci->fh, "#endif /* __cplusplus */\n\n");
 
+  fprintf(ci->fh, "#if defined(_WIN32) && !defined(%s_COMPILATION)\n", rinfo->dll_name);
+  fprintf(ci->fh, "# define %s_IMPORT __declspec(dllimport)\n", rinfo->dll_name);
+  fprintf(ci->fh, "#else\n");
+  fprintf(ci->fh, "# define %s_IMPORT\n", rinfo->dll_name);
+  fprintf(ci->fh, "#endif\n");
+
   /* Do all the typedefs, etc. */
   fprintf(ci->fh, "\n/** typedefs **/\n");
   ch_output_types(tree, rinfo, ci);
@@ -175,7 +181,7 @@
     /* FIXME: hackish ? */
     fprintf(ci->fh, "#include <orbit/orb-core/orbit-interface.h>\n\n");
 
-    ch_output_itypes(tree, ci);
+    ch_output_itypes(tree, rinfo, ci);
   }
 
   if (rinfo->idata) {
@@ -362,7 +368,7 @@
     } else {
     	fprintf(ci->fh, "#define %s__freekids CORBA_Object__freekids\n", fullname);
     	fprintf(ci->fh, "typedef CORBA_Object %s;\n", fullname);
-    	fprintf(ci->fh, "extern CORBA_unsigned_long %s__classid;\n", fullname);
+    	fprintf(ci->fh, "extern %s_IMPORT CORBA_unsigned_long %s__classid;\n", rinfo->dll_name, fullname);
 	ch_type_alloc_and_tc(tree, rinfo, ci, FALSE);
     }
 
@@ -770,9 +776,9 @@
   }
 
   fprintf (ci->fh, "#ifdef ORBIT_IDL_C_IMODULE_%s\n", ci->c_base_name);
-  fprintf (ci->fh, "static\n");
+  fprintf (ci->fh, "#ifdef _WIN32\nextern\n#else\nstatic\n#endif\n");
   fprintf (ci->fh, "#else\n");
-  fprintf (ci->fh, "extern\n");
+  fprintf (ci->fh, "extern %s_IMPORT\n", rinfo->dll_name);
   fprintf (ci->fh, "#endif\n");
   fprintf (ci->fh, "ORBIT2_MAYBE_CONST struct CORBA_TypeCode_struct TC_%s_struct;\n", ctmp);
 
@@ -1130,7 +1136,7 @@
 }
 
 static void
-ch_output_itypes (IDL_tree tree, OIDL_C_Info *ci)
+ch_output_itypes (IDL_tree tree, OIDL_Run_Info* rinfo, OIDL_C_Info *ci)
 {
 	static int num_methods = 0;
 
@@ -1139,12 +1145,12 @@
 
 	switch(IDL_NODE_TYPE(tree)) {
 	case IDLN_MODULE:
-		ch_output_itypes (IDL_MODULE(tree).definition_list, ci);
+		ch_output_itypes (IDL_MODULE(tree).definition_list, rinfo, ci);
 		break;
 	case IDLN_LIST: {
 		IDL_tree sub;
 		for (sub = tree; sub; sub = IDL_LIST(sub).next)
-			ch_output_itypes (IDL_LIST(sub).data, ci);
+			ch_output_itypes (IDL_LIST(sub).data, rinfo, ci);
 	}
 	break;
 	case IDLN_ATTR_DCL: {
@@ -1156,9 +1162,9 @@
 		    curitem = IDL_LIST(curitem).next) {
 			ai = IDL_LIST(curitem).data->data;
 	
-			ch_output_itypes (ai->op1, ci);
+			ch_output_itypes (ai->op1, rinfo, ci);
 			if(ai->op2)
-				ch_output_itypes (ai->op2, ci);
+				ch_output_itypes (ai->op2, rinfo, ci);
 		}
 	}
 	break;
@@ -1169,13 +1175,13 @@
 		id = IDL_ns_ident_to_qstring (IDL_IDENT_TO_NS (
 			IDL_INTERFACE (tree).ident), "_", 0);
 
-		ch_output_itypes (IDL_INTERFACE(tree).body, ci);
+		ch_output_itypes (IDL_INTERFACE(tree).body, rinfo, ci);
       
 		fprintf (ci->fh, "#ifdef ORBIT_IDL_C_IMODULE_%s\n",
 			 ci->c_base_name);
-		fprintf (ci->fh, "static \n");
+        fprintf (ci->fh, "#ifdef _WIN32\nextern\n#else\nstatic\n#endif\n");
 		fprintf (ci->fh, "#else\n");
-		fprintf (ci->fh, "extern \n");
+		fprintf (ci->fh, "extern %s_IMPORT\n", rinfo->dll_name);
 		fprintf (ci->fh, "#endif\n");
 		fprintf (ci->fh, "ORBit_IInterface %s__iinterface;\n", id);
 
@@ -1186,9 +1192,9 @@
 		else {
 			fprintf (ci->fh, "#ifdef ORBIT_IDL_C_IMODULE_%s\n",
 				 ci->c_base_name);
-			fprintf (ci->fh, "static \n");
+            fprintf (ci->fh, "#ifdef _WIN32\nextern\n#else\nstatic\n#endif\n");
 			fprintf (ci->fh, "#else\n");
-			fprintf (ci->fh, "extern \n");
+			fprintf (ci->fh, "extern %s_IMPORT\n", rinfo->dll_name);
 			fprintf (ci->fh, "#endif\n");
 			fprintf (ci->fh, "ORBit_IMethod %s__imethods[%s_IMETHODS_LEN];\n", id, id);
 		}
diff -ru ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-c-imodule.c ORBit2-2.14.16/src/idl-compiler/orbit-idl-c-imodule.c
--- ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-c-imodule.c	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/idl-compiler/orbit-idl-c-imodule.c	2008-12-04 09:41:41 +0100
@@ -209,6 +209,7 @@
 	fprintf (ci->fh, "#include \"%s-common.c\"\n\n", ci->base_name);
 
 	fprintf (ci->fh, "#include <orbit/orb-core/orbit-small.h>\n\n");
+	fprintf (ci->fh, "#ifdef __cplusplus\nextern \"C\" {\n#endif\n");
 
 	fprintf (ci->fh, "static CORBA_TypeCode %s__itypes[] = {\n",
 		 ci->c_base_name);
@@ -231,4 +232,5 @@
 	fprintf (ci->fh, " { %u, %u, %s__itypes, FALSE }\n",
 		 count, count, ci->c_base_name);
 	fprintf (ci->fh, "};\n\n");
+	fprintf (ci->fh, "#ifdef __cplusplus\n} /* extern \"C\" */\n#endif\n");
 }
diff -ru ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-c-skels.c ORBit2-2.14.16/src/idl-compiler/orbit-idl-c-skels.c
--- ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-c-skels.c	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/idl-compiler/orbit-idl-c-skels.c	2008-12-04 09:41:41 +0100
@@ -289,6 +289,9 @@
 	fprintf (ci->fh, "#include <string.h>\n");
 	fprintf (ci->fh, "#define ORBIT2_STUBS_API\n");
 	fprintf (ci->fh, "#include \"%s.h\"\n\n", ci->base_name);
+	fprintf (ci->fh, "#ifdef __cplusplus\nextern \"C\" {\n#endif\n");
 
 	ck_output_poastuff (tree, rinfo, ci);
+
+	fprintf (ci->fh, "#ifdef __cplusplus\n} /* extern \"C\" */\n#endif\n");
 }
diff -ru ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-c-stubs.c ORBit2-2.14.16/src/idl-compiler/orbit-idl-c-stubs.c
--- ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-c-stubs.c	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/idl-compiler/orbit-idl-c-stubs.c	2008-12-04 09:41:41 +0100
@@ -164,6 +164,9 @@
 	fprintf (ci->fh, "#include <string.h>\n");
 	fprintf (ci->fh, "#define ORBIT2_STUBS_API\n");
 	fprintf (ci->fh, "#include \"%s.h\"\n\n", ci->base_name);
+	fprintf (ci->fh, "#ifdef __cplusplus\nextern \"C\" {\n#endif\n");
 
 	cs_output_stubs (tree, ci, NULL);
+
+	fprintf (ci->fh, "#ifdef __cplusplus\n} /* extern \"C\" */\n#endif\n");
 }
diff -ru ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-main.c ORBit2-2.14.16/src/idl-compiler/orbit-idl-main.c
--- ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl-main.c	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/idl-compiler/orbit-idl-main.c	2008-12-04 09:41:41 +0100
@@ -63,6 +63,7 @@
 static gboolean cl_onlytop = FALSE;
 static char *cl_deps_file = NULL;
 static char *cl_output_directory = "";
+static char *cl_dll_name = "GENERIC_DLL";
 static char **idl_files = NULL;
 
 #define BASE_CPP_ARGS "-D__ORBIT_IDL__ "
@@ -156,6 +157,7 @@
   { "deps", 0, 0,  G_OPTION_ARG_FILENAME, &cl_deps_file, N_("Generate dependency info suitable for inclusion in Makefile"), N_("FILENAME") },
   { "headerguardprefix", 0, 0, G_OPTION_ARG_STRING, &cl_header_guard_prefix, N_("Prefix for #ifdef header guards. Sometimes useful to avoid conflicts."), N_("PREFIX") },
   { "output-dir", 0, 0, G_OPTION_ARG_FILENAME, &cl_output_directory, N_("Where to put generated files. This directory must exist."), N_("DIR") },
+  { "dll-define", 0, 0, G_OPTION_ARG_STRING, &cl_dll_name, N_("The name of the define to build Win32 DLL's."), N_("DEFINE") },
   { G_OPTION_REMAINING, 0, 0, G_OPTION_ARG_FILENAME_ARRAY, &idl_files, NULL, N_("<IDL files>") },
   { NULL }
 };
@@ -261,13 +263,14 @@
   rinfo.output_language = cl_output_lang;
   rinfo.header_guard_prefix = cl_header_guard_prefix;
   rinfo.output_directory = cl_output_directory;
+  rinfo.dll_name = cl_dll_name;
   rinfo.backend_directory = cl_backend_dir;
   rinfo.onlytop = cl_onlytop;
   rinfo.idata = !cl_disable_idata;
   
   printf ("orbit-idl-2 " VERSION " compiling\n");
-  printf (" %s mode, %s preprocessor errors, passes: %s%s%s%s%s%s\n\n",
-	  rinfo.is_pidl ? "pidl" : "",
+  printf (" %s mode, dll %s, %s preprocessor errors, passes: %s%s%s%s%s%s\n\n",
+	  rinfo.is_pidl ? "pidl" : "", rinfo.dll_name,
 	  rinfo.show_cpp_errors ? "show" : "hide",
 	  cl_disable_stubs ? "" : "stubs ",
 	  cl_disable_skels ? "" : "skels ",
diff -ru ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl3-types.h ORBit2-2.14.16/src/idl-compiler/orbit-idl3-types.h
--- ORBit2-2.14.16.orig/src/idl-compiler/orbit-idl3-types.h	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/idl-compiler/orbit-idl3-types.h	2008-12-04 09:41:41 +0100
@@ -34,6 +34,7 @@
   char *deps_file;
   char *header_guard_prefix;
   char *output_directory;
+  char *dll_name;
   gboolean onlytop;
   gboolean idata;
 
diff -ru ORBit2-2.14.16.orig/src/orb/GIOP/Makefile.am ORBit2-2.14.16/src/orb/GIOP/Makefile.am
--- ORBit2-2.14.16.orig/src/orb/GIOP/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/orb/GIOP/Makefile.am	2008-12-04 09:41:41 +0100
@@ -5,7 +5,7 @@
 	$(WARN_CFLAGS)			\
 	$(DISABLE_DEPRECATED_CFLAGS)	\
 	$(LOCAL_LINC_CFLAGS)		\
-	$(ORBIT_CFLAGS)
+	$(ORBIT_CFLAGS) -DORBIT2_COMPILATION
 
 noinst_LTLIBRARIES=libGIOP.la
 
diff -ru ORBit2-2.14.16.orig/src/orb/Makefile.am ORBit2-2.14.16/src/orb/Makefile.am
--- ORBit2-2.14.16.orig/src/orb/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/orb/Makefile.am	2008-12-04 09:41:41 +0100
@@ -11,7 +11,7 @@
 	-DORBIT2_INTERNAL_API			\
 	$(WARN_CFLAGS)				\
 	$(LOCAL_LINC_CFLAGS)			\
-	$(ORBIT_CFLAGS)
+	$(ORBIT_CFLAGS) -DORBIT2_COMPILATION
 
 libORBit_2_la_SOURCES=orbit-init.c orbit-init.h
 
diff -ru ORBit2-2.14.16.orig/src/orb/dynamic/Makefile.am ORBit2-2.14.16/src/orb/dynamic/Makefile.am
--- ORBit2-2.14.16.orig/src/orb/dynamic/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/orb/dynamic/Makefile.am	2008-12-04 09:41:41 +0100
@@ -14,7 +14,7 @@
 	$(WARN_CFLAGS)					\
 	$(DISABLE_DEPRECATED_CFLAGS)			\
 	$(LOCAL_LINC_CFLAGS)				\
-	$(ORBIT_CFLAGS)
+	$(ORBIT_CFLAGS) -DORBIT2_COMPILATION
 
 IDL_FLAGS= -I$(top_srcdir)/src/idl/CORBA_PIDL			\
 	--define=Object=OObject --define=TypeCode=TTypeCode	\
@@ -22,7 +22,7 @@
 	-I$(top_srcdir)/src/idl/misc				\
 	-I$(top_srcdir)/src/orb/orb-core			\
 	--noskels --nodefskels --nostubs --noidata --noheaders  \
-	--showcpperrors
+	--showcpperrors --dll-define=ORBIT2
 IDL_FILES=dynamic-defs.idl
 include $(top_srcdir)/Makefile.shared
 
diff -ru ORBit2-2.14.16.orig/src/orb/orb-core/Makefile.am ORBit2-2.14.16/src/orb/orb-core/Makefile.am
--- ORBit2-2.14.16.orig/src/orb/orb-core/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/orb/orb-core/Makefile.am	2008-12-04 09:41:41 +0100
@@ -14,7 +14,7 @@
 	$(WARN_CFLAGS)					\
 	$(DISABLE_DEPRECATED_CFLAGS)			\
 	$(LOCAL_LINC_CFLAGS)				\
-	$(ORBIT_CFLAGS)
+	$(ORBIT_CFLAGS) -DORBIT2_COMPILATION
 
 HTTP_FILES = orbhttp.c orbhttp.h
 
@@ -68,12 +68,12 @@
 	-D_PRE_3_0_COMPILER_                                    \
 	--noskels --nodefskels --nostubs --noidata --noheaders	\
 	--define=Object=OObject --define=TypeCode=TTypeCode	\
-        --showcpperrors
+        --showcpperrors --dll-define=ORBIT2
 IDL_FILES=corba-defs.idl iop-defs.idl corba-ops.idl orbit-interface.idl
 IDL_DIR=$(top_srcdir)/src/orb/orb-core/
 include $(top_srcdir)/Makefile.shared
 
-IDL_FLAGS_CORBA_OPS = --showcpperrors
+IDL_FLAGS_CORBA_OPS = --showcpperrors --dll-define=ORBIT2
 $(OPS_IDLOUT_C): $(OPS_IDLOUT_H)
 
 $(OPS_IDLOUT_H): corba-ops.idl $(IDL_COMPILER)
@@ -85,7 +85,7 @@
 	sed -e "s,Z,_,g" corba-ops-skels.c > corba-ops-skels.c.out;				\
 	mv corba-ops-skels.c.out corba-ops-skels.c;
 
-IDL_FLAGS_INTERFACE = --nostubs --noskels --showcpperrors --noheaders
+IDL_FLAGS_INTERFACE = --nostubs --noskels --showcpperrors --noheaders --dll-define=ORBIT2
 $(IFACE_IDLOUT) : orbit-interface.idl $(IDL_COMPILER)
 	-(rm -f $(IFACE_IDLOUT) || true) > /dev/null
 	$(IDL_COMPILER) $(IDL_FLAGS_INTERFACE) --deps .deps/orbit-interface.idl.P $<
diff -ru ORBit2-2.14.16.orig/src/orb/poa/Makefile.am ORBit2-2.14.16/src/orb/poa/Makefile.am
--- ORBit2-2.14.16.orig/src/orb/poa/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/orb/poa/Makefile.am	2008-12-04 09:41:41 +0100
@@ -15,7 +15,7 @@
 	$(WARN_CFLAGS)					\
 	$(DISABLE_DEPRECATED_CFLAGS)			\
 	$(LOCAL_LINC_CFLAGS)				\
-	$(ORBIT_CFLAGS)
+	$(ORBIT_CFLAGS) -DORBIT2_COMPILATION
 
 POA_IDLOUT=poa-defs-common.c
 
@@ -41,7 +41,7 @@
 	-D_PRE_3_0_COMPILER_                                    \
 	--noskels --nodefskels --nostubs --noidata --noheaders  \
 	--define=Object=OObject --define=TypeCode=TTypeCode	\
-        --showcpperrors
+        --showcpperrors --dll-define=ORBIT2
 IDL_FILES=poa-defs.idl
 include $(top_srcdir)/Makefile.shared
 
diff -ru ORBit2-2.14.16.orig/src/orb/util/Makefile.am ORBit2-2.14.16/src/orb/util/Makefile.am
--- ORBit2-2.14.16.orig/src/orb/util/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/orb/util/Makefile.am	2008-12-04 09:41:41 +0100
@@ -8,7 +8,7 @@
 	$(WARN_CFLAGS)					\
 	$(DISABLE_DEPRECATED_CFLAGS)			\
 	$(LOCAL_LINC_CFLAGS)				\
-	$(ORBIT_CFLAGS)
+	$(ORBIT_CFLAGS) -DORBIT2_COMPILATION
 
 liborb_util_la_SOURCES= \
 	orbit-purify.h  \
diff -ru ORBit2-2.14.16.orig/src/services/name/Makefile.am ORBit2-2.14.16/src/services/name/Makefile.am
--- ORBit2-2.14.16.orig/src/services/name/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/services/name/Makefile.am	2008-12-04 09:41:41 +0100
@@ -1,7 +1,9 @@
 if OS_WIN32
 libm=
+cxxflag=-xc++
 else
 libm=-lm
+cxxflag=
 endif
 
 libORBitservicesincludedir = $(includedir)/orbit-2.0/ORBitservices
@@ -10,7 +12,7 @@
 
 noinst_PROGRAMS = name-client-2 orbit-name-server-2
 
-lib_LIBRARIES = libname-server-2.a
+lib_LIBRARIES = libname-server-2.a libname-server-2-skels.a
 
 libORBitCosNaming_2_la_SOURCES = 					\
 	CosNaming-common.c 						\
@@ -26,6 +28,8 @@
 	$(ORBIT_NAME_LIBS)						\
 	$(top_builddir)/src/orb/libORBit-2.la
 
+libORBitCosNaming_2_la_CFLAGS = -DORBIT2CosNaming_COMPILATION $(cxxflag)
+
 INCLUDES =				\
 	-I.				\
 	-I$(top_builddir)/include	\
@@ -42,7 +46,7 @@
 	CosNaming-stubs.c 						\
 	CosNaming-skels.c
 
-IDL_FLAGS=--showcpperrors
+IDL_FLAGS=--showcpperrors --dll-define=ORBIT2CosNaming
 IDL_FILES=CosNaming.idl
 include $(top_srcdir)/Makefile.shared
 
@@ -62,14 +66,18 @@
 name_client_2_DEPENDENCIES = $(DEPS) CosNaming.h
 name_client_2_LDADD = $(LDADDS)
 
-libname_server_2_a_SOURCES = orbit-name-server.c CosNaming-skels.c \
+libname_server_2_skels_a_SOURCES = CosNaming-skels.c
+libname_server_2_skels_a_DEPENDENCIES = $(DEPS) CosNaming.h
+libname_server_2_skels_a_CFLAGS = $(cxxflag)
+
+libname_server_2_a_SOURCES = orbit-name-server.c \
 			   name-support.c name-support.h
 libname_server_2_a_DEPENDENCIES = $(DEPS) CosNaming.h
 
 orbit_name_server_2_SOURCES = boot.c
 orbit_name_server_2_LDFLAGS = $(FLAGS)
 orbit_name_server_2_DEPENDENCIES = $(DEPS) CosNaming.h
-orbit_name_server_2_LDADD = libname-server-2.a $(LDADDS)
+orbit_name_server_2_LDADD = libname-server-2.a libname-server-2-skels.a $(LDADDS)
 
 $(libname_server_2_a_OBJECTS) $(name_client_2_OBJECTS) $(orbit_name_server_2_OBJECTS) : CosNaming.h
 
diff -ru ORBit2-2.14.16.orig/src/services/name/name-support.c ORBit2-2.14.16/src/services/name/name-support.c
--- ORBit2-2.14.16.orig/src/services/name/name-support.c	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/services/name/name-support.c	2008-12-04 09:41:41 +0100
@@ -1,5 +1,9 @@
 #include "name-support.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 CosNaming_Name*
 ORBit_string_to_CosNaming_Name (const CORBA_char *string,
 				CORBA_Environment * ev)
@@ -141,3 +145,7 @@
   return retval;
 }
 
+#ifdef __cplusplus
+} /* extern "C" */
+#endif
+
diff -ru ORBit2-2.14.16.orig/src/services/name/orbit-name-server.c ORBit2-2.14.16/src/services/name/orbit-name-server.c
--- ORBit2-2.14.16.orig/src/services/name/orbit-name-server.c	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/src/services/name/orbit-name-server.c	2008-12-04 09:41:41 +0100
@@ -28,6 +28,10 @@
 #include <signal.h>
 #include <sys/types.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /*** App-specific servant structures ***/
 
 typedef struct
@@ -826,3 +830,8 @@
 
 	return retval;
 }
+
+#ifdef __cplusplus
+} /* extern "C" */
+#endif
+
diff -ru ORBit2-2.14.16.orig/test/Makefile.am ORBit2-2.14.16/test/Makefile.am
--- ORBit2-2.14.16.orig/test/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/test/Makefile.am	2008-12-04 10:33:49 +0100
@@ -18,6 +18,8 @@
 
 check_PROGRAMS = test-dynany test-mem test-performance test-giop test-corbaloc
 
+noinst_LIBRARIES = libtest-any.a libtest-echo.a libtest-empty.a libtest-timeout.a libtest-test1.a libtest-dynany.a
+
 TESTS =  timeout.sh $(check_PROGRAMS)
 
 LDADD = $(top_builddir)/src/orb/libORBit-2.la $(ORBIT_LIBS)
@@ -34,40 +36,94 @@
 	test-giop.c		\
 	test-giop-frag.h
 
+if OS_WIN32
+cxxflag = -xc++
+else
+cxxflag =
+endif
+
 TEST_ANY_IDLOUT=test-any.h test-any-common.c test-any-stubs.c test-any-skels.c
-test_any_client_SOURCES=test-any-client.c $(TEST_ANY_IDLOUT)
-test_any_server_SOURCES=test-any-server.c $(TEST_ANY_IDLOUT)
+libtest_any_a_SOURCES=$(TEST_ANY_IDLOUT)
+libtest_any_a_CFLAGS=$(cxxflag)
+
+test_any_client_SOURCES=test-any-client.c
+test_any_client_DEPENDENCIES=libtest-any.a
+test_any_client_LDADD=libtest-any.a $(LDADD)
+
+test_any_server_SOURCES=test-any-server.c
+test_any_server_DEPENDENCIES=libtest-any.a
+test_any_server_LDADD=libtest-any.a $(LDADD)
 $(srcdir)/test-any-client.c $(srcdir)/test-any-server.c: test-any.h
 
 ECHO_IDLOUT=echo.h echo-common.c echo-stubs.c echo-skels.c
-echo_client_SOURCES=echo-client.c $(ECHO_IDLOUT)
-echo_client_t_SOURCES=echo-client-t.c $(ECHO_IDLOUT)
-echo_server_SOURCES=echo-server.c echo-srv.c $(ECHO_IDLOUT) echo-share.h$
+libtest_echo_a_SOURCES=$(ECHO_IDLOUT)
+libtest_echo_a_CFLAGS=$(cxxflag)
+
+echo_client_SOURCES=echo-client.c
+echo_client_DEPENDENCIES=libtest-echo.a
+echo_client_LDADD=libtest-echo.a $(LDADD)
+
+echo_client_t_SOURCES=echo-client-t.c
+echo_client_t_DEPENDENCIES=libtest-echo.a
+echo_client_t_LDADD=libtest-echo.a $(LDADD)
+
+echo_server_SOURCES=echo-server.c echo-srv.c echo-share.h$
+echo_server_DEPENDENCIES=libtest-echo.a
+echo_server_LDADD=libtest-echo.a $(LDADD)
+
 $(srcdir)/echo-client-t.c $(srcdir)/echo-client.c $(srcdir)/echo-server.c $(srcdir)/echo-srv.c: echo.h
 
 EMPTY_IDLOUT=empty.h empty-common.c empty-stubs.c empty-skels.c
-empty_client_SOURCES=empty-client.c $(EMPTY_IDLOUT)
-empty_server_SOURCES=empty-server.c $(EMPTY_IDLOUT)
+libtest_empty_a_SOURCES=$(EMPTY_IDLOUT)
+libtest_empty_a_CFLAGS=$(cxxflag)
+
+empty_client_SOURCES=empty-client.c
+empty_client_DEPENDENCIES=libtest-empty.a
+empty_client_LDADD=libtest-empty.a $(LDADD)
+
+empty_server_SOURCES=empty-server.c
+empty_server_DEPENDENCIES=libtest-empty.a
+empty_server_LDADD=libtest-empty.a $(LDADD)
 $(srcdir)/empty-client.c $(srcdir)/empty-server.c: empty.h
 
 TIMEOUT_IDLOUT=timeout.h timeout-common.c timeout-stubs.c timeout-skels.c
-timeout_client_SOURCES=timeout-client.c $(TIMEOUT_IDLOUT)
-timeout_server_SOURCES=timeout-server.c $(TIMEOUT_IDLOUT)
+libtest_timeout_a_SOURCES=$(TIMEOUT_IDLOUT)
+libtest_timeout_a_CFLAGS=$(cxxflag)
+
+timeout_client_SOURCES=timeout-client.c
+timeout_client_DEPENDENCIES=libtest-timeout.a
+timeout_client_LDADD=libtest-timeout.a $(LDADD)
+
+timeout_server_SOURCES=timeout-server.c
+timeout_server_DEPENDENCIES=libtest-timeout.a
+timeout_server_LDADD=libtest-timeout.a $(LDADD)
 $(srcdir)/timeout-client.c $(srcdir)/timeout-server.c: timeout.h
 
 IDLOUT=test1-stubs.c test1-skels.c test1-common.c test1.h
-test1_SOURCES=test1.c $(IDLOUT)
+libtest_test1_a_SOURCES=$(IDLOUT)
+libtest_test1_a_CFLAGS=$(cxxflag)
+
+test1_SOURCES=test1.c
+test1_DEPENDENCIES=libtest-test1.a
+test1_LDADD=libtest-test1.a $(LDADD)
 $(srcdir)/test1.c: test1.h
 
 test_mem_SOURCES=test-mem.c $(IDLOUT)
+test_mem_DEPENDENCIES=libtest-test1.a
+test_mem_LDADD=libtest-test1.a $(LDADD)
 
 ior_decode_2_SOURCES=ior-decode.c
 
 typelib_dump_SOURCES=typelib-dump.c
 
 DYNANY_IDLOUT=dynany.h dynany-common.c dynany-skels.c dynany-stubs.c
+libtest_dynany_a_SOURCES=$(DYNANY_IDLOUT)
+libtest_dynany_a_CFLAGS=$(cxxflag)
+
 test_dynany_SOURCES=$(DYNANY_IDLOUT) test-dynany.c
 test_dynany_LDFLAGS=$(LIBM)
+test_dynany_DEPENDENCIES=libtest-dynany.a
+test_dynany_LDADD=libtest-dynany.a $(LDADD)
 
 IDL_FLAGS =  --showcpperrors
 IDL_FILES=echo.idl empty.idl test-any.idl test1.idl dynany.idl timeout.idl
diff -ru ORBit2-2.14.16.orig/test/everything/Makefile.am ORBit2-2.14.16/test/everything/Makefile.am
--- ORBit2-2.14.16.orig/test/everything/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/test/everything/Makefile.am	2008-12-04 09:41:41 +0100
@@ -1,4 +1,11 @@
 check_PROGRAMS = client server
+noinst_LIBRARIES = libtest-everything-built.a
+
+if OS_WIN32
+cxxflag=-xc++
+else
+cxxflag=
+endif
 
 EVERYTHING_BUILT = \
 	everything.h        \
@@ -34,11 +41,15 @@
 
 #  -lefence
 
+libtest_everything_built_a_SOURCES = $(EVERYTHING_BUILT)
+libtest_everything_built_a_DEPENDENCIES = $(EVERYTHING_BUILT)
+libtest_everything_built_a_CFLAGS = $(cxxflag) -DTEST_EVERYTHING_COMPILATION
+
 client.c : $(included_src)
 	touch client.c
-client_SOURCES=${EVERYTHING_BUILT} client.c everything.idl constants.h
-client_DEPENDENCIES=${EVERYTHING_BUILT} $(included_src)
-client_LDADD = $(LDADD)
+client_SOURCES=client.c everything.idl constants.h
+client_DEPENDENCIES=$(included_src) libtest-everything-built.a
+client_LDADD = $(LDADD) libtest-everything-built.a
 client_LDFLAGS = -static -module
 
 included_src = \
@@ -65,13 +76,16 @@
         $(top_builddir)/src/orb/libORBit-2.la \
 	$(ORBIT_LIBS)
 
-server_SOURCES=server.c ${EVERYTHING_BUILT} everything.idl constants.h
+Everything_module_la_CFLAGS = $(cxxflag) -DTEST_EVERYTHING_COMPILATION
+
+server_SOURCES=server.c everything.idl constants.h
 server_LDFLAGS = -static -module
 server.c : $(included_src)
 	touch server.c
-server_DEPENDENCIES=${EVERYTHING_BUILT} $(included_src)
+server_DEPENDENCIES=$(included_src) libtest-everything-built.a
+server_LDADD = libtest-everything-built.a
 
-IDL_FLAGS= --showcpperrors --add-imodule
+IDL_FLAGS= --showcpperrors --add-imodule --dll-define=TEST_EVERYTHING
 IDL_FILES= everything.idl
 include $(top_srcdir)/Makefile.shared
 
diff -ru ORBit2-2.14.16.orig/test/inhibit/Makefile.am ORBit2-2.14.16/test/inhibit/Makefile.am
--- ORBit2-2.14.16.orig/test/inhibit/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/test/inhibit/Makefile.am	2008-12-04 09:41:41 +0100
@@ -1,6 +1,14 @@
 noinst_PROGRAMS=					\
 	test-inhibit
 
+noinst_LIBRARIES= libtest-inhibit-foo.a libtest-inhibit-baa.a
+
+if OS_WIN32
+cxxflag = -xc++
+else
+cxxflag =
+endif
+
 INCLUDES =                           \
 	-I$(top_srcdir)/include	     \
 	-I$(top_builddir)/include    \
@@ -13,9 +21,17 @@
 FOO_IDLOUT=foo.h foo-stubs.c foo-skels.c foo-common.c
 BAA_IDLOUT=baa.h baa-stubs.c baa-skels.c baa-common.c
 
-test_inhibit_SOURCES=$(FOO_IDLOUT) $(BAA_IDLOUT) test-inhibit.c
+libtest_inhibit_foo_a_SOURCES=$(FOO_IDLOUT)
+libtest_inhibit_foo_a_CFLAGS=$(cxxflag)
+
+libtest_inhibit_baa_a_SOURCES=$(BAA_IDLOUT)
+libtest_inhibit_baa_a_CFLAGS=$(cxxflag)
+
+test_inhibit_SOURCES=test-inhibit.c
+test_inhibit_DEPENDENCIES=libtest-inhibit-foo.a libtest-inhibit-baa.a
+test_inhibit_LDADD=libtest-inhibit-foo.a libtest-inhibit-baa.a $(LDADD)
 
-IDL_FLAGS=--showcpperrors -I$(srcdir)
+IDL_FLAGS=--showcpperrors -I$(srcdir) --dll-define=$(*)
 IDL_FILES=foo.idl baa.idl
 include $(top_srcdir)/Makefile.shared
 
diff -ru ORBit2-2.14.16.orig/test/poa/Makefile.am ORBit2-2.14.16/test/poa/Makefile.am
--- ORBit2-2.14.16.orig/test/poa/Makefile.am	2008-12-04 09:51:36 +0100
+++ ORBit2-2.14.16/test/poa/Makefile.am	2008-12-04 09:41:41 +0100
@@ -35,32 +35,77 @@
 
 check_PROGRAMS = $(TESTS)
 
+noinst_LIBRARIES = libtest-poa.a
+
 LDADD = $(top_builddir)/src/orb/libORBit-2.la $(ORBIT_LIBS)
 
 POATEST_IDLOUT = poatest.h poatest-common.c poatest-skels.c poatest-stubs.c
 
-IDL_FLAGS=--showcpperrors
+IDL_FLAGS=--showcpperrors --dll-define=POATEST
 IDL_FILES=poatest.idl
 include $(top_srcdir)/Makefile.shared
 
+if OS_WIN32
+cxxflag = -xc++
+else
+cxxflag =
+endif
+
+libtest_poa_a_SOURCES = $(POATEST_IDLOUT)
+libtest_poa_a_CFLAGS = $(cxxflag) -DPOATEST_COMPILATION
+
 common_srcs = 			\
-	$(POATEST_IDLOUT)	\
 	poatest-basic-shell.c	\
 	poatest-basic-shell.h	\
 	poatest-exception.h
 
 test_poa_SOURCES = test-poa.c
+test_poa_DEPENDENCIES = libtest-poa.a
+test_poa_LDADD = libtest-poa.a $(LDADD)
+
 poatest_basic01_SOURCES = $(common_srcs) poatest-basic01.c 
+poatest_basic01_DEPENDENCIES = libtest-poa.a
+poatest_basic01_LDADD = libtest-poa.a $(LDADD)
+
 poatest_basic02_SOURCES = $(common_srcs) poatest-basic02.c 
+poatest_basic02_DEPENDENCIES = libtest-poa.a
+poatest_basic02_LDADD = libtest-poa.a $(LDADD)
+
 poatest_basic03_SOURCES = $(common_srcs) poatest-basic03.c 
+poatest_basic03_DEPENDENCIES = libtest-poa.a
+poatest_basic03_LDADD = libtest-poa.a $(LDADD)
+
 poatest_basic04_SOURCES = $(common_srcs) poatest-basic04.c 
+poatest_basic04_DEPENDENCIES = libtest-poa.a
+poatest_basic04_LDADD = libtest-poa.a $(LDADD)
+
 poatest_basic05_SOURCES = $(common_srcs) poatest-basic05.c 
+poatest_basic05_DEPENDENCIES = libtest-poa.a
+poatest_basic05_LDADD = libtest-poa.a $(LDADD)
+
 poatest_basic06_SOURCES = $(common_srcs) poatest-basic06.c 
+poatest_basic06_DEPENDENCIES = libtest-poa.a
+poatest_basic06_LDADD = libtest-poa.a $(LDADD)
+
 poatest_basic07_SOURCES = $(common_srcs) poatest-basic07.c 
+poatest_basic07_DEPENDENCIES = libtest-poa.a
+poatest_basic07_LDADD = libtest-poa.a $(LDADD)
+
 poatest_basic08_SOURCES = $(common_srcs) poatest-basic08.c 
+poatest_basic08_DEPENDENCIES = libtest-poa.a
+poatest_basic08_LDADD = libtest-poa.a $(LDADD)
+
 poatest_basic09_SOURCES = $(common_srcs) poatest-basic09.c 
+poatest_basic09_DEPENDENCIES = libtest-poa.a
+poatest_basic09_LDADD = libtest-poa.a $(LDADD)
+
 poatest_basic10_SOURCES = $(common_srcs) poatest-basic10.c 
+poatest_basic10_DEPENDENCIES = libtest-poa.a
+poatest_basic10_LDADD = libtest-poa.a $(LDADD)
+
 poatest_basic11_SOURCES = $(common_srcs) poatest-basic11.c 
+poatest_basic11_DEPENDENCIES = libtest-poa.a
+poatest_basic11_LDADD = libtest-poa.a $(LDADD)
 
 README : $(POA_TESTS_SRCS)
 	awk -F'*' 'BEGIN { intest = 0; } \
