--- cache.h	2008-12-22 14:41:50.000000000 +0000
+++ cache.h	2008-12-22 14:42:32.000000000 +0000
@@ -42,7 +42,7 @@
  * happens that everybody shares the same bit representation
  * in the UNIX world (and apparently wider too..)
  */
-#define S_IFGITLINK	0160000
+#define S_IFGITLINK	0140000
 #define S_ISGITLINK(m)	(((m) & S_IFMT) == S_IFGITLINK)
 
 /*
--- git-compat-util.h	2008-12-22 18:35:58.000000000 +0000
+++ git-compat-util.h	2008-12-22 18:39:21.000000000 +0000
@@ -82,6 +82,9 @@
 #include <arpa/inet.h>
 #include <netdb.h>
 #include <pwd.h>
+#ifdef __MINT__
+#include "compat/mint.h"
+#endif
 #ifndef __INTERIX
 #include <inttypes.h>
 #else
--- /dev/null	2008-12-07 20:01:52.000000000 +0000
+++ compat/mint.h	2008-12-23 11:52:10.000000000 +0000
@@ -0,0 +1,2 @@
+#define SA_RESTART 0
+#define sockaddr_storage sockaddr
