diff -ru -x '*.Po' -x '*.Plo' libXfont-1.3.4.orig/src/FreeType/ftfuncs.c libXfont-1.3.4/src/FreeType/ftfuncs.c
--- libXfont-1.3.4.orig/src/FreeType/ftfuncs.c	2009-01-19 10:34:16 +0100
+++ libXfont-1.3.4/src/FreeType/ftfuncs.c	2009-01-19 10:48:18 +0100
@@ -32,8 +32,6 @@
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
-#include <X11/fonts/fontmisc.h>
-
 #ifndef FONTMODULE
 #include <string.h>
 #include <math.h>
@@ -69,6 +67,7 @@
 #include "ft.h"
 #include "ftfuncs.h"
 #include "xttcap.h"
+#include <X11/fonts/fontmisc.h>
 
 /* Work around FreeType bug */
 #define WORK_AROUND_UPM 2048
diff -ru -x '*.Po' -x '*.Plo' libXfont-1.3.4.orig/src/bitmap/pcfread.c libXfont-1.3.4/src/bitmap/pcfread.c
--- libXfont-1.3.4.orig/src/bitmap/pcfread.c	2009-01-19 10:34:15 +0100
+++ libXfont-1.3.4/src/bitmap/pcfread.c	2009-01-19 10:38:45 +0100
@@ -45,7 +45,13 @@
 #endif
 
 #include <stdarg.h>
+
+#ifndef _WIN32
 #include <stdint.h>
+#else
+#include <limits.h>
+#define INT32_MAX INT_MAX
+#endif
 
 void
 pcfError(const char* message, ...)
diff -ru -x '*.Po' -x '*.Plo' libXfont-1.3.4.orig/src/fontfile/catalogue.c libXfont-1.3.4/src/fontfile/catalogue.c
--- libXfont-1.3.4.orig/src/fontfile/catalogue.c	2009-01-19 10:34:16 +0100
+++ libXfont-1.3.4/src/fontfile/catalogue.c	2009-01-19 10:47:28 +0100
@@ -158,7 +158,12 @@
     while (entry = readdir(dir), entry != NULL)
     {
 	snprintf(link, sizeof link, "%s/%s", path, entry->d_name);
+	#ifndef _WIN32
 	len = readlink(link, dest, sizeof dest - 1);
+	#else
+	strcpy(dest, link);
+	len = strlen(dest);
+	#endif
 	if (len < 0)
 	    continue;
 
diff -ru -x '*.Po' -x '*.Plo' libXfont-1.3.4.orig/src/fontfile/gunzip.c libXfont-1.3.4/src/fontfile/gunzip.c
--- libXfont-1.3.4.orig/src/fontfile/gunzip.c	2009-01-19 10:34:16 +0100
+++ libXfont-1.3.4/src/fontfile/gunzip.c	2009-01-19 10:47:51 +0100
@@ -7,9 +7,9 @@
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
-#include <X11/fonts/fontmisc.h>
 #include <X11/fonts/bufio.h>
 #include <zlib.h>
+#include <X11/fonts/fontmisc.h>
 
 typedef struct _xzip_buf {
   z_stream z;
diff -ru -x '*.Po' -x '*.Plo' libXfont-1.3.4.orig/src/fontfile/register.c libXfont-1.3.4/src/fontfile/register.c
--- libXfont-1.3.4.orig/src/fontfile/register.c	2009-01-19 10:34:16 +0100
+++ libXfont-1.3.4/src/fontfile/register.c	2009-01-19 10:48:02 +0100
@@ -39,10 +39,10 @@
 #define XFONT_BITMAP 1
 #endif
 
-#include <X11/fonts/fontmisc.h>
 #include <X11/fonts/fntfilst.h>
 #include <X11/fonts/bitmap.h>
 #include <X11/fonts/fontmod.h>
+#include <X11/fonts/fontmisc.h>
 
 /*
  * Translate monolithic build symbols to modular build symbols.
diff -ru -x '*.Po' -x '*.Plo' libXfont-1.3.4.orig/src/util/atom.c libXfont-1.3.4/src/util/atom.c
--- libXfont-1.3.4.orig/src/util/atom.c	2009-01-19 10:34:16 +0100
+++ libXfont-1.3.4/src/util/atom.c	2009-01-19 10:48:29 +0100
@@ -36,8 +36,8 @@
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
-#include <X11/fonts/fontmisc.h>
 #include "stubs.h"
+#include <X11/fonts/fontmisc.h>
 
 typedef struct _AtomList {
     char		*name;
diff -ru -x '*.Po' -x '*.Plo' libXfont-1.3.4.orig/src/util/fontxlfd.c libXfont-1.3.4/src/util/fontxlfd.c
--- libXfont-1.3.4.orig/src/util/fontxlfd.c	2009-01-19 10:34:16 +0100
+++ libXfont-1.3.4/src/util/fontxlfd.c	2009-01-19 10:48:39 +0100
@@ -36,7 +36,6 @@
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
-#include	<X11/fonts/fontmisc.h>
 #include	<X11/fonts/fontstruct.h>
 #include	<X11/fonts/fontxlfd.h>
 #include	<X11/fonts/fontutil.h>
@@ -51,6 +50,7 @@
 #endif
 #include	<ctype.h>
 #include	<stdio.h>	/* for sprintf() */
+#include	<X11/fonts/fontmisc.h>
 
 static char *
 GetInt(char *ptr, int *val)
diff -ru -x '*.Po' -x '*.Plo' libXfont-1.3.4.orig/src/util/miscutil.c libXfont-1.3.4/src/util/miscutil.c
--- libXfont-1.3.4.orig/src/util/miscutil.c	2009-01-19 10:34:16 +0100
+++ libXfont-1.3.4/src/util/miscutil.c	2009-01-19 10:42:49 +0100
@@ -34,12 +34,12 @@
 #endif
 #include <X11/Xosdefs.h>
 #include <stdlib.h>
-#include <X11/fonts/fontmisc.h>
 #include "stubs.h"
 
 #define XK_LATIN1
 #include    <X11/keysymdef.h>
 
+#include <X11/fonts/fontmisc.h>
 
 #ifdef __SUNPRO_C
 #pragma weak serverGeneration
