diff --git a/ChangeLog b/ChangeLog
index 622bce0..1e94451 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,13 @@
 2008-03-19  Vincent Untz  <vuntz@gnome.org>
 
+	List windows that demands attention in the tasklist when they're in the
+	current workspace but they're not visible in the viewport.
+	Fix bug #520124
+
+	* libwnck/tasklist.c: (wnck_task_state_changed): just do it
+
+2008-03-19  Vincent Untz  <vuntz@gnome.org>
+
 	Fix crash on logout because of some race condition.
 	Fix bug #517750
 
diff --git a/libwnck/tasklist.c b/libwnck/tasklist.c
index d687ee3..1d7e4df 100644
--- a/libwnck/tasklist.c
+++ b/libwnck/tasklist.c
@@ -3397,7 +3397,9 @@ wnck_task_state_changed (WnckWindow     *window,
         wnck_screen_get_active_workspace (tasklist->priv->screen);
 
       if (active_workspace                              &&
-          active_workspace != wnck_window_get_workspace (window))
+          (active_workspace != wnck_window_get_workspace (window) ||
+	   (wnck_workspace_is_virtual (active_workspace) &&
+	    !wnck_window_is_in_viewport (window, active_workspace))))
         {
           wnck_tasklist_update_lists (tasklist);
           gtk_widget_queue_resize (GTK_WIDGET (tasklist));
