diff --git a/ChangeLog b/ChangeLog
index 1e94451..b041087 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,15 @@
 2008-03-19  Vincent Untz  <vuntz@gnome.org>
 
+	Unstick windows when moving them in another workspace (viewport mode),
+	else the windows are just outside the visible screen.
+	Fix bug #520704.
+	Patch by Travis Watkins <amaranth@ubuntu.com>
+
+	* libwnck/window-action-menu.c: (item_activated_callback): well, do
+	what I already described
+
+2008-03-19  Vincent Untz  <vuntz@gnome.org>
+
 	List windows that demands attention in the tasklist when they're in the
 	current workspace but they're not visible in the viewport.
 	Fix bug #520124
diff --git a/libwnck/window-action-menu.c b/libwnck/window-action-menu.c
index 5921b9d..d0b34e0 100644
--- a/libwnck/window-action-menu.c
+++ b/libwnck/window-action-menu.c
@@ -208,13 +208,13 @@ item_activated_callback (GtkWidget *menu_item,
       if (!viewport_mode)
         wnck_window_pin (window);
       else
-        wnck_window_stick(window);
+        wnck_window_stick (window);
       break;
     case UNPIN:
       if (!viewport_mode)
         wnck_window_unpin (window);
       else
-        wnck_window_unstick(window);
+        wnck_window_unstick (window);
       break;
     case LEFT:
       if (!viewport_mode)
@@ -230,6 +230,7 @@ item_activated_callback (GtkWidget *menu_item,
 
           width = wnck_screen_get_width (screen);
           wnck_window_get_geometry (window, &xw, &yw, &ww, &hw);
+          wnck_window_unstick (window);
           wnck_window_set_geometry (window, 0,
                                     WNCK_WINDOW_CHANGE_X | WNCK_WINDOW_CHANGE_Y,
                                     xw - width, yw,
@@ -250,6 +251,7 @@ item_activated_callback (GtkWidget *menu_item,
 
           width = wnck_screen_get_width (screen);
           wnck_window_get_geometry (window, &xw, &yw, &ww, &hw);
+          wnck_window_unstick (window);
           wnck_window_set_geometry (window, 0,
                                     WNCK_WINDOW_CHANGE_X | WNCK_WINDOW_CHANGE_Y,
                                     xw + width, yw,
@@ -270,6 +272,7 @@ item_activated_callback (GtkWidget *menu_item,
 
           height = wnck_screen_get_height (screen);
           wnck_window_get_geometry (window, &xw, &yw, &ww, &hw);
+          wnck_window_unstick (window);
           wnck_window_set_geometry (window, 0,
                                     WNCK_WINDOW_CHANGE_X | WNCK_WINDOW_CHANGE_Y,
                                     xw, yw - height,
@@ -290,6 +293,7 @@ item_activated_callback (GtkWidget *menu_item,
 
           height = wnck_screen_get_height (screen);
           wnck_window_get_geometry (window, &xw, &yw, &ww, &hw);
+          wnck_window_unstick (window);
           wnck_window_set_geometry (window, 0,
                                     WNCK_WINDOW_CHANGE_X | WNCK_WINDOW_CHANGE_Y,
                                     xw, yw + height,
@@ -336,6 +340,7 @@ item_activated_callback (GtkWidget *menu_item,
           xw += viewport_x;
           yw += viewport_y;
 
+          wnck_window_unstick (window);
           wnck_window_set_geometry (window, 0,
                                     WNCK_WINDOW_CHANGE_X | WNCK_WINDOW_CHANGE_Y,
                                     xw + new_viewport_x,
