diff --git a/ChangeLog b/ChangeLog
index b041087..e56cdfe 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,13 @@
 2008-03-19  Vincent Untz  <vuntz@gnome.org>
 
+	Fix moving a window to an arbitrary workspace in viewport mode.
+	Fix bug #520726.
+
+	* libwnck/window-action-menu.c: (item_activated_callback): get the
+	maths right.
+
+2008-03-19  Vincent Untz  <vuntz@gnome.org>
+
 	Unstick windows when moving them in another workspace (viewport mode),
 	else the windows are just outside the visible screen.
 	Fix bug #520704.
diff --git a/libwnck/window-action-menu.c b/libwnck/window-action-menu.c
index d0b34e0..0e1108a 100644
--- a/libwnck/window-action-menu.c
+++ b/libwnck/window-action-menu.c
@@ -337,14 +337,11 @@ item_activated_callback (GtkWidget *menu_item,
           viewport_x = wnck_workspace_get_viewport_x (workspace);
           viewport_y = wnck_workspace_get_viewport_y (workspace);
 
-          xw += viewport_x;
-          yw += viewport_y;
-
           wnck_window_unstick (window);
           wnck_window_set_geometry (window, 0,
                                     WNCK_WINDOW_CHANGE_X | WNCK_WINDOW_CHANGE_Y,
-                                    xw + new_viewport_x,
-                                    yw + new_viewport_y,
+                                    xw + new_viewport_x - viewport_x,
+                                    yw + new_viewport_y - viewport_y,
                                     ww, hw);
         }
       break;
