diff -ru cmus-2.2.0.orig/debug.h cmus-2.2.0/debug.h
--- cmus-2.2.0.orig/debug.h	2008-03-18 08:39:58 +0100
+++ cmus-2.2.0/debug.h	2008-03-18 08:40:20 +0100
@@ -8,6 +8,10 @@
 #include "compiler.h"
 #include "config/debug.h"
 
+#if defined(__INTERIX)
+# include <stdint.h>
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
diff -ru cmus-2.2.0.orig/utils.h cmus-2.2.0/utils.h
--- cmus-2.2.0.orig/utils.h	2008-03-18 08:39:58 +0100
+++ cmus-2.2.0/utils.h	2008-03-18 08:43:00 +0100
@@ -85,6 +85,10 @@
 	return s.st_mtime;
 }
 
+#if defined(__INTERIX)
+# define nanosleep(x, y) usleep( ((x)->tv_sec * 1000000) + ((x)->tv_nsec/1000))
+#endif
+
 static inline void ns_sleep(int ns)
 {
 	struct timespec req;
