diff -ru -x '*.Po' x11vnc-0.9.3.orig/libvncclient/listen.c x11vnc-0.9.3/libvncclient/listen.c
--- x11vnc-0.9.3.orig/libvncclient/listen.c	2008-03-11 09:55:02 +0100
+++ x11vnc-0.9.3/libvncclient/listen.c	2008-03-11 10:05:00 +0100
@@ -67,7 +67,11 @@
 
     /* reap any zombies */
     int status, pid;
+#if defined (__INTERIX)
+    while ((pid= waitpid(-1, &status, WNOHANG))>0);
+#else
     while ((pid= wait3(&status, WNOHANG, (struct rusage *)0))>0);
+#endif
 
     /* TODO: callback for discard any events (like X11 events) */
 
diff -ru -x '*.Po' x11vnc-0.9.3.orig/libvncserver/sockets.c x11vnc-0.9.3/libvncserver/sockets.c
--- x11vnc-0.9.3.orig/libvncserver/sockets.c	2008-03-11 09:55:02 +0100
+++ x11vnc-0.9.3/libvncserver/sockets.c	2008-03-11 10:00:05 +0100
@@ -62,6 +62,11 @@
 #include <unistd.h>
 #endif
 
+#if defined(__INTERIX)
+// for INADDR_LOOPBACK
+# include <rpc/types.h>
+#endif
+
 #if defined(__linux__) && defined(NEED_TIMEVAL)
 struct timeval 
 {
diff -ru -x '*.Po' x11vnc-0.9.3.orig/x11vnc/remote.c x11vnc-0.9.3/x11vnc/remote.c
--- x11vnc-0.9.3.orig/x11vnc/remote.c	2008-03-11 09:55:02 +0100
+++ x11vnc-0.9.3/x11vnc/remote.c	2008-03-11 10:01:15 +0100
@@ -28,6 +28,11 @@
 #include "userinput.h"
 #include "avahi.h"
 
+#if defined(__INTERIX)
+// for INADDR_LOOPBACK
+# include <rpc/types.h>
+#endif
+
 int send_remote_cmd(char *cmd, int query, int wait);
 int do_remote_query(char *remote_cmd, char *query_cmd, int remote_sync,
     int qdefault);
diff -ru -x '*.Po' x11vnc-0.9.3.orig/x11vnc/sslhelper.c x11vnc-0.9.3/x11vnc/sslhelper.c
--- x11vnc-0.9.3.orig/x11vnc/sslhelper.c	2008-03-11 09:55:02 +0100
+++ x11vnc-0.9.3/x11vnc/sslhelper.c	2008-03-11 10:01:47 +0100
@@ -8,6 +8,11 @@
 #include "connections.h"
 #include "sslcmds.h"
 
+#if defined(__INTERIX)
+// for INADDR_LOOPBACK
+# include <rpc/types.h>
+#endif
+
 #define OPENSSL_INETD 1
 #define OPENSSL_VNC   2
 #define OPENSSL_HTTPS 3
diff -ru -x '*.Po' x11vnc-0.9.3.orig/x11vnc/x11vnc.c x11vnc-0.9.3/x11vnc/x11vnc.c
--- x11vnc-0.9.3.orig/x11vnc/x11vnc.c	2008-03-11 09:55:02 +0100
+++ x11vnc-0.9.3/x11vnc/x11vnc.c	2008-03-11 10:02:36 +0100
@@ -145,6 +145,11 @@
 #include "pm.h"
 #include "solid.h"
 
+#if defined(__INTERIX)
+// for INADDR_LOOPBACK
+# include <rpc/types.h>
+#endif
+
 /*
  * main routine for the x11vnc program
  */
